<!DOCTYPE html>
<html lang="zh-Hans">
<head>
  <meta charset="UTF-8">
<meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=2">
<meta name="theme-color" content="#222">
<meta name="generator" content="Hexo 5.4.0">
  <link rel="apple-touch-icon" sizes="180x180" href="/blog/images/apple-touch-icon-next.png">
  <link rel="icon" type="image/png" sizes="32x32" href="/blog/images/favicon-32x32-next.png">
  <link rel="icon" type="image/png" sizes="16x16" href="/blog/images/favicon-16x16-next.png">
  <link rel="mask-icon" href="/blog/images/logo.svg" color="#222">
  <meta name="google-site-verification" content="1osTo_gOe1SW9Hevqtz8vl8HfSgTO4DSMZ0DiMPMX6w">

<link rel="stylesheet" href="/blog/css/main.css">


<link rel="stylesheet" href="/blog/lib/font-awesome/css/all.min.css">

<script id="hexo-configurations">
    var NexT = window.NexT || {};
    var CONFIG = {"hostname":"wangc1106.github.io","root":"/blog/","scheme":"Gemini","version":"7.8.0","exturl":false,"sidebar":{"position":"left","display":"post","padding":18,"offset":12,"onmobile":false},"copycode":{"enable":false,"show_result":false,"style":null},"back2top":{"enable":true,"sidebar":true,"scrollpercent":"ture"},"bookmark":{"enable":false,"color":"#222","save":"auto"},"fancybox":false,"mediumzoom":false,"lazyload":false,"pangu":false,"comments":{"style":"tabs","active":null,"storage":true,"lazyload":false,"nav":null},"algolia":{"hits":{"per_page":10},"labels":{"input_placeholder":"Search for Posts","hits_empty":"We didn't find any results for the search: ${query}","hits_stats":"${hits} results found in ${time} ms"}},"localsearch":{"enable":true,"trigger":"auto","top_n_per_article":1,"unescape":false,"preload":false},"motion":{"enable":true,"async":false,"transition":{"post_block":"fadeIn","post_header":"slideDownIn","post_body":"slideDownIn","coll_header":"slideLeftIn","sidebar":"slideUpIn"}},"path":"search.xml"};
  </script>

  <meta name="description" content="è¾…åŠ©ç»„è£…æ¡ˆä¾‹">
<meta property="og:type" content="article">
<meta property="og:title" content="HiFi-6-[HiFi-HiC-assembly-method]">
<meta property="og:url" content="https://wangc1106.github.io/blog/2022/03/14/2022-03-14-HiFi-6-HiC-assembly/index.html">
<meta property="og:site_name" content="Banner&#39;s Blog">
<meta property="og:description" content="è¾…åŠ©ç»„è£…æ¡ˆä¾‹">
<meta property="og:locale">
<meta property="og:image" content="https://wangc1106.github.io/blog/2022/03/14/2022-03-14-HiFi-6-HiC-assembly/1.png">
<meta property="article:published_time" content="2022-03-14T08:09:34.778Z">
<meta property="article:modified_time" content="2022-03-14T13:51:43.936Z">
<meta property="article:author" content="Wang C">
<meta property="article:tag" content="denovo-III">
<meta name="twitter:card" content="summary">
<meta name="twitter:image" content="https://wangc1106.github.io/blog/2022/03/14/2022-03-14-HiFi-6-HiC-assembly/1.png">

<link rel="canonical" href="https://wangc1106.github.io/blog/2022/03/14/2022-03-14-HiFi-6-HiC-assembly/">


<script id="page-configurations">
  // https://hexo.io/docs/variables.html
  CONFIG.page = {
    sidebar: "",
    isHome : false,
    isPost : true,
    lang   : 'zh-Hans'
  };
</script>

  <title>HiFi-6-[HiFi-HiC-assembly-method] | Banner's Blog</title>
  






  <noscript>
  <style>
  .use-motion .brand,
  .use-motion .menu-item,
  .sidebar-inner,
  .use-motion .post-block,
  .use-motion .pagination,
  .use-motion .comments,
  .use-motion .post-header,
  .use-motion .post-body,
  .use-motion .collection-header { opacity: initial; }

  .use-motion .site-title,
  .use-motion .site-subtitle {
    opacity: initial;
    top: initial;
  }

  .use-motion .logo-line-before i { left: initial; }
  .use-motion .logo-line-after i { right: initial; }
  </style>
</noscript>

</head>

<body itemscope itemtype="http://schema.org/WebPage">
  <div class="container use-motion">
    <div class="headband"></div>

    <header class="header" itemscope itemtype="http://schema.org/WPHeader">
      <div class="header-inner"><div class="site-brand-container">
  <div class="site-nav-toggle">
    <div class="toggle" aria-label="Toggle navigation bar">
      <span class="toggle-line toggle-line-first"></span>
      <span class="toggle-line toggle-line-middle"></span>
      <span class="toggle-line toggle-line-last"></span>
    </div>
  </div>

  <div class="site-meta">

    <a href="/blog/" class="brand" rel="start">
      <span class="logo-line-before"><i></i></span>
      <h1 class="site-title">Banner's Blog</h1>
      <span class="logo-line-after"><i></i></span>
    </a>
      <p class="site-subtitle" itemprop="description">Dream big work hard</p>
  </div>

  <div class="site-nav-right">
    <div class="toggle popup-trigger">
        <i class="fa fa-search fa-fw fa-lg"></i>
    </div>
  </div>
</div>




<nav class="site-nav">
  <ul id="menu" class="main-menu menu">
        <li class="menu-item menu-item-home">

    <a href="/blog/" rel="section"><i class="home fa-fw"></i>Home</a>

  </li>
        <li class="menu-item menu-item-categories">

    <a href="/blog/categories/" rel="section"><i class="th fa-fw"></i>Categories</a>

  </li>
        <li class="menu-item menu-item-tags">

    <a href="/blog/tags/" rel="section"><i class="tags fa-fw"></i>Tags</a>

  </li>
        <li class="menu-item menu-item-about">

    <a href="https://wangc1106.github.io/" rel="section"><i class="user fa-fw"></i>About</a>

  </li>
      <li class="menu-item menu-item-search">
        <a role="button" class="popup-trigger"><i class="fa fa-search fa-fw"></i>Search
        </a>
      </li>
  </ul>
</nav>



  <div class="search-pop-overlay">
    <div class="popup search-popup">
        <div class="search-header">
  <span class="search-icon">
    <i class="fa fa-search"></i>
  </span>
  <div class="search-input-container">
    <input autocomplete="off" autocapitalize="off"
           placeholder="Searching..." spellcheck="false"
           type="search" class="search-input">
  </div>
  <span class="popup-btn-close">
    <i class="fa fa-times-circle"></i>
  </span>
</div>
<div id="search-result">
  <div id="no-result">
    <i class="fa fa-spinner fa-pulse fa-5x fa-fw"></i>
  </div>
</div>

    </div>
  </div>

</div>
    </header>

    


    <main class="main">
      <div class="main-inner">
        <div class="content-wrap">
          

          <div class="content post posts-expand">
            

    
  
  
  <article itemscope itemtype="http://schema.org/Article" class="post-block" lang="zh-Hans">
    <link itemprop="mainEntityOfPage" href="https://wangc1106.github.io/blog/2022/03/14/2022-03-14-HiFi-6-HiC-assembly/">

    <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
      <meta itemprop="image" content="/blog/images/banner.png">
      <meta itemprop="name" content="Wang C">
      <meta itemprop="description" content="Make more time">
    </span>

    <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
      <meta itemprop="name" content="Banner's Blog">
    </span>
      <header class="post-header">
        <h1 class="post-title" itemprop="name headline">
          HiFi-6-[HiFi-HiC-assembly-method]
        </h1>

        <div class="post-meta">
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-calendar"></i>
              </span>
              <span class="post-meta-item-text">Posted on</span>
              

              <time title="Created: 2022-03-14 16:09:34 / Modified: 21:51:43" itemprop="dateCreated datePublished" datetime="2022-03-14T16:09:34+08:00">2022-03-14</time>
            </span>
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-folder"></i>
              </span>
              <span class="post-meta-item-text">In</span>
                <span itemprop="about" itemscope itemtype="http://schema.org/Thing">
                  <a href="/blog/categories/Software/" itemprop="url" rel="index"><span itemprop="name">Software</span></a>
                </span>
            </span>

          
            <span id="/blog/2022/03/14/2022-03-14-HiFi-6-HiC-assembly/" class="post-meta-item leancloud_visitors" data-flag-title="HiFi-6-[HiFi-HiC-assembly-method]" title="Views">
              <span class="post-meta-item-icon">
                <i class="fa fa-eye"></i>
              </span>
              <span class="post-meta-item-text">Views: </span>
              <span class="leancloud-visitors-count"></span>
            </span>
            <span class="post-meta-item" title="Symbols count in article">
              <span class="post-meta-item-icon">
                <i class="far fa-file-word"></i>
              </span>
              <span>17k</span>
            </span>
            <span class="post-meta-item" title="Reading time">
              <span class="post-meta-item-icon">
                <i class="far fa-clock"></i>
              </span>
              <span>16 mins.</span>
            </span>

        </div>
      </header>

    
    
    
    <div class="post-body" itemprop="articleBody">

      
        <p>è¾…åŠ©ç»„è£…æ¡ˆä¾‹</p>
<span id="more"></span>

<h1 id="1-2016-05-np-ç¢§å†¬èŒ„"><a href="#1-2016-05-np-ç¢§å†¬èŒ„" class="headerlink" title="1. 2016.05-np ç¢§å†¬èŒ„"></a>1. 2016.05-np ç¢§å†¬èŒ„</h1><p><strong>Insight into the evolution of the Solanaceae from the parental genomes of Petunia hybrida</strong></p>
<h2 id="Genome-sequencing-assembly-and-annotation"><a href="#Genome-sequencing-assembly-and-annotation" class="headerlink" title="Genome sequencing, assembly and annotation."></a>Genome sequencing, assembly and annotation.</h2><p>Plants were grown and DNA was extracted following the methods described at Supplementary Note 1.</p>
<p>Illumina libraries with 0.17-, 0.35-, 0.5-, 0.8-, 1-, 2-, 5-, 8- and 15-kb inserts were sequenced at BGI-Shenzghen and University of Illinois, Roy J. Carver Biotechnology. PacBio P. axillaris DNA library was sequenced with P4/C2 chemistry.</p>
<p>Illumina reads were processed using <strong>Fastq-mcf</strong> (<strong>quality filtering</strong>; <a target="_blank" rel="noopener" href="https://code.google.com/p/eautils/wiki/FastqMcf">https://code.google.com/p/eautils/wiki/FastqMcf</a>), <strong>PRINSEQ</strong> (<strong>duplication filtering</strong>; http:// prinseq.sourceforge.net/) and <strong>Musket</strong> (<strong>error correction</strong>; <a target="_blank" rel="noopener" href="http://musket.sourceforge/">http://musket.sourceforge</a>. net/). <strong>Pacbio reads</strong> were processed using the <strong>SMRT Analysis pipeline</strong> (v.2.0.1; <a target="_blank" rel="noopener" href="https://github.com/PacificBiosciences/SMRT-Analysis">https://github.com/PacificBiosciences/SMRT-Analysis</a>).</p>
<p>Both genomes were assembled with <strong>SOAPdenovo</strong> with different k-mer sizes. For both genomes, <strong>k-mer = 79</strong> showed the best statistics. Gaps between contigs were completed using <strong>GapCloser</strong> . Additionally for P. axillaris, <strong>PacBio reads</strong> were integrated in <strong>four different steps</strong>: (1) <strong>Rescaffolding</strong> of the <strong>Illumina contigs</strong> using the <strong>PacBio reads</strong> and the <strong>AHA assembler</strong>; (2) <strong>Gap filling</strong> using <strong>PBJelly</strong>; (3) <strong>Rescaffolding</strong> using the Illumina <strong>pair data</strong> and <strong>SSPACE</strong>; (4) Last round of <strong>gap filling</strong> using <strong>PBJelly</strong> . </p>
<p><strong>Genome size estimation</strong> was performed through the k-mers abundance distribution (<strong>k-mer = 31</strong>). <strong>Heterozygosity</strong> was <strong>estimated</strong> mapping the Illumina reads to the assemblies using <strong>Bowtie2</strong>, calling SNPs using <strong>FreeBayes</strong> and  annotating the SNPs using <strong>SnpEff</strong> .</p>
<p>The genome structural annotation was performed using <strong>Maker-P</strong>: (1) <strong>SNAP</strong> and <strong>Augustus</strong> as ab initio gene predictors; (2) Exonerate as experimental based predictor with 454 and Illumina RNASeq reads and protein sequences from different protein datasets. RNAseq Illumina data was mapped using <strong>Tophat2</strong> . tRNAs were annotated using <strong>tRNAscan</strong> (<a target="_blank" rel="noopener" href="http://lowelab.ucsc.edu/tRNAscan-SE/">http://lowelab.ucsc.edu/tRNAscan-SE/</a>).</p>
<p>The <strong>gene functional annotation</strong> was performed by <strong>sequence homology search</strong> with different protein datasets using <strong>BlastP</strong> and protein domains search using <strong>InterProScan</strong> . Functional annotations were integrated using <strong>AHRD</strong> (<a target="_blank" rel="noopener" href="https://github.com/groupschoof/AHRD">https://github.com/groupschoof/AHRD</a>). See Supplementary Note 1.</p>
<h2 id="Repetitive-elements-analysis"><a href="#Repetitive-elements-analysis" class="headerlink" title="Repetitive elements analysis"></a>Repetitive elements analysis</h2><p>Repeat annotation was performed using RepeatModeler (v1.0.8; <a target="_blank" rel="noopener" href="http://www.repeatmasker.org/RepeatModeler.html">http://www.repeatmasker.org/RepeatModeler.html</a>), RepeatMasker (v4.0.5; <a target="_blank" rel="noopener" href="http://www.repeatmasker.org/">http://www.repeatmasker.org</a>) with the repeat database Repbase (release 20140131; <a target="_blank" rel="noopener" href="http://www.girinst.org/repbase/">http://www.girinst.org/repbase/</a>) and Geneious (v7.1.4; <a target="_blank" rel="noopener" href="http://www.geneious.com/">http://www.geneious.com</a>). Identification of PVCV-like and EPRV elements was performed using BlastN and TBlastN . The identified sequences were aligned with ClustalW (MEGA5 package;<a target="_blank" rel="noopener" href="http://www.megasoftware.net/">http://www.megasoftware.net/</a>) and then manually curated. RepeatExplorer (<a target="_blank" rel="noopener" href="http://www.repeatexplorer.org/">http://www.repeatexplorer.org/</a>) and other methods were used to extend the analysis to unassembled repeats. Fluorescent in situ hybridization was performed in root tips from young P. axillaris and P. inflata plants for 5S rDNA and three PVCV viral probes following the procedure described in Supplementary Note 2.</p>
<p>The detection of dTph1 loci in P. hybrida W138 was performed through a BLAST40 search of the P. axillaris and P. inflata dTph1 elements including the 500 bp of flanking sequence against the TFS W138 collection. Polymorphisms found in the genomic flanking regions were used to identify the species of origin. dTph1 elements were identified in a P. axillaris population using a modification of the methodology described in Supplementary Note 3.</p>
<p>##Whole-genome duplication, tandem duplications and gene family analysis</p>
<p>Whole-genome collinear analysis was performed using SynMap and microsynteny analysis were performed using GEvo in the comparative genomics platform, CoGe .See Supplementary Note 5.</p>
<p>The gene family analysis included Solanum lycopersicum, S. tuberosum, Nicotiana benthamiana and Arabidopsis thaliana protein sets using BlastP (v2.2.27) on an all-versus-all comparison and grouping the genes into families with OrthoMCL, v2.0.8. See Supplementary Note 4.</p>
<h2 id="Gene-data-mining"><a href="#Gene-data-mining" class="headerlink" title="Gene data mining"></a>Gene data mining</h2><p>The specific identification of genes for P. axillaris and P. inflata genomes for colour and scent, root-specific pathways, self-incompatibility and circadian clock was performed through a BlastN/BlastP sequence homology search. Blast GUI, JBrowser (<a target="_blank" rel="noopener" href="http://jbrowse.org/">http://jbrowse.org/</a>) and WebApollo (<a target="_blank" rel="noopener" href="http://genomearchitect/">http://genomearchitect</a>. org/) were installed in a server to search and manually curate the gene structures of the identified genes. See Supplementary Notes 7, 8 and 10â€“12.</p>
<h1 id="2-2017-04-ng-å±±ç¾ŠğŸ"><a href="#2-2017-04-ng-å±±ç¾ŠğŸ" class="headerlink" title="2. 2017.04-ng å±±ç¾ŠğŸ"></a>2. 2017.04-ng å±±ç¾ŠğŸ</h1><p>Published in final edited form as:Nat Genet. 2017 April ; 49(4): 643â€“650. doi:10.1038/ng.3802.</p>
<p><strong>Single-molecule sequencing and chromatin conformation capture enable</strong> <strong>de novo</strong> <strong>reference assembly of the domestic goat genome</strong></p>
<p>â€‹        é¦–æ¬¡åœ¨äººä¹‹å¤–çš„ç‰©ç§ä¸­ç›´æ¥de novoç»„è£…å‡ºæŸ“è‰²ä½“æ°´å¹³çš„åŸºå› ç»„ã€‚åˆ©ç”¨<font color="blue"><strong>Pacbio+BioNano+Hi-C+Illumina</strong> å¤šå±‚çº§</font>çš„ç­–ç•¥ï¼Œç»„è£…äº†ä¸€åªå±±ç¾Š(Capra hircus)çš„åŸºå› ç»„ï¼Œè·å¾—<font color="blue"><strong>31</strong>ä¸ª<strong>scaffold</strong>ï¼Œ<strong>663</strong>ä¸ª<strong>gap</strong>åŒºï¼Œscaffold <strong>N50</strong> æƒŠäººåœ°è¾¾åˆ°<strong>87M</strong>ï¼Œå‡ ä¹<strong>one scaffold one chromosome</strong></font>ï¼Œå¹¶ä¸”æˆåŠŸåœ°ç»„è£…å‡º<font color="blue"><strong>å…ç–«åŸºå› åŒº</strong>å’Œ<strong>å¤§éƒ¨åˆ†é‡å¤åºåˆ—å®¶æ—</strong></font>ã€‚</p>
<p>â€‹        <font color="red">äººçš„GRCh38ç‰ˆæœ¬æœ‰<strong>24</strong>ä¸ª<strong>scaffold</strong>ï¼Œ<strong>169</strong>ä¸ª<strong>unplacedscaffold</strong>ï¼Œ<strong>832ä¸ªgap</strong>åŒº</font>ã€‚</p>
<p>â€‹        ä»<strong>96å¤´</strong>å±±ç¾Š(<strong>6ä¸ªå“ç§</strong>)ä¸­ï¼Œåˆ©ç”¨Illuminaçš„ Caprine53K <font color="blue"><strong>SNP èŠ¯ç‰‡ç­›é€‰</strong></font>å‡º<font color="blue"><strong>åŸºå› å‹çº¯å’Œåº¦æœ€é«˜</strong></font>çš„ä¸ªä½“ç”¨æ¥<font color="blue"><strong>ç»„è£…</strong></font>(San Clemente breed)ã€‚</p>
<h2 id="data"><a href="#data" class="headerlink" title="data"></a>data</h2><ol>
<li><p>Pacbioæ•°æ®</p>
<p>ä¸€å…±äº§ç”Ÿ465ä¸ªSMRT cellçš„æ•°æ®ã€‚å…¶ä¸­ä½¿ç”¨P5-C3è¯•å‰‚çš„311ä¸ªcell;ä½¿ç”¨P4-C2è¯•å‰‚çš„142ä¸ªcellï¼ŒXL-C2è¯•å‰‚çš„12ä¸ªcellã€‚æ•°æ®é‡194Gï¼Œè¦†ç›–æ·±åº¦è¾¾<strong>69X</strong>ï¼Œ<strong>subread</strong>çš„<strong>å¹³å‡</strong>é•¿åº¦ä¸º<strong>5,110bp</strong>ã€‚</p>
</li>
<li><p>BioNano æ•°æ®</p>
<p>ç”±äºé‡‡æ ·æ‰€ç”¨å±±ç¾Šæ„å¤–æ­»äº¡ï¼Œå…¶DNAä¸èƒ½æ»¡è¶³Irsy optical mapping æµ‹åºéœ€æ±‚ï¼Œåªå¥½ç”¨å…¶<strong>é›„æ€§åä»£</strong>é‡‡æ ·ã€‚Optical mapä¸€å…±äº§ç”Ÿ256Gbæ•°æ®ï¼Œè¦†ç›–<strong>98 X</strong>ã€‚</p>
</li>
<li><p>Hi-Cæ•°æ®</p>
<p>é‡‡æ ·åŠ¨ç‰©æ˜¯æœ€åˆé‚£åªå±±ç¾Šã€‚Hi-Cå»ºåº“ï¼Œåºåˆ—ç”¨ç‰©ç†æ–¹æ³•æ‰“æ–­æˆ300-500bpé•¿åº¦ï¼ŒPE101æµ‹åºï¼Œäº§ç”Ÿ<strong>115M reads</strong>çš„æ•°æ®é‡ã€‚</p>
</li>
<li><p>Illumina æ•°æ®</p>
<p>é‡‡æ ·æœ€åˆé‚£åªå±±ç¾Šï¼ŒPE251å»ºåº“æµ‹åºï¼Œè·å¾—<strong>23X</strong>çš„æ•°æ®è¦†ç›–ï¼Œç”¨æ¥åšæœ€åçš„<font color="red"><strong>é”™è¯¯æ ¡æ­£</strong></font></p>
</li>
</ol>
<h2 id="ç»„è£…ç­–ç•¥åŠç»„è£…ç»“æœ"><a href="#ç»„è£…ç­–ç•¥åŠç»„è£…ç»“æœ" class="headerlink" title="ç»„è£…ç­–ç•¥åŠç»„è£…ç»“æœ"></a><strong>ç»„è£…ç­–ç•¥åŠç»„è£…ç»“æœ</strong></h2><ul>
<li><p>å…ˆç”¨Pacbioçš„ long-read æ•°æ®æ„å»ºcontigï¼Œ ä½¿ç”¨<font color="blue"><strong>Celera Assembler</strong> PacBio corrected Reads æµç¨‹</font>ç»„è£…ã€‚è·å¾—<font color="blue"><strong>3,074</strong>ä¸ªcontig(2.63G)ï¼Œ<strong>N50 4.159M</strong>b</font>ã€‚</p>
</li>
<li><p>æ¥ä¸‹æ¥é‡‡ç”¨<font color="blue"><strong>Irys optical mapping</strong>æ•°æ®æ„å»º<strong>scaffold</strong></font>ï¼Œä½¿ç”¨è½¯ä»¶ä¸º<font color="blue"><strong>IrysView</strong></font>. äº§ç”Ÿ<font color="blue"><strong>842</strong>ä¸ªscaffoldï¼Œscaffold N50ä¸º<strong>13.4M</strong>bï¼Œcontig N50ä¸º10.858Mï¼Œæœ€é•¿scaffoldä¸º66.728Mb</font>ã€‚ åˆ©ç”¨<font color="red"><strong>radiation hybrid(RH) mapæ¯”è¾ƒ</strong>å‘ç°</font>ï¼Œ<font color="blue">PacBio-Irys è”åˆ</font>çš„æ–¹æ³•å·²ç»<font color="blue">å®Œæ•´ç»„è£…å‡º<strong>20å·æŸ“è‰²ä½“</strong></font>ã€‚</p>
</li>
<li><p>æ¥ä¸‹æ¥ç”¨Hi-Cæ•°æ®ï¼Œè°ƒç”¨ç”¨<font color="red"><strong>Lachesisè½¯ä»¶åŒ…</strong></font>ï¼Œç»“åˆPacBio-Irysçš„ç»“æœï¼Œå³æ•´åˆçš„PacBio-Irys-PGA (PBIP)æ–¹æ³•ï¼Œè·å¾—è¾ƒå®Œç¾çš„ç»„è£…ç»“æœã€‚<font color="blue">Scaffold N50</font> è¾¾åˆ°æƒŠäººçš„<font color="blue"><strong>87.347Mb</strong></font>(è¿œè¶…ä»Šå¹´å°–å»é²ˆï¼Œå¤§çŒ©çŒ©N50 20Mbçš„é«˜èµ·ç‚¹)ï¼Œæ€»å…±è·å¾—<font color="blue"><strong>31æ¡</strong>scaffold</font>ï¼Œ(<font color="red">31,261æ¡&lt;50kbpçš„**degenerate scaffold**ä¸è®¡å…¥</font> )ã€‚</p>
</li>
<li><p>æœ€ååˆ©ç”¨Illuminaæ•°æ®åš<font color="red"><strong>ä¸€è‡´æ€§æ ¡æ­£</strong>å’Œæœ€åçš„<strong>è¡¥æ´</strong></font>ï¼Œæœ€åä½¿å¾—contigçš„æ•°ç›®ä¸‹é™åˆ°680æ¡ï¼Œäº§ç”Ÿçš„gapæ•°ç›®ä¸º663ä¸ªã€‚</p>
</li>
<li><p>åˆ©ç”¨Kraken v0.10.5 <font color="red">å»é™¤æœ‰<strong>ç—…æ¯’å’Œç»†èŒæ±¡æŸ“</strong>çš„åºåˆ—</font>ï¼Œå»æ‰æœ‰NCBI vectoræ±¡æŸ“çš„åºåˆ—ã€‚è·å¾—æœ€ç»ˆçš„åŸºå› ç»„ ç‰ˆæœ¬<strong>ARS1</strong>ã€‚</p>
</li>
</ul>
<h2 id="ç»„è£…è¯„ä¼°åŠæ¯”è¾ƒåˆ†æ"><a href="#ç»„è£…è¯„ä¼°åŠæ¯”è¾ƒåˆ†æ" class="headerlink" title="ç»„è£…è¯„ä¼°åŠæ¯”è¾ƒåˆ†æ"></a><strong>ç»„è£…è¯„ä¼°åŠæ¯”è¾ƒåˆ†æ</strong></h2><p>ç”±äºå±±ç¾Šä¹‹å‰å·²ç»äº§ç”Ÿè¿‡ä¸¤ä¸ªåŸºå› ç»„ç‰ˆæœ¬<strong>CHIR_1.0ã€CHIR_2.0</strong>ï¼Œå¯ä»¥ç”¨æ¥åšæ¯”è¾ƒ åˆ†æï¼Œå¹¶ä¸”æœ‰RH mappingæ•°æ®ç”¨æ¥è¯„ä¼°ã€æ ¡æ­£ARS1çš„ç»„è£…ç‰ˆæœ¬ã€‚ <strong>é€šè¿‡å„ç§å‚æ•°æ¯”è¾ƒå‘ç°ï¼ŒARS1å‡ä¼˜äºCHIR_2.0ç‰ˆæœ¬ã€‚</strong></p>
<p>1ã€é€šè¿‡æ¯”å¯¹å‘ç°ï¼ŒCHIR_2.0 æ¯”CHIR_1.0æœ‰æ›´å°‘çš„putative deletion(2,735 vs 1 0,256)å’Œduplication(115 vs 290)ã€‚è€Œ2.0ç‰ˆæœ¬ç›¸è¾ƒ1.0æœ‰æ›´å¤šçš„inversion(215 v s 4)ã€‚é€šè¿‡æ¯”è¾ƒï¼Œå‘ç°ARS1ç‰ˆæœ¬æœ‰å¤§å¹…åº¦åœ°æå‡ï¼Œè¾ƒ2.0ç‰ˆæœ¬å°‘4å€çš„deletionï¼Œ å°‘50å€inversionã€‚</p>
<p>2ã€CHIR_2.0èƒ½å¡«è¡¥CHIR_1.0 94.6%çš„gapï¼Œè€Œå‰©ä¸‹çš„é‚£äº›gapé€šè¿‡åˆ†æARS1å¾—çŸ¥ï¼Œæ˜¯CHIR_1.0çš„ç»„è£…é”™è¯¯ï¼Œè€ŒRHæ•°æ®ä¹Ÿæ”¯æŒäº†è¿™ä¸€è®ºæ–­ã€‚ 3ã€BUSCOæ˜¯åˆ©ç”¨å•æ‹·è´orthologæ¥è¯„ä¼°ç»„è£…çš„å¯é æ€§ï¼Œå’ŒCHIR_2.0ç›¸æ¯”ï¼ŒARS1æœ‰æ›´é«˜çš„BUSCO scoreã€‚</p>
<p>4ã€å±±ç¾Š52k SNP èŠ¯ç‰‡ä¸­ï¼Œæœ‰1,723ä¸ªSNPæ¢é’ˆç›®å‰åªèƒ½å®šä½åœ¨CHIR_2.0ç»„è£…ç‰ˆæœ¬ ä¸­çš„unplaced contigsä¸Šã€‚è€ŒARS1ç‰ˆæœ¬èƒ½æŠŠå…¶ä¸­çš„90%(1,552/1,723)ä»¥ä¸Šçš„S NPæ¢é’ˆå®šä½åœ¨æŸ“è‰²ä½“ä¸Šã€‚å¦å¤–å‘ç°æœ‰26ä¸ªlow call-rate çš„SNPmarkeråœ¨ARS1ç»„è£…ç‰ˆæœ¬ä¸Šå®šä½æ¯”è¾ƒæ¨¡æ£±ä¸¤å¯ã€‚è¿™ä¹Ÿå°±è§£é‡Šäº†ä¸ºä»€ä¹ˆè¿™äº›makeråœ¨èŠ¯ç‰‡ä¸Šæœ‰è¾ƒå·®çš„call -rate. </p>
<p>5ã€é€šè¿‡æ¯”è¾ƒå‘ç°ï¼ŒCHIR_2.0ç‰ˆæœ¬ä¸­çš„3,495ä¸ªå†…å«å­æˆ–å¤–æ˜¾å­æœ‰gapçš„åŸºå› ï¼Œåœ¨ARS1ä¸­éƒ½å¾—åˆ°è¡¥å…¨ã€‚åŒæ—¶ä¹Ÿç¡®è®¤1,926ä¸ªé¢„æµ‹çš„å¤–æ˜¾å­åœ¨CHIR_1.0æˆ–è€…2.0ä¸­æœ‰gapï¼Œ ä½†åœ¨ARS1ä¸­å¾—åˆ°ä¿®å¤ã€‚ </p>
<p>6ã€ç”±äºå…ç–«åŸºå› åŒºé«˜åº¦å¤šæ€æ€§å’Œé‡å¤æ€§ï¼Œç”¨äºŒä»£æµ‹åºæ•°æ®å¾ˆéš¾ç»„è£…èµ·æ¥ï¼Œä½†æ˜¯é€šè¿‡ åˆ†æå‘ç°ï¼ŒARS1ç‰ˆæœ¬å°±å¾ˆå¥½åœ°æŠŠLRCå’ŒNKCåŸºå› (å…ç–«åŠŸèƒ½ç›¸å…³åŸºå› )å®šä½åœ¨ä¸€ä¸ªç‹¬ç«‹çš„å¸¸æŸ“è‰²ä½“scaffoldä¸Šã€‚ </p>
<p>7ã€é€šè¿‡å’Œä¸€ä»£ã€äºŒä»£ç»„è£…ç­–ç•¥ç›¸æ¯”ï¼ŒARS1ç‰ˆæœ¬åœ¨é‡å¤åºåˆ—çš„ç»„è£…ä¸Šè·å¾—äº†æå¤§è¿›æ­¥ã€‚å¯¹æŸäº›å¼‚æŸ“è‰²ä½“æˆ–å¼‚æŸ“è‰²è´¨çš„åŒºåŸŸä¹Ÿè·å¾—äº†è¾ƒå¥½çš„ç»„è£…è¦†ç›–ã€‚æ¯”å¦‚ï¼Œåœ¨6æ¡å¸¸ æŸ“è‰²ä½“ä¸Šç»„è£…å‡º&gt;5kbpçš„ç«¯ç²’åºåˆ—ã€‚é€šè¿‡åˆ†æå‘ç°ï¼Œ 15æ¡æŸ“è‰²ä½“scaffoldï¼Œåœ¨ç€ä¸ç²’åŒºåŸŸç»„è£…å‡ºå¤§äº2kbpé•¿åº¦çš„é‡å¤åŒºï¼Œè€Œå…¶ä¸­çš„7æ¡ï¼Œæ›´æ˜¯è£…å‡ºæ¥8kbpä»¥ä¸Šçš„é‡å¤åŒºã€‚æ›´ä»¤äººæƒŠè®¶çš„æ˜¯ï¼Œ19å·å’Œ23å·æŸ“è‰²ä½“éƒ½ç»„è£…å‡ºé«˜åº¦é‡å¤çš„ç€ä¸ç²’å’Œç«¯ç²’åŒºåŸŸï¼Œ è´¯é€šäº†æœ‰ç»“æ„æ€§å¼‚æŸ“è‰²è´¨åŒºçš„æŸ“è‰²ä½“ã€‚ </p>
<p>8ã€åœ¨ARS1ç‰ˆæœ¬ä¸­é‰´å®šå‡ºå¤§äº12kbpçš„é‡å¤æ¨¡å¼åºåˆ—å¤šè¾¾105æ¡ã€‚å¯¹äºé‡å¤åºåˆ—å®¶ æ—çš„é‰´å®šï¼Œåœ¨ARS1ä¸­è·å¾—çš„æ¥è¿‘å…¨é•¿çš„BovB LINE åºåˆ—æ¯”CHIR_2.0å¤š60%ä»¥ä¸Šã€‚ åœ¨CHIR_2.0ç‰ˆæœ¬é‡Œï¼Œè¢«ARS1æˆåŠŸè¡¥å…¨çš„gapä¸­çš„43.6%ä¸BovBé‡å¤åºåˆ—ä¸€è‡´(é•¿ åº¦å¤§äº3.5kbp)ï¼Œå³æ„å‘³ç€äºŒä»£æµ‹åºçš„gapåŒºå¤§å¤šéƒ½æ˜¯é‡å¤åºåˆ—ï¼Œç”¨ä¸‰ä»£å°±èƒ½è¾ƒå®¹æ˜“æµ‹é€šã€‚ </p>
<p>9ã€å…³äºæ€§æŸ“è‰²ä½“çš„åˆ†æã€‚é€šè¿‡æ¯”å¯¹å‘ç°ï¼Œä¸¤ä¸ªä¸åŒçš„scaffoldæ¯”å¯¹åˆ°XæŸ“è‰²ä½“çš„ä¸åŒä½†è¿ç»­çš„åŒºåŸŸï¼Œå é¢„æœŸXæŸ“è‰²ä½“å¤§å°(150Mb)çš„86%å·¦å³ã€‚é€šè¿‡å’Œè‡ªå·±çš„ã€è·¨ç‰©ç§ç‰›çš„YæŸ“è‰²ä½“æ¯”å¯¹ï¼Œæœ€ç»ˆç¡®å®šå‡º10MbåŒºåŸŸçš„åºåˆ—ï¼Œå YæŸ“è‰²ä½“é¢„ä¼°å¤§å°çš„50% å·¦å³ã€‚é€šè¿‡æ¯”å¯¹ç‰›å’Œç¾Šçš„YæŸ“è‰²ä½“ä¸Šçš„åŸºå› ï¼Œå‘ç°åœ¨ç›®å‰çš„scaffoldä¸­èƒ½æ‰¾åˆ°16%ï¼Œ è€Œåœ¨å…ˆå‰è¿‡æ»¤æ‰çš„degenerate contigä¸­èƒ½æ‰¾åˆ°84%çš„åŸºå› ã€‚è€ƒè™‘åˆ°YæŸ“è‰²ä½“çš„å¼‚æŸ“ è‰²è´¨å±æ€§åŠX,YæŸ“è‰²ä½“çš„æ‹Ÿå¸¸æŸ“è‰²åŒºåŸŸç»å¸¸å‘ç°äº¤æ¢ï¼Œå®åœ¨å¤ªè¿‡å¤æ‚ï¼Œè¶…å‡ºç›®å‰çš„ç»„è£…å’Œè®¤çŸ¥æ°´å¹³ï¼Œè¡¨ç¤ºå®åœ¨æ— èƒ½ä¸ºåŠ›ã€‚</p>
<p>10ã€åˆ©ç”¨ä¹‹å‰RH mapping dataï¼Œå¯¹ç»„è£…çš„å¤§éƒ¨åˆ†ç¯èŠ‚è¿›è¡Œäº†æ ¡æ­£åŠè¯„ä¼°ã€‚æ¯”å¦‚å¤šæ¬¡è¾…åŠ©è§£å†³scaffold conflictçš„é—®é¢˜ã€‚</p>
<h2 id="åŸºå› ç»„æ³¨é‡Š"><a href="#åŸºå› ç»„æ³¨é‡Š" class="headerlink" title="åŸºå› ç»„æ³¨é‡Š"></a><strong>åŸºå› ç»„æ³¨é‡Š</strong></h2><p>å¤šç§æ–¹æ³•ç»“åˆæ³¨é‡ŠåŸºå› é›†ã€‚<br>1.<strong>RNA-seqçš„æ–¹æ³•:</strong> 6ä¸ªç»„ç»‡(å¤§å¤šå’Œè„‘ç»„ç»‡ç›¸å…³)RNA-seqæµ‹åºã€13ä¸ªSRAä¸‹è½½ æ•°æ®ï¼Œåˆ©ç”¨stringtieã€cufflinkså’ŒTrinity(åŸºäºæ— å‚ç»„è£…)ã€‚æœ€åç”¨PASAè½¯ä»¶æ•´åˆåœ¨ä¸€èµ·;</p>
<p>2.<strong>ç”¨exonerateå’Œtblastn è½¯ä»¶</strong>æ¯”å¯¹åˆ°å‡ ä¸ªè¿‘ç¼˜ç‰©ç§çš„EnsemblåŸºå› é›†ä¸Šï¼Œè·å¾—åŒ æºé¢„æµ‹åŸºå› é›†;</p>
<p>3.<strong>ç”¨Braker1åšAb initio é¢„æµ‹</strong>;</p>
<p>4.<strong>CHIR_1.0 ç‰ˆæœ¬çš„æ³¨é‡ŠåŸºå› é›†</strong>; æœ€åç”¨<strong>EVM+PASA</strong>æŠŠä»¥ä¸Š4ç§æ•°æ®æ•´åˆæˆä¸€ä¸ªæœ€ç»ˆçš„åŸºå› é›†(è®¾ç½®çš„æƒé‡ä¸ºRNAseq &gt; cDNA/protein &gt; ab initio gene predictions)ã€‚</p>
<h2 id="å¯ç¤ºå½•"><a href="#å¯ç¤ºå½•" class="headerlink" title="å¯ç¤ºå½•"></a><strong>å¯ç¤ºå½•</strong></h2><p>1ã€Pacbio RSIIæ˜¯ä¸€ä¸ªè¾ƒä¸ºç†æƒ³ç»„è£…å¹³å°ï¼Œäº§ç”Ÿå¹³å‡14Kbpå·¦å³çš„è¯»é•¿ï¼Œæœ€é«˜åˆ°60 Kbpï¼Œå¯ä»¥ç»„è£…å‡ºè¾ƒä¸ºç†æƒ³çš„åŸºå› ç»„ç‰ˆæœ¬ã€‚ç”±äºä¸‰ä»£çš„é•¿è¯»é•¿ä¼˜åŠ¿ï¼ŒåŒæ—¶å¯ä»¥è¾ƒå¥½åœ°ç»„è£…å‡ºåŸºå› ç»„ä¸­å¤§éƒ¨åˆ†é«˜åº¦é‡å¤åŒº(ä¸»è¦é›†ä¸­åœ¨ç€ä¸ç²’å’Œç«¯ç²’åŒºåŸŸ);</p>
<p>2ã€BioNano çš„Irys optical mapping æ˜¯ä¸€ç§éå¸¸æœ‰æ•ˆçš„ï¼Œæˆæœ¬æ¯”è¾ƒé€‚ä¸­çš„æ„å»ºscaffoldçš„å¹³å°ã€‚å’ŒHi-Cæ¯”ä¼˜åŠ¿åœ¨äºå®šä½é”™è¯¯è¾ƒå°‘ï¼Œç¼ºç‚¹æ˜¯å¯¹N50çš„æé«˜å¹¶ä¸æ˜¯ç‰¹åˆ«æ˜¾è‘—ï¼Œä¸€èˆ¬åœ¨2å€å·¦å³ã€‚å¦‚æœæ­£åé“¾çš„Nt.BsqIé…¶åˆ‡ä½ç‚¹ç¦»å¾—æ¯”è¾ƒè¿‘ï¼Œå¸¸å¸¸å¯¼è‡´åŒé“¾ æ–­è£‚ï¼Œé™åˆ¶äº†Optical Map scaffoldçš„å¤§å°ã€‚æ‰€ä»¥ï¼ŒBioNanoå¯¹è¾ƒé•¿çš„scaffoldæ•ˆæœæ¯”è¾ƒå¥½ã€‚ç”±äºOptical mapå’Œpacbioäº§ç”Ÿçš„é”™è¯¯ç‰¹å¾å¹¶ä¸ä¸€æ ·ï¼Œä¸¤è€…ç»“åˆï¼Œå¯ä»¥ ç›¸äº’æ ¡æ­£ï¼Œè·å¾—è¾ƒå¥½çš„ç»„è£…æŒ‡æ ‡;</p>
<p>3ã€Hi-Cæ˜¯ç›®å‰éå¸¸ç«çˆ†çš„ä¸€ä¸ªæŠ€æœ¯ï¼Œåœ¨å¤šä¸ªé¢†åŸŸéƒ½æœ‰å¾ˆå¥½çš„åº”ç”¨ã€‚åœ¨ç»„è£…æ–¹é¢ï¼Œ å®ƒèƒ½è·å¾—æŸ“è‰²ä½“çº§åˆ«çš„scaffoldï¼Œä½†æ˜¯ä¹Ÿå®¹æ˜“è·å¾—è¾ƒé«˜çš„contigå®šå‘é”™è¯¯ã€‚è¿™ä¹Ÿæ˜¯ ä¸ºä»€ä¹ˆæ–‡ç« ä¸­ï¼Œå…ˆç”¨optical mappingï¼Œåç”¨Hi-Cåšscaffoldingçš„é‡è¦åŸå› ä¹‹ä¸€ã€‚ ä½œè€…å»ºè®®åœ¨ä»¥åçš„å®éªŒä¸­ï¼Œå¯ä»¥é€‰æ‹©è¾ƒçŸ­çš„è¯†åˆ«ä½ç‚¹çš„é™åˆ¶å†…åˆ‡é…¶(æˆ–è€…DNase Hi -C)æ¥æé«˜Hi-Cäº¤è”çš„å¯†åº¦ï¼Œé™ä½å®šå‘é”™è¯¯ã€‚</p>
<p>4ã€åˆ©ç”¨åŒæ ·çš„æ–¹æ³•å’Œå¹³å°ï¼Œåšç±»ä¼¼çš„åŸºå› ç»„é¡¹ç›®ï¼Œç”¨Pacbioå¹³å°å’Œscaffoldingå¹³ å°ï¼Œç›®å‰å¤§çº¦èŠ±è´¹åœ¨$100,000ä¸Šä¸‹,è€Œä¸€èˆ¬çš„çŸ­readså¹³å°æµ‹åºåŠ scaffolding å¹³å°ï¼Œåªéœ€è¦å…¶ä¸‰åˆ†ä¹‹ä¸€çš„ä»·æ ¼ã€‚ä½†æ˜¯ç”¨Pacbioå¹³å°å´è·å¾—äº†é«˜è´¨é‡çš„ã€è¿ç»­æ€§å¥½çš„åŸºå› ç»„åºåˆ—ã€‚ç›®å‰Pacbio æ–°ä¸€ä»£çš„sequelå¹³å°å·²ç»æ¨å‡ºï¼Œå•ä¸ªcellçš„äº§é‡æ¯”RSIIæ é«˜7å€ä»¥ä¸Šï¼Œæµ‹åºæˆæœ¬å°†ä¼šå¾—åˆ°å¤§å¹…åº¦é™ä½ï¼Œç›¸ä¿¡ä¼šæˆä¸ºæœªæ¥åŸºå› ç»„ç ”ç©¶çš„ä¸»æµå·¥å…·ä¹‹ä¸€ã€‚</p>
<p>5ã€æœ¬æ–‡çš„ç ”ç©¶ç­–ç•¥ç¡®å®ä¼šå¯è¿ªåæ¥çš„ç ”ç©¶è€…ï¼Œä¸ºde novoç»„è£…æ ‘ç«‹æ–°æ ‡æ†ã€‚Pacbio æµ‹åºæŠ€æœ¯è”åˆå¤šç§scaffolding å¹³å°(æ¯”å¦‚BioNanoã€Hi-Cï¼Œmating-pairedlibr aryï¼Œ10X genomics)çš„ä½¿ç”¨ï¼Œ ä¼šå¼•é¢†de novo ç»„è£…è¿›å…¥ near finished genome æˆ–è€…finished genomeçš„æ–°æ—¶ä»£ã€‚</p>
<p>6ã€ä½†æ˜¯ï¼Œå°½ç®¡åšäº†å¾ˆå¤šåŠªåŠ›ï¼ŒARS1ä»ç„¶æ˜¯ä¸€ä¸ªå•ä½“å‹æ··åˆå‹çš„ç»„è£…ç‰ˆæœ¬ã€‚åœ¨æœªæ¥ï¼Œå¯ä»¥åˆ©ç”¨single moleculeå’ŒHi-CæŠ€æœ¯è·å¾—å•ä½“å‹åˆ†æœŸ(haplotype phasing) çš„å‚è€ƒåŸºå› ç»„ã€‚å¯¹äºç»„æˆå‹å¼‚æŸ“è‰²è´¨åŒºï¼Œå°¤å…¶æ˜¯ç€ä¸ç²’å’Œç«¯ç²’éƒ¨åˆ†ï¼Œä»ç„¶æ˜¯ç»„è£…çš„ç›²åŒºï¼Œå³ä½¿åœ¨äººçš„åŸºå› ç»„ä¸­è¿™äº›åŒºåŸŸä¹Ÿæ²¡æœ‰è¢«å®Œæ•´ç»„è£…å‡ºæ¥ã€‚</p>
<h1 id="3-2020-09-cell-å¤§è±†æ³›åŸºå› ç»„"><a href="#3-2020-09-cell-å¤§è±†æ³›åŸºå› ç»„" class="headerlink" title="3. 2020.09-cell å¤§è±†æ³›åŸºå› ç»„"></a>3. 2020.09-cell å¤§è±†æ³›åŸºå› ç»„</h1><p>Pan-Genome of Wild and Cultivated Soybeans</p>
<p>Soybean is one of the most important vegetable oil and protein feed crops. To capture the entire genomic diversity, it is needed to construct a complete high-quality pan-genome from diverse soybean accessions. In this study, we performed individual de novo genome assemblies for <strong>26 representative soybeans</strong> that were selected from <strong>2,898 deeply sequenced accessions</strong>. Using these assembled genomes together with <strong>three previously reported genomes</strong>, we constructed a <strong>graph-based genome</strong> and performed pan-genome analysis, which identified <strong>numerous genetic variations that cannot be detected by direct mapping of short sequence reads onto a single reference genome</strong>. The <strong>structural variations</strong> from the 2,898 accessions that were <strong>genotyped</strong> based on the graph-based genome and the <strong>RNA-seq data</strong> from the representative 26 accessions helped to <strong>link genetic variations to candidate genes</strong> that are responsible for important traits. This pan-genome resource will promote evolutionary and functional genomics studies in soybean.</p>
<h2 id="SNP-Calling-and-Phylogenetic-Analyses"><a href="#SNP-Calling-and-Phylogenetic-Analyses" class="headerlink" title="SNP Calling and Phylogenetic Analyses"></a>SNP Calling and Phylogenetic Analyses</h2><h2 id="Genome-Assembly"><a href="#Genome-Assembly" class="headerlink" title="Genome Assembly"></a>Genome Assembly</h2><p><em>De novo</em> assembly was conducted referring to a <font color="blue"><strong>reported pipeline</strong></font> (Du and Liang, 2019; Shen et al., 2019). In brief, <font color="red"><strong>Canu</strong></font> (Koren et al., 2017) v1.7.1 was used to assemble PacBio subreads to PacBio contigs, after which <font color="red"><strong>HiSeq</strong></font> reads were used for <font color="red"><strong>error correction</strong></font>. <strong>Bio-nano optical maps</strong> were assembled into consensus physical maps by <strong>BioNano Solve</strong> v3.0.1 (<a target="_blank" rel="noopener" href="https://bionanogenomics.com/">https://bionanogenomics.com/</a>, Solve_06082017Rel). Then <font color="red"><strong>HERA</strong></font> (Du and Liang, 2019) was used to combine PacBio contigs and Bionano based physical maps to <font color="blue"><strong>PacBio-BioNano hybrid scaffolds</strong></font>. To <font color="red"><strong>anchor</strong></font> hybrid scaffolds <font color="blue">into <strong>chromosomes</strong></font>, the <font color="blue"><strong>Hi-C</strong></font> sequencing data were <font color="red"><strong>aligned into scaffolds</strong></font> by <font color="red"><strong>Juicer</strong></font> (Durand et al., 2016) v1.5 and <font color="red"><strong>3D-DNA</strong></font> (Dudchenko et al., 2017).</p>
<h2 id="Repeat-Analysis-and-Gene-Annotation"><a href="#Repeat-Analysis-and-Gene-Annotation" class="headerlink" title="Repeat Analysis and Gene Annotation"></a>Repeat Analysis and Gene Annotation</h2><h2 id="Synteny-Analysis"><a href="#Synteny-Analysis" class="headerlink" title="Synteny Analysis"></a>Synteny Analysis</h2><p>Structural Variation Identification</p>
<p>Genetic Variation Analysis</p>
<p>Gene and miRNA Expression</p>
<p>Core and Dispensable Gene Family Clustering</p>
<p>CHS gene unit identification</p>
<p>Gene Fusion Event Identification</p>
<p>QUANTIFICATION AND STATISTICAL ANALYSIS</p>
<h1 id="4-2019-nc-å¤§è±†æ³›åŸºå› ç»„-ç»„è£…å‚è€ƒ1"><a href="#4-2019-nc-å¤§è±†æ³›åŸºå› ç»„-ç»„è£…å‚è€ƒ1" class="headerlink" title="4. 2019-nc å¤§è±†æ³›åŸºå› ç»„-ç»„è£…å‚è€ƒ1"></a>4. 2019-nc å¤§è±†æ³›åŸºå› ç»„-ç»„è£…å‚è€ƒ1</h1><p>HERA</p>
<p>#5. 2019.09-scienceChina å¤§è±†æ³›åŸºå› ç»„-ç»„è£…å‚è€ƒ2</p>
<p><img src="/blog/2022/03/14/2022-03-14-HiFi-6-HiC-assembly/1.png" alt="1"></p>
<p>For genome assembly, we sequenced additional 40 Gb SMRT reads from 5 cells for this update. Therefore, a total of <strong>120Ã— PacBio reads</strong>, <strong>365Ã— Bionano optical maps</strong> marked by BssSI, <strong>275Ã— Bionano optical maps</strong> marked by BspQI, <strong>45Ã— HiSeq</strong> reads and <strong>125Ã—</strong> chromosome conformation capture sequencing (<strong>Hi-C</strong>) reads were used for the new genome assembly (Figure 1A). To make best of the sequencing data, we also <font color="red"><strong>adopted a new genome assembly pipeline</strong> by <strong>adding HERA</strong></font> (Du and Liang, 2018) to improve the sequence contiguity and reduce errors by accurately assembling the repetitive genome regions (Figure 1A). Briefly, <font color="blue">the whole assembly process <strong>differed from</strong> previous pipeline</font> at </p>
<p>(1) using <font color="red"><strong>CANU</strong></font> (v1.7.1) to replace Smrtmake for assembling PacBio subreads to PacBio contigs; </p>
<p>(2) using <font color="red"><strong>HERA</strong></font> to generate longer contigs; </p>
<p>(3) using <font color="red"><strong>Juicer and 3D-DNA to replace HiC-Pro and LACHESIS</strong></font> to anchor the hybrid scaffolds into chromosomes with Hi-C reads.</p>
<h1 id="6-2020-05-nc-åŒæºå››å€ä½“ç´«èŠ±è‹œè“¿"><a href="#6-2020-05-nc-åŒæºå››å€ä½“ç´«èŠ±è‹œè“¿" class="headerlink" title="6. 2020.05-nc åŒæºå››å€ä½“ç´«èŠ±è‹œè“¿"></a>6. 2020.05-nc åŒæºå››å€ä½“ç´«èŠ±è‹œè“¿</h1><p>HiFiåŠ©åŠ›åŒæºå››å€ä½“ç´«èŠ±è‹œè“¿åŸºå› ç»„ç»„è£…ï¼ŒNC</p>
<blockquote>
<p>Allele-aware chromosome-level genome assembly and efficient transgene-free genome editing for the autotetraploid cultivated alfalfa</p>
</blockquote>
<p>ï¼ˆ1ï¼‰ä½¿ç”¨Canué»˜è®¤å‚æ•°ï¼Œåˆ©ç”¨CCS clean readsç»„è£…contigsã€‚ç»„è£…å¾—åˆ°çš„Contig N50å€¼ä¸º459kbï¼Œæ€»é•¿åº¦ä¸º3.15GBã€‚</p>
<blockquote>
<p>We assembled contigs from CCS clean reads using Canu, with default parameters. The N50 values of the contig sets were 459 kb, with total lengths 3154 Mb.</p>
</blockquote>
<p>ï¼ˆ2ï¼‰ä½¿ç”¨HiC-Proå°†Hi-C readsä¸contigs è¿›è¡Œæ¯”å¯¹ï¼Œäº§ç”Ÿæ¯”å¯¹BAMæ–‡ä»¶ã€‚</p>
<blockquote>
<p>Hi-C reads were aligned to contigs using <a target="_blank" rel="noopener" href="https://pubmed.ncbi.nlm.nih.gov/26619908/">HiC-Pro(2015)</a>, yielding an alignment BAM file.</p>
</blockquote>
<p>ï¼ˆ3ï¼‰ä½¿ç”¨æ³¨é‡Šçš„è’ºè—œè‹œè“¿è›‹ç™½ä½œä¸ºå‚è€ƒï¼Œå®Œå…¨åŸºäºåŒæºçš„ç­–ç•¥æ³¨é‡Šcontigsã€‚å¯¹138,729ä¸ªåŒæºåŸºå› è¿›è¡Œäº†ç»“æ„æ³¨é‡Šã€‚ç”¨MCscanç”¨äºé‰´å®šcontigså’Œå‚è€ƒåŸºå› ç»„ä¹‹é—´çš„å…±çº¿æ€§ã€‚æ˜¾ç¤ºç´«èŠ±è‹œè“¿å’Œè’ºè—œè‹œè“¿ä¹‹é—´çš„é«˜å…±çº¿æ€§ã€‚ã€ä¸‹è½½çˆ¶æ¯æœ¬petunia refã€‘</p>
<blockquote>
<p>Contigs were annotated with a <font color="red"><strong>solely homology-based strategy</strong></font>, using annotated Medicago truncatula proteins as references. 138,729 homologous genes were structurally annotated. <font color="red"><strong>MCscan</strong> in <strong>Jcvi</strong></font> (<a target="_blank" rel="noopener" href="https://zenodo.org/record/31631#.XpkUyTOeask">https://zenodo.org/record/31631#.XpkUyTOeask</a>) was used to identify <font color="red"><strong>synteny blocks</strong></font> between <strong>contigs and the reference</strong> genome. <font color="red"><strong>Contigs syntenic</strong> to M. truncatula were <strong>stacked and aligned</strong> to M. truncatula chromosomes</font>. The syntenic contigs are summarized in Supplementary Table 4.</p>
</blockquote>
<p>ï¼ˆ4ï¼‰ä½¿ç”¨å†…éƒ¨è„šæœ¬å¤„ç†BAMæ–‡ä»¶ï¼Œå»é™¤ç­‰ä½åŸºå› contigsä¹‹é—´çš„linksã€‚ä½¿ç”¨ALLHiCè½¯ä»¶ï¼Œ<font color="red">æå–ã€èšç±»å’Œé‡æ’Contigs</font> (Contigs syntenicä¸è’ºè—œè‹œè“¿æŸ“è‰²ä½“ä¸€è‡´)ï¼Œå¾—åˆ°åŸå§‹çš„scaffoldsã€‚</p>
<blockquote>
<p>An <strong>in-house script</strong> was used to <font color="red">prune the BAM</font> file and <font color="red">discard links between allelic contigs</font>. Contigs syntenic to one chromosome of M. truncatula, e.g., chr1, were extracted, sub-clustered and reordered using <font color="red">ALLHiC</font>, yielding a raw scaffold set.</p>
</blockquote>
<p>ï¼ˆ5ï¼‰<a target="_blank" rel="noopener" href="https://pubmed.ncbi.nlm.nih.gov/27467250/">Juicebox</a>ç”¨äºä»¥å›¾å½¢å’Œäº¤äº’æ–¹å¼å¾®è°ƒç»„è£…çš„scaffoldsã€‚å‰ªè£äº†40ä¸ªæ€»é•¿åº¦è¾¾1800Mbçš„scaffolds ã€‚</p>
<blockquote>
<p>Juicebox45 was used for fine-tuning assembled scaffolds in a graphic and inter-active fashion. Forty scaffolds with a total length of 1800 Mb were cropped (Supplementary Table 5).</p>
</blockquote>
<p>ï¼ˆ6ï¼‰ åŸºäºç»„è£…çš„scaffoldï¼Œé€šè¿‡Hi-Cæ•°æ®ï¼Œæ¯ä¸ªunplaced contigè¢«åˆ†é…åˆ°äº’ä½œæœ€å¼ºçš„é‚£äº›contig clusteré‡Œã€‚</p>
<blockquote>
<p>Based on this scaffold assembly, each unplaced contig was assigned to the contig cluster, to which the contig was most connected by Hi-C data.</p>
</blockquote>
<p>ï¼ˆ7ï¼‰ä½¿ç”¨ALLHiCå¯¹é‚£äº›contig clusterså†æ¬¡è¿›è¡Œé‡æ’å’Œæ„å»ºscaffoldã€‚</p>
<blockquote>
<p>Those contig clusters were reordered and scaffolded using ALLHiC. </p>
</blockquote>
<p>ï¼ˆ8ï¼‰ä½¿ç”¨Juiceboxå¯¹scaffoldsè¿›è¡Œå¾®è°ƒï¼Œå¹¶ä»scaffoldsä¸Šå»é™¤ä¸ä¸€è‡´çš„contigsï¼Œäº§ç”Ÿæœ€ç»ˆçš„æŸ“è‰²ä½“åŸºå› ç»„ï¼Œå…¶åŒ…å«32æ¡æŸ“è‰²ä½“(8ä¸ªåŒæºç»„ï¼Œæ¯ä¸ªç»„ä¸­æœ‰4ä¸ªç­‰ä½åŸºå› æŸ“è‰²ä½“)ï¼Œæ€»é•¿åº¦ä¸º2738Mbï¼Œå’Œ419MbæœªæŒ‚è½½åˆ°æŸ“è‰²ä½“æ°´å¹³çš„åºåˆ—ã€‚</p>
<blockquote>
<p>Using Juicebox, scaffolds were fine-tuned and discordant contigs were removed from scaffolds, and the final chromosome assembly was generated, containing 32 chromosomes with a total length of 2738 Mb (Supplementary Table 6).</p>
</blockquote>
<h1 id="7-2021-11-gb-æ¤°å­ğŸ¥¥"><a href="#7-2021-11-gb-æ¤°å­ğŸ¥¥" class="headerlink" title="7. 2021.11-gb æ¤°å­ğŸ¥¥"></a>7. 2021.11-gb æ¤°å­ğŸ¥¥</h1><p>High-quality reference genome sequences of two coconut cultivars provide insights into evolution of monocot chromosomes and differentiation of fiber content and plant height</p>
<p>è¯¥ç ”ç©¶ç»˜åˆ¶ä¸¤ä¸ªå‚è€ƒçº§åˆ«æ¤°å­åŸºå› ç»„ã€‚æ­ç¤ºå•å­å¶æ¤ç‰©æŸ“è‰²ä½“çš„è¿›åŒ–è¿‡ç¨‹å’Œæ¤°å­æŸ“è‰²ä½“çš„å½¢æˆè¿‡ç¨‹ï¼Œåˆ†æé«˜çŸ®ä¸¤ç§æ¤°å­è€ç›ï¼Œæ ªé«˜ï¼Œçº¤ç»´å’Œè„‚è´¨å«é‡ç­‰å…³é”®æ€§çŠ¶çš„é—ä¼ å·®å¼‚ï¼ŒåŒæ—¶å‘ç°æ¤°å­è¿›åŒ–è¿‡ç¨‹ä¸­çº¦åœ¨400ä¸‡å¹´å‰è‡ªç„¶å‘ç”Ÿçš„ç»¿è‰²é©å‘½ã€‚é«˜æ¤°å­2.39GçŸ®æ¤°å­2.40Gã€‚åŸºäºæ¤°å­ï¼Œç´«èï¼Œè èç­‰åŸºå› ç»„çš„å…±çº¿æ€§å…³ç³»ï¼Œé‡æ–°æ„å»ºäº†å«æœ‰10æ¡æŸ“è‰²ä½“çš„å¤å•å­å¶æ¤ç‰©åŸºå› ç»„æ ¸å‹ï¼Œå¹¶å¯¹åŸºå› ç»„æ ¸å‹çš„è¿›åŒ–è·¯çº¿è¿›è¡Œæ¨æ–­ã€‚è¿›ä¸€æ­¥åˆ©ç”¨å¶é—´è·è¡¨å‹è¿›è¡Œå…¨åŸºå› ç»„å…³è”åˆ†æï¼Œé‰´å®šå‡º12å·æŸ“è‰²ä½“ä¸ŠGA20oxåŸºå› ä¸æ¤°å­æ ªé«˜æ˜¾è‘—ç›¸å…³ï¼Œå¹¶å­˜åœ¨æ‹·è´æ•°å˜å¼‚ï¼›æ•´åˆå¤šç§æ‰‹æ®µéªŒè¯HA20oxä»¥åŠèµ¤éœ‰ç´ ä»£è°¢é€”å¾„åœ¨é«˜çŸ®ç§çš„å·®å¼‚å¹¶å¯¼è‡´æ ªé«˜å·®å¼‚ã€‚400ä¸‡å¹´å‰ï¼Œç»¿è‰²é©å‘½ã€‚</p>
<h2 id="Hi-C-based-genome-assembly"><a href="#Hi-C-based-genome-assembly" class="headerlink" title="Hi-C-based genome assembly"></a>Hi-C-based genome assembly</h2><p>Both Cn. tall and Cn. dwarf genomes were assembled de novo using SMARTdenovo (<a target="_blank" rel="noopener" href="https://github.com/ruanjue/smartdenovo">https://github.com/ruanjue/smartdenovo</a>) [-k 19 -J 3000] based on <font color="red"><strong>Nanopore long reads</strong></font> <strong>corrected</strong> with <font color="red"><strong>NextDenovo</strong></font> (<a target="_blank" rel="noopener" href="https://github.com/Nextomics/NextDenovo">https://github.com/Nextomics/NextDenovo</a>). Raw contigs were <font color="blue"><strong>polished</strong></font> using <font color="blue"><strong>Nanopore long reads</strong></font> and <font color="blue"><strong>Illumina short reads</strong></font> with <font color="red"><strong>Next-Polish</strong></font> (v1.2.2) [49]. Then polished contigs were <font color="blue"><strong>anchored</strong></font> into chromosomes by using <font color="red"><strong>HiC-Pro (v2.11.1)</strong></font> [50] and <font color="red"><strong>LACHESIS</strong></font>[51]. Finally, the assembled genomes were <font color="blue"><strong>manual correction</strong></font> with <font color="red"><strong>Juicebox</strong></font> (v1.11.08).</p>
<p>To assess the quality of assembled genomes, raw Illumina <font color="blue"><strong>paired-end</strong> reads and <strong>RNA-seq</strong></font> reads from multiple tissues were <font color="blue"><strong>mapping</strong></font> to the genomes using <font color="red"><strong>BWA (v0.7.17-r1188)</strong></font> [52] and <font color="red"><strong>HISAT2 (v2.1.0)</strong></font> [53], respectively. In addition, Benchmarking Universal Single-Copy Orthologs (<font color="red"><strong>BUSCO</strong></font>) was also used to assess the genome completeness base on Embryophyta Plant database (odb10) [54].</p>
<h1 id="8-2021-11-nc-é‡‘ç²Ÿå…°åŸºå› ç»„"><a href="#8-2021-11-nc-é‡‘ç²Ÿå…°åŸºå› ç»„" class="headerlink" title="8. 2021.11-nc é‡‘ç²Ÿå…°åŸºå› ç»„"></a>8. 2021.11-nc é‡‘ç²Ÿå…°åŸºå› ç»„</h1><p>å¡«è¡¥æ ¸å¿ƒè¢«å­æ¤ç‰©æœ€åä¸»è¦åˆ†æ”¯-é‡‘ç²Ÿå…°ç›®çš„åŸºå› ç»„ä¿¡æ¯ï¼Œæä¾›è¢«å­æ¤ç‰©æ—©æœŸæ¼”åŒ–è¯æ®ã€‚åˆ˜å»ºå…¨å›¢é˜Ÿè‡´åŠ›äºè§£æç±»ç¾¤æ¼”åŒ–ï¼Œæ¢ç©¶ä¸åŒç‰©ç§é—´åˆ†åŒ–é€‚åº”é—ä¼ æœºåˆ¶ã€‚</p>
<p>è¢«å­æ¤ç‰©ï¼ˆAngiosperms or flowering plantsï¼‰æ˜¯åœ°çƒä¸Šåˆ†å¸ƒèŒƒå›´æœ€å¹¿ã€å¤šæ ·åŒ–æœ€é«˜ã€é€‚åº”æ€§æœ€å¼ºçš„é™†ç”Ÿæ¤ç‰©ç±»ç¾¤ï¼Œæå¤§åœ°å½±å“äº†å…¶ä»–ç”Ÿç‰©ç±»ç¾¤çš„æ¼”åŒ–è¿›ç¨‹ï¼Œç„¶è€Œï¼Œç”±äºè¢«å­æ¤ç‰©åœ¨æ—©æœŸç»å†äº†å¿«é€Ÿçš„è¾å°„è¿›åŒ–ï¼Œå…¶å†…éƒ¨æ¼”åŒ–å…³ç³»è¿Ÿè¿Ÿæœªè¢«è§£å†³ï¼Œè¿™ä¸€é—®é¢˜ä¹Ÿè¢«ç§°ä¸ºè¾¾å°”æ–‡çš„â€œæ¼äººä¹‹è°œâ€ã€‚ç»è¿‡å‰äººé•¿æ—¶æœŸçš„åŠªåŠ›ï¼Œç›®å‰è¢«å­æ¤ç‰©è¢«åˆ’åˆ†ä¸ºï¼šæ— æ²¹æ¨Ÿç›®ã€ç¡è²ç›®ã€æœ¨å…°è—¤ç›®å’Œæ ¸å¿ƒè¢«å­æ¤ç‰©ï¼ˆMesangiospermaeï¼‰ã€‚æ— æ²¹æ¨Ÿç›®ã€ç¡è²ç›®ã€æœ¨å…°è—¤ç›®å…±åŒè¢«ç§°ä¸ºANA gradeï¼Œæ˜¯è¢«å­æ¤ç‰©çš„åŸºéƒ¨ç±»ç¾¤ï¼Œç³»ç»Ÿå‘è‚²å…³ç³»ç›¸å¯¹ç¨³å®šï¼Œè€Œæ ¸å¿ƒè¢«å­æ¤ç‰©åŒ…å«äº†~99.95%çš„è¢«å­æ¤ç‰©ï¼Œåˆå¯è¢«åˆ’åˆ†ä¸º5å¤§ç±»ç¾¤ï¼šåŒå­å¶ã€å•å­å¶ã€æœ¨å…°ç±»ã€é‡‘é±¼è—»ç›®å’Œé‡‘ç²Ÿå…°ç›®ï¼Œå®ƒä»¬ä¹‹é—´çš„æ¼”åŒ–å…³ç³»ä¸€ç›´å­˜åœ¨äº‰è®®ã€‚è¿‘å¹´æ¥ï¼Œéšç€å„ç±»ç¾¤ä»£è¡¨ç‰©ç§å…¨åŸºå› ç»„æµ‹åºç›¸ç»§å®Œæˆï¼Œå…¶å„è‡ªçš„è¿›åŒ–å†ç¨‹å‡æœ‰äº†åšå®è€Œå¯é çš„è¯æ®ï¼Œåˆ˜å»ºå…¨å›¢é˜Ÿåœ¨2020å¹´é¦–æ¬¡æŠ¥é“äº†é‡‘é±¼è—»å’ŒèŠ¡å®çš„åŸºå› ç»„ï¼Œè¯æ˜äº†æ ¸å¿ƒè¢«å­æ¤ç‰©å†…éƒ¨å­˜åœ¨å¹¿æ³›çš„ä¸å®Œå…¨è°±ç³»åˆ†é€‰å’Œæ‚äº¤äº‹ä»¶æ˜¯å¯¼è‡´æ ‘å½¢å†²çªçš„ä¸»è¦åŸå› ã€‚ä½†æ˜¯é‡‘ç²Ÿå…°ç›®ä¸€ç›´æœªæœ‰åŸºå› ç»„æŠ¥é“ï¼Œå…¶æ¼”åŒ–å†ç¨‹ä»¥åŠæ•´ä¸ªæ ¸å¿ƒè¢«å­æ¤ç‰©çš„æ¼”åŒ–è¿‡ç¨‹è¿˜æœ‰å¾…æ·±å…¥åˆ†æã€‚ä¸ºæ­¤è¯¥å›¢é˜Ÿå†æ¬¡åˆ©ç”¨Nanoporeã€Illuminaå’ŒHi-CæŠ€æœ¯ï¼Œæ„å»ºäº†ç¬¬ä¸€ä¸ªé‡‘ç²Ÿå…°ç›®æ¤ç‰©ï¼ˆå››å·é‡‘ç²Ÿå…°ï¼Œ<em>Chloranthus sessilifolius</em>ï¼‰çš„é«˜è´¨é‡æŸ“è‰²ä½“çº§åˆ«åŸºå› ç»„ï¼ŒContig N50é«˜è¾¾53.74 Mbï¼ˆå›¾1ï¼‰ï¼Œä½¿ç”¨å¤šç§åˆ†ææ–¹æ³•å¯¹ä¸Šè¿°é—®é¢˜è¿›è¡Œäº†æ·±å…¥çš„é˜è¿°ã€‚</p>
<p>é€šè¿‡ä½¿ç”¨å¤šç§ç³»ç»Ÿå‘è‚²åŸºå› ç»„å­¦åˆ†æï¼ŒåŒ…æ‹¬å•æ‹·è´åŸºå› ã€ä½æ‹·è´åŸºå› å’Œå…±çº¿æ€§åŸºå› æ•°æ®é›†ï¼Œè¿›è¡Œäº†ä¸²è”æ ‘å’Œæº¯ç¥–æ ‘çš„æ„å»ºï¼ŒåŒæ—¶è¯„ä¼°äº†é•¿æå¸å¼•ã€ç‰©ç§é€‰æ‹©å’ŒåŒæºèšç±»æ–¹æ³•ç­‰çš„å½±å“ï¼Œä½¿ç”¨å¤šå¥—æ•°æ®é›†åˆå’Œå¤šç§åˆ†ææ–¹æ³•ï¼Œè·å¾—äº†é«˜å¯ä¿¡åº¦çš„æ‹“æ‰‘ç»“æ„ï¼Œå³é‡‘ç²Ÿå…°ç›®å’Œæœ¨å…°ç±»å…·æœ‰æœ€è¿‘çš„äº²ç¼˜å…³ç³»ï¼Œä¸”ä»–ä»¬ä¸€èµ·ä¸é‡‘é±¼è—»ç›®+åŒå­å¶æ¤ç‰©æ„æˆå§Šå¦¹å…³ç³»ï¼Œè€Œå•å­å¶æ¤ç‰©æ˜¯å…¶å®ƒæ‰€æœ‰æ ¸å¿ƒè¢«å­æ¤ç‰©çš„å§Šå¦¹æï¼ˆå›¾2ï¼‰ã€‚</p>
<p>åŒæ—¶ï¼Œä½œè€…ä¹Ÿå‘ç°äº†å¤§é‡çš„åŸºå› æ ‘æˆ–å¶ç»¿ä½“æ ‘ä¸ç‰©ç§æ ‘ä¸ä¸€è‡´çš„æƒ…å†µï¼Œå› æ­¤ç€é‡è¯„ä¼°äº†ä¸å®Œå…¨è°±ç³»åˆ†é€‰ï¼ˆILSï¼‰å’Œæ‚äº¤äº‹ä»¶å¯¹æ­¤å†²çªçš„è´¡çŒ®ã€‚é€šè¿‡PhyloNetworksåˆ†æé‰´å®šåˆ°äº†ä¸‰æ¬¡å¯èƒ½çš„æ‚äº¤äº‹ä»¶ï¼Œå¯ä»¥è§£æéƒ¨åˆ†çš„æ ‘å½¢å†²çªã€‚ç„¶è€Œå€¼å¾—æ³¨æ„çš„æ˜¯äº”å¤§ç±»ç¾¤ä¹‹é—´çš„åˆ†åŒ–æ—¶é—´ä»…æœ‰23ä¸ªç™¾ä¸‡å¹´ï¼ˆ158~135 Myaï¼‰ï¼Œå¦‚æ­¤çŸ­çš„åˆ†åŒ–æ—¶é—´ï¼ŒILSå¯èƒ½å‘æŒ¥äº†æ›´ä¸ºä¸»è¦çš„ä½œç”¨ã€‚ä½¿ç”¨thetaå‚æ•°å¯¹æ¯ä¸ªå†…éƒ¨åˆ†æ”¯å‘ç”ŸILSçš„æ¦‚ç‡è¿›è¡Œè¯„ä¼°å‘ç°ï¼Œé™¤äº†æœ¨å…°ç±»å’Œé‡‘ç²Ÿå…°ç›®çš„å§Šå¦¹å…³ç³»è¾ƒä¸ºç¨³å®šå¤–ï¼Œå…¶ä»–ç±»ç¾¤é—´éƒ½å…·æœ‰è¾ƒé«˜çš„thetaå€¼ï¼Œè€Œé‡‘é±¼è—»ç›®+åŒå­å¶çš„ç¥–å…ˆåˆ†æå…·æœ‰è¾ƒé«˜çš„tehtaå€¼ï¼Œå‡è¡¨æ˜äº†ILSåœ¨æ ¸å¿ƒè¢«å­æ¤ç‰©å†…éƒ¨çš„æ¼”åŒ–ä¸­å‘æŒ¥äº†é‡è¦ä½œç”¨ï¼ˆå›¾3 aå’Œbï¼‰ã€‚åŸºäºPhyabseå’ŒDendroPyè¿›è¡Œæ ‘å½¢æ¨¡æ‹Ÿè¡¨æ˜ï¼Œå½“å­˜åœ¨ILSæ—¶æ¨¡æ‹Ÿçš„æ ‘å½¢å’ŒçœŸå®æ ‘å½¢å…·æœ‰é«˜åº¦ä¸€è‡´æ€§ï¼ˆR2ï¼0.99ï¼‰ï¼Œè€Œä¸å­˜åœ¨ILSçš„æ¨¡æ‹Ÿæ ‘å½¢ä¸çœŸå®æ ‘å½¢åˆ™å…·æœ‰è¾ƒä½çš„ç›¸å…³æ€§ï¼Œè¿›ä¸€æ­¥è¡¨æ˜äº†ILSåœ¨æ ¸å¿ƒè¢«å­æ¤ç‰©çš„å¿«é€Ÿè¾å°„ä¸­å‘æŒ¥äº†é‡è¦ä½œç”¨ï¼ˆå›¾3 c-fï¼‰ã€‚è¿›ä¸€æ­¥å¯¹æ ‘å½¢å¼‚è´¨æ€§è¿›è¡ŒåŒºåˆ†è¡¨æ˜ï¼ŒILSå¯åŸºæœ¬è§£é‡Šé‡‘é±¼è—»ç›®+åŒå­å¶çš„ç¥–å…ˆæä¸Šçš„æ ‘å½¢å†²çªï¼Œä¹Ÿå°±æ˜¯é‡‘é±¼è—»ç›®+åŒå­å¶ã€æœ¨å…°ç±»+é‡‘ç²Ÿå…°å’Œå•å­å¶ä¹‹é—´çš„å†²çªï¼Œè€Œå…¶ä»–æ ¸å¿ƒè¢«å­æ¤ç‰©ç±»ç¾¤é—´çš„å†²çªè¿˜æœ‰å…¶ä»–å› ç´ ï¼ˆä¾‹å¦‚æ‚äº¤ï¼‰çš„å‚ä¸ã€‚</p>
<p>æ­¤å¤–ï¼Œæœ¬ç ”ç©¶è¿˜é‰´å®šäº†å››å·é‡‘ç²Ÿå…°ç‹¬æœ‰çš„ä¸€æ¬¡WGDäº‹ä»¶ï¼ŒåŒæ—¶5å¤§æ ¸å¿ƒè¢«å­æ¤ç‰©åˆ†æçš„åŠ å€äº‹ä»¶éƒ½ä¸å…±äº«ï¼›é‰´å®šäº†èŠ±å‘è‚²ç›¸å…³çš„åŸºå› å®¶æ—ï¼Œè§£æäº†å…¶ç‰¹æœ‰èŠ±å™¨å®˜çš„å¯èƒ½å½¢æˆåŸå› ï¼›é‰´å®šäº†èœç±»åˆæˆå’Œæœ¨è´¨éƒ¨å½¢æˆç›¸å…³çš„åŸºå› å®¶æ—ï¼Œä¸ºä»Šåç ”ç©¶å…¶ç®€å•æœ¨è´¨éƒ¨å½¢æˆå’Œèœç±»ç‰©ç§åˆæˆé€šè·¯æ¼”åŒ–æä¾›äº†åŸºç¡€ã€‚</p>
<p>ç»¼ä¸Šæ‰€è¿°ï¼Œè¯¥ç ”ç©¶æŠ¥é“äº†é«˜è´¨é‡çš„å››å·é‡‘ç²Ÿå…°åŸºå› ç»„ï¼Œå¡«è¡¥äº†æ ¸å¿ƒè¢«å­æ¤ç‰©é‡‘ç²Ÿå…°ç›®çš„åŸºå› ç»„ä¿¡æ¯ï¼Œå¹¶é€šè¿‡å¤šç§åˆ†æç­–ç•¥å’Œæ–¹æ³•ï¼Œæ·±å…¥çš„æ­ç¤ºäº†æ ¸å¿ƒè¢«å­æ¤ç‰©é—´çš„å¤æ‚æ¼”åŒ–å†ç¨‹ï¼Œè¯¥æˆæœå¯¹äºæ—©æœŸè¢«å­æ¤ç‰©çš„èµ·æºã€è¾å°„è¿›åŒ–ä»¥åŠé€‚åº”æ€§æ¼”åŒ–æä¾›æ•°æ®æ”¯æ’‘å’Œç†è®ºåŸºç¡€ã€‚</p>
<h2 id="Genome-size-estimate-and-assembly"><a href="#Genome-size-estimate-and-assembly" class="headerlink" title="Genome size estimate and assembly."></a>Genome size estimate and assembly.</h2><p>To estimate the genome size of C. sessilifolius, we surveyed 150 bp paired-end reads, computed 21 bp K-mer frequencies using <font color="red"><strong>Jellyfish</strong></font>, and exported the resulting histogram into <font color="red">findGSE</font>. <font color="red"><strong>Nextdenovo</strong></font> (https:// github.com/Nextomics/Nextdenovo) was selected for <font color="blue"><strong>correcting reads</strong></font> with para- meters â€œread_cutoff=2k, seed_cutoff=30k, blocksize=1.5gâ€ and then <font color="red"><strong>Smartdenovo</strong></font> (<a target="_blank" rel="noopener" href="https://github.com/ruanjue/smartdenovo">https://github.com/ruanjue/smartdenovo</a>) for <font color="blue"><strong>de novo assembly</strong></font> with parameters â€œwtpre -J 3,000; wtzmo-k 21 -z 10 -Z 16 -U -1 -m 0.1 -A 1000; wtclp -d 3 -k 300 -m 0.1 -FT; wtlay -w 300 -s 200 -m 0.1 -r 0.95 -c 1â€. <font color="blue">The preliminary contigs were further <strong>polished</strong> by aligning the Illumina <strong>short</strong> <strong>reads</strong> to the contigs</font> using <font color="red"><strong>Nextpolish</strong></font>. After four rounds of successive iterative correction, the final genome sequence was obtained. The GC content and sequencing coverage analyses were applied to evaluate the presence of contamination. The quality of the genome assembly was also assessed using <font color="red"><strong>BUSCO</strong></font> (Benchmarking Universal Single-Copy Orthologs) with the embry- ophyta_odb10 database. The clean <font color="blue">Hi-C</font> data were mapped to contig sequences by <font color="red"><strong>Bowtie2</strong></font> and 354 Mb valid interaction pairs were extracted. Based on those chromatin interactions, <font color="red"><strong>LACHESIS</strong></font> was employed to <font color="blue"><strong>cluster</strong>, <strong>order</strong>, and <strong>orient</strong> the <strong>contigs</strong> into <strong>pseudo-chromosomes</strong></font>.</p>

    </div>

    
    
    
        

<div>
<ul class="post-copyright">
  <li class="post-copyright-author">
    <strong>Post author:  </strong>Wang C
  </li>
  <li class="post-copyright-link">
    <strong>Post link: </strong>
    <a href="https://wangc1106.github.io/blog/2022/03/14/2022-03-14-HiFi-6-HiC-assembly/" title="HiFi-6-[HiFi-HiC-assembly-method]">https://wangc1106.github.io/blog/2022/03/14/2022-03-14-HiFi-6-HiC-assembly/</a>
  </li>
  <li class="post-copyright-license">
    <strong>Copyright Notice:  </strong>All articles in this blog are licensed under <a href="https://creativecommons.org/licenses/by-nc-sa/4.0/" rel="noopener" target="_blank"><i class="fab fa-fw fa-creative-commons"></i>BY-NC-SA</a> unless stating additionally.
  </li>
</ul>
</div>


      <footer class="post-footer">
          <div class="post-tags">
              <a href="/blog/tags/denovo-III/" rel="tag"># denovo-III</a>
          </div>

        


        
    <div class="post-nav">
      <div class="post-nav-item">
    <a href="/blog/2022/03/09/2022-03-09-HiFi-5-HiC-Pro/" rel="prev" title="HiFi-5-[HiC-Pro]-[MCscan]-[ALLHiC]">
      <i class="fa fa-chevron-left"></i> HiFi-5-[HiC-Pro]-[MCscan]-[ALLHiC]
    </a></div>
      <div class="post-nav-item"></div>
    </div>
      </footer>
    
  </article>
  
  
  



          </div>
          

<script>
  window.addEventListener('tabs:register', () => {
    let { activeClass } = CONFIG.comments;
    if (CONFIG.comments.storage) {
      activeClass = localStorage.getItem('comments_active') || activeClass;
    }
    if (activeClass) {
      let activeTab = document.querySelector(`a[href="#comment-${activeClass}"]`);
      if (activeTab) {
        activeTab.click();
      }
    }
  });
  if (CONFIG.comments.storage) {
    window.addEventListener('tabs:click', event => {
      if (!event.target.matches('.tabs-comment .tab-content .tab-pane')) return;
      let commentClass = event.target.classList[1];
      localStorage.setItem('comments_active', commentClass);
    });
  }
</script>

        </div>
          
  
  <div class="toggle sidebar-toggle">
    <span class="toggle-line toggle-line-first"></span>
    <span class="toggle-line toggle-line-middle"></span>
    <span class="toggle-line toggle-line-last"></span>
  </div>

  <aside class="sidebar">
    <div class="sidebar-inner">

      <ul class="sidebar-nav motion-element">
        <li class="sidebar-nav-toc">
          Table of Contents
        </li>
        <li class="sidebar-nav-overview">
          Overview
        </li>
      </ul>

      <!--noindex-->
      <div class="post-toc-wrap sidebar-panel">
          <div class="post-toc motion-element"><ol class="nav"><li class="nav-item nav-level-1"><a class="nav-link" href="#1-2016-05-np-%E7%A2%A7%E5%86%AC%E8%8C%84"><span class="nav-number">1.</span> <span class="nav-text">1. 2016.05-np ç¢§å†¬èŒ„</span></a><ol class="nav-child"><li class="nav-item nav-level-2"><a class="nav-link" href="#Genome-sequencing-assembly-and-annotation"><span class="nav-number">1.1.</span> <span class="nav-text">Genome sequencing, assembly and annotation.</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#Repetitive-elements-analysis"><span class="nav-number">1.2.</span> <span class="nav-text">Repetitive elements analysis</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#Gene-data-mining"><span class="nav-number">1.3.</span> <span class="nav-text">Gene data mining</span></a></li></ol></li><li class="nav-item nav-level-1"><a class="nav-link" href="#2-2017-04-ng-%E5%B1%B1%E7%BE%8A%F0%9F%90%90"><span class="nav-number">2.</span> <span class="nav-text">2. 2017.04-ng å±±ç¾ŠğŸ</span></a><ol class="nav-child"><li class="nav-item nav-level-2"><a class="nav-link" href="#data"><span class="nav-number">2.1.</span> <span class="nav-text">data</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#%E7%BB%84%E8%A3%85%E7%AD%96%E7%95%A5%E5%8F%8A%E7%BB%84%E8%A3%85%E7%BB%93%E6%9E%9C"><span class="nav-number">2.2.</span> <span class="nav-text">ç»„è£…ç­–ç•¥åŠç»„è£…ç»“æœ</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#%E7%BB%84%E8%A3%85%E8%AF%84%E4%BC%B0%E5%8F%8A%E6%AF%94%E8%BE%83%E5%88%86%E6%9E%90"><span class="nav-number">2.3.</span> <span class="nav-text">ç»„è£…è¯„ä¼°åŠæ¯”è¾ƒåˆ†æ</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#%E5%9F%BA%E5%9B%A0%E7%BB%84%E6%B3%A8%E9%87%8A"><span class="nav-number">2.4.</span> <span class="nav-text">åŸºå› ç»„æ³¨é‡Š</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#%E5%90%AF%E7%A4%BA%E5%BD%95"><span class="nav-number">2.5.</span> <span class="nav-text">å¯ç¤ºå½•</span></a></li></ol></li><li class="nav-item nav-level-1"><a class="nav-link" href="#3-2020-09-cell-%E5%A4%A7%E8%B1%86%E6%B3%9B%E5%9F%BA%E5%9B%A0%E7%BB%84"><span class="nav-number">3.</span> <span class="nav-text">3. 2020.09-cell å¤§è±†æ³›åŸºå› ç»„</span></a><ol class="nav-child"><li class="nav-item nav-level-2"><a class="nav-link" href="#SNP-Calling-and-Phylogenetic-Analyses"><span class="nav-number">3.1.</span> <span class="nav-text">SNP Calling and Phylogenetic Analyses</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#Genome-Assembly"><span class="nav-number">3.2.</span> <span class="nav-text">Genome Assembly</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#Repeat-Analysis-and-Gene-Annotation"><span class="nav-number">3.3.</span> <span class="nav-text">Repeat Analysis and Gene Annotation</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#Synteny-Analysis"><span class="nav-number">3.4.</span> <span class="nav-text">Synteny Analysis</span></a></li></ol></li><li class="nav-item nav-level-1"><a class="nav-link" href="#4-2019-nc-%E5%A4%A7%E8%B1%86%E6%B3%9B%E5%9F%BA%E5%9B%A0%E7%BB%84-%E7%BB%84%E8%A3%85%E5%8F%82%E8%80%831"><span class="nav-number">4.</span> <span class="nav-text">4. 2019-nc å¤§è±†æ³›åŸºå› ç»„-ç»„è£…å‚è€ƒ1</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#6-2020-05-nc-%E5%90%8C%E6%BA%90%E5%9B%9B%E5%80%8D%E4%BD%93%E7%B4%AB%E8%8A%B1%E8%8B%9C%E8%93%BF"><span class="nav-number">5.</span> <span class="nav-text">6. 2020.05-nc åŒæºå››å€ä½“ç´«èŠ±è‹œè“¿</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#7-2021-11-gb-%E6%A4%B0%E5%AD%90%F0%9F%A5%A5"><span class="nav-number">6.</span> <span class="nav-text">7. 2021.11-gb æ¤°å­ğŸ¥¥</span></a><ol class="nav-child"><li class="nav-item nav-level-2"><a class="nav-link" href="#Hi-C-based-genome-assembly"><span class="nav-number">6.1.</span> <span class="nav-text">Hi-C-based genome assembly</span></a></li></ol></li><li class="nav-item nav-level-1"><a class="nav-link" href="#8-2021-11-nc-%E9%87%91%E7%B2%9F%E5%85%B0%E5%9F%BA%E5%9B%A0%E7%BB%84"><span class="nav-number">7.</span> <span class="nav-text">8. 2021.11-nc é‡‘ç²Ÿå…°åŸºå› ç»„</span></a><ol class="nav-child"><li class="nav-item nav-level-2"><a class="nav-link" href="#Genome-size-estimate-and-assembly"><span class="nav-number">7.1.</span> <span class="nav-text">Genome size estimate and assembly.</span></a></li></ol></li></ol></div>
      </div>
      <!--/noindex-->

      <div class="site-overview-wrap sidebar-panel">
        <div class="site-author motion-element" itemprop="author" itemscope itemtype="http://schema.org/Person">
    <img class="site-author-image" itemprop="image" alt="Wang C"
      src="/blog/images/banner.png">
  <p class="site-author-name" itemprop="name">Wang C</p>
  <div class="site-description" itemprop="description">Make more time</div>
</div>
<div class="site-state-wrap motion-element">
  <nav class="site-state">
      <div class="site-state-item site-state-posts">
          <a href="/blog/archives">
          <span class="site-state-item-count">32</span>
          <span class="site-state-item-name">posts</span>
        </a>
      </div>
      <div class="site-state-item site-state-categories">
            <a href="/blog/categories/">
          
        <span class="site-state-item-count">11</span>
        <span class="site-state-item-name">categories</span></a>
      </div>
      <div class="site-state-item site-state-tags">
            <a href="/blog/tags/">
          
        <span class="site-state-item-count">18</span>
        <span class="site-state-item-name">tags</span></a>
      </div>
  </nav>
</div>
  <div class="links-of-author motion-element">
      <span class="links-of-author-item">
        <a href="https://github.com/wangc1106" title="GitHub â†’ https:&#x2F;&#x2F;github.com&#x2F;wangc1106" rel="noopener" target="_blank"><i class="fab fa-github fa-fw"></i>GitHub</a>
      </span>
      <span class="links-of-author-item">
        <a href="mailto:wangchen116@126.com" title="E-Mail â†’ mailto:wangchen116@126.com" rel="noopener" target="_blank"><i class="fa fa-envelope fa-fw"></i>E-Mail</a>
      </span>
  </div>
  <div class="cc-license motion-element" itemprop="license">
    <a href="https://creativecommons.org/licenses/by-nc-sa/4.0/" class="cc-opacity" rel="noopener" target="_blank"><img src="/blog/images/cc-by-nc-sa.svg" alt="Creative Commons"></a>
  </div>



      </div>
        <div class="back-to-top motion-element">
          <i class="fa fa-arrow-up"></i>
          <span>0%</span>
        </div>

    </div>
  </aside>
  <div id="sidebar-dimmer"></div>


      </div>
    </main>

    <footer class="footer">
      <div class="footer-inner">
        

        

<div class="copyright">
  
  &copy; 2021 â€“ 
  <span itemprop="copyrightYear">2022</span>
  <span class="with-love">
    <i class="fas fa-dna"></i>
  </span>
  <span class="author" itemprop="copyrightHolder">Wang C</span>
</div>

        






<script>
  (function() {
    function leancloudSelector(url) {
      url = encodeURI(url);
      return document.getElementById(url).querySelector('.leancloud-visitors-count');
    }

    function addCount(Counter) {
      var visitors = document.querySelector('.leancloud_visitors');
      var url = decodeURI(visitors.id);
      var title = visitors.dataset.flagTitle;

      Counter('get', '/classes/Counter?where=' + encodeURIComponent(JSON.stringify({ url })))
        .then(response => response.json())
        .then(({ results }) => {
          if (results.length > 0) {
            var counter = results[0];
            leancloudSelector(url).innerText = counter.time + 1;
            Counter('put', '/classes/Counter/' + counter.objectId, { time: { '__op': 'Increment', 'amount': 1 } })
              .catch(error => {
                console.error('Failed to save visitor count', error);
              });
          } else {
              Counter('post', '/classes/Counter', { title, url, time: 1 })
                .then(response => response.json())
                .then(() => {
                  leancloudSelector(url).innerText = 1;
                })
                .catch(error => {
                  console.error('Failed to create', error);
                });
          }
        })
        .catch(error => {
          console.error('LeanCloud Counter Error', error);
        });
    }

    function showTime(Counter) {
      var visitors = document.querySelectorAll('.leancloud_visitors');
      var entries = [...visitors].map(element => {
        return decodeURI(element.id);
      });

      Counter('get', '/classes/Counter?where=' + encodeURIComponent(JSON.stringify({ url: { '$in': entries } })))
        .then(response => response.json())
        .then(({ results }) => {
          for (let url of entries) {
            let target = results.find(item => item.url === url);
            leancloudSelector(url).innerText = target ? target.time : 0;
          }
        })
        .catch(error => {
          console.error('LeanCloud Counter Error', error);
        });
    }

    let { app_id, app_key, server_url } = {"enable":true,"app_id":"UqIYufmNk2sE8MiBKKWc4Kja-gzGzoHsz","app_key":"iYpeWQcmNqpHwr2uVlnJYVji","server_url":null,"security":false};
    function fetchData(api_server) {
      var Counter = (method, url, data) => {
        return fetch(`${api_server}/1.1${url}`, {
          method,
          headers: {
            'X-LC-Id'     : app_id,
            'X-LC-Key'    : app_key,
            'Content-Type': 'application/json',
          },
          body: JSON.stringify(data)
        });
      };
      if (CONFIG.page.isPost) {
        if (CONFIG.hostname !== location.hostname) return;
        addCount(Counter);
      } else if (document.querySelectorAll('.post-title-link').length >= 1) {
        showTime(Counter);
      }
    }

    let api_server = app_id.slice(-9) !== '-MdYXbMMI' ? server_url : `https://${app_id.slice(0, 8).toLowerCase()}.api.lncldglobal.com`;

    if (api_server) {
      fetchData(api_server);
    } else {
      fetch('https://app-router.leancloud.cn/2/route?appId=' + app_id)
        .then(response => response.json())
        .then(({ api_server }) => {
          fetchData('https://' + api_server);
        });
    }
  })();
</script>


      </div>
      <div style="height: 200px">
      <script type="text/javascript" src="//rf.revolvermaps.com/0/0/8.js?i=5knf34gs07t&amp;m=0&amp;c=ff0000&amp;cr1=ffffff&amp;f=arial&amp;l=33" async="async"></script>
      </div>
    </footer>
  </div>

  
  <script src="/blog/lib/anime.min.js"></script>
  <script src="/blog/lib/velocity/velocity.min.js"></script>
  <script src="/blog/lib/velocity/velocity.ui.min.js"></script>

<script src="/blog/js/utils.js"></script>

<script src="/blog/js/motion.js"></script>


<script src="/blog/js/schemes/pisces.js"></script>


<script src="/blog/js/next-boot.js"></script>




  




  
<script src="/blog/js/local-search.js"></script>













  

  

</body>
</html>
