<!DOCTYPE html>
<html lang="zh-Hans">
<head>
  <meta charset="UTF-8">
<meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=2">
<meta name="theme-color" content="#222">
<meta name="generator" content="Hexo 5.4.0">
  <link rel="apple-touch-icon" sizes="180x180" href="/blog/images/apple-touch-icon-next.png">
  <link rel="icon" type="image/png" sizes="32x32" href="/blog/images/favicon-32x32-next.png">
  <link rel="icon" type="image/png" sizes="16x16" href="/blog/images/favicon-16x16-next.png">
  <link rel="mask-icon" href="/blog/images/logo.svg" color="#222">
  <meta name="google-site-verification" content="1osTo_gOe1SW9Hevqtz8vl8HfSgTO4DSMZ0DiMPMX6w">

<link rel="stylesheet" href="/blog/css/main.css">


<link rel="stylesheet" href="/blog/lib/font-awesome/css/all.min.css">

<script id="hexo-configurations">
    var NexT = window.NexT || {};
    var CONFIG = {"hostname":"wangc1106.github.io","root":"/blog/","scheme":"Gemini","version":"7.8.0","exturl":false,"sidebar":{"position":"left","display":"post","padding":18,"offset":12,"onmobile":false},"copycode":{"enable":false,"show_result":false,"style":null},"back2top":{"enable":true,"sidebar":true,"scrollpercent":"ture"},"bookmark":{"enable":false,"color":"#222","save":"auto"},"fancybox":false,"mediumzoom":false,"lazyload":false,"pangu":false,"comments":{"style":"tabs","active":null,"storage":true,"lazyload":false,"nav":null},"algolia":{"hits":{"per_page":10},"labels":{"input_placeholder":"Search for Posts","hits_empty":"We didn't find any results for the search: ${query}","hits_stats":"${hits} results found in ${time} ms"}},"localsearch":{"enable":true,"trigger":"auto","top_n_per_article":1,"unescape":false,"preload":false},"motion":{"enable":true,"async":false,"transition":{"post_block":"fadeIn","post_header":"slideDownIn","post_body":"slideDownIn","coll_header":"slideLeftIn","sidebar":"slideUpIn"}},"path":"search.xml"};
  </script>

  <meta name="description" content="辅助组装案例">
<meta property="og:type" content="article">
<meta property="og:title" content="HiFi-6-[HiFi-HiC-assembly-method]">
<meta property="og:url" content="https://wangc1106.github.io/blog/2022/03/14/2022-03-14-HiFi-6-HiC-assembly/index.html">
<meta property="og:site_name" content="Banner&#39;s Blog">
<meta property="og:description" content="辅助组装案例">
<meta property="og:locale">
<meta property="og:image" content="https://wangc1106.github.io/blog/2022/03/14/2022-03-14-HiFi-6-HiC-assembly/1.png">
<meta property="og:image" content="https://wangc1106.github.io/blog/2022/03/14/2022-03-14-HiFi-6-HiC-assembly/2.png">
<meta property="og:image" content="https://wangc1106.github.io/blog/2022/03/14/2022-03-14-HiFi-6-HiC-assembly/3.png">
<meta property="og:image" content="https://wangc1106.github.io/blog/2022/03/14/2022-03-14-HiFi-6-HiC-assembly/4.png">
<meta property="og:image" content="https://wangc1106.github.io/blog/2022/03/14/2022-03-14-HiFi-6-HiC-assembly/5.png">
<meta property="og:image" content="https://wangc1106.github.io/blog/2022/03/14/2022-03-14-HiFi-6-HiC-assembly/6.png">
<meta property="og:image" content="https://wangc1106.github.io/blog/2022/03/14/2022-03-14-HiFi-6-HiC-assembly/7.png">
<meta property="og:image" content="https://wangc1106.github.io/blog/2022/03/14/2022-03-14-HiFi-6-HiC-assembly/8.png">
<meta property="og:image" content="https://wangc1106.github.io/blog/2022/03/14/2022-03-14-HiFi-6-HiC-assembly/9.png">
<meta property="og:image" content="https://wangc1106.github.io/blog/2022/03/14/2022-03-14-HiFi-6-HiC-assembly/10.png">
<meta property="og:image" content="https://wangc1106.github.io/blog/2022/03/14/2022-03-14-HiFi-6-HiC-assembly/11.png">
<meta property="article:published_time" content="2022-03-14T08:09:34.778Z">
<meta property="article:modified_time" content="2022-03-15T14:19:04.334Z">
<meta property="article:author" content="Wang C">
<meta property="article:tag" content="denovo-III">
<meta name="twitter:card" content="summary">
<meta name="twitter:image" content="https://wangc1106.github.io/blog/2022/03/14/2022-03-14-HiFi-6-HiC-assembly/1.png">

<link rel="canonical" href="https://wangc1106.github.io/blog/2022/03/14/2022-03-14-HiFi-6-HiC-assembly/">


<script id="page-configurations">
  // https://hexo.io/docs/variables.html
  CONFIG.page = {
    sidebar: "",
    isHome : false,
    isPost : true,
    lang   : 'zh-Hans'
  };
</script>

  <title>HiFi-6-[HiFi-HiC-assembly-method] | Banner's Blog</title>
  






  <noscript>
  <style>
  .use-motion .brand,
  .use-motion .menu-item,
  .sidebar-inner,
  .use-motion .post-block,
  .use-motion .pagination,
  .use-motion .comments,
  .use-motion .post-header,
  .use-motion .post-body,
  .use-motion .collection-header { opacity: initial; }

  .use-motion .site-title,
  .use-motion .site-subtitle {
    opacity: initial;
    top: initial;
  }

  .use-motion .logo-line-before i { left: initial; }
  .use-motion .logo-line-after i { right: initial; }
  </style>
</noscript>

</head>

<body itemscope itemtype="http://schema.org/WebPage">
  <div class="container use-motion">
    <div class="headband"></div>

    <header class="header" itemscope itemtype="http://schema.org/WPHeader">
      <div class="header-inner"><div class="site-brand-container">
  <div class="site-nav-toggle">
    <div class="toggle" aria-label="Toggle navigation bar">
      <span class="toggle-line toggle-line-first"></span>
      <span class="toggle-line toggle-line-middle"></span>
      <span class="toggle-line toggle-line-last"></span>
    </div>
  </div>

  <div class="site-meta">

    <a href="/blog/" class="brand" rel="start">
      <span class="logo-line-before"><i></i></span>
      <h1 class="site-title">Banner's Blog</h1>
      <span class="logo-line-after"><i></i></span>
    </a>
      <p class="site-subtitle" itemprop="description">Dream big work hard</p>
  </div>

  <div class="site-nav-right">
    <div class="toggle popup-trigger">
        <i class="fa fa-search fa-fw fa-lg"></i>
    </div>
  </div>
</div>




<nav class="site-nav">
  <ul id="menu" class="main-menu menu">
        <li class="menu-item menu-item-home">

    <a href="/blog/" rel="section"><i class="home fa-fw"></i>Home</a>

  </li>
        <li class="menu-item menu-item-categories">

    <a href="/blog/categories/" rel="section"><i class="th fa-fw"></i>Categories</a>

  </li>
        <li class="menu-item menu-item-tags">

    <a href="/blog/tags/" rel="section"><i class="tags fa-fw"></i>Tags</a>

  </li>
        <li class="menu-item menu-item-about">

    <a href="https://wangc1106.github.io/" rel="section"><i class="user fa-fw"></i>About</a>

  </li>
      <li class="menu-item menu-item-search">
        <a role="button" class="popup-trigger"><i class="fa fa-search fa-fw"></i>Search
        </a>
      </li>
  </ul>
</nav>



  <div class="search-pop-overlay">
    <div class="popup search-popup">
        <div class="search-header">
  <span class="search-icon">
    <i class="fa fa-search"></i>
  </span>
  <div class="search-input-container">
    <input autocomplete="off" autocapitalize="off"
           placeholder="Searching..." spellcheck="false"
           type="search" class="search-input">
  </div>
  <span class="popup-btn-close">
    <i class="fa fa-times-circle"></i>
  </span>
</div>
<div id="search-result">
  <div id="no-result">
    <i class="fa fa-spinner fa-pulse fa-5x fa-fw"></i>
  </div>
</div>

    </div>
  </div>

</div>
    </header>

    


    <main class="main">
      <div class="main-inner">
        <div class="content-wrap">
          

          <div class="content post posts-expand">
            

    
  
  
  <article itemscope itemtype="http://schema.org/Article" class="post-block" lang="zh-Hans">
    <link itemprop="mainEntityOfPage" href="https://wangc1106.github.io/blog/2022/03/14/2022-03-14-HiFi-6-HiC-assembly/">

    <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
      <meta itemprop="image" content="/blog/images/banner.png">
      <meta itemprop="name" content="Wang C">
      <meta itemprop="description" content="Make more time">
    </span>

    <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
      <meta itemprop="name" content="Banner's Blog">
    </span>
      <header class="post-header">
        <h1 class="post-title" itemprop="name headline">
          HiFi-6-[HiFi-HiC-assembly-method]
        </h1>

        <div class="post-meta">
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-calendar"></i>
              </span>
              <span class="post-meta-item-text">Posted on</span>

              <time title="Created: 2022-03-14 16:09:34" itemprop="dateCreated datePublished" datetime="2022-03-14T16:09:34+08:00">2022-03-14</time>
            </span>
              <span class="post-meta-item">
                <span class="post-meta-item-icon">
                  <i class="far fa-calendar-check"></i>
                </span>
                <span class="post-meta-item-text">Edited on</span>
                <time title="Modified: 2022-03-15 22:19:04" itemprop="dateModified" datetime="2022-03-15T22:19:04+08:00">2022-03-15</time>
              </span>
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-folder"></i>
              </span>
              <span class="post-meta-item-text">In</span>
                <span itemprop="about" itemscope itemtype="http://schema.org/Thing">
                  <a href="/blog/categories/Software/" itemprop="url" rel="index"><span itemprop="name">Software</span></a>
                </span>
            </span>

          
            <span id="/blog/2022/03/14/2022-03-14-HiFi-6-HiC-assembly/" class="post-meta-item leancloud_visitors" data-flag-title="HiFi-6-[HiFi-HiC-assembly-method]" title="Views">
              <span class="post-meta-item-icon">
                <i class="fa fa-eye"></i>
              </span>
              <span class="post-meta-item-text">Views: </span>
              <span class="leancloud-visitors-count"></span>
            </span>
            <span class="post-meta-item" title="Symbols count in article">
              <span class="post-meta-item-icon">
                <i class="far fa-file-word"></i>
              </span>
              <span>48k</span>
            </span>
            <span class="post-meta-item" title="Reading time">
              <span class="post-meta-item-icon">
                <i class="far fa-clock"></i>
              </span>
              <span>44 mins.</span>
            </span>

        </div>
      </header>

    
    
    
    <div class="post-body" itemprop="articleBody">

      
        <p>辅助组装案例</p>
<span id="more"></span>

<h1 id="1-2016-05-np-碧冬茄"><a href="#1-2016-05-np-碧冬茄" class="headerlink" title="1. 2016.05-np 碧冬茄"></a>1. 2016.05-np 碧冬茄</h1><p><strong>Insight into the evolution of the Solanaceae from the parental genomes of Petunia hybrida</strong></p>
<h2 id="Genome-sequencing-assembly-and-annotation"><a href="#Genome-sequencing-assembly-and-annotation" class="headerlink" title="Genome sequencing, assembly and annotation."></a>Genome sequencing, assembly and annotation.</h2><p>Plants were grown and DNA was extracted following the methods described at Supplementary Note 1.</p>
<p>Illumina libraries with 0.17-, 0.35-, 0.5-, 0.8-, 1-, 2-, 5-, 8- and 15-kb inserts were sequenced at BGI-Shenzghen and University of Illinois, Roy J. Carver Biotechnology. PacBio P. axillaris DNA library was sequenced with P4/C2 chemistry.</p>
<p>Illumina reads were processed using <strong>Fastq-mcf</strong> (<strong>quality filtering</strong>; <a target="_blank" rel="noopener" href="https://code.google.com/p/eautils/wiki/FastqMcf">https://code.google.com/p/eautils/wiki/FastqMcf</a>), <strong>PRINSEQ</strong> (<strong>duplication filtering</strong>; http:// prinseq.sourceforge.net/) and <strong>Musket</strong> (<strong>error correction</strong>; <a target="_blank" rel="noopener" href="http://musket.sourceforge/">http://musket.sourceforge</a>. net/). <strong>Pacbio reads</strong> were processed using the <strong>SMRT Analysis pipeline</strong> (v.2.0.1; <a target="_blank" rel="noopener" href="https://github.com/PacificBiosciences/SMRT-Analysis">https://github.com/PacificBiosciences/SMRT-Analysis</a>).</p>
<p>Both genomes were assembled with <strong>SOAPdenovo</strong> with different k-mer sizes. For both genomes, <strong>k-mer = 79</strong> showed the best statistics. Gaps between contigs were completed using <strong>GapCloser</strong> . Additionally for P. axillaris, <strong>PacBio reads</strong> were integrated in <strong>four different steps</strong>: (1) <strong>Rescaffolding</strong> of the <strong>Illumina contigs</strong> using the <strong>PacBio reads</strong> and the <strong>AHA assembler</strong>; (2) <strong>Gap filling</strong> using <strong>PBJelly</strong>; (3) <strong>Rescaffolding</strong> using the Illumina <strong>pair data</strong> and <strong>SSPACE</strong>; (4) Last round of <strong>gap filling</strong> using <strong>PBJelly</strong> . </p>
<p><strong>Genome size estimation</strong> was performed through the k-mers abundance distribution (<strong>k-mer = 31</strong>). <strong>Heterozygosity</strong> was <strong>estimated</strong> mapping the Illumina reads to the assemblies using <strong>Bowtie2</strong>, calling SNPs using <strong>FreeBayes</strong> and  annotating the SNPs using <strong>SnpEff</strong> .</p>
<p>The genome structural annotation was performed using <strong>Maker-P</strong>: (1) <strong>SNAP</strong> and <strong>Augustus</strong> as ab initio gene predictors; (2) Exonerate as experimental based predictor with 454 and Illumina RNASeq reads and protein sequences from different protein datasets. RNAseq Illumina data was mapped using <strong>Tophat2</strong> . tRNAs were annotated using <strong>tRNAscan</strong> (<a target="_blank" rel="noopener" href="http://lowelab.ucsc.edu/tRNAscan-SE/">http://lowelab.ucsc.edu/tRNAscan-SE/</a>).</p>
<p>The <strong>gene functional annotation</strong> was performed by <strong>sequence homology search</strong> with different protein datasets using <strong>BlastP</strong> and protein domains search using <strong>InterProScan</strong> . Functional annotations were integrated using <strong>AHRD</strong> (<a target="_blank" rel="noopener" href="https://github.com/groupschoof/AHRD">https://github.com/groupschoof/AHRD</a>). See Supplementary Note 1.</p>
<h2 id="Repetitive-elements-analysis"><a href="#Repetitive-elements-analysis" class="headerlink" title="Repetitive elements analysis"></a>Repetitive elements analysis</h2><p>Repeat annotation was performed using RepeatModeler (v1.0.8; <a target="_blank" rel="noopener" href="http://www.repeatmasker.org/RepeatModeler.html">http://www.repeatmasker.org/RepeatModeler.html</a>), RepeatMasker (v4.0.5; <a target="_blank" rel="noopener" href="http://www.repeatmasker.org/">http://www.repeatmasker.org</a>) with the repeat database Repbase (release 20140131; <a target="_blank" rel="noopener" href="http://www.girinst.org/repbase/">http://www.girinst.org/repbase/</a>) and Geneious (v7.1.4; <a target="_blank" rel="noopener" href="http://www.geneious.com/">http://www.geneious.com</a>). Identification of PVCV-like and EPRV elements was performed using BlastN and TBlastN . The identified sequences were aligned with ClustalW (MEGA5 package;<a target="_blank" rel="noopener" href="http://www.megasoftware.net/">http://www.megasoftware.net/</a>) and then manually curated. RepeatExplorer (<a target="_blank" rel="noopener" href="http://www.repeatexplorer.org/">http://www.repeatexplorer.org/</a>) and other methods were used to extend the analysis to unassembled repeats. Fluorescent in situ hybridization was performed in root tips from young P. axillaris and P. inflata plants for 5S rDNA and three PVCV viral probes following the procedure described in Supplementary Note 2.</p>
<p>The detection of dTph1 loci in P. hybrida W138 was performed through a BLAST40 search of the P. axillaris and P. inflata dTph1 elements including the 500 bp of flanking sequence against the TFS W138 collection. Polymorphisms found in the genomic flanking regions were used to identify the species of origin. dTph1 elements were identified in a P. axillaris population using a modification of the methodology described in Supplementary Note 3.</p>
<p>##Whole-genome duplication, tandem duplications and gene family analysis</p>
<p>Whole-genome collinear analysis was performed using SynMap and microsynteny analysis were performed using GEvo in the comparative genomics platform, CoGe .See Supplementary Note 5.</p>
<p>The gene family analysis included Solanum lycopersicum, S. tuberosum, Nicotiana benthamiana and Arabidopsis thaliana protein sets using BlastP (v2.2.27) on an all-versus-all comparison and grouping the genes into families with OrthoMCL, v2.0.8. See Supplementary Note 4.</p>
<h2 id="Gene-data-mining"><a href="#Gene-data-mining" class="headerlink" title="Gene data mining"></a>Gene data mining</h2><p>The specific identification of genes for P. axillaris and P. inflata genomes for colour and scent, root-specific pathways, self-incompatibility and circadian clock was performed through a BlastN/BlastP sequence homology search. Blast GUI, JBrowser (<a target="_blank" rel="noopener" href="http://jbrowse.org/">http://jbrowse.org/</a>) and WebApollo (<a target="_blank" rel="noopener" href="http://genomearchitect/">http://genomearchitect</a>. org/) were installed in a server to search and manually curate the gene structures of the identified genes. See Supplementary Notes 7, 8 and 10–12.</p>
<h1 id="2-2017-04-ng-山羊🐐"><a href="#2-2017-04-ng-山羊🐐" class="headerlink" title="2. 2017.04-ng 山羊🐐"></a>2. 2017.04-ng 山羊🐐</h1><p>Published in final edited form as:Nat Genet. 2017 April ; 49(4): 643–650. doi:10.1038/ng.3802.</p>
<p><strong>Single-molecule sequencing and chromatin conformation capture enable</strong> <strong>de novo</strong> <strong>reference assembly of the domestic goat genome</strong></p>
<p>​        首次在人之外的物种中直接de novo组装出染色体水平的基因组。利用<font color="blue"><strong>Pacbio+BioNano+Hi-C+Illumina</strong> 多层级</font>的策略，组装了一只山羊(Capra hircus)的基因组，获得<font color="blue"><strong>31</strong>个<strong>scaffold</strong>，<strong>663</strong>个<strong>gap</strong>区，scaffold <strong>N50</strong> 惊人地达到<strong>87M</strong>，几乎<strong>one scaffold one chromosome</strong></font>，并且成功地组装出<font color="blue"><strong>免疫基因区</strong>和<strong>大部分重复序列家族</strong></font>。</p>
<p>​        <font color="red">人的GRCh38版本有<strong>24</strong>个<strong>scaffold</strong>，<strong>169</strong>个<strong>unplacedscaffold</strong>，<strong>832个gap</strong>区</font>。</p>
<p>​        从<strong>96头</strong>山羊(<strong>6个品种</strong>)中，利用Illumina的 Caprine53K <font color="blue"><strong>SNP 芯片筛选</strong></font>出<font color="blue"><strong>基因型纯和度最高</strong></font>的个体用来<font color="blue"><strong>组装</strong></font>(San Clemente breed)。</p>
<h2 id="data"><a href="#data" class="headerlink" title="data"></a>data</h2><ol>
<li><p>Pacbio数据</p>
<p>一共产生465个SMRT cell的数据。其中使用P5-C3试剂的311个cell;使用P4-C2试剂的142个cell，XL-C2试剂的12个cell。数据量194G，覆盖深度达<strong>69X</strong>，<strong>subread</strong>的<strong>平均</strong>长度为<strong>5,110bp</strong>。</p>
</li>
<li><p>BioNano 数据</p>
<p>由于采样所用山羊意外死亡，其DNA不能满足Irsy optical mapping 测序需求，只好用其<strong>雄性后代</strong>采样。Optical map一共产生256Gb数据，覆盖<strong>98 X</strong>。</p>
</li>
<li><p>Hi-C数据</p>
<p>采样动物是最初那只山羊。Hi-C建库，序列用物理方法打断成300-500bp长度，PE101测序，产生<strong>115M reads</strong>的数据量。</p>
</li>
<li><p>Illumina 数据</p>
<p>采样最初那只山羊，PE251建库测序，获得<strong>23X</strong>的数据覆盖，用来做最后的<font color="red"><strong>错误校正</strong></font></p>
</li>
</ol>
<h2 id="组装策略及组装结果"><a href="#组装策略及组装结果" class="headerlink" title="组装策略及组装结果"></a><strong>组装策略及组装结果</strong></h2><ul>
<li><p>先用Pacbio的 long-read 数据构建contig， 使用<font color="blue"><strong>Celera Assembler</strong> PacBio corrected Reads 流程</font>组装。获得<font color="blue"><strong>3,074</strong>个contig(2.63G)，<strong>N50 4.159M</strong>b</font>。</p>
</li>
<li><p>接下来采用<font color="blue"><strong>Irys optical mapping</strong>数据构建<strong>scaffold</strong></font>，使用软件为<font color="blue"><strong>IrysView</strong></font>. 产生<font color="blue"><strong>842</strong>个scaffold，scaffold N50为<strong>13.4M</strong>b，contig N50为10.858M，最长scaffold为66.728Mb</font>。 利用<font color="red"><strong>radiation hybrid(RH) map比较</strong>发现</font>，<font color="blue">PacBio-Irys 联合</font>的方法已经<font color="blue">完整组装出<strong>20号染色体</strong></font>。</p>
</li>
<li><p>接下来用Hi-C数据，调用用<font color="red"><strong>Lachesis软件包</strong></font>，结合PacBio-Irys的结果，即整合的PacBio-Irys-PGA (PBIP)方法，获得较完美的组装结果。<font color="blue">Scaffold N50</font> 达到惊人的<font color="blue"><strong>87.347Mb</strong></font>(远超今年尖吻鲈，大猩猩N50 20Mb的高起点)，总共获得<font color="blue"><strong>31条</strong>scaffold</font>，(<font color="red">31,261条&lt;50kbp的**degenerate scaffold**不计入</font> )。</p>
</li>
<li><p>最后利用Illumina数据做<font color="red"><strong>一致性校正</strong>和最后的<strong>补洞</strong></font>，最后使得contig的数目下降到680条，产生的gap数目为663个。</p>
</li>
<li><p>利用Kraken v0.10.5 <font color="red">去除有<strong>病毒和细菌污染</strong>的序列</font>，去掉有NCBI vector污染的序列。获得最终的基因组 版本<strong>ARS1</strong>。</p>
</li>
</ul>
<h2 id="组装评估及比较分析"><a href="#组装评估及比较分析" class="headerlink" title="组装评估及比较分析"></a><strong>组装评估及比较分析</strong></h2><p>由于山羊之前已经产生过两个基因组版本<strong>CHIR_1.0、CHIR_2.0</strong>，可以用来做比较 分析，并且有RH mapping数据用来评估、校正ARS1的组装版本。 <strong>通过各种参数比较发现，ARS1均优于CHIR_2.0版本。</strong></p>
<p>1、通过比对发现，CHIR_2.0 比CHIR_1.0有更少的putative deletion(2,735 vs 1 0,256)和duplication(115 vs 290)。而2.0版本相较1.0有更多的inversion(215 v s 4)。通过比较，发现ARS1版本有大幅度地提升，较2.0版本少4倍的deletion， 少50倍inversion。</p>
<p>2、CHIR_2.0能填补CHIR_1.0 94.6%的gap，而剩下的那些gap通过分析ARS1得知，是CHIR_1.0的组装错误，而RH数据也支持了这一论断。 3、BUSCO是利用单拷贝ortholog来评估组装的可靠性，和CHIR_2.0相比，ARS1有更高的BUSCO score。</p>
<p>4、山羊52k SNP 芯片中，有1,723个SNP探针目前只能定位在CHIR_2.0组装版本 中的unplaced contigs上。而ARS1版本能把其中的90%(1,552/1,723)以上的S NP探针定位在染色体上。另外发现有26个low call-rate 的SNPmarker在ARS1组装版本上定位比较模棱两可。这也就解释了为什么这些maker在芯片上有较差的call -rate. </p>
<p>5、通过比较发现，CHIR_2.0版本中的3,495个内含子或外显子有gap的基因，在ARS1中都得到补全。同时也确认1,926个预测的外显子在CHIR_1.0或者2.0中有gap， 但在ARS1中得到修复。 </p>
<p>6、由于免疫基因区高度多态性和重复性，用二代测序数据很难组装起来，但是通过 分析发现，ARS1版本就很好地把LRC和NKC基因(免疫功能相关基因)定位在一个独立的常染色体scaffold上。 </p>
<p>7、通过和一代、二代组装策略相比，ARS1版本在重复序列的组装上获得了极大进步。对某些异染色体或异染色质的区域也获得了较好的组装覆盖。比如，在6条常 染色体上组装出&gt;5kbp的端粒序列。通过分析发现， 15条染色体scaffold，在着丝粒区域组装出大于2kbp长度的重复区，而其中的7条，更是装出来8kbp以上的重复区。更令人惊讶的是，19号和23号染色体都组装出高度重复的着丝粒和端粒区域， 贯通了有结构性异染色质区的染色体。 </p>
<p>8、在ARS1版本中鉴定出大于12kbp的重复模式序列多达105条。对于重复序列家 族的鉴定，在ARS1中获得的接近全长的BovB LINE 序列比CHIR_2.0多60%以上。 在CHIR_2.0版本里，被ARS1成功补全的gap中的43.6%与BovB重复序列一致(长 度大于3.5kbp)，即意味着二代测序的gap区大多都是重复序列，用三代就能较容易测通。 </p>
<p>9、关于性染色体的分析。通过比对发现，两个不同的scaffold比对到X染色体的不同但连续的区域，占预期X染色体大小(150Mb)的86%左右。通过和自己的、跨物种牛的Y染色体比对，最终确定出10Mb区域的序列，占Y染色体预估大小的50% 左右。通过比对牛和羊的Y染色体上的基因，发现在目前的scaffold中能找到16%， 而在先前过滤掉的degenerate contig中能找到84%的基因。考虑到Y染色体的异染 色质属性及X,Y染色体的拟常染色区域经常发现交换，实在太过复杂，超出目前的组装和认知水平，表示实在无能为力。</p>
<p>10、利用之前RH mapping data，对组装的大部分环节进行了校正及评估。比如多次辅助解决scaffold conflict的问题。</p>
<h2 id="基因组注释"><a href="#基因组注释" class="headerlink" title="基因组注释"></a><strong>基因组注释</strong></h2><p>多种方法结合注释基因集。<br>1.<strong>RNA-seq的方法:</strong> 6个组织(大多和脑组织相关)RNA-seq测序、13个SRA下载 数据，利用stringtie、cufflinks和Trinity(基于无参组装)。最后用PASA软件整合在一起;</p>
<p>2.<strong>用exonerate和tblastn 软件</strong>比对到几个近缘物种的Ensembl基因集上，获得同 源预测基因集;</p>
<p>3.<strong>用Braker1做Ab initio 预测</strong>;</p>
<p>4.<strong>CHIR_1.0 版本的注释基因集</strong>; 最后用<strong>EVM+PASA</strong>把以上4种数据整合成一个最终的基因集(设置的权重为RNAseq &gt; cDNA/protein &gt; ab initio gene predictions)。</p>
<h2 id="启示录"><a href="#启示录" class="headerlink" title="启示录"></a><strong>启示录</strong></h2><p>1、Pacbio RSII是一个较为理想组装平台，产生平均14Kbp左右的读长，最高到60 Kbp，可以组装出较为理想的基因组版本。由于三代的长读长优势，同时可以较好地组装出基因组中大部分高度重复区(主要集中在着丝粒和端粒区域);</p>
<p>2、BioNano 的Irys optical mapping 是一种非常有效的，成本比较适中的构建scaffold的平台。和Hi-C比优势在于定位错误较少，缺点是对N50的提高并不是特别显著，一般在2倍左右。如果正反链的Nt.BsqI酶切位点离得比较近，常常导致双链 断裂，限制了Optical Map scaffold的大小。所以，BioNano对较长的scaffold效果比较好。由于Optical map和pacbio产生的错误特征并不一样，两者结合，可以 相互校正，获得较好的组装指标;</p>
<p>3、Hi-C是目前非常火爆的一个技术，在多个领域都有很好的应用。在组装方面， 它能获得染色体级别的scaffold，但是也容易获得较高的contig定向错误。这也是 为什么文章中，先用optical mapping，后用Hi-C做scaffolding的重要原因之一。 作者建议在以后的实验中，可以选择较短的识别位点的限制内切酶(或者DNase Hi -C)来提高Hi-C交联的密度，降低定向错误。</p>
<p>4、利用同样的方法和平台，做类似的基因组项目，用Pacbio平台和scaffolding平 台，目前大约花费在$100,000上下,而一般的短reads平台测序加scaffolding 平台，只需要其三分之一的价格。但是用Pacbio平台却获得了高质量的、连续性好的基因组序列。目前Pacbio 新一代的sequel平台已经推出，单个cell的产量比RSII提 高7倍以上，测序成本将会得到大幅度降低，相信会成为未来基因组研究的主流工具之一。</p>
<p>5、本文的研究策略确实会启迪后来的研究者，为de novo组装树立新标杆。Pacbio 测序技术联合多种scaffolding 平台(比如BioNano、Hi-C，mating-pairedlibr ary，10X genomics)的使用， 会引领de novo 组装进入 near finished genome 或者finished genome的新时代。</p>
<p>6、但是，尽管做了很多努力，ARS1仍然是一个单体型混合型的组装版本。在未来，可以利用single molecule和Hi-C技术获得单体型分期(haplotype phasing) 的参考基因组。对于组成型异染色质区，尤其是着丝粒和端粒部分，仍然是组装的盲区，即使在人的基因组中这些区域也没有被完整组装出来。</p>
<h1 id="3-2020-09-cell-大豆泛基因组"><a href="#3-2020-09-cell-大豆泛基因组" class="headerlink" title="3. 2020.09-cell 大豆泛基因组"></a>3. 2020.09-cell 大豆泛基因组</h1><p>Pan-Genome of Wild and Cultivated Soybeans</p>
<p>Soybean is one of the most important vegetable oil and protein feed crops. To capture the entire genomic diversity, it is needed to construct a complete high-quality pan-genome from diverse soybean accessions. In this study, we performed individual de novo genome assemblies for <strong>26 representative soybeans</strong> that were selected from <strong>2,898 deeply sequenced accessions</strong>. Using these assembled genomes together with <strong>three previously reported genomes</strong>, we constructed a <strong>graph-based genome</strong> and performed pan-genome analysis, which identified <strong>numerous genetic variations that cannot be detected by direct mapping of short sequence reads onto a single reference genome</strong>. The <strong>structural variations</strong> from the 2,898 accessions that were <strong>genotyped</strong> based on the graph-based genome and the <strong>RNA-seq data</strong> from the representative 26 accessions helped to <strong>link genetic variations to candidate genes</strong> that are responsible for important traits. This pan-genome resource will promote evolutionary and functional genomics studies in soybean.</p>
<h2 id="SNP-Calling-and-Phylogenetic-Analyses"><a href="#SNP-Calling-and-Phylogenetic-Analyses" class="headerlink" title="SNP Calling and Phylogenetic Analyses"></a>SNP Calling and Phylogenetic Analyses</h2><h2 id="Genome-Assembly"><a href="#Genome-Assembly" class="headerlink" title="Genome Assembly"></a>Genome Assembly</h2><p><em>De novo</em> assembly was conducted referring to a <font color="blue"><strong>reported pipeline</strong></font> (Du and Liang, 2019; Shen et al., 2019). In brief, <font color="red"><strong>Canu</strong></font> (Koren et al., 2017) v1.7.1 was used to assemble PacBio subreads to PacBio contigs, after which <font color="red"><strong>HiSeq</strong></font> reads were used for <font color="red"><strong>error correction</strong></font>. <strong>Bio-nano optical maps</strong> were assembled into consensus physical maps by <strong>BioNano Solve</strong> v3.0.1 (<a target="_blank" rel="noopener" href="https://bionanogenomics.com/">https://bionanogenomics.com/</a>, Solve_06082017Rel). Then <font color="red"><strong>HERA</strong></font> (Du and Liang, 2019) was used to combine PacBio contigs and Bionano based physical maps to <font color="blue"><strong>PacBio-BioNano hybrid scaffolds</strong></font>. To <font color="red"><strong>anchor</strong></font> hybrid scaffolds <font color="blue">into <strong>chromosomes</strong></font>, the <font color="blue"><strong>Hi-C</strong></font> sequencing data were <font color="red"><strong>aligned into scaffolds</strong></font> by <font color="red"><strong>Juicer</strong></font> (Durand et al., 2016) v1.5 and <font color="red"><strong>3D-DNA</strong></font> (Dudchenko et al., 2017).</p>
<h2 id="Repeat-Analysis-and-Gene-Annotation"><a href="#Repeat-Analysis-and-Gene-Annotation" class="headerlink" title="Repeat Analysis and Gene Annotation"></a>Repeat Analysis and Gene Annotation</h2><h2 id="Synteny-Analysis"><a href="#Synteny-Analysis" class="headerlink" title="Synteny Analysis"></a>Synteny Analysis</h2><p>Structural Variation Identification</p>
<p>Genetic Variation Analysis</p>
<p>Gene and miRNA Expression</p>
<p>Core and Dispensable Gene Family Clustering</p>
<p>CHS gene unit identification</p>
<p>Gene Fusion Event Identification</p>
<p>QUANTIFICATION AND STATISTICAL ANALYSIS</p>
<h1 id="4-2019-nc-大豆泛基因组-组装参考1"><a href="#4-2019-nc-大豆泛基因组-组装参考1" class="headerlink" title="4. 2019-nc 大豆泛基因组-组装参考1"></a>4. 2019-nc 大豆泛基因组-组装参考1</h1><p>HERA</p>
<h1 id="5-2019-09-scienceChina-大豆泛基因组-组装参考2"><a href="#5-2019-09-scienceChina-大豆泛基因组-组装参考2" class="headerlink" title="5. 2019.09-scienceChina 大豆泛基因组-组装参考2"></a>5. 2019.09-scienceChina 大豆泛基因组-组装参考2</h1><p><img src="/blog/2022/03/14/2022-03-14-HiFi-6-HiC-assembly/1.png" alt="1"></p>
<p>For genome assembly, we sequenced additional 40 Gb SMRT reads from 5 cells for this update. Therefore, a total of <strong>120× PacBio reads</strong>, <strong>365× Bionano optical maps</strong> marked by BssSI, <strong>275× Bionano optical maps</strong> marked by BspQI, <strong>45× HiSeq</strong> reads and <strong>125×</strong> chromosome conformation capture sequencing (<strong>Hi-C</strong>) reads were used for the new genome assembly (Figure 1A). To make best of the sequencing data, we also <font color="red"><strong>adopted a new genome assembly pipeline</strong> by <strong>adding HERA</strong></font> (Du and Liang, 2018) to improve the sequence contiguity and reduce errors by accurately assembling the repetitive genome regions (Figure 1A). Briefly, <font color="blue">the whole assembly process <strong>differed from</strong> previous pipeline</font> at </p>
<p>(1) using <font color="red"><strong>CANU</strong></font> (v1.7.1) to replace Smrtmake for assembling PacBio subreads to PacBio contigs; </p>
<p>(2) using <font color="red"><strong>HERA</strong></font> to generate longer contigs; </p>
<p>(3) using <font color="red"><strong>Juicer and 3D-DNA to replace HiC-Pro and LACHESIS</strong></font> to anchor the hybrid scaffolds into chromosomes with Hi-C reads.</p>
<h1 id="6-2020-05-nc-同源四倍体紫花苜蓿"><a href="#6-2020-05-nc-同源四倍体紫花苜蓿" class="headerlink" title="6. 2020.05-nc 同源四倍体紫花苜蓿"></a>6. 2020.05-nc 同源四倍体紫花苜蓿</h1><p>HiFi助力同源四倍体紫花苜蓿基因组组装，NC</p>
<blockquote>
<p>Allele-aware chromosome-level genome assembly and efficient transgene-free genome editing for the autotetraploid cultivated alfalfa</p>
</blockquote>
<p>（1）使用Canu默认参数，利用CCS clean reads组装contigs。组装得到的Contig N50值为459kb，总长度为3.15GB。</p>
<blockquote>
<p>We assembled contigs from CCS clean reads using Canu, with default parameters. The N50 values of the contig sets were 459 kb, with total lengths 3154 Mb.</p>
</blockquote>
<p>（2）使用HiC-Pro将Hi-C reads与contigs 进行比对，产生比对BAM文件。</p>
<blockquote>
<p>Hi-C reads were aligned to contigs using <a target="_blank" rel="noopener" href="https://pubmed.ncbi.nlm.nih.gov/26619908/">HiC-Pro(2015)</a>, yielding an alignment BAM file.</p>
</blockquote>
<p>（3）使用注释的蒺藜苜蓿蛋白作为参考，完全基于同源的策略注释contigs。对138,729个同源基因进行了结构注释。用MCscan用于鉴定contigs和参考基因组之间的共线性。显示紫花苜蓿和蒺藜苜蓿之间的高共线性。【下载父母本petunia ref】</p>
<blockquote>
<p>Contigs were annotated with a <font color="red"><strong>solely homology-based strategy</strong></font>, using annotated Medicago truncatula proteins as references. 138,729 homologous genes were structurally annotated. <font color="red"><strong>MCscan</strong> in <strong>Jcvi</strong></font> (<a target="_blank" rel="noopener" href="https://zenodo.org/record/31631#.XpkUyTOeask">https://zenodo.org/record/31631#.XpkUyTOeask</a>) was used to identify <font color="red"><strong>synteny blocks</strong></font> between <strong>contigs and the reference</strong> genome. <font color="red"><strong>Contigs syntenic</strong> to M. truncatula were <strong>stacked and aligned</strong> to M. truncatula chromosomes</font>. The syntenic contigs are summarized in Supplementary Table 4.</p>
</blockquote>
<p>（4）使用内部脚本处理BAM文件，去除等位基因contigs之间的links。使用ALLHiC软件，<font color="red">提取、聚类和重排Contigs</font> (Contigs syntenic与蒺藜苜蓿染色体一致)，得到原始的scaffolds。</p>
<blockquote>
<p>An <strong>in-house script</strong> was used to <font color="red">prune the BAM</font> file and <font color="red">discard links between allelic contigs</font>. Contigs syntenic to one chromosome of M. truncatula, e.g., chr1, were extracted, sub-clustered and reordered using <font color="red">ALLHiC</font>, yielding a raw scaffold set.</p>
</blockquote>
<p>（5）<a target="_blank" rel="noopener" href="https://pubmed.ncbi.nlm.nih.gov/27467250/">Juicebox</a>用于以图形和交互方式微调组装的scaffolds。剪裁了40个总长度达1800Mb的scaffolds 。</p>
<blockquote>
<p>Juicebox45 was used for fine-tuning assembled scaffolds in a graphic and inter-active fashion. Forty scaffolds with a total length of 1800 Mb were cropped (Supplementary Table 5).</p>
</blockquote>
<p>（6） 基于组装的scaffold，通过Hi-C数据，每个unplaced contig被分配到互作最强的那些contig cluster里。</p>
<blockquote>
<p>Based on this scaffold assembly, each unplaced contig was assigned to the contig cluster, to which the contig was most connected by Hi-C data.</p>
</blockquote>
<p>（7）使用ALLHiC对那些contig clusters再次进行重排和构建scaffold。</p>
<blockquote>
<p>Those contig clusters were reordered and scaffolded using ALLHiC. </p>
</blockquote>
<p>（8）使用Juicebox对scaffolds进行微调，并从scaffolds上去除不一致的contigs，产生最终的染色体基因组，其包含32条染色体(8个同源组，每个组中有4个等位基因染色体)，总长度为2738Mb，和419Mb未挂载到染色体水平的序列。</p>
<blockquote>
<p>Using Juicebox, scaffolds were fine-tuned and discordant contigs were removed from scaffolds, and the final chromosome assembly was generated, containing 32 chromosomes with a total length of 2738 Mb (Supplementary Table 6).</p>
</blockquote>
<h1 id="7-2021-11-gb-椰子🥥"><a href="#7-2021-11-gb-椰子🥥" class="headerlink" title="7. 2021.11-gb 椰子🥥"></a>7. 2021.11-gb 椰子🥥</h1><p>High-quality reference genome sequences of two coconut cultivars provide insights into evolution of monocot chromosomes and differentiation of fiber content and plant height</p>
<p>该研究绘制两个参考级别椰子基因组。揭示单子叶植物染色体的进化过程和椰子染色体的形成过程，分析高矮两种椰子耐盐，株高，纤维和脂质含量等关键性状的遗传差异，同时发现椰子进化过程中约在400万年前自然发生的绿色革命。高椰子2.39G矮椰子2.40G。基于椰子，紫萍，菠萝等基因组的共线性关系，重新构建了含有10条染色体的古单子叶植物基因组核型，并对基因组核型的进化路线进行推断。进一步利用叶间距表型进行全基因组关联分析，鉴定出12号染色体上GA20ox基因与椰子株高显著相关，并存在拷贝数变异；整合多种手段验证HA20ox以及赤霉素代谢途径在高矮种的差异并导致株高差异。400万年前，绿色革命。</p>
<h2 id="Hi-C-based-genome-assembly"><a href="#Hi-C-based-genome-assembly" class="headerlink" title="Hi-C-based genome assembly"></a>Hi-C-based genome assembly</h2><p>Both Cn. tall and Cn. dwarf genomes were assembled de novo using SMARTdenovo (<a target="_blank" rel="noopener" href="https://github.com/ruanjue/smartdenovo">https://github.com/ruanjue/smartdenovo</a>) [-k 19 -J 3000] based on <font color="red"><strong>Nanopore long reads</strong></font> <strong>corrected</strong> with <font color="red"><strong>NextDenovo</strong></font> (<a target="_blank" rel="noopener" href="https://github.com/Nextomics/NextDenovo">https://github.com/Nextomics/NextDenovo</a>). Raw contigs were <font color="blue"><strong>polished</strong></font> using <font color="blue"><strong>Nanopore long reads</strong></font> and <font color="blue"><strong>Illumina short reads</strong></font> with <font color="red"><strong>Next-Polish</strong></font> (v1.2.2) [49]. Then polished contigs were <font color="blue"><strong>anchored</strong></font> into chromosomes by using <font color="red"><strong>HiC-Pro (v2.11.1)</strong></font> [50] and <font color="red"><strong>LACHESIS</strong></font>[51]. Finally, the assembled genomes were <font color="blue"><strong>manual correction</strong></font> with <font color="red"><strong>Juicebox</strong></font> (v1.11.08).</p>
<p>To assess the quality of assembled genomes, raw Illumina <font color="blue"><strong>paired-end</strong> reads and <strong>RNA-seq</strong></font> reads from multiple tissues were <font color="blue"><strong>mapping</strong></font> to the genomes using <font color="red"><strong>BWA (v0.7.17-r1188)</strong></font> [52] and <font color="red"><strong>HISAT2 (v2.1.0)</strong></font> [53], respectively. In addition, Benchmarking Universal Single-Copy Orthologs (<font color="red"><strong>BUSCO</strong></font>) was also used to assess the genome completeness base on Embryophyta Plant database (odb10) [54].</p>
<h1 id="8-2021-11-nc-金粟兰基因组"><a href="#8-2021-11-nc-金粟兰基因组" class="headerlink" title="8. 2021.11-nc 金粟兰基因组"></a>8. 2021.11-nc 金粟兰基因组</h1><p>填补核心被子植物最后主要分支-金粟兰目的基因组信息，提供被子植物早期演化证据。刘建全团队致力于解析类群演化，探究不同物种间分化适应遗传机制。</p>
<p>被子植物（Angiosperms or flowering plants）是地球上分布范围最广、多样化最高、适应性最强的陆生植物类群，极大地影响了其他生物类群的演化进程，然而，由于被子植物在早期经历了快速的辐射进化，其内部演化关系迟迟未被解决，这一问题也被称为达尔文的“恼人之谜”。经过前人长时期的努力，目前被子植物被划分为：无油樟目、睡莲目、木兰藤目和核心被子植物（Mesangiospermae）。无油樟目、睡莲目、木兰藤目共同被称为ANA grade，是被子植物的基部类群，系统发育关系相对稳定，而核心被子植物包含了~99.95%的被子植物，又可被划分为5大类群：双子叶、单子叶、木兰类、金鱼藻目和金粟兰目，它们之间的演化关系一直存在争议。近年来，随着各类群代表物种全基因组测序相继完成，其各自的进化历程均有了坚实而可靠的证据，刘建全团队在2020年首次报道了金鱼藻和芡实的基因组，证明了核心被子植物内部存在广泛的不完全谱系分选和杂交事件是导致树形冲突的主要原因。但是金粟兰目一直未有基因组报道，其演化历程以及整个核心被子植物的演化过程还有待深入分析。为此该团队再次利用Nanopore、Illumina和Hi-C技术，构建了第一个金粟兰目植物（四川金粟兰，<em>Chloranthus sessilifolius</em>）的高质量染色体级别基因组，Contig N50高达53.74 Mb（图1），使用多种分析方法对上述问题进行了深入的阐述。</p>
<p>通过使用多种系统发育基因组学分析，包括单拷贝基因、低拷贝基因和共线性基因数据集，进行了串联树和溯祖树的构建，同时评估了长枝吸引、物种选择和同源聚类方法等的影响，使用多套数据集合和多种分析方法，获得了高可信度的拓扑结构，即金粟兰目和木兰类具有最近的亲缘关系，且他们一起与金鱼藻目+双子叶植物构成姊妹关系，而单子叶植物是其它所有核心被子植物的姊妹枝（图2）。</p>
<p>同时，作者也发现了大量的基因树或叶绿体树与物种树不一致的情况，因此着重评估了不完全谱系分选（ILS）和杂交事件对此冲突的贡献。通过PhyloNetworks分析鉴定到了三次可能的杂交事件，可以解析部分的树形冲突。然而值得注意的是五大类群之间的分化时间仅有23个百万年（158~135 Mya），如此短的分化时间，ILS可能发挥了更为主要的作用。使用theta参数对每个内部分支发生ILS的概率进行评估发现，除了木兰类和金粟兰目的姊妹关系较为稳定外，其他类群间都具有较高的theta值，而金鱼藻目+双子叶的祖先分枝具有较高的tehta值，均表明了ILS在核心被子植物内部的演化中发挥了重要作用（图3 a和b）。基于Phyabse和DendroPy进行树形模拟表明，当存在ILS时模拟的树形和真实树形具有高度一致性（R2＞0.99），而不存在ILS的模拟树形与真实树形则具有较低的相关性，进一步表明了ILS在核心被子植物的快速辐射中发挥了重要作用（图3 c-f）。进一步对树形异质性进行区分表明，ILS可基本解释金鱼藻目+双子叶的祖先枝上的树形冲突，也就是金鱼藻目+双子叶、木兰类+金粟兰和单子叶之间的冲突，而其他核心被子植物类群间的冲突还有其他因素（例如杂交）的参与。</p>
<p>此外，本研究还鉴定了四川金粟兰独有的一次WGD事件，同时5大核心被子植物分枝的加倍事件都不共享；鉴定了花发育相关的基因家族，解析了其特有花器官的可能形成原因；鉴定了萜类合成和木质部形成相关的基因家族，为今后研究其简单木质部形成和萜类物种合成通路演化提供了基础。</p>
<p>综上所述，该研究报道了高质量的四川金粟兰基因组，填补了核心被子植物金粟兰目的基因组信息，并通过多种分析策略和方法，深入的揭示了核心被子植物间的复杂演化历程，该成果对于早期被子植物的起源、辐射进化以及适应性演化提供数据支撑和理论基础。</p>
<h2 id="Genome-size-estimate-and-assembly"><a href="#Genome-size-estimate-and-assembly" class="headerlink" title="Genome size estimate and assembly."></a>Genome size estimate and assembly.</h2><p>To estimate the genome size of C. sessilifolius, we surveyed 150 bp paired-end reads, computed 21 bp K-mer frequencies using <font color="red"><strong>Jellyfish</strong></font>, and exported the resulting histogram into <font color="red">findGSE</font>. <font color="red"><strong>Nextdenovo</strong></font> (https:// github.com/Nextomics/Nextdenovo) was selected for <font color="blue"><strong>correcting reads</strong></font> with para- meters “read_cutoff=2k, seed_cutoff=30k, blocksize=1.5g” and then <font color="red"><strong>Smartdenovo</strong></font> (<a target="_blank" rel="noopener" href="https://github.com/ruanjue/smartdenovo">https://github.com/ruanjue/smartdenovo</a>) for <font color="blue"><strong>de novo assembly</strong></font> with parameters “wtpre -J 3,000; wtzmo-k 21 -z 10 -Z 16 -U -1 -m 0.1 -A 1000; wtclp -d 3 -k 300 -m 0.1 -FT; wtlay -w 300 -s 200 -m 0.1 -r 0.95 -c 1”. <font color="blue">The preliminary contigs were further <strong>polished</strong> by aligning the Illumina <strong>short</strong> <strong>reads</strong> to the contigs</font> using <font color="red"><strong>Nextpolish</strong></font>. After four rounds of successive iterative correction, the final genome sequence was obtained. The GC content and sequencing coverage analyses were applied to evaluate the presence of contamination. The quality of the genome assembly was also assessed using <font color="red"><strong>BUSCO</strong></font> (Benchmarking Universal Single-Copy Orthologs) with the embry- ophyta_odb10 database. The clean <font color="blue">Hi-C</font> data were mapped to contig sequences by <font color="red"><strong>Bowtie2</strong></font> and 354 Mb valid interaction pairs were extracted. Based on those chromatin interactions, <font color="red"><strong>LACHESIS</strong></font> was employed to <font color="blue"><strong>cluster</strong>, <strong>order</strong>, and <strong>orient</strong> the <strong>contigs</strong> into <strong>pseudo-chromosomes</strong></font>.</p>
<h1 id="9-2022-01-gb-油茶"><a href="#9-2022-01-gb-油茶" class="headerlink" title="9. 2022.01-gb 油茶"></a>9. 2022.01-gb 油茶</h1><blockquote>
<p>The genome of oil-Camellia and population genomics analysis provide insights into seed oil domestication</p>
</blockquote>
<p>油茶是我国传统木本油料树种，具有2300余年栽培食用历史。自上世纪起，经过四代科技工作者艰苦努力，我国油茶主栽良种衍生出数百个品种。但受制于<font color="blue">多倍性、长时效</font>特性，油茶育种工作效率不高。</p>
<p>2022年1月10日，中国林业科学研究院亚热带林业研究所（亚林所）研究员<strong>姚小华团队</strong>和<strong>殷恒福团队</strong>在<strong>《基因组生物学》（Genome Biology）</strong>发表最新论文。</p>
<p>该研究成功组装全球首个染色体级别高质量油茶基因组，揭示<font color="red">油茶物种<strong>进化历史</strong>及其<strong>种子含高油脂</strong>、<strong>高不饱和脂肪酸</strong>的<strong>驯化机制</strong></font>，建立了油脂性状早期选择技术体系，为加快良种选育效率、保障我国粮油安全奠定了基础。</p>
<h2 id="抱子怀胎的油茶良种育种缓慢"><a href="#抱子怀胎的油茶良种育种缓慢" class="headerlink" title="抱子怀胎的油茶良种育种缓慢"></a>抱子怀胎的油茶良种育种缓慢</h2><p>良种是农业生产的“芯片”。作为我国四大油料作物之一，油茶是农民增收、粮油安全、生态建设的“重要法宝”。我国油茶育种历经了选择育种和定向杂交育种阶段，主栽良种由家系品种、农家品种走向无性系良种，衍生出数百个品种，构成了我国油茶产业的发展基础。</p>
<p>但由于油茶结实之前要经历漫长的童期，而且从<font color="blue">开花到果实成熟需要整整1年时间</font>，俗称‘抱子怀胎’，导致油茶育种年限长，新品种选育缓慢，良种选育速度无法满足产业发展的需求，已成为阻碍油茶产业发展的重要因素。</p>
<p>我国长期面临食用油供需压力，对外依存度高达60%。油茶能够利用我国亚热带区域的宜林山地种植，不与粮争地，是保障我国粮油安全、实现乡村振兴的重要抓手。</p>
<p>油茶泛指山茶科山茶属植物中油脂含量高，具有一定栽培面积的，有经济栽培价值的物种的总称，是我国主推的重要木本油料树种。我国现有油茶种植面积7000万亩，以普通油茶（Camellia oleifera）为主，年总产值1200亿元左右。</p>
<p>油茶籽油中不饱和脂肪酸达90%以上，其中以单不饱和脂肪酸油酸为主，富含角鲨烯、维生素E、甾醇、多酚等功能成分，具有极高的保健价值，被誉为“液体黄金”。</p>
<p>然而研究发现, 油茶基因组具有<font color="red"><strong>杂合率高、重复序列占比大</strong>的特点，其杂合率达到了<strong>2.52%</strong></font>，远高于大部分已测序的物种，这为基因组组装带来了巨大挑战。</p>
<h2 id="染色体级别组装"><a href="#染色体级别组装" class="headerlink" title="染色体级别组装"></a>染色体级别组装</h2><p>经过四年多的努力，该团队获得大小为<font color="red">2.95 GB，Contig N50为<strong>1.002 MB</strong></font>的二倍体油茶（CON）的基因组图谱，并将基因组<font color="red">锚定到15条染色体</font>，锚定率达到<font color="red">91.33%</font>，获得了质量优良的油茶基因组图谱。</p>
<p><img src="/blog/2022/03/14/2022-03-14-HiFi-6-HiC-assembly/2.png" alt="油茶基因组图谱及其进化"></p>
<p>同时，利用控制杂交的群体建立<font color="red"><strong>高密度的遗传标记连锁图谱</strong></font>，对<font color="red">基因组<strong>图谱组装</strong>进行进一步<strong>校正</strong></font>，为未来重要性状的基因定位提供支撑。</p>
<p>油茶物种起源与分布主要集中在我国南方亚热带地区。通过对油茶基因组的比较分析，发现油茶基因组共发生<font color="red"><strong>两次全基因组复制</strong>事件</font>：第一次为古老的<font color="red">γ复制</font>，第二次发生在<font color="red"><strong>山茶属与猕猴桃属</strong>分化前不久</font>。</p>
<p>而山茶属形成之后并没有发生特殊的全基因组复制事件。通过山茶—猕猴桃、山茶—柿、柿—猕猴桃<font color="red"><strong>属间同源突变率的比较</strong></font>印证这一观点。</p>
<p><img src="/blog/2022/03/14/2022-03-14-HiFi-6-HiC-assembly/3.png" alt="油茶基因组重复序列分析"></p>
<p>山茶属包括200多个物种，包括以<strong>普通油茶</strong>为代表的油茶组、以<strong>茶</strong>为代表的茶组和以<strong>山茶花</strong>为代表的山茶组等。系统发育和分子钟标定分析发现，<strong>油茶和茶的分化</strong>发生在大约17.3百万年前。</p>
<p>在长期栽培驯化中，区别于以叶用为目的的茶和以观赏为目的的山茶，油茶在人工选择作用下进化成以种子油脂为主要栽培目的的木本油料树种，使得油茶成为研究植物种子<font color="red"><strong>油脂性状驯化</strong></font>的绝佳材料。</p>
<p>为进一步揭示油茶油脂性状驯化的分子机制，该团队利用全国分布区内的<font color="red">221个代表品种</font>开展<font color="red"><strong>群体转录组测序和关联分析</strong></font>。结果发现在长期<strong>驯化</strong>过程中，油茶果实逐渐<strong>增大</strong>，种仁油脂含量显著<strong>提高</strong>，油脂品质不断<strong>优化</strong>，形成了目前主栽的果大、含油率高、油脂品质优良的栽培类型。</p>
<p>进一步分析发现驯化过程中，油茶基因组中<font color="red"><strong>油脂代谢通路</strong>中的多个<strong>基因</strong>受到了<strong>显著的人工选择</strong></font>。此外，<font color="red">胁迫应答、激素生物合成</font>等通路也受到显著选择。</p>
<p>为了挖掘高度可信的油脂代谢关键基因，联合采用<font color="red"><strong>基因组遗传变异、基因表达水平和表型性状变异关联分析</strong></font>。令人惊讶的是，在<font color="red">筛选出的<strong>21个候选</strong>基因</font>中，有<font color="red"><strong>14个</strong></font>属于油脂合成和分解代谢通路基因。</p>
<p><img src="/blog/2022/03/14/2022-03-14-HiFi-6-HiC-assembly/4.png" alt="油茶品种群体关联分析发掘油脂产量与品质的调控基因"></p>
<p>其中，油脂代谢通路中的8个基因和植物激素相关转录因子IAA26在长期<strong>栽培驯化</strong>过程中<strong>形成</strong><font color="red"><strong>显著</strong>的<strong>单核苷酸多态性</strong>和<strong>表达水平差异</strong></font>。“这8个基因既包括<font color="red">油脂合成</font>基因，也有<font color="red">油脂分解代谢</font>基因，这表明油茶<font color="red"><strong>油脂性状</strong></font>是由参与合成和分解的<font color="red"><strong>两类基因协同调控</strong></font>的结果。</p>
<h2 id="Genome-sequencing-de-novo-assembly-and-annotation"><a href="#Genome-sequencing-de-novo-assembly-and-annotation" class="headerlink" title="Genome sequencing, de novo assembly, and annotation"></a>Genome sequencing, de novo assembly, and annotation</h2><p>The genomic DNA of CON was prepared by using the young <strong>leaves</strong>. Ten 20 kb de novo SMRTbell libraries were constructed according to the standard manufacturer’s protocol and used for <font color="red">SMRT PacBio</font> genome sequencing. A total of 27,876,348 reads (total size of <strong>320 G</strong>) were generated and used for initial assembly by the <font color="red"><strong>Falcon</strong></font> (v0.3.0) pipeline. The <font color="red"><strong>HaploMerger2</strong></font> (v20180603) program (default parameters) was used to <strong>reduce the redundancy</strong>, and <font color="red"><strong>Arrow program</strong></font> with default parameters was used to <strong>correct</strong> the sequencing <strong>errors</strong>. Further, a total of <font color="red">210 G</font> clean data generated by the <strong>Illumina</strong> Nova- Seq6000 platform was used to <font color="red">correct</font> the PacBio reads.</p>
<p>To circumvent the high heterozygosity of the CON genome, a <font color="red"><strong>hybrid assembly of strategy</strong></font> was used to construct the high-quality reference genome. The details of the <font color="red"><strong>BioNano, 10X Genomics, and Hi-C</strong></font> sequencing procedures were described in the Additional File 3: Method S1; and the hybrid assembly approach was described in <strong>Additional File 2</strong>: Fig. S3. Finally, to construct a chromosome-scale reference genome, the Hi-C chromosomal interaction was created using HiC-pro [38] software (v2.5.0) (<strong>Additional File 2</strong>: Fig. S4). </p>
<p>The <font color="red">repetitive elements</font> in the CON genome, including tandem repeats and interspersed repeats, were identified. Tandem repeats were discovered by <font color="red"><strong>Tandem Repeats Finder</strong></font> v4.07b. Interspersed repeats in the genome were identified using an integration of <strong>independent homology searching and de novo predictions</strong> (See details in Additional File 3: Method S2). Non-coding RNA genes (ncRNA) were annotated in this study (see details in Additional File 3: Method S2) and the results were shown in Additional File 1: Table S4.</p>
<p>We <font color="red">annotated</font> the assembled genome through combining three different approaches: <font color="red"><strong>ab initio prediction, homology-based prediction,</strong> and <strong>transcriptome alignment</strong></font>. To ob- tain the transcriptome data, the total RNA of seven different tissues from CON was se- quenced using the Illumina NovaSeq platform. The transcripts homolog prediction was performed initially by MAKER (v2.31.10); Augustus (v3.3.1) and SNAP (v2006-07-28) were used for de novo prediction. Finally, both homolog and de novo prediction results were integrated using MAKER and resulted in the final gene models (Additional File 1: Table S5; see details in Additional File 3: Method S2)</p>
<p><font color="red"><strong>Functional annotation</strong></font> was achieved by comparing predicted proteins against public databases, including NCBI non-redundant protein sequences database (Nr), SwissProt (201709) [39], eggNOG [39], KEGG (v84) [40], Interpro (v5.16-55.0) [41], and GO [42] using Blast (v2.2.3) [43].</p>
<p><img src="/blog/2022/03/14/2022-03-14-HiFi-6-HiC-assembly/5.png" alt="5"></p>
<p>PacBio+illumina : Falcon+Arrow+BWA+Pilon+HaploMerger-&gt;contigs</p>
<p>scaffold+HiC : <font color="red"><strong>JUICER+3D-DNA</strong></font></p>
<h1 id="10-2022-01-HorticRes-甘菊"><a href="#10-2022-01-HorticRes-甘菊" class="headerlink" title="10. 2022.01-HorticRes 甘菊"></a>10. 2022.01-HorticRes 甘菊</h1><p><strong>The</strong> <em><strong>chrysanthemum lavandulifolium</strong></em> <strong>genome and the molecular mechanism underlying diverse capitulum types</strong></p>
<p><strong>菊花</strong>(<em>Chrysanthemum ×morifolium</em> Ramat.)是世界著名的观赏植物，具有千姿百态的花型。实际菊花的花是指由<font color="red"><strong>外围的舌状花</strong></font>和<font color="red"><strong>盘心的管状花</strong></font>共同构成的<font color="red"><strong>头状花序</strong></font>。其花型由头状花序上舌状花和管状花的形态和相对数量决定。解析同一头状花序上舌状花和管状花分化的分子调控机制不仅可以为阐明菊花复杂的头状花序形态奠定基础，也将为菊科植物头状花序发育提供新见解。但目前关于菊花花型的研究受到其复杂遗传背景的限制，导致无法充分利用菊花丰富的基因资源进行花型定向育种。因此，获得高质量的菊花及其近缘种的基因组信息并在此基础上<font color="red">研究<strong>头状花序发育</strong>的<strong>分子调控机制</strong></font>显得尤为必要。</p>
<p>近日，<em><strong>Horticulture Research</strong></em>在线发表了<strong>北京林业大学戴思兰团队</strong>题为<em><strong>The Chrysanthemum lavandulifolium genome and the molecular mechanism underlying diverse capitulum types</strong></em> 的研究论文。该论文完成<font color="red"><strong>菊花近缘野生种</strong>之一<strong>甘菊</strong></font>(C. lavandulifolium)的全基因测序工作，获得了染色体水平上的高质量甘菊参考基因组，结合<font color="red"><strong>3种不同类型菊科植物头状花序</strong>的<strong>转录组数据</strong></font>初步解析了头状花序发育的<strong>分子调控机制</strong>，为实现人工调控菊花花型奠定了坚实的分子理论基础。</p>
<p>甘菊为菊属植物中的<strong>二倍体物种</strong>，采用<strong>Illumina + Pacbio + Hi-C</strong>测序技术对其进行全基因测序，获得了2.60 Gb的染色体水平的参考基因组。其中<font color="red">94.46%的序列锚定到<strong>9条染色体</strong>上</font>。甘菊基因组中重复序列占基因组的67.69%，<em>Cypsy</em>和<em>Copia</em>的占比分别为37.92%和29.06%，插入时间在1.25百万年前。<font color="red"><strong>比较基因组分析</strong></font>结果表明，甘菊经历了<font color="red"><strong>两次全基因组复制(WGD)事件</strong></font>，其中<font color="red"><strong>最近一次</strong>是所有<strong>菊科植物共有</strong></font>，而<font color="red"><strong>较早一次</strong>是核心双子叶植物共有的<strong>γ事件</strong></font>，**甘菊自身没有发生WGD，其基因组演化的动力来源主要是<font color="red">串联重复事件</font>**。</p>
<p><img src="/blog/2022/03/14/2022-03-14-HiFi-6-HiC-assembly/6.png" alt="overview and genome evolution"></p>
<p>基于<font color="blue">甘菊头状花序发育的6个重要时期</font>和<font color="blue">其他菊科植物不同类型头状花序发育关键时期</font>的<font color="red">转录组</font>有参分析发现，MADS-box、TCP、NAC和LOB基因家族<font color="red">可能参与<strong>管状花和舌状花的分化</strong></font>。值得注意的是，<em>NAM</em>和<em>LOB30</em>高表达于舌管兼备型的头状花序中，而在全舌型和全管型的头状花序中表达量相对较低，这表明其可能是参与两类小花分化的关键基因。结合关键基因在全舌型、全管型以及舌管兼备型的头状花序中的表达模式和蛋白互作模式，初步推测并构建了不同类型头状花序发育可能的调控机制。在全管型头状花序中，<em>CUC2</em>与<em>LFY</em>和<em>AG</em><font color="red"><strong>共同调控</strong>管状花原基的起始</font>。而CUC2和CUC3的<font color="red"><strong>互作及协同表达</strong>则会<strong>促进</strong>全舌型头状花序中舌状花原基的起始</font>。在舌管兼备型的头状花序中，<em>NAM</em>和<em>LOB30</em>的表达则参与<font color="red"><strong>调控</strong>舌状花和管状花原基的分化</font>。LOB30可以与LFY, TFL1, CUC2, CUC3和NAM互作，表明其在基因调控网络中的核心位置。总之，<em>NAM</em>和<em>LOB</em>不仅可以与花序分生组织相关基因如<em>LFY</em> 等互作，也可以与两类小花身份决定基因如<em>CYC2</em>-<em>LIKE</em>等基因互作，这表明<em><strong>NAM</strong></em><strong>和LOB30在头状花序上舌状花和管状花原基分化调控中的关键角色</strong>。</p>
<p><img src="/blog/2022/03/14/2022-03-14-HiFi-6-HiC-assembly/7.png" alt="The expression patterns of NAM/CUC and LOB30 homologous genes in ten asteraceae species"></p>
<p><img src="/blog/2022/03/14/2022-03-14-HiFi-6-HiC-assembly/8.png" alt="the probable gene regulation mechanism in the development of different capitula types"></p>
<p><strong>高质量甘菊参考基因组的获得不仅可以为栽培菊花基因组的破译提供有效的参考，更为解析菊花乃至菊科植物多样的生物学性状提供丰富的基因资源。</strong></p>
<p>北京林业大学戴思兰课题组一直致力于菊花研究，基于<font color="blue">植物系统学研究方法</font>对<font color="blue">菊属植物种间<strong>亲缘关系</strong>和菊花<strong>品种起源</strong></font>进行探索，在<font color="red"><strong>种质资源评价，花色、花型、开花期</strong>和<strong>抗逆性</strong>等观赏性状形成的<strong>分子机理</strong></font>，菊花<font color="blue">优异种质创制</font>以及<font color="blue">产业化栽培技术</font>等开展全面研究，取得了一系列重要突破性进展和研究成果。</p>
<p>2.6G：scaffold Num 178；scaN50 300M；N90 212M；Longest 322M</p>
<h2 id="Genome-assembly-and-quality-assessment"><a href="#Genome-assembly-and-quality-assessment" class="headerlink" title="Genome assembly and quality assessment"></a><strong>Genome assembly and quality assessment</strong></h2><p>For the <font color="blue">PacBio RSII</font> platform data, <font color="blue">longer subreads were <strong>selected</strong></font> by the <font color="blue">error correction module</font> of <font color="red"><strong>canu v1.5</strong></font> [35]. <font color="blue">Raw overlapping subreads were <strong>detected</strong></font> through the highly sensitive overlap detection program <font color="red"><strong>MHAP v2.1</strong></font> [36], and the <font color="blue">error correction of these data was <strong>carried</strong></font> out by the <font color="red"><strong>Falcon</strong></font> sense method v0.40 (“correct- edErrorRate = 0.025”) [37]. The error-corrected subreads were used to <font color="blue">generate a <strong>draft assembly</strong></font> in <font color="red"><strong>WTDBG v2.5</strong></font> (<a target="_blank" rel="noopener" href="https://github.com/ruanjue/wtdbg">https://github.com/ruanjue/wtdbg</a>). <font color="blue"><strong>Iterative polishing</strong></font> by <font color="red"><strong>Pilon v1.22</strong></font> [38] was achieved by <font color="blue"><strong>aligning</strong></font> adapter-trimmed and paired-end <font color="blue">Illumina <strong>reads</strong> to the <strong>PacBio draft genome</strong></font>. <font color="red"><strong>Clean nanopore data</strong></font> were acquired via sequencing on the PromethION platform, and these data were <font color="blue">corrected with the same method</font> described above. The draft genome assembled by <font color="red"><strong>WTDBG v2.5</strong></font> (https:// github.com/ruanjue/wtdbg) was <font color="blue"><strong>corrected</strong> three times</font> by <font color="red"><strong>Racon v1.3.3</strong></font> [39] by aligning adapter-trimmed and paired-end Illumina reads to the <font color="blue"><strong>Nanopore draft genome</strong></font>. Then, the <font color="blue">PacBio draft genome as a <strong>query input</strong></font> was <font color="blue"><strong>aligned against</strong> the <strong>Nanopore draft genome</strong></font> using <font color="red"><strong>MUMmer v4.0.0</strong></font> [40]. The PacBio draft genome and Nanopore <font color="blue"><strong>draft genome</strong></font> were then <font color="blue"><strong>merged</strong></font> using <font color="red"><strong>quickmerge v0.3.0</strong></font> [21]. This <font color="blue"><strong>merged draft genome</strong></font> was <font color="blue"><strong>polished</strong></font> by <font color="red"><strong>Racon v1.3.3</strong></font> [39] and <font color="red"><strong>Pilon v1.22</strong></font> [38]. The <font color="blue">mapping depth</font> was obtained by <font color="blue"><strong>aligning</strong> corrected Nanopore sequencing data</font> to the merged assembly by <font color="red"><strong>minimap2 v2.17</strong></font> [41] with default parameters. Then, <font color="red"><strong>purge haplotigs v1.0.4</strong></font> [22] was used to <font color="blue"><strong>eliminate</strong> redundancy according to the <strong>coverage depth</strong></font> and obtain the <font color="blue"><strong>purged haplotig genome</strong></font>. Ultimately, a <em>C. lavandulifolium</em> genome with a total length of <strong>3.10 G</strong>b was obtained. The second-generation sequencing data, core gene completeness and BUSCOs were evaluated to verify the accuracy of the genome assembly.</p>
<h2 id="Hi-C-sequencing-and-assistant-assembly"><a href="#Hi-C-sequencing-and-assistant-assembly" class="headerlink" title="Hi-C sequencing and assistant assembly"></a><strong>Hi-C sequencing and assistant assembly</strong></h2><p>Hi-C is a technology derived from chromosome conformation capture technology that utilizes high- throughput sequencing data and is mainly used to assist in genome assembly. We constructed Hi-C fragment libraries with insert sizes of 300–700 bp, as illustrated in Rao et al. [42], and sequenced them using the Illumina platform [42]. <font color="blue"><strong>Before</strong> <strong>chromosome assembly</strong>, we first performed a <strong>preassembly</strong> for the <strong>error correction of scaffolds</strong>, which required <strong>splitting the scaffolds into segments of 50 kb</strong>, on average</font>. Then, the Hi-C data were <font color="blue">mapped</font> to these segments using <font color="red"><strong>BWA</strong></font> aligner v0.7.10-r789 [43]. We <font color="red"><strong>retained</strong> the <strong>uniquely mapped</strong> data</font> to assemble the genome using <font color="red"><strong>LACHESIS</strong></font> [44] with the following <font color="green"><strong>parameters</strong>: CLUSTER_MIN_RE_SITES = 80; CLUSTER_MAX_LINK_DENSITY = 2; CLUSTER_NONINFORMATIVE_RATIO = 2; ORDER_MIN_N_RES_IN_TRUN = 16; ORDER_MIN_N_RES_IN_SHREDS = 16</font>. To further address the redundant sequences, we manually checked any two segments that showed inconsistent connections with the raw scaffold. The detailed workflow schema for the assembly pipeline of the chromosome-scale <em>C. lavandulifolium</em> genome is shown in Supplementary Figure 24.</p>
<p><img src="/blog/2022/03/14/2022-03-14-HiFi-6-HiC-assembly/9.png" alt="9"></p>
<h1 id="11-2022-01-HorticRes-粉葛"><a href="#11-2022-01-HorticRes-粉葛" class="headerlink" title="11. 2022.01-HorticRes 粉葛"></a>11. 2022.01-HorticRes 粉葛</h1><p><em><strong>Chromosomal-level genome and multi-omics dataset of Pueraria lobata var. thomsonii provide new insights into legume family and the isoflavone and puerarin biosynthesis pathways</strong></em> </p>
<p>粉葛基因组及多组学分析为<font color="red">葛根素</font>等功效成分的生物合成研究提供新见解</p>
<p>广西壮族自治区农业科学院经济作物研究所严华兵团队。该研究利用<strong>PacBio</strong>、<strong>Illumina</strong>测序以及<strong>Hi-C</strong>技术测序组装，<strong>获得了首个<font color="red">豆科葛属药食同源植物粉葛</font>染色体级别的基因组</strong>，标志着粉葛研究迈入基因组时代，开启了<font color="blue">葛属<strong>进化</strong>、野葛<strong>驯化</strong>、粉葛<strong>品种改良</strong></font>之旅，助力广西特色优势粉葛产业高质量发展。</p>
<p>鉴于粉葛<font color="blue">杂合度较高</font>，研究者选用了PacBio和Hi-C测序，构建的粉葛基因组大小为<strong>1.38Gb</strong>，<font color="blue">Contig N50=598kb</font>，并将<font color="blue">99.3%的序列<strong>锚定</strong>到11条染色体上</font>，BUSCO评估<font color="blue">92.9%</font>。通过注释，共获得45270个蛋白编码基因，其中<font color="blue"><strong>94.4%的基因</strong>可以得到<strong>功能注释</strong></font>，基因组中<font color="blue">重复序列</font>占比为62.7%。将粉葛<font color="red">与<strong>16个近缘物种</strong>(包含5个豆科植物)进行<strong>比较基因组</strong>分析</font>，结果表明6个豆科植物共有基因家族为11204个，粉葛基因组中<font color="blue"><strong>显著扩张的4743个基因家族</strong>主要富集在与<strong>类黄酮、生物碱、甾醇和萜类</strong>等生物合成相关的通路</font>；粉葛<font color="red"><strong>特有</strong>基因家族</font>为2373个，主要<font color="blue">富集在与<strong>萜类</strong>生物合成相关的通路</font>；粉葛基因组中受到<font color="blue"><strong>显著正选择</strong>的基因</font>共有34个，富集在<font color="blue"><strong>昼夜节律、同源重组、淀粉代谢和蔗糖代谢</strong></font>等途径。</p>
<p><font color="red"><strong>系统进化分析</strong></font>表明，粉葛<strong>与大豆亲缘关系最近</strong>，两者在20.1个百万年前产生分化。通过<font color="red"><strong>Ks</strong>和<strong>4DTV分析</strong></font>，<strong>粉葛共经历两次WGD事件</strong>，一次WGD事件是<font color="blue">大豆与粉葛<strong>共有</strong></font>，发生在44.5个百万年前；另一次WGD事件是在粉葛与大豆分化后，<font color="blue">粉葛<strong>独自</strong>经历</font>，发生时间大致在4.8个百万年前。</p>
<p>通过对高葛根素ZG-19和低葛根素ZG-39进行<font color="red"><strong>转录组</strong>和<strong>代谢组</strong>分析</font>，研究者检测到了614种225种差异代谢物(DMs)，1814个差异表达基因(DEG)，DMs和DEG的丰富功能类别<strong>重叠</strong>，这说明它们都是与<font color="blue">类黄酮、异黄酮和ABC转运相关的基因或代谢物</font>。进一步分析代谢物与基因表达的相关系数，结果表明代谢物和基因对在样本中高度相关，60%的显著相关性涉及上调的代谢物和下调或不变的基因，在15%的显著相关性中，代谢物和基因表达的变化方向相同。此外，研究者在异黄酮生物合成途径中发现了大量的DMs和DEG。<strong>这充分解析了粉葛中异黄酮的生物合成途径。</strong></p>
<p>通过<font color="red"><strong>同源基因搜索</strong></font>，研究者发现<font color="blue">编码<strong>葛根素合成途径</strong>中关键酶的9个基因家族</font>在粉葛中都有所<font color="blue">扩张</font>；通过分析糖基转移酶家族中催化糖基化修饰的基因，共鉴定出104个GT基因，有13个基因与8-C-葡萄糖基转移酶(8-C-GT)同源，其中6个与先前研究的催化大豆苷元C-糖基化为葛根素的PIUGT43基因同源；编码大豆异黄酮合酶(IFS)的基因(CHR11G3854.1)催化着葛根素合成的中间代谢物大豆苷元的合成，被鉴定为与葛根素的合成途径高度相关。总之，上述分析<strong>初步解析了粉葛中葛根素的生物合成途径。</strong></p>
<p><strong>综上，该研究通过构建高质量的粉葛基因组解析了粉葛基因组的<font color="red">进化特征</font>；通过多组学分析深入解析了粉葛中<font color="red">重要次生代谢物异黄酮、葛根素等生物合成途径</font>，从而为粉葛的资源利用、遗传育种等研究提供了新见解。</strong></p>
<p><strong>严华兵研究员团队在葛根上的研究进展：</strong></p>
<p><strong>严华兵</strong>团队近年<strong>在全球葛根资源收集与鉴定评价、葛属资源分类、葛根<font color="red">基因组与分子生物学</font>、粉葛和野葛品种选育、健康种苗生产、高产高效栽培等方面取得了一系列的成果</strong>。团队目前广泛收集全球葛属种质资源419份，包括野葛、粉葛、葛麻姆、大花葛、泰葛、苦葛、红葛、须弥葛、食用葛等；通过开发<strong>葛SSR分子标记</strong>，构建了广西葛核心种质库；通过<strong>广泛靶向代谢组</strong>解析葛属葛种野葛、粉葛和葛麻姆等3个变种块根中影响<strong>食用品质和药用品质</strong>的<strong>代谢差异</strong>；结合表型鉴定通过叶绿体基因组研究，揭示了葛及其<strong>近缘种之间的系统发育关系</strong>；挖掘了调控<strong>葛根素合成代谢相关的结构基因和转录因子</strong>，并正在开展相关基因功能验证工作；选育出适合开发葛花茶、高葛根素粉葛、无渣粉葛、药用野葛等系列葛根新品种，并逐步建立配套种苗繁育和高效栽培技术。</p>
<p>1.37G：contig Num 5145；conN50 593kb；</p>
<h2 id="Genome-survey"><a href="#Genome-survey" class="headerlink" title="Genome survey"></a>Genome survey</h2><p>Short reads produced by the Illumina 6000 platform were <font color="blue">quality-filtered</font> by <font color="red"><strong>HTQC</strong></font> [23] (version v1.92.310) using the following method. Adaptors were firstly removed from the sequencing reads, and read pairs with any one end having an average quality &lt;20 were discarded. Ends of reads were trimmed if average quality was &lt;20 in the sliding window size of 5 bp, and read pairs with <font color="red">any end <strong>shorter than 75 bp</strong> were removed</font>. The quality-filtered reads were used for estimation of genome size. The <strong>17-mer</strong> occurrence distribution of sequencing reads was generated from short libraries with <font color="red"><strong>Jellyfish</strong></font> [24] (v2).</p>
<h2 id="Genome-assembly"><a href="#Genome-assembly" class="headerlink" title="Genome assembly"></a>Genome assembly</h2><p>Subreads generated with the above long-read third-generation sequencing were used for genome assembly of <em>P. lobata</em>. The draft assembly of the genome was made using <font color="red"><strong>Falcon (v0.3.0)</strong></font> [25]. To <font color="blue"><strong>correct</strong> errors</font> in the primary assembly, <font color="red"><strong>Racon</strong></font> (v1.44) tools were used to <font color="blue"><strong>polish</strong></font> the genome [26]. <font color="blue">Illumina-derived <strong>short reads</strong></font> were used to <font color="blue"><strong>correct</strong></font> any <font color="blue"><strong>remaining errors</strong></font> by <font color="red"><strong>NextPolish (v1.1.0)</strong></font> [27].</p>
<p><font color="blue">To scaffold the contigs</font>, 425 011 698 clean-read pairs were sequenced from the Hi-C library. <font color="red"><strong>Fastp</strong></font> (0.19.5) [28] was used to filter Hi-C data, including <font color="blue"><strong>removing adaptors</strong> and l<strong>ow-quality</strong> reads</font>. <font color="red"><strong>HiC-Pro</strong></font> Proto was used to obtain effective reads [29]. After filtering, <font color="blue">6%【?】 of high-quality and effective Hi-C data was retained</font>. The <font color="blue">resulting <strong>effective reads</strong></font> were <strong>mapped</strong> to the <strong>assembled and polished</strong> <em>P. thomsonii</em> <strong>genome</strong> using <font color="red"><strong>BWA</strong></font> (bwa-0.7.17) [30] with default parameters. <font color="red"><strong>ALL-HiC</strong></font> software was used to <font color="blue"><strong>anchor</strong> scaffolds to chromosomes</font> to obtain the chromosome-level <em>P. thomsonii</em> assembly.</p>
<h1 id="12-2022-03-ng-四倍体土豆"><a href="#12-2022-03-ng-四倍体土豆" class="headerlink" title="12. 2022.03-ng 四倍体土豆"></a>12. 2022.03-ng 四倍体土豆</h1><blockquote>
<p>Chromosome-scale and haplotype-resolved genome assembly of a tetraploid potato cultivar</p>
</blockquote>
<p>##<strong>Genome size estimation.</strong> </p>
<p>After <strong>trimming</strong> off 10× genomics <font color="blue"><strong>barcodes and hexamers</strong></font> from the 370.3-Gb reads combined from the 10× single-cell CNV libraries and single-molecule libraries, <em>k</em>-mer counting (<em>k</em> = 21) was performed with <font color="red"><strong>Jellyfish</strong></font> (v.2.2.10). The <em>k</em>-mer histogram was provided to <font color="red"><strong>findGSE</strong></font> (v.1.0)39 to estimate the haploid/tetraploid genome size of ‘Otava’ under the <font color="blue">heterozygous mode</font> (with ‘exp_hom = 200’; Extended Data Fig. 1).</p>
<p>##<strong>Initial tetraploid genome assembly, polishing and purging.</strong> </p>
<p>The initial assembly of the tetraploid genome was performed using <font color="red"><strong>hifiasm (v.0.7)</strong></font> with <strong>default settings</strong> with the 102 Gb raw PacBio HiFi reads of ‘Otava’, where the output consisting of <font color="red"><strong>unitigs</strong> (locally haplotype-resolved contigs)</font> was selected for <strong>further processing</strong>. </p>
<p>Then the <font color="blue">short reads</font> from the two 10× single-cell CNV libraries were <font color="blue"><strong>aligned</strong> to the assembly</font> using <font color="red"><strong>bowtie2 (v.2.2.8)</strong></font>. These <font color="blue">alignments were used to <strong>polish</strong> the <strong>assembly</strong></font> with <font color="red"><strong>pilon (v.1.22)</strong></font> with <font color="green"><strong>options</strong> of <strong>–fixbases –changes –diploid –mindepth 0.8</strong></font>. </p>
<p>Further, <font color="blue"><strong>short</strong> reads</font> of the two 10× single-cell CNV libraries, additional 10× single-molecule libraries <strong>and</strong> PacBio <font color="blue"><strong>HiFi</strong> reads</font> were <font color="blue"><strong>all aligned</strong> to the <strong>polished assembly</strong></font> (using <font color="red"><strong>bowtie2</strong></font> and <font color="red"><strong>minimap2</strong></font> (v.2.17-r491) respectively). Among <font color="blue">17,153 raw contigs, <strong>9,041 contigs</strong></font> which were <font color="blue">longer than <strong>50 kb</strong></font> and with an <font color="blue">average sequencing depth over <strong>80×</strong> were <strong>kept</strong></font>. </p>
<p><font color="blue">HiFi reads were <strong>re-aligned</strong> to the purged assembly and contigs covered less than 3× were <strong>removed</strong></font>. </p>
<p>The HiFi reads-based <font color="red"><strong>purging process</strong></font> was repeated for <font color="blue"><strong>five rounds</strong></font> to get an initial assembly of <font color="blue">6,366 contigs</font> for subsequent analysis (Supplementary Fig. 1).</p>
<p>##<strong>Haplotype-specific PacBio HiFi read separation and haplotype assembly.</strong></p>
<p>HiFi reads were <font color="blue">classified into 48 groups</font> on the basis of alignments to the 50-kb coverage markers using customized code. Specifically, to assign a read to a marker, at least 500 bp of the read had to be aligned to the marker (reads aligning two neighboring markers were assigned to the marker with a larger overlapping size). Reads overlapping non-haplotig marker were randomly assigned to one of the marker-associated groups.</p>
<p><font color="red"><strong>Each set</strong> of HiFi reads</font> was independently assembled using <font color="red"><strong>hifiasm</strong></font> (v.0.7) with default settings. The resulting contigs were first polished with <font color="red">short reads using <strong>pilon</strong></font> with <font color="green"><strong>–fix bases –changes –diploid –mindepth 0.8</strong></font> and then with <font color="red">HiFi reads using <strong>racon</strong></font> (v.1.4.10)46 with<font color="green"> <strong>-u –no-trimming</strong></font>.</p>
<p>##<strong>Evaluation of haplotyping accuracy.</strong> </p>
<p>For <font color="blue">each haplotype assembly</font> and the sequencing data of the parental genomes, <em>k</em>-mers (<em>k</em> = 21) were counted using <font color="red"><strong>KMC</strong></font>. Specifically, <em>k</em>-mers found in ‘Hera’ but not in ‘Stieglitz’ (with a coverage<br>of 6–12), as well as <em>k</em>-mers found in ‘Stieglitz’ but not in ‘Hera’ (with a coverage of 5–11) were selected using kmc_tools simple. For each haplotype, the sets of assembled <em>k</em>-mers were intersected with the two sets of parental-specific <em>k</em>-mers (using kmc_tools simple with subfunction intersect), which revealed <strong><em>k</em>-mers common with either of the parental genomes</strong>. </p>
<p>As a haplotype can only be inherited from one of the parents, it is expected to find parental-specific *k-*mers only of one parent. The overall haplotyping precision was determined as the total number of correctly phased <em>k</em>-mers divided by the total number of <em>k</em>-mers investigated in the <font color="red"><strong>48 haplotype assemblies</strong></font>. Note, this was done <font color="rblueed">before and after contig <strong>polishing</strong></font>, where we observed the same haplotyping accuracy.</p>
<p>##<strong>Haplotype-specific contig scaffolding using group-specific Hi-C reads.</strong> </p>
<p><font color="blue">Each haplotype-specific contig-level assembly</font> was indexed with <font color="red"><strong>bwa index</strong></font> (with -a bwtsw) (v.0.7.15-r1140) and <font color="red"><strong>samtools faidx</strong></font>. The haplotype-specific Hi-C read pairs were aligned using <font color="red"><strong>bwa aln</strong> and <strong>bwa sampe</strong></font>. Aligned reads (in pairs) were <strong>converted into BAM</strong> files using samtools view with options of <font color="green">-b -F12</font>. The BAM files were filtered with <font color="red"><strong>filterBAM_forHiC.pl</strong></font> (from ALLHiC package, v.0.9.13) to <font color="blue"><strong>remove nonuniquely mapped reads</strong></font>. Then BAM files were converted to bed files using <font color="red"><strong>bamToBed</strong></font> (from bedtools package) and <font color="blue"><strong>sorted</strong> by read name</font>. The <strong>bed files</strong> were provided to <font color="red"><em><strong>SALSA2</strong></em></font> (run with <font color="green">-s 100000000 -m yes -i 10 -e DNASE</font>). Potential chimeric contigs were broken at the chimeric sites given by SALSA2 output file of input_breaks, leading to a <font color="blue"><strong>new set</strong> of contigs for each of the 48 original groups</font>.</p>
<p>For <font color="blue">each new group of contigs</font>, the above process of <font color="red"><strong>contig indexing, Hi-C read alignment</strong> and <strong>BAM filtering</strong></font> was repeated. Then, for each haplotype, <font color="red"><strong>ALLHiC_ partition</strong></font> was run with <font color="green">-e GATC -k 1 -m 25</font>; <font color="red"><strong>allhic extract</strong></font> was run with <font color="green">–RE GATC</font>; allhic optimize and <font color="red"><strong>ALLHiC_build</strong></font> were run with <font color="green">default settings</font>; the chromosome contact map was visualized with <font color="red"><strong>ALLHiC_plot</strong></font> at <font color="green">1-Mb</font> resolution, where obvious mis-placement/orientation of large contigs were visually identified and manually corrected (Extended Data Fig. 3).</p>
<h1 id="13-2022-03-ng-六倍体小麦"><a href="#13-2022-03-ng-六倍体小麦" class="headerlink" title="13. 2022.03-ng 六倍体小麦"></a>13. 2022.03-ng 六倍体小麦</h1><blockquote>
<p>Long-read genome sequencing of bread wheat facilitates disease resistance gene cloning</p>
</blockquote>
<table>
<thead>
<tr>
<th>Genomic feature</th>
<th>Value</th>
</tr>
</thead>
<tbody><tr>
<td><em><strong>Length of HiFi assembly</strong></em></td>
<td>14.66 Gb</td>
</tr>
<tr>
<td>Number of contigs</td>
<td>5,055</td>
</tr>
<tr>
<td>Length of contig N50</td>
<td>30.22 Mb</td>
</tr>
<tr>
<td>Length of contig N90</td>
<td>5.5 Mb</td>
</tr>
<tr>
<td><strong><em>Length of hybrid assembly</em>a</strong></td>
<td>14.68 Gb</td>
</tr>
<tr>
<td><em><strong>Length of hybrid scaffolds</strong></em></td>
<td>14.46 Gb</td>
</tr>
<tr>
<td>Number of hybrid scaffolds</td>
<td>324</td>
</tr>
<tr>
<td>Length of hybrid scaffold N50</td>
<td>204.26 Mb</td>
</tr>
<tr>
<td>Gap size</td>
<td>18.8 Mb (0.13%)</td>
</tr>
<tr>
<td><em><strong>Length of pseudomolecule assembly</strong></em></td>
<td>14.68 Gb</td>
</tr>
<tr>
<td><em><strong>Total length of anchored pseudomolecules</strong></em></td>
<td>14.45 Gb</td>
</tr>
<tr>
<td>Number of anchored hybrid scaffolds or contigs</td>
<td>717</td>
</tr>
<tr>
<td>Gap size</td>
<td>11.6 Mb (0.08%)</td>
</tr>
<tr>
<td>Number of high-confidence genes</td>
<td>110,383</td>
</tr>
<tr>
<td><em><strong>Total length of unanchored chromosome</strong></em></td>
<td>224.01 Mb</td>
</tr>
<tr>
<td>Number of unanchored hybrid scaffolds or contigs</td>
<td>3,910</td>
</tr>
<tr>
<td>Gap size</td>
<td>8.1 Mb (3.6%)</td>
</tr>
<tr>
<td>Number of high-confidence genes</td>
<td>6,455</td>
</tr>
<tr>
<td><em><strong>BUSCO</strong></em></td>
<td></td>
</tr>
<tr>
<td>Complete</td>
<td>99.4%</td>
</tr>
<tr>
<td>Duplicated</td>
<td>96.3%</td>
</tr>
<tr>
<td>Fragmented</td>
<td>0.1%</td>
</tr>
<tr>
<td>Missing</td>
<td>0.5%</td>
</tr>
</tbody></table>
<h2 id="Genome-assembly-and-validation"><a href="#Genome-assembly-and-validation" class="headerlink" title="Genome assembly and validation."></a><strong>Genome assembly and validation.</strong></h2><p>The PacBio HiFi reads were assembled using <font color="red"><strong>hifiasm12 (v.0.11)</strong></font> with <font color="green">default parameters</font>. Hybrid scaffolding incorporating the PacBio contigs and the <font color="blue">optical map</font> was performed using the <font color="red"><strong>hybridScaffold pipeline</strong></font> (Bionano Solve 3.6) with default parameters. For the pseudomolecule construction, the Omni-C reads were incorporated using <font color="red"><strong>Juicer tools (v.1.6)</strong></font> and <font color="red"><strong>3D-DNA (v.180114)</strong></font>. </p>
<p>In brief, the preprocessing of the Omni-C reads was performed with juicer.sh (parameter: -s none). The ‘merged_nodups.txt’ output file corresponding to the Hi-C contacts with duplicates removed was subsequently used with run-asm-pipeline.sh (parameter: -r 0) as input to produce the ‘.hic’ and ‘.assembly’ files. These files were uploaded into Juicebox (v.1.11.08) to visualize the Hi-C map and for manual curation. </p>
<p>As a final step, the script <font color="red"><strong>run-asm-pipeline-post-review.sh</strong></font> (default parameters) was used to save the final Hi-C contact map and to output the final Kariega assembly (21 pseudomolecules and 1 unanchored pseudochromosome). To <font color="blue"><strong>validate</strong> the genome assembly</font>, we remapped the optical map onto the pseudomolecule using the <font color="red"><strong>hybridScaffold pipeline</strong></font> (Bionano Solve 3.6), and the final pseudomolecules were compared with the recent bread wheat assemblies of Chinese Spring (IWGSC RefSeq v.2.1) and the assemblies of the 10+ Wheat Genomes Project using <font color="red"><strong>MashMap</strong></font> (v.2.0; parameter: -s 300000–pi 98).</p>
<h1 id="14-2019-10-二倍体巴特利特梨-【四倍体土豆参考hic组装1】"><a href="#14-2019-10-二倍体巴特利特梨-【四倍体土豆参考hic组装1】" class="headerlink" title="14.  2019.10 二倍体巴特利特梨 【四倍体土豆参考hic组装1】"></a>14.  2019.10 二倍体巴特利特梨 【四倍体土豆参考hic组装1】</h1><blockquote>
<p>Pseudo-chromosome–length genome assembly of a double haploid “Bartlett” pear (<em>Pyrus communis</em> L.)</p>
</blockquote>
<p><strong>Genome assembly and scaffolding</strong></p>
<p>The genome assembly workflow began with <em>de novo</em> assembly of contigs from the PacBio long reads using 2 tools, <font color="red"><strong>Canu</strong></font> (version 1.5) and <font color="red"><strong>Falcon</strong></font> (version 0.5). For each assembler the most important assembly parameters were systematically varied (Supplementary Methods), as defined by the tool developers, and by consideration of assembly theory (e.g., overlap length, overlap identity for overlap layout consensus assembly). <font color="blue"><strong>Optimal settings</strong> were <strong>selected</strong></font> by <strong>comparison of assembly statistics</strong> (total size assembled and contig N50) and by <strong>alignment of Illumina PE</strong> data to the assembly with <font color="red"><strong>bowtie2</strong></font> (using the “very fast” pre-set). For <font color="blue">all PacBio assemblies</font> the consensus step was performed by running <font color="red"><strong>Quiver纠错</strong></font> (Genomic Consensus version 2.3.3) (with default parameters) on <font color="blue">raw PacBio contigs</font> and using the full 63× of PacBio data.</p>
<p>Assembled contigs were further joined into scaffolds using a combination of <strong>BioNano optical mapping</strong> data, <strong>Hi-C</strong> chromatin conformation capture data, and <strong>genetic maps.</strong> The best <font color="blue">assemblies from Canu and Falcon</font> were <font color="blue">independently combined with</font> BioNano optical mapping data using the <font color="red">IrysView</font> software to develop the Canu + BioNano (CB) and Falcon + BioNano (FB) assemblies, respectively. The BioNano scaffolding process identified conflicts between the assembled contigs and the optical map, indicating some degree of misassembly in both Canu and Falcon results.</p>
<p><strong>Assembly polishing</strong></p>
<p><font color="red"><strong>Pilon</strong></font> (version1.21) [31] was <font color="blue">run iteratively</font> on the assembly, with <font color="blue">Illumina sequence realigned to the polished assembly</font> at each iteration and then <font color="blue">alignments passed to Pilon to call the next consensus</font>. Alignments for Pilon were produced using <font color="red"><strong>BWA mem</strong></font> (v0.7.17) with default settings. <em>k</em>-mer spectrum comparisons were made using the <em>k</em>-mer analysis toolkit (<font color="red"><strong>KAT</strong></font>) (version 2.3.4) [13] (KAT comp) at a <em>k</em>-mer size of 32, and <font color="blue">the metric used to assess each iteration was the number of <strong>k-mers shared between</strong> the <strong>assembly</strong> and the <strong>Illumina</strong> reads</font>. </p>
<p>In a second consensus phase, <font color="blue">RNA-Seq</font> reads were aligned as single end (SE), to the genome using <font color="red"><strong>Hisat</strong></font> (version 2.1.0)  with default parameters. This time the effectiveness of consensus calling was <font color="blue">assessed by analysis of <strong>full-length alignments</strong> of assembled RNA-Seq transcripts</font>. All transcripts designated as “complete” by Evigene  were aligned to the genome with BLAT  (version 3.4, with [34] minimum match identity 90%). Alignments were filtered to <strong>retain only full-length alignments</strong> (i.e., from query start to query end). </p>
<p>Finally, the number of gaps in the alignments (query gaps + target gaps) was used as a metric with the rationale that this serves as a proxy for the number of indels in alignments of assembled messenger RNA sequence.</p>
<p><strong>Scaffold clustering and genome anchoring using Hi-C</strong></p>
<p>Hi-C reads were aligned to the polished scaffolds in CB with <font color="red"><strong>Bowtie2</strong></font> (version 2.3.3.1) . Each of the paired reads was aligned independently; then these SE alignments were subsequently merged as recommended by the LACHESIS developers. Based on the alignments, <font color="red">CB scaffolds were <strong>arranged</strong> into <strong>17 ordered</strong> and <strong>oriented</strong> <strong>clusters</strong> using the <strong>LACHESIS</strong> software</font>. As an <font color="blue">internal check</font>, the process was completed on <font color="blue">2 different random 95% sub-samplings of the Hi-C data</font>, as well as on the full data set. The clusters produced by <strong>all 3</strong> of these LACHESIS runs were <font color="blue">identical</font>. <font color="blue">LACHESIS produces groups of scaffolds</font> that are ordered and oriented relative to each other. These scaffold groupings were <font color="blue">compared with the genetic map</font> and the <strong>consistency</strong> of these sources of information was <strong>assessed</strong>. <font color="blue">The SNP probe mapping </font>at the scaffold <strong>validation step</strong> was compared with the clusters produced by LACHESIS.</p>
<p><strong>Illumina assembly</strong></p>
<p>The Illumina data were also assembled on their own, using the de Bruijn graph–based assembler <font color="red"><strong>SOAPdenovo2</strong></font> (version 2.04). <font color="blue"><strong>This assembly</strong></font> was <font color="blue"><strong>used in various ways</strong></font> during the course of the pear genome project (for further scaffold validation, for training the <em>ab initio</em> gene predictors, etc.). <font color="red"><strong>【借鉴】The Illumina data were assembled twice</strong></font>. The <font color="red">first pass</font> contigs were screened using the Kraken software and an index built from the entire RefSeq database. <font color="red">Reads aligning to contaminant contigs were <strong>removed</strong></font> and the remaining data were assembled again.</p>
<h1 id="15-2021-05-nbt-人-【四倍体土豆参考hic组装2】"><a href="#15-2021-05-nbt-人-【四倍体土豆参考hic组装2】" class="headerlink" title="15.  2021.05-nbt 人 【四倍体土豆参考hic组装2】"></a>15.  2021.05-nbt 人 【四倍体土豆参考hic组装2】</h1><blockquote>
<p>Chromosome-scale, haplotype-resolved assembly of human genomes</p>
</blockquote>
<p><img src="/blog/2022/03/14/2022-03-14-HiFi-6-HiC-assembly/10.png" alt="10"></p>
<p>outline of the phased assembly algorithm, <font color="red"><strong>DipAsm</strong></font></p>
<p>Assemble HiFi reads into unphased contigs using <font color="red"><strong>Peregrine</strong></font> (1);</p>
<p> <font color="blue"><strong>group</strong> and <strong>order</strong></font> contigs into scaffolds with <font color="blue"><strong>Hi-C</strong> data</font> using <font color="blue"><strong>Hirise/3D-DNA</strong></font> (3D de novo assembly) (2);</p>
<p> <font color="blue"><strong>map HiFi reads</strong> to scaffolds</font> and <font color="blue"><strong>call heterozygous SNPs</strong></font> using <font color="red"><strong>DeepVariant</strong></font> (3);</p>
<p> <font color="blue"><strong>phase</strong></font> heterozygous SNP calls with both HiFi and Hi-C data using <font color="red"><strong>WhatsHap plus HapCUT2</strong></font> (4);</p>
<p> partition reads based on their phase using <font color="red"><strong>WhatsHap</strong></font> (5);</p>
<p> assemble partitioned reads into <font color="blue"><strong>phased contigs</strong></font> using <font color="red"><strong>Peregrine</strong></font> (6).</p>
<p><strong>Phased sequence assembly.</strong> </p>
<p>We ran <font color="red"><strong>Peregrine v.0.1.5.2</strong></font> with the following command line: ‘peregrine asm reads.lst 24 24 24 24 24 24 24 24 24 –with-consensus –shimmer-r 3 –best_n_ovlp 8–output asm’, where file ‘reads. lst’ gives the list of input read files and directory ‘asm’ holds the output assembly. </p>
<p>We <font color="blue"><strong>mapped Hi-C reads to contigs</strong></font> with <font color="red"><strong>BWA-MEM</strong></font> v.0.7.17 and <font color="blue"><strong>scaffolded</strong></font> the Peregrine contigs with <font color="red"><strong>juicer v.1.5</strong></font> and <font color="red"><strong>3D-DNA v.180922</strong></font>. We preprocessed data with <font color="green"><strong>‘juicer.sh -d juicer -p chrom.sizes -y cut-sites.txt -z contigs.fa -D’</strong></font>, where file <strong>‘cut-sites.txt’</strong> was <strong>generated</strong> using the <strong>generate_site_positions_Arima. py</strong> script, which <strong>outputs merged_nodups.txt</strong>. The scaffolds were produced with <font color="green"><strong>‘run-asm-pipeline.sh -m haploid contigs.fa merged_nodups.txt’</strong></font>. </p>
<p>We then called small variants using DeepVariant v.0.8.0 with the pretrained PacBio model. </p>
<p>We mapped Hi-C reads to the scaffolds and ran HapCUT2 v.1.1 over heterozygous SNP sites to obtain sparse phasing at the chromosome scale. The resulting haplotypes were then combined with PacBio HiFi data using WhatsHap v.0.18, with default parameters, to generate fine-scale, chromosome-long phasing. </p>
<p>We partitioned HiFi reads based on the phases of SNPs residing on these reads, and ran Peregrine again for reads on the same haplotype from the same scaffold. This provided the final phased assembly.</p>
<h1 id="16-2018-11-ng-同源多倍体甘蔗-【四倍体土豆参考hic组装3】"><a href="#16-2018-11-ng-同源多倍体甘蔗-【四倍体土豆参考hic组装3】" class="headerlink" title="16.  2018.11-ng 同源多倍体甘蔗 【四倍体土豆参考hic组装3】"></a>16.  2018.11-ng 同源多倍体甘蔗 【四倍体土豆参考hic组装3】</h1><blockquote>
<p>Allele-defined genome of theautopolyploid sugarcane <em>Saccharum spontaneum</em> L.</p>
</blockquote>
<p><strong>材料</strong></p>
<p>栽培甘蔗的<font color="blue">单倍型个体</font><em>S.spontaneum</em>“AP85-441”(1n = 4x = 32) 进行基因组<em>denovo</em>测序研究。<br>64份甘蔗（<em>Saccharum spontaneum</em>）群体材料进行<font color="blue">遗传多样性研究</font>。</p>
<p><strong>方法</strong></p>
<p>二代280-bp和500-bp(PE) 小片段文库（Illumina HiSeq 2500）<br>BAC文库（Illumina HiSeq 2500，PE250）；三代大片段文库~20kb SMRTbellTM（<font color="blue">PacbioRSII</font>）<br>Hi-C文库（<font color="blue">HindIII酶切</font>），500-700bp小片段文库（HiSeqX ten，PE150）</p>
<p><strong>主要结果</strong>：</p>
<p><strong>甘蔗基因组denovo</strong><br>利用BAC文库进行二代测序，ALLPATH-LG,SPAdes 和SOAPdenovo2进行初步组装，进一步利用<font color="blue">87X 三代PacBio数据进行<strong>纠错</strong></font>，<font color="red"><strong>CANU</strong></font>组装，组装基因组3.13G，<font color="blue">contigN50=45 kb</font>（流式评估基因组大小：3.36G）；通过Hi-C染色体挂载技术，针对<font color="red"><strong>多倍体染色挂载最新算法ALLHIC</strong></font>，进一步将contig水平提升至染色体水平，<strong>挂载率达到93%<strong>；通过BUSCO和二代数据回比评估基因组完整性高达</strong>97.3%<strong>。再通过<font color="red"><strong>MAKER</strong></font>对同源多倍体甘蔗<font color="blue">特异等位基因进行</font></strong>手动注释</strong>，以获得高质量的甘蔗基因组</p>
<p>**基础染色体数目的减少 *<em>禾本科染色体数进化（高粱n= 10到甘蔗n= 8）<br>AP85-441基因组的组装显示了</em>S.spontaneum*的染色体数目从10降到8，而这与频繁复制的古复制染色体对相关，通过与高粱的聚类比对，发现高粱祖先5染色体和8号染色体同源物经历了染色体裂变（图2）。SbChr05（A12）的祖先染色体断裂分为两个主要部分，即C5S（A12S）和C5L（A12L），分别转移到SbChr06（A2）和SbChr07（A5）的祖先染色体；SbChr8（A11）的祖先染色体断裂为两个主要的部分，即C8S（A11S）和C8L（A11L），分别转移到SbChr09（A6）和SbChr02（A7+ A9）的祖先染色体中。SbChr8和SsChr5之间及SbChr5和SsChr7之间近乎同源的短片段是在高粱与甘蔗分化前，高粱SSA形成于13.4MYA同源基因的残留物，同时发现，S5中较小的SSA区域和S8中SSA的较大区域在重排的AP85-441基因组中也是保守的。</p>
<p><strong>S.spontaneum的多倍体化分析</strong><br>进一步研究发现了甘蔗（<em>Saccharum spontaneum</em> L.）染色体由高粱（<em>Sorghum</em>）演变进化过程中经历了染色体断裂与融合后，发生了两次WGD事件（图3）；对甘蔗的基因组的多倍化研究中发现，通过两次全基因组复制事件的发生，导致了甘蔗染色体的自发加倍过程；对同源多倍体甘蔗的特异等位基因的鉴定作为本研究的另一亮点，文章在鉴定等位基因的同时，进一步分析研究了等位基因的表达模式，结果发现不同单倍型表达模式相似，并无明显差异；对甘蔗C4光合作用的研究中鉴定了24个基因，7种关键酶参与NADP-MEC4途径；在甘蔗中蔗糖对的积累的研究中发现与高粱相比甘蔗中的液泡膜糖转运蛋白（TSTs，一种蔗糖转运蛋白）发生了基因家族的扩张，所以推测TSTs参与蔗糖在液泡中的积累；<em>S.spontaneum</em>为现代栽培甘蔗提供了抗病性基因，在对其研究中发现，甘蔗中的80％的NBS编码基因位于四个重排染色体（Ss02，Ss05，Ss06和Ss07）上，其中51％位于重排区域。</p>
<p><strong>S.spontaneum的起源与遗传多样性分析</strong><br>通过对64份甘蔗（<em>Saccharum spontaneum</em>）群体材料遗传多样性研究，发现其具有广泛的自然部分范围，遍及亚洲，印度次大陆，地中海和非洲，且自然种群具有广泛的表型，遗传和倍性水平多样性（图4）。研究发现，与高粱相比具有大规模染色体重排的<em>S. spontaneum</em>区域具有比非重排区域更高的遗传多样性，并且可能经历了更强的平衡选择。虽然几条单独的染色体没有显示出显著差异，但是比较所有染色体上的平均值显示重排区域中的核苷酸多样性（0.00025±0.00003）远高于非重排区域中的核苷酸多样性（0.00021±0.00001），染色体重排区域的Tajima’s D（-0.659±0.052）远远高于非重排区（-0.720±0.011），重排区的SNP密度(360.27±48.41) 高于非重排区域（297.46±12.65, P=0.001798）；此外，GO富集显示，染色体非重排区主要富集到与基本生命周期，光合作用，呼吸作用和ATP合成的通路上。重排区主要富集在GO生物合成和代谢，跨膜转运和离子结合通路上。</p>
<p><em>S. spontaneum</em>重排区具有高度的遗传多样性将更适应环境压力，例如对各种非生物胁迫（干旱，盐度，碱性，金属离子等）的反应，这些受到次生细胞生物合成和代谢，跨膜转运和离子结合的控制的基因在这些区域可以检测到。在多倍化事件后，重排区域经历了更强的平衡选择。</p>
<p><strong>讨论</strong></p>
<p>本研究利用复杂基因组Hi-C建库测序技术，并结合ALLHiC算法成功组装了高复杂同源多倍体甘蔗基因组AP85-441，高质量的甘蔗基因组的获得，为后续研究其蔗糖含量，抗性等性状及农业育种基础奠定了基础。</p>
<p><strong>Genome assembly overview.</strong> </p>
<p>The sugarcane AP85-441 contig-level assembly incorporated sequencing data from a mixture of sequencing technologies (Supplementary Fig. 1), including BAC pools sequenced with Illumina HiSeq 2500 and whole-genome shotgun sequencing with PacBio RS II as well as Hi-C reads, followed by Illumina short reads polishing. Each BAC pool was independently assembled using <font color="red">ALLPATHS-LG, SPAdes and SOAPdenovo2</font>, and best results were retained. For PacBio assembly, <font color="red"><strong>Canu v1.511</strong></font> was used, as it is capable<br>of avoiding collapsed repetitive regions and haplotypes. Self-correction was performed with parameter corOutCoverage = 100, which allowed us to correct all of the input PacBio reads. The corrected reads, along with BAC-assembled contigs, were imported to the assembly step. <font color="blue">Chromosomal assembly</font> was constructed based on proximity-guided assembly using our newly developed program, <font color="red"><strong>ALLHIC</strong>, which is designed for <strong>polyploid</strong> genome scaffolding</font> (see Supplementary Note for details).</p>
<p><img src="/blog/2022/03/14/2022-03-14-HiFi-6-HiC-assembly/11.png" alt="11"></p>
<h1 id="17-2019-08-np-ALLHiC【四倍体土豆参考hic组装4】"><a href="#17-2019-08-np-ALLHiC【四倍体土豆参考hic组装4】" class="headerlink" title="17.  2019.08-np ALLHiC【四倍体土豆参考hic组装4】"></a>17.  2019.08-np ALLHiC【四倍体土豆参考hic组装4】</h1><blockquote>
<p>Assembly of allele-aware, chromosomal-scale autopolyploid genomes based on Hi-C data</p>
</blockquote>
<p><strong>existing approaches are mostly designed for diploid genomes and often with the aim of reconstructing a haploid representation, thereby having limited power to reconstruct chromosomes for autopolyploid genomes. We developed a novel algorithm (ALLHiC) that is capable of building allele-aware, chromosomal-scale assembly for <font color="red">auto-polyploid genomes</font> using Hi-C paired-end reads with innovative ‘prune’ and ‘optimize’ steps. Application on simulated data showed that ALLHiC can phase allelic contigs and substantially improve ordering and orientation when compared to other mainstream Hi-C assemblers</strong></p>
<h1 id="18-2020-05-nc-紫花苜蓿同源四倍体-06【四倍体土豆参考hic组装5】"><a href="#18-2020-05-nc-紫花苜蓿同源四倍体-06【四倍体土豆参考hic组装5】" class="headerlink" title="18. 2020.05-nc 紫花苜蓿同源四倍体-06【四倍体土豆参考hic组装5】"></a>18. 2020.05-nc 紫花苜蓿同源四倍体-06【四倍体土豆参考hic组装5】</h1><h1 id="19-2020-10-ng-二倍体土豆-【四倍体土豆参考hic组装6】"><a href="#19-2020-10-ng-二倍体土豆-【四倍体土豆参考hic组装6】" class="headerlink" title="19. 2020.10-ng 二倍体土豆 【四倍体土豆参考hic组装6】"></a>19. 2020.10-ng 二倍体土豆 【四倍体土豆参考hic组装6】</h1><blockquote>
<p>Haplotype-resolved genome analyses of a heterozygous diploid potato</p>
</blockquote>
<p>ref of 12: Hi-C-genetic-maps-for-<font color="red">correction</font></p>
<p>马铃薯(Solanum tuberosum L.)是全球重要的块茎作物。人们正在努力将这种作物从无性繁殖的四倍体转化为种子繁殖的、自交系为基础的杂交作物，但这一过程需要对马铃薯基因组有更好的了解。</p>
<p><strong>Genome assembly and phasing using PacBio CCS reads.</strong> </p>
<p>A total of 29 Gb CCS reads were assembled using <font color="red"><strong>Canu</strong></font> (v1.91) with the <font color="green"><strong>parameter</strong> –pacbio-hifi</font>.<br>Canu <font color="blue">generated <strong>two</strong> assemblies</font> composed of contigs and unitigs (Supplementary Table 4), and the <font color="blue"><strong>unitig</strong> assembly consisted of the <strong>contigs that split at any alternative paths</strong> in the assembly graph</font>. The contig assembly had longer continuity but more chimeric fragments嵌合体片段 as revealed in the genetic mapping analysis. To avoid the mis-joining of two haplotypes, <font color="red"><strong>the unitig assembly</strong></font> rather than the contig assembly was chosen for the <font color="red"><strong>subsequent analysis</strong></font>. The unitigs were then <font color="blue"><strong>polished</strong> iteratively using <strong>two rounds</strong></font> of <font color="red"><strong>Pilon</strong></font> with ~150 Gb of WGS <strong>Illumina</strong> data, generating the genome <strong>draft RHgv2</strong>.</p>
<p>Similarly, the sequenced reads of RH selfing progeny后代 were mapped to unitigs of RHgv2 to perform <font color="blue"><strong>genetic grouping</strong></font>. Because the unitigs were relatively long (N50 = 2 Mb), windows with a size of 200 kb rather than the whole unitig were used. If the adjacent windows of one unitig showed contrary read distribution, the unitig was defined as chimeric and broken between windows; 40 chimeric unitigs with a total length of 95 Mb were broken. In total, <font color="blue">1.31 Gb of 1.53 Gb sequences were assigned to 24 linkage groups</font>.</p>
<p>After merging, 141 Mb sequences and 5,252 annotated genes of RHgv1 were added to the RHgv2, yielding a 1.67 Gb genome draft with 1.54 Gb sequences assigned to 24 groups, termed RHgv3 (Supplementary Table 6). The sequences from the RHgv1 and RHgv2 assemblies were named as ontctg* and unitig* in the AGP file, respectively (Supplementary Data 3).</p>
<p><strong>Construction of pseudochromosomes.</strong> </p>
<p>As no approach generated satisfactory results on the RH genome, we introduced the group information derived from the genetic mapping to assist the Hi-C application on chromosome-level assembly. The process was performed on RHgv3 including three steps as follows:</p>
<ol>
<li><font color="blue"><strong>Align</strong></font>. The 24 previously determined groups were divided into <font color="blue"><strong>two haplotypes</strong> to <strong>generate</strong> <strong>two</strong> <strong>pseudohaploid genome drafts</strong></font>. The 132 Mb sequences that could not be assigned to any group were added to two pseudohaploid genomes. Total <font color="blue">Hi-C reads were <strong>aligned</strong> to <strong>each</strong> pseudohaploid genome using <strong>HiC-Pro</strong> to calculate the <strong>contact frequency</strong></font>. This step <font color="blue">yielded <strong>two bam files</strong> for the two pseudohaploid genomes</font>.</li>
<li><font color="blue"><strong>Rescue</strong></font>. Using the <font color="blue">bam file as input</font>, the <em>rescue</em> function in <font color="red"><strong>ALLHiC</strong></font> was applied to assign unplaced sequences to known groups. Because the 132 Mb <strong>unplaced sequences</strong> were <strong>added</strong> to two pseudohaploid genomes and <strong>processed twice</strong>, the rescued <strong>results were redundant</strong>. For every <strong>unplaced sequence</strong>, we <strong>considered its best Hi-C signal density</strong> to decide the group to which it belonged. After this step, the sequence content of 24 groups was updated with an <strong>extra 75.6 Mb</strong> sequences assigned to proper groups.</li>
<li><font color="blue"><strong>Optimize and build</strong></font>. For each pseudohaploid genome, using <font color="red">the bam file and the updated group file as input</font>, the <font color="red"><strong><em>optimize</em> function in ALLHiC</strong></font> decidedthe <strong>order</strong> and <strong>orientation</strong> of scaffolds for each group; thus, the <font color="red"><strong><em>build</em> function</strong></font> generated fasta sequences on that basis. By performing this step, we identified the pseudochromosomes for 24 groups. The order and orientation of scaffolds on chromosomes are provided in Supplementary Data 3.</li>
</ol>
<h1 id="20-2021-12-gb-杏树【四倍体土豆参考hic组装7】"><a href="#20-2021-12-gb-杏树【四倍体土豆参考hic组装7】" class="headerlink" title="20. 2021.12-gb 杏树【四倍体土豆参考hic组装7】"></a>20. 2021.12-gb 杏树【四倍体土豆参考hic组装7】</h1><blockquote>
<p>Gamete binning: chromosome-level and haplotype-resolved genome assembly enabled by high-throughput single-cell sequencing of gamete genomes</p>
</blockquote>
<p>ref of 12: Hi-C-genetic-maps-for-correction</p>
<p>一种基于单倍体配子单细胞测序的方法。基于445个个体花粉粒的全基因组测序，组装了一株二倍体杏树的两个基因组。</p>
<p><strong>Genome size estimation</strong></p>
<p>After trimming off 10x Genomics barcodes and hexamers from the 61.7 Gb reads of the two 10x sc-CNV libraries, k-mer counting (k=21) was performed with <font color="red"><strong>Jellyfish</strong></font>. The k-mer histogram was provided to<font color="red"> <strong>findGSE</strong></font> to estimate the size of the “Rojo Pasión” genome under the heterozygous mode (with “exp_hom=200”; Add- itional file 1: Fig. S3).</p>
<p><strong>Preliminary diploid-genome assembly and curation</strong></p>
<p>With the 19.9 G raw PacBio reads of “Rojo Pasión” (Additional file 1: Fig. S2), a pre-liminary diploid assembly was constructed using <font color="red"><strong>Canu</strong></font> (with options “genome- Size=242500000 corMhapSensitivity=high corMinCoverage=0 corOutCoverage=100 correctedErrorRate=0.105”).</p>
<p>All raw <strong>Illumina reads</strong> from the 10x libraries were <font color="red">firstly <strong>aligned</strong> to the <strong>initial assembly</strong></font> using <font color="red"><strong>bowtie2</strong></font>. Then, the purge haplotigs pipeline was used to remove haplotigs (i.e., haplotype-specific contigs inflating the true haploid genome) based on statistical analysis of sequencing depth and identify primary contigs to build up a curated haploid assembly. To reduce the false-positive rate in defining haplotigs, each haplotig was blasted to the curated assembly; if over 50% of the haplotig could not be covered by any primary contigs, it was re-collected as a primary contig.</p>

    </div>

    
    
    
        

<div>
<ul class="post-copyright">
  <li class="post-copyright-author">
    <strong>Post author:  </strong>Wang C
  </li>
  <li class="post-copyright-link">
    <strong>Post link: </strong>
    <a href="https://wangc1106.github.io/blog/2022/03/14/2022-03-14-HiFi-6-HiC-assembly/" title="HiFi-6-[HiFi-HiC-assembly-method]">https://wangc1106.github.io/blog/2022/03/14/2022-03-14-HiFi-6-HiC-assembly/</a>
  </li>
  <li class="post-copyright-license">
    <strong>Copyright Notice:  </strong>All articles in this blog are licensed under <a href="https://creativecommons.org/licenses/by-nc-sa/4.0/" rel="noopener" target="_blank"><i class="fab fa-fw fa-creative-commons"></i>BY-NC-SA</a> unless stating additionally.
  </li>
</ul>
</div>


      <footer class="post-footer">
          <div class="post-tags">
              <a href="/blog/tags/denovo-III/" rel="tag"># denovo-III</a>
          </div>

        


        
    <div class="post-nav">
      <div class="post-nav-item">
    <a href="/blog/2022/03/09/2022-03-09-HiFi-5-HiC-Pro/" rel="prev" title="HiFi-5-[HiC-Pro]-[MCscan]-[ALLHiC]">
      <i class="fa fa-chevron-left"></i> HiFi-5-[HiC-Pro]-[MCscan]-[ALLHiC]
    </a></div>
      <div class="post-nav-item"></div>
    </div>
      </footer>
    
  </article>
  
  
  



          </div>
          

<script>
  window.addEventListener('tabs:register', () => {
    let { activeClass } = CONFIG.comments;
    if (CONFIG.comments.storage) {
      activeClass = localStorage.getItem('comments_active') || activeClass;
    }
    if (activeClass) {
      let activeTab = document.querySelector(`a[href="#comment-${activeClass}"]`);
      if (activeTab) {
        activeTab.click();
      }
    }
  });
  if (CONFIG.comments.storage) {
    window.addEventListener('tabs:click', event => {
      if (!event.target.matches('.tabs-comment .tab-content .tab-pane')) return;
      let commentClass = event.target.classList[1];
      localStorage.setItem('comments_active', commentClass);
    });
  }
</script>

        </div>
          
  
  <div class="toggle sidebar-toggle">
    <span class="toggle-line toggle-line-first"></span>
    <span class="toggle-line toggle-line-middle"></span>
    <span class="toggle-line toggle-line-last"></span>
  </div>

  <aside class="sidebar">
    <div class="sidebar-inner">

      <ul class="sidebar-nav motion-element">
        <li class="sidebar-nav-toc">
          Table of Contents
        </li>
        <li class="sidebar-nav-overview">
          Overview
        </li>
      </ul>

      <!--noindex-->
      <div class="post-toc-wrap sidebar-panel">
          <div class="post-toc motion-element"><ol class="nav"><li class="nav-item nav-level-1"><a class="nav-link" href="#1-2016-05-np-%E7%A2%A7%E5%86%AC%E8%8C%84"><span class="nav-number">1.</span> <span class="nav-text">1. 2016.05-np 碧冬茄</span></a><ol class="nav-child"><li class="nav-item nav-level-2"><a class="nav-link" href="#Genome-sequencing-assembly-and-annotation"><span class="nav-number">1.1.</span> <span class="nav-text">Genome sequencing, assembly and annotation.</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#Repetitive-elements-analysis"><span class="nav-number">1.2.</span> <span class="nav-text">Repetitive elements analysis</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#Gene-data-mining"><span class="nav-number">1.3.</span> <span class="nav-text">Gene data mining</span></a></li></ol></li><li class="nav-item nav-level-1"><a class="nav-link" href="#2-2017-04-ng-%E5%B1%B1%E7%BE%8A%F0%9F%90%90"><span class="nav-number">2.</span> <span class="nav-text">2. 2017.04-ng 山羊🐐</span></a><ol class="nav-child"><li class="nav-item nav-level-2"><a class="nav-link" href="#data"><span class="nav-number">2.1.</span> <span class="nav-text">data</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#%E7%BB%84%E8%A3%85%E7%AD%96%E7%95%A5%E5%8F%8A%E7%BB%84%E8%A3%85%E7%BB%93%E6%9E%9C"><span class="nav-number">2.2.</span> <span class="nav-text">组装策略及组装结果</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#%E7%BB%84%E8%A3%85%E8%AF%84%E4%BC%B0%E5%8F%8A%E6%AF%94%E8%BE%83%E5%88%86%E6%9E%90"><span class="nav-number">2.3.</span> <span class="nav-text">组装评估及比较分析</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#%E5%9F%BA%E5%9B%A0%E7%BB%84%E6%B3%A8%E9%87%8A"><span class="nav-number">2.4.</span> <span class="nav-text">基因组注释</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#%E5%90%AF%E7%A4%BA%E5%BD%95"><span class="nav-number">2.5.</span> <span class="nav-text">启示录</span></a></li></ol></li><li class="nav-item nav-level-1"><a class="nav-link" href="#3-2020-09-cell-%E5%A4%A7%E8%B1%86%E6%B3%9B%E5%9F%BA%E5%9B%A0%E7%BB%84"><span class="nav-number">3.</span> <span class="nav-text">3. 2020.09-cell 大豆泛基因组</span></a><ol class="nav-child"><li class="nav-item nav-level-2"><a class="nav-link" href="#SNP-Calling-and-Phylogenetic-Analyses"><span class="nav-number">3.1.</span> <span class="nav-text">SNP Calling and Phylogenetic Analyses</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#Genome-Assembly"><span class="nav-number">3.2.</span> <span class="nav-text">Genome Assembly</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#Repeat-Analysis-and-Gene-Annotation"><span class="nav-number">3.3.</span> <span class="nav-text">Repeat Analysis and Gene Annotation</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#Synteny-Analysis"><span class="nav-number">3.4.</span> <span class="nav-text">Synteny Analysis</span></a></li></ol></li><li class="nav-item nav-level-1"><a class="nav-link" href="#4-2019-nc-%E5%A4%A7%E8%B1%86%E6%B3%9B%E5%9F%BA%E5%9B%A0%E7%BB%84-%E7%BB%84%E8%A3%85%E5%8F%82%E8%80%831"><span class="nav-number">4.</span> <span class="nav-text">4. 2019-nc 大豆泛基因组-组装参考1</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#5-2019-09-scienceChina-%E5%A4%A7%E8%B1%86%E6%B3%9B%E5%9F%BA%E5%9B%A0%E7%BB%84-%E7%BB%84%E8%A3%85%E5%8F%82%E8%80%832"><span class="nav-number">5.</span> <span class="nav-text">5. 2019.09-scienceChina 大豆泛基因组-组装参考2</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#6-2020-05-nc-%E5%90%8C%E6%BA%90%E5%9B%9B%E5%80%8D%E4%BD%93%E7%B4%AB%E8%8A%B1%E8%8B%9C%E8%93%BF"><span class="nav-number">6.</span> <span class="nav-text">6. 2020.05-nc 同源四倍体紫花苜蓿</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#7-2021-11-gb-%E6%A4%B0%E5%AD%90%F0%9F%A5%A5"><span class="nav-number">7.</span> <span class="nav-text">7. 2021.11-gb 椰子🥥</span></a><ol class="nav-child"><li class="nav-item nav-level-2"><a class="nav-link" href="#Hi-C-based-genome-assembly"><span class="nav-number">7.1.</span> <span class="nav-text">Hi-C-based genome assembly</span></a></li></ol></li><li class="nav-item nav-level-1"><a class="nav-link" href="#8-2021-11-nc-%E9%87%91%E7%B2%9F%E5%85%B0%E5%9F%BA%E5%9B%A0%E7%BB%84"><span class="nav-number">8.</span> <span class="nav-text">8. 2021.11-nc 金粟兰基因组</span></a><ol class="nav-child"><li class="nav-item nav-level-2"><a class="nav-link" href="#Genome-size-estimate-and-assembly"><span class="nav-number">8.1.</span> <span class="nav-text">Genome size estimate and assembly.</span></a></li></ol></li><li class="nav-item nav-level-1"><a class="nav-link" href="#9-2022-01-gb-%E6%B2%B9%E8%8C%B6"><span class="nav-number">9.</span> <span class="nav-text">9. 2022.01-gb 油茶</span></a><ol class="nav-child"><li class="nav-item nav-level-2"><a class="nav-link" href="#%E6%8A%B1%E5%AD%90%E6%80%80%E8%83%8E%E7%9A%84%E6%B2%B9%E8%8C%B6%E8%89%AF%E7%A7%8D%E8%82%B2%E7%A7%8D%E7%BC%93%E6%85%A2"><span class="nav-number">9.1.</span> <span class="nav-text">抱子怀胎的油茶良种育种缓慢</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#%E6%9F%93%E8%89%B2%E4%BD%93%E7%BA%A7%E5%88%AB%E7%BB%84%E8%A3%85"><span class="nav-number">9.2.</span> <span class="nav-text">染色体级别组装</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#Genome-sequencing-de-novo-assembly-and-annotation"><span class="nav-number">9.3.</span> <span class="nav-text">Genome sequencing, de novo assembly, and annotation</span></a></li></ol></li><li class="nav-item nav-level-1"><a class="nav-link" href="#10-2022-01-HorticRes-%E7%94%98%E8%8F%8A"><span class="nav-number">10.</span> <span class="nav-text">10. 2022.01-HorticRes 甘菊</span></a><ol class="nav-child"><li class="nav-item nav-level-2"><a class="nav-link" href="#Genome-assembly-and-quality-assessment"><span class="nav-number">10.1.</span> <span class="nav-text">Genome assembly and quality assessment</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#Hi-C-sequencing-and-assistant-assembly"><span class="nav-number">10.2.</span> <span class="nav-text">Hi-C sequencing and assistant assembly</span></a></li></ol></li><li class="nav-item nav-level-1"><a class="nav-link" href="#11-2022-01-HorticRes-%E7%B2%89%E8%91%9B"><span class="nav-number">11.</span> <span class="nav-text">11. 2022.01-HorticRes 粉葛</span></a><ol class="nav-child"><li class="nav-item nav-level-2"><a class="nav-link" href="#Genome-survey"><span class="nav-number">11.1.</span> <span class="nav-text">Genome survey</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#Genome-assembly"><span class="nav-number">11.2.</span> <span class="nav-text">Genome assembly</span></a></li></ol></li><li class="nav-item nav-level-1"><a class="nav-link" href="#12-2022-03-ng-%E5%9B%9B%E5%80%8D%E4%BD%93%E5%9C%9F%E8%B1%86"><span class="nav-number">12.</span> <span class="nav-text">12. 2022.03-ng 四倍体土豆</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#13-2022-03-ng-%E5%85%AD%E5%80%8D%E4%BD%93%E5%B0%8F%E9%BA%A6"><span class="nav-number">13.</span> <span class="nav-text">13. 2022.03-ng 六倍体小麦</span></a><ol class="nav-child"><li class="nav-item nav-level-2"><a class="nav-link" href="#Genome-assembly-and-validation"><span class="nav-number">13.1.</span> <span class="nav-text">Genome assembly and validation.</span></a></li></ol></li><li class="nav-item nav-level-1"><a class="nav-link" href="#14-2019-10-%E4%BA%8C%E5%80%8D%E4%BD%93%E5%B7%B4%E7%89%B9%E5%88%A9%E7%89%B9%E6%A2%A8-%E3%80%90%E5%9B%9B%E5%80%8D%E4%BD%93%E5%9C%9F%E8%B1%86%E5%8F%82%E8%80%83hic%E7%BB%84%E8%A3%851%E3%80%91"><span class="nav-number">14.</span> <span class="nav-text">14.  2019.10 二倍体巴特利特梨 【四倍体土豆参考hic组装1】</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#15-2021-05-nbt-%E4%BA%BA-%E3%80%90%E5%9B%9B%E5%80%8D%E4%BD%93%E5%9C%9F%E8%B1%86%E5%8F%82%E8%80%83hic%E7%BB%84%E8%A3%852%E3%80%91"><span class="nav-number">15.</span> <span class="nav-text">15.  2021.05-nbt 人 【四倍体土豆参考hic组装2】</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#16-2018-11-ng-%E5%90%8C%E6%BA%90%E5%A4%9A%E5%80%8D%E4%BD%93%E7%94%98%E8%94%97-%E3%80%90%E5%9B%9B%E5%80%8D%E4%BD%93%E5%9C%9F%E8%B1%86%E5%8F%82%E8%80%83hic%E7%BB%84%E8%A3%853%E3%80%91"><span class="nav-number">16.</span> <span class="nav-text">16.  2018.11-ng 同源多倍体甘蔗 【四倍体土豆参考hic组装3】</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#17-2019-08-np-ALLHiC%E3%80%90%E5%9B%9B%E5%80%8D%E4%BD%93%E5%9C%9F%E8%B1%86%E5%8F%82%E8%80%83hic%E7%BB%84%E8%A3%854%E3%80%91"><span class="nav-number">17.</span> <span class="nav-text">17.  2019.08-np ALLHiC【四倍体土豆参考hic组装4】</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#18-2020-05-nc-%E7%B4%AB%E8%8A%B1%E8%8B%9C%E8%93%BF%E5%90%8C%E6%BA%90%E5%9B%9B%E5%80%8D%E4%BD%93-06%E3%80%90%E5%9B%9B%E5%80%8D%E4%BD%93%E5%9C%9F%E8%B1%86%E5%8F%82%E8%80%83hic%E7%BB%84%E8%A3%855%E3%80%91"><span class="nav-number">18.</span> <span class="nav-text">18. 2020.05-nc 紫花苜蓿同源四倍体-06【四倍体土豆参考hic组装5】</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#19-2020-10-ng-%E4%BA%8C%E5%80%8D%E4%BD%93%E5%9C%9F%E8%B1%86-%E3%80%90%E5%9B%9B%E5%80%8D%E4%BD%93%E5%9C%9F%E8%B1%86%E5%8F%82%E8%80%83hic%E7%BB%84%E8%A3%856%E3%80%91"><span class="nav-number">19.</span> <span class="nav-text">19. 2020.10-ng 二倍体土豆 【四倍体土豆参考hic组装6】</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#20-2021-12-gb-%E6%9D%8F%E6%A0%91%E3%80%90%E5%9B%9B%E5%80%8D%E4%BD%93%E5%9C%9F%E8%B1%86%E5%8F%82%E8%80%83hic%E7%BB%84%E8%A3%857%E3%80%91"><span class="nav-number">20.</span> <span class="nav-text">20. 2021.12-gb 杏树【四倍体土豆参考hic组装7】</span></a></li></ol></div>
      </div>
      <!--/noindex-->

      <div class="site-overview-wrap sidebar-panel">
        <div class="site-author motion-element" itemprop="author" itemscope itemtype="http://schema.org/Person">
    <img class="site-author-image" itemprop="image" alt="Wang C"
      src="/blog/images/banner.png">
  <p class="site-author-name" itemprop="name">Wang C</p>
  <div class="site-description" itemprop="description">Make more time</div>
</div>
<div class="site-state-wrap motion-element">
  <nav class="site-state">
      <div class="site-state-item site-state-posts">
          <a href="/blog/archives">
          <span class="site-state-item-count">32</span>
          <span class="site-state-item-name">posts</span>
        </a>
      </div>
      <div class="site-state-item site-state-categories">
            <a href="/blog/categories/">
          
        <span class="site-state-item-count">11</span>
        <span class="site-state-item-name">categories</span></a>
      </div>
      <div class="site-state-item site-state-tags">
            <a href="/blog/tags/">
          
        <span class="site-state-item-count">18</span>
        <span class="site-state-item-name">tags</span></a>
      </div>
  </nav>
</div>
  <div class="links-of-author motion-element">
      <span class="links-of-author-item">
        <a href="https://github.com/wangc1106" title="GitHub → https:&#x2F;&#x2F;github.com&#x2F;wangc1106" rel="noopener" target="_blank"><i class="fab fa-github fa-fw"></i>GitHub</a>
      </span>
      <span class="links-of-author-item">
        <a href="mailto:wangchen116@126.com" title="E-Mail → mailto:wangchen116@126.com" rel="noopener" target="_blank"><i class="fa fa-envelope fa-fw"></i>E-Mail</a>
      </span>
  </div>
  <div class="cc-license motion-element" itemprop="license">
    <a href="https://creativecommons.org/licenses/by-nc-sa/4.0/" class="cc-opacity" rel="noopener" target="_blank"><img src="/blog/images/cc-by-nc-sa.svg" alt="Creative Commons"></a>
  </div>



      </div>
        <div class="back-to-top motion-element">
          <i class="fa fa-arrow-up"></i>
          <span>0%</span>
        </div>

    </div>
  </aside>
  <div id="sidebar-dimmer"></div>


      </div>
    </main>

    <footer class="footer">
      <div class="footer-inner">
        

        

<div class="copyright">
  
  &copy; 2021 – 
  <span itemprop="copyrightYear">2022</span>
  <span class="with-love">
    <i class="fas fa-dna"></i>
  </span>
  <span class="author" itemprop="copyrightHolder">Wang C</span>
</div>

        






<script>
  (function() {
    function leancloudSelector(url) {
      url = encodeURI(url);
      return document.getElementById(url).querySelector('.leancloud-visitors-count');
    }

    function addCount(Counter) {
      var visitors = document.querySelector('.leancloud_visitors');
      var url = decodeURI(visitors.id);
      var title = visitors.dataset.flagTitle;

      Counter('get', '/classes/Counter?where=' + encodeURIComponent(JSON.stringify({ url })))
        .then(response => response.json())
        .then(({ results }) => {
          if (results.length > 0) {
            var counter = results[0];
            leancloudSelector(url).innerText = counter.time + 1;
            Counter('put', '/classes/Counter/' + counter.objectId, { time: { '__op': 'Increment', 'amount': 1 } })
              .catch(error => {
                console.error('Failed to save visitor count', error);
              });
          } else {
              Counter('post', '/classes/Counter', { title, url, time: 1 })
                .then(response => response.json())
                .then(() => {
                  leancloudSelector(url).innerText = 1;
                })
                .catch(error => {
                  console.error('Failed to create', error);
                });
          }
        })
        .catch(error => {
          console.error('LeanCloud Counter Error', error);
        });
    }

    function showTime(Counter) {
      var visitors = document.querySelectorAll('.leancloud_visitors');
      var entries = [...visitors].map(element => {
        return decodeURI(element.id);
      });

      Counter('get', '/classes/Counter?where=' + encodeURIComponent(JSON.stringify({ url: { '$in': entries } })))
        .then(response => response.json())
        .then(({ results }) => {
          for (let url of entries) {
            let target = results.find(item => item.url === url);
            leancloudSelector(url).innerText = target ? target.time : 0;
          }
        })
        .catch(error => {
          console.error('LeanCloud Counter Error', error);
        });
    }

    let { app_id, app_key, server_url } = {"enable":true,"app_id":"UqIYufmNk2sE8MiBKKWc4Kja-gzGzoHsz","app_key":"iYpeWQcmNqpHwr2uVlnJYVji","server_url":null,"security":false};
    function fetchData(api_server) {
      var Counter = (method, url, data) => {
        return fetch(`${api_server}/1.1${url}`, {
          method,
          headers: {
            'X-LC-Id'     : app_id,
            'X-LC-Key'    : app_key,
            'Content-Type': 'application/json',
          },
          body: JSON.stringify(data)
        });
      };
      if (CONFIG.page.isPost) {
        if (CONFIG.hostname !== location.hostname) return;
        addCount(Counter);
      } else if (document.querySelectorAll('.post-title-link').length >= 1) {
        showTime(Counter);
      }
    }

    let api_server = app_id.slice(-9) !== '-MdYXbMMI' ? server_url : `https://${app_id.slice(0, 8).toLowerCase()}.api.lncldglobal.com`;

    if (api_server) {
      fetchData(api_server);
    } else {
      fetch('https://app-router.leancloud.cn/2/route?appId=' + app_id)
        .then(response => response.json())
        .then(({ api_server }) => {
          fetchData('https://' + api_server);
        });
    }
  })();
</script>


      </div>
      <div style="height: 200px">
      <script type="text/javascript" src="//rf.revolvermaps.com/0/0/8.js?i=5knf34gs07t&amp;m=0&amp;c=ff0000&amp;cr1=ffffff&amp;f=arial&amp;l=33" async="async"></script>
      </div>
    </footer>
  </div>

  
  <script src="/blog/lib/anime.min.js"></script>
  <script src="/blog/lib/velocity/velocity.min.js"></script>
  <script src="/blog/lib/velocity/velocity.ui.min.js"></script>

<script src="/blog/js/utils.js"></script>

<script src="/blog/js/motion.js"></script>


<script src="/blog/js/schemes/pisces.js"></script>


<script src="/blog/js/next-boot.js"></script>




  




  
<script src="/blog/js/local-search.js"></script>













  

  

</body>
</html>
