<!DOCTYPE html>
<html lang="zh-Hans">
<head>
  <meta charset="UTF-8">
<meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=2">
<meta name="theme-color" content="#222">
<meta name="generator" content="Hexo 5.4.0">
  <link rel="apple-touch-icon" sizes="180x180" href="/blog/images/apple-touch-icon-next.png">
  <link rel="icon" type="image/png" sizes="32x32" href="/blog/images/favicon-32x32-next.png">
  <link rel="icon" type="image/png" sizes="16x16" href="/blog/images/favicon-16x16-next.png">
  <link rel="mask-icon" href="/blog/images/logo.svg" color="#222">
  <meta name="google-site-verification" content="1osTo_gOe1SW9Hevqtz8vl8HfSgTO4DSMZ0DiMPMX6w">

<link rel="stylesheet" href="/blog/css/main.css">


<link rel="stylesheet" href="/blog/lib/font-awesome/css/all.min.css">

<script id="hexo-configurations">
    var NexT = window.NexT || {};
    var CONFIG = {"hostname":"wangc1106.github.io","root":"/blog/","scheme":"Gemini","version":"7.8.0","exturl":false,"sidebar":{"position":"left","display":"post","padding":18,"offset":12,"onmobile":false},"copycode":{"enable":false,"show_result":false,"style":null},"back2top":{"enable":true,"sidebar":true,"scrollpercent":"ture"},"bookmark":{"enable":false,"color":"#222","save":"auto"},"fancybox":false,"mediumzoom":false,"lazyload":false,"pangu":false,"comments":{"style":"tabs","active":null,"storage":true,"lazyload":false,"nav":null},"algolia":{"hits":{"per_page":10},"labels":{"input_placeholder":"Search for Posts","hits_empty":"We didn't find any results for the search: ${query}","hits_stats":"${hits} results found in ${time} ms"}},"localsearch":{"enable":true,"trigger":"auto","top_n_per_article":1,"unescape":false,"preload":false},"motion":{"enable":true,"async":false,"transition":{"post_block":"fadeIn","post_header":"slideDownIn","post_body":"slideDownIn","coll_header":"slideLeftIn","sidebar":"slideUpIn"}},"path":"search.xml"};
  </script>

  <meta name="description" content="辅助组装案例">
<meta property="og:type" content="article">
<meta property="og:title" content="HiFi-6-[HiFi-HiC-assembly-method]">
<meta property="og:url" content="https://wangc1106.github.io/blog/2022/03/14/2022-03-14-HiFi-6-HiC-assembly/index.html">
<meta property="og:site_name" content="Banner&#39;s Blog">
<meta property="og:description" content="辅助组装案例">
<meta property="og:locale">
<meta property="og:image" content="https://wangc1106.github.io/blog/2022/03/14/2022-03-14-HiFi-6-HiC-assembly/1.png">
<meta property="article:published_time" content="2022-03-14T08:09:34.778Z">
<meta property="article:modified_time" content="2022-03-14T13:51:43.936Z">
<meta property="article:author" content="Wang C">
<meta property="article:tag" content="denovo-III">
<meta name="twitter:card" content="summary">
<meta name="twitter:image" content="https://wangc1106.github.io/blog/2022/03/14/2022-03-14-HiFi-6-HiC-assembly/1.png">

<link rel="canonical" href="https://wangc1106.github.io/blog/2022/03/14/2022-03-14-HiFi-6-HiC-assembly/">


<script id="page-configurations">
  // https://hexo.io/docs/variables.html
  CONFIG.page = {
    sidebar: "",
    isHome : false,
    isPost : true,
    lang   : 'zh-Hans'
  };
</script>

  <title>HiFi-6-[HiFi-HiC-assembly-method] | Banner's Blog</title>
  






  <noscript>
  <style>
  .use-motion .brand,
  .use-motion .menu-item,
  .sidebar-inner,
  .use-motion .post-block,
  .use-motion .pagination,
  .use-motion .comments,
  .use-motion .post-header,
  .use-motion .post-body,
  .use-motion .collection-header { opacity: initial; }

  .use-motion .site-title,
  .use-motion .site-subtitle {
    opacity: initial;
    top: initial;
  }

  .use-motion .logo-line-before i { left: initial; }
  .use-motion .logo-line-after i { right: initial; }
  </style>
</noscript>

</head>

<body itemscope itemtype="http://schema.org/WebPage">
  <div class="container use-motion">
    <div class="headband"></div>

    <header class="header" itemscope itemtype="http://schema.org/WPHeader">
      <div class="header-inner"><div class="site-brand-container">
  <div class="site-nav-toggle">
    <div class="toggle" aria-label="Toggle navigation bar">
      <span class="toggle-line toggle-line-first"></span>
      <span class="toggle-line toggle-line-middle"></span>
      <span class="toggle-line toggle-line-last"></span>
    </div>
  </div>

  <div class="site-meta">

    <a href="/blog/" class="brand" rel="start">
      <span class="logo-line-before"><i></i></span>
      <h1 class="site-title">Banner's Blog</h1>
      <span class="logo-line-after"><i></i></span>
    </a>
      <p class="site-subtitle" itemprop="description">Dream big work hard</p>
  </div>

  <div class="site-nav-right">
    <div class="toggle popup-trigger">
        <i class="fa fa-search fa-fw fa-lg"></i>
    </div>
  </div>
</div>




<nav class="site-nav">
  <ul id="menu" class="main-menu menu">
        <li class="menu-item menu-item-home">

    <a href="/blog/" rel="section"><i class="home fa-fw"></i>Home</a>

  </li>
        <li class="menu-item menu-item-categories">

    <a href="/blog/categories/" rel="section"><i class="th fa-fw"></i>Categories</a>

  </li>
        <li class="menu-item menu-item-tags">

    <a href="/blog/tags/" rel="section"><i class="tags fa-fw"></i>Tags</a>

  </li>
        <li class="menu-item menu-item-about">

    <a href="https://wangc1106.github.io/" rel="section"><i class="user fa-fw"></i>About</a>

  </li>
      <li class="menu-item menu-item-search">
        <a role="button" class="popup-trigger"><i class="fa fa-search fa-fw"></i>Search
        </a>
      </li>
  </ul>
</nav>



  <div class="search-pop-overlay">
    <div class="popup search-popup">
        <div class="search-header">
  <span class="search-icon">
    <i class="fa fa-search"></i>
  </span>
  <div class="search-input-container">
    <input autocomplete="off" autocapitalize="off"
           placeholder="Searching..." spellcheck="false"
           type="search" class="search-input">
  </div>
  <span class="popup-btn-close">
    <i class="fa fa-times-circle"></i>
  </span>
</div>
<div id="search-result">
  <div id="no-result">
    <i class="fa fa-spinner fa-pulse fa-5x fa-fw"></i>
  </div>
</div>

    </div>
  </div>

</div>
    </header>

    


    <main class="main">
      <div class="main-inner">
        <div class="content-wrap">
          

          <div class="content post posts-expand">
            

    
  
  
  <article itemscope itemtype="http://schema.org/Article" class="post-block" lang="zh-Hans">
    <link itemprop="mainEntityOfPage" href="https://wangc1106.github.io/blog/2022/03/14/2022-03-14-HiFi-6-HiC-assembly/">

    <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
      <meta itemprop="image" content="/blog/images/banner.png">
      <meta itemprop="name" content="Wang C">
      <meta itemprop="description" content="Make more time">
    </span>

    <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
      <meta itemprop="name" content="Banner's Blog">
    </span>
      <header class="post-header">
        <h1 class="post-title" itemprop="name headline">
          HiFi-6-[HiFi-HiC-assembly-method]
        </h1>

        <div class="post-meta">
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-calendar"></i>
              </span>
              <span class="post-meta-item-text">Posted on</span>
              

              <time title="Created: 2022-03-14 16:09:34 / Modified: 21:51:43" itemprop="dateCreated datePublished" datetime="2022-03-14T16:09:34+08:00">2022-03-14</time>
            </span>
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-folder"></i>
              </span>
              <span class="post-meta-item-text">In</span>
                <span itemprop="about" itemscope itemtype="http://schema.org/Thing">
                  <a href="/blog/categories/Software/" itemprop="url" rel="index"><span itemprop="name">Software</span></a>
                </span>
            </span>

          
            <span id="/blog/2022/03/14/2022-03-14-HiFi-6-HiC-assembly/" class="post-meta-item leancloud_visitors" data-flag-title="HiFi-6-[HiFi-HiC-assembly-method]" title="Views">
              <span class="post-meta-item-icon">
                <i class="fa fa-eye"></i>
              </span>
              <span class="post-meta-item-text">Views: </span>
              <span class="leancloud-visitors-count"></span>
            </span>
            <span class="post-meta-item" title="Symbols count in article">
              <span class="post-meta-item-icon">
                <i class="far fa-file-word"></i>
              </span>
              <span>17k</span>
            </span>
            <span class="post-meta-item" title="Reading time">
              <span class="post-meta-item-icon">
                <i class="far fa-clock"></i>
              </span>
              <span>16 mins.</span>
            </span>

        </div>
      </header>

    
    
    
    <div class="post-body" itemprop="articleBody">

      
        <p>辅助组装案例</p>
<span id="more"></span>

<h1 id="1-2016-05-np-碧冬茄"><a href="#1-2016-05-np-碧冬茄" class="headerlink" title="1. 2016.05-np 碧冬茄"></a>1. 2016.05-np 碧冬茄</h1><p><strong>Insight into the evolution of the Solanaceae from the parental genomes of Petunia hybrida</strong></p>
<h2 id="Genome-sequencing-assembly-and-annotation"><a href="#Genome-sequencing-assembly-and-annotation" class="headerlink" title="Genome sequencing, assembly and annotation."></a>Genome sequencing, assembly and annotation.</h2><p>Plants were grown and DNA was extracted following the methods described at Supplementary Note 1.</p>
<p>Illumina libraries with 0.17-, 0.35-, 0.5-, 0.8-, 1-, 2-, 5-, 8- and 15-kb inserts were sequenced at BGI-Shenzghen and University of Illinois, Roy J. Carver Biotechnology. PacBio P. axillaris DNA library was sequenced with P4/C2 chemistry.</p>
<p>Illumina reads were processed using <strong>Fastq-mcf</strong> (<strong>quality filtering</strong>; <a target="_blank" rel="noopener" href="https://code.google.com/p/eautils/wiki/FastqMcf">https://code.google.com/p/eautils/wiki/FastqMcf</a>), <strong>PRINSEQ</strong> (<strong>duplication filtering</strong>; http:// prinseq.sourceforge.net/) and <strong>Musket</strong> (<strong>error correction</strong>; <a target="_blank" rel="noopener" href="http://musket.sourceforge/">http://musket.sourceforge</a>. net/). <strong>Pacbio reads</strong> were processed using the <strong>SMRT Analysis pipeline</strong> (v.2.0.1; <a target="_blank" rel="noopener" href="https://github.com/PacificBiosciences/SMRT-Analysis">https://github.com/PacificBiosciences/SMRT-Analysis</a>).</p>
<p>Both genomes were assembled with <strong>SOAPdenovo</strong> with different k-mer sizes. For both genomes, <strong>k-mer = 79</strong> showed the best statistics. Gaps between contigs were completed using <strong>GapCloser</strong> . Additionally for P. axillaris, <strong>PacBio reads</strong> were integrated in <strong>four different steps</strong>: (1) <strong>Rescaffolding</strong> of the <strong>Illumina contigs</strong> using the <strong>PacBio reads</strong> and the <strong>AHA assembler</strong>; (2) <strong>Gap filling</strong> using <strong>PBJelly</strong>; (3) <strong>Rescaffolding</strong> using the Illumina <strong>pair data</strong> and <strong>SSPACE</strong>; (4) Last round of <strong>gap filling</strong> using <strong>PBJelly</strong> . </p>
<p><strong>Genome size estimation</strong> was performed through the k-mers abundance distribution (<strong>k-mer = 31</strong>). <strong>Heterozygosity</strong> was <strong>estimated</strong> mapping the Illumina reads to the assemblies using <strong>Bowtie2</strong>, calling SNPs using <strong>FreeBayes</strong> and  annotating the SNPs using <strong>SnpEff</strong> .</p>
<p>The genome structural annotation was performed using <strong>Maker-P</strong>: (1) <strong>SNAP</strong> and <strong>Augustus</strong> as ab initio gene predictors; (2) Exonerate as experimental based predictor with 454 and Illumina RNASeq reads and protein sequences from different protein datasets. RNAseq Illumina data was mapped using <strong>Tophat2</strong> . tRNAs were annotated using <strong>tRNAscan</strong> (<a target="_blank" rel="noopener" href="http://lowelab.ucsc.edu/tRNAscan-SE/">http://lowelab.ucsc.edu/tRNAscan-SE/</a>).</p>
<p>The <strong>gene functional annotation</strong> was performed by <strong>sequence homology search</strong> with different protein datasets using <strong>BlastP</strong> and protein domains search using <strong>InterProScan</strong> . Functional annotations were integrated using <strong>AHRD</strong> (<a target="_blank" rel="noopener" href="https://github.com/groupschoof/AHRD">https://github.com/groupschoof/AHRD</a>). See Supplementary Note 1.</p>
<h2 id="Repetitive-elements-analysis"><a href="#Repetitive-elements-analysis" class="headerlink" title="Repetitive elements analysis"></a>Repetitive elements analysis</h2><p>Repeat annotation was performed using RepeatModeler (v1.0.8; <a target="_blank" rel="noopener" href="http://www.repeatmasker.org/RepeatModeler.html">http://www.repeatmasker.org/RepeatModeler.html</a>), RepeatMasker (v4.0.5; <a target="_blank" rel="noopener" href="http://www.repeatmasker.org/">http://www.repeatmasker.org</a>) with the repeat database Repbase (release 20140131; <a target="_blank" rel="noopener" href="http://www.girinst.org/repbase/">http://www.girinst.org/repbase/</a>) and Geneious (v7.1.4; <a target="_blank" rel="noopener" href="http://www.geneious.com/">http://www.geneious.com</a>). Identification of PVCV-like and EPRV elements was performed using BlastN and TBlastN . The identified sequences were aligned with ClustalW (MEGA5 package;<a target="_blank" rel="noopener" href="http://www.megasoftware.net/">http://www.megasoftware.net/</a>) and then manually curated. RepeatExplorer (<a target="_blank" rel="noopener" href="http://www.repeatexplorer.org/">http://www.repeatexplorer.org/</a>) and other methods were used to extend the analysis to unassembled repeats. Fluorescent in situ hybridization was performed in root tips from young P. axillaris and P. inflata plants for 5S rDNA and three PVCV viral probes following the procedure described in Supplementary Note 2.</p>
<p>The detection of dTph1 loci in P. hybrida W138 was performed through a BLAST40 search of the P. axillaris and P. inflata dTph1 elements including the 500 bp of flanking sequence against the TFS W138 collection. Polymorphisms found in the genomic flanking regions were used to identify the species of origin. dTph1 elements were identified in a P. axillaris population using a modification of the methodology described in Supplementary Note 3.</p>
<p>##Whole-genome duplication, tandem duplications and gene family analysis</p>
<p>Whole-genome collinear analysis was performed using SynMap and microsynteny analysis were performed using GEvo in the comparative genomics platform, CoGe .See Supplementary Note 5.</p>
<p>The gene family analysis included Solanum lycopersicum, S. tuberosum, Nicotiana benthamiana and Arabidopsis thaliana protein sets using BlastP (v2.2.27) on an all-versus-all comparison and grouping the genes into families with OrthoMCL, v2.0.8. See Supplementary Note 4.</p>
<h2 id="Gene-data-mining"><a href="#Gene-data-mining" class="headerlink" title="Gene data mining"></a>Gene data mining</h2><p>The specific identification of genes for P. axillaris and P. inflata genomes for colour and scent, root-specific pathways, self-incompatibility and circadian clock was performed through a BlastN/BlastP sequence homology search. Blast GUI, JBrowser (<a target="_blank" rel="noopener" href="http://jbrowse.org/">http://jbrowse.org/</a>) and WebApollo (<a target="_blank" rel="noopener" href="http://genomearchitect/">http://genomearchitect</a>. org/) were installed in a server to search and manually curate the gene structures of the identified genes. See Supplementary Notes 7, 8 and 10–12.</p>
<h1 id="2-2017-04-ng-山羊🐐"><a href="#2-2017-04-ng-山羊🐐" class="headerlink" title="2. 2017.04-ng 山羊🐐"></a>2. 2017.04-ng 山羊🐐</h1><p>Published in final edited form as:Nat Genet. 2017 April ; 49(4): 643–650. doi:10.1038/ng.3802.</p>
<p><strong>Single-molecule sequencing and chromatin conformation capture enable</strong> <strong>de novo</strong> <strong>reference assembly of the domestic goat genome</strong></p>
<p>​        首次在人之外的物种中直接de novo组装出染色体水平的基因组。利用<font color="blue"><strong>Pacbio+BioNano+Hi-C+Illumina</strong> 多层级</font>的策略，组装了一只山羊(Capra hircus)的基因组，获得<font color="blue"><strong>31</strong>个<strong>scaffold</strong>，<strong>663</strong>个<strong>gap</strong>区，scaffold <strong>N50</strong> 惊人地达到<strong>87M</strong>，几乎<strong>one scaffold one chromosome</strong></font>，并且成功地组装出<font color="blue"><strong>免疫基因区</strong>和<strong>大部分重复序列家族</strong></font>。</p>
<p>​        <font color="red">人的GRCh38版本有<strong>24</strong>个<strong>scaffold</strong>，<strong>169</strong>个<strong>unplacedscaffold</strong>，<strong>832个gap</strong>区</font>。</p>
<p>​        从<strong>96头</strong>山羊(<strong>6个品种</strong>)中，利用Illumina的 Caprine53K <font color="blue"><strong>SNP 芯片筛选</strong></font>出<font color="blue"><strong>基因型纯和度最高</strong></font>的个体用来<font color="blue"><strong>组装</strong></font>(San Clemente breed)。</p>
<h2 id="data"><a href="#data" class="headerlink" title="data"></a>data</h2><ol>
<li><p>Pacbio数据</p>
<p>一共产生465个SMRT cell的数据。其中使用P5-C3试剂的311个cell;使用P4-C2试剂的142个cell，XL-C2试剂的12个cell。数据量194G，覆盖深度达<strong>69X</strong>，<strong>subread</strong>的<strong>平均</strong>长度为<strong>5,110bp</strong>。</p>
</li>
<li><p>BioNano 数据</p>
<p>由于采样所用山羊意外死亡，其DNA不能满足Irsy optical mapping 测序需求，只好用其<strong>雄性后代</strong>采样。Optical map一共产生256Gb数据，覆盖<strong>98 X</strong>。</p>
</li>
<li><p>Hi-C数据</p>
<p>采样动物是最初那只山羊。Hi-C建库，序列用物理方法打断成300-500bp长度，PE101测序，产生<strong>115M reads</strong>的数据量。</p>
</li>
<li><p>Illumina 数据</p>
<p>采样最初那只山羊，PE251建库测序，获得<strong>23X</strong>的数据覆盖，用来做最后的<font color="red"><strong>错误校正</strong></font></p>
</li>
</ol>
<h2 id="组装策略及组装结果"><a href="#组装策略及组装结果" class="headerlink" title="组装策略及组装结果"></a><strong>组装策略及组装结果</strong></h2><ul>
<li><p>先用Pacbio的 long-read 数据构建contig， 使用<font color="blue"><strong>Celera Assembler</strong> PacBio corrected Reads 流程</font>组装。获得<font color="blue"><strong>3,074</strong>个contig(2.63G)，<strong>N50 4.159M</strong>b</font>。</p>
</li>
<li><p>接下来采用<font color="blue"><strong>Irys optical mapping</strong>数据构建<strong>scaffold</strong></font>，使用软件为<font color="blue"><strong>IrysView</strong></font>. 产生<font color="blue"><strong>842</strong>个scaffold，scaffold N50为<strong>13.4M</strong>b，contig N50为10.858M，最长scaffold为66.728Mb</font>。 利用<font color="red"><strong>radiation hybrid(RH) map比较</strong>发现</font>，<font color="blue">PacBio-Irys 联合</font>的方法已经<font color="blue">完整组装出<strong>20号染色体</strong></font>。</p>
</li>
<li><p>接下来用Hi-C数据，调用用<font color="red"><strong>Lachesis软件包</strong></font>，结合PacBio-Irys的结果，即整合的PacBio-Irys-PGA (PBIP)方法，获得较完美的组装结果。<font color="blue">Scaffold N50</font> 达到惊人的<font color="blue"><strong>87.347Mb</strong></font>(远超今年尖吻鲈，大猩猩N50 20Mb的高起点)，总共获得<font color="blue"><strong>31条</strong>scaffold</font>，(<font color="red">31,261条&lt;50kbp的**degenerate scaffold**不计入</font> )。</p>
</li>
<li><p>最后利用Illumina数据做<font color="red"><strong>一致性校正</strong>和最后的<strong>补洞</strong></font>，最后使得contig的数目下降到680条，产生的gap数目为663个。</p>
</li>
<li><p>利用Kraken v0.10.5 <font color="red">去除有<strong>病毒和细菌污染</strong>的序列</font>，去掉有NCBI vector污染的序列。获得最终的基因组 版本<strong>ARS1</strong>。</p>
</li>
</ul>
<h2 id="组装评估及比较分析"><a href="#组装评估及比较分析" class="headerlink" title="组装评估及比较分析"></a><strong>组装评估及比较分析</strong></h2><p>由于山羊之前已经产生过两个基因组版本<strong>CHIR_1.0、CHIR_2.0</strong>，可以用来做比较 分析，并且有RH mapping数据用来评估、校正ARS1的组装版本。 <strong>通过各种参数比较发现，ARS1均优于CHIR_2.0版本。</strong></p>
<p>1、通过比对发现，CHIR_2.0 比CHIR_1.0有更少的putative deletion(2,735 vs 1 0,256)和duplication(115 vs 290)。而2.0版本相较1.0有更多的inversion(215 v s 4)。通过比较，发现ARS1版本有大幅度地提升，较2.0版本少4倍的deletion， 少50倍inversion。</p>
<p>2、CHIR_2.0能填补CHIR_1.0 94.6%的gap，而剩下的那些gap通过分析ARS1得知，是CHIR_1.0的组装错误，而RH数据也支持了这一论断。 3、BUSCO是利用单拷贝ortholog来评估组装的可靠性，和CHIR_2.0相比，ARS1有更高的BUSCO score。</p>
<p>4、山羊52k SNP 芯片中，有1,723个SNP探针目前只能定位在CHIR_2.0组装版本 中的unplaced contigs上。而ARS1版本能把其中的90%(1,552/1,723)以上的S NP探针定位在染色体上。另外发现有26个low call-rate 的SNPmarker在ARS1组装版本上定位比较模棱两可。这也就解释了为什么这些maker在芯片上有较差的call -rate. </p>
<p>5、通过比较发现，CHIR_2.0版本中的3,495个内含子或外显子有gap的基因，在ARS1中都得到补全。同时也确认1,926个预测的外显子在CHIR_1.0或者2.0中有gap， 但在ARS1中得到修复。 </p>
<p>6、由于免疫基因区高度多态性和重复性，用二代测序数据很难组装起来，但是通过 分析发现，ARS1版本就很好地把LRC和NKC基因(免疫功能相关基因)定位在一个独立的常染色体scaffold上。 </p>
<p>7、通过和一代、二代组装策略相比，ARS1版本在重复序列的组装上获得了极大进步。对某些异染色体或异染色质的区域也获得了较好的组装覆盖。比如，在6条常 染色体上组装出&gt;5kbp的端粒序列。通过分析发现， 15条染色体scaffold，在着丝粒区域组装出大于2kbp长度的重复区，而其中的7条，更是装出来8kbp以上的重复区。更令人惊讶的是，19号和23号染色体都组装出高度重复的着丝粒和端粒区域， 贯通了有结构性异染色质区的染色体。 </p>
<p>8、在ARS1版本中鉴定出大于12kbp的重复模式序列多达105条。对于重复序列家 族的鉴定，在ARS1中获得的接近全长的BovB LINE 序列比CHIR_2.0多60%以上。 在CHIR_2.0版本里，被ARS1成功补全的gap中的43.6%与BovB重复序列一致(长 度大于3.5kbp)，即意味着二代测序的gap区大多都是重复序列，用三代就能较容易测通。 </p>
<p>9、关于性染色体的分析。通过比对发现，两个不同的scaffold比对到X染色体的不同但连续的区域，占预期X染色体大小(150Mb)的86%左右。通过和自己的、跨物种牛的Y染色体比对，最终确定出10Mb区域的序列，占Y染色体预估大小的50% 左右。通过比对牛和羊的Y染色体上的基因，发现在目前的scaffold中能找到16%， 而在先前过滤掉的degenerate contig中能找到84%的基因。考虑到Y染色体的异染 色质属性及X,Y染色体的拟常染色区域经常发现交换，实在太过复杂，超出目前的组装和认知水平，表示实在无能为力。</p>
<p>10、利用之前RH mapping data，对组装的大部分环节进行了校正及评估。比如多次辅助解决scaffold conflict的问题。</p>
<h2 id="基因组注释"><a href="#基因组注释" class="headerlink" title="基因组注释"></a><strong>基因组注释</strong></h2><p>多种方法结合注释基因集。<br>1.<strong>RNA-seq的方法:</strong> 6个组织(大多和脑组织相关)RNA-seq测序、13个SRA下载 数据，利用stringtie、cufflinks和Trinity(基于无参组装)。最后用PASA软件整合在一起;</p>
<p>2.<strong>用exonerate和tblastn 软件</strong>比对到几个近缘物种的Ensembl基因集上，获得同 源预测基因集;</p>
<p>3.<strong>用Braker1做Ab initio 预测</strong>;</p>
<p>4.<strong>CHIR_1.0 版本的注释基因集</strong>; 最后用<strong>EVM+PASA</strong>把以上4种数据整合成一个最终的基因集(设置的权重为RNAseq &gt; cDNA/protein &gt; ab initio gene predictions)。</p>
<h2 id="启示录"><a href="#启示录" class="headerlink" title="启示录"></a><strong>启示录</strong></h2><p>1、Pacbio RSII是一个较为理想组装平台，产生平均14Kbp左右的读长，最高到60 Kbp，可以组装出较为理想的基因组版本。由于三代的长读长优势，同时可以较好地组装出基因组中大部分高度重复区(主要集中在着丝粒和端粒区域);</p>
<p>2、BioNano 的Irys optical mapping 是一种非常有效的，成本比较适中的构建scaffold的平台。和Hi-C比优势在于定位错误较少，缺点是对N50的提高并不是特别显著，一般在2倍左右。如果正反链的Nt.BsqI酶切位点离得比较近，常常导致双链 断裂，限制了Optical Map scaffold的大小。所以，BioNano对较长的scaffold效果比较好。由于Optical map和pacbio产生的错误特征并不一样，两者结合，可以 相互校正，获得较好的组装指标;</p>
<p>3、Hi-C是目前非常火爆的一个技术，在多个领域都有很好的应用。在组装方面， 它能获得染色体级别的scaffold，但是也容易获得较高的contig定向错误。这也是 为什么文章中，先用optical mapping，后用Hi-C做scaffolding的重要原因之一。 作者建议在以后的实验中，可以选择较短的识别位点的限制内切酶(或者DNase Hi -C)来提高Hi-C交联的密度，降低定向错误。</p>
<p>4、利用同样的方法和平台，做类似的基因组项目，用Pacbio平台和scaffolding平 台，目前大约花费在$100,000上下,而一般的短reads平台测序加scaffolding 平台，只需要其三分之一的价格。但是用Pacbio平台却获得了高质量的、连续性好的基因组序列。目前Pacbio 新一代的sequel平台已经推出，单个cell的产量比RSII提 高7倍以上，测序成本将会得到大幅度降低，相信会成为未来基因组研究的主流工具之一。</p>
<p>5、本文的研究策略确实会启迪后来的研究者，为de novo组装树立新标杆。Pacbio 测序技术联合多种scaffolding 平台(比如BioNano、Hi-C，mating-pairedlibr ary，10X genomics)的使用， 会引领de novo 组装进入 near finished genome 或者finished genome的新时代。</p>
<p>6、但是，尽管做了很多努力，ARS1仍然是一个单体型混合型的组装版本。在未来，可以利用single molecule和Hi-C技术获得单体型分期(haplotype phasing) 的参考基因组。对于组成型异染色质区，尤其是着丝粒和端粒部分，仍然是组装的盲区，即使在人的基因组中这些区域也没有被完整组装出来。</p>
<h1 id="3-2020-09-cell-大豆泛基因组"><a href="#3-2020-09-cell-大豆泛基因组" class="headerlink" title="3. 2020.09-cell 大豆泛基因组"></a>3. 2020.09-cell 大豆泛基因组</h1><p>Pan-Genome of Wild and Cultivated Soybeans</p>
<p>Soybean is one of the most important vegetable oil and protein feed crops. To capture the entire genomic diversity, it is needed to construct a complete high-quality pan-genome from diverse soybean accessions. In this study, we performed individual de novo genome assemblies for <strong>26 representative soybeans</strong> that were selected from <strong>2,898 deeply sequenced accessions</strong>. Using these assembled genomes together with <strong>three previously reported genomes</strong>, we constructed a <strong>graph-based genome</strong> and performed pan-genome analysis, which identified <strong>numerous genetic variations that cannot be detected by direct mapping of short sequence reads onto a single reference genome</strong>. The <strong>structural variations</strong> from the 2,898 accessions that were <strong>genotyped</strong> based on the graph-based genome and the <strong>RNA-seq data</strong> from the representative 26 accessions helped to <strong>link genetic variations to candidate genes</strong> that are responsible for important traits. This pan-genome resource will promote evolutionary and functional genomics studies in soybean.</p>
<h2 id="SNP-Calling-and-Phylogenetic-Analyses"><a href="#SNP-Calling-and-Phylogenetic-Analyses" class="headerlink" title="SNP Calling and Phylogenetic Analyses"></a>SNP Calling and Phylogenetic Analyses</h2><h2 id="Genome-Assembly"><a href="#Genome-Assembly" class="headerlink" title="Genome Assembly"></a>Genome Assembly</h2><p><em>De novo</em> assembly was conducted referring to a <font color="blue"><strong>reported pipeline</strong></font> (Du and Liang, 2019; Shen et al., 2019). In brief, <font color="red"><strong>Canu</strong></font> (Koren et al., 2017) v1.7.1 was used to assemble PacBio subreads to PacBio contigs, after which <font color="red"><strong>HiSeq</strong></font> reads were used for <font color="red"><strong>error correction</strong></font>. <strong>Bio-nano optical maps</strong> were assembled into consensus physical maps by <strong>BioNano Solve</strong> v3.0.1 (<a target="_blank" rel="noopener" href="https://bionanogenomics.com/">https://bionanogenomics.com/</a>, Solve_06082017Rel). Then <font color="red"><strong>HERA</strong></font> (Du and Liang, 2019) was used to combine PacBio contigs and Bionano based physical maps to <font color="blue"><strong>PacBio-BioNano hybrid scaffolds</strong></font>. To <font color="red"><strong>anchor</strong></font> hybrid scaffolds <font color="blue">into <strong>chromosomes</strong></font>, the <font color="blue"><strong>Hi-C</strong></font> sequencing data were <font color="red"><strong>aligned into scaffolds</strong></font> by <font color="red"><strong>Juicer</strong></font> (Durand et al., 2016) v1.5 and <font color="red"><strong>3D-DNA</strong></font> (Dudchenko et al., 2017).</p>
<h2 id="Repeat-Analysis-and-Gene-Annotation"><a href="#Repeat-Analysis-and-Gene-Annotation" class="headerlink" title="Repeat Analysis and Gene Annotation"></a>Repeat Analysis and Gene Annotation</h2><h2 id="Synteny-Analysis"><a href="#Synteny-Analysis" class="headerlink" title="Synteny Analysis"></a>Synteny Analysis</h2><p>Structural Variation Identification</p>
<p>Genetic Variation Analysis</p>
<p>Gene and miRNA Expression</p>
<p>Core and Dispensable Gene Family Clustering</p>
<p>CHS gene unit identification</p>
<p>Gene Fusion Event Identification</p>
<p>QUANTIFICATION AND STATISTICAL ANALYSIS</p>
<h1 id="4-2019-nc-大豆泛基因组-组装参考1"><a href="#4-2019-nc-大豆泛基因组-组装参考1" class="headerlink" title="4. 2019-nc 大豆泛基因组-组装参考1"></a>4. 2019-nc 大豆泛基因组-组装参考1</h1><p>HERA</p>
<p>#5. 2019.09-scienceChina 大豆泛基因组-组装参考2</p>
<p><img src="/blog/2022/03/14/2022-03-14-HiFi-6-HiC-assembly/1.png" alt="1"></p>
<p>For genome assembly, we sequenced additional 40 Gb SMRT reads from 5 cells for this update. Therefore, a total of <strong>120× PacBio reads</strong>, <strong>365× Bionano optical maps</strong> marked by BssSI, <strong>275× Bionano optical maps</strong> marked by BspQI, <strong>45× HiSeq</strong> reads and <strong>125×</strong> chromosome conformation capture sequencing (<strong>Hi-C</strong>) reads were used for the new genome assembly (Figure 1A). To make best of the sequencing data, we also <font color="red"><strong>adopted a new genome assembly pipeline</strong> by <strong>adding HERA</strong></font> (Du and Liang, 2018) to improve the sequence contiguity and reduce errors by accurately assembling the repetitive genome regions (Figure 1A). Briefly, <font color="blue">the whole assembly process <strong>differed from</strong> previous pipeline</font> at </p>
<p>(1) using <font color="red"><strong>CANU</strong></font> (v1.7.1) to replace Smrtmake for assembling PacBio subreads to PacBio contigs; </p>
<p>(2) using <font color="red"><strong>HERA</strong></font> to generate longer contigs; </p>
<p>(3) using <font color="red"><strong>Juicer and 3D-DNA to replace HiC-Pro and LACHESIS</strong></font> to anchor the hybrid scaffolds into chromosomes with Hi-C reads.</p>
<h1 id="6-2020-05-nc-同源四倍体紫花苜蓿"><a href="#6-2020-05-nc-同源四倍体紫花苜蓿" class="headerlink" title="6. 2020.05-nc 同源四倍体紫花苜蓿"></a>6. 2020.05-nc 同源四倍体紫花苜蓿</h1><p>HiFi助力同源四倍体紫花苜蓿基因组组装，NC</p>
<blockquote>
<p>Allele-aware chromosome-level genome assembly and efficient transgene-free genome editing for the autotetraploid cultivated alfalfa</p>
</blockquote>
<p>（1）使用Canu默认参数，利用CCS clean reads组装contigs。组装得到的Contig N50值为459kb，总长度为3.15GB。</p>
<blockquote>
<p>We assembled contigs from CCS clean reads using Canu, with default parameters. The N50 values of the contig sets were 459 kb, with total lengths 3154 Mb.</p>
</blockquote>
<p>（2）使用HiC-Pro将Hi-C reads与contigs 进行比对，产生比对BAM文件。</p>
<blockquote>
<p>Hi-C reads were aligned to contigs using <a target="_blank" rel="noopener" href="https://pubmed.ncbi.nlm.nih.gov/26619908/">HiC-Pro(2015)</a>, yielding an alignment BAM file.</p>
</blockquote>
<p>（3）使用注释的蒺藜苜蓿蛋白作为参考，完全基于同源的策略注释contigs。对138,729个同源基因进行了结构注释。用MCscan用于鉴定contigs和参考基因组之间的共线性。显示紫花苜蓿和蒺藜苜蓿之间的高共线性。【下载父母本petunia ref】</p>
<blockquote>
<p>Contigs were annotated with a <font color="red"><strong>solely homology-based strategy</strong></font>, using annotated Medicago truncatula proteins as references. 138,729 homologous genes were structurally annotated. <font color="red"><strong>MCscan</strong> in <strong>Jcvi</strong></font> (<a target="_blank" rel="noopener" href="https://zenodo.org/record/31631#.XpkUyTOeask">https://zenodo.org/record/31631#.XpkUyTOeask</a>) was used to identify <font color="red"><strong>synteny blocks</strong></font> between <strong>contigs and the reference</strong> genome. <font color="red"><strong>Contigs syntenic</strong> to M. truncatula were <strong>stacked and aligned</strong> to M. truncatula chromosomes</font>. The syntenic contigs are summarized in Supplementary Table 4.</p>
</blockquote>
<p>（4）使用内部脚本处理BAM文件，去除等位基因contigs之间的links。使用ALLHiC软件，<font color="red">提取、聚类和重排Contigs</font> (Contigs syntenic与蒺藜苜蓿染色体一致)，得到原始的scaffolds。</p>
<blockquote>
<p>An <strong>in-house script</strong> was used to <font color="red">prune the BAM</font> file and <font color="red">discard links between allelic contigs</font>. Contigs syntenic to one chromosome of M. truncatula, e.g., chr1, were extracted, sub-clustered and reordered using <font color="red">ALLHiC</font>, yielding a raw scaffold set.</p>
</blockquote>
<p>（5）<a target="_blank" rel="noopener" href="https://pubmed.ncbi.nlm.nih.gov/27467250/">Juicebox</a>用于以图形和交互方式微调组装的scaffolds。剪裁了40个总长度达1800Mb的scaffolds 。</p>
<blockquote>
<p>Juicebox45 was used for fine-tuning assembled scaffolds in a graphic and inter-active fashion. Forty scaffolds with a total length of 1800 Mb were cropped (Supplementary Table 5).</p>
</blockquote>
<p>（6） 基于组装的scaffold，通过Hi-C数据，每个unplaced contig被分配到互作最强的那些contig cluster里。</p>
<blockquote>
<p>Based on this scaffold assembly, each unplaced contig was assigned to the contig cluster, to which the contig was most connected by Hi-C data.</p>
</blockquote>
<p>（7）使用ALLHiC对那些contig clusters再次进行重排和构建scaffold。</p>
<blockquote>
<p>Those contig clusters were reordered and scaffolded using ALLHiC. </p>
</blockquote>
<p>（8）使用Juicebox对scaffolds进行微调，并从scaffolds上去除不一致的contigs，产生最终的染色体基因组，其包含32条染色体(8个同源组，每个组中有4个等位基因染色体)，总长度为2738Mb，和419Mb未挂载到染色体水平的序列。</p>
<blockquote>
<p>Using Juicebox, scaffolds were fine-tuned and discordant contigs were removed from scaffolds, and the final chromosome assembly was generated, containing 32 chromosomes with a total length of 2738 Mb (Supplementary Table 6).</p>
</blockquote>
<h1 id="7-2021-11-gb-椰子🥥"><a href="#7-2021-11-gb-椰子🥥" class="headerlink" title="7. 2021.11-gb 椰子🥥"></a>7. 2021.11-gb 椰子🥥</h1><p>High-quality reference genome sequences of two coconut cultivars provide insights into evolution of monocot chromosomes and differentiation of fiber content and plant height</p>
<p>该研究绘制两个参考级别椰子基因组。揭示单子叶植物染色体的进化过程和椰子染色体的形成过程，分析高矮两种椰子耐盐，株高，纤维和脂质含量等关键性状的遗传差异，同时发现椰子进化过程中约在400万年前自然发生的绿色革命。高椰子2.39G矮椰子2.40G。基于椰子，紫萍，菠萝等基因组的共线性关系，重新构建了含有10条染色体的古单子叶植物基因组核型，并对基因组核型的进化路线进行推断。进一步利用叶间距表型进行全基因组关联分析，鉴定出12号染色体上GA20ox基因与椰子株高显著相关，并存在拷贝数变异；整合多种手段验证HA20ox以及赤霉素代谢途径在高矮种的差异并导致株高差异。400万年前，绿色革命。</p>
<h2 id="Hi-C-based-genome-assembly"><a href="#Hi-C-based-genome-assembly" class="headerlink" title="Hi-C-based genome assembly"></a>Hi-C-based genome assembly</h2><p>Both Cn. tall and Cn. dwarf genomes were assembled de novo using SMARTdenovo (<a target="_blank" rel="noopener" href="https://github.com/ruanjue/smartdenovo">https://github.com/ruanjue/smartdenovo</a>) [-k 19 -J 3000] based on <font color="red"><strong>Nanopore long reads</strong></font> <strong>corrected</strong> with <font color="red"><strong>NextDenovo</strong></font> (<a target="_blank" rel="noopener" href="https://github.com/Nextomics/NextDenovo">https://github.com/Nextomics/NextDenovo</a>). Raw contigs were <font color="blue"><strong>polished</strong></font> using <font color="blue"><strong>Nanopore long reads</strong></font> and <font color="blue"><strong>Illumina short reads</strong></font> with <font color="red"><strong>Next-Polish</strong></font> (v1.2.2) [49]. Then polished contigs were <font color="blue"><strong>anchored</strong></font> into chromosomes by using <font color="red"><strong>HiC-Pro (v2.11.1)</strong></font> [50] and <font color="red"><strong>LACHESIS</strong></font>[51]. Finally, the assembled genomes were <font color="blue"><strong>manual correction</strong></font> with <font color="red"><strong>Juicebox</strong></font> (v1.11.08).</p>
<p>To assess the quality of assembled genomes, raw Illumina <font color="blue"><strong>paired-end</strong> reads and <strong>RNA-seq</strong></font> reads from multiple tissues were <font color="blue"><strong>mapping</strong></font> to the genomes using <font color="red"><strong>BWA (v0.7.17-r1188)</strong></font> [52] and <font color="red"><strong>HISAT2 (v2.1.0)</strong></font> [53], respectively. In addition, Benchmarking Universal Single-Copy Orthologs (<font color="red"><strong>BUSCO</strong></font>) was also used to assess the genome completeness base on Embryophyta Plant database (odb10) [54].</p>
<h1 id="8-2021-11-nc-金粟兰基因组"><a href="#8-2021-11-nc-金粟兰基因组" class="headerlink" title="8. 2021.11-nc 金粟兰基因组"></a>8. 2021.11-nc 金粟兰基因组</h1><p>填补核心被子植物最后主要分支-金粟兰目的基因组信息，提供被子植物早期演化证据。刘建全团队致力于解析类群演化，探究不同物种间分化适应遗传机制。</p>
<p>被子植物（Angiosperms or flowering plants）是地球上分布范围最广、多样化最高、适应性最强的陆生植物类群，极大地影响了其他生物类群的演化进程，然而，由于被子植物在早期经历了快速的辐射进化，其内部演化关系迟迟未被解决，这一问题也被称为达尔文的“恼人之谜”。经过前人长时期的努力，目前被子植物被划分为：无油樟目、睡莲目、木兰藤目和核心被子植物（Mesangiospermae）。无油樟目、睡莲目、木兰藤目共同被称为ANA grade，是被子植物的基部类群，系统发育关系相对稳定，而核心被子植物包含了~99.95%的被子植物，又可被划分为5大类群：双子叶、单子叶、木兰类、金鱼藻目和金粟兰目，它们之间的演化关系一直存在争议。近年来，随着各类群代表物种全基因组测序相继完成，其各自的进化历程均有了坚实而可靠的证据，刘建全团队在2020年首次报道了金鱼藻和芡实的基因组，证明了核心被子植物内部存在广泛的不完全谱系分选和杂交事件是导致树形冲突的主要原因。但是金粟兰目一直未有基因组报道，其演化历程以及整个核心被子植物的演化过程还有待深入分析。为此该团队再次利用Nanopore、Illumina和Hi-C技术，构建了第一个金粟兰目植物（四川金粟兰，<em>Chloranthus sessilifolius</em>）的高质量染色体级别基因组，Contig N50高达53.74 Mb（图1），使用多种分析方法对上述问题进行了深入的阐述。</p>
<p>通过使用多种系统发育基因组学分析，包括单拷贝基因、低拷贝基因和共线性基因数据集，进行了串联树和溯祖树的构建，同时评估了长枝吸引、物种选择和同源聚类方法等的影响，使用多套数据集合和多种分析方法，获得了高可信度的拓扑结构，即金粟兰目和木兰类具有最近的亲缘关系，且他们一起与金鱼藻目+双子叶植物构成姊妹关系，而单子叶植物是其它所有核心被子植物的姊妹枝（图2）。</p>
<p>同时，作者也发现了大量的基因树或叶绿体树与物种树不一致的情况，因此着重评估了不完全谱系分选（ILS）和杂交事件对此冲突的贡献。通过PhyloNetworks分析鉴定到了三次可能的杂交事件，可以解析部分的树形冲突。然而值得注意的是五大类群之间的分化时间仅有23个百万年（158~135 Mya），如此短的分化时间，ILS可能发挥了更为主要的作用。使用theta参数对每个内部分支发生ILS的概率进行评估发现，除了木兰类和金粟兰目的姊妹关系较为稳定外，其他类群间都具有较高的theta值，而金鱼藻目+双子叶的祖先分枝具有较高的tehta值，均表明了ILS在核心被子植物内部的演化中发挥了重要作用（图3 a和b）。基于Phyabse和DendroPy进行树形模拟表明，当存在ILS时模拟的树形和真实树形具有高度一致性（R2＞0.99），而不存在ILS的模拟树形与真实树形则具有较低的相关性，进一步表明了ILS在核心被子植物的快速辐射中发挥了重要作用（图3 c-f）。进一步对树形异质性进行区分表明，ILS可基本解释金鱼藻目+双子叶的祖先枝上的树形冲突，也就是金鱼藻目+双子叶、木兰类+金粟兰和单子叶之间的冲突，而其他核心被子植物类群间的冲突还有其他因素（例如杂交）的参与。</p>
<p>此外，本研究还鉴定了四川金粟兰独有的一次WGD事件，同时5大核心被子植物分枝的加倍事件都不共享；鉴定了花发育相关的基因家族，解析了其特有花器官的可能形成原因；鉴定了萜类合成和木质部形成相关的基因家族，为今后研究其简单木质部形成和萜类物种合成通路演化提供了基础。</p>
<p>综上所述，该研究报道了高质量的四川金粟兰基因组，填补了核心被子植物金粟兰目的基因组信息，并通过多种分析策略和方法，深入的揭示了核心被子植物间的复杂演化历程，该成果对于早期被子植物的起源、辐射进化以及适应性演化提供数据支撑和理论基础。</p>
<h2 id="Genome-size-estimate-and-assembly"><a href="#Genome-size-estimate-and-assembly" class="headerlink" title="Genome size estimate and assembly."></a>Genome size estimate and assembly.</h2><p>To estimate the genome size of C. sessilifolius, we surveyed 150 bp paired-end reads, computed 21 bp K-mer frequencies using <font color="red"><strong>Jellyfish</strong></font>, and exported the resulting histogram into <font color="red">findGSE</font>. <font color="red"><strong>Nextdenovo</strong></font> (https:// github.com/Nextomics/Nextdenovo) was selected for <font color="blue"><strong>correcting reads</strong></font> with para- meters “read_cutoff=2k, seed_cutoff=30k, blocksize=1.5g” and then <font color="red"><strong>Smartdenovo</strong></font> (<a target="_blank" rel="noopener" href="https://github.com/ruanjue/smartdenovo">https://github.com/ruanjue/smartdenovo</a>) for <font color="blue"><strong>de novo assembly</strong></font> with parameters “wtpre -J 3,000; wtzmo-k 21 -z 10 -Z 16 -U -1 -m 0.1 -A 1000; wtclp -d 3 -k 300 -m 0.1 -FT; wtlay -w 300 -s 200 -m 0.1 -r 0.95 -c 1”. <font color="blue">The preliminary contigs were further <strong>polished</strong> by aligning the Illumina <strong>short</strong> <strong>reads</strong> to the contigs</font> using <font color="red"><strong>Nextpolish</strong></font>. After four rounds of successive iterative correction, the final genome sequence was obtained. The GC content and sequencing coverage analyses were applied to evaluate the presence of contamination. The quality of the genome assembly was also assessed using <font color="red"><strong>BUSCO</strong></font> (Benchmarking Universal Single-Copy Orthologs) with the embry- ophyta_odb10 database. The clean <font color="blue">Hi-C</font> data were mapped to contig sequences by <font color="red"><strong>Bowtie2</strong></font> and 354 Mb valid interaction pairs were extracted. Based on those chromatin interactions, <font color="red"><strong>LACHESIS</strong></font> was employed to <font color="blue"><strong>cluster</strong>, <strong>order</strong>, and <strong>orient</strong> the <strong>contigs</strong> into <strong>pseudo-chromosomes</strong></font>.</p>

    </div>

    
    
    
        

<div>
<ul class="post-copyright">
  <li class="post-copyright-author">
    <strong>Post author:  </strong>Wang C
  </li>
  <li class="post-copyright-link">
    <strong>Post link: </strong>
    <a href="https://wangc1106.github.io/blog/2022/03/14/2022-03-14-HiFi-6-HiC-assembly/" title="HiFi-6-[HiFi-HiC-assembly-method]">https://wangc1106.github.io/blog/2022/03/14/2022-03-14-HiFi-6-HiC-assembly/</a>
  </li>
  <li class="post-copyright-license">
    <strong>Copyright Notice:  </strong>All articles in this blog are licensed under <a href="https://creativecommons.org/licenses/by-nc-sa/4.0/" rel="noopener" target="_blank"><i class="fab fa-fw fa-creative-commons"></i>BY-NC-SA</a> unless stating additionally.
  </li>
</ul>
</div>


      <footer class="post-footer">
          <div class="post-tags">
              <a href="/blog/tags/denovo-III/" rel="tag"># denovo-III</a>
          </div>

        


        
    <div class="post-nav">
      <div class="post-nav-item">
    <a href="/blog/2022/03/09/2022-03-09-HiFi-5-HiC-Pro/" rel="prev" title="HiFi-5-[HiC-Pro]-[MCscan]-[ALLHiC]">
      <i class="fa fa-chevron-left"></i> HiFi-5-[HiC-Pro]-[MCscan]-[ALLHiC]
    </a></div>
      <div class="post-nav-item"></div>
    </div>
      </footer>
    
  </article>
  
  
  



          </div>
          

<script>
  window.addEventListener('tabs:register', () => {
    let { activeClass } = CONFIG.comments;
    if (CONFIG.comments.storage) {
      activeClass = localStorage.getItem('comments_active') || activeClass;
    }
    if (activeClass) {
      let activeTab = document.querySelector(`a[href="#comment-${activeClass}"]`);
      if (activeTab) {
        activeTab.click();
      }
    }
  });
  if (CONFIG.comments.storage) {
    window.addEventListener('tabs:click', event => {
      if (!event.target.matches('.tabs-comment .tab-content .tab-pane')) return;
      let commentClass = event.target.classList[1];
      localStorage.setItem('comments_active', commentClass);
    });
  }
</script>

        </div>
          
  
  <div class="toggle sidebar-toggle">
    <span class="toggle-line toggle-line-first"></span>
    <span class="toggle-line toggle-line-middle"></span>
    <span class="toggle-line toggle-line-last"></span>
  </div>

  <aside class="sidebar">
    <div class="sidebar-inner">

      <ul class="sidebar-nav motion-element">
        <li class="sidebar-nav-toc">
          Table of Contents
        </li>
        <li class="sidebar-nav-overview">
          Overview
        </li>
      </ul>

      <!--noindex-->
      <div class="post-toc-wrap sidebar-panel">
          <div class="post-toc motion-element"><ol class="nav"><li class="nav-item nav-level-1"><a class="nav-link" href="#1-2016-05-np-%E7%A2%A7%E5%86%AC%E8%8C%84"><span class="nav-number">1.</span> <span class="nav-text">1. 2016.05-np 碧冬茄</span></a><ol class="nav-child"><li class="nav-item nav-level-2"><a class="nav-link" href="#Genome-sequencing-assembly-and-annotation"><span class="nav-number">1.1.</span> <span class="nav-text">Genome sequencing, assembly and annotation.</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#Repetitive-elements-analysis"><span class="nav-number">1.2.</span> <span class="nav-text">Repetitive elements analysis</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#Gene-data-mining"><span class="nav-number">1.3.</span> <span class="nav-text">Gene data mining</span></a></li></ol></li><li class="nav-item nav-level-1"><a class="nav-link" href="#2-2017-04-ng-%E5%B1%B1%E7%BE%8A%F0%9F%90%90"><span class="nav-number">2.</span> <span class="nav-text">2. 2017.04-ng 山羊🐐</span></a><ol class="nav-child"><li class="nav-item nav-level-2"><a class="nav-link" href="#data"><span class="nav-number">2.1.</span> <span class="nav-text">data</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#%E7%BB%84%E8%A3%85%E7%AD%96%E7%95%A5%E5%8F%8A%E7%BB%84%E8%A3%85%E7%BB%93%E6%9E%9C"><span class="nav-number">2.2.</span> <span class="nav-text">组装策略及组装结果</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#%E7%BB%84%E8%A3%85%E8%AF%84%E4%BC%B0%E5%8F%8A%E6%AF%94%E8%BE%83%E5%88%86%E6%9E%90"><span class="nav-number">2.3.</span> <span class="nav-text">组装评估及比较分析</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#%E5%9F%BA%E5%9B%A0%E7%BB%84%E6%B3%A8%E9%87%8A"><span class="nav-number">2.4.</span> <span class="nav-text">基因组注释</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#%E5%90%AF%E7%A4%BA%E5%BD%95"><span class="nav-number">2.5.</span> <span class="nav-text">启示录</span></a></li></ol></li><li class="nav-item nav-level-1"><a class="nav-link" href="#3-2020-09-cell-%E5%A4%A7%E8%B1%86%E6%B3%9B%E5%9F%BA%E5%9B%A0%E7%BB%84"><span class="nav-number">3.</span> <span class="nav-text">3. 2020.09-cell 大豆泛基因组</span></a><ol class="nav-child"><li class="nav-item nav-level-2"><a class="nav-link" href="#SNP-Calling-and-Phylogenetic-Analyses"><span class="nav-number">3.1.</span> <span class="nav-text">SNP Calling and Phylogenetic Analyses</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#Genome-Assembly"><span class="nav-number">3.2.</span> <span class="nav-text">Genome Assembly</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#Repeat-Analysis-and-Gene-Annotation"><span class="nav-number">3.3.</span> <span class="nav-text">Repeat Analysis and Gene Annotation</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#Synteny-Analysis"><span class="nav-number">3.4.</span> <span class="nav-text">Synteny Analysis</span></a></li></ol></li><li class="nav-item nav-level-1"><a class="nav-link" href="#4-2019-nc-%E5%A4%A7%E8%B1%86%E6%B3%9B%E5%9F%BA%E5%9B%A0%E7%BB%84-%E7%BB%84%E8%A3%85%E5%8F%82%E8%80%831"><span class="nav-number">4.</span> <span class="nav-text">4. 2019-nc 大豆泛基因组-组装参考1</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#6-2020-05-nc-%E5%90%8C%E6%BA%90%E5%9B%9B%E5%80%8D%E4%BD%93%E7%B4%AB%E8%8A%B1%E8%8B%9C%E8%93%BF"><span class="nav-number">5.</span> <span class="nav-text">6. 2020.05-nc 同源四倍体紫花苜蓿</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#7-2021-11-gb-%E6%A4%B0%E5%AD%90%F0%9F%A5%A5"><span class="nav-number">6.</span> <span class="nav-text">7. 2021.11-gb 椰子🥥</span></a><ol class="nav-child"><li class="nav-item nav-level-2"><a class="nav-link" href="#Hi-C-based-genome-assembly"><span class="nav-number">6.1.</span> <span class="nav-text">Hi-C-based genome assembly</span></a></li></ol></li><li class="nav-item nav-level-1"><a class="nav-link" href="#8-2021-11-nc-%E9%87%91%E7%B2%9F%E5%85%B0%E5%9F%BA%E5%9B%A0%E7%BB%84"><span class="nav-number">7.</span> <span class="nav-text">8. 2021.11-nc 金粟兰基因组</span></a><ol class="nav-child"><li class="nav-item nav-level-2"><a class="nav-link" href="#Genome-size-estimate-and-assembly"><span class="nav-number">7.1.</span> <span class="nav-text">Genome size estimate and assembly.</span></a></li></ol></li></ol></div>
      </div>
      <!--/noindex-->

      <div class="site-overview-wrap sidebar-panel">
        <div class="site-author motion-element" itemprop="author" itemscope itemtype="http://schema.org/Person">
    <img class="site-author-image" itemprop="image" alt="Wang C"
      src="/blog/images/banner.png">
  <p class="site-author-name" itemprop="name">Wang C</p>
  <div class="site-description" itemprop="description">Make more time</div>
</div>
<div class="site-state-wrap motion-element">
  <nav class="site-state">
      <div class="site-state-item site-state-posts">
          <a href="/blog/archives">
          <span class="site-state-item-count">32</span>
          <span class="site-state-item-name">posts</span>
        </a>
      </div>
      <div class="site-state-item site-state-categories">
            <a href="/blog/categories/">
          
        <span class="site-state-item-count">11</span>
        <span class="site-state-item-name">categories</span></a>
      </div>
      <div class="site-state-item site-state-tags">
            <a href="/blog/tags/">
          
        <span class="site-state-item-count">18</span>
        <span class="site-state-item-name">tags</span></a>
      </div>
  </nav>
</div>
  <div class="links-of-author motion-element">
      <span class="links-of-author-item">
        <a href="https://github.com/wangc1106" title="GitHub → https:&#x2F;&#x2F;github.com&#x2F;wangc1106" rel="noopener" target="_blank"><i class="fab fa-github fa-fw"></i>GitHub</a>
      </span>
      <span class="links-of-author-item">
        <a href="mailto:wangchen116@126.com" title="E-Mail → mailto:wangchen116@126.com" rel="noopener" target="_blank"><i class="fa fa-envelope fa-fw"></i>E-Mail</a>
      </span>
  </div>
  <div class="cc-license motion-element" itemprop="license">
    <a href="https://creativecommons.org/licenses/by-nc-sa/4.0/" class="cc-opacity" rel="noopener" target="_blank"><img src="/blog/images/cc-by-nc-sa.svg" alt="Creative Commons"></a>
  </div>



      </div>
        <div class="back-to-top motion-element">
          <i class="fa fa-arrow-up"></i>
          <span>0%</span>
        </div>

    </div>
  </aside>
  <div id="sidebar-dimmer"></div>


      </div>
    </main>

    <footer class="footer">
      <div class="footer-inner">
        

        

<div class="copyright">
  
  &copy; 2021 – 
  <span itemprop="copyrightYear">2022</span>
  <span class="with-love">
    <i class="fas fa-dna"></i>
  </span>
  <span class="author" itemprop="copyrightHolder">Wang C</span>
</div>

        






<script>
  (function() {
    function leancloudSelector(url) {
      url = encodeURI(url);
      return document.getElementById(url).querySelector('.leancloud-visitors-count');
    }

    function addCount(Counter) {
      var visitors = document.querySelector('.leancloud_visitors');
      var url = decodeURI(visitors.id);
      var title = visitors.dataset.flagTitle;

      Counter('get', '/classes/Counter?where=' + encodeURIComponent(JSON.stringify({ url })))
        .then(response => response.json())
        .then(({ results }) => {
          if (results.length > 0) {
            var counter = results[0];
            leancloudSelector(url).innerText = counter.time + 1;
            Counter('put', '/classes/Counter/' + counter.objectId, { time: { '__op': 'Increment', 'amount': 1 } })
              .catch(error => {
                console.error('Failed to save visitor count', error);
              });
          } else {
              Counter('post', '/classes/Counter', { title, url, time: 1 })
                .then(response => response.json())
                .then(() => {
                  leancloudSelector(url).innerText = 1;
                })
                .catch(error => {
                  console.error('Failed to create', error);
                });
          }
        })
        .catch(error => {
          console.error('LeanCloud Counter Error', error);
        });
    }

    function showTime(Counter) {
      var visitors = document.querySelectorAll('.leancloud_visitors');
      var entries = [...visitors].map(element => {
        return decodeURI(element.id);
      });

      Counter('get', '/classes/Counter?where=' + encodeURIComponent(JSON.stringify({ url: { '$in': entries } })))
        .then(response => response.json())
        .then(({ results }) => {
          for (let url of entries) {
            let target = results.find(item => item.url === url);
            leancloudSelector(url).innerText = target ? target.time : 0;
          }
        })
        .catch(error => {
          console.error('LeanCloud Counter Error', error);
        });
    }

    let { app_id, app_key, server_url } = {"enable":true,"app_id":"UqIYufmNk2sE8MiBKKWc4Kja-gzGzoHsz","app_key":"iYpeWQcmNqpHwr2uVlnJYVji","server_url":null,"security":false};
    function fetchData(api_server) {
      var Counter = (method, url, data) => {
        return fetch(`${api_server}/1.1${url}`, {
          method,
          headers: {
            'X-LC-Id'     : app_id,
            'X-LC-Key'    : app_key,
            'Content-Type': 'application/json',
          },
          body: JSON.stringify(data)
        });
      };
      if (CONFIG.page.isPost) {
        if (CONFIG.hostname !== location.hostname) return;
        addCount(Counter);
      } else if (document.querySelectorAll('.post-title-link').length >= 1) {
        showTime(Counter);
      }
    }

    let api_server = app_id.slice(-9) !== '-MdYXbMMI' ? server_url : `https://${app_id.slice(0, 8).toLowerCase()}.api.lncldglobal.com`;

    if (api_server) {
      fetchData(api_server);
    } else {
      fetch('https://app-router.leancloud.cn/2/route?appId=' + app_id)
        .then(response => response.json())
        .then(({ api_server }) => {
          fetchData('https://' + api_server);
        });
    }
  })();
</script>


      </div>
      <div style="height: 200px">
      <script type="text/javascript" src="//rf.revolvermaps.com/0/0/8.js?i=5knf34gs07t&amp;m=0&amp;c=ff0000&amp;cr1=ffffff&amp;f=arial&amp;l=33" async="async"></script>
      </div>
    </footer>
  </div>

  
  <script src="/blog/lib/anime.min.js"></script>
  <script src="/blog/lib/velocity/velocity.min.js"></script>
  <script src="/blog/lib/velocity/velocity.ui.min.js"></script>

<script src="/blog/js/utils.js"></script>

<script src="/blog/js/motion.js"></script>


<script src="/blog/js/schemes/pisces.js"></script>


<script src="/blog/js/next-boot.js"></script>




  




  
<script src="/blog/js/local-search.js"></script>













  

  

</body>
</html>
