<!DOCTYPE html>
<html lang="zh-Hans">
<head>
  <meta charset="UTF-8">
<meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=2">
<meta name="theme-color" content="#222">
<meta name="generator" content="Hexo 5.4.0">
  <link rel="apple-touch-icon" sizes="180x180" href="/blog/images/apple-touch-icon-next.png">
  <link rel="icon" type="image/png" sizes="32x32" href="/blog/images/favicon-32x32-next.png">
  <link rel="icon" type="image/png" sizes="16x16" href="/blog/images/favicon-16x16-next.png">
  <link rel="mask-icon" href="/blog/images/logo.svg" color="#222">
  <meta name="google-site-verification" content="1osTo_gOe1SW9Hevqtz8vl8HfSgTO4DSMZ0DiMPMX6w">

<link rel="stylesheet" href="/blog/css/main.css">


<link rel="stylesheet" href="/blog/lib/font-awesome/css/all.min.css">

<script id="hexo-configurations">
    var NexT = window.NexT || {};
    var CONFIG = {"hostname":"wangc1106.github.io","root":"/blog/","scheme":"Gemini","version":"7.8.0","exturl":false,"sidebar":{"position":"left","display":"post","padding":18,"offset":12,"onmobile":false},"copycode":{"enable":false,"show_result":false,"style":null},"back2top":{"enable":true,"sidebar":true,"scrollpercent":"ture"},"bookmark":{"enable":false,"color":"#222","save":"auto"},"fancybox":false,"mediumzoom":false,"lazyload":false,"pangu":false,"comments":{"style":"tabs","active":null,"storage":true,"lazyload":false,"nav":null},"algolia":{"hits":{"per_page":10},"labels":{"input_placeholder":"Search for Posts","hits_empty":"We didn't find any results for the search: ${query}","hits_stats":"${hits} results found in ${time} ms"}},"localsearch":{"enable":true,"trigger":"auto","top_n_per_article":1,"unescape":false,"preload":false},"motion":{"enable":true,"async":false,"transition":{"post_block":"fadeIn","post_header":"slideDownIn","post_body":"slideDownIn","coll_header":"slideLeftIn","sidebar":"slideUpIn"}},"path":"search.xml"};
  </script>

  <meta name="description" content="è¾…åŠ©ç»„è£…æ¡ˆä¾‹">
<meta property="og:type" content="article">
<meta property="og:title" content="HiFi-6-[HiFi-HiC-assembly-method]">
<meta property="og:url" content="https://wangc1106.github.io/blog/2022/03/14/2022-03-14-HiFi-6-HiC-assembly/index.html">
<meta property="og:site_name" content="Banner&#39;s Blog">
<meta property="og:description" content="è¾…åŠ©ç»„è£…æ¡ˆä¾‹">
<meta property="og:locale">
<meta property="og:image" content="https://wangc1106.github.io/blog/2022/03/14/2022-03-14-HiFi-6-HiC-assembly/1.png">
<meta property="og:image" content="https://wangc1106.github.io/blog/2022/03/14/2022-03-14-HiFi-6-HiC-assembly/2.png">
<meta property="og:image" content="https://wangc1106.github.io/blog/2022/03/14/2022-03-14-HiFi-6-HiC-assembly/3.png">
<meta property="og:image" content="https://wangc1106.github.io/blog/2022/03/14/2022-03-14-HiFi-6-HiC-assembly/4.png">
<meta property="og:image" content="https://wangc1106.github.io/blog/2022/03/14/2022-03-14-HiFi-6-HiC-assembly/5.png">
<meta property="og:image" content="https://wangc1106.github.io/blog/2022/03/14/2022-03-14-HiFi-6-HiC-assembly/6.png">
<meta property="og:image" content="https://wangc1106.github.io/blog/2022/03/14/2022-03-14-HiFi-6-HiC-assembly/7.png">
<meta property="og:image" content="https://wangc1106.github.io/blog/2022/03/14/2022-03-14-HiFi-6-HiC-assembly/8.png">
<meta property="og:image" content="https://wangc1106.github.io/blog/2022/03/14/2022-03-14-HiFi-6-HiC-assembly/9.png">
<meta property="og:image" content="https://wangc1106.github.io/blog/2022/03/14/2022-03-14-HiFi-6-HiC-assembly/10.png">
<meta property="og:image" content="https://wangc1106.github.io/blog/2022/03/14/2022-03-14-HiFi-6-HiC-assembly/11.png">
<meta property="article:published_time" content="2022-03-14T08:09:34.778Z">
<meta property="article:modified_time" content="2022-03-15T14:19:04.334Z">
<meta property="article:author" content="Wang C">
<meta property="article:tag" content="denovo-III">
<meta name="twitter:card" content="summary">
<meta name="twitter:image" content="https://wangc1106.github.io/blog/2022/03/14/2022-03-14-HiFi-6-HiC-assembly/1.png">

<link rel="canonical" href="https://wangc1106.github.io/blog/2022/03/14/2022-03-14-HiFi-6-HiC-assembly/">


<script id="page-configurations">
  // https://hexo.io/docs/variables.html
  CONFIG.page = {
    sidebar: "",
    isHome : false,
    isPost : true,
    lang   : 'zh-Hans'
  };
</script>

  <title>HiFi-6-[HiFi-HiC-assembly-method] | Banner's Blog</title>
  






  <noscript>
  <style>
  .use-motion .brand,
  .use-motion .menu-item,
  .sidebar-inner,
  .use-motion .post-block,
  .use-motion .pagination,
  .use-motion .comments,
  .use-motion .post-header,
  .use-motion .post-body,
  .use-motion .collection-header { opacity: initial; }

  .use-motion .site-title,
  .use-motion .site-subtitle {
    opacity: initial;
    top: initial;
  }

  .use-motion .logo-line-before i { left: initial; }
  .use-motion .logo-line-after i { right: initial; }
  </style>
</noscript>

</head>

<body itemscope itemtype="http://schema.org/WebPage">
  <div class="container use-motion">
    <div class="headband"></div>

    <header class="header" itemscope itemtype="http://schema.org/WPHeader">
      <div class="header-inner"><div class="site-brand-container">
  <div class="site-nav-toggle">
    <div class="toggle" aria-label="Toggle navigation bar">
      <span class="toggle-line toggle-line-first"></span>
      <span class="toggle-line toggle-line-middle"></span>
      <span class="toggle-line toggle-line-last"></span>
    </div>
  </div>

  <div class="site-meta">

    <a href="/blog/" class="brand" rel="start">
      <span class="logo-line-before"><i></i></span>
      <h1 class="site-title">Banner's Blog</h1>
      <span class="logo-line-after"><i></i></span>
    </a>
      <p class="site-subtitle" itemprop="description">Dream big work hard</p>
  </div>

  <div class="site-nav-right">
    <div class="toggle popup-trigger">
        <i class="fa fa-search fa-fw fa-lg"></i>
    </div>
  </div>
</div>




<nav class="site-nav">
  <ul id="menu" class="main-menu menu">
        <li class="menu-item menu-item-home">

    <a href="/blog/" rel="section"><i class="home fa-fw"></i>Home</a>

  </li>
        <li class="menu-item menu-item-categories">

    <a href="/blog/categories/" rel="section"><i class="th fa-fw"></i>Categories</a>

  </li>
        <li class="menu-item menu-item-tags">

    <a href="/blog/tags/" rel="section"><i class="tags fa-fw"></i>Tags</a>

  </li>
        <li class="menu-item menu-item-about">

    <a href="https://wangc1106.github.io/" rel="section"><i class="user fa-fw"></i>About</a>

  </li>
      <li class="menu-item menu-item-search">
        <a role="button" class="popup-trigger"><i class="fa fa-search fa-fw"></i>Search
        </a>
      </li>
  </ul>
</nav>



  <div class="search-pop-overlay">
    <div class="popup search-popup">
        <div class="search-header">
  <span class="search-icon">
    <i class="fa fa-search"></i>
  </span>
  <div class="search-input-container">
    <input autocomplete="off" autocapitalize="off"
           placeholder="Searching..." spellcheck="false"
           type="search" class="search-input">
  </div>
  <span class="popup-btn-close">
    <i class="fa fa-times-circle"></i>
  </span>
</div>
<div id="search-result">
  <div id="no-result">
    <i class="fa fa-spinner fa-pulse fa-5x fa-fw"></i>
  </div>
</div>

    </div>
  </div>

</div>
    </header>

    


    <main class="main">
      <div class="main-inner">
        <div class="content-wrap">
          

          <div class="content post posts-expand">
            

    
  
  
  <article itemscope itemtype="http://schema.org/Article" class="post-block" lang="zh-Hans">
    <link itemprop="mainEntityOfPage" href="https://wangc1106.github.io/blog/2022/03/14/2022-03-14-HiFi-6-HiC-assembly/">

    <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
      <meta itemprop="image" content="/blog/images/banner.png">
      <meta itemprop="name" content="Wang C">
      <meta itemprop="description" content="Make more time">
    </span>

    <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
      <meta itemprop="name" content="Banner's Blog">
    </span>
      <header class="post-header">
        <h1 class="post-title" itemprop="name headline">
          HiFi-6-[HiFi-HiC-assembly-method]
        </h1>

        <div class="post-meta">
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-calendar"></i>
              </span>
              <span class="post-meta-item-text">Posted on</span>

              <time title="Created: 2022-03-14 16:09:34" itemprop="dateCreated datePublished" datetime="2022-03-14T16:09:34+08:00">2022-03-14</time>
            </span>
              <span class="post-meta-item">
                <span class="post-meta-item-icon">
                  <i class="far fa-calendar-check"></i>
                </span>
                <span class="post-meta-item-text">Edited on</span>
                <time title="Modified: 2022-03-15 22:19:04" itemprop="dateModified" datetime="2022-03-15T22:19:04+08:00">2022-03-15</time>
              </span>
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-folder"></i>
              </span>
              <span class="post-meta-item-text">In</span>
                <span itemprop="about" itemscope itemtype="http://schema.org/Thing">
                  <a href="/blog/categories/Software/" itemprop="url" rel="index"><span itemprop="name">Software</span></a>
                </span>
            </span>

          
            <span id="/blog/2022/03/14/2022-03-14-HiFi-6-HiC-assembly/" class="post-meta-item leancloud_visitors" data-flag-title="HiFi-6-[HiFi-HiC-assembly-method]" title="Views">
              <span class="post-meta-item-icon">
                <i class="fa fa-eye"></i>
              </span>
              <span class="post-meta-item-text">Views: </span>
              <span class="leancloud-visitors-count"></span>
            </span>
            <span class="post-meta-item" title="Symbols count in article">
              <span class="post-meta-item-icon">
                <i class="far fa-file-word"></i>
              </span>
              <span>48k</span>
            </span>
            <span class="post-meta-item" title="Reading time">
              <span class="post-meta-item-icon">
                <i class="far fa-clock"></i>
              </span>
              <span>44 mins.</span>
            </span>

        </div>
      </header>

    
    
    
    <div class="post-body" itemprop="articleBody">

      
        <p>è¾…åŠ©ç»„è£…æ¡ˆä¾‹</p>
<span id="more"></span>

<h1 id="1-2016-05-np-ç¢§å†¬èŒ„"><a href="#1-2016-05-np-ç¢§å†¬èŒ„" class="headerlink" title="1. 2016.05-np ç¢§å†¬èŒ„"></a>1. 2016.05-np ç¢§å†¬èŒ„</h1><p><strong>Insight into the evolution of the Solanaceae from the parental genomes of Petunia hybrida</strong></p>
<h2 id="Genome-sequencing-assembly-and-annotation"><a href="#Genome-sequencing-assembly-and-annotation" class="headerlink" title="Genome sequencing, assembly and annotation."></a>Genome sequencing, assembly and annotation.</h2><p>Plants were grown and DNA was extracted following the methods described at Supplementary Note 1.</p>
<p>Illumina libraries with 0.17-, 0.35-, 0.5-, 0.8-, 1-, 2-, 5-, 8- and 15-kb inserts were sequenced at BGI-Shenzghen and University of Illinois, Roy J. Carver Biotechnology. PacBio P. axillaris DNA library was sequenced with P4/C2 chemistry.</p>
<p>Illumina reads were processed using <strong>Fastq-mcf</strong> (<strong>quality filtering</strong>; <a target="_blank" rel="noopener" href="https://code.google.com/p/eautils/wiki/FastqMcf">https://code.google.com/p/eautils/wiki/FastqMcf</a>), <strong>PRINSEQ</strong> (<strong>duplication filtering</strong>; http:// prinseq.sourceforge.net/) and <strong>Musket</strong> (<strong>error correction</strong>; <a target="_blank" rel="noopener" href="http://musket.sourceforge/">http://musket.sourceforge</a>. net/). <strong>Pacbio reads</strong> were processed using the <strong>SMRT Analysis pipeline</strong> (v.2.0.1; <a target="_blank" rel="noopener" href="https://github.com/PacificBiosciences/SMRT-Analysis">https://github.com/PacificBiosciences/SMRT-Analysis</a>).</p>
<p>Both genomes were assembled with <strong>SOAPdenovo</strong> with different k-mer sizes. For both genomes, <strong>k-mer = 79</strong> showed the best statistics. Gaps between contigs were completed using <strong>GapCloser</strong> . Additionally for P. axillaris, <strong>PacBio reads</strong> were integrated in <strong>four different steps</strong>: (1) <strong>Rescaffolding</strong> of the <strong>Illumina contigs</strong> using the <strong>PacBio reads</strong> and the <strong>AHA assembler</strong>; (2) <strong>Gap filling</strong> using <strong>PBJelly</strong>; (3) <strong>Rescaffolding</strong> using the Illumina <strong>pair data</strong> and <strong>SSPACE</strong>; (4) Last round of <strong>gap filling</strong> using <strong>PBJelly</strong> . </p>
<p><strong>Genome size estimation</strong> was performed through the k-mers abundance distribution (<strong>k-mer = 31</strong>). <strong>Heterozygosity</strong> was <strong>estimated</strong> mapping the Illumina reads to the assemblies using <strong>Bowtie2</strong>, calling SNPs using <strong>FreeBayes</strong> and  annotating the SNPs using <strong>SnpEff</strong> .</p>
<p>The genome structural annotation was performed using <strong>Maker-P</strong>: (1) <strong>SNAP</strong> and <strong>Augustus</strong> as ab initio gene predictors; (2) Exonerate as experimental based predictor with 454 and Illumina RNASeq reads and protein sequences from different protein datasets. RNAseq Illumina data was mapped using <strong>Tophat2</strong> . tRNAs were annotated using <strong>tRNAscan</strong> (<a target="_blank" rel="noopener" href="http://lowelab.ucsc.edu/tRNAscan-SE/">http://lowelab.ucsc.edu/tRNAscan-SE/</a>).</p>
<p>The <strong>gene functional annotation</strong> was performed by <strong>sequence homology search</strong> with different protein datasets using <strong>BlastP</strong> and protein domains search using <strong>InterProScan</strong> . Functional annotations were integrated using <strong>AHRD</strong> (<a target="_blank" rel="noopener" href="https://github.com/groupschoof/AHRD">https://github.com/groupschoof/AHRD</a>). See Supplementary Note 1.</p>
<h2 id="Repetitive-elements-analysis"><a href="#Repetitive-elements-analysis" class="headerlink" title="Repetitive elements analysis"></a>Repetitive elements analysis</h2><p>Repeat annotation was performed using RepeatModeler (v1.0.8; <a target="_blank" rel="noopener" href="http://www.repeatmasker.org/RepeatModeler.html">http://www.repeatmasker.org/RepeatModeler.html</a>), RepeatMasker (v4.0.5; <a target="_blank" rel="noopener" href="http://www.repeatmasker.org/">http://www.repeatmasker.org</a>) with the repeat database Repbase (release 20140131; <a target="_blank" rel="noopener" href="http://www.girinst.org/repbase/">http://www.girinst.org/repbase/</a>) and Geneious (v7.1.4; <a target="_blank" rel="noopener" href="http://www.geneious.com/">http://www.geneious.com</a>). Identification of PVCV-like and EPRV elements was performed using BlastN and TBlastN . The identified sequences were aligned with ClustalW (MEGA5 package;<a target="_blank" rel="noopener" href="http://www.megasoftware.net/">http://www.megasoftware.net/</a>) and then manually curated. RepeatExplorer (<a target="_blank" rel="noopener" href="http://www.repeatexplorer.org/">http://www.repeatexplorer.org/</a>) and other methods were used to extend the analysis to unassembled repeats. Fluorescent in situ hybridization was performed in root tips from young P. axillaris and P. inflata plants for 5S rDNA and three PVCV viral probes following the procedure described in Supplementary Note 2.</p>
<p>The detection of dTph1 loci in P. hybrida W138 was performed through a BLAST40 search of the P. axillaris and P. inflata dTph1 elements including the 500 bp of flanking sequence against the TFS W138 collection. Polymorphisms found in the genomic flanking regions were used to identify the species of origin. dTph1 elements were identified in a P. axillaris population using a modification of the methodology described in Supplementary Note 3.</p>
<p>##Whole-genome duplication, tandem duplications and gene family analysis</p>
<p>Whole-genome collinear analysis was performed using SynMap and microsynteny analysis were performed using GEvo in the comparative genomics platform, CoGe .See Supplementary Note 5.</p>
<p>The gene family analysis included Solanum lycopersicum, S. tuberosum, Nicotiana benthamiana and Arabidopsis thaliana protein sets using BlastP (v2.2.27) on an all-versus-all comparison and grouping the genes into families with OrthoMCL, v2.0.8. See Supplementary Note 4.</p>
<h2 id="Gene-data-mining"><a href="#Gene-data-mining" class="headerlink" title="Gene data mining"></a>Gene data mining</h2><p>The specific identification of genes for P. axillaris and P. inflata genomes for colour and scent, root-specific pathways, self-incompatibility and circadian clock was performed through a BlastN/BlastP sequence homology search. Blast GUI, JBrowser (<a target="_blank" rel="noopener" href="http://jbrowse.org/">http://jbrowse.org/</a>) and WebApollo (<a target="_blank" rel="noopener" href="http://genomearchitect/">http://genomearchitect</a>. org/) were installed in a server to search and manually curate the gene structures of the identified genes. See Supplementary Notes 7, 8 and 10â€“12.</p>
<h1 id="2-2017-04-ng-å±±ç¾ŠğŸ"><a href="#2-2017-04-ng-å±±ç¾ŠğŸ" class="headerlink" title="2. 2017.04-ng å±±ç¾ŠğŸ"></a>2. 2017.04-ng å±±ç¾ŠğŸ</h1><p>Published in final edited form as:Nat Genet. 2017 April ; 49(4): 643â€“650. doi:10.1038/ng.3802.</p>
<p><strong>Single-molecule sequencing and chromatin conformation capture enable</strong> <strong>de novo</strong> <strong>reference assembly of the domestic goat genome</strong></p>
<p>â€‹        é¦–æ¬¡åœ¨äººä¹‹å¤–çš„ç‰©ç§ä¸­ç›´æ¥de novoç»„è£…å‡ºæŸ“è‰²ä½“æ°´å¹³çš„åŸºå› ç»„ã€‚åˆ©ç”¨<font color="blue"><strong>Pacbio+BioNano+Hi-C+Illumina</strong> å¤šå±‚çº§</font>çš„ç­–ç•¥ï¼Œç»„è£…äº†ä¸€åªå±±ç¾Š(Capra hircus)çš„åŸºå› ç»„ï¼Œè·å¾—<font color="blue"><strong>31</strong>ä¸ª<strong>scaffold</strong>ï¼Œ<strong>663</strong>ä¸ª<strong>gap</strong>åŒºï¼Œscaffold <strong>N50</strong> æƒŠäººåœ°è¾¾åˆ°<strong>87M</strong>ï¼Œå‡ ä¹<strong>one scaffold one chromosome</strong></font>ï¼Œå¹¶ä¸”æˆåŠŸåœ°ç»„è£…å‡º<font color="blue"><strong>å…ç–«åŸºå› åŒº</strong>å’Œ<strong>å¤§éƒ¨åˆ†é‡å¤åºåˆ—å®¶æ—</strong></font>ã€‚</p>
<p>â€‹        <font color="red">äººçš„GRCh38ç‰ˆæœ¬æœ‰<strong>24</strong>ä¸ª<strong>scaffold</strong>ï¼Œ<strong>169</strong>ä¸ª<strong>unplacedscaffold</strong>ï¼Œ<strong>832ä¸ªgap</strong>åŒº</font>ã€‚</p>
<p>â€‹        ä»<strong>96å¤´</strong>å±±ç¾Š(<strong>6ä¸ªå“ç§</strong>)ä¸­ï¼Œåˆ©ç”¨Illuminaçš„ Caprine53K <font color="blue"><strong>SNP èŠ¯ç‰‡ç­›é€‰</strong></font>å‡º<font color="blue"><strong>åŸºå› å‹çº¯å’Œåº¦æœ€é«˜</strong></font>çš„ä¸ªä½“ç”¨æ¥<font color="blue"><strong>ç»„è£…</strong></font>(San Clemente breed)ã€‚</p>
<h2 id="data"><a href="#data" class="headerlink" title="data"></a>data</h2><ol>
<li><p>Pacbioæ•°æ®</p>
<p>ä¸€å…±äº§ç”Ÿ465ä¸ªSMRT cellçš„æ•°æ®ã€‚å…¶ä¸­ä½¿ç”¨P5-C3è¯•å‰‚çš„311ä¸ªcell;ä½¿ç”¨P4-C2è¯•å‰‚çš„142ä¸ªcellï¼ŒXL-C2è¯•å‰‚çš„12ä¸ªcellã€‚æ•°æ®é‡194Gï¼Œè¦†ç›–æ·±åº¦è¾¾<strong>69X</strong>ï¼Œ<strong>subread</strong>çš„<strong>å¹³å‡</strong>é•¿åº¦ä¸º<strong>5,110bp</strong>ã€‚</p>
</li>
<li><p>BioNano æ•°æ®</p>
<p>ç”±äºé‡‡æ ·æ‰€ç”¨å±±ç¾Šæ„å¤–æ­»äº¡ï¼Œå…¶DNAä¸èƒ½æ»¡è¶³Irsy optical mapping æµ‹åºéœ€æ±‚ï¼Œåªå¥½ç”¨å…¶<strong>é›„æ€§åä»£</strong>é‡‡æ ·ã€‚Optical mapä¸€å…±äº§ç”Ÿ256Gbæ•°æ®ï¼Œè¦†ç›–<strong>98 X</strong>ã€‚</p>
</li>
<li><p>Hi-Cæ•°æ®</p>
<p>é‡‡æ ·åŠ¨ç‰©æ˜¯æœ€åˆé‚£åªå±±ç¾Šã€‚Hi-Cå»ºåº“ï¼Œåºåˆ—ç”¨ç‰©ç†æ–¹æ³•æ‰“æ–­æˆ300-500bpé•¿åº¦ï¼ŒPE101æµ‹åºï¼Œäº§ç”Ÿ<strong>115M reads</strong>çš„æ•°æ®é‡ã€‚</p>
</li>
<li><p>Illumina æ•°æ®</p>
<p>é‡‡æ ·æœ€åˆé‚£åªå±±ç¾Šï¼ŒPE251å»ºåº“æµ‹åºï¼Œè·å¾—<strong>23X</strong>çš„æ•°æ®è¦†ç›–ï¼Œç”¨æ¥åšæœ€åçš„<font color="red"><strong>é”™è¯¯æ ¡æ­£</strong></font></p>
</li>
</ol>
<h2 id="ç»„è£…ç­–ç•¥åŠç»„è£…ç»“æœ"><a href="#ç»„è£…ç­–ç•¥åŠç»„è£…ç»“æœ" class="headerlink" title="ç»„è£…ç­–ç•¥åŠç»„è£…ç»“æœ"></a><strong>ç»„è£…ç­–ç•¥åŠç»„è£…ç»“æœ</strong></h2><ul>
<li><p>å…ˆç”¨Pacbioçš„ long-read æ•°æ®æ„å»ºcontigï¼Œ ä½¿ç”¨<font color="blue"><strong>Celera Assembler</strong> PacBio corrected Reads æµç¨‹</font>ç»„è£…ã€‚è·å¾—<font color="blue"><strong>3,074</strong>ä¸ªcontig(2.63G)ï¼Œ<strong>N50 4.159M</strong>b</font>ã€‚</p>
</li>
<li><p>æ¥ä¸‹æ¥é‡‡ç”¨<font color="blue"><strong>Irys optical mapping</strong>æ•°æ®æ„å»º<strong>scaffold</strong></font>ï¼Œä½¿ç”¨è½¯ä»¶ä¸º<font color="blue"><strong>IrysView</strong></font>. äº§ç”Ÿ<font color="blue"><strong>842</strong>ä¸ªscaffoldï¼Œscaffold N50ä¸º<strong>13.4M</strong>bï¼Œcontig N50ä¸º10.858Mï¼Œæœ€é•¿scaffoldä¸º66.728Mb</font>ã€‚ åˆ©ç”¨<font color="red"><strong>radiation hybrid(RH) mapæ¯”è¾ƒ</strong>å‘ç°</font>ï¼Œ<font color="blue">PacBio-Irys è”åˆ</font>çš„æ–¹æ³•å·²ç»<font color="blue">å®Œæ•´ç»„è£…å‡º<strong>20å·æŸ“è‰²ä½“</strong></font>ã€‚</p>
</li>
<li><p>æ¥ä¸‹æ¥ç”¨Hi-Cæ•°æ®ï¼Œè°ƒç”¨ç”¨<font color="red"><strong>Lachesisè½¯ä»¶åŒ…</strong></font>ï¼Œç»“åˆPacBio-Irysçš„ç»“æœï¼Œå³æ•´åˆçš„PacBio-Irys-PGA (PBIP)æ–¹æ³•ï¼Œè·å¾—è¾ƒå®Œç¾çš„ç»„è£…ç»“æœã€‚<font color="blue">Scaffold N50</font> è¾¾åˆ°æƒŠäººçš„<font color="blue"><strong>87.347Mb</strong></font>(è¿œè¶…ä»Šå¹´å°–å»é²ˆï¼Œå¤§çŒ©çŒ©N50 20Mbçš„é«˜èµ·ç‚¹)ï¼Œæ€»å…±è·å¾—<font color="blue"><strong>31æ¡</strong>scaffold</font>ï¼Œ(<font color="red">31,261æ¡&lt;50kbpçš„**degenerate scaffold**ä¸è®¡å…¥</font> )ã€‚</p>
</li>
<li><p>æœ€ååˆ©ç”¨Illuminaæ•°æ®åš<font color="red"><strong>ä¸€è‡´æ€§æ ¡æ­£</strong>å’Œæœ€åçš„<strong>è¡¥æ´</strong></font>ï¼Œæœ€åä½¿å¾—contigçš„æ•°ç›®ä¸‹é™åˆ°680æ¡ï¼Œäº§ç”Ÿçš„gapæ•°ç›®ä¸º663ä¸ªã€‚</p>
</li>
<li><p>åˆ©ç”¨Kraken v0.10.5 <font color="red">å»é™¤æœ‰<strong>ç—…æ¯’å’Œç»†èŒæ±¡æŸ“</strong>çš„åºåˆ—</font>ï¼Œå»æ‰æœ‰NCBI vectoræ±¡æŸ“çš„åºåˆ—ã€‚è·å¾—æœ€ç»ˆçš„åŸºå› ç»„ ç‰ˆæœ¬<strong>ARS1</strong>ã€‚</p>
</li>
</ul>
<h2 id="ç»„è£…è¯„ä¼°åŠæ¯”è¾ƒåˆ†æ"><a href="#ç»„è£…è¯„ä¼°åŠæ¯”è¾ƒåˆ†æ" class="headerlink" title="ç»„è£…è¯„ä¼°åŠæ¯”è¾ƒåˆ†æ"></a><strong>ç»„è£…è¯„ä¼°åŠæ¯”è¾ƒåˆ†æ</strong></h2><p>ç”±äºå±±ç¾Šä¹‹å‰å·²ç»äº§ç”Ÿè¿‡ä¸¤ä¸ªåŸºå› ç»„ç‰ˆæœ¬<strong>CHIR_1.0ã€CHIR_2.0</strong>ï¼Œå¯ä»¥ç”¨æ¥åšæ¯”è¾ƒ åˆ†æï¼Œå¹¶ä¸”æœ‰RH mappingæ•°æ®ç”¨æ¥è¯„ä¼°ã€æ ¡æ­£ARS1çš„ç»„è£…ç‰ˆæœ¬ã€‚ <strong>é€šè¿‡å„ç§å‚æ•°æ¯”è¾ƒå‘ç°ï¼ŒARS1å‡ä¼˜äºCHIR_2.0ç‰ˆæœ¬ã€‚</strong></p>
<p>1ã€é€šè¿‡æ¯”å¯¹å‘ç°ï¼ŒCHIR_2.0 æ¯”CHIR_1.0æœ‰æ›´å°‘çš„putative deletion(2,735 vs 1 0,256)å’Œduplication(115 vs 290)ã€‚è€Œ2.0ç‰ˆæœ¬ç›¸è¾ƒ1.0æœ‰æ›´å¤šçš„inversion(215 v s 4)ã€‚é€šè¿‡æ¯”è¾ƒï¼Œå‘ç°ARS1ç‰ˆæœ¬æœ‰å¤§å¹…åº¦åœ°æå‡ï¼Œè¾ƒ2.0ç‰ˆæœ¬å°‘4å€çš„deletionï¼Œ å°‘50å€inversionã€‚</p>
<p>2ã€CHIR_2.0èƒ½å¡«è¡¥CHIR_1.0 94.6%çš„gapï¼Œè€Œå‰©ä¸‹çš„é‚£äº›gapé€šè¿‡åˆ†æARS1å¾—çŸ¥ï¼Œæ˜¯CHIR_1.0çš„ç»„è£…é”™è¯¯ï¼Œè€ŒRHæ•°æ®ä¹Ÿæ”¯æŒäº†è¿™ä¸€è®ºæ–­ã€‚ 3ã€BUSCOæ˜¯åˆ©ç”¨å•æ‹·è´orthologæ¥è¯„ä¼°ç»„è£…çš„å¯é æ€§ï¼Œå’ŒCHIR_2.0ç›¸æ¯”ï¼ŒARS1æœ‰æ›´é«˜çš„BUSCO scoreã€‚</p>
<p>4ã€å±±ç¾Š52k SNP èŠ¯ç‰‡ä¸­ï¼Œæœ‰1,723ä¸ªSNPæ¢é’ˆç›®å‰åªèƒ½å®šä½åœ¨CHIR_2.0ç»„è£…ç‰ˆæœ¬ ä¸­çš„unplaced contigsä¸Šã€‚è€ŒARS1ç‰ˆæœ¬èƒ½æŠŠå…¶ä¸­çš„90%(1,552/1,723)ä»¥ä¸Šçš„S NPæ¢é’ˆå®šä½åœ¨æŸ“è‰²ä½“ä¸Šã€‚å¦å¤–å‘ç°æœ‰26ä¸ªlow call-rate çš„SNPmarkeråœ¨ARS1ç»„è£…ç‰ˆæœ¬ä¸Šå®šä½æ¯”è¾ƒæ¨¡æ£±ä¸¤å¯ã€‚è¿™ä¹Ÿå°±è§£é‡Šäº†ä¸ºä»€ä¹ˆè¿™äº›makeråœ¨èŠ¯ç‰‡ä¸Šæœ‰è¾ƒå·®çš„call -rate. </p>
<p>5ã€é€šè¿‡æ¯”è¾ƒå‘ç°ï¼ŒCHIR_2.0ç‰ˆæœ¬ä¸­çš„3,495ä¸ªå†…å«å­æˆ–å¤–æ˜¾å­æœ‰gapçš„åŸºå› ï¼Œåœ¨ARS1ä¸­éƒ½å¾—åˆ°è¡¥å…¨ã€‚åŒæ—¶ä¹Ÿç¡®è®¤1,926ä¸ªé¢„æµ‹çš„å¤–æ˜¾å­åœ¨CHIR_1.0æˆ–è€…2.0ä¸­æœ‰gapï¼Œ ä½†åœ¨ARS1ä¸­å¾—åˆ°ä¿®å¤ã€‚ </p>
<p>6ã€ç”±äºå…ç–«åŸºå› åŒºé«˜åº¦å¤šæ€æ€§å’Œé‡å¤æ€§ï¼Œç”¨äºŒä»£æµ‹åºæ•°æ®å¾ˆéš¾ç»„è£…èµ·æ¥ï¼Œä½†æ˜¯é€šè¿‡ åˆ†æå‘ç°ï¼ŒARS1ç‰ˆæœ¬å°±å¾ˆå¥½åœ°æŠŠLRCå’ŒNKCåŸºå› (å…ç–«åŠŸèƒ½ç›¸å…³åŸºå› )å®šä½åœ¨ä¸€ä¸ªç‹¬ç«‹çš„å¸¸æŸ“è‰²ä½“scaffoldä¸Šã€‚ </p>
<p>7ã€é€šè¿‡å’Œä¸€ä»£ã€äºŒä»£ç»„è£…ç­–ç•¥ç›¸æ¯”ï¼ŒARS1ç‰ˆæœ¬åœ¨é‡å¤åºåˆ—çš„ç»„è£…ä¸Šè·å¾—äº†æå¤§è¿›æ­¥ã€‚å¯¹æŸäº›å¼‚æŸ“è‰²ä½“æˆ–å¼‚æŸ“è‰²è´¨çš„åŒºåŸŸä¹Ÿè·å¾—äº†è¾ƒå¥½çš„ç»„è£…è¦†ç›–ã€‚æ¯”å¦‚ï¼Œåœ¨6æ¡å¸¸ æŸ“è‰²ä½“ä¸Šç»„è£…å‡º&gt;5kbpçš„ç«¯ç²’åºåˆ—ã€‚é€šè¿‡åˆ†æå‘ç°ï¼Œ 15æ¡æŸ“è‰²ä½“scaffoldï¼Œåœ¨ç€ä¸ç²’åŒºåŸŸç»„è£…å‡ºå¤§äº2kbpé•¿åº¦çš„é‡å¤åŒºï¼Œè€Œå…¶ä¸­çš„7æ¡ï¼Œæ›´æ˜¯è£…å‡ºæ¥8kbpä»¥ä¸Šçš„é‡å¤åŒºã€‚æ›´ä»¤äººæƒŠè®¶çš„æ˜¯ï¼Œ19å·å’Œ23å·æŸ“è‰²ä½“éƒ½ç»„è£…å‡ºé«˜åº¦é‡å¤çš„ç€ä¸ç²’å’Œç«¯ç²’åŒºåŸŸï¼Œ è´¯é€šäº†æœ‰ç»“æ„æ€§å¼‚æŸ“è‰²è´¨åŒºçš„æŸ“è‰²ä½“ã€‚ </p>
<p>8ã€åœ¨ARS1ç‰ˆæœ¬ä¸­é‰´å®šå‡ºå¤§äº12kbpçš„é‡å¤æ¨¡å¼åºåˆ—å¤šè¾¾105æ¡ã€‚å¯¹äºé‡å¤åºåˆ—å®¶ æ—çš„é‰´å®šï¼Œåœ¨ARS1ä¸­è·å¾—çš„æ¥è¿‘å…¨é•¿çš„BovB LINE åºåˆ—æ¯”CHIR_2.0å¤š60%ä»¥ä¸Šã€‚ åœ¨CHIR_2.0ç‰ˆæœ¬é‡Œï¼Œè¢«ARS1æˆåŠŸè¡¥å…¨çš„gapä¸­çš„43.6%ä¸BovBé‡å¤åºåˆ—ä¸€è‡´(é•¿ åº¦å¤§äº3.5kbp)ï¼Œå³æ„å‘³ç€äºŒä»£æµ‹åºçš„gapåŒºå¤§å¤šéƒ½æ˜¯é‡å¤åºåˆ—ï¼Œç”¨ä¸‰ä»£å°±èƒ½è¾ƒå®¹æ˜“æµ‹é€šã€‚ </p>
<p>9ã€å…³äºæ€§æŸ“è‰²ä½“çš„åˆ†æã€‚é€šè¿‡æ¯”å¯¹å‘ç°ï¼Œä¸¤ä¸ªä¸åŒçš„scaffoldæ¯”å¯¹åˆ°XæŸ“è‰²ä½“çš„ä¸åŒä½†è¿ç»­çš„åŒºåŸŸï¼Œå é¢„æœŸXæŸ“è‰²ä½“å¤§å°(150Mb)çš„86%å·¦å³ã€‚é€šè¿‡å’Œè‡ªå·±çš„ã€è·¨ç‰©ç§ç‰›çš„YæŸ“è‰²ä½“æ¯”å¯¹ï¼Œæœ€ç»ˆç¡®å®šå‡º10MbåŒºåŸŸçš„åºåˆ—ï¼Œå YæŸ“è‰²ä½“é¢„ä¼°å¤§å°çš„50% å·¦å³ã€‚é€šè¿‡æ¯”å¯¹ç‰›å’Œç¾Šçš„YæŸ“è‰²ä½“ä¸Šçš„åŸºå› ï¼Œå‘ç°åœ¨ç›®å‰çš„scaffoldä¸­èƒ½æ‰¾åˆ°16%ï¼Œ è€Œåœ¨å…ˆå‰è¿‡æ»¤æ‰çš„degenerate contigä¸­èƒ½æ‰¾åˆ°84%çš„åŸºå› ã€‚è€ƒè™‘åˆ°YæŸ“è‰²ä½“çš„å¼‚æŸ“ è‰²è´¨å±æ€§åŠX,YæŸ“è‰²ä½“çš„æ‹Ÿå¸¸æŸ“è‰²åŒºåŸŸç»å¸¸å‘ç°äº¤æ¢ï¼Œå®åœ¨å¤ªè¿‡å¤æ‚ï¼Œè¶…å‡ºç›®å‰çš„ç»„è£…å’Œè®¤çŸ¥æ°´å¹³ï¼Œè¡¨ç¤ºå®åœ¨æ— èƒ½ä¸ºåŠ›ã€‚</p>
<p>10ã€åˆ©ç”¨ä¹‹å‰RH mapping dataï¼Œå¯¹ç»„è£…çš„å¤§éƒ¨åˆ†ç¯èŠ‚è¿›è¡Œäº†æ ¡æ­£åŠè¯„ä¼°ã€‚æ¯”å¦‚å¤šæ¬¡è¾…åŠ©è§£å†³scaffold conflictçš„é—®é¢˜ã€‚</p>
<h2 id="åŸºå› ç»„æ³¨é‡Š"><a href="#åŸºå› ç»„æ³¨é‡Š" class="headerlink" title="åŸºå› ç»„æ³¨é‡Š"></a><strong>åŸºå› ç»„æ³¨é‡Š</strong></h2><p>å¤šç§æ–¹æ³•ç»“åˆæ³¨é‡ŠåŸºå› é›†ã€‚<br>1.<strong>RNA-seqçš„æ–¹æ³•:</strong> 6ä¸ªç»„ç»‡(å¤§å¤šå’Œè„‘ç»„ç»‡ç›¸å…³)RNA-seqæµ‹åºã€13ä¸ªSRAä¸‹è½½ æ•°æ®ï¼Œåˆ©ç”¨stringtieã€cufflinkså’ŒTrinity(åŸºäºæ— å‚ç»„è£…)ã€‚æœ€åç”¨PASAè½¯ä»¶æ•´åˆåœ¨ä¸€èµ·;</p>
<p>2.<strong>ç”¨exonerateå’Œtblastn è½¯ä»¶</strong>æ¯”å¯¹åˆ°å‡ ä¸ªè¿‘ç¼˜ç‰©ç§çš„EnsemblåŸºå› é›†ä¸Šï¼Œè·å¾—åŒ æºé¢„æµ‹åŸºå› é›†;</p>
<p>3.<strong>ç”¨Braker1åšAb initio é¢„æµ‹</strong>;</p>
<p>4.<strong>CHIR_1.0 ç‰ˆæœ¬çš„æ³¨é‡ŠåŸºå› é›†</strong>; æœ€åç”¨<strong>EVM+PASA</strong>æŠŠä»¥ä¸Š4ç§æ•°æ®æ•´åˆæˆä¸€ä¸ªæœ€ç»ˆçš„åŸºå› é›†(è®¾ç½®çš„æƒé‡ä¸ºRNAseq &gt; cDNA/protein &gt; ab initio gene predictions)ã€‚</p>
<h2 id="å¯ç¤ºå½•"><a href="#å¯ç¤ºå½•" class="headerlink" title="å¯ç¤ºå½•"></a><strong>å¯ç¤ºå½•</strong></h2><p>1ã€Pacbio RSIIæ˜¯ä¸€ä¸ªè¾ƒä¸ºç†æƒ³ç»„è£…å¹³å°ï¼Œäº§ç”Ÿå¹³å‡14Kbpå·¦å³çš„è¯»é•¿ï¼Œæœ€é«˜åˆ°60 Kbpï¼Œå¯ä»¥ç»„è£…å‡ºè¾ƒä¸ºç†æƒ³çš„åŸºå› ç»„ç‰ˆæœ¬ã€‚ç”±äºä¸‰ä»£çš„é•¿è¯»é•¿ä¼˜åŠ¿ï¼ŒåŒæ—¶å¯ä»¥è¾ƒå¥½åœ°ç»„è£…å‡ºåŸºå› ç»„ä¸­å¤§éƒ¨åˆ†é«˜åº¦é‡å¤åŒº(ä¸»è¦é›†ä¸­åœ¨ç€ä¸ç²’å’Œç«¯ç²’åŒºåŸŸ);</p>
<p>2ã€BioNano çš„Irys optical mapping æ˜¯ä¸€ç§éå¸¸æœ‰æ•ˆçš„ï¼Œæˆæœ¬æ¯”è¾ƒé€‚ä¸­çš„æ„å»ºscaffoldçš„å¹³å°ã€‚å’ŒHi-Cæ¯”ä¼˜åŠ¿åœ¨äºå®šä½é”™è¯¯è¾ƒå°‘ï¼Œç¼ºç‚¹æ˜¯å¯¹N50çš„æé«˜å¹¶ä¸æ˜¯ç‰¹åˆ«æ˜¾è‘—ï¼Œä¸€èˆ¬åœ¨2å€å·¦å³ã€‚å¦‚æœæ­£åé“¾çš„Nt.BsqIé…¶åˆ‡ä½ç‚¹ç¦»å¾—æ¯”è¾ƒè¿‘ï¼Œå¸¸å¸¸å¯¼è‡´åŒé“¾ æ–­è£‚ï¼Œé™åˆ¶äº†Optical Map scaffoldçš„å¤§å°ã€‚æ‰€ä»¥ï¼ŒBioNanoå¯¹è¾ƒé•¿çš„scaffoldæ•ˆæœæ¯”è¾ƒå¥½ã€‚ç”±äºOptical mapå’Œpacbioäº§ç”Ÿçš„é”™è¯¯ç‰¹å¾å¹¶ä¸ä¸€æ ·ï¼Œä¸¤è€…ç»“åˆï¼Œå¯ä»¥ ç›¸äº’æ ¡æ­£ï¼Œè·å¾—è¾ƒå¥½çš„ç»„è£…æŒ‡æ ‡;</p>
<p>3ã€Hi-Cæ˜¯ç›®å‰éå¸¸ç«çˆ†çš„ä¸€ä¸ªæŠ€æœ¯ï¼Œåœ¨å¤šä¸ªé¢†åŸŸéƒ½æœ‰å¾ˆå¥½çš„åº”ç”¨ã€‚åœ¨ç»„è£…æ–¹é¢ï¼Œ å®ƒèƒ½è·å¾—æŸ“è‰²ä½“çº§åˆ«çš„scaffoldï¼Œä½†æ˜¯ä¹Ÿå®¹æ˜“è·å¾—è¾ƒé«˜çš„contigå®šå‘é”™è¯¯ã€‚è¿™ä¹Ÿæ˜¯ ä¸ºä»€ä¹ˆæ–‡ç« ä¸­ï¼Œå…ˆç”¨optical mappingï¼Œåç”¨Hi-Cåšscaffoldingçš„é‡è¦åŸå› ä¹‹ä¸€ã€‚ ä½œè€…å»ºè®®åœ¨ä»¥åçš„å®éªŒä¸­ï¼Œå¯ä»¥é€‰æ‹©è¾ƒçŸ­çš„è¯†åˆ«ä½ç‚¹çš„é™åˆ¶å†…åˆ‡é…¶(æˆ–è€…DNase Hi -C)æ¥æé«˜Hi-Cäº¤è”çš„å¯†åº¦ï¼Œé™ä½å®šå‘é”™è¯¯ã€‚</p>
<p>4ã€åˆ©ç”¨åŒæ ·çš„æ–¹æ³•å’Œå¹³å°ï¼Œåšç±»ä¼¼çš„åŸºå› ç»„é¡¹ç›®ï¼Œç”¨Pacbioå¹³å°å’Œscaffoldingå¹³ å°ï¼Œç›®å‰å¤§çº¦èŠ±è´¹åœ¨$100,000ä¸Šä¸‹,è€Œä¸€èˆ¬çš„çŸ­readså¹³å°æµ‹åºåŠ scaffolding å¹³å°ï¼Œåªéœ€è¦å…¶ä¸‰åˆ†ä¹‹ä¸€çš„ä»·æ ¼ã€‚ä½†æ˜¯ç”¨Pacbioå¹³å°å´è·å¾—äº†é«˜è´¨é‡çš„ã€è¿ç»­æ€§å¥½çš„åŸºå› ç»„åºåˆ—ã€‚ç›®å‰Pacbio æ–°ä¸€ä»£çš„sequelå¹³å°å·²ç»æ¨å‡ºï¼Œå•ä¸ªcellçš„äº§é‡æ¯”RSIIæ é«˜7å€ä»¥ä¸Šï¼Œæµ‹åºæˆæœ¬å°†ä¼šå¾—åˆ°å¤§å¹…åº¦é™ä½ï¼Œç›¸ä¿¡ä¼šæˆä¸ºæœªæ¥åŸºå› ç»„ç ”ç©¶çš„ä¸»æµå·¥å…·ä¹‹ä¸€ã€‚</p>
<p>5ã€æœ¬æ–‡çš„ç ”ç©¶ç­–ç•¥ç¡®å®ä¼šå¯è¿ªåæ¥çš„ç ”ç©¶è€…ï¼Œä¸ºde novoç»„è£…æ ‘ç«‹æ–°æ ‡æ†ã€‚Pacbio æµ‹åºæŠ€æœ¯è”åˆå¤šç§scaffolding å¹³å°(æ¯”å¦‚BioNanoã€Hi-Cï¼Œmating-pairedlibr aryï¼Œ10X genomics)çš„ä½¿ç”¨ï¼Œ ä¼šå¼•é¢†de novo ç»„è£…è¿›å…¥ near finished genome æˆ–è€…finished genomeçš„æ–°æ—¶ä»£ã€‚</p>
<p>6ã€ä½†æ˜¯ï¼Œå°½ç®¡åšäº†å¾ˆå¤šåŠªåŠ›ï¼ŒARS1ä»ç„¶æ˜¯ä¸€ä¸ªå•ä½“å‹æ··åˆå‹çš„ç»„è£…ç‰ˆæœ¬ã€‚åœ¨æœªæ¥ï¼Œå¯ä»¥åˆ©ç”¨single moleculeå’ŒHi-CæŠ€æœ¯è·å¾—å•ä½“å‹åˆ†æœŸ(haplotype phasing) çš„å‚è€ƒåŸºå› ç»„ã€‚å¯¹äºç»„æˆå‹å¼‚æŸ“è‰²è´¨åŒºï¼Œå°¤å…¶æ˜¯ç€ä¸ç²’å’Œç«¯ç²’éƒ¨åˆ†ï¼Œä»ç„¶æ˜¯ç»„è£…çš„ç›²åŒºï¼Œå³ä½¿åœ¨äººçš„åŸºå› ç»„ä¸­è¿™äº›åŒºåŸŸä¹Ÿæ²¡æœ‰è¢«å®Œæ•´ç»„è£…å‡ºæ¥ã€‚</p>
<h1 id="3-2020-09-cell-å¤§è±†æ³›åŸºå› ç»„"><a href="#3-2020-09-cell-å¤§è±†æ³›åŸºå› ç»„" class="headerlink" title="3. 2020.09-cell å¤§è±†æ³›åŸºå› ç»„"></a>3. 2020.09-cell å¤§è±†æ³›åŸºå› ç»„</h1><p>Pan-Genome of Wild and Cultivated Soybeans</p>
<p>Soybean is one of the most important vegetable oil and protein feed crops. To capture the entire genomic diversity, it is needed to construct a complete high-quality pan-genome from diverse soybean accessions. In this study, we performed individual de novo genome assemblies for <strong>26 representative soybeans</strong> that were selected from <strong>2,898 deeply sequenced accessions</strong>. Using these assembled genomes together with <strong>three previously reported genomes</strong>, we constructed a <strong>graph-based genome</strong> and performed pan-genome analysis, which identified <strong>numerous genetic variations that cannot be detected by direct mapping of short sequence reads onto a single reference genome</strong>. The <strong>structural variations</strong> from the 2,898 accessions that were <strong>genotyped</strong> based on the graph-based genome and the <strong>RNA-seq data</strong> from the representative 26 accessions helped to <strong>link genetic variations to candidate genes</strong> that are responsible for important traits. This pan-genome resource will promote evolutionary and functional genomics studies in soybean.</p>
<h2 id="SNP-Calling-and-Phylogenetic-Analyses"><a href="#SNP-Calling-and-Phylogenetic-Analyses" class="headerlink" title="SNP Calling and Phylogenetic Analyses"></a>SNP Calling and Phylogenetic Analyses</h2><h2 id="Genome-Assembly"><a href="#Genome-Assembly" class="headerlink" title="Genome Assembly"></a>Genome Assembly</h2><p><em>De novo</em> assembly was conducted referring to a <font color="blue"><strong>reported pipeline</strong></font> (Du and Liang, 2019; Shen et al., 2019). In brief, <font color="red"><strong>Canu</strong></font> (Koren et al., 2017) v1.7.1 was used to assemble PacBio subreads to PacBio contigs, after which <font color="red"><strong>HiSeq</strong></font> reads were used for <font color="red"><strong>error correction</strong></font>. <strong>Bio-nano optical maps</strong> were assembled into consensus physical maps by <strong>BioNano Solve</strong> v3.0.1 (<a target="_blank" rel="noopener" href="https://bionanogenomics.com/">https://bionanogenomics.com/</a>, Solve_06082017Rel). Then <font color="red"><strong>HERA</strong></font> (Du and Liang, 2019) was used to combine PacBio contigs and Bionano based physical maps to <font color="blue"><strong>PacBio-BioNano hybrid scaffolds</strong></font>. To <font color="red"><strong>anchor</strong></font> hybrid scaffolds <font color="blue">into <strong>chromosomes</strong></font>, the <font color="blue"><strong>Hi-C</strong></font> sequencing data were <font color="red"><strong>aligned into scaffolds</strong></font> by <font color="red"><strong>Juicer</strong></font> (Durand et al., 2016) v1.5 and <font color="red"><strong>3D-DNA</strong></font> (Dudchenko et al., 2017).</p>
<h2 id="Repeat-Analysis-and-Gene-Annotation"><a href="#Repeat-Analysis-and-Gene-Annotation" class="headerlink" title="Repeat Analysis and Gene Annotation"></a>Repeat Analysis and Gene Annotation</h2><h2 id="Synteny-Analysis"><a href="#Synteny-Analysis" class="headerlink" title="Synteny Analysis"></a>Synteny Analysis</h2><p>Structural Variation Identification</p>
<p>Genetic Variation Analysis</p>
<p>Gene and miRNA Expression</p>
<p>Core and Dispensable Gene Family Clustering</p>
<p>CHS gene unit identification</p>
<p>Gene Fusion Event Identification</p>
<p>QUANTIFICATION AND STATISTICAL ANALYSIS</p>
<h1 id="4-2019-nc-å¤§è±†æ³›åŸºå› ç»„-ç»„è£…å‚è€ƒ1"><a href="#4-2019-nc-å¤§è±†æ³›åŸºå› ç»„-ç»„è£…å‚è€ƒ1" class="headerlink" title="4. 2019-nc å¤§è±†æ³›åŸºå› ç»„-ç»„è£…å‚è€ƒ1"></a>4. 2019-nc å¤§è±†æ³›åŸºå› ç»„-ç»„è£…å‚è€ƒ1</h1><p>HERA</p>
<h1 id="5-2019-09-scienceChina-å¤§è±†æ³›åŸºå› ç»„-ç»„è£…å‚è€ƒ2"><a href="#5-2019-09-scienceChina-å¤§è±†æ³›åŸºå› ç»„-ç»„è£…å‚è€ƒ2" class="headerlink" title="5. 2019.09-scienceChina å¤§è±†æ³›åŸºå› ç»„-ç»„è£…å‚è€ƒ2"></a>5. 2019.09-scienceChina å¤§è±†æ³›åŸºå› ç»„-ç»„è£…å‚è€ƒ2</h1><p><img src="/blog/2022/03/14/2022-03-14-HiFi-6-HiC-assembly/1.png" alt="1"></p>
<p>For genome assembly, we sequenced additional 40 Gb SMRT reads from 5 cells for this update. Therefore, a total of <strong>120Ã— PacBio reads</strong>, <strong>365Ã— Bionano optical maps</strong> marked by BssSI, <strong>275Ã— Bionano optical maps</strong> marked by BspQI, <strong>45Ã— HiSeq</strong> reads and <strong>125Ã—</strong> chromosome conformation capture sequencing (<strong>Hi-C</strong>) reads were used for the new genome assembly (Figure 1A). To make best of the sequencing data, we also <font color="red"><strong>adopted a new genome assembly pipeline</strong> by <strong>adding HERA</strong></font> (Du and Liang, 2018) to improve the sequence contiguity and reduce errors by accurately assembling the repetitive genome regions (Figure 1A). Briefly, <font color="blue">the whole assembly process <strong>differed from</strong> previous pipeline</font> at </p>
<p>(1) using <font color="red"><strong>CANU</strong></font> (v1.7.1) to replace Smrtmake for assembling PacBio subreads to PacBio contigs; </p>
<p>(2) using <font color="red"><strong>HERA</strong></font> to generate longer contigs; </p>
<p>(3) using <font color="red"><strong>Juicer and 3D-DNA to replace HiC-Pro and LACHESIS</strong></font> to anchor the hybrid scaffolds into chromosomes with Hi-C reads.</p>
<h1 id="6-2020-05-nc-åŒæºå››å€ä½“ç´«èŠ±è‹œè“¿"><a href="#6-2020-05-nc-åŒæºå››å€ä½“ç´«èŠ±è‹œè“¿" class="headerlink" title="6. 2020.05-nc åŒæºå››å€ä½“ç´«èŠ±è‹œè“¿"></a>6. 2020.05-nc åŒæºå››å€ä½“ç´«èŠ±è‹œè“¿</h1><p>HiFiåŠ©åŠ›åŒæºå››å€ä½“ç´«èŠ±è‹œè“¿åŸºå› ç»„ç»„è£…ï¼ŒNC</p>
<blockquote>
<p>Allele-aware chromosome-level genome assembly and efficient transgene-free genome editing for the autotetraploid cultivated alfalfa</p>
</blockquote>
<p>ï¼ˆ1ï¼‰ä½¿ç”¨Canué»˜è®¤å‚æ•°ï¼Œåˆ©ç”¨CCS clean readsç»„è£…contigsã€‚ç»„è£…å¾—åˆ°çš„Contig N50å€¼ä¸º459kbï¼Œæ€»é•¿åº¦ä¸º3.15GBã€‚</p>
<blockquote>
<p>We assembled contigs from CCS clean reads using Canu, with default parameters. The N50 values of the contig sets were 459 kb, with total lengths 3154 Mb.</p>
</blockquote>
<p>ï¼ˆ2ï¼‰ä½¿ç”¨HiC-Proå°†Hi-C readsä¸contigs è¿›è¡Œæ¯”å¯¹ï¼Œäº§ç”Ÿæ¯”å¯¹BAMæ–‡ä»¶ã€‚</p>
<blockquote>
<p>Hi-C reads were aligned to contigs using <a target="_blank" rel="noopener" href="https://pubmed.ncbi.nlm.nih.gov/26619908/">HiC-Pro(2015)</a>, yielding an alignment BAM file.</p>
</blockquote>
<p>ï¼ˆ3ï¼‰ä½¿ç”¨æ³¨é‡Šçš„è’ºè—œè‹œè“¿è›‹ç™½ä½œä¸ºå‚è€ƒï¼Œå®Œå…¨åŸºäºåŒæºçš„ç­–ç•¥æ³¨é‡Šcontigsã€‚å¯¹138,729ä¸ªåŒæºåŸºå› è¿›è¡Œäº†ç»“æ„æ³¨é‡Šã€‚ç”¨MCscanç”¨äºé‰´å®šcontigså’Œå‚è€ƒåŸºå› ç»„ä¹‹é—´çš„å…±çº¿æ€§ã€‚æ˜¾ç¤ºç´«èŠ±è‹œè“¿å’Œè’ºè—œè‹œè“¿ä¹‹é—´çš„é«˜å…±çº¿æ€§ã€‚ã€ä¸‹è½½çˆ¶æ¯æœ¬petunia refã€‘</p>
<blockquote>
<p>Contigs were annotated with a <font color="red"><strong>solely homology-based strategy</strong></font>, using annotated Medicago truncatula proteins as references. 138,729 homologous genes were structurally annotated. <font color="red"><strong>MCscan</strong> in <strong>Jcvi</strong></font> (<a target="_blank" rel="noopener" href="https://zenodo.org/record/31631#.XpkUyTOeask">https://zenodo.org/record/31631#.XpkUyTOeask</a>) was used to identify <font color="red"><strong>synteny blocks</strong></font> between <strong>contigs and the reference</strong> genome. <font color="red"><strong>Contigs syntenic</strong> to M. truncatula were <strong>stacked and aligned</strong> to M. truncatula chromosomes</font>. The syntenic contigs are summarized in Supplementary Table 4.</p>
</blockquote>
<p>ï¼ˆ4ï¼‰ä½¿ç”¨å†…éƒ¨è„šæœ¬å¤„ç†BAMæ–‡ä»¶ï¼Œå»é™¤ç­‰ä½åŸºå› contigsä¹‹é—´çš„linksã€‚ä½¿ç”¨ALLHiCè½¯ä»¶ï¼Œ<font color="red">æå–ã€èšç±»å’Œé‡æ’Contigs</font> (Contigs syntenicä¸è’ºè—œè‹œè“¿æŸ“è‰²ä½“ä¸€è‡´)ï¼Œå¾—åˆ°åŸå§‹çš„scaffoldsã€‚</p>
<blockquote>
<p>An <strong>in-house script</strong> was used to <font color="red">prune the BAM</font> file and <font color="red">discard links between allelic contigs</font>. Contigs syntenic to one chromosome of M. truncatula, e.g., chr1, were extracted, sub-clustered and reordered using <font color="red">ALLHiC</font>, yielding a raw scaffold set.</p>
</blockquote>
<p>ï¼ˆ5ï¼‰<a target="_blank" rel="noopener" href="https://pubmed.ncbi.nlm.nih.gov/27467250/">Juicebox</a>ç”¨äºä»¥å›¾å½¢å’Œäº¤äº’æ–¹å¼å¾®è°ƒç»„è£…çš„scaffoldsã€‚å‰ªè£äº†40ä¸ªæ€»é•¿åº¦è¾¾1800Mbçš„scaffolds ã€‚</p>
<blockquote>
<p>Juicebox45 was used for fine-tuning assembled scaffolds in a graphic and inter-active fashion. Forty scaffolds with a total length of 1800 Mb were cropped (Supplementary Table 5).</p>
</blockquote>
<p>ï¼ˆ6ï¼‰ åŸºäºç»„è£…çš„scaffoldï¼Œé€šè¿‡Hi-Cæ•°æ®ï¼Œæ¯ä¸ªunplaced contigè¢«åˆ†é…åˆ°äº’ä½œæœ€å¼ºçš„é‚£äº›contig clusteré‡Œã€‚</p>
<blockquote>
<p>Based on this scaffold assembly, each unplaced contig was assigned to the contig cluster, to which the contig was most connected by Hi-C data.</p>
</blockquote>
<p>ï¼ˆ7ï¼‰ä½¿ç”¨ALLHiCå¯¹é‚£äº›contig clusterså†æ¬¡è¿›è¡Œé‡æ’å’Œæ„å»ºscaffoldã€‚</p>
<blockquote>
<p>Those contig clusters were reordered and scaffolded using ALLHiC. </p>
</blockquote>
<p>ï¼ˆ8ï¼‰ä½¿ç”¨Juiceboxå¯¹scaffoldsè¿›è¡Œå¾®è°ƒï¼Œå¹¶ä»scaffoldsä¸Šå»é™¤ä¸ä¸€è‡´çš„contigsï¼Œäº§ç”Ÿæœ€ç»ˆçš„æŸ“è‰²ä½“åŸºå› ç»„ï¼Œå…¶åŒ…å«32æ¡æŸ“è‰²ä½“(8ä¸ªåŒæºç»„ï¼Œæ¯ä¸ªç»„ä¸­æœ‰4ä¸ªç­‰ä½åŸºå› æŸ“è‰²ä½“)ï¼Œæ€»é•¿åº¦ä¸º2738Mbï¼Œå’Œ419MbæœªæŒ‚è½½åˆ°æŸ“è‰²ä½“æ°´å¹³çš„åºåˆ—ã€‚</p>
<blockquote>
<p>Using Juicebox, scaffolds were fine-tuned and discordant contigs were removed from scaffolds, and the final chromosome assembly was generated, containing 32 chromosomes with a total length of 2738 Mb (Supplementary Table 6).</p>
</blockquote>
<h1 id="7-2021-11-gb-æ¤°å­ğŸ¥¥"><a href="#7-2021-11-gb-æ¤°å­ğŸ¥¥" class="headerlink" title="7. 2021.11-gb æ¤°å­ğŸ¥¥"></a>7. 2021.11-gb æ¤°å­ğŸ¥¥</h1><p>High-quality reference genome sequences of two coconut cultivars provide insights into evolution of monocot chromosomes and differentiation of fiber content and plant height</p>
<p>è¯¥ç ”ç©¶ç»˜åˆ¶ä¸¤ä¸ªå‚è€ƒçº§åˆ«æ¤°å­åŸºå› ç»„ã€‚æ­ç¤ºå•å­å¶æ¤ç‰©æŸ“è‰²ä½“çš„è¿›åŒ–è¿‡ç¨‹å’Œæ¤°å­æŸ“è‰²ä½“çš„å½¢æˆè¿‡ç¨‹ï¼Œåˆ†æé«˜çŸ®ä¸¤ç§æ¤°å­è€ç›ï¼Œæ ªé«˜ï¼Œçº¤ç»´å’Œè„‚è´¨å«é‡ç­‰å…³é”®æ€§çŠ¶çš„é—ä¼ å·®å¼‚ï¼ŒåŒæ—¶å‘ç°æ¤°å­è¿›åŒ–è¿‡ç¨‹ä¸­çº¦åœ¨400ä¸‡å¹´å‰è‡ªç„¶å‘ç”Ÿçš„ç»¿è‰²é©å‘½ã€‚é«˜æ¤°å­2.39GçŸ®æ¤°å­2.40Gã€‚åŸºäºæ¤°å­ï¼Œç´«èï¼Œè èç­‰åŸºå› ç»„çš„å…±çº¿æ€§å…³ç³»ï¼Œé‡æ–°æ„å»ºäº†å«æœ‰10æ¡æŸ“è‰²ä½“çš„å¤å•å­å¶æ¤ç‰©åŸºå› ç»„æ ¸å‹ï¼Œå¹¶å¯¹åŸºå› ç»„æ ¸å‹çš„è¿›åŒ–è·¯çº¿è¿›è¡Œæ¨æ–­ã€‚è¿›ä¸€æ­¥åˆ©ç”¨å¶é—´è·è¡¨å‹è¿›è¡Œå…¨åŸºå› ç»„å…³è”åˆ†æï¼Œé‰´å®šå‡º12å·æŸ“è‰²ä½“ä¸ŠGA20oxåŸºå› ä¸æ¤°å­æ ªé«˜æ˜¾è‘—ç›¸å…³ï¼Œå¹¶å­˜åœ¨æ‹·è´æ•°å˜å¼‚ï¼›æ•´åˆå¤šç§æ‰‹æ®µéªŒè¯HA20oxä»¥åŠèµ¤éœ‰ç´ ä»£è°¢é€”å¾„åœ¨é«˜çŸ®ç§çš„å·®å¼‚å¹¶å¯¼è‡´æ ªé«˜å·®å¼‚ã€‚400ä¸‡å¹´å‰ï¼Œç»¿è‰²é©å‘½ã€‚</p>
<h2 id="Hi-C-based-genome-assembly"><a href="#Hi-C-based-genome-assembly" class="headerlink" title="Hi-C-based genome assembly"></a>Hi-C-based genome assembly</h2><p>Both Cn. tall and Cn. dwarf genomes were assembled de novo using SMARTdenovo (<a target="_blank" rel="noopener" href="https://github.com/ruanjue/smartdenovo">https://github.com/ruanjue/smartdenovo</a>) [-k 19 -J 3000] based on <font color="red"><strong>Nanopore long reads</strong></font> <strong>corrected</strong> with <font color="red"><strong>NextDenovo</strong></font> (<a target="_blank" rel="noopener" href="https://github.com/Nextomics/NextDenovo">https://github.com/Nextomics/NextDenovo</a>). Raw contigs were <font color="blue"><strong>polished</strong></font> using <font color="blue"><strong>Nanopore long reads</strong></font> and <font color="blue"><strong>Illumina short reads</strong></font> with <font color="red"><strong>Next-Polish</strong></font> (v1.2.2) [49]. Then polished contigs were <font color="blue"><strong>anchored</strong></font> into chromosomes by using <font color="red"><strong>HiC-Pro (v2.11.1)</strong></font> [50] and <font color="red"><strong>LACHESIS</strong></font>[51]. Finally, the assembled genomes were <font color="blue"><strong>manual correction</strong></font> with <font color="red"><strong>Juicebox</strong></font> (v1.11.08).</p>
<p>To assess the quality of assembled genomes, raw Illumina <font color="blue"><strong>paired-end</strong> reads and <strong>RNA-seq</strong></font> reads from multiple tissues were <font color="blue"><strong>mapping</strong></font> to the genomes using <font color="red"><strong>BWA (v0.7.17-r1188)</strong></font> [52] and <font color="red"><strong>HISAT2 (v2.1.0)</strong></font> [53], respectively. In addition, Benchmarking Universal Single-Copy Orthologs (<font color="red"><strong>BUSCO</strong></font>) was also used to assess the genome completeness base on Embryophyta Plant database (odb10) [54].</p>
<h1 id="8-2021-11-nc-é‡‘ç²Ÿå…°åŸºå› ç»„"><a href="#8-2021-11-nc-é‡‘ç²Ÿå…°åŸºå› ç»„" class="headerlink" title="8. 2021.11-nc é‡‘ç²Ÿå…°åŸºå› ç»„"></a>8. 2021.11-nc é‡‘ç²Ÿå…°åŸºå› ç»„</h1><p>å¡«è¡¥æ ¸å¿ƒè¢«å­æ¤ç‰©æœ€åä¸»è¦åˆ†æ”¯-é‡‘ç²Ÿå…°ç›®çš„åŸºå› ç»„ä¿¡æ¯ï¼Œæä¾›è¢«å­æ¤ç‰©æ—©æœŸæ¼”åŒ–è¯æ®ã€‚åˆ˜å»ºå…¨å›¢é˜Ÿè‡´åŠ›äºè§£æç±»ç¾¤æ¼”åŒ–ï¼Œæ¢ç©¶ä¸åŒç‰©ç§é—´åˆ†åŒ–é€‚åº”é—ä¼ æœºåˆ¶ã€‚</p>
<p>è¢«å­æ¤ç‰©ï¼ˆAngiosperms or flowering plantsï¼‰æ˜¯åœ°çƒä¸Šåˆ†å¸ƒèŒƒå›´æœ€å¹¿ã€å¤šæ ·åŒ–æœ€é«˜ã€é€‚åº”æ€§æœ€å¼ºçš„é™†ç”Ÿæ¤ç‰©ç±»ç¾¤ï¼Œæå¤§åœ°å½±å“äº†å…¶ä»–ç”Ÿç‰©ç±»ç¾¤çš„æ¼”åŒ–è¿›ç¨‹ï¼Œç„¶è€Œï¼Œç”±äºè¢«å­æ¤ç‰©åœ¨æ—©æœŸç»å†äº†å¿«é€Ÿçš„è¾å°„è¿›åŒ–ï¼Œå…¶å†…éƒ¨æ¼”åŒ–å…³ç³»è¿Ÿè¿Ÿæœªè¢«è§£å†³ï¼Œè¿™ä¸€é—®é¢˜ä¹Ÿè¢«ç§°ä¸ºè¾¾å°”æ–‡çš„â€œæ¼äººä¹‹è°œâ€ã€‚ç»è¿‡å‰äººé•¿æ—¶æœŸçš„åŠªåŠ›ï¼Œç›®å‰è¢«å­æ¤ç‰©è¢«åˆ’åˆ†ä¸ºï¼šæ— æ²¹æ¨Ÿç›®ã€ç¡è²ç›®ã€æœ¨å…°è—¤ç›®å’Œæ ¸å¿ƒè¢«å­æ¤ç‰©ï¼ˆMesangiospermaeï¼‰ã€‚æ— æ²¹æ¨Ÿç›®ã€ç¡è²ç›®ã€æœ¨å…°è—¤ç›®å…±åŒè¢«ç§°ä¸ºANA gradeï¼Œæ˜¯è¢«å­æ¤ç‰©çš„åŸºéƒ¨ç±»ç¾¤ï¼Œç³»ç»Ÿå‘è‚²å…³ç³»ç›¸å¯¹ç¨³å®šï¼Œè€Œæ ¸å¿ƒè¢«å­æ¤ç‰©åŒ…å«äº†~99.95%çš„è¢«å­æ¤ç‰©ï¼Œåˆå¯è¢«åˆ’åˆ†ä¸º5å¤§ç±»ç¾¤ï¼šåŒå­å¶ã€å•å­å¶ã€æœ¨å…°ç±»ã€é‡‘é±¼è—»ç›®å’Œé‡‘ç²Ÿå…°ç›®ï¼Œå®ƒä»¬ä¹‹é—´çš„æ¼”åŒ–å…³ç³»ä¸€ç›´å­˜åœ¨äº‰è®®ã€‚è¿‘å¹´æ¥ï¼Œéšç€å„ç±»ç¾¤ä»£è¡¨ç‰©ç§å…¨åŸºå› ç»„æµ‹åºç›¸ç»§å®Œæˆï¼Œå…¶å„è‡ªçš„è¿›åŒ–å†ç¨‹å‡æœ‰äº†åšå®è€Œå¯é çš„è¯æ®ï¼Œåˆ˜å»ºå…¨å›¢é˜Ÿåœ¨2020å¹´é¦–æ¬¡æŠ¥é“äº†é‡‘é±¼è—»å’ŒèŠ¡å®çš„åŸºå› ç»„ï¼Œè¯æ˜äº†æ ¸å¿ƒè¢«å­æ¤ç‰©å†…éƒ¨å­˜åœ¨å¹¿æ³›çš„ä¸å®Œå…¨è°±ç³»åˆ†é€‰å’Œæ‚äº¤äº‹ä»¶æ˜¯å¯¼è‡´æ ‘å½¢å†²çªçš„ä¸»è¦åŸå› ã€‚ä½†æ˜¯é‡‘ç²Ÿå…°ç›®ä¸€ç›´æœªæœ‰åŸºå› ç»„æŠ¥é“ï¼Œå…¶æ¼”åŒ–å†ç¨‹ä»¥åŠæ•´ä¸ªæ ¸å¿ƒè¢«å­æ¤ç‰©çš„æ¼”åŒ–è¿‡ç¨‹è¿˜æœ‰å¾…æ·±å…¥åˆ†æã€‚ä¸ºæ­¤è¯¥å›¢é˜Ÿå†æ¬¡åˆ©ç”¨Nanoporeã€Illuminaå’ŒHi-CæŠ€æœ¯ï¼Œæ„å»ºäº†ç¬¬ä¸€ä¸ªé‡‘ç²Ÿå…°ç›®æ¤ç‰©ï¼ˆå››å·é‡‘ç²Ÿå…°ï¼Œ<em>Chloranthus sessilifolius</em>ï¼‰çš„é«˜è´¨é‡æŸ“è‰²ä½“çº§åˆ«åŸºå› ç»„ï¼ŒContig N50é«˜è¾¾53.74 Mbï¼ˆå›¾1ï¼‰ï¼Œä½¿ç”¨å¤šç§åˆ†ææ–¹æ³•å¯¹ä¸Šè¿°é—®é¢˜è¿›è¡Œäº†æ·±å…¥çš„é˜è¿°ã€‚</p>
<p>é€šè¿‡ä½¿ç”¨å¤šç§ç³»ç»Ÿå‘è‚²åŸºå› ç»„å­¦åˆ†æï¼ŒåŒ…æ‹¬å•æ‹·è´åŸºå› ã€ä½æ‹·è´åŸºå› å’Œå…±çº¿æ€§åŸºå› æ•°æ®é›†ï¼Œè¿›è¡Œäº†ä¸²è”æ ‘å’Œæº¯ç¥–æ ‘çš„æ„å»ºï¼ŒåŒæ—¶è¯„ä¼°äº†é•¿æå¸å¼•ã€ç‰©ç§é€‰æ‹©å’ŒåŒæºèšç±»æ–¹æ³•ç­‰çš„å½±å“ï¼Œä½¿ç”¨å¤šå¥—æ•°æ®é›†åˆå’Œå¤šç§åˆ†ææ–¹æ³•ï¼Œè·å¾—äº†é«˜å¯ä¿¡åº¦çš„æ‹“æ‰‘ç»“æ„ï¼Œå³é‡‘ç²Ÿå…°ç›®å’Œæœ¨å…°ç±»å…·æœ‰æœ€è¿‘çš„äº²ç¼˜å…³ç³»ï¼Œä¸”ä»–ä»¬ä¸€èµ·ä¸é‡‘é±¼è—»ç›®+åŒå­å¶æ¤ç‰©æ„æˆå§Šå¦¹å…³ç³»ï¼Œè€Œå•å­å¶æ¤ç‰©æ˜¯å…¶å®ƒæ‰€æœ‰æ ¸å¿ƒè¢«å­æ¤ç‰©çš„å§Šå¦¹æï¼ˆå›¾2ï¼‰ã€‚</p>
<p>åŒæ—¶ï¼Œä½œè€…ä¹Ÿå‘ç°äº†å¤§é‡çš„åŸºå› æ ‘æˆ–å¶ç»¿ä½“æ ‘ä¸ç‰©ç§æ ‘ä¸ä¸€è‡´çš„æƒ…å†µï¼Œå› æ­¤ç€é‡è¯„ä¼°äº†ä¸å®Œå…¨è°±ç³»åˆ†é€‰ï¼ˆILSï¼‰å’Œæ‚äº¤äº‹ä»¶å¯¹æ­¤å†²çªçš„è´¡çŒ®ã€‚é€šè¿‡PhyloNetworksåˆ†æé‰´å®šåˆ°äº†ä¸‰æ¬¡å¯èƒ½çš„æ‚äº¤äº‹ä»¶ï¼Œå¯ä»¥è§£æéƒ¨åˆ†çš„æ ‘å½¢å†²çªã€‚ç„¶è€Œå€¼å¾—æ³¨æ„çš„æ˜¯äº”å¤§ç±»ç¾¤ä¹‹é—´çš„åˆ†åŒ–æ—¶é—´ä»…æœ‰23ä¸ªç™¾ä¸‡å¹´ï¼ˆ158~135 Myaï¼‰ï¼Œå¦‚æ­¤çŸ­çš„åˆ†åŒ–æ—¶é—´ï¼ŒILSå¯èƒ½å‘æŒ¥äº†æ›´ä¸ºä¸»è¦çš„ä½œç”¨ã€‚ä½¿ç”¨thetaå‚æ•°å¯¹æ¯ä¸ªå†…éƒ¨åˆ†æ”¯å‘ç”ŸILSçš„æ¦‚ç‡è¿›è¡Œè¯„ä¼°å‘ç°ï¼Œé™¤äº†æœ¨å…°ç±»å’Œé‡‘ç²Ÿå…°ç›®çš„å§Šå¦¹å…³ç³»è¾ƒä¸ºç¨³å®šå¤–ï¼Œå…¶ä»–ç±»ç¾¤é—´éƒ½å…·æœ‰è¾ƒé«˜çš„thetaå€¼ï¼Œè€Œé‡‘é±¼è—»ç›®+åŒå­å¶çš„ç¥–å…ˆåˆ†æå…·æœ‰è¾ƒé«˜çš„tehtaå€¼ï¼Œå‡è¡¨æ˜äº†ILSåœ¨æ ¸å¿ƒè¢«å­æ¤ç‰©å†…éƒ¨çš„æ¼”åŒ–ä¸­å‘æŒ¥äº†é‡è¦ä½œç”¨ï¼ˆå›¾3 aå’Œbï¼‰ã€‚åŸºäºPhyabseå’ŒDendroPyè¿›è¡Œæ ‘å½¢æ¨¡æ‹Ÿè¡¨æ˜ï¼Œå½“å­˜åœ¨ILSæ—¶æ¨¡æ‹Ÿçš„æ ‘å½¢å’ŒçœŸå®æ ‘å½¢å…·æœ‰é«˜åº¦ä¸€è‡´æ€§ï¼ˆR2ï¼0.99ï¼‰ï¼Œè€Œä¸å­˜åœ¨ILSçš„æ¨¡æ‹Ÿæ ‘å½¢ä¸çœŸå®æ ‘å½¢åˆ™å…·æœ‰è¾ƒä½çš„ç›¸å…³æ€§ï¼Œè¿›ä¸€æ­¥è¡¨æ˜äº†ILSåœ¨æ ¸å¿ƒè¢«å­æ¤ç‰©çš„å¿«é€Ÿè¾å°„ä¸­å‘æŒ¥äº†é‡è¦ä½œç”¨ï¼ˆå›¾3 c-fï¼‰ã€‚è¿›ä¸€æ­¥å¯¹æ ‘å½¢å¼‚è´¨æ€§è¿›è¡ŒåŒºåˆ†è¡¨æ˜ï¼ŒILSå¯åŸºæœ¬è§£é‡Šé‡‘é±¼è—»ç›®+åŒå­å¶çš„ç¥–å…ˆæä¸Šçš„æ ‘å½¢å†²çªï¼Œä¹Ÿå°±æ˜¯é‡‘é±¼è—»ç›®+åŒå­å¶ã€æœ¨å…°ç±»+é‡‘ç²Ÿå…°å’Œå•å­å¶ä¹‹é—´çš„å†²çªï¼Œè€Œå…¶ä»–æ ¸å¿ƒè¢«å­æ¤ç‰©ç±»ç¾¤é—´çš„å†²çªè¿˜æœ‰å…¶ä»–å› ç´ ï¼ˆä¾‹å¦‚æ‚äº¤ï¼‰çš„å‚ä¸ã€‚</p>
<p>æ­¤å¤–ï¼Œæœ¬ç ”ç©¶è¿˜é‰´å®šäº†å››å·é‡‘ç²Ÿå…°ç‹¬æœ‰çš„ä¸€æ¬¡WGDäº‹ä»¶ï¼ŒåŒæ—¶5å¤§æ ¸å¿ƒè¢«å­æ¤ç‰©åˆ†æçš„åŠ å€äº‹ä»¶éƒ½ä¸å…±äº«ï¼›é‰´å®šäº†èŠ±å‘è‚²ç›¸å…³çš„åŸºå› å®¶æ—ï¼Œè§£æäº†å…¶ç‰¹æœ‰èŠ±å™¨å®˜çš„å¯èƒ½å½¢æˆåŸå› ï¼›é‰´å®šäº†èœç±»åˆæˆå’Œæœ¨è´¨éƒ¨å½¢æˆç›¸å…³çš„åŸºå› å®¶æ—ï¼Œä¸ºä»Šåç ”ç©¶å…¶ç®€å•æœ¨è´¨éƒ¨å½¢æˆå’Œèœç±»ç‰©ç§åˆæˆé€šè·¯æ¼”åŒ–æä¾›äº†åŸºç¡€ã€‚</p>
<p>ç»¼ä¸Šæ‰€è¿°ï¼Œè¯¥ç ”ç©¶æŠ¥é“äº†é«˜è´¨é‡çš„å››å·é‡‘ç²Ÿå…°åŸºå› ç»„ï¼Œå¡«è¡¥äº†æ ¸å¿ƒè¢«å­æ¤ç‰©é‡‘ç²Ÿå…°ç›®çš„åŸºå› ç»„ä¿¡æ¯ï¼Œå¹¶é€šè¿‡å¤šç§åˆ†æç­–ç•¥å’Œæ–¹æ³•ï¼Œæ·±å…¥çš„æ­ç¤ºäº†æ ¸å¿ƒè¢«å­æ¤ç‰©é—´çš„å¤æ‚æ¼”åŒ–å†ç¨‹ï¼Œè¯¥æˆæœå¯¹äºæ—©æœŸè¢«å­æ¤ç‰©çš„èµ·æºã€è¾å°„è¿›åŒ–ä»¥åŠé€‚åº”æ€§æ¼”åŒ–æä¾›æ•°æ®æ”¯æ’‘å’Œç†è®ºåŸºç¡€ã€‚</p>
<h2 id="Genome-size-estimate-and-assembly"><a href="#Genome-size-estimate-and-assembly" class="headerlink" title="Genome size estimate and assembly."></a>Genome size estimate and assembly.</h2><p>To estimate the genome size of C. sessilifolius, we surveyed 150 bp paired-end reads, computed 21 bp K-mer frequencies using <font color="red"><strong>Jellyfish</strong></font>, and exported the resulting histogram into <font color="red">findGSE</font>. <font color="red"><strong>Nextdenovo</strong></font> (https:// github.com/Nextomics/Nextdenovo) was selected for <font color="blue"><strong>correcting reads</strong></font> with para- meters â€œread_cutoff=2k, seed_cutoff=30k, blocksize=1.5gâ€ and then <font color="red"><strong>Smartdenovo</strong></font> (<a target="_blank" rel="noopener" href="https://github.com/ruanjue/smartdenovo">https://github.com/ruanjue/smartdenovo</a>) for <font color="blue"><strong>de novo assembly</strong></font> with parameters â€œwtpre -J 3,000; wtzmo-k 21 -z 10 -Z 16 -U -1 -m 0.1 -A 1000; wtclp -d 3 -k 300 -m 0.1 -FT; wtlay -w 300 -s 200 -m 0.1 -r 0.95 -c 1â€. <font color="blue">The preliminary contigs were further <strong>polished</strong> by aligning the Illumina <strong>short</strong> <strong>reads</strong> to the contigs</font> using <font color="red"><strong>Nextpolish</strong></font>. After four rounds of successive iterative correction, the final genome sequence was obtained. The GC content and sequencing coverage analyses were applied to evaluate the presence of contamination. The quality of the genome assembly was also assessed using <font color="red"><strong>BUSCO</strong></font> (Benchmarking Universal Single-Copy Orthologs) with the embry- ophyta_odb10 database. The clean <font color="blue">Hi-C</font> data were mapped to contig sequences by <font color="red"><strong>Bowtie2</strong></font> and 354 Mb valid interaction pairs were extracted. Based on those chromatin interactions, <font color="red"><strong>LACHESIS</strong></font> was employed to <font color="blue"><strong>cluster</strong>, <strong>order</strong>, and <strong>orient</strong> the <strong>contigs</strong> into <strong>pseudo-chromosomes</strong></font>.</p>
<h1 id="9-2022-01-gb-æ²¹èŒ¶"><a href="#9-2022-01-gb-æ²¹èŒ¶" class="headerlink" title="9. 2022.01-gb æ²¹èŒ¶"></a>9. 2022.01-gb æ²¹èŒ¶</h1><blockquote>
<p>The genome of oil-Camellia and population genomics analysis provide insights into seed oil domestication</p>
</blockquote>
<p>æ²¹èŒ¶æ˜¯æˆ‘å›½ä¼ ç»Ÿæœ¨æœ¬æ²¹æ–™æ ‘ç§ï¼Œå…·æœ‰2300ä½™å¹´æ ½åŸ¹é£Ÿç”¨å†å²ã€‚è‡ªä¸Šä¸–çºªèµ·ï¼Œç»è¿‡å››ä»£ç§‘æŠ€å·¥ä½œè€…è‰°è‹¦åŠªåŠ›ï¼Œæˆ‘å›½æ²¹èŒ¶ä¸»æ ½è‰¯ç§è¡ç”Ÿå‡ºæ•°ç™¾ä¸ªå“ç§ã€‚ä½†å—åˆ¶äº<font color="blue">å¤šå€æ€§ã€é•¿æ—¶æ•ˆ</font>ç‰¹æ€§ï¼Œæ²¹èŒ¶è‚²ç§å·¥ä½œæ•ˆç‡ä¸é«˜ã€‚</p>
<p>2022å¹´1æœˆ10æ—¥ï¼Œä¸­å›½æ—ä¸šç§‘å­¦ç ”ç©¶é™¢äºšçƒ­å¸¦æ—ä¸šç ”ç©¶æ‰€ï¼ˆäºšæ—æ‰€ï¼‰ç ”ç©¶å‘˜<strong>å§šå°åå›¢é˜Ÿ</strong>å’Œ<strong>æ®·æ’ç¦å›¢é˜Ÿ</strong>åœ¨<strong>ã€ŠåŸºå› ç»„ç”Ÿç‰©å­¦ã€‹ï¼ˆGenome Biologyï¼‰</strong>å‘è¡¨æœ€æ–°è®ºæ–‡ã€‚</p>
<p>è¯¥ç ”ç©¶æˆåŠŸç»„è£…å…¨çƒé¦–ä¸ªæŸ“è‰²ä½“çº§åˆ«é«˜è´¨é‡æ²¹èŒ¶åŸºå› ç»„ï¼Œæ­ç¤º<font color="red">æ²¹èŒ¶ç‰©ç§<strong>è¿›åŒ–å†å²</strong>åŠå…¶<strong>ç§å­å«é«˜æ²¹è„‚</strong>ã€<strong>é«˜ä¸é¥±å’Œè„‚è‚ªé…¸</strong>çš„<strong>é©¯åŒ–æœºåˆ¶</strong></font>ï¼Œå»ºç«‹äº†æ²¹è„‚æ€§çŠ¶æ—©æœŸé€‰æ‹©æŠ€æœ¯ä½“ç³»ï¼Œä¸ºåŠ å¿«è‰¯ç§é€‰è‚²æ•ˆç‡ã€ä¿éšœæˆ‘å›½ç²®æ²¹å®‰å…¨å¥ å®šäº†åŸºç¡€ã€‚</p>
<h2 id="æŠ±å­æ€€èƒçš„æ²¹èŒ¶è‰¯ç§è‚²ç§ç¼“æ…¢"><a href="#æŠ±å­æ€€èƒçš„æ²¹èŒ¶è‰¯ç§è‚²ç§ç¼“æ…¢" class="headerlink" title="æŠ±å­æ€€èƒçš„æ²¹èŒ¶è‰¯ç§è‚²ç§ç¼“æ…¢"></a>æŠ±å­æ€€èƒçš„æ²¹èŒ¶è‰¯ç§è‚²ç§ç¼“æ…¢</h2><p>è‰¯ç§æ˜¯å†œä¸šç”Ÿäº§çš„â€œèŠ¯ç‰‡â€ã€‚ä½œä¸ºæˆ‘å›½å››å¤§æ²¹æ–™ä½œç‰©ä¹‹ä¸€ï¼Œæ²¹èŒ¶æ˜¯å†œæ°‘å¢æ”¶ã€ç²®æ²¹å®‰å…¨ã€ç”Ÿæ€å»ºè®¾çš„â€œé‡è¦æ³•å®â€ã€‚æˆ‘å›½æ²¹èŒ¶è‚²ç§å†ç»äº†é€‰æ‹©è‚²ç§å’Œå®šå‘æ‚äº¤è‚²ç§é˜¶æ®µï¼Œä¸»æ ½è‰¯ç§ç”±å®¶ç³»å“ç§ã€å†œå®¶å“ç§èµ°å‘æ— æ€§ç³»è‰¯ç§ï¼Œè¡ç”Ÿå‡ºæ•°ç™¾ä¸ªå“ç§ï¼Œæ„æˆäº†æˆ‘å›½æ²¹èŒ¶äº§ä¸šçš„å‘å±•åŸºç¡€ã€‚</p>
<p>ä½†ç”±äºæ²¹èŒ¶ç»“å®ä¹‹å‰è¦ç»å†æ¼«é•¿çš„ç«¥æœŸï¼Œè€Œä¸”ä»<font color="blue">å¼€èŠ±åˆ°æœå®æˆç†Ÿéœ€è¦æ•´æ•´1å¹´æ—¶é—´</font>ï¼Œä¿—ç§°â€˜æŠ±å­æ€€èƒâ€™ï¼Œå¯¼è‡´æ²¹èŒ¶è‚²ç§å¹´é™é•¿ï¼Œæ–°å“ç§é€‰è‚²ç¼“æ…¢ï¼Œè‰¯ç§é€‰è‚²é€Ÿåº¦æ— æ³•æ»¡è¶³äº§ä¸šå‘å±•çš„éœ€æ±‚ï¼Œå·²æˆä¸ºé˜»ç¢æ²¹èŒ¶äº§ä¸šå‘å±•çš„é‡è¦å› ç´ ã€‚</p>
<p>æˆ‘å›½é•¿æœŸé¢ä¸´é£Ÿç”¨æ²¹ä¾›éœ€å‹åŠ›ï¼Œå¯¹å¤–ä¾å­˜åº¦é«˜è¾¾60%ã€‚æ²¹èŒ¶èƒ½å¤Ÿåˆ©ç”¨æˆ‘å›½äºšçƒ­å¸¦åŒºåŸŸçš„å®œæ—å±±åœ°ç§æ¤ï¼Œä¸ä¸ç²®äº‰åœ°ï¼Œæ˜¯ä¿éšœæˆ‘å›½ç²®æ²¹å®‰å…¨ã€å®ç°ä¹¡æ‘æŒ¯å…´çš„é‡è¦æŠ“æ‰‹ã€‚</p>
<p>æ²¹èŒ¶æ³›æŒ‡å±±èŒ¶ç§‘å±±èŒ¶å±æ¤ç‰©ä¸­æ²¹è„‚å«é‡é«˜ï¼Œå…·æœ‰ä¸€å®šæ ½åŸ¹é¢ç§¯çš„ï¼Œæœ‰ç»æµæ ½åŸ¹ä»·å€¼çš„ç‰©ç§çš„æ€»ç§°ï¼Œæ˜¯æˆ‘å›½ä¸»æ¨çš„é‡è¦æœ¨æœ¬æ²¹æ–™æ ‘ç§ã€‚æˆ‘å›½ç°æœ‰æ²¹èŒ¶ç§æ¤é¢ç§¯7000ä¸‡äº©ï¼Œä»¥æ™®é€šæ²¹èŒ¶ï¼ˆCamellia oleiferaï¼‰ä¸ºä¸»ï¼Œå¹´æ€»äº§å€¼1200äº¿å…ƒå·¦å³ã€‚</p>
<p>æ²¹èŒ¶ç±½æ²¹ä¸­ä¸é¥±å’Œè„‚è‚ªé…¸è¾¾90%ä»¥ä¸Šï¼Œå…¶ä¸­ä»¥å•ä¸é¥±å’Œè„‚è‚ªé…¸æ²¹é…¸ä¸ºä¸»ï¼Œå¯Œå«è§’é²¨çƒ¯ã€ç»´ç”Ÿç´ Eã€ç”¾é†‡ã€å¤šé…šç­‰åŠŸèƒ½æˆåˆ†ï¼Œå…·æœ‰æé«˜çš„ä¿å¥ä»·å€¼ï¼Œè¢«èª‰ä¸ºâ€œæ¶²ä½“é»„é‡‘â€ã€‚</p>
<p>ç„¶è€Œç ”ç©¶å‘ç°, æ²¹èŒ¶åŸºå› ç»„å…·æœ‰<font color="red"><strong>æ‚åˆç‡é«˜ã€é‡å¤åºåˆ—å æ¯”å¤§</strong>çš„ç‰¹ç‚¹ï¼Œå…¶æ‚åˆç‡è¾¾åˆ°äº†<strong>2.52%</strong></font>ï¼Œè¿œé«˜äºå¤§éƒ¨åˆ†å·²æµ‹åºçš„ç‰©ç§ï¼Œè¿™ä¸ºåŸºå› ç»„ç»„è£…å¸¦æ¥äº†å·¨å¤§æŒ‘æˆ˜ã€‚</p>
<h2 id="æŸ“è‰²ä½“çº§åˆ«ç»„è£…"><a href="#æŸ“è‰²ä½“çº§åˆ«ç»„è£…" class="headerlink" title="æŸ“è‰²ä½“çº§åˆ«ç»„è£…"></a>æŸ“è‰²ä½“çº§åˆ«ç»„è£…</h2><p>ç»è¿‡å››å¹´å¤šçš„åŠªåŠ›ï¼Œè¯¥å›¢é˜Ÿè·å¾—å¤§å°ä¸º<font color="red">2.95 GBï¼ŒContig N50ä¸º<strong>1.002 MB</strong></font>çš„äºŒå€ä½“æ²¹èŒ¶ï¼ˆCONï¼‰çš„åŸºå› ç»„å›¾è°±ï¼Œå¹¶å°†åŸºå› ç»„<font color="red">é”šå®šåˆ°15æ¡æŸ“è‰²ä½“</font>ï¼Œé”šå®šç‡è¾¾åˆ°<font color="red">91.33%</font>ï¼Œè·å¾—äº†è´¨é‡ä¼˜è‰¯çš„æ²¹èŒ¶åŸºå› ç»„å›¾è°±ã€‚</p>
<p><img src="/blog/2022/03/14/2022-03-14-HiFi-6-HiC-assembly/2.png" alt="æ²¹èŒ¶åŸºå› ç»„å›¾è°±åŠå…¶è¿›åŒ–"></p>
<p>åŒæ—¶ï¼Œåˆ©ç”¨æ§åˆ¶æ‚äº¤çš„ç¾¤ä½“å»ºç«‹<font color="red"><strong>é«˜å¯†åº¦çš„é—ä¼ æ ‡è®°è¿é”å›¾è°±</strong></font>ï¼Œå¯¹<font color="red">åŸºå› ç»„<strong>å›¾è°±ç»„è£…</strong>è¿›è¡Œè¿›ä¸€æ­¥<strong>æ ¡æ­£</strong></font>ï¼Œä¸ºæœªæ¥é‡è¦æ€§çŠ¶çš„åŸºå› å®šä½æä¾›æ”¯æ’‘ã€‚</p>
<p>æ²¹èŒ¶ç‰©ç§èµ·æºä¸åˆ†å¸ƒä¸»è¦é›†ä¸­åœ¨æˆ‘å›½å—æ–¹äºšçƒ­å¸¦åœ°åŒºã€‚é€šè¿‡å¯¹æ²¹èŒ¶åŸºå› ç»„çš„æ¯”è¾ƒåˆ†æï¼Œå‘ç°æ²¹èŒ¶åŸºå› ç»„å…±å‘ç”Ÿ<font color="red"><strong>ä¸¤æ¬¡å…¨åŸºå› ç»„å¤åˆ¶</strong>äº‹ä»¶</font>ï¼šç¬¬ä¸€æ¬¡ä¸ºå¤è€çš„<font color="red">Î³å¤åˆ¶</font>ï¼Œç¬¬äºŒæ¬¡å‘ç”Ÿåœ¨<font color="red"><strong>å±±èŒ¶å±ä¸çŒ•çŒ´æ¡ƒå±</strong>åˆ†åŒ–å‰ä¸ä¹…</font>ã€‚</p>
<p>è€Œå±±èŒ¶å±å½¢æˆä¹‹åå¹¶æ²¡æœ‰å‘ç”Ÿç‰¹æ®Šçš„å…¨åŸºå› ç»„å¤åˆ¶äº‹ä»¶ã€‚é€šè¿‡å±±èŒ¶â€”çŒ•çŒ´æ¡ƒã€å±±èŒ¶â€”æŸ¿ã€æŸ¿â€”çŒ•çŒ´æ¡ƒ<font color="red"><strong>å±é—´åŒæºçªå˜ç‡çš„æ¯”è¾ƒ</strong></font>å°è¯è¿™ä¸€è§‚ç‚¹ã€‚</p>
<p><img src="/blog/2022/03/14/2022-03-14-HiFi-6-HiC-assembly/3.png" alt="æ²¹èŒ¶åŸºå› ç»„é‡å¤åºåˆ—åˆ†æ"></p>
<p>å±±èŒ¶å±åŒ…æ‹¬200å¤šä¸ªç‰©ç§ï¼ŒåŒ…æ‹¬ä»¥<strong>æ™®é€šæ²¹èŒ¶</strong>ä¸ºä»£è¡¨çš„æ²¹èŒ¶ç»„ã€ä»¥<strong>èŒ¶</strong>ä¸ºä»£è¡¨çš„èŒ¶ç»„å’Œä»¥<strong>å±±èŒ¶èŠ±</strong>ä¸ºä»£è¡¨çš„å±±èŒ¶ç»„ç­‰ã€‚ç³»ç»Ÿå‘è‚²å’Œåˆ†å­é’Ÿæ ‡å®šåˆ†æå‘ç°ï¼Œ<strong>æ²¹èŒ¶å’ŒèŒ¶çš„åˆ†åŒ–</strong>å‘ç”Ÿåœ¨å¤§çº¦17.3ç™¾ä¸‡å¹´å‰ã€‚</p>
<p>åœ¨é•¿æœŸæ ½åŸ¹é©¯åŒ–ä¸­ï¼ŒåŒºåˆ«äºä»¥å¶ç”¨ä¸ºç›®çš„çš„èŒ¶å’Œä»¥è§‚èµä¸ºç›®çš„çš„å±±èŒ¶ï¼Œæ²¹èŒ¶åœ¨äººå·¥é€‰æ‹©ä½œç”¨ä¸‹è¿›åŒ–æˆä»¥ç§å­æ²¹è„‚ä¸ºä¸»è¦æ ½åŸ¹ç›®çš„çš„æœ¨æœ¬æ²¹æ–™æ ‘ç§ï¼Œä½¿å¾—æ²¹èŒ¶æˆä¸ºç ”ç©¶æ¤ç‰©ç§å­<font color="red"><strong>æ²¹è„‚æ€§çŠ¶é©¯åŒ–</strong></font>çš„ç»ä½³ææ–™ã€‚</p>
<p>ä¸ºè¿›ä¸€æ­¥æ­ç¤ºæ²¹èŒ¶æ²¹è„‚æ€§çŠ¶é©¯åŒ–çš„åˆ†å­æœºåˆ¶ï¼Œè¯¥å›¢é˜Ÿåˆ©ç”¨å…¨å›½åˆ†å¸ƒåŒºå†…çš„<font color="red">221ä¸ªä»£è¡¨å“ç§</font>å¼€å±•<font color="red"><strong>ç¾¤ä½“è½¬å½•ç»„æµ‹åºå’Œå…³è”åˆ†æ</strong></font>ã€‚ç»“æœå‘ç°åœ¨é•¿æœŸ<strong>é©¯åŒ–</strong>è¿‡ç¨‹ä¸­ï¼Œæ²¹èŒ¶æœå®é€æ¸<strong>å¢å¤§</strong>ï¼Œç§ä»æ²¹è„‚å«é‡æ˜¾è‘—<strong>æé«˜</strong>ï¼Œæ²¹è„‚å“è´¨ä¸æ–­<strong>ä¼˜åŒ–</strong>ï¼Œå½¢æˆäº†ç›®å‰ä¸»æ ½çš„æœå¤§ã€å«æ²¹ç‡é«˜ã€æ²¹è„‚å“è´¨ä¼˜è‰¯çš„æ ½åŸ¹ç±»å‹ã€‚</p>
<p>è¿›ä¸€æ­¥åˆ†æå‘ç°é©¯åŒ–è¿‡ç¨‹ä¸­ï¼Œæ²¹èŒ¶åŸºå› ç»„ä¸­<font color="red"><strong>æ²¹è„‚ä»£è°¢é€šè·¯</strong>ä¸­çš„å¤šä¸ª<strong>åŸºå› </strong>å—åˆ°äº†<strong>æ˜¾è‘—çš„äººå·¥é€‰æ‹©</strong></font>ã€‚æ­¤å¤–ï¼Œ<font color="red">èƒè¿«åº”ç­”ã€æ¿€ç´ ç”Ÿç‰©åˆæˆ</font>ç­‰é€šè·¯ä¹Ÿå—åˆ°æ˜¾è‘—é€‰æ‹©ã€‚</p>
<p>ä¸ºäº†æŒ–æ˜é«˜åº¦å¯ä¿¡çš„æ²¹è„‚ä»£è°¢å…³é”®åŸºå› ï¼Œè”åˆé‡‡ç”¨<font color="red"><strong>åŸºå› ç»„é—ä¼ å˜å¼‚ã€åŸºå› è¡¨è¾¾æ°´å¹³å’Œè¡¨å‹æ€§çŠ¶å˜å¼‚å…³è”åˆ†æ</strong></font>ã€‚ä»¤äººæƒŠè®¶çš„æ˜¯ï¼Œåœ¨<font color="red">ç­›é€‰å‡ºçš„<strong>21ä¸ªå€™é€‰</strong>åŸºå› </font>ä¸­ï¼Œæœ‰<font color="red"><strong>14ä¸ª</strong></font>å±äºæ²¹è„‚åˆæˆå’Œåˆ†è§£ä»£è°¢é€šè·¯åŸºå› ã€‚</p>
<p><img src="/blog/2022/03/14/2022-03-14-HiFi-6-HiC-assembly/4.png" alt="æ²¹èŒ¶å“ç§ç¾¤ä½“å…³è”åˆ†æå‘æ˜æ²¹è„‚äº§é‡ä¸å“è´¨çš„è°ƒæ§åŸºå› "></p>
<p>å…¶ä¸­ï¼Œæ²¹è„‚ä»£è°¢é€šè·¯ä¸­çš„8ä¸ªåŸºå› å’Œæ¤ç‰©æ¿€ç´ ç›¸å…³è½¬å½•å› å­IAA26åœ¨é•¿æœŸ<strong>æ ½åŸ¹é©¯åŒ–</strong>è¿‡ç¨‹ä¸­<strong>å½¢æˆ</strong><font color="red"><strong>æ˜¾è‘—</strong>çš„<strong>å•æ ¸è‹·é…¸å¤šæ€æ€§</strong>å’Œ<strong>è¡¨è¾¾æ°´å¹³å·®å¼‚</strong></font>ã€‚â€œè¿™8ä¸ªåŸºå› æ—¢åŒ…æ‹¬<font color="red">æ²¹è„‚åˆæˆ</font>åŸºå› ï¼Œä¹Ÿæœ‰<font color="red">æ²¹è„‚åˆ†è§£ä»£è°¢</font>åŸºå› ï¼Œè¿™è¡¨æ˜æ²¹èŒ¶<font color="red"><strong>æ²¹è„‚æ€§çŠ¶</strong></font>æ˜¯ç”±å‚ä¸åˆæˆå’Œåˆ†è§£çš„<font color="red"><strong>ä¸¤ç±»åŸºå› ååŒè°ƒæ§</strong></font>çš„ç»“æœã€‚</p>
<h2 id="Genome-sequencing-de-novo-assembly-and-annotation"><a href="#Genome-sequencing-de-novo-assembly-and-annotation" class="headerlink" title="Genome sequencing, de novo assembly, and annotation"></a>Genome sequencing, de novo assembly, and annotation</h2><p>The genomic DNA of CON was prepared by using the young <strong>leaves</strong>. Ten 20 kb de novo SMRTbell libraries were constructed according to the standard manufacturerâ€™s protocol and used for <font color="red">SMRT PacBio</font> genome sequencing. A total of 27,876,348 reads (total size of <strong>320 G</strong>) were generated and used for initial assembly by the <font color="red"><strong>Falcon</strong></font> (v0.3.0) pipeline. The <font color="red"><strong>HaploMerger2</strong></font> (v20180603) program (default parameters) was used to <strong>reduce the redundancy</strong>, and <font color="red"><strong>Arrow program</strong></font> with default parameters was used to <strong>correct</strong> the sequencing <strong>errors</strong>. Further, a total of <font color="red">210 G</font> clean data generated by the <strong>Illumina</strong> Nova- Seq6000 platform was used to <font color="red">correct</font> the PacBio reads.</p>
<p>To circumvent the high heterozygosity of the CON genome, a <font color="red"><strong>hybrid assembly of strategy</strong></font> was used to construct the high-quality reference genome. The details of the <font color="red"><strong>BioNano, 10X Genomics, and Hi-C</strong></font> sequencing procedures were described in the Additional File 3: Method S1; and the hybrid assembly approach was described in <strong>Additional File 2</strong>: Fig. S3. Finally, to construct a chromosome-scale reference genome, the Hi-C chromosomal interaction was created using HiC-pro [38] software (v2.5.0) (<strong>Additional File 2</strong>: Fig. S4). </p>
<p>The <font color="red">repetitive elements</font> in the CON genome, including tandem repeats and interspersed repeats, were identified. Tandem repeats were discovered by <font color="red"><strong>Tandem Repeats Finder</strong></font> v4.07b. Interspersed repeats in the genome were identified using an integration of <strong>independent homology searching and de novo predictions</strong> (See details in Additional File 3: Method S2). Non-coding RNA genes (ncRNA) were annotated in this study (see details in Additional File 3: Method S2) and the results were shown in Additional File 1: Table S4.</p>
<p>We <font color="red">annotated</font> the assembled genome through combining three different approaches: <font color="red"><strong>ab initio prediction, homology-based prediction,</strong> and <strong>transcriptome alignment</strong></font>. To ob- tain the transcriptome data, the total RNA of seven different tissues from CON was se- quenced using the Illumina NovaSeq platform. The transcripts homolog prediction was performed initially by MAKER (v2.31.10); Augustus (v3.3.1) and SNAP (v2006-07-28) were used for de novo prediction. Finally, both homolog and de novo prediction results were integrated using MAKER and resulted in the final gene models (Additional File 1: Table S5; see details in Additional File 3: Method S2)</p>
<p><font color="red"><strong>Functional annotation</strong></font> was achieved by comparing predicted proteins against public databases, including NCBI non-redundant protein sequences database (Nr), SwissProt (201709) [39], eggNOG [39], KEGG (v84) [40], Interpro (v5.16-55.0) [41], and GO [42] using Blast (v2.2.3) [43].</p>
<p><img src="/blog/2022/03/14/2022-03-14-HiFi-6-HiC-assembly/5.png" alt="5"></p>
<p>PacBio+illumina : Falcon+Arrow+BWA+Pilon+HaploMerger-&gt;contigs</p>
<p>scaffold+HiC : <font color="red"><strong>JUICER+3D-DNA</strong></font></p>
<h1 id="10-2022-01-HorticRes-ç”˜èŠ"><a href="#10-2022-01-HorticRes-ç”˜èŠ" class="headerlink" title="10. 2022.01-HorticRes ç”˜èŠ"></a>10. 2022.01-HorticRes ç”˜èŠ</h1><p><strong>The</strong> <em><strong>chrysanthemum lavandulifolium</strong></em> <strong>genome and the molecular mechanism underlying diverse capitulum types</strong></p>
<p><strong>èŠèŠ±</strong>(<em>Chrysanthemum Ã—morifolium</em> Ramat.)æ˜¯ä¸–ç•Œè‘—åçš„è§‚èµæ¤ç‰©ï¼Œå…·æœ‰åƒå§¿ç™¾æ€çš„èŠ±å‹ã€‚å®é™…èŠèŠ±çš„èŠ±æ˜¯æŒ‡ç”±<font color="red"><strong>å¤–å›´çš„èˆŒçŠ¶èŠ±</strong></font>å’Œ<font color="red"><strong>ç›˜å¿ƒçš„ç®¡çŠ¶èŠ±</strong></font>å…±åŒæ„æˆçš„<font color="red"><strong>å¤´çŠ¶èŠ±åº</strong></font>ã€‚å…¶èŠ±å‹ç”±å¤´çŠ¶èŠ±åºä¸ŠèˆŒçŠ¶èŠ±å’Œç®¡çŠ¶èŠ±çš„å½¢æ€å’Œç›¸å¯¹æ•°é‡å†³å®šã€‚è§£æåŒä¸€å¤´çŠ¶èŠ±åºä¸ŠèˆŒçŠ¶èŠ±å’Œç®¡çŠ¶èŠ±åˆ†åŒ–çš„åˆ†å­è°ƒæ§æœºåˆ¶ä¸ä»…å¯ä»¥ä¸ºé˜æ˜èŠèŠ±å¤æ‚çš„å¤´çŠ¶èŠ±åºå½¢æ€å¥ å®šåŸºç¡€ï¼Œä¹Ÿå°†ä¸ºèŠç§‘æ¤ç‰©å¤´çŠ¶èŠ±åºå‘è‚²æä¾›æ–°è§è§£ã€‚ä½†ç›®å‰å…³äºèŠèŠ±èŠ±å‹çš„ç ”ç©¶å—åˆ°å…¶å¤æ‚é—ä¼ èƒŒæ™¯çš„é™åˆ¶ï¼Œå¯¼è‡´æ— æ³•å……åˆ†åˆ©ç”¨èŠèŠ±ä¸°å¯Œçš„åŸºå› èµ„æºè¿›è¡ŒèŠ±å‹å®šå‘è‚²ç§ã€‚å› æ­¤ï¼Œè·å¾—é«˜è´¨é‡çš„èŠèŠ±åŠå…¶è¿‘ç¼˜ç§çš„åŸºå› ç»„ä¿¡æ¯å¹¶åœ¨æ­¤åŸºç¡€ä¸Š<font color="red">ç ”ç©¶<strong>å¤´çŠ¶èŠ±åºå‘è‚²</strong>çš„<strong>åˆ†å­è°ƒæ§æœºåˆ¶</strong></font>æ˜¾å¾—å°¤ä¸ºå¿…è¦ã€‚</p>
<p>è¿‘æ—¥ï¼Œ<em><strong>Horticulture Research</strong></em>åœ¨çº¿å‘è¡¨äº†<strong>åŒ—äº¬æ—ä¸šå¤§å­¦æˆ´æ€å…°å›¢é˜Ÿ</strong>é¢˜ä¸º<em><strong>The Chrysanthemum lavandulifolium genome and the molecular mechanism underlying diverse capitulum types</strong></em> çš„ç ”ç©¶è®ºæ–‡ã€‚è¯¥è®ºæ–‡å®Œæˆ<font color="red"><strong>èŠèŠ±è¿‘ç¼˜é‡ç”Ÿç§</strong>ä¹‹ä¸€<strong>ç”˜èŠ</strong></font>(C. lavandulifolium)çš„å…¨åŸºå› æµ‹åºå·¥ä½œï¼Œè·å¾—äº†æŸ“è‰²ä½“æ°´å¹³ä¸Šçš„é«˜è´¨é‡ç”˜èŠå‚è€ƒåŸºå› ç»„ï¼Œç»“åˆ<font color="red"><strong>3ç§ä¸åŒç±»å‹èŠç§‘æ¤ç‰©å¤´çŠ¶èŠ±åº</strong>çš„<strong>è½¬å½•ç»„æ•°æ®</strong></font>åˆæ­¥è§£æäº†å¤´çŠ¶èŠ±åºå‘è‚²çš„<strong>åˆ†å­è°ƒæ§æœºåˆ¶</strong>ï¼Œä¸ºå®ç°äººå·¥è°ƒæ§èŠèŠ±èŠ±å‹å¥ å®šäº†åšå®çš„åˆ†å­ç†è®ºåŸºç¡€ã€‚</p>
<p>ç”˜èŠä¸ºèŠå±æ¤ç‰©ä¸­çš„<strong>äºŒå€ä½“ç‰©ç§</strong>ï¼Œé‡‡ç”¨<strong>Illumina + Pacbio + Hi-C</strong>æµ‹åºæŠ€æœ¯å¯¹å…¶è¿›è¡Œå…¨åŸºå› æµ‹åºï¼Œè·å¾—äº†2.60 Gbçš„æŸ“è‰²ä½“æ°´å¹³çš„å‚è€ƒåŸºå› ç»„ã€‚å…¶ä¸­<font color="red">94.46%çš„åºåˆ—é”šå®šåˆ°<strong>9æ¡æŸ“è‰²ä½“</strong>ä¸Š</font>ã€‚ç”˜èŠåŸºå› ç»„ä¸­é‡å¤åºåˆ—å åŸºå› ç»„çš„67.69%ï¼Œ<em>Cypsy</em>å’Œ<em>Copia</em>çš„å æ¯”åˆ†åˆ«ä¸º37.92%å’Œ29.06%ï¼Œæ’å…¥æ—¶é—´åœ¨1.25ç™¾ä¸‡å¹´å‰ã€‚<font color="red"><strong>æ¯”è¾ƒåŸºå› ç»„åˆ†æ</strong></font>ç»“æœè¡¨æ˜ï¼Œç”˜èŠç»å†äº†<font color="red"><strong>ä¸¤æ¬¡å…¨åŸºå› ç»„å¤åˆ¶(WGD)äº‹ä»¶</strong></font>ï¼Œå…¶ä¸­<font color="red"><strong>æœ€è¿‘ä¸€æ¬¡</strong>æ˜¯æ‰€æœ‰<strong>èŠç§‘æ¤ç‰©å…±æœ‰</strong></font>ï¼Œè€Œ<font color="red"><strong>è¾ƒæ—©ä¸€æ¬¡</strong>æ˜¯æ ¸å¿ƒåŒå­å¶æ¤ç‰©å…±æœ‰çš„<strong>Î³äº‹ä»¶</strong></font>ï¼Œ**ç”˜èŠè‡ªèº«æ²¡æœ‰å‘ç”ŸWGDï¼Œå…¶åŸºå› ç»„æ¼”åŒ–çš„åŠ¨åŠ›æ¥æºä¸»è¦æ˜¯<font color="red">ä¸²è”é‡å¤äº‹ä»¶</font>**ã€‚</p>
<p><img src="/blog/2022/03/14/2022-03-14-HiFi-6-HiC-assembly/6.png" alt="overview and genome evolution"></p>
<p>åŸºäº<font color="blue">ç”˜èŠå¤´çŠ¶èŠ±åºå‘è‚²çš„6ä¸ªé‡è¦æ—¶æœŸ</font>å’Œ<font color="blue">å…¶ä»–èŠç§‘æ¤ç‰©ä¸åŒç±»å‹å¤´çŠ¶èŠ±åºå‘è‚²å…³é”®æ—¶æœŸ</font>çš„<font color="red">è½¬å½•ç»„</font>æœ‰å‚åˆ†æå‘ç°ï¼ŒMADS-boxã€TCPã€NACå’ŒLOBåŸºå› å®¶æ—<font color="red">å¯èƒ½å‚ä¸<strong>ç®¡çŠ¶èŠ±å’ŒèˆŒçŠ¶èŠ±çš„åˆ†åŒ–</strong></font>ã€‚å€¼å¾—æ³¨æ„çš„æ˜¯ï¼Œ<em>NAM</em>å’Œ<em>LOB30</em>é«˜è¡¨è¾¾äºèˆŒç®¡å…¼å¤‡å‹çš„å¤´çŠ¶èŠ±åºä¸­ï¼Œè€Œåœ¨å…¨èˆŒå‹å’Œå…¨ç®¡å‹çš„å¤´çŠ¶èŠ±åºä¸­è¡¨è¾¾é‡ç›¸å¯¹è¾ƒä½ï¼Œè¿™è¡¨æ˜å…¶å¯èƒ½æ˜¯å‚ä¸ä¸¤ç±»å°èŠ±åˆ†åŒ–çš„å…³é”®åŸºå› ã€‚ç»“åˆå…³é”®åŸºå› åœ¨å…¨èˆŒå‹ã€å…¨ç®¡å‹ä»¥åŠèˆŒç®¡å…¼å¤‡å‹çš„å¤´çŠ¶èŠ±åºä¸­çš„è¡¨è¾¾æ¨¡å¼å’Œè›‹ç™½äº’ä½œæ¨¡å¼ï¼Œåˆæ­¥æ¨æµ‹å¹¶æ„å»ºäº†ä¸åŒç±»å‹å¤´çŠ¶èŠ±åºå‘è‚²å¯èƒ½çš„è°ƒæ§æœºåˆ¶ã€‚åœ¨å…¨ç®¡å‹å¤´çŠ¶èŠ±åºä¸­ï¼Œ<em>CUC2</em>ä¸<em>LFY</em>å’Œ<em>AG</em><font color="red"><strong>å…±åŒè°ƒæ§</strong>ç®¡çŠ¶èŠ±åŸåŸºçš„èµ·å§‹</font>ã€‚è€ŒCUC2å’ŒCUC3çš„<font color="red"><strong>äº’ä½œåŠååŒè¡¨è¾¾</strong>åˆ™ä¼š<strong>ä¿ƒè¿›</strong>å…¨èˆŒå‹å¤´çŠ¶èŠ±åºä¸­èˆŒçŠ¶èŠ±åŸåŸºçš„èµ·å§‹</font>ã€‚åœ¨èˆŒç®¡å…¼å¤‡å‹çš„å¤´çŠ¶èŠ±åºä¸­ï¼Œ<em>NAM</em>å’Œ<em>LOB30</em>çš„è¡¨è¾¾åˆ™å‚ä¸<font color="red"><strong>è°ƒæ§</strong>èˆŒçŠ¶èŠ±å’Œç®¡çŠ¶èŠ±åŸåŸºçš„åˆ†åŒ–</font>ã€‚LOB30å¯ä»¥ä¸LFY, TFL1, CUC2, CUC3å’ŒNAMäº’ä½œï¼Œè¡¨æ˜å…¶åœ¨åŸºå› è°ƒæ§ç½‘ç»œä¸­çš„æ ¸å¿ƒä½ç½®ã€‚æ€»ä¹‹ï¼Œ<em>NAM</em>å’Œ<em>LOB</em>ä¸ä»…å¯ä»¥ä¸èŠ±åºåˆ†ç”Ÿç»„ç»‡ç›¸å…³åŸºå› å¦‚<em>LFY</em> ç­‰äº’ä½œï¼Œä¹Ÿå¯ä»¥ä¸ä¸¤ç±»å°èŠ±èº«ä»½å†³å®šåŸºå› å¦‚<em>CYC2</em>-<em>LIKE</em>ç­‰åŸºå› äº’ä½œï¼Œè¿™è¡¨æ˜<em><strong>NAM</strong></em><strong>å’ŒLOB30åœ¨å¤´çŠ¶èŠ±åºä¸ŠèˆŒçŠ¶èŠ±å’Œç®¡çŠ¶èŠ±åŸåŸºåˆ†åŒ–è°ƒæ§ä¸­çš„å…³é”®è§’è‰²</strong>ã€‚</p>
<p><img src="/blog/2022/03/14/2022-03-14-HiFi-6-HiC-assembly/7.png" alt="The expression patterns of NAM/CUC and LOB30 homologous genes in ten asteraceae species"></p>
<p><img src="/blog/2022/03/14/2022-03-14-HiFi-6-HiC-assembly/8.png" alt="the probable gene regulation mechanism in the development of different capitula types"></p>
<p><strong>é«˜è´¨é‡ç”˜èŠå‚è€ƒåŸºå› ç»„çš„è·å¾—ä¸ä»…å¯ä»¥ä¸ºæ ½åŸ¹èŠèŠ±åŸºå› ç»„çš„ç ´è¯‘æä¾›æœ‰æ•ˆçš„å‚è€ƒï¼Œæ›´ä¸ºè§£æèŠèŠ±ä¹ƒè‡³èŠç§‘æ¤ç‰©å¤šæ ·çš„ç”Ÿç‰©å­¦æ€§çŠ¶æä¾›ä¸°å¯Œçš„åŸºå› èµ„æºã€‚</strong></p>
<p>åŒ—äº¬æ—ä¸šå¤§å­¦æˆ´æ€å…°è¯¾é¢˜ç»„ä¸€ç›´è‡´åŠ›äºèŠèŠ±ç ”ç©¶ï¼ŒåŸºäº<font color="blue">æ¤ç‰©ç³»ç»Ÿå­¦ç ”ç©¶æ–¹æ³•</font>å¯¹<font color="blue">èŠå±æ¤ç‰©ç§é—´<strong>äº²ç¼˜å…³ç³»</strong>å’ŒèŠèŠ±<strong>å“ç§èµ·æº</strong></font>è¿›è¡Œæ¢ç´¢ï¼Œåœ¨<font color="red"><strong>ç§è´¨èµ„æºè¯„ä»·ï¼ŒèŠ±è‰²ã€èŠ±å‹ã€å¼€èŠ±æœŸ</strong>å’Œ<strong>æŠ—é€†æ€§</strong>ç­‰è§‚èµæ€§çŠ¶å½¢æˆçš„<strong>åˆ†å­æœºç†</strong></font>ï¼ŒèŠèŠ±<font color="blue">ä¼˜å¼‚ç§è´¨åˆ›åˆ¶</font>ä»¥åŠ<font color="blue">äº§ä¸šåŒ–æ ½åŸ¹æŠ€æœ¯</font>ç­‰å¼€å±•å…¨é¢ç ”ç©¶ï¼Œå–å¾—äº†ä¸€ç³»åˆ—é‡è¦çªç ´æ€§è¿›å±•å’Œç ”ç©¶æˆæœã€‚</p>
<p>2.6Gï¼šscaffold Num 178ï¼›scaN50 300Mï¼›N90 212Mï¼›Longest 322M</p>
<h2 id="Genome-assembly-and-quality-assessment"><a href="#Genome-assembly-and-quality-assessment" class="headerlink" title="Genome assembly and quality assessment"></a><strong>Genome assembly and quality assessment</strong></h2><p>For the <font color="blue">PacBio RSII</font> platform data, <font color="blue">longer subreads were <strong>selected</strong></font> by the <font color="blue">error correction module</font> of <font color="red"><strong>canu v1.5</strong></font> [35]. <font color="blue">Raw overlapping subreads were <strong>detected</strong></font> through the highly sensitive overlap detection program <font color="red"><strong>MHAP v2.1</strong></font> [36], and the <font color="blue">error correction of these data was <strong>carried</strong></font> out by the <font color="red"><strong>Falcon</strong></font> sense method v0.40 (â€œcorrect- edErrorRate = 0.025â€) [37]. The error-corrected subreads were used to <font color="blue">generate a <strong>draft assembly</strong></font> in <font color="red"><strong>WTDBG v2.5</strong></font> (<a target="_blank" rel="noopener" href="https://github.com/ruanjue/wtdbg">https://github.com/ruanjue/wtdbg</a>). <font color="blue"><strong>Iterative polishing</strong></font> by <font color="red"><strong>Pilon v1.22</strong></font> [38] was achieved by <font color="blue"><strong>aligning</strong></font> adapter-trimmed and paired-end <font color="blue">Illumina <strong>reads</strong> to the <strong>PacBio draft genome</strong></font>. <font color="red"><strong>Clean nanopore data</strong></font> were acquired via sequencing on the PromethION platform, and these data were <font color="blue">corrected with the same method</font> described above. The draft genome assembled by <font color="red"><strong>WTDBG v2.5</strong></font> (https:// github.com/ruanjue/wtdbg) was <font color="blue"><strong>corrected</strong> three times</font> by <font color="red"><strong>Racon v1.3.3</strong></font> [39] by aligning adapter-trimmed and paired-end Illumina reads to the <font color="blue"><strong>Nanopore draft genome</strong></font>. Then, the <font color="blue">PacBio draft genome as a <strong>query input</strong></font> was <font color="blue"><strong>aligned against</strong> the <strong>Nanopore draft genome</strong></font> using <font color="red"><strong>MUMmer v4.0.0</strong></font> [40]. The PacBio draft genome and Nanopore <font color="blue"><strong>draft genome</strong></font> were then <font color="blue"><strong>merged</strong></font> using <font color="red"><strong>quickmerge v0.3.0</strong></font> [21]. This <font color="blue"><strong>merged draft genome</strong></font> was <font color="blue"><strong>polished</strong></font> by <font color="red"><strong>Racon v1.3.3</strong></font> [39] and <font color="red"><strong>Pilon v1.22</strong></font> [38]. The <font color="blue">mapping depth</font> was obtained by <font color="blue"><strong>aligning</strong> corrected Nanopore sequencing data</font> to the merged assembly by <font color="red"><strong>minimap2 v2.17</strong></font> [41] with default parameters. Then, <font color="red"><strong>purge haplotigs v1.0.4</strong></font> [22] was used to <font color="blue"><strong>eliminate</strong> redundancy according to the <strong>coverage depth</strong></font> and obtain the <font color="blue"><strong>purged haplotig genome</strong></font>. Ultimately, a <em>C. lavandulifolium</em> genome with a total length of <strong>3.10 G</strong>b was obtained. The second-generation sequencing data, core gene completeness and BUSCOs were evaluated to verify the accuracy of the genome assembly.</p>
<h2 id="Hi-C-sequencing-and-assistant-assembly"><a href="#Hi-C-sequencing-and-assistant-assembly" class="headerlink" title="Hi-C sequencing and assistant assembly"></a><strong>Hi-C sequencing and assistant assembly</strong></h2><p>Hi-C is a technology derived from chromosome conformation capture technology that utilizes high- throughput sequencing data and is mainly used to assist in genome assembly. We constructed Hi-C fragment libraries with insert sizes of 300â€“700 bp, as illustrated in Rao et al. [42], and sequenced them using the Illumina platform [42]. <font color="blue"><strong>Before</strong> <strong>chromosome assembly</strong>, we first performed a <strong>preassembly</strong> for the <strong>error correction of scaffolds</strong>, which required <strong>splitting the scaffolds into segments of 50 kb</strong>, on average</font>. Then, the Hi-C data were <font color="blue">mapped</font> to these segments using <font color="red"><strong>BWA</strong></font> aligner v0.7.10-r789 [43]. We <font color="red"><strong>retained</strong> the <strong>uniquely mapped</strong> data</font> to assemble the genome using <font color="red"><strong>LACHESIS</strong></font> [44] with the following <font color="green"><strong>parameters</strong>: CLUSTER_MIN_RE_SITES = 80; CLUSTER_MAX_LINK_DENSITY = 2; CLUSTER_NONINFORMATIVE_RATIO = 2; ORDER_MIN_N_RES_IN_TRUN = 16; ORDER_MIN_N_RES_IN_SHREDS = 16</font>. To further address the redundant sequences, we manually checked any two segments that showed inconsistent connections with the raw scaffold. The detailed workflow schema for the assembly pipeline of the chromosome-scale <em>C. lavandulifolium</em> genome is shown in Supplementary Figure 24.</p>
<p><img src="/blog/2022/03/14/2022-03-14-HiFi-6-HiC-assembly/9.png" alt="9"></p>
<h1 id="11-2022-01-HorticRes-ç²‰è‘›"><a href="#11-2022-01-HorticRes-ç²‰è‘›" class="headerlink" title="11. 2022.01-HorticRes ç²‰è‘›"></a>11. 2022.01-HorticRes ç²‰è‘›</h1><p><em><strong>Chromosomal-level genome and multi-omics dataset of Pueraria lobata var. thomsonii provide new insights into legume family and the isoflavone and puerarin biosynthesis pathways</strong></em> </p>
<p>ç²‰è‘›åŸºå› ç»„åŠå¤šç»„å­¦åˆ†æä¸º<font color="red">è‘›æ ¹ç´ </font>ç­‰åŠŸæ•ˆæˆåˆ†çš„ç”Ÿç‰©åˆæˆç ”ç©¶æä¾›æ–°è§è§£</p>
<p>å¹¿è¥¿å£®æ—è‡ªæ²»åŒºå†œä¸šç§‘å­¦é™¢ç»æµä½œç‰©ç ”ç©¶æ‰€ä¸¥åå…µå›¢é˜Ÿã€‚è¯¥ç ”ç©¶åˆ©ç”¨<strong>PacBio</strong>ã€<strong>Illumina</strong>æµ‹åºä»¥åŠ<strong>Hi-C</strong>æŠ€æœ¯æµ‹åºç»„è£…ï¼Œ<strong>è·å¾—äº†é¦–ä¸ª<font color="red">è±†ç§‘è‘›å±è¯é£ŸåŒæºæ¤ç‰©ç²‰è‘›</font>æŸ“è‰²ä½“çº§åˆ«çš„åŸºå› ç»„</strong>ï¼Œæ ‡å¿—ç€ç²‰è‘›ç ”ç©¶è¿ˆå…¥åŸºå› ç»„æ—¶ä»£ï¼Œå¼€å¯äº†<font color="blue">è‘›å±<strong>è¿›åŒ–</strong>ã€é‡è‘›<strong>é©¯åŒ–</strong>ã€ç²‰è‘›<strong>å“ç§æ”¹è‰¯</strong></font>ä¹‹æ—…ï¼ŒåŠ©åŠ›å¹¿è¥¿ç‰¹è‰²ä¼˜åŠ¿ç²‰è‘›äº§ä¸šé«˜è´¨é‡å‘å±•ã€‚</p>
<p>é‰´äºç²‰è‘›<font color="blue">æ‚åˆåº¦è¾ƒé«˜</font>ï¼Œç ”ç©¶è€…é€‰ç”¨äº†PacBioå’ŒHi-Cæµ‹åºï¼Œæ„å»ºçš„ç²‰è‘›åŸºå› ç»„å¤§å°ä¸º<strong>1.38Gb</strong>ï¼Œ<font color="blue">Contig N50=598kb</font>ï¼Œå¹¶å°†<font color="blue">99.3%çš„åºåˆ—<strong>é”šå®š</strong>åˆ°11æ¡æŸ“è‰²ä½“ä¸Š</font>ï¼ŒBUSCOè¯„ä¼°<font color="blue">92.9%</font>ã€‚é€šè¿‡æ³¨é‡Šï¼Œå…±è·å¾—45270ä¸ªè›‹ç™½ç¼–ç åŸºå› ï¼Œå…¶ä¸­<font color="blue"><strong>94.4%çš„åŸºå› </strong>å¯ä»¥å¾—åˆ°<strong>åŠŸèƒ½æ³¨é‡Š</strong></font>ï¼ŒåŸºå› ç»„ä¸­<font color="blue">é‡å¤åºåˆ—</font>å æ¯”ä¸º62.7%ã€‚å°†ç²‰è‘›<font color="red">ä¸<strong>16ä¸ªè¿‘ç¼˜ç‰©ç§</strong>(åŒ…å«5ä¸ªè±†ç§‘æ¤ç‰©)è¿›è¡Œ<strong>æ¯”è¾ƒåŸºå› ç»„</strong>åˆ†æ</font>ï¼Œç»“æœè¡¨æ˜6ä¸ªè±†ç§‘æ¤ç‰©å…±æœ‰åŸºå› å®¶æ—ä¸º11204ä¸ªï¼Œç²‰è‘›åŸºå› ç»„ä¸­<font color="blue"><strong>æ˜¾è‘—æ‰©å¼ çš„4743ä¸ªåŸºå› å®¶æ—</strong>ä¸»è¦å¯Œé›†åœ¨ä¸<strong>ç±»é»„é…®ã€ç”Ÿç‰©ç¢±ã€ç”¾é†‡å’Œèœç±»</strong>ç­‰ç”Ÿç‰©åˆæˆç›¸å…³çš„é€šè·¯</font>ï¼›ç²‰è‘›<font color="red"><strong>ç‰¹æœ‰</strong>åŸºå› å®¶æ—</font>ä¸º2373ä¸ªï¼Œä¸»è¦<font color="blue">å¯Œé›†åœ¨ä¸<strong>èœç±»</strong>ç”Ÿç‰©åˆæˆç›¸å…³çš„é€šè·¯</font>ï¼›ç²‰è‘›åŸºå› ç»„ä¸­å—åˆ°<font color="blue"><strong>æ˜¾è‘—æ­£é€‰æ‹©</strong>çš„åŸºå› </font>å…±æœ‰34ä¸ªï¼Œå¯Œé›†åœ¨<font color="blue"><strong>æ˜¼å¤œèŠ‚å¾‹ã€åŒæºé‡ç»„ã€æ·€ç²‰ä»£è°¢å’Œè”—ç³–ä»£è°¢</strong></font>ç­‰é€”å¾„ã€‚</p>
<p><font color="red"><strong>ç³»ç»Ÿè¿›åŒ–åˆ†æ</strong></font>è¡¨æ˜ï¼Œç²‰è‘›<strong>ä¸å¤§è±†äº²ç¼˜å…³ç³»æœ€è¿‘</strong>ï¼Œä¸¤è€…åœ¨20.1ä¸ªç™¾ä¸‡å¹´å‰äº§ç”Ÿåˆ†åŒ–ã€‚é€šè¿‡<font color="red"><strong>Ks</strong>å’Œ<strong>4DTVåˆ†æ</strong></font>ï¼Œ<strong>ç²‰è‘›å…±ç»å†ä¸¤æ¬¡WGDäº‹ä»¶</strong>ï¼Œä¸€æ¬¡WGDäº‹ä»¶æ˜¯<font color="blue">å¤§è±†ä¸ç²‰è‘›<strong>å…±æœ‰</strong></font>ï¼Œå‘ç”Ÿåœ¨44.5ä¸ªç™¾ä¸‡å¹´å‰ï¼›å¦ä¸€æ¬¡WGDäº‹ä»¶æ˜¯åœ¨ç²‰è‘›ä¸å¤§è±†åˆ†åŒ–åï¼Œ<font color="blue">ç²‰è‘›<strong>ç‹¬è‡ª</strong>ç»å†</font>ï¼Œå‘ç”Ÿæ—¶é—´å¤§è‡´åœ¨4.8ä¸ªç™¾ä¸‡å¹´å‰ã€‚</p>
<p>é€šè¿‡å¯¹é«˜è‘›æ ¹ç´ ZG-19å’Œä½è‘›æ ¹ç´ ZG-39è¿›è¡Œ<font color="red"><strong>è½¬å½•ç»„</strong>å’Œ<strong>ä»£è°¢ç»„</strong>åˆ†æ</font>ï¼Œç ”ç©¶è€…æ£€æµ‹åˆ°äº†614ç§225ç§å·®å¼‚ä»£è°¢ç‰©(DMs)ï¼Œ1814ä¸ªå·®å¼‚è¡¨è¾¾åŸºå› (DEG)ï¼ŒDMså’ŒDEGçš„ä¸°å¯ŒåŠŸèƒ½ç±»åˆ«<strong>é‡å </strong>ï¼Œè¿™è¯´æ˜å®ƒä»¬éƒ½æ˜¯ä¸<font color="blue">ç±»é»„é…®ã€å¼‚é»„é…®å’ŒABCè½¬è¿ç›¸å…³çš„åŸºå› æˆ–ä»£è°¢ç‰©</font>ã€‚è¿›ä¸€æ­¥åˆ†æä»£è°¢ç‰©ä¸åŸºå› è¡¨è¾¾çš„ç›¸å…³ç³»æ•°ï¼Œç»“æœè¡¨æ˜ä»£è°¢ç‰©å’ŒåŸºå› å¯¹åœ¨æ ·æœ¬ä¸­é«˜åº¦ç›¸å…³ï¼Œ60%çš„æ˜¾è‘—ç›¸å…³æ€§æ¶‰åŠä¸Šè°ƒçš„ä»£è°¢ç‰©å’Œä¸‹è°ƒæˆ–ä¸å˜çš„åŸºå› ï¼Œåœ¨15%çš„æ˜¾è‘—ç›¸å…³æ€§ä¸­ï¼Œä»£è°¢ç‰©å’ŒåŸºå› è¡¨è¾¾çš„å˜åŒ–æ–¹å‘ç›¸åŒã€‚æ­¤å¤–ï¼Œç ”ç©¶è€…åœ¨å¼‚é»„é…®ç”Ÿç‰©åˆæˆé€”å¾„ä¸­å‘ç°äº†å¤§é‡çš„DMså’ŒDEGã€‚<strong>è¿™å……åˆ†è§£æäº†ç²‰è‘›ä¸­å¼‚é»„é…®çš„ç”Ÿç‰©åˆæˆé€”å¾„ã€‚</strong></p>
<p>é€šè¿‡<font color="red"><strong>åŒæºåŸºå› æœç´¢</strong></font>ï¼Œç ”ç©¶è€…å‘ç°<font color="blue">ç¼–ç <strong>è‘›æ ¹ç´ åˆæˆé€”å¾„</strong>ä¸­å…³é”®é…¶çš„9ä¸ªåŸºå› å®¶æ—</font>åœ¨ç²‰è‘›ä¸­éƒ½æœ‰æ‰€<font color="blue">æ‰©å¼ </font>ï¼›é€šè¿‡åˆ†æç³–åŸºè½¬ç§»é…¶å®¶æ—ä¸­å‚¬åŒ–ç³–åŸºåŒ–ä¿®é¥°çš„åŸºå› ï¼Œå…±é‰´å®šå‡º104ä¸ªGTåŸºå› ï¼Œæœ‰13ä¸ªåŸºå› ä¸8-C-è‘¡è„ç³–åŸºè½¬ç§»é…¶(8-C-GT)åŒæºï¼Œå…¶ä¸­6ä¸ªä¸å…ˆå‰ç ”ç©¶çš„å‚¬åŒ–å¤§è±†è‹·å…ƒC-ç³–åŸºåŒ–ä¸ºè‘›æ ¹ç´ çš„PIUGT43åŸºå› åŒæºï¼›ç¼–ç å¤§è±†å¼‚é»„é…®åˆé…¶(IFS)çš„åŸºå› (CHR11G3854.1)å‚¬åŒ–ç€è‘›æ ¹ç´ åˆæˆçš„ä¸­é—´ä»£è°¢ç‰©å¤§è±†è‹·å…ƒçš„åˆæˆï¼Œè¢«é‰´å®šä¸ºä¸è‘›æ ¹ç´ çš„åˆæˆé€”å¾„é«˜åº¦ç›¸å…³ã€‚æ€»ä¹‹ï¼Œä¸Šè¿°åˆ†æ<strong>åˆæ­¥è§£æäº†ç²‰è‘›ä¸­è‘›æ ¹ç´ çš„ç”Ÿç‰©åˆæˆé€”å¾„ã€‚</strong></p>
<p><strong>ç»¼ä¸Šï¼Œè¯¥ç ”ç©¶é€šè¿‡æ„å»ºé«˜è´¨é‡çš„ç²‰è‘›åŸºå› ç»„è§£æäº†ç²‰è‘›åŸºå› ç»„çš„<font color="red">è¿›åŒ–ç‰¹å¾</font>ï¼›é€šè¿‡å¤šç»„å­¦åˆ†ææ·±å…¥è§£æäº†ç²‰è‘›ä¸­<font color="red">é‡è¦æ¬¡ç”Ÿä»£è°¢ç‰©å¼‚é»„é…®ã€è‘›æ ¹ç´ ç­‰ç”Ÿç‰©åˆæˆé€”å¾„</font>ï¼Œä»è€Œä¸ºç²‰è‘›çš„èµ„æºåˆ©ç”¨ã€é—ä¼ è‚²ç§ç­‰ç ”ç©¶æä¾›äº†æ–°è§è§£ã€‚</strong></p>
<p><strong>ä¸¥åå…µç ”ç©¶å‘˜å›¢é˜Ÿåœ¨è‘›æ ¹ä¸Šçš„ç ”ç©¶è¿›å±•ï¼š</strong></p>
<p><strong>ä¸¥åå…µ</strong>å›¢é˜Ÿè¿‘å¹´<strong>åœ¨å…¨çƒè‘›æ ¹èµ„æºæ”¶é›†ä¸é‰´å®šè¯„ä»·ã€è‘›å±èµ„æºåˆ†ç±»ã€è‘›æ ¹<font color="red">åŸºå› ç»„ä¸åˆ†å­ç”Ÿç‰©å­¦</font>ã€ç²‰è‘›å’Œé‡è‘›å“ç§é€‰è‚²ã€å¥åº·ç§è‹—ç”Ÿäº§ã€é«˜äº§é«˜æ•ˆæ ½åŸ¹ç­‰æ–¹é¢å–å¾—äº†ä¸€ç³»åˆ—çš„æˆæœ</strong>ã€‚å›¢é˜Ÿç›®å‰å¹¿æ³›æ”¶é›†å…¨çƒè‘›å±ç§è´¨èµ„æº419ä»½ï¼ŒåŒ…æ‹¬é‡è‘›ã€ç²‰è‘›ã€è‘›éº»å§†ã€å¤§èŠ±è‘›ã€æ³°è‘›ã€è‹¦è‘›ã€çº¢è‘›ã€é¡»å¼¥è‘›ã€é£Ÿç”¨è‘›ç­‰ï¼›é€šè¿‡å¼€å‘<strong>è‘›SSRåˆ†å­æ ‡è®°</strong>ï¼Œæ„å»ºäº†å¹¿è¥¿è‘›æ ¸å¿ƒç§è´¨åº“ï¼›é€šè¿‡<strong>å¹¿æ³›é¶å‘ä»£è°¢ç»„</strong>è§£æè‘›å±è‘›ç§é‡è‘›ã€ç²‰è‘›å’Œè‘›éº»å§†ç­‰3ä¸ªå˜ç§å—æ ¹ä¸­å½±å“<strong>é£Ÿç”¨å“è´¨å’Œè¯ç”¨å“è´¨</strong>çš„<strong>ä»£è°¢å·®å¼‚</strong>ï¼›ç»“åˆè¡¨å‹é‰´å®šé€šè¿‡å¶ç»¿ä½“åŸºå› ç»„ç ”ç©¶ï¼Œæ­ç¤ºäº†è‘›åŠå…¶<strong>è¿‘ç¼˜ç§ä¹‹é—´çš„ç³»ç»Ÿå‘è‚²å…³ç³»</strong>ï¼›æŒ–æ˜äº†è°ƒæ§<strong>è‘›æ ¹ç´ åˆæˆä»£è°¢ç›¸å…³çš„ç»“æ„åŸºå› å’Œè½¬å½•å› å­</strong>ï¼Œå¹¶æ­£åœ¨å¼€å±•ç›¸å…³åŸºå› åŠŸèƒ½éªŒè¯å·¥ä½œï¼›é€‰è‚²å‡ºé€‚åˆå¼€å‘è‘›èŠ±èŒ¶ã€é«˜è‘›æ ¹ç´ ç²‰è‘›ã€æ— æ¸£ç²‰è‘›ã€è¯ç”¨é‡è‘›ç­‰ç³»åˆ—è‘›æ ¹æ–°å“ç§ï¼Œå¹¶é€æ­¥å»ºç«‹é…å¥—ç§è‹—ç¹è‚²å’Œé«˜æ•ˆæ ½åŸ¹æŠ€æœ¯ã€‚</p>
<p>1.37Gï¼šcontig Num 5145ï¼›conN50 593kbï¼›</p>
<h2 id="Genome-survey"><a href="#Genome-survey" class="headerlink" title="Genome survey"></a>Genome survey</h2><p>Short reads produced by the Illumina 6000 platform were <font color="blue">quality-filtered</font> by <font color="red"><strong>HTQC</strong></font> [23] (version v1.92.310) using the following method. Adaptors were firstly removed from the sequencing reads, and read pairs with any one end having an average quality &lt;20 were discarded. Ends of reads were trimmed if average quality was &lt;20 in the sliding window size of 5 bp, and read pairs with <font color="red">any end <strong>shorter than 75 bp</strong> were removed</font>. The quality-filtered reads were used for estimation of genome size. The <strong>17-mer</strong> occurrence distribution of sequencing reads was generated from short libraries with <font color="red"><strong>Jellyfish</strong></font> [24] (v2).</p>
<h2 id="Genome-assembly"><a href="#Genome-assembly" class="headerlink" title="Genome assembly"></a>Genome assembly</h2><p>Subreads generated with the above long-read third-generation sequencing were used for genome assembly of <em>P. lobata</em>. The draft assembly of the genome was made using <font color="red"><strong>Falcon (v0.3.0)</strong></font> [25]. To <font color="blue"><strong>correct</strong> errors</font> in the primary assembly, <font color="red"><strong>Racon</strong></font> (v1.44) tools were used to <font color="blue"><strong>polish</strong></font> the genome [26]. <font color="blue">Illumina-derived <strong>short reads</strong></font> were used to <font color="blue"><strong>correct</strong></font> any <font color="blue"><strong>remaining errors</strong></font> by <font color="red"><strong>NextPolish (v1.1.0)</strong></font> [27].</p>
<p><font color="blue">To scaffold the contigs</font>, 425 011 698 clean-read pairs were sequenced from the Hi-C library. <font color="red"><strong>Fastp</strong></font> (0.19.5) [28] was used to filter Hi-C data, including <font color="blue"><strong>removing adaptors</strong> and l<strong>ow-quality</strong> reads</font>. <font color="red"><strong>HiC-Pro</strong></font> Proto was used to obtain effective reads [29]. After filtering, <font color="blue">6%ã€?ã€‘ of high-quality and effective Hi-C data was retained</font>. The <font color="blue">resulting <strong>effective reads</strong></font> were <strong>mapped</strong> to the <strong>assembled and polished</strong> <em>P. thomsonii</em> <strong>genome</strong> using <font color="red"><strong>BWA</strong></font> (bwa-0.7.17) [30] with default parameters. <font color="red"><strong>ALL-HiC</strong></font> software was used to <font color="blue"><strong>anchor</strong> scaffolds to chromosomes</font> to obtain the chromosome-level <em>P. thomsonii</em> assembly.</p>
<h1 id="12-2022-03-ng-å››å€ä½“åœŸè±†"><a href="#12-2022-03-ng-å››å€ä½“åœŸè±†" class="headerlink" title="12. 2022.03-ng å››å€ä½“åœŸè±†"></a>12. 2022.03-ng å››å€ä½“åœŸè±†</h1><blockquote>
<p>Chromosome-scale and haplotype-resolved genome assembly of a tetraploid potato cultivar</p>
</blockquote>
<p>##<strong>Genome size estimation.</strong> </p>
<p>After <strong>trimming</strong> off 10Ã— genomics <font color="blue"><strong>barcodes and hexamers</strong></font> from the 370.3-Gb reads combined from the 10Ã— single-cell CNV libraries and single-molecule libraries, <em>k</em>-mer counting (<em>k</em> = 21) was performed with <font color="red"><strong>Jellyfish</strong></font> (v.2.2.10). The <em>k</em>-mer histogram was provided to <font color="red"><strong>findGSE</strong></font> (v.1.0)39 to estimate the haploid/tetraploid genome size of â€˜Otavaâ€™ under the <font color="blue">heterozygous mode</font> (with â€˜exp_hom = 200â€™; Extended Data Fig. 1).</p>
<p>##<strong>Initial tetraploid genome assembly, polishing and purging.</strong> </p>
<p>The initial assembly of the tetraploid genome was performed using <font color="red"><strong>hifiasm (v.0.7)</strong></font> with <strong>default settings</strong> with the 102 Gb raw PacBio HiFi reads of â€˜Otavaâ€™, where the output consisting of <font color="red"><strong>unitigs</strong> (locally haplotype-resolved contigs)</font> was selected for <strong>further processing</strong>. </p>
<p>Then the <font color="blue">short reads</font> from the two 10Ã— single-cell CNV libraries were <font color="blue"><strong>aligned</strong> to the assembly</font> using <font color="red"><strong>bowtie2 (v.2.2.8)</strong></font>. These <font color="blue">alignments were used to <strong>polish</strong> the <strong>assembly</strong></font> with <font color="red"><strong>pilon (v.1.22)</strong></font> with <font color="green"><strong>options</strong> of <strong>â€“fixbases â€“changes â€“diploid â€“mindepth 0.8</strong></font>. </p>
<p>Further, <font color="blue"><strong>short</strong> reads</font> of the two 10Ã— single-cell CNV libraries, additional 10Ã— single-molecule libraries <strong>and</strong> PacBio <font color="blue"><strong>HiFi</strong> reads</font> were <font color="blue"><strong>all aligned</strong> to the <strong>polished assembly</strong></font> (using <font color="red"><strong>bowtie2</strong></font> and <font color="red"><strong>minimap2</strong></font> (v.2.17-r491) respectively). Among <font color="blue">17,153 raw contigs, <strong>9,041 contigs</strong></font> which were <font color="blue">longer than <strong>50 kb</strong></font> and with an <font color="blue">average sequencing depth over <strong>80Ã—</strong> were <strong>kept</strong></font>. </p>
<p><font color="blue">HiFi reads were <strong>re-aligned</strong> to the purged assembly and contigs covered less than 3Ã— were <strong>removed</strong></font>. </p>
<p>The HiFi reads-based <font color="red"><strong>purging process</strong></font> was repeated for <font color="blue"><strong>five rounds</strong></font> to get an initial assembly of <font color="blue">6,366 contigs</font> for subsequent analysis (Supplementary Fig. 1).</p>
<p>##<strong>Haplotype-specific PacBio HiFi read separation and haplotype assembly.</strong></p>
<p>HiFi reads were <font color="blue">classified into 48 groups</font> on the basis of alignments to the 50-kb coverage markers using customized code. Specifically, to assign a read to a marker, at least 500 bp of the read had to be aligned to the marker (reads aligning two neighboring markers were assigned to the marker with a larger overlapping size). Reads overlapping non-haplotig marker were randomly assigned to one of the marker-associated groups.</p>
<p><font color="red"><strong>Each set</strong> of HiFi reads</font> was independently assembled using <font color="red"><strong>hifiasm</strong></font> (v.0.7) with default settings. The resulting contigs were first polished with <font color="red">short reads using <strong>pilon</strong></font> with <font color="green"><strong>â€“fix bases â€“changes â€“diploid â€“mindepth 0.8</strong></font> and then with <font color="red">HiFi reads using <strong>racon</strong></font> (v.1.4.10)46 with<font color="green"> <strong>-u â€“no-trimming</strong></font>.</p>
<p>##<strong>Evaluation of haplotyping accuracy.</strong> </p>
<p>For <font color="blue">each haplotype assembly</font> and the sequencing data of the parental genomes, <em>k</em>-mers (<em>k</em> = 21) were counted using <font color="red"><strong>KMC</strong></font>. Specifically, <em>k</em>-mers found in â€˜Heraâ€™ but not in â€˜Stieglitzâ€™ (with a coverage<br>of 6â€“12), as well as <em>k</em>-mers found in â€˜Stieglitzâ€™ but not in â€˜Heraâ€™ (with a coverage of 5â€“11) were selected using kmc_tools simple. For each haplotype, the sets of assembled <em>k</em>-mers were intersected with the two sets of parental-specific <em>k</em>-mers (using kmc_tools simple with subfunction intersect), which revealed <strong><em>k</em>-mers common with either of the parental genomes</strong>. </p>
<p>As a haplotype can only be inherited from one of the parents, it is expected to find parental-specific *k-*mers only of one parent. The overall haplotyping precision was determined as the total number of correctly phased <em>k</em>-mers divided by the total number of <em>k</em>-mers investigated in the <font color="red"><strong>48 haplotype assemblies</strong></font>. Note, this was done <font color="rblueed">before and after contig <strong>polishing</strong></font>, where we observed the same haplotyping accuracy.</p>
<p>##<strong>Haplotype-specific contig scaffolding using group-specific Hi-C reads.</strong> </p>
<p><font color="blue">Each haplotype-specific contig-level assembly</font> was indexed with <font color="red"><strong>bwa index</strong></font> (with -a bwtsw) (v.0.7.15-r1140) and <font color="red"><strong>samtools faidx</strong></font>. The haplotype-specific Hi-C read pairs were aligned using <font color="red"><strong>bwa aln</strong> and <strong>bwa sampe</strong></font>. Aligned reads (in pairs) were <strong>converted into BAM</strong> files using samtools view with options of <font color="green">-b -F12</font>. The BAM files were filtered with <font color="red"><strong>filterBAM_forHiC.pl</strong></font> (from ALLHiC package, v.0.9.13) to <font color="blue"><strong>remove nonuniquely mapped reads</strong></font>. Then BAM files were converted to bed files using <font color="red"><strong>bamToBed</strong></font> (from bedtools package) and <font color="blue"><strong>sorted</strong> by read name</font>. The <strong>bed files</strong> were provided to <font color="red"><em><strong>SALSA2</strong></em></font> (run with <font color="green">-s 100000000 -m yes -i 10 -e DNASE</font>). Potential chimeric contigs were broken at the chimeric sites given by SALSA2 output file of input_breaks, leading to a <font color="blue"><strong>new set</strong> of contigs for each of the 48 original groups</font>.</p>
<p>For <font color="blue">each new group of contigs</font>, the above process of <font color="red"><strong>contig indexing, Hi-C read alignment</strong> and <strong>BAM filtering</strong></font> was repeated. Then, for each haplotype, <font color="red"><strong>ALLHiC_ partition</strong></font> was run with <font color="green">-e GATC -k 1 -m 25</font>; <font color="red"><strong>allhic extract</strong></font> was run with <font color="green">â€“RE GATC</font>; allhic optimize and <font color="red"><strong>ALLHiC_build</strong></font> were run with <font color="green">default settings</font>; the chromosome contact map was visualized with <font color="red"><strong>ALLHiC_plot</strong></font> at <font color="green">1-Mb</font> resolution, where obvious mis-placement/orientation of large contigs were visually identified and manually corrected (Extended Data Fig. 3).</p>
<h1 id="13-2022-03-ng-å…­å€ä½“å°éº¦"><a href="#13-2022-03-ng-å…­å€ä½“å°éº¦" class="headerlink" title="13. 2022.03-ng å…­å€ä½“å°éº¦"></a>13. 2022.03-ng å…­å€ä½“å°éº¦</h1><blockquote>
<p>Long-read genome sequencing of bread wheat facilitates disease resistance gene cloning</p>
</blockquote>
<table>
<thead>
<tr>
<th>Genomic feature</th>
<th>Value</th>
</tr>
</thead>
<tbody><tr>
<td><em><strong>Length of HiFi assembly</strong></em></td>
<td>14.66 Gb</td>
</tr>
<tr>
<td>Number of contigs</td>
<td>5,055</td>
</tr>
<tr>
<td>Length of contig N50</td>
<td>30.22 Mb</td>
</tr>
<tr>
<td>Length of contig N90</td>
<td>5.5 Mb</td>
</tr>
<tr>
<td><strong><em>Length of hybrid assembly</em>a</strong></td>
<td>14.68 Gb</td>
</tr>
<tr>
<td><em><strong>Length of hybrid scaffolds</strong></em></td>
<td>14.46 Gb</td>
</tr>
<tr>
<td>Number of hybrid scaffolds</td>
<td>324</td>
</tr>
<tr>
<td>Length of hybrid scaffold N50</td>
<td>204.26 Mb</td>
</tr>
<tr>
<td>Gap size</td>
<td>18.8 Mb (0.13%)</td>
</tr>
<tr>
<td><em><strong>Length of pseudomolecule assembly</strong></em></td>
<td>14.68 Gb</td>
</tr>
<tr>
<td><em><strong>Total length of anchored pseudomolecules</strong></em></td>
<td>14.45 Gb</td>
</tr>
<tr>
<td>Number of anchored hybrid scaffolds or contigs</td>
<td>717</td>
</tr>
<tr>
<td>Gap size</td>
<td>11.6 Mb (0.08%)</td>
</tr>
<tr>
<td>Number of high-confidence genes</td>
<td>110,383</td>
</tr>
<tr>
<td><em><strong>Total length of unanchored chromosome</strong></em></td>
<td>224.01 Mb</td>
</tr>
<tr>
<td>Number of unanchored hybrid scaffolds or contigs</td>
<td>3,910</td>
</tr>
<tr>
<td>Gap size</td>
<td>8.1 Mb (3.6%)</td>
</tr>
<tr>
<td>Number of high-confidence genes</td>
<td>6,455</td>
</tr>
<tr>
<td><em><strong>BUSCO</strong></em></td>
<td></td>
</tr>
<tr>
<td>Complete</td>
<td>99.4%</td>
</tr>
<tr>
<td>Duplicated</td>
<td>96.3%</td>
</tr>
<tr>
<td>Fragmented</td>
<td>0.1%</td>
</tr>
<tr>
<td>Missing</td>
<td>0.5%</td>
</tr>
</tbody></table>
<h2 id="Genome-assembly-and-validation"><a href="#Genome-assembly-and-validation" class="headerlink" title="Genome assembly and validation."></a><strong>Genome assembly and validation.</strong></h2><p>The PacBio HiFi reads were assembled using <font color="red"><strong>hifiasm12 (v.0.11)</strong></font> with <font color="green">default parameters</font>. Hybrid scaffolding incorporating the PacBio contigs and the <font color="blue">optical map</font> was performed using the <font color="red"><strong>hybridScaffold pipeline</strong></font> (Bionano Solve 3.6) with default parameters. For the pseudomolecule construction, the Omni-C reads were incorporated using <font color="red"><strong>Juicer tools (v.1.6)</strong></font> and <font color="red"><strong>3D-DNA (v.180114)</strong></font>. </p>
<p>In brief, the preprocessing of the Omni-C reads was performed with juicer.sh (parameter: -s none). The â€˜merged_nodups.txtâ€™ output file corresponding to the Hi-C contacts with duplicates removed was subsequently used with run-asm-pipeline.sh (parameter: -r 0) as input to produce the â€˜.hicâ€™ and â€˜.assemblyâ€™ files. These files were uploaded into Juicebox (v.1.11.08) to visualize the Hi-C map and for manual curation. </p>
<p>As a final step, the script <font color="red"><strong>run-asm-pipeline-post-review.sh</strong></font> (default parameters) was used to save the final Hi-C contact map and to output the final Kariega assembly (21 pseudomolecules and 1 unanchored pseudochromosome). To <font color="blue"><strong>validate</strong> the genome assembly</font>, we remapped the optical map onto the pseudomolecule using the <font color="red"><strong>hybridScaffold pipeline</strong></font> (Bionano Solve 3.6), and the final pseudomolecules were compared with the recent bread wheat assemblies of Chinese Spring (IWGSC RefSeq v.2.1) and the assemblies of the 10+ Wheat Genomes Project using <font color="red"><strong>MashMap</strong></font> (v.2.0; parameter: -s 300000â€“pi 98).</p>
<h1 id="14-2019-10-äºŒå€ä½“å·´ç‰¹åˆ©ç‰¹æ¢¨-ã€å››å€ä½“åœŸè±†å‚è€ƒhicç»„è£…1ã€‘"><a href="#14-2019-10-äºŒå€ä½“å·´ç‰¹åˆ©ç‰¹æ¢¨-ã€å››å€ä½“åœŸè±†å‚è€ƒhicç»„è£…1ã€‘" class="headerlink" title="14.  2019.10 äºŒå€ä½“å·´ç‰¹åˆ©ç‰¹æ¢¨ ã€å››å€ä½“åœŸè±†å‚è€ƒhicç»„è£…1ã€‘"></a>14.  2019.10 äºŒå€ä½“å·´ç‰¹åˆ©ç‰¹æ¢¨ ã€å››å€ä½“åœŸè±†å‚è€ƒhicç»„è£…1ã€‘</h1><blockquote>
<p>Pseudo-chromosomeâ€“length genome assembly of a double haploid â€œBartlettâ€ pear (<em>Pyrus communis</em> L.)</p>
</blockquote>
<p><strong>Genome assembly and scaffolding</strong></p>
<p>The genome assembly workflow began with <em>de novo</em> assembly of contigs from the PacBio long reads using 2 tools, <font color="red"><strong>Canu</strong></font> (version 1.5) and <font color="red"><strong>Falcon</strong></font> (version 0.5). For each assembler the most important assembly parameters were systematically varied (Supplementary Methods), as defined by the tool developers, and by consideration of assembly theory (e.g., overlap length, overlap identity for overlap layout consensus assembly). <font color="blue"><strong>Optimal settings</strong> were <strong>selected</strong></font> by <strong>comparison of assembly statistics</strong> (total size assembled and contig N50) and by <strong>alignment of Illumina PE</strong> data to the assembly with <font color="red"><strong>bowtie2</strong></font> (using the â€œvery fastâ€ pre-set). For <font color="blue">all PacBio assemblies</font> the consensus step was performed by running <font color="red"><strong>Quiverçº é”™</strong></font> (Genomic Consensus version 2.3.3) (with default parameters) on <font color="blue">raw PacBio contigs</font> and using the full 63Ã— of PacBio data.</p>
<p>Assembled contigs were further joined into scaffolds using a combination of <strong>BioNano optical mapping</strong> data, <strong>Hi-C</strong> chromatin conformation capture data, and <strong>genetic maps.</strong> The best <font color="blue">assemblies from Canu and Falcon</font> were <font color="blue">independently combined with</font> BioNano optical mapping data using the <font color="red">IrysView</font> software to develop the Canu + BioNano (CB) and Falcon + BioNano (FB) assemblies, respectively. The BioNano scaffolding process identified conflicts between the assembled contigs and the optical map, indicating some degree of misassembly in both Canu and Falcon results.</p>
<p><strong>Assembly polishing</strong></p>
<p><font color="red"><strong>Pilon</strong></font> (version1.21) [31] was <font color="blue">run iteratively</font> on the assembly, with <font color="blue">Illumina sequence realigned to the polished assembly</font> at each iteration and then <font color="blue">alignments passed to Pilon to call the next consensus</font>. Alignments for Pilon were produced using <font color="red"><strong>BWA mem</strong></font> (v0.7.17) with default settings. <em>k</em>-mer spectrum comparisons were made using the <em>k</em>-mer analysis toolkit (<font color="red"><strong>KAT</strong></font>) (version 2.3.4) [13] (KAT comp) at a <em>k</em>-mer size of 32, and <font color="blue">the metric used to assess each iteration was the number of <strong>k-mers shared between</strong> the <strong>assembly</strong> and the <strong>Illumina</strong> reads</font>. </p>
<p>In a second consensus phase, <font color="blue">RNA-Seq</font> reads were aligned as single end (SE), to the genome using <font color="red"><strong>Hisat</strong></font> (version 2.1.0)  with default parameters. This time the effectiveness of consensus calling was <font color="blue">assessed by analysis of <strong>full-length alignments</strong> of assembled RNA-Seq transcripts</font>. All transcripts designated as â€œcompleteâ€ by Evigene  were aligned to the genome with BLAT  (version 3.4, with [34] minimum match identity 90%). Alignments were filtered to <strong>retain only full-length alignments</strong> (i.e., from query start to query end). </p>
<p>Finally, the number of gaps in the alignments (query gaps + target gaps) was used as a metric with the rationale that this serves as a proxy for the number of indels in alignments of assembled messenger RNA sequence.</p>
<p><strong>Scaffold clustering and genome anchoring using Hi-C</strong></p>
<p>Hi-C reads were aligned to the polished scaffolds in CB with <font color="red"><strong>Bowtie2</strong></font> (version 2.3.3.1) . Each of the paired reads was aligned independently; then these SE alignments were subsequently merged as recommended by the LACHESIS developers. Based on the alignments, <font color="red">CB scaffolds were <strong>arranged</strong> into <strong>17 ordered</strong> and <strong>oriented</strong> <strong>clusters</strong> using the <strong>LACHESIS</strong> software</font>. As an <font color="blue">internal check</font>, the process was completed on <font color="blue">2 different random 95% sub-samplings of the Hi-C data</font>, as well as on the full data set. The clusters produced by <strong>all 3</strong> of these LACHESIS runs were <font color="blue">identical</font>. <font color="blue">LACHESIS produces groups of scaffolds</font> that are ordered and oriented relative to each other. These scaffold groupings were <font color="blue">compared with the genetic map</font> and the <strong>consistency</strong> of these sources of information was <strong>assessed</strong>. <font color="blue">The SNP probe mapping </font>at the scaffold <strong>validation step</strong> was compared with the clusters produced by LACHESIS.</p>
<p><strong>Illumina assembly</strong></p>
<p>The Illumina data were also assembled on their own, using the de Bruijn graphâ€“based assembler <font color="red"><strong>SOAPdenovo2</strong></font> (version 2.04). <font color="blue"><strong>This assembly</strong></font> was <font color="blue"><strong>used in various ways</strong></font> during the course of the pear genome project (for further scaffold validation, for training the <em>ab initio</em> gene predictors, etc.). <font color="red"><strong>ã€å€Ÿé‰´ã€‘The Illumina data were assembled twice</strong></font>. The <font color="red">first pass</font> contigs were screened using the Kraken software and an index built from the entire RefSeq database. <font color="red">Reads aligning to contaminant contigs were <strong>removed</strong></font> and the remaining data were assembled again.</p>
<h1 id="15-2021-05-nbt-äºº-ã€å››å€ä½“åœŸè±†å‚è€ƒhicç»„è£…2ã€‘"><a href="#15-2021-05-nbt-äºº-ã€å››å€ä½“åœŸè±†å‚è€ƒhicç»„è£…2ã€‘" class="headerlink" title="15.  2021.05-nbt äºº ã€å››å€ä½“åœŸè±†å‚è€ƒhicç»„è£…2ã€‘"></a>15.  2021.05-nbt äºº ã€å››å€ä½“åœŸè±†å‚è€ƒhicç»„è£…2ã€‘</h1><blockquote>
<p>Chromosome-scale, haplotype-resolved assembly of human genomes</p>
</blockquote>
<p><img src="/blog/2022/03/14/2022-03-14-HiFi-6-HiC-assembly/10.png" alt="10"></p>
<p>outline of the phased assembly algorithm, <font color="red"><strong>DipAsm</strong></font></p>
<p>Assemble HiFi reads into unphased contigs using <font color="red"><strong>Peregrine</strong></font> (1);</p>
<p> <font color="blue"><strong>group</strong> and <strong>order</strong></font> contigs into scaffolds with <font color="blue"><strong>Hi-C</strong> data</font> using <font color="blue"><strong>Hirise/3D-DNA</strong></font> (3D de novo assembly) (2);</p>
<p> <font color="blue"><strong>map HiFi reads</strong> to scaffolds</font> and <font color="blue"><strong>call heterozygous SNPs</strong></font> using <font color="red"><strong>DeepVariant</strong></font> (3);</p>
<p> <font color="blue"><strong>phase</strong></font> heterozygous SNP calls with both HiFi and Hi-C data using <font color="red"><strong>WhatsHap plus HapCUT2</strong></font> (4);</p>
<p> partition reads based on their phase using <font color="red"><strong>WhatsHap</strong></font> (5);</p>
<p> assemble partitioned reads into <font color="blue"><strong>phased contigs</strong></font> using <font color="red"><strong>Peregrine</strong></font> (6).</p>
<p><strong>Phased sequence assembly.</strong> </p>
<p>We ran <font color="red"><strong>Peregrine v.0.1.5.2</strong></font> with the following command line: â€˜peregrine asm reads.lst 24 24 24 24 24 24 24 24 24 â€“with-consensus â€“shimmer-r 3 â€“best_n_ovlp 8â€“output asmâ€™, where file â€˜reads. lstâ€™ gives the list of input read files and directory â€˜asmâ€™ holds the output assembly. </p>
<p>We <font color="blue"><strong>mapped Hi-C reads to contigs</strong></font> with <font color="red"><strong>BWA-MEM</strong></font> v.0.7.17 and <font color="blue"><strong>scaffolded</strong></font> the Peregrine contigs with <font color="red"><strong>juicer v.1.5</strong></font> and <font color="red"><strong>3D-DNA v.180922</strong></font>. We preprocessed data with <font color="green"><strong>â€˜juicer.sh -d juicer -p chrom.sizes -y cut-sites.txt -z contigs.fa -Dâ€™</strong></font>, where file <strong>â€˜cut-sites.txtâ€™</strong> was <strong>generated</strong> using the <strong>generate_site_positions_Arima. py</strong> script, which <strong>outputs merged_nodups.txt</strong>. The scaffolds were produced with <font color="green"><strong>â€˜run-asm-pipeline.sh -m haploid contigs.fa merged_nodups.txtâ€™</strong></font>. </p>
<p>We then called small variants using DeepVariant v.0.8.0 with the pretrained PacBio model. </p>
<p>We mapped Hi-C reads to the scaffolds and ran HapCUT2 v.1.1 over heterozygous SNP sites to obtain sparse phasing at the chromosome scale. The resulting haplotypes were then combined with PacBio HiFi data using WhatsHap v.0.18, with default parameters, to generate fine-scale, chromosome-long phasing. </p>
<p>We partitioned HiFi reads based on the phases of SNPs residing on these reads, and ran Peregrine again for reads on the same haplotype from the same scaffold. This provided the final phased assembly.</p>
<h1 id="16-2018-11-ng-åŒæºå¤šå€ä½“ç”˜è”—-ã€å››å€ä½“åœŸè±†å‚è€ƒhicç»„è£…3ã€‘"><a href="#16-2018-11-ng-åŒæºå¤šå€ä½“ç”˜è”—-ã€å››å€ä½“åœŸè±†å‚è€ƒhicç»„è£…3ã€‘" class="headerlink" title="16.  2018.11-ng åŒæºå¤šå€ä½“ç”˜è”— ã€å››å€ä½“åœŸè±†å‚è€ƒhicç»„è£…3ã€‘"></a>16.  2018.11-ng åŒæºå¤šå€ä½“ç”˜è”— ã€å››å€ä½“åœŸè±†å‚è€ƒhicç»„è£…3ã€‘</h1><blockquote>
<p>Allele-defined genome of theautopolyploid sugarcane <em>Saccharum spontaneum</em> L.</p>
</blockquote>
<p><strong>ææ–™</strong></p>
<p>æ ½åŸ¹ç”˜è”—çš„<font color="blue">å•å€å‹ä¸ªä½“</font><em>S.spontaneum</em>â€œAP85-441â€(1n = 4x = 32) è¿›è¡ŒåŸºå› ç»„<em>denovo</em>æµ‹åºç ”ç©¶ã€‚<br>64ä»½ç”˜è”—ï¼ˆ<em>Saccharum spontaneum</em>ï¼‰ç¾¤ä½“ææ–™è¿›è¡Œ<font color="blue">é—ä¼ å¤šæ ·æ€§ç ”ç©¶</font>ã€‚</p>
<p><strong>æ–¹æ³•</strong></p>
<p>äºŒä»£280-bpå’Œ500-bp(PE) å°ç‰‡æ®µæ–‡åº“ï¼ˆIllumina HiSeq 2500ï¼‰<br>BACæ–‡åº“ï¼ˆIllumina HiSeq 2500ï¼ŒPE250ï¼‰ï¼›ä¸‰ä»£å¤§ç‰‡æ®µæ–‡åº“~20kb SMRTbellTMï¼ˆ<font color="blue">PacbioRSII</font>ï¼‰<br>Hi-Cæ–‡åº“ï¼ˆ<font color="blue">HindIIIé…¶åˆ‡</font>ï¼‰ï¼Œ500-700bpå°ç‰‡æ®µæ–‡åº“ï¼ˆHiSeqX tenï¼ŒPE150ï¼‰</p>
<p><strong>ä¸»è¦ç»“æœ</strong>ï¼š</p>
<p><strong>ç”˜è”—åŸºå› ç»„denovo</strong><br>åˆ©ç”¨BACæ–‡åº“è¿›è¡ŒäºŒä»£æµ‹åºï¼ŒALLPATH-LG,SPAdes å’ŒSOAPdenovo2è¿›è¡Œåˆæ­¥ç»„è£…ï¼Œè¿›ä¸€æ­¥åˆ©ç”¨<font color="blue">87X ä¸‰ä»£PacBioæ•°æ®è¿›è¡Œ<strong>çº é”™</strong></font>ï¼Œ<font color="red"><strong>CANU</strong></font>ç»„è£…ï¼Œç»„è£…åŸºå› ç»„3.13Gï¼Œ<font color="blue">contigN50=45 kb</font>ï¼ˆæµå¼è¯„ä¼°åŸºå› ç»„å¤§å°ï¼š3.36Gï¼‰ï¼›é€šè¿‡Hi-CæŸ“è‰²ä½“æŒ‚è½½æŠ€æœ¯ï¼Œé’ˆå¯¹<font color="red"><strong>å¤šå€ä½“æŸ“è‰²æŒ‚è½½æœ€æ–°ç®—æ³•ALLHIC</strong></font>ï¼Œè¿›ä¸€æ­¥å°†contigæ°´å¹³æå‡è‡³æŸ“è‰²ä½“æ°´å¹³ï¼Œ<strong>æŒ‚è½½ç‡è¾¾åˆ°93%<strong>ï¼›é€šè¿‡BUSCOå’ŒäºŒä»£æ•°æ®å›æ¯”è¯„ä¼°åŸºå› ç»„å®Œæ•´æ€§é«˜è¾¾</strong>97.3%<strong>ã€‚å†é€šè¿‡<font color="red"><strong>MAKER</strong></font>å¯¹åŒæºå¤šå€ä½“ç”˜è”—<font color="blue">ç‰¹å¼‚ç­‰ä½åŸºå› è¿›è¡Œ</font></strong>æ‰‹åŠ¨æ³¨é‡Š</strong>ï¼Œä»¥è·å¾—é«˜è´¨é‡çš„ç”˜è”—åŸºå› ç»„</p>
<p>**åŸºç¡€æŸ“è‰²ä½“æ•°ç›®çš„å‡å°‘ *<em>ç¦¾æœ¬ç§‘æŸ“è‰²ä½“æ•°è¿›åŒ–ï¼ˆé«˜ç²±n= 10åˆ°ç”˜è”—n= 8ï¼‰<br>AP85-441åŸºå› ç»„çš„ç»„è£…æ˜¾ç¤ºäº†</em>S.spontaneum*çš„æŸ“è‰²ä½“æ•°ç›®ä»10é™åˆ°8ï¼Œè€Œè¿™ä¸é¢‘ç¹å¤åˆ¶çš„å¤å¤åˆ¶æŸ“è‰²ä½“å¯¹ç›¸å…³ï¼Œé€šè¿‡ä¸é«˜ç²±çš„èšç±»æ¯”å¯¹ï¼Œå‘ç°é«˜ç²±ç¥–å…ˆ5æŸ“è‰²ä½“å’Œ8å·æŸ“è‰²ä½“åŒæºç‰©ç»å†äº†æŸ“è‰²ä½“è£‚å˜ï¼ˆå›¾2ï¼‰ã€‚SbChr05ï¼ˆA12ï¼‰çš„ç¥–å…ˆæŸ“è‰²ä½“æ–­è£‚åˆ†ä¸ºä¸¤ä¸ªä¸»è¦éƒ¨åˆ†ï¼Œå³C5Sï¼ˆA12Sï¼‰å’ŒC5Lï¼ˆA12Lï¼‰ï¼Œåˆ†åˆ«è½¬ç§»åˆ°SbChr06ï¼ˆA2ï¼‰å’ŒSbChr07ï¼ˆA5ï¼‰çš„ç¥–å…ˆæŸ“è‰²ä½“ï¼›SbChr8ï¼ˆA11ï¼‰çš„ç¥–å…ˆæŸ“è‰²ä½“æ–­è£‚ä¸ºä¸¤ä¸ªä¸»è¦çš„éƒ¨åˆ†ï¼Œå³C8Sï¼ˆA11Sï¼‰å’ŒC8Lï¼ˆA11Lï¼‰ï¼Œåˆ†åˆ«è½¬ç§»åˆ°SbChr09ï¼ˆA6ï¼‰å’ŒSbChr02ï¼ˆA7+ A9ï¼‰çš„ç¥–å…ˆæŸ“è‰²ä½“ä¸­ã€‚SbChr8å’ŒSsChr5ä¹‹é—´åŠSbChr5å’ŒSsChr7ä¹‹é—´è¿‘ä¹åŒæºçš„çŸ­ç‰‡æ®µæ˜¯åœ¨é«˜ç²±ä¸ç”˜è”—åˆ†åŒ–å‰ï¼Œé«˜ç²±SSAå½¢æˆäº13.4MYAåŒæºåŸºå› çš„æ®‹ç•™ç‰©ï¼ŒåŒæ—¶å‘ç°ï¼ŒS5ä¸­è¾ƒå°çš„SSAåŒºåŸŸå’ŒS8ä¸­SSAçš„è¾ƒå¤§åŒºåŸŸåœ¨é‡æ’çš„AP85-441åŸºå› ç»„ä¸­ä¹Ÿæ˜¯ä¿å®ˆçš„ã€‚</p>
<p><strong>S.spontaneumçš„å¤šå€ä½“åŒ–åˆ†æ</strong><br>è¿›ä¸€æ­¥ç ”ç©¶å‘ç°äº†ç”˜è”—ï¼ˆ<em>Saccharum spontaneum</em> L.ï¼‰æŸ“è‰²ä½“ç”±é«˜ç²±ï¼ˆ<em>Sorghum</em>ï¼‰æ¼”å˜è¿›åŒ–è¿‡ç¨‹ä¸­ç»å†äº†æŸ“è‰²ä½“æ–­è£‚ä¸èåˆåï¼Œå‘ç”Ÿäº†ä¸¤æ¬¡WGDäº‹ä»¶ï¼ˆå›¾3ï¼‰ï¼›å¯¹ç”˜è”—çš„åŸºå› ç»„çš„å¤šå€åŒ–ç ”ç©¶ä¸­å‘ç°ï¼Œé€šè¿‡ä¸¤æ¬¡å…¨åŸºå› ç»„å¤åˆ¶äº‹ä»¶çš„å‘ç”Ÿï¼Œå¯¼è‡´äº†ç”˜è”—æŸ“è‰²ä½“çš„è‡ªå‘åŠ å€è¿‡ç¨‹ï¼›å¯¹åŒæºå¤šå€ä½“ç”˜è”—çš„ç‰¹å¼‚ç­‰ä½åŸºå› çš„é‰´å®šä½œä¸ºæœ¬ç ”ç©¶çš„å¦ä¸€äº®ç‚¹ï¼Œæ–‡ç« åœ¨é‰´å®šç­‰ä½åŸºå› çš„åŒæ—¶ï¼Œè¿›ä¸€æ­¥åˆ†æç ”ç©¶äº†ç­‰ä½åŸºå› çš„è¡¨è¾¾æ¨¡å¼ï¼Œç»“æœå‘ç°ä¸åŒå•å€å‹è¡¨è¾¾æ¨¡å¼ç›¸ä¼¼ï¼Œå¹¶æ— æ˜æ˜¾å·®å¼‚ï¼›å¯¹ç”˜è”—C4å…‰åˆä½œç”¨çš„ç ”ç©¶ä¸­é‰´å®šäº†24ä¸ªåŸºå› ï¼Œ7ç§å…³é”®é…¶å‚ä¸NADP-MEC4é€”å¾„ï¼›åœ¨ç”˜è”—ä¸­è”—ç³–å¯¹çš„ç§¯ç´¯çš„ç ”ç©¶ä¸­å‘ç°ä¸é«˜ç²±ç›¸æ¯”ç”˜è”—ä¸­çš„æ¶²æ³¡è†œç³–è½¬è¿è›‹ç™½ï¼ˆTSTsï¼Œä¸€ç§è”—ç³–è½¬è¿è›‹ç™½ï¼‰å‘ç”Ÿäº†åŸºå› å®¶æ—çš„æ‰©å¼ ï¼Œæ‰€ä»¥æ¨æµ‹TSTså‚ä¸è”—ç³–åœ¨æ¶²æ³¡ä¸­çš„ç§¯ç´¯ï¼›<em>S.spontaneum</em>ä¸ºç°ä»£æ ½åŸ¹ç”˜è”—æä¾›äº†æŠ—ç—…æ€§åŸºå› ï¼Œåœ¨å¯¹å…¶ç ”ç©¶ä¸­å‘ç°ï¼Œç”˜è”—ä¸­çš„80ï¼…çš„NBSç¼–ç åŸºå› ä½äºå››ä¸ªé‡æ’æŸ“è‰²ä½“ï¼ˆSs02ï¼ŒSs05ï¼ŒSs06å’ŒSs07ï¼‰ä¸Šï¼Œå…¶ä¸­51ï¼…ä½äºé‡æ’åŒºåŸŸã€‚</p>
<p><strong>S.spontaneumçš„èµ·æºä¸é—ä¼ å¤šæ ·æ€§åˆ†æ</strong><br>é€šè¿‡å¯¹64ä»½ç”˜è”—ï¼ˆ<em>Saccharum spontaneum</em>ï¼‰ç¾¤ä½“ææ–™é—ä¼ å¤šæ ·æ€§ç ”ç©¶ï¼Œå‘ç°å…¶å…·æœ‰å¹¿æ³›çš„è‡ªç„¶éƒ¨åˆ†èŒƒå›´ï¼ŒéåŠäºšæ´²ï¼Œå°åº¦æ¬¡å¤§é™†ï¼Œåœ°ä¸­æµ·å’Œéæ´²ï¼Œä¸”è‡ªç„¶ç§ç¾¤å…·æœ‰å¹¿æ³›çš„è¡¨å‹ï¼Œé—ä¼ å’Œå€æ€§æ°´å¹³å¤šæ ·æ€§ï¼ˆå›¾4ï¼‰ã€‚ç ”ç©¶å‘ç°ï¼Œä¸é«˜ç²±ç›¸æ¯”å…·æœ‰å¤§è§„æ¨¡æŸ“è‰²ä½“é‡æ’çš„<em>S. spontaneum</em>åŒºåŸŸå…·æœ‰æ¯”éé‡æ’åŒºåŸŸæ›´é«˜çš„é—ä¼ å¤šæ ·æ€§ï¼Œå¹¶ä¸”å¯èƒ½ç»å†äº†æ›´å¼ºçš„å¹³è¡¡é€‰æ‹©ã€‚è™½ç„¶å‡ æ¡å•ç‹¬çš„æŸ“è‰²ä½“æ²¡æœ‰æ˜¾ç¤ºå‡ºæ˜¾è‘—å·®å¼‚ï¼Œä½†æ˜¯æ¯”è¾ƒæ‰€æœ‰æŸ“è‰²ä½“ä¸Šçš„å¹³å‡å€¼æ˜¾ç¤ºé‡æ’åŒºåŸŸä¸­çš„æ ¸è‹·é…¸å¤šæ ·æ€§ï¼ˆ0.00025Â±0.00003ï¼‰è¿œé«˜äºéé‡æ’åŒºåŸŸä¸­çš„æ ¸è‹·é…¸å¤šæ ·æ€§ï¼ˆ0.00021Â±0.00001ï¼‰ï¼ŒæŸ“è‰²ä½“é‡æ’åŒºåŸŸçš„Tajimaâ€™s Dï¼ˆ-0.659Â±0.052ï¼‰è¿œè¿œé«˜äºéé‡æ’åŒºï¼ˆ-0.720Â±0.011ï¼‰ï¼Œé‡æ’åŒºçš„SNPå¯†åº¦(360.27Â±48.41) é«˜äºéé‡æ’åŒºåŸŸï¼ˆ297.46Â±12.65, P=0.001798ï¼‰ï¼›æ­¤å¤–ï¼ŒGOå¯Œé›†æ˜¾ç¤ºï¼ŒæŸ“è‰²ä½“éé‡æ’åŒºä¸»è¦å¯Œé›†åˆ°ä¸åŸºæœ¬ç”Ÿå‘½å‘¨æœŸï¼Œå…‰åˆä½œç”¨ï¼Œå‘¼å¸ä½œç”¨å’ŒATPåˆæˆçš„é€šè·¯ä¸Šã€‚é‡æ’åŒºä¸»è¦å¯Œé›†åœ¨GOç”Ÿç‰©åˆæˆå’Œä»£è°¢ï¼Œè·¨è†œè½¬è¿å’Œç¦»å­ç»“åˆé€šè·¯ä¸Šã€‚</p>
<p><em>S. spontaneum</em>é‡æ’åŒºå…·æœ‰é«˜åº¦çš„é—ä¼ å¤šæ ·æ€§å°†æ›´é€‚åº”ç¯å¢ƒå‹åŠ›ï¼Œä¾‹å¦‚å¯¹å„ç§éç”Ÿç‰©èƒè¿«ï¼ˆå¹²æ—±ï¼Œç›åº¦ï¼Œç¢±æ€§ï¼Œé‡‘å±ç¦»å­ç­‰ï¼‰çš„ååº”ï¼Œè¿™äº›å—åˆ°æ¬¡ç”Ÿç»†èƒç”Ÿç‰©åˆæˆå’Œä»£è°¢ï¼Œè·¨è†œè½¬è¿å’Œç¦»å­ç»“åˆçš„æ§åˆ¶çš„åŸºå› åœ¨è¿™äº›åŒºåŸŸå¯ä»¥æ£€æµ‹åˆ°ã€‚åœ¨å¤šå€åŒ–äº‹ä»¶åï¼Œé‡æ’åŒºåŸŸç»å†äº†æ›´å¼ºçš„å¹³è¡¡é€‰æ‹©ã€‚</p>
<p><strong>è®¨è®º</strong></p>
<p>æœ¬ç ”ç©¶åˆ©ç”¨å¤æ‚åŸºå› ç»„Hi-Cå»ºåº“æµ‹åºæŠ€æœ¯ï¼Œå¹¶ç»“åˆALLHiCç®—æ³•æˆåŠŸç»„è£…äº†é«˜å¤æ‚åŒæºå¤šå€ä½“ç”˜è”—åŸºå› ç»„AP85-441ï¼Œé«˜è´¨é‡çš„ç”˜è”—åŸºå› ç»„çš„è·å¾—ï¼Œä¸ºåç»­ç ”ç©¶å…¶è”—ç³–å«é‡ï¼ŒæŠ—æ€§ç­‰æ€§çŠ¶åŠå†œä¸šè‚²ç§åŸºç¡€å¥ å®šäº†åŸºç¡€ã€‚</p>
<p><strong>Genome assembly overview.</strong> </p>
<p>The sugarcane AP85-441 contig-level assembly incorporated sequencing data from a mixture of sequencing technologies (Supplementary Fig. 1), including BAC pools sequenced with Illumina HiSeq 2500 and whole-genome shotgun sequencing with PacBio RS II as well as Hi-C reads, followed by Illumina short reads polishing. Each BAC pool was independently assembled using <font color="red">ALLPATHS-LG, SPAdes and SOAPdenovo2</font>, and best results were retained. For PacBio assembly, <font color="red"><strong>Canu v1.511</strong></font> was used, as it is capable<br>of avoiding collapsed repetitive regions and haplotypes. Self-correction was performed with parameter corOutCoverage = 100, which allowed us to correct all of the input PacBio reads. The corrected reads, along with BAC-assembled contigs, were imported to the assembly step. <font color="blue">Chromosomal assembly</font> was constructed based on proximity-guided assembly using our newly developed program, <font color="red"><strong>ALLHIC</strong>, which is designed for <strong>polyploid</strong> genome scaffolding</font> (see Supplementary Note for details).</p>
<p><img src="/blog/2022/03/14/2022-03-14-HiFi-6-HiC-assembly/11.png" alt="11"></p>
<h1 id="17-2019-08-np-ALLHiCã€å››å€ä½“åœŸè±†å‚è€ƒhicç»„è£…4ã€‘"><a href="#17-2019-08-np-ALLHiCã€å››å€ä½“åœŸè±†å‚è€ƒhicç»„è£…4ã€‘" class="headerlink" title="17.  2019.08-np ALLHiCã€å››å€ä½“åœŸè±†å‚è€ƒhicç»„è£…4ã€‘"></a>17.  2019.08-np ALLHiCã€å››å€ä½“åœŸè±†å‚è€ƒhicç»„è£…4ã€‘</h1><blockquote>
<p>Assembly of allele-aware, chromosomal-scale autopolyploid genomes based on Hi-C data</p>
</blockquote>
<p><strong>existing approaches are mostly designed for diploid genomes and often with the aim of reconstructing a haploid representation, thereby having limited power to reconstruct chromosomes for autopolyploid genomes. We developed a novel algorithm (ALLHiC) that is capable of building allele-aware, chromosomal-scale assembly for <font color="red">auto-polyploid genomes</font> using Hi-C paired-end reads with innovative â€˜pruneâ€™ and â€˜optimizeâ€™ steps. Application on simulated data showed that ALLHiC can phase allelic contigs and substantially improve ordering and orientation when compared to other mainstream Hi-C assemblers</strong></p>
<h1 id="18-2020-05-nc-ç´«èŠ±è‹œè“¿åŒæºå››å€ä½“-06ã€å››å€ä½“åœŸè±†å‚è€ƒhicç»„è£…5ã€‘"><a href="#18-2020-05-nc-ç´«èŠ±è‹œè“¿åŒæºå››å€ä½“-06ã€å››å€ä½“åœŸè±†å‚è€ƒhicç»„è£…5ã€‘" class="headerlink" title="18. 2020.05-nc ç´«èŠ±è‹œè“¿åŒæºå››å€ä½“-06ã€å››å€ä½“åœŸè±†å‚è€ƒhicç»„è£…5ã€‘"></a>18. 2020.05-nc ç´«èŠ±è‹œè“¿åŒæºå››å€ä½“-06ã€å››å€ä½“åœŸè±†å‚è€ƒhicç»„è£…5ã€‘</h1><h1 id="19-2020-10-ng-äºŒå€ä½“åœŸè±†-ã€å››å€ä½“åœŸè±†å‚è€ƒhicç»„è£…6ã€‘"><a href="#19-2020-10-ng-äºŒå€ä½“åœŸè±†-ã€å››å€ä½“åœŸè±†å‚è€ƒhicç»„è£…6ã€‘" class="headerlink" title="19. 2020.10-ng äºŒå€ä½“åœŸè±† ã€å››å€ä½“åœŸè±†å‚è€ƒhicç»„è£…6ã€‘"></a>19. 2020.10-ng äºŒå€ä½“åœŸè±† ã€å››å€ä½“åœŸè±†å‚è€ƒhicç»„è£…6ã€‘</h1><blockquote>
<p>Haplotype-resolved genome analyses of a heterozygous diploid potato</p>
</blockquote>
<p>ref of 12: Hi-C-genetic-maps-for-<font color="red">correction</font></p>
<p>é©¬é“ƒè–¯(Solanum tuberosum L.)æ˜¯å…¨çƒé‡è¦çš„å—èŒä½œç‰©ã€‚äººä»¬æ­£åœ¨åŠªåŠ›å°†è¿™ç§ä½œç‰©ä»æ— æ€§ç¹æ®–çš„å››å€ä½“è½¬åŒ–ä¸ºç§å­ç¹æ®–çš„ã€è‡ªäº¤ç³»ä¸ºåŸºç¡€çš„æ‚äº¤ä½œç‰©ï¼Œä½†è¿™ä¸€è¿‡ç¨‹éœ€è¦å¯¹é©¬é“ƒè–¯åŸºå› ç»„æœ‰æ›´å¥½çš„äº†è§£ã€‚</p>
<p><strong>Genome assembly and phasing using PacBio CCS reads.</strong> </p>
<p>A total of 29 Gb CCS reads were assembled using <font color="red"><strong>Canu</strong></font> (v1.91) with the <font color="green"><strong>parameter</strong> â€“pacbio-hifi</font>.<br>Canu <font color="blue">generated <strong>two</strong> assemblies</font> composed of contigs and unitigs (Supplementary Table 4), and the <font color="blue"><strong>unitig</strong> assembly consisted of the <strong>contigs that split at any alternative paths</strong> in the assembly graph</font>. The contig assembly had longer continuity but more chimeric fragmentsåµŒåˆä½“ç‰‡æ®µ as revealed in the genetic mapping analysis. To avoid the mis-joining of two haplotypes, <font color="red"><strong>the unitig assembly</strong></font> rather than the contig assembly was chosen for the <font color="red"><strong>subsequent analysis</strong></font>. The unitigs were then <font color="blue"><strong>polished</strong> iteratively using <strong>two rounds</strong></font> of <font color="red"><strong>Pilon</strong></font> with ~150 Gb of WGS <strong>Illumina</strong> data, generating the genome <strong>draft RHgv2</strong>.</p>
<p>Similarly, the sequenced reads of RH selfing progenyåä»£ were mapped to unitigs of RHgv2 to perform <font color="blue"><strong>genetic grouping</strong></font>. Because the unitigs were relatively long (N50 = 2 Mb), windows with a size of 200 kb rather than the whole unitig were used. If the adjacent windows of one unitig showed contrary read distribution, the unitig was defined as chimeric and broken between windows; 40 chimeric unitigs with a total length of 95 Mb were broken. In total, <font color="blue">1.31 Gb of 1.53 Gb sequences were assigned to 24 linkage groups</font>.</p>
<p>After merging, 141 Mb sequences and 5,252 annotated genes of RHgv1 were added to the RHgv2, yielding a 1.67 Gb genome draft with 1.54 Gb sequences assigned to 24 groups, termed RHgv3 (Supplementary Table 6). The sequences from the RHgv1 and RHgv2 assemblies were named as ontctg* and unitig* in the AGP file, respectively (Supplementary Data 3).</p>
<p><strong>Construction of pseudochromosomes.</strong> </p>
<p>As no approach generated satisfactory results on the RH genome, we introduced the group information derived from the genetic mapping to assist the Hi-C application on chromosome-level assembly. The process was performed on RHgv3 including three steps as follows:</p>
<ol>
<li><font color="blue"><strong>Align</strong></font>. The 24 previously determined groups were divided into <font color="blue"><strong>two haplotypes</strong> to <strong>generate</strong> <strong>two</strong> <strong>pseudohaploid genome drafts</strong></font>. The 132 Mb sequences that could not be assigned to any group were added to two pseudohaploid genomes. Total <font color="blue">Hi-C reads were <strong>aligned</strong> to <strong>each</strong> pseudohaploid genome using <strong>HiC-Pro</strong> to calculate the <strong>contact frequency</strong></font>. This step <font color="blue">yielded <strong>two bam files</strong> for the two pseudohaploid genomes</font>.</li>
<li><font color="blue"><strong>Rescue</strong></font>. Using the <font color="blue">bam file as input</font>, the <em>rescue</em> function in <font color="red"><strong>ALLHiC</strong></font> was applied to assign unplaced sequences to known groups. Because the 132 Mb <strong>unplaced sequences</strong> were <strong>added</strong> to two pseudohaploid genomes and <strong>processed twice</strong>, the rescued <strong>results were redundant</strong>. For every <strong>unplaced sequence</strong>, we <strong>considered its best Hi-C signal density</strong> to decide the group to which it belonged. After this step, the sequence content of 24 groups was updated with an <strong>extra 75.6 Mb</strong> sequences assigned to proper groups.</li>
<li><font color="blue"><strong>Optimize and build</strong></font>. For each pseudohaploid genome, using <font color="red">the bam file and the updated group file as input</font>, the <font color="red"><strong><em>optimize</em> function in ALLHiC</strong></font> decidedthe <strong>order</strong> and <strong>orientation</strong> of scaffolds for each group; thus, the <font color="red"><strong><em>build</em> function</strong></font> generated fasta sequences on that basis. By performing this step, we identified the pseudochromosomes for 24 groups. The order and orientation of scaffolds on chromosomes are provided in Supplementary Data 3.</li>
</ol>
<h1 id="20-2021-12-gb-ææ ‘ã€å››å€ä½“åœŸè±†å‚è€ƒhicç»„è£…7ã€‘"><a href="#20-2021-12-gb-ææ ‘ã€å››å€ä½“åœŸè±†å‚è€ƒhicç»„è£…7ã€‘" class="headerlink" title="20. 2021.12-gb ææ ‘ã€å››å€ä½“åœŸè±†å‚è€ƒhicç»„è£…7ã€‘"></a>20. 2021.12-gb ææ ‘ã€å››å€ä½“åœŸè±†å‚è€ƒhicç»„è£…7ã€‘</h1><blockquote>
<p>Gamete binning: chromosome-level and haplotype-resolved genome assembly enabled by high-throughput single-cell sequencing of gamete genomes</p>
</blockquote>
<p>ref of 12: Hi-C-genetic-maps-for-correction</p>
<p>ä¸€ç§åŸºäºå•å€ä½“é…å­å•ç»†èƒæµ‹åºçš„æ–¹æ³•ã€‚åŸºäº445ä¸ªä¸ªä½“èŠ±ç²‰ç²’çš„å…¨åŸºå› ç»„æµ‹åºï¼Œç»„è£…äº†ä¸€æ ªäºŒå€ä½“ææ ‘çš„ä¸¤ä¸ªåŸºå› ç»„ã€‚</p>
<p><strong>Genome size estimation</strong></p>
<p>After trimming off 10x Genomics barcodes and hexamers from the 61.7 Gb reads of the two 10x sc-CNV libraries, k-mer counting (k=21) was performed with <font color="red"><strong>Jellyfish</strong></font>. The k-mer histogram was provided to<font color="red"> <strong>findGSE</strong></font> to estimate the size of the â€œRojo PasioÌnâ€ genome under the heterozygous mode (with â€œexp_hom=200â€; Add- itional file 1: Fig. S3).</p>
<p><strong>Preliminary diploid-genome assembly and curation</strong></p>
<p>With the 19.9 G raw PacBio reads of â€œRojo PasioÌnâ€ (Additional file 1: Fig. S2), a pre-liminary diploid assembly was constructed using <font color="red"><strong>Canu</strong></font> (with options â€œgenome- Size=242500000 corMhapSensitivity=high corMinCoverage=0 corOutCoverage=100 correctedErrorRate=0.105â€).</p>
<p>All raw <strong>Illumina reads</strong> from the 10x libraries were <font color="red">firstly <strong>aligned</strong> to the <strong>initial assembly</strong></font> using <font color="red"><strong>bowtie2</strong></font>. Then, the purge haplotigs pipeline was used to remove haplotigs (i.e., haplotype-specific contigs inflating the true haploid genome) based on statistical analysis of sequencing depth and identify primary contigs to build up a curated haploid assembly. To reduce the false-positive rate in defining haplotigs, each haplotig was blasted to the curated assembly; if over 50% of the haplotig could not be covered by any primary contigs, it was re-collected as a primary contig.</p>

    </div>

    
    
    
        

<div>
<ul class="post-copyright">
  <li class="post-copyright-author">
    <strong>Post author:  </strong>Wang C
  </li>
  <li class="post-copyright-link">
    <strong>Post link: </strong>
    <a href="https://wangc1106.github.io/blog/2022/03/14/2022-03-14-HiFi-6-HiC-assembly/" title="HiFi-6-[HiFi-HiC-assembly-method]">https://wangc1106.github.io/blog/2022/03/14/2022-03-14-HiFi-6-HiC-assembly/</a>
  </li>
  <li class="post-copyright-license">
    <strong>Copyright Notice:  </strong>All articles in this blog are licensed under <a href="https://creativecommons.org/licenses/by-nc-sa/4.0/" rel="noopener" target="_blank"><i class="fab fa-fw fa-creative-commons"></i>BY-NC-SA</a> unless stating additionally.
  </li>
</ul>
</div>


      <footer class="post-footer">
          <div class="post-tags">
              <a href="/blog/tags/denovo-III/" rel="tag"># denovo-III</a>
          </div>

        


        
    <div class="post-nav">
      <div class="post-nav-item">
    <a href="/blog/2022/03/09/2022-03-09-HiFi-5-HiC-Pro/" rel="prev" title="HiFi-5-[HiC-Pro]-[MCscan]-[ALLHiC]">
      <i class="fa fa-chevron-left"></i> HiFi-5-[HiC-Pro]-[MCscan]-[ALLHiC]
    </a></div>
      <div class="post-nav-item"></div>
    </div>
      </footer>
    
  </article>
  
  
  



          </div>
          

<script>
  window.addEventListener('tabs:register', () => {
    let { activeClass } = CONFIG.comments;
    if (CONFIG.comments.storage) {
      activeClass = localStorage.getItem('comments_active') || activeClass;
    }
    if (activeClass) {
      let activeTab = document.querySelector(`a[href="#comment-${activeClass}"]`);
      if (activeTab) {
        activeTab.click();
      }
    }
  });
  if (CONFIG.comments.storage) {
    window.addEventListener('tabs:click', event => {
      if (!event.target.matches('.tabs-comment .tab-content .tab-pane')) return;
      let commentClass = event.target.classList[1];
      localStorage.setItem('comments_active', commentClass);
    });
  }
</script>

        </div>
          
  
  <div class="toggle sidebar-toggle">
    <span class="toggle-line toggle-line-first"></span>
    <span class="toggle-line toggle-line-middle"></span>
    <span class="toggle-line toggle-line-last"></span>
  </div>

  <aside class="sidebar">
    <div class="sidebar-inner">

      <ul class="sidebar-nav motion-element">
        <li class="sidebar-nav-toc">
          Table of Contents
        </li>
        <li class="sidebar-nav-overview">
          Overview
        </li>
      </ul>

      <!--noindex-->
      <div class="post-toc-wrap sidebar-panel">
          <div class="post-toc motion-element"><ol class="nav"><li class="nav-item nav-level-1"><a class="nav-link" href="#1-2016-05-np-%E7%A2%A7%E5%86%AC%E8%8C%84"><span class="nav-number">1.</span> <span class="nav-text">1. 2016.05-np ç¢§å†¬èŒ„</span></a><ol class="nav-child"><li class="nav-item nav-level-2"><a class="nav-link" href="#Genome-sequencing-assembly-and-annotation"><span class="nav-number">1.1.</span> <span class="nav-text">Genome sequencing, assembly and annotation.</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#Repetitive-elements-analysis"><span class="nav-number">1.2.</span> <span class="nav-text">Repetitive elements analysis</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#Gene-data-mining"><span class="nav-number">1.3.</span> <span class="nav-text">Gene data mining</span></a></li></ol></li><li class="nav-item nav-level-1"><a class="nav-link" href="#2-2017-04-ng-%E5%B1%B1%E7%BE%8A%F0%9F%90%90"><span class="nav-number">2.</span> <span class="nav-text">2. 2017.04-ng å±±ç¾ŠğŸ</span></a><ol class="nav-child"><li class="nav-item nav-level-2"><a class="nav-link" href="#data"><span class="nav-number">2.1.</span> <span class="nav-text">data</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#%E7%BB%84%E8%A3%85%E7%AD%96%E7%95%A5%E5%8F%8A%E7%BB%84%E8%A3%85%E7%BB%93%E6%9E%9C"><span class="nav-number">2.2.</span> <span class="nav-text">ç»„è£…ç­–ç•¥åŠç»„è£…ç»“æœ</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#%E7%BB%84%E8%A3%85%E8%AF%84%E4%BC%B0%E5%8F%8A%E6%AF%94%E8%BE%83%E5%88%86%E6%9E%90"><span class="nav-number">2.3.</span> <span class="nav-text">ç»„è£…è¯„ä¼°åŠæ¯”è¾ƒåˆ†æ</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#%E5%9F%BA%E5%9B%A0%E7%BB%84%E6%B3%A8%E9%87%8A"><span class="nav-number">2.4.</span> <span class="nav-text">åŸºå› ç»„æ³¨é‡Š</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#%E5%90%AF%E7%A4%BA%E5%BD%95"><span class="nav-number">2.5.</span> <span class="nav-text">å¯ç¤ºå½•</span></a></li></ol></li><li class="nav-item nav-level-1"><a class="nav-link" href="#3-2020-09-cell-%E5%A4%A7%E8%B1%86%E6%B3%9B%E5%9F%BA%E5%9B%A0%E7%BB%84"><span class="nav-number">3.</span> <span class="nav-text">3. 2020.09-cell å¤§è±†æ³›åŸºå› ç»„</span></a><ol class="nav-child"><li class="nav-item nav-level-2"><a class="nav-link" href="#SNP-Calling-and-Phylogenetic-Analyses"><span class="nav-number">3.1.</span> <span class="nav-text">SNP Calling and Phylogenetic Analyses</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#Genome-Assembly"><span class="nav-number">3.2.</span> <span class="nav-text">Genome Assembly</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#Repeat-Analysis-and-Gene-Annotation"><span class="nav-number">3.3.</span> <span class="nav-text">Repeat Analysis and Gene Annotation</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#Synteny-Analysis"><span class="nav-number">3.4.</span> <span class="nav-text">Synteny Analysis</span></a></li></ol></li><li class="nav-item nav-level-1"><a class="nav-link" href="#4-2019-nc-%E5%A4%A7%E8%B1%86%E6%B3%9B%E5%9F%BA%E5%9B%A0%E7%BB%84-%E7%BB%84%E8%A3%85%E5%8F%82%E8%80%831"><span class="nav-number">4.</span> <span class="nav-text">4. 2019-nc å¤§è±†æ³›åŸºå› ç»„-ç»„è£…å‚è€ƒ1</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#5-2019-09-scienceChina-%E5%A4%A7%E8%B1%86%E6%B3%9B%E5%9F%BA%E5%9B%A0%E7%BB%84-%E7%BB%84%E8%A3%85%E5%8F%82%E8%80%832"><span class="nav-number">5.</span> <span class="nav-text">5. 2019.09-scienceChina å¤§è±†æ³›åŸºå› ç»„-ç»„è£…å‚è€ƒ2</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#6-2020-05-nc-%E5%90%8C%E6%BA%90%E5%9B%9B%E5%80%8D%E4%BD%93%E7%B4%AB%E8%8A%B1%E8%8B%9C%E8%93%BF"><span class="nav-number">6.</span> <span class="nav-text">6. 2020.05-nc åŒæºå››å€ä½“ç´«èŠ±è‹œè“¿</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#7-2021-11-gb-%E6%A4%B0%E5%AD%90%F0%9F%A5%A5"><span class="nav-number">7.</span> <span class="nav-text">7. 2021.11-gb æ¤°å­ğŸ¥¥</span></a><ol class="nav-child"><li class="nav-item nav-level-2"><a class="nav-link" href="#Hi-C-based-genome-assembly"><span class="nav-number">7.1.</span> <span class="nav-text">Hi-C-based genome assembly</span></a></li></ol></li><li class="nav-item nav-level-1"><a class="nav-link" href="#8-2021-11-nc-%E9%87%91%E7%B2%9F%E5%85%B0%E5%9F%BA%E5%9B%A0%E7%BB%84"><span class="nav-number">8.</span> <span class="nav-text">8. 2021.11-nc é‡‘ç²Ÿå…°åŸºå› ç»„</span></a><ol class="nav-child"><li class="nav-item nav-level-2"><a class="nav-link" href="#Genome-size-estimate-and-assembly"><span class="nav-number">8.1.</span> <span class="nav-text">Genome size estimate and assembly.</span></a></li></ol></li><li class="nav-item nav-level-1"><a class="nav-link" href="#9-2022-01-gb-%E6%B2%B9%E8%8C%B6"><span class="nav-number">9.</span> <span class="nav-text">9. 2022.01-gb æ²¹èŒ¶</span></a><ol class="nav-child"><li class="nav-item nav-level-2"><a class="nav-link" href="#%E6%8A%B1%E5%AD%90%E6%80%80%E8%83%8E%E7%9A%84%E6%B2%B9%E8%8C%B6%E8%89%AF%E7%A7%8D%E8%82%B2%E7%A7%8D%E7%BC%93%E6%85%A2"><span class="nav-number">9.1.</span> <span class="nav-text">æŠ±å­æ€€èƒçš„æ²¹èŒ¶è‰¯ç§è‚²ç§ç¼“æ…¢</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#%E6%9F%93%E8%89%B2%E4%BD%93%E7%BA%A7%E5%88%AB%E7%BB%84%E8%A3%85"><span class="nav-number">9.2.</span> <span class="nav-text">æŸ“è‰²ä½“çº§åˆ«ç»„è£…</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#Genome-sequencing-de-novo-assembly-and-annotation"><span class="nav-number">9.3.</span> <span class="nav-text">Genome sequencing, de novo assembly, and annotation</span></a></li></ol></li><li class="nav-item nav-level-1"><a class="nav-link" href="#10-2022-01-HorticRes-%E7%94%98%E8%8F%8A"><span class="nav-number">10.</span> <span class="nav-text">10. 2022.01-HorticRes ç”˜èŠ</span></a><ol class="nav-child"><li class="nav-item nav-level-2"><a class="nav-link" href="#Genome-assembly-and-quality-assessment"><span class="nav-number">10.1.</span> <span class="nav-text">Genome assembly and quality assessment</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#Hi-C-sequencing-and-assistant-assembly"><span class="nav-number">10.2.</span> <span class="nav-text">Hi-C sequencing and assistant assembly</span></a></li></ol></li><li class="nav-item nav-level-1"><a class="nav-link" href="#11-2022-01-HorticRes-%E7%B2%89%E8%91%9B"><span class="nav-number">11.</span> <span class="nav-text">11. 2022.01-HorticRes ç²‰è‘›</span></a><ol class="nav-child"><li class="nav-item nav-level-2"><a class="nav-link" href="#Genome-survey"><span class="nav-number">11.1.</span> <span class="nav-text">Genome survey</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#Genome-assembly"><span class="nav-number">11.2.</span> <span class="nav-text">Genome assembly</span></a></li></ol></li><li class="nav-item nav-level-1"><a class="nav-link" href="#12-2022-03-ng-%E5%9B%9B%E5%80%8D%E4%BD%93%E5%9C%9F%E8%B1%86"><span class="nav-number">12.</span> <span class="nav-text">12. 2022.03-ng å››å€ä½“åœŸè±†</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#13-2022-03-ng-%E5%85%AD%E5%80%8D%E4%BD%93%E5%B0%8F%E9%BA%A6"><span class="nav-number">13.</span> <span class="nav-text">13. 2022.03-ng å…­å€ä½“å°éº¦</span></a><ol class="nav-child"><li class="nav-item nav-level-2"><a class="nav-link" href="#Genome-assembly-and-validation"><span class="nav-number">13.1.</span> <span class="nav-text">Genome assembly and validation.</span></a></li></ol></li><li class="nav-item nav-level-1"><a class="nav-link" href="#14-2019-10-%E4%BA%8C%E5%80%8D%E4%BD%93%E5%B7%B4%E7%89%B9%E5%88%A9%E7%89%B9%E6%A2%A8-%E3%80%90%E5%9B%9B%E5%80%8D%E4%BD%93%E5%9C%9F%E8%B1%86%E5%8F%82%E8%80%83hic%E7%BB%84%E8%A3%851%E3%80%91"><span class="nav-number">14.</span> <span class="nav-text">14.  2019.10 äºŒå€ä½“å·´ç‰¹åˆ©ç‰¹æ¢¨ ã€å››å€ä½“åœŸè±†å‚è€ƒhicç»„è£…1ã€‘</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#15-2021-05-nbt-%E4%BA%BA-%E3%80%90%E5%9B%9B%E5%80%8D%E4%BD%93%E5%9C%9F%E8%B1%86%E5%8F%82%E8%80%83hic%E7%BB%84%E8%A3%852%E3%80%91"><span class="nav-number">15.</span> <span class="nav-text">15.  2021.05-nbt äºº ã€å››å€ä½“åœŸè±†å‚è€ƒhicç»„è£…2ã€‘</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#16-2018-11-ng-%E5%90%8C%E6%BA%90%E5%A4%9A%E5%80%8D%E4%BD%93%E7%94%98%E8%94%97-%E3%80%90%E5%9B%9B%E5%80%8D%E4%BD%93%E5%9C%9F%E8%B1%86%E5%8F%82%E8%80%83hic%E7%BB%84%E8%A3%853%E3%80%91"><span class="nav-number">16.</span> <span class="nav-text">16.  2018.11-ng åŒæºå¤šå€ä½“ç”˜è”— ã€å››å€ä½“åœŸè±†å‚è€ƒhicç»„è£…3ã€‘</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#17-2019-08-np-ALLHiC%E3%80%90%E5%9B%9B%E5%80%8D%E4%BD%93%E5%9C%9F%E8%B1%86%E5%8F%82%E8%80%83hic%E7%BB%84%E8%A3%854%E3%80%91"><span class="nav-number">17.</span> <span class="nav-text">17.  2019.08-np ALLHiCã€å››å€ä½“åœŸè±†å‚è€ƒhicç»„è£…4ã€‘</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#18-2020-05-nc-%E7%B4%AB%E8%8A%B1%E8%8B%9C%E8%93%BF%E5%90%8C%E6%BA%90%E5%9B%9B%E5%80%8D%E4%BD%93-06%E3%80%90%E5%9B%9B%E5%80%8D%E4%BD%93%E5%9C%9F%E8%B1%86%E5%8F%82%E8%80%83hic%E7%BB%84%E8%A3%855%E3%80%91"><span class="nav-number">18.</span> <span class="nav-text">18. 2020.05-nc ç´«èŠ±è‹œè“¿åŒæºå››å€ä½“-06ã€å››å€ä½“åœŸè±†å‚è€ƒhicç»„è£…5ã€‘</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#19-2020-10-ng-%E4%BA%8C%E5%80%8D%E4%BD%93%E5%9C%9F%E8%B1%86-%E3%80%90%E5%9B%9B%E5%80%8D%E4%BD%93%E5%9C%9F%E8%B1%86%E5%8F%82%E8%80%83hic%E7%BB%84%E8%A3%856%E3%80%91"><span class="nav-number">19.</span> <span class="nav-text">19. 2020.10-ng äºŒå€ä½“åœŸè±† ã€å››å€ä½“åœŸè±†å‚è€ƒhicç»„è£…6ã€‘</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#20-2021-12-gb-%E6%9D%8F%E6%A0%91%E3%80%90%E5%9B%9B%E5%80%8D%E4%BD%93%E5%9C%9F%E8%B1%86%E5%8F%82%E8%80%83hic%E7%BB%84%E8%A3%857%E3%80%91"><span class="nav-number">20.</span> <span class="nav-text">20. 2021.12-gb ææ ‘ã€å››å€ä½“åœŸè±†å‚è€ƒhicç»„è£…7ã€‘</span></a></li></ol></div>
      </div>
      <!--/noindex-->

      <div class="site-overview-wrap sidebar-panel">
        <div class="site-author motion-element" itemprop="author" itemscope itemtype="http://schema.org/Person">
    <img class="site-author-image" itemprop="image" alt="Wang C"
      src="/blog/images/banner.png">
  <p class="site-author-name" itemprop="name">Wang C</p>
  <div class="site-description" itemprop="description">Make more time</div>
</div>
<div class="site-state-wrap motion-element">
  <nav class="site-state">
      <div class="site-state-item site-state-posts">
          <a href="/blog/archives">
          <span class="site-state-item-count">32</span>
          <span class="site-state-item-name">posts</span>
        </a>
      </div>
      <div class="site-state-item site-state-categories">
            <a href="/blog/categories/">
          
        <span class="site-state-item-count">11</span>
        <span class="site-state-item-name">categories</span></a>
      </div>
      <div class="site-state-item site-state-tags">
            <a href="/blog/tags/">
          
        <span class="site-state-item-count">18</span>
        <span class="site-state-item-name">tags</span></a>
      </div>
  </nav>
</div>
  <div class="links-of-author motion-element">
      <span class="links-of-author-item">
        <a href="https://github.com/wangc1106" title="GitHub â†’ https:&#x2F;&#x2F;github.com&#x2F;wangc1106" rel="noopener" target="_blank"><i class="fab fa-github fa-fw"></i>GitHub</a>
      </span>
      <span class="links-of-author-item">
        <a href="mailto:wangchen116@126.com" title="E-Mail â†’ mailto:wangchen116@126.com" rel="noopener" target="_blank"><i class="fa fa-envelope fa-fw"></i>E-Mail</a>
      </span>
  </div>
  <div class="cc-license motion-element" itemprop="license">
    <a href="https://creativecommons.org/licenses/by-nc-sa/4.0/" class="cc-opacity" rel="noopener" target="_blank"><img src="/blog/images/cc-by-nc-sa.svg" alt="Creative Commons"></a>
  </div>



      </div>
        <div class="back-to-top motion-element">
          <i class="fa fa-arrow-up"></i>
          <span>0%</span>
        </div>

    </div>
  </aside>
  <div id="sidebar-dimmer"></div>


      </div>
    </main>

    <footer class="footer">
      <div class="footer-inner">
        

        

<div class="copyright">
  
  &copy; 2021 â€“ 
  <span itemprop="copyrightYear">2022</span>
  <span class="with-love">
    <i class="fas fa-dna"></i>
  </span>
  <span class="author" itemprop="copyrightHolder">Wang C</span>
</div>

        






<script>
  (function() {
    function leancloudSelector(url) {
      url = encodeURI(url);
      return document.getElementById(url).querySelector('.leancloud-visitors-count');
    }

    function addCount(Counter) {
      var visitors = document.querySelector('.leancloud_visitors');
      var url = decodeURI(visitors.id);
      var title = visitors.dataset.flagTitle;

      Counter('get', '/classes/Counter?where=' + encodeURIComponent(JSON.stringify({ url })))
        .then(response => response.json())
        .then(({ results }) => {
          if (results.length > 0) {
            var counter = results[0];
            leancloudSelector(url).innerText = counter.time + 1;
            Counter('put', '/classes/Counter/' + counter.objectId, { time: { '__op': 'Increment', 'amount': 1 } })
              .catch(error => {
                console.error('Failed to save visitor count', error);
              });
          } else {
              Counter('post', '/classes/Counter', { title, url, time: 1 })
                .then(response => response.json())
                .then(() => {
                  leancloudSelector(url).innerText = 1;
                })
                .catch(error => {
                  console.error('Failed to create', error);
                });
          }
        })
        .catch(error => {
          console.error('LeanCloud Counter Error', error);
        });
    }

    function showTime(Counter) {
      var visitors = document.querySelectorAll('.leancloud_visitors');
      var entries = [...visitors].map(element => {
        return decodeURI(element.id);
      });

      Counter('get', '/classes/Counter?where=' + encodeURIComponent(JSON.stringify({ url: { '$in': entries } })))
        .then(response => response.json())
        .then(({ results }) => {
          for (let url of entries) {
            let target = results.find(item => item.url === url);
            leancloudSelector(url).innerText = target ? target.time : 0;
          }
        })
        .catch(error => {
          console.error('LeanCloud Counter Error', error);
        });
    }

    let { app_id, app_key, server_url } = {"enable":true,"app_id":"UqIYufmNk2sE8MiBKKWc4Kja-gzGzoHsz","app_key":"iYpeWQcmNqpHwr2uVlnJYVji","server_url":null,"security":false};
    function fetchData(api_server) {
      var Counter = (method, url, data) => {
        return fetch(`${api_server}/1.1${url}`, {
          method,
          headers: {
            'X-LC-Id'     : app_id,
            'X-LC-Key'    : app_key,
            'Content-Type': 'application/json',
          },
          body: JSON.stringify(data)
        });
      };
      if (CONFIG.page.isPost) {
        if (CONFIG.hostname !== location.hostname) return;
        addCount(Counter);
      } else if (document.querySelectorAll('.post-title-link').length >= 1) {
        showTime(Counter);
      }
    }

    let api_server = app_id.slice(-9) !== '-MdYXbMMI' ? server_url : `https://${app_id.slice(0, 8).toLowerCase()}.api.lncldglobal.com`;

    if (api_server) {
      fetchData(api_server);
    } else {
      fetch('https://app-router.leancloud.cn/2/route?appId=' + app_id)
        .then(response => response.json())
        .then(({ api_server }) => {
          fetchData('https://' + api_server);
        });
    }
  })();
</script>


      </div>
      <div style="height: 200px">
      <script type="text/javascript" src="//rf.revolvermaps.com/0/0/8.js?i=5knf34gs07t&amp;m=0&amp;c=ff0000&amp;cr1=ffffff&amp;f=arial&amp;l=33" async="async"></script>
      </div>
    </footer>
  </div>

  
  <script src="/blog/lib/anime.min.js"></script>
  <script src="/blog/lib/velocity/velocity.min.js"></script>
  <script src="/blog/lib/velocity/velocity.ui.min.js"></script>

<script src="/blog/js/utils.js"></script>

<script src="/blog/js/motion.js"></script>


<script src="/blog/js/schemes/pisces.js"></script>


<script src="/blog/js/next-boot.js"></script>




  




  
<script src="/blog/js/local-search.js"></script>













  

  

</body>
</html>
