<!DOCTYPE html>
<html lang="zh-Hans">
<head>
  <meta charset="UTF-8">
<meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=2">
<meta name="theme-color" content="#222">
<meta name="generator" content="Hexo 5.4.0">
  <link rel="apple-touch-icon" sizes="180x180" href="/blog/images/apple-touch-icon-next.png">
  <link rel="icon" type="image/png" sizes="32x32" href="/blog/images/favicon-32x32-next.png">
  <link rel="icon" type="image/png" sizes="16x16" href="/blog/images/favicon-16x16-next.png">
  <link rel="mask-icon" href="/blog/images/logo.svg" color="#222">
  <meta name="google-site-verification" content="1osTo_gOe1SW9Hevqtz8vl8HfSgTO4DSMZ0DiMPMX6w">

<link rel="stylesheet" href="/blog/css/main.css">


<link rel="stylesheet" href="/blog/lib/font-awesome/css/all.min.css">

<script id="hexo-configurations">
    var NexT = window.NexT || {};
    var CONFIG = {"hostname":"wangc1106.github.io","root":"/blog/","scheme":"Gemini","version":"7.8.0","exturl":false,"sidebar":{"position":"left","display":"post","padding":18,"offset":12,"onmobile":false},"copycode":{"enable":false,"show_result":false,"style":null},"back2top":{"enable":true,"sidebar":true,"scrollpercent":"ture"},"bookmark":{"enable":false,"color":"#222","save":"auto"},"fancybox":false,"mediumzoom":false,"lazyload":false,"pangu":false,"comments":{"style":"tabs","active":null,"storage":true,"lazyload":false,"nav":null},"algolia":{"hits":{"per_page":10},"labels":{"input_placeholder":"Search for Posts","hits_empty":"We didn't find any results for the search: ${query}","hits_stats":"${hits} results found in ${time} ms"}},"localsearch":{"enable":true,"trigger":"auto","top_n_per_article":1,"unescape":false,"preload":false},"motion":{"enable":true,"async":false,"transition":{"post_block":"fadeIn","post_header":"slideDownIn","post_body":"slideDownIn","coll_header":"slideLeftIn","sidebar":"slideUpIn"}},"path":"search.xml"};
  </script>

  <meta name="description" content="CUT&amp;Tag Data Processing and Analysis Tutorial Ye Zheng, Kami Ahmad, Steven Henikoff [following NC] https:&#x2F;&#x2F;yezhengstat.github.io&#x2F;CUTTag_tutorial&#x2F;#223_Intepret_the_quality_check_results https:&#x2F;&#x2F;www">
<meta property="og:type" content="article">
<meta property="og:title" content="【CUT&amp;Tag-pipeline】">
<meta property="og:url" content="https://wangc1106.github.io/blog/2022/03/21/2022-03-21-CUTTag/index.html">
<meta property="og:site_name" content="Banner&#39;s Blog">
<meta property="og:description" content="CUT&amp;Tag Data Processing and Analysis Tutorial Ye Zheng, Kami Ahmad, Steven Henikoff [following NC] https:&#x2F;&#x2F;yezhengstat.github.io&#x2F;CUTTag_tutorial&#x2F;#223_Intepret_the_quality_check_results https:&#x2F;&#x2F;www">
<meta property="og:locale">
<meta property="og:image" content="https://wangc1106.github.io/blog/2022/03/21/2022-03-21-CUTTag/1.png">
<meta property="og:image" content="https://wangc1106.github.io/blog/2022/03/21/2022-03-21-CUTTag/2.png">
<meta property="og:image" content="https://wangc1106.github.io/blog/2022/03/21/2022-03-21-CUTTag/3.png">
<meta property="og:image" content="https://wangc1106.github.io/blog/2022/03/21/2022-03-21-CUTTag/4.png">
<meta property="og:image" content="https://wangc1106.github.io/blog/2022/03/21/2022-03-21-CUTTag/5.png">
<meta property="og:image" content="https://wangc1106.github.io/blog/2022/03/21/2022-03-21-CUTTag/6.png">
<meta property="og:image" content="https://wangc1106.github.io/blog/2022/03/21/2022-03-21-CUTTag/7.png">
<meta property="og:image" content="https://wangc1106.github.io/blog/2022/03/21/2022-03-21-CUTTag/8.png">
<meta property="og:image" content="https://wangc1106.github.io/blog/2022/03/21/2022-03-21-CUTTag/9.png">
<meta property="og:image" content="https://wangc1106.github.io/blog/2022/03/21/2022-03-21-CUTTag/10.png">
<meta property="article:published_time" content="2022-03-21T05:58:54.780Z">
<meta property="article:modified_time" content="2022-03-21T15:45:40.167Z">
<meta property="article:author" content="Wang C">
<meta property="article:tag" content="pipeline">
<meta name="twitter:card" content="summary">
<meta name="twitter:image" content="https://wangc1106.github.io/blog/2022/03/21/2022-03-21-CUTTag/1.png">

<link rel="canonical" href="https://wangc1106.github.io/blog/2022/03/21/2022-03-21-CUTTag/">


<script id="page-configurations">
  // https://hexo.io/docs/variables.html
  CONFIG.page = {
    sidebar: "",
    isHome : false,
    isPost : true,
    lang   : 'zh-Hans'
  };
</script>

  <title>【CUT&Tag-pipeline】 | Banner's Blog</title>
  






  <noscript>
  <style>
  .use-motion .brand,
  .use-motion .menu-item,
  .sidebar-inner,
  .use-motion .post-block,
  .use-motion .pagination,
  .use-motion .comments,
  .use-motion .post-header,
  .use-motion .post-body,
  .use-motion .collection-header { opacity: initial; }

  .use-motion .site-title,
  .use-motion .site-subtitle {
    opacity: initial;
    top: initial;
  }

  .use-motion .logo-line-before i { left: initial; }
  .use-motion .logo-line-after i { right: initial; }
  </style>
</noscript>

</head>

<body itemscope itemtype="http://schema.org/WebPage">
  <div class="container use-motion">
    <div class="headband"></div>

    <header class="header" itemscope itemtype="http://schema.org/WPHeader">
      <div class="header-inner"><div class="site-brand-container">
  <div class="site-nav-toggle">
    <div class="toggle" aria-label="Toggle navigation bar">
      <span class="toggle-line toggle-line-first"></span>
      <span class="toggle-line toggle-line-middle"></span>
      <span class="toggle-line toggle-line-last"></span>
    </div>
  </div>

  <div class="site-meta">

    <a href="/blog/" class="brand" rel="start">
      <span class="logo-line-before"><i></i></span>
      <h1 class="site-title">Banner's Blog</h1>
      <span class="logo-line-after"><i></i></span>
    </a>
      <p class="site-subtitle" itemprop="description">Dream big work hard</p>
  </div>

  <div class="site-nav-right">
    <div class="toggle popup-trigger">
        <i class="fa fa-search fa-fw fa-lg"></i>
    </div>
  </div>
</div>




<nav class="site-nav">
  <ul id="menu" class="main-menu menu">
        <li class="menu-item menu-item-home">

    <a href="/blog/" rel="section"><i class="home fa-fw"></i>Home</a>

  </li>
        <li class="menu-item menu-item-categories">

    <a href="/blog/categories/" rel="section"><i class="th fa-fw"></i>Categories</a>

  </li>
        <li class="menu-item menu-item-tags">

    <a href="/blog/tags/" rel="section"><i class="tags fa-fw"></i>Tags</a>

  </li>
        <li class="menu-item menu-item-about">

    <a href="https://wangc1106.github.io/" rel="section"><i class="user fa-fw"></i>About</a>

  </li>
      <li class="menu-item menu-item-search">
        <a role="button" class="popup-trigger"><i class="fa fa-search fa-fw"></i>Search
        </a>
      </li>
  </ul>
</nav>



  <div class="search-pop-overlay">
    <div class="popup search-popup">
        <div class="search-header">
  <span class="search-icon">
    <i class="fa fa-search"></i>
  </span>
  <div class="search-input-container">
    <input autocomplete="off" autocapitalize="off"
           placeholder="Searching..." spellcheck="false"
           type="search" class="search-input">
  </div>
  <span class="popup-btn-close">
    <i class="fa fa-times-circle"></i>
  </span>
</div>
<div id="search-result">
  <div id="no-result">
    <i class="fa fa-spinner fa-pulse fa-5x fa-fw"></i>
  </div>
</div>

    </div>
  </div>

</div>
    </header>

    


    <main class="main">
      <div class="main-inner">
        <div class="content-wrap">
          

          <div class="content post posts-expand">
            

    
  
  
  <article itemscope itemtype="http://schema.org/Article" class="post-block" lang="zh-Hans">
    <link itemprop="mainEntityOfPage" href="https://wangc1106.github.io/blog/2022/03/21/2022-03-21-CUTTag/">

    <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
      <meta itemprop="image" content="/blog/images/banner.png">
      <meta itemprop="name" content="Wang C">
      <meta itemprop="description" content="Make more time">
    </span>

    <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
      <meta itemprop="name" content="Banner's Blog">
    </span>
      <header class="post-header">
        <h1 class="post-title" itemprop="name headline">
          【CUT&Tag-pipeline】
        </h1>

        <div class="post-meta">
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-calendar"></i>
              </span>
              <span class="post-meta-item-text">Posted on</span>
              

              <time title="Created: 2022-03-21 13:58:54 / Modified: 23:45:40" itemprop="dateCreated datePublished" datetime="2022-03-21T13:58:54+08:00">2022-03-21</time>
            </span>
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-folder"></i>
              </span>
              <span class="post-meta-item-text">In</span>
                <span itemprop="about" itemscope itemtype="http://schema.org/Thing">
                  <a href="/blog/categories/CUT-Tag/" itemprop="url" rel="index"><span itemprop="name">CUT&Tag</span></a>
                </span>
            </span>

          
            <span id="/blog/2022/03/21/2022-03-21-CUTTag/" class="post-meta-item leancloud_visitors" data-flag-title="【CUT&Tag-pipeline】" title="Views">
              <span class="post-meta-item-icon">
                <i class="fa fa-eye"></i>
              </span>
              <span class="post-meta-item-text">Views: </span>
              <span class="leancloud-visitors-count"></span>
            </span>
            <span class="post-meta-item" title="Symbols count in article">
              <span class="post-meta-item-icon">
                <i class="far fa-file-word"></i>
              </span>
              <span>46k</span>
            </span>
            <span class="post-meta-item" title="Reading time">
              <span class="post-meta-item-icon">
                <i class="far fa-clock"></i>
              </span>
              <span>42 mins.</span>
            </span>

        </div>
      </header>

    
    
    
    <div class="post-body" itemprop="articleBody">

      
        <p><strong>CUT&amp;Tag Data Processing and Analysis Tutorial</strong></p>
<p>Ye Zheng, Kami Ahmad, Steven Henikoff [following NC]</p>
<p><a target="_blank" rel="noopener" href="https://yezhengstat.github.io/CUTTag_tutorial/#223_Intepret_the_quality_check_results">https://yezhengstat.github.io/CUTTag_tutorial/#223_Intepret_the_quality_check_results</a></p>
<p><a target="_blank" rel="noopener" href="https://www.jianshu.com/p/1a23656a0713">https://www.jianshu.com/p/1a23656a0713</a></p>
<p><a target="_blank" rel="noopener" href="https://mp.weixin.qq.com/s?__biz=MzUxNTkzODIzMg==&mid=2247484049&idx=2&sn=b74aa66a9c9e4bd36a8a1d50e2834226&chksm=f9ae4046ced9c950efee443e1f0692e37dbce2b856e4f2f841ecd723affcff9c67c698ebe2fa&scene=21#wechat_redirect">wechat</a></p>
<span id="more"></span>

<hr>
<h1 id="I-Introduction"><a href="#I-Introduction" class="headerlink" title="I. Introduction"></a>I. Introduction</h1><h2 id="1-1-Overview-of-CUT-amp-Tag"><a href="#1-1-Overview-of-CUT-amp-Tag" class="headerlink" title="1.1. Overview of CUT&amp;Tag"></a>1.1. Overview of CUT&amp;Tag</h2><p>All dynamic processes that take place on DNA in the eukaryotic nucleus occur in the context of a chromatin landscape that comprises nucleosomes and their modifications, transcription factors, and chromatin-associated complexes. A variety of chromatin features mark sites of activating and silencing transcriptional regulatory elements and chromatin domains that differ between cell types and change during development.</p>
<p>The mapping of chromatin features genome-wide has traditionally been performed using chromatin immunoprecipitation (ChIP), in which chromatin is <strong>cross-linked</strong> and <strong>solubilized</strong>, and an <strong>antibody</strong> to a <strong>protein or modification of interest</strong> is used to <strong>immunoprecipitate the bound DNA</strong> (<font color="green"><strong>Fig. 1a</strong></font>). Very little has changed in the way <strong>ChIP</strong> is most generally performed since it was first described <strong>35 years ago</strong>, and remains fraught with <strong>signal-to-noise issues</strong> and <strong>artifacts</strong>. An alternative chromatin profiling strategy is <strong>enzyme tethering in situ</strong> whereby the chromatin protein or modification of interest is targeted by an <strong>antibody or fusion protein</strong>. Then, the underlying DNA is marked or cleaved, and a succession of enzyme-tethering methods have been introduced over the past two decades. <font color="green"><strong>Cleavage Under Targets &amp; Tagmentation (CUT&amp;Tag)</strong></font> is a tethering method that uses a <font color="green"><strong>protein-A-Tn5 (pA-Tn5) transposome fusion protein</strong></font> (<font color="green"><strong>Fig. 1b</strong></font>). In CUT&amp;Tag, permeabilized cells or nuclei are incubated with antibody to a specified chromatin protein, and then <strong>pA-Tn5 loaded with mosaic end adaptors</strong> is successively tethered to antibody-bound sites. Activation of the transposome by adding magnesium ions[<strong>Mg2+</strong>] results in the <strong>integration of the adaptors into nearby DNA</strong>. These are then amplified to generate sequencing libraries. Antibody-tethered Tn5-based methods achieve high sensitivity owing to stringent washing of samples after pA-Tn5 tethering and the high efficiency of adaptor integration. The <font color="green"><strong>improved signal-to-noise</strong></font> relative to ChIP-seq translates to an order-of-magnitude reduction in the amount of sequencing required to map chromatin features, allowing sample pooling (typically up to 90 samples) for paired-end sequencing on Illumina NGS sequencers by barcoded PCR of libraries.</p>
<p><img src="/blog/2022/03/21/2022-03-21-CUTTag/1.png" alt="Figure 1. Differences between immunoprecipitation and in antibody-targeted chromatin profiling strategies. A. ChIP-seq experimental procedure. B. CUT&amp;Tag experimental procedure. Cells and nuclei are indicated in grey, chromatin as red nucleosomes, and a specific chromatin protein in green."></p>
<p>Cells and nuclei are indicated in grey, chromatin as red nucleosomes, and a specific chromatin protein in green.</p>
<h2 id="1-2-Objectives"><a href="#1-2-Objectives" class="headerlink" title="1.2. Objectives"></a>1.2. Objectives</h2><p>This tutorial is designed for processing and analyzing CUT&amp;Tag data following the <a target="_blank" rel="noopener" href="https://www.protocols.io/view/bench-top-cut-amp-tag-bcuhiwt6/abstract">Bench top CUT&amp;Tag V.3 protocol</a> <strong>NC</strong>. The illustration data used in <strong>this tutorial</strong> is the profiling of <strong>histone modifications</strong> in the human lymphoma K562 cell line, but the tutorial is <strong>generally applicable to</strong> any chromatin protein, including <strong>transcription factors, RNA polymerase II</strong>, and epitope-tagged proteins.</p>
<h2 id="1-3-CUT-amp-Tag-data-processing-and-analysis-outline"><a href="#1-3-CUT-amp-Tag-data-processing-and-analysis-outline" class="headerlink" title="1.3 CUT&amp;Tag data processing and analysis outline."></a>1.3 CUT&amp;Tag data processing and analysis outline.</h2><p><img src="/blog/2022/03/21/2022-03-21-CUTTag/2.png" alt="Figure 2. CUT&amp;Tag data processing and analysis."></p>
<ol start="3">
<li><strong>Alignment</strong>: Bowtie2 alignment; Duplicates; Fragment size; Replicate reproducibility</li>
</ol>
<blockquote>
<p>Bowtie2, Samtools, Picard</p>
</blockquote>
<ol start="4">
<li><strong>Filtering and Conversion</strong>: Sam to bam; bam to bed</li>
</ol>
<blockquote>
<p>Samtools, Bedtools</p>
</blockquote>
<ol start="5">
<li><strong>Spike-in Calibration?:</strong> Sam to bam; Bam to bed</li>
</ol>
<blockquote>
<p>Bowtie2, Bedtools</p>
</blockquote>
<ol start="6">
<li><strong>Peak Calling:</strong> SEACR peak calling; Fragment propotion in peak regions(FRiPs)</li>
</ol>
<blockquote>
<p>SEACR, R: chromVAR</p>
</blockquote>
<ol start="7">
<li>Visualization: Genome browser; Heatmap</li>
</ol>
<blockquote>
<p>IGV, UCSC browser, Deeptools</p>
</blockquote>
<ol start="8">
<li>Different Analysis</li>
<li>Additional alternatives</li>
</ol>
<h2 id="1-4-Requirements"><a href="#1-4-Requirements" class="headerlink" title="1.4. Requirements"></a>1.4. Requirements</h2><ul>
<li>R (versions &gt;= 3.6)<ul>
<li>dplyr</li>
<li>stringr</li>
<li>ggplot2</li>
<li>viridis</li>
<li>GenomicRanges</li>
<li><strong>chromVAR</strong></li>
<li>DESeq2</li>
<li><strong>ggpubr</strong></li>
<li><strong>corrplot</strong></li>
<li>ChIPseqSpikeInFree [Optional]</li>
</ul>
</li>
</ul>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br></pre></td><td class="code"><pre><span class="line">library(dplyr)</span><br><span class="line">library(stringr)</span><br><span class="line">library(ggplot2)</span><br><span class="line">library(viridis)</span><br><span class="line">library(GenomicRanges)</span><br><span class="line">library(chromVAR) <span class="comment">## For FRiP analysis and differential analysis</span></span><br><span class="line">library(DESeq2) <span class="comment">## For differential analysis section</span></span><br><span class="line">library(ggpubr) <span class="comment">## For customizing figures</span></span><br><span class="line">library(corrplot) <span class="comment">## For correlation plot</span></span><br></pre></td></tr></table></figure>

<ul>
<li>FastQC(version &gt;= 0.11.9)</li>
<li>Bowtie2 (version &gt;= 2.3.4.3)</li>
<li>samtools (version &gt;= 1.10)</li>
<li>bedtools (version &gt;= 2.29.1)</li>
<li>Picard (version &gt;= 2.18.29)</li>
<li><strong>SEACR (version &gt;= 1.3)</strong></li>
<li>deepTools (version &gt;= 2.0)</li>
</ul>
<h2 id="1-5-Data-Downloading"><a href="#1-5-Data-Downloading" class="headerlink" title="1.5. Data Downloading"></a>1.5. Data Downloading</h2><ol>
<li>Using <a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/sra/docs/sradownload/">SRA Toolkit</a></li>
<li>Download through <a href="file:///Users/yezheng/Downloads/downloading_fastq_GEO.pdf">European Nucleotide Archive</a>. New ENA Browser: <a target="_blank" rel="noopener" href="https://www.ebi.ac.uk/ena/browser/view">https://www.ebi.ac.uk/ena/browser/view</a>. We are using this option as illustration.</li>
</ol>
<ul>
<li><strong>H3K27me3</strong>:<ul>
<li>SH_Hs_K27m3_NX_0918 as replicate 1: GEO accession: GSE145187, SRA entry: SRX8754646</li>
<li>SH_Hs_K27m3_Xpc_0107 as replicate 2: GEO accession: GSE145187, SRA entry: SRX7713678</li>
</ul>
</li>
<li><strong>H3K4me3</strong>:<ul>
<li>SH_Hs_K4m3_NX_0918 as replicate 1: GEO accession: GSE145187, SRA entry: SRX7713692</li>
<li>SH_Hs_K4m3_Xpc_0107 as replicate 2: GEO accession: GSE145187, SRA entry: SRX7713696</li>
</ul>
</li>
<li><strong>IgG</strong>:<ul>
<li>SH_Hs_IgG_1x_0924 as replicate 1:GEO accession: GSE145187, SRA entry: SRX8468909</li>
<li>SH_Hs_IgG_20181224 as replicate 2: GEO accession: GSM3680227, SRA entry: SRX5545346</li>
</ul>
</li>
</ul>
<figure class="highlight sh"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#example</span></span><br><span class="line">wget -O <span class="variable">$projPath</span>/data/IgG_rep2/IgG_rep2_R1_001.fastq.gz ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR875/001/SRR8754611/SRR8754611_1.fastq.gz</span><br><span class="line">wget -O <span class="variable">$projPath</span>/data/IgG_rep2/IgG_rep2_R2_001.fastq.gz ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR875/001/SRR8754611/SRR8754611_2.fastq.gz</span><br></pre></td></tr></table></figure>



<hr>
<h1 id="II-Data-Pre-processing"><a href="#II-Data-Pre-processing" class="headerlink" title="II. Data Pre-processing"></a>II. Data Pre-processing</h1><p>FASTQC</p>
<p><strong>Per base sequence content fails the FastQC quality check.</strong></p>
<p><strong>The discordant sequence content at the begining of the reads are common phenomenon for CUT&amp;Tag reads. Failing to pass the Per base seuqnence content does not mean your data failed.</strong></p>
<ul>
<li>It can be due to the Tn5 preference.</li>
<li>What you might be detecting is <font color="green">the 10-bp</font> periodicity that shows up as a sawtooth pattern in the length distribution. If so, <font color="blue"><strong>this is normal and will not affect alignment or peak calling</strong></font>. In any case we do not recommend trimming as the bowtie2 parameters that we list <font color="blue"><strong>will give accurate mapping information without trimming</strong></font>.</li>
</ul>
<h2 id="2-2-Merge-technical-replicates-lanes-if-needed-Optional"><a href="#2-2-Merge-technical-replicates-lanes-if-needed-Optional" class="headerlink" title="2.2. Merge technical replicates/lanes if needed [Optional]"></a>2.2. Merge technical replicates/lanes if needed [Optional]</h2><p>Sometimes, samples are often sequenced across <strong>multiple lanes for efficiency</strong> and can be pooled before alignment. <font color="blue">If you want to <strong>check the reproducibility</strong> between sequences of different lanes of the same sample</font>, you can skip this step and <strong>align</strong> each sequencing file (fastq file) <strong>respectively</strong>.</p>
<figure class="highlight sh"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#combination of the data</span></span><br><span class="line">cat <span class="variable">$&#123;projPath&#125;</span>/data/<span class="variable">$&#123;histName&#125;</span>/*_R1_*.fastq.gz &gt;<span class="variable">$&#123;projPath&#125;</span>/fastq/<span class="variable">$&#123;histName&#125;</span>_R1.fastq.gz</span><br></pre></td></tr></table></figure>

<hr>
<h1 id="III-Alignment"><a href="#III-Alignment" class="headerlink" title="III. Alignment"></a>III. Alignment</h1><h2 id="3-1-Bowtie2-alignment-required"><a href="#3-1-Bowtie2-alignment-required" class="headerlink" title="3.1. Bowtie2 alignment [required]"></a>3.1. Bowtie2 alignment [required]</h2><p>The structure of <font color="blue"><strong>CUT&amp;Tag insert libraries</strong></font> with <font color="blue"><strong>Tn5 adapters</strong></font> and <font color="blue"><strong>barcoded PCR primers</strong></font> is shown below:</p>
<p><img src="/blog/2022/03/21/2022-03-21-CUTTag/3.png" alt="Figure 4. CUT&amp;Tag insert libraries with the sequence of adapters."></p>
<p>Our standard pipeline is to perform <font color="blue">single-index 25x25 PE Illumina sequencing??</font> on up to 90 pooled samples on a single HiSeq 2500 flowcell, where <font color="blue">each sample</font> has a <font color="blue">unique PCR primer barcode</font>. Amounts for <strong>each library</strong> are adjusted to provide <strong>~5 million</strong> paired-end reads, which provides high-quality profiling for abundant chromatin features with a specific and high-yield antibody. Less abundant features typically require fewer reads, while <strong>lower-quality antibodies</strong> may increase the number of reads needed for generating robust chromatin profiles. A thorough discussion of feature recall and <strong>sequencing depths for CUT&amp;Tag</strong> has been published (Kaya-Okur et al 2020).</p>
<h3 id="3-1-1-Alignment-to-ref"><a href="#3-1-1-Alignment-to-ref" class="headerlink" title="3.1.1 Alignment to ref."></a>3.1.1 Alignment to <font color="red">ref</font>.</h3><figure class="highlight sh"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br></pre></td><td class="code"><pre><span class="line">bowtie2 \</span><br><span class="line">--end-to-end \</span><br><span class="line">--very-sensitive \</span><br><span class="line"><span class="comment">#runs a little faster, only consider alignment status of pairs per se</span></span><br><span class="line">--no-mixed \</span><br><span class="line"><span class="comment">#disablelooks for discordant alignments if it cannot find any concordant alignments. </span></span><br><span class="line">--no-discordant \</span><br><span class="line">--phred33 \</span><br><span class="line"><span class="comment">#-I The minimum fragment length for valid paired-end alignments. Default: 0</span></span><br><span class="line">-I 10 \</span><br><span class="line">-X 700 \</span><br><span class="line">-p <span class="variable">$&#123;cores&#125;</span> \</span><br><span class="line">-x <span class="variable">$&#123;ref&#125;</span> \</span><br><span class="line">-1 <span class="variable">$&#123;projPath&#125;</span>/fastq/<span class="variable">$&#123;histName&#125;</span>_R1.fastq.gz -2 <span class="variable">$&#123;projPath&#125;</span>/fastq/<span class="variable">$&#123;histName&#125;</span>_R2.fastq.gz \</span><br><span class="line">-S <span class="variable">$&#123;projPath&#125;</span>/alignment/sam/<span class="variable">$&#123;histName&#125;</span>_bowtie2.sam &amp;&gt; <span class="variable">$&#123;projPath&#125;</span>/alignment/sam/bowtie2_summary/<span class="variable">$&#123;histName&#125;</span>_bowtie2.txt</span><br></pre></td></tr></table></figure>

<p><code>--end-to-end --very-sensitive --no-mixed --no-discordant --phred33 -I 10 -X 700</code> for mapping of <strong>inserts 10-700 bp</strong> in length.</p>
<p><font color="red"><strong>Critical step</strong></font>: There is no need to trim reads from out standard 25x25 PE sequencing?, as adapter sequences will not be included in reads of inserts &gt;25 bp. However, for users <strong>performing longer sequencing</strong>, reads will need to be trimmed by Cutadapt and mapped by <code>--local --very-sensitive --no-mixed --no-discordant --phred33 -I 10 -X 700</code> to ignore any remaining adapter sequence at the 3’ ends of reads during mapping.</p>
<p><font color="red">–local</font> In this mode, Bowtie 2 does not require that the entire read align from one end to the other. Rather, some characters may be omitted (“soft clipped”) from the ends <strong>in order to achieve the greatest possible alignment score</strong>. The match bonus <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-options-ma"><code>--ma</code></a>is used in this mode, and the best possible alignment score is equal to the match bonus (<a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-options-ma"><code>--ma</code></a>) times the length of the read. Specifying <code>--local</code> and one of the presets (e.g. <code>--local --very-fast</code>) is equivalent to specifying the local version of the preset (<code>--very-fast-local</code>). This is mutually exclusive with <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-options-end-to-end"><code>--end-to-end</code></a>. <code>--end-to-end</code> is the default mode.</p>
<h3 id="3-1-2-Alignment-to-spike-in-genome-for-spike-in-calibration-optional-recommended"><a href="#3-1-2-Alignment-to-spike-in-genome-for-spike-in-calibration-optional-recommended" class="headerlink" title="3.1.2 Alignment to spike-in genome for spike-in calibration [optional/recommended]"></a>3.1.2 Alignment to <font color="red">spike-in genome</font> for spike-in calibration [optional/recommended]</h3><p>This section is <strong>optional</strong> but <strong>recommended</strong> depending on your experimental protocol.</p>
<p><font color="red"><strong>E. coli</strong> DNA is <strong>carried along with</strong> bacterially-produced <strong>pA-Tn5 protein</strong></font> and gets tagmented non-specifically during the reaction. <strong>The fraction of total reads that map to the E.coli genome</strong> <font color="blue"><strong>depends on</strong></font> the <strong>yield of epitope-targeted CUT&amp;Tag</strong>, and so <font color="blue"><strong>depends on</strong></font> the <strong>number of cells used and the abundance of that epitope in chromatin</strong>. Since a constant amount of pATn5 is added to CUT&amp;Tag reactions and brings along a fixed amount of E. coli DNA, E. coli reads can be used to <font color="red"><strong>normalize epitope abundance</strong></font> in a set of experiments. More discussion, please see Section V.</p>
<figure class="highlight sh"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># download E.coli K12 MG1655</span></span><br><span class="line">bowtie2-build ecoli.fa ecoli</span><br></pre></td></tr></table></figure>

<figure class="highlight sh"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">##== linux command ==##</span></span><br><span class="line">spikeInRef=/path/to/index/eoli</span><br><span class="line">chromSize=<span class="string">&quot;hg38.chrom.size&quot;</span></span><br><span class="line"><span class="comment"># download ecoli K-12 substr.MG1655[INSDC:U00096.3;size:4.64M;]</span></span><br><span class="line">bowtie2-build path/to/Ecoli/fasta/Ecoli.fa /path/to/bowtie2Index/Ecoli</span><br><span class="line">bowtie2 --end-to-end --very-sensitive --no-overlap --no-dovetail --no-mixed --no-discordant --phred33 -I 10 -X 700 -p <span class="variable">$&#123;cores&#125;</span> -x <span class="variable">$&#123;spikeInRef&#125;</span> -1 _R1.fastq.gz -2 _R2.fastq.gz -S _bowtie2_spikeIn.sam &amp;&gt; log_bowtie2_spikeIn.txt</span><br><span class="line"></span><br><span class="line"><span class="comment">#-f output alignments with bits</span></span><br><span class="line"><span class="comment">#-F Do not output alignments with any bits set in INT present in the FLAG field. INT can be specified in hex by beginning with 0x&#x27; (i.e. /^0x[0-9A-F]+/) or in octal by beginning with0&#x27; (i.e./^0 [0-7]+/) </span></span><br><span class="line"><span class="comment"># FLAGS 0x2 PROPER_PAIR each segment properly aligned according to the aligner</span></span><br><span class="line"><span class="comment"># FLAGS 0x4 UNMAP segment unmapped</span></span><br><span class="line">seqDepthDouble=`samtools view -F 0x04 _bowtie2_spikeIn.sam | wc -l`</span><br><span class="line">seqDepth=$((seqDepthDouble/<span class="number">2</span>))</span><br><span class="line"><span class="built_in">echo</span> <span class="variable">$seqDepth</span> &gt; _bowtie2_spikeIn.seqDepth</span><br></pre></td></tr></table></figure>

<ul>
<li>For spike-in normalization, reads are aligned to the <strong>E. coli genome U00096.3</strong> with two more parameters <code>--no-overlap</code>and <code>--no-dovetail</code>(<code>--end-to-end --very-sensitive --no-overlap --no-dovetail --no-mixed --no-discordant --phred33 -I 10 -X 700</code>) to avoid possible cross-mapping of the experimental genome to that of the carry-over E. coli DNA that is used for calibration.</li>
</ul>
<h3 id="3-1-3-Alignment-summary"><a href="#3-1-3-Alignment-summary" class="headerlink" title="3.1.3 Alignment summary"></a>3.1.3 Alignment summary</h3><p>For more detailed parameters explanation, users can refer to the <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml">bowite2 manual</a>.</p>
<p>Bowtie2 alignment results summary is saved at <code>log</code> .</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line">2984630 reads; of these:</span><br><span class="line">  2984630 (100.00%) were paired; of these:</span><br><span class="line">    125110 (4.19%) aligned concordantly 0 times</span><br><span class="line">    2360430 (79.09%) aligned concordantly exactly 1 time</span><br><span class="line">    499090 (16.72%) aligned concordantly &gt;1 times</span><br><span class="line">95.81% overall alignment rate</span><br></pre></td></tr></table></figure>

<ul>
<li>2984640 is the <strong>sequencing depth?</strong>, i.e., total number of paired reads.</li>
<li>125110 is the number of read pairs that fail to be mapped.</li>
<li>2360430 + 499090 is the number of read paris that are successfully mapped.</li>
<li>95.81% is the overall alignment rate</li>
</ul>
<h2 id="3-2-Report-sequencing-mapping-summary-required"><a href="#3-2-Report-sequencing-mapping-summary-required" class="headerlink" title="3.2 Report sequencing mapping summary [required]"></a>3.2 Report sequencing mapping summary [required]</h2><p>Summarize the raw reads and uniquely mapping reads to report the efficiency of alignment. Alignment frequencies are expected to be &gt;80% for high-quality data. <strong>CUT&amp;Tag</strong> data typically has <font color="red"><strong>very low backgrounds</strong></font>, so as few as 1 million mapped fragments can give robust profiles for a histone modification in the human genome. Profiling of less-abundant transcription factors and chromatin proteins may require 10 times as many mapped fragments for downstream analysis.</p>
<p>We can evaluate the following metrics:</p>
<ul>
<li>Sequencing depth</li>
<li>Alignment rate</li>
<li>Number of mappable fragments</li>
<li>Duplication rate</li>
<li>Unique library size</li>
<li>Fragment size distribution</li>
</ul>
<h3 id="3-2-1-Sequencing-depth"><a href="#3-2-1-Sequencing-depth" class="headerlink" title="3.2.1 Sequencing depth"></a>3.2.1 Sequencing depth</h3><p>code in this part, <font color="red"><strong>no use</strong></font>. Summary all reads, mapped reads, and mapping ratio</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">##=== R command ===## </span></span><br><span class="line"><span class="comment">## Path to the project and histone list</span></span><br><span class="line">projPath = <span class="string">&quot;///CUTTag_tutorial&quot;</span></span><br><span class="line">sampleList = <span class="built_in">c</span>(<span class="string">&quot;K27me3_rep1&quot;</span>, <span class="string">&quot;K27me3_rep2&quot;</span>, <span class="string">&quot;K4me3_rep1&quot;</span>, <span class="string">&quot;K4me3_rep2&quot;</span>, <span class="string">&quot;IgG_rep1&quot;</span>, <span class="string">&quot;IgG_rep2&quot;</span>)</span><br><span class="line">histList = <span class="built_in">c</span>(<span class="string">&quot;K27me3&quot;</span>, <span class="string">&quot;K4me3&quot;</span>, <span class="string">&quot;IgG&quot;</span>)</span><br><span class="line"></span><br><span class="line"><span class="comment">## Collect the alignment results from the bowtie2 alignment summary files</span></span><br><span class="line">alignResult = <span class="built_in">c</span>()</span><br><span class="line"><span class="keyword">for</span>(hist <span class="keyword">in</span> sampleList)&#123;</span><br><span class="line">  alignRes = read.table(paste0(projPath, <span class="string">&quot;/alignment/sam/bowtie2_summary/&quot;</span>, hist, <span class="string">&quot;_bowtie2.txt&quot;</span>), header = <span class="literal">FALSE</span>, fill = <span class="literal">TRUE</span>)</span><br><span class="line">  alignRate = substr(alignRes$V1[<span class="number">6</span>], <span class="number">1</span>, nchar(<span class="built_in">as.character</span>(alignRes$V1[<span class="number">6</span>]))-<span class="number">1</span>)</span><br><span class="line">  histInfo = strsplit(hist, <span class="string">&quot;_&quot;</span>)[[<span class="number">1</span>]]</span><br><span class="line">  alignResult = data.frame(Histone = histInfo[<span class="number">1</span>], Replicate = histInfo[<span class="number">2</span>], </span><br><span class="line">                           SequencingDepth = alignRes$V1[<span class="number">1</span>] %&gt;% <span class="built_in">as.character</span> %&gt;% <span class="built_in">as.numeric</span>, </span><br><span class="line">                           MappedFragNum_hg38 = alignRes$V1[<span class="number">4</span>] %&gt;% <span class="built_in">as.character</span> %&gt;% <span class="built_in">as.numeric</span> + alignRes$V1[<span class="number">5</span>] %&gt;% <span class="built_in">as.character</span> %&gt;% <span class="built_in">as.numeric</span>, </span><br><span class="line">                           AlignmentRate_hg38 = alignRate %&gt;% <span class="built_in">as.numeric</span>)  %&gt;% rbind(alignResult, .)</span><br><span class="line">&#125;</span><br><span class="line">alignResult$Histone = factor(alignResult$Histone, levels = histList)</span><br><span class="line">alignResult %&gt;% mutate(AlignmentRate_hg38 = paste0(AlignmentRate_hg38, <span class="string">&quot;%&quot;</span>))</span><br></pre></td></tr></table></figure>

<h3 id="3-2-2-Spike-in-alignment"><a href="#3-2-2-Spike-in-alignment" class="headerlink" title="3.2.2 Spike-in alignment"></a>3.2.2 Spike-in alignment</h3><p>Summary of bowtie2 log file</p>
<h3 id="3-2-3-Summarize-the-alignment-to-hg38-and-E-coli"><a href="#3-2-3-Summarize-the-alignment-to-hg38-and-E-coli" class="headerlink" title="3.2.3 Summarize the alignment to hg38 and E.coli"></a>3.2.3 Summarize the alignment to hg38 and E.coli</h3><p>Summary of bowtie2 log file(ref and ecoli)</p>
<h3 id="3-2-4-Visualizing-the-sequencing-depth-and-alignment-results"><a href="#3-2-4-Visualizing-the-sequencing-depth-and-alignment-results" class="headerlink" title="3.2.4 Visualizing the sequencing depth and alignment results"></a>3.2.4 Visualizing the sequencing depth and alignment results</h3><p><font color="red"><strong>No use</strong></font> . barplot learning</p>
<figure class="highlight sh"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">## Generate sequencing depth boxplot</span></span><br><span class="line">fig3A = alignResult %&gt;% ggplot(aes(x = Histone, y = SequencingDepth/1000000, fill = Histone)) +</span><br><span class="line">    geom_boxplot() +</span><br><span class="line">    geom_jitter(aes(color = Replicate), position = position_jitter(0.15)) +</span><br><span class="line">    scale_fill_viridis(discrete = TRUE, begin = 0.1, end = 0.9, option = <span class="string">&quot;magma&quot;</span>, alpha = 0.8) +</span><br><span class="line">    scale_color_viridis(discrete = TRUE, begin = 0.1, end = 0.9) +</span><br><span class="line">    theme_bw(base_size = 18) +</span><br><span class="line">    ylab(<span class="string">&quot;Sequencing Depth per Million&quot;</span>) +</span><br><span class="line">    xlab(<span class="string">&quot;&quot;</span>) + </span><br><span class="line">    ggtitle(<span class="string">&quot;A. Sequencing Depth&quot;</span>)</span><br></pre></td></tr></table></figure>

<p>In a typical <strong>CUT&amp;Tag experiment</strong> targeting the abundant <strong>H3K27me3</strong> histone modification in <strong>65,000</strong> <strong>K562 cells</strong>, the <font color="red"><strong>percentage</strong> of <strong>E. coli</strong> reads range from ~<strong>0.01% to 10%</strong></font>. With <font color="red"><strong>fewer cells</strong> or <strong>less abundant epitopes</strong></font>, E. coli reads can comprise[<strong>more</strong>] as much as 70% or the total mapped reads. For <font color="red"><strong>IgG</strong> controls</font>, the percentage of <strong>E. coli</strong> reads is typically <font color="red"><strong>much higher</strong></font> than that for an abundant histone modification.</p>
<h2 id="3-3-Remove-duplicates-optional-required"><a href="#3-3-Remove-duplicates-optional-required" class="headerlink" title="3.3. Remove duplicates [optional/required]"></a>3.3. Remove duplicates [optional/required]</h2><p>CUT&amp;Tag integrates adapters into DNA in the vicinity of the antibody-tethered pA-Tn5, and the exact sites of integration are affected by the accessibility of surrounding DNA. For this reason fragments that share exact starting and ending positions are expected to be common, and <font color="red">such ‘duplicates’ <strong>may not be due to duplication during PCR</strong></font>. In practice, we have found that the <font color="red">apparent <strong>duplication rate is low for high quality CUT&amp;Tag datasets</strong></font>, and even the apparent ‘duplicate’ fragments are <strong>likely to be true fragments</strong>. Thus, we <font color="red"><strong>do not recommend removing</strong> the duplicates</font>. In experiments with very small amounts of material or where PCR duplication is suspected, duplicates can be removed. The following commands show how to check the duplication rate using <a target="_blank" rel="noopener" href="https://broadinstitute.github.io/picard/">Picard</a>.</p>
<figure class="highlight sh"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">##== linux command ==##</span></span><br><span class="line"><span class="comment">## depending on how you load picard and your server environment, the picardCMD can be different. Adjust accordingly.</span></span><br><span class="line">picardCMD=<span class="string">&quot;java -jar picard.jar&quot;</span></span><br><span class="line">mkdir -p <span class="variable">$projPath</span>/alignment/removeDuplicate/picard_summary</span><br><span class="line"></span><br><span class="line"><span class="comment">## Sort by coordinate</span></span><br><span class="line"><span class="variable">$picardCMD</span> SortSam I=<span class="variable">$projPath</span>/alignment/sam/<span class="variable">$&#123;histName&#125;</span>_bowtie2.sam O=<span class="variable">$projPath</span>/alignment/sam/<span class="variable">$&#123;histName&#125;</span>_bowtie2.sorted.sam SORT_ORDER=coordinate</span><br><span class="line"></span><br><span class="line"><span class="comment">## mark duplicates</span></span><br><span class="line"><span class="variable">$picardCMD</span> MarkDuplicates I=<span class="variable">$projPath</span>/alignment/sam/<span class="variable">$&#123;histName&#125;</span>_bowtie2.sorted.sam O=<span class="variable">$projPath</span>/alignment/removeDuplicate/<span class="variable">$&#123;histName&#125;</span>_bowtie2.sorted.dupMarked.sam METRICS_FILE=<span class="variable">$projPath</span>/alignment/removeDuplicate/picard_summary/<span class="variable">$&#123;histName&#125;</span>_picard.dupMark.txt</span><br><span class="line"></span><br><span class="line"><span class="comment">## remove duplicates</span></span><br><span class="line">picardCMD MarkDuplicates I=<span class="variable">$projPath</span>/alignment/sam/<span class="variable">$&#123;histName&#125;</span>_bowtie2.sorted.sam O=<span class="variable">$projPath</span>/alignment/removeDuplicate/<span class="variable">$&#123;histName&#125;</span>_bowtie2.sorted.rmDup.sam REMOVE_DUPLICATES=<span class="literal">true</span> METRICS_FILE=<span class="variable">$projPath</span>/alignment/removeDuplicate/picard_summary/<span class="variable">$&#123;histName&#125;</span>_picard.rmDup.txt</span><br></pre></td></tr></table></figure>

<p>We summarize the apparent duplication rate and calculate the unique library size without duplicates.</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">##=== R command ===## </span></span><br><span class="line"><span class="comment">## Summarize the duplication information from the picard summary outputs.</span></span><br><span class="line">library(magrittr)</span><br><span class="line">libry(dplyr)</span><br><span class="line">libry)(idyverse</span><br><span class="line">sampleList = <span class="built_in">c</span>(<span class="string">&quot;K27me3_rep1&quot;</span>, <span class="string">&quot;K27me3_rep2&quot;</span>, <span class="string">&quot;K4me3_rep1&quot;</span>, <span class="string">&quot;K4me3_rep2&quot;</span>, <span class="string">&quot;IgG_rep1&quot;</span>, <span class="string">&quot;IgG_rep2&quot;</span>)</span><br><span class="line">histList = <span class="built_in">c</span>(<span class="string">&quot;K27me3&quot;</span>, <span class="string">&quot;K4me3&quot;</span>, <span class="string">&quot;IgG&quot;</span>)</span><br><span class="line"></span><br><span class="line">dupResult = <span class="built_in">c</span>()</span><br><span class="line"><span class="keyword">for</span>(hist <span class="keyword">in</span> sampleList)&#123;</span><br><span class="line">  dupRes = read.table(paste0(projPath, <span class="string">&quot;/alignment/removeDuplicate/picard_summary/&quot;</span>, hist, <span class="string">&quot;_picard.rmDup.txt&quot;</span>), header = <span class="literal">TRUE</span>, fill = <span class="literal">TRUE</span>)</span><br><span class="line">  </span><br><span class="line">  histInfo = strsplit(hist, <span class="string">&quot;_&quot;</span>)[[<span class="number">1</span>]]</span><br><span class="line">  dupResult = data.frame(Histone = histInfo[<span class="number">1</span>], Replicate = histInfo[<span class="number">2</span>], MappedFragNum_hg38 = dupRes$READ_PAIRS_EXAMINED[<span class="number">1</span>] %&gt;% <span class="built_in">as.character</span> %&gt;% <span class="built_in">as.numeric</span>, DuplicationRate = dupRes$PERCENT_DUPLICATION[<span class="number">1</span>] %&gt;% <span class="built_in">as.character</span> %&gt;% <span class="built_in">as.numeric</span> * <span class="number">100</span>, EstimatedLibrarySize = dupRes$ESTIMATED_LIBRARY_SIZE[<span class="number">1</span>] %&gt;% <span class="built_in">as.character</span> %&gt;% <span class="built_in">as.numeric</span>) %&gt;% mutate(UniqueFragNum = MappedFragNum_hg38 * (<span class="number">1</span>-DuplicationRate/<span class="number">100</span>))  %&gt;% rbind(dupResult, .)</span><br><span class="line">&#125;</span><br><span class="line">dupResult$Histone = factor(dupResult$Histone, levels = histList)</span><br><span class="line">alignDupSummary = left_join(alignSummary, dupResult, by = <span class="built_in">c</span>(<span class="string">&quot;Histone&quot;</span>, <span class="string">&quot;Replicate&quot;</span>, <span class="string">&quot;MappedFragNum_hg38&quot;</span>)) %&gt;% mutate(DuplicationRate = paste0(DuplicationRate, <span class="string">&quot;%&quot;</span>))</span><br><span class="line">alignDupSummary</span><br></pre></td></tr></table></figure>

<p>计算总reads，比对到ref和ecoli的frag和ratio，计算<strong>duplicationrate，estimatedlibrarysize和uniqueFragNum</strong></p>
<ul>
<li>In these example datasets, the <strong>IgG</strong> control samples have <strong>relatively high duplication rates</strong>, since reads in this sample derive from <strong>non-specific tagmentation</strong> in the CUT&amp;Tag reactions. Therefore, <font color="red">it is appropriate to <strong>remove the duplicates</strong> from the <strong>IgG</strong> datasets before downstream analysis</font>.</li>
<li>The <font color="red"><strong>estimated library size</strong></font> are the estimated <strong>number of unique molecules in the library based on PE duplication</strong> calculated by <strong>Picard</strong>.</li>
<li>The estimated library sizes is proportional to the <strong>abundance of the targeted epitope</strong> and to the <strong>quality of the antibody</strong> used, while the estimated library sizes of <font color="red">IgG</font> samples are expected to be <font color="red">very low</font>.</li>
<li><strong>Unique fragment number</strong> is calculated by the MappedFragNum_hg38 * (1-DuplicationRate/100).</li>
</ul>
<p><font color="red"><strong>no use</strong>, code</font></p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">##=== R command ===## </span></span><br><span class="line"><span class="comment">## generate boxplot figure for the  duplication rate</span></span><br><span class="line">fig4A = dupResult %&gt;% ggplot(aes(x = Histone, y = DuplicationRate, fill = Histone)) +</span><br><span class="line">    geom_boxplot() +</span><br><span class="line">    geom_jitter(aes(color = Replicate), position = position_jitter(<span class="number">0.15</span>)) +</span><br><span class="line">    scale_fill_viridis(discrete = <span class="literal">TRUE</span>, begin = <span class="number">0.1</span>, end = <span class="number">0.9</span>, option = <span class="string">&quot;magma&quot;</span>, alpha = <span class="number">0.8</span>) +</span><br><span class="line">    scale_color_viridis(discrete = <span class="literal">TRUE</span>, begin = <span class="number">0.1</span>, end = <span class="number">0.9</span>) +</span><br><span class="line">    theme_bw(base_size = <span class="number">18</span>) +</span><br><span class="line">    ylab(<span class="string">&quot;Duplication Rate (*100%)&quot;</span>) +</span><br><span class="line">    xlab(<span class="string">&quot;&quot;</span>) </span><br><span class="line"><span class="comment">## generate boxplot figure for the  Estimated Library Size</span></span><br><span class="line">fig4B = dupResult %&gt;% ggplot(aes(x = Histone, y = EstimatedLibrarySize, fill = Histone)) +</span><br><span class="line">    geom_boxplot() +</span><br><span class="line">    geom_jitter(aes(color = Replicate), position = position_jitter(<span class="number">0.15</span>)) +</span><br><span class="line">    scale_fill_viridis(discrete = <span class="literal">TRUE</span>, begin = <span class="number">0.1</span>, end = <span class="number">0.9</span>, option = <span class="string">&quot;magma&quot;</span>, alpha = <span class="number">0.8</span>) +</span><br><span class="line">    scale_color_viridis(discrete = <span class="literal">TRUE</span>, begin = <span class="number">0.1</span>, end = <span class="number">0.9</span>) +</span><br><span class="line">    theme_bw(base_size = <span class="number">18</span>) +</span><br><span class="line">    ylab(<span class="string">&quot;Estimated Library Size&quot;</span>) +</span><br><span class="line">    xlab(<span class="string">&quot;&quot;</span>) </span><br><span class="line"><span class="comment">## generate boxplot figure for the  Unique Fragments</span></span><br><span class="line">fig4C = dupResult %&gt;% ggplot(aes(x = Histone, y = UniqueFragNum, fill = Histone)) +</span><br><span class="line">    geom_boxplot() +</span><br><span class="line">    geom_jitter(aes(color = Replicate), position = position_jitter(<span class="number">0.15</span>)) +</span><br><span class="line">    scale_fill_viridis(discrete = <span class="literal">TRUE</span>, begin = <span class="number">0.1</span>, end = <span class="number">0.9</span>, option = <span class="string">&quot;magma&quot;</span>, alpha = <span class="number">0.8</span>) +</span><br><span class="line">    scale_color_viridis(discrete = <span class="literal">TRUE</span>, begin = <span class="number">0.1</span>, end = <span class="number">0.9</span>) +</span><br><span class="line">    theme_bw(base_size = <span class="number">18</span>) +</span><br><span class="line">    ylab(<span class="string">&quot;# of Unique Fragments&quot;</span>) +</span><br><span class="line">    xlab(<span class="string">&quot;&quot;</span>)</span><br><span class="line"></span><br><span class="line">ggarrange(fig4A, fig4B, fig4C, ncol = <span class="number">3</span>, common.legend = <span class="literal">TRUE</span>, legend=<span class="string">&quot;bottom&quot;</span>)</span><br></pre></td></tr></table></figure>

<h2 id="3-4-Assess-mapped-fragment-size-distribution-Required"><a href="#3-4-Assess-mapped-fragment-size-distribution-Required" class="headerlink" title="3.4. Assess mapped fragment size distribution [Required]"></a>3.4. Assess mapped fragment size distribution [Required]</h2><p>CUT&amp;Tag inserts adapters on either side of chromatin particles in the vicinity of the tethered enzyme, although tagmentation within chromatin particles can also occur. So, CUT&amp;Tag reactions targeting a histone modification predominantly results in fragments that are nucleosomal lengths (~180 bp), or multiples of that length. CUT&amp;Tag targeting transcription factors predominantly produce nucleosome-sized fragments and variable amounts of shorter fragments, from neighboring nucleosomes and the factor-bound site, respectively. Tagmentation of DNA on the surface of nucleosomes also occurs, and plotting fragment lengths with single-basepair resolution reveal a 10-bp sawtooth periodicity, which is typical of successful CUT&amp;Tag experiments.</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line">##== linux command ==##</span><br><span class="line">mkdir -p $projPath/alignment/sam/fragmentLen</span><br><span class="line"></span><br><span class="line">## Extract the 9th column from the alignment sam file which is the fragment length</span><br><span class="line">samtools view -F 0x04 $projPath/alignment/sam/$&#123;histName&#125;_bowtie2.sam | awk -F&#x27;\t&#x27; &#x27;function abs(x)&#123;return ((x &lt; 0.0) ? -x : x)&#125; &#123;print abs($9)&#125;&#x27; | sort | uniq -c | awk -v OFS=&quot;\t&quot; &#x27;&#123;print $2, $1/2&#125;&#x27; &gt;$projPath/alignment/sam/fragmentLen/$&#123;histName&#125;_fragmentLen.txt</span><br></pre></td></tr></table></figure>

<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">##=== R command ===## </span></span><br><span class="line"><span class="comment">## Collect the fragment size information</span></span><br><span class="line">fragLen = <span class="built_in">c</span>()</span><br><span class="line"><span class="keyword">for</span>(hist <span class="keyword">in</span> sampleList)&#123;</span><br><span class="line">  </span><br><span class="line">  histInfo = strsplit(hist, <span class="string">&quot;_&quot;</span>)[[<span class="number">1</span>]]</span><br><span class="line">  fragLen = read.table(paste0(projPath, <span class="string">&quot;/alignment/sam/fragmentLen/&quot;</span>, hist, <span class="string">&quot;_fragmentLen.txt&quot;</span>), header = <span class="literal">FALSE</span>) %&gt;% mutate(fragLen = V1 %&gt;% <span class="built_in">as.numeric</span>, fragCount = V2 %&gt;% <span class="built_in">as.numeric</span>, Weight = <span class="built_in">as.numeric</span>(V2)/<span class="built_in">sum</span>(<span class="built_in">as.numeric</span>(V2)), Histone = histInfo[<span class="number">1</span>], Replicate = histInfo[<span class="number">2</span>], sampleInfo = hist) %&gt;% rbind(fragLen, .) </span><br><span class="line">&#125;</span><br><span class="line">fragLen$sampleInfo = factor(fragLen$sampleInfo, levels = sampleList)</span><br><span class="line">fragLen$Histone = factor(fragLen$Histone, levels = histList)</span><br><span class="line"><span class="comment">## Generate the fragment size density plot (violin plot)</span></span><br><span class="line">fig5A = fragLen %&gt;% ggplot(aes(x = sampleInfo, y = fragLen, weight = Weight, fill = Histone)) +</span><br><span class="line">    geom_violin(bw = <span class="number">5</span>) +</span><br><span class="line">    scale_y_continuous(breaks = seq(<span class="number">0</span>, <span class="number">800</span>, <span class="number">50</span>)) +</span><br><span class="line">    scale_fill_viridis(discrete = <span class="literal">TRUE</span>, begin = <span class="number">0.1</span>, end = <span class="number">0.9</span>, option = <span class="string">&quot;magma&quot;</span>, alpha = <span class="number">0.8</span>) +</span><br><span class="line">    scale_color_viridis(discrete = <span class="literal">TRUE</span>, begin = <span class="number">0.1</span>, end = <span class="number">0.9</span>) +</span><br><span class="line">    theme_bw(base_size = <span class="number">20</span>) +</span><br><span class="line">    ggpubr::rotate_x_text(angle = <span class="number">20</span>) +</span><br><span class="line">    ylab(<span class="string">&quot;Fragment Length&quot;</span>) +</span><br><span class="line">    xlab(<span class="string">&quot;&quot;</span>)</span><br><span class="line"></span><br><span class="line">fig5B = fragLen %&gt;% ggplot(aes(x = fragLen, y = fragCount, color = Histone, group = sampleInfo, linetype = Replicate)) +</span><br><span class="line">  geom_line(size = <span class="number">1</span>) +</span><br><span class="line">  scale_color_viridis(discrete = <span class="literal">TRUE</span>, begin = <span class="number">0.1</span>, end = <span class="number">0.9</span>, option = <span class="string">&quot;magma&quot;</span>) +</span><br><span class="line">  theme_bw(base_size = <span class="number">20</span>) +</span><br><span class="line">  xlab(<span class="string">&quot;Fragment Length&quot;</span>) +</span><br><span class="line">  ylab(<span class="string">&quot;Count&quot;</span>) +</span><br><span class="line">  coord_cartesian(xlim = <span class="built_in">c</span>(<span class="number">0</span>, <span class="number">500</span>))</span><br><span class="line"></span><br><span class="line">ggarrange(fig5A, fig5B, ncol = <span class="number">2</span>)</span><br></pre></td></tr></table></figure>

<p><img src="/blog/2022/03/21/2022-03-21-CUTTag/4.png"></p>
<ul>
<li>The smaller fragments (50-100 bp) can be due to that tethered Tn5 can tagment on the surface of a nucleosome as well as in linker regions(NFR?), so the <strong>small fragments might not be background.</strong></li>
</ul>
<h2 id="3-5-Assess-replicate-reproducibility"><a href="#3-5-Assess-replicate-reproducibility" class="headerlink" title="3.5 Assess replicate reproducibility"></a>3.5 Assess replicate reproducibility</h2><p>Data reproducibility between replicates is assessed by <strong>correlation</strong> analysis of <strong>mapped read counts</strong> across the genome. For the simplicity of implementation, we will postpone this analysis after Section IV when the file format has been converted into fragment bed files.</p>
<hr>
<h1 id="IV-Alignments-results-filtering-and-file-format-conversion"><a href="#IV-Alignments-results-filtering-and-file-format-conversion" class="headerlink" title="IV. Alignments results filtering and file format conversion"></a>IV. Alignments results filtering and file format conversion</h1><h2 id="4-1-Filtering-mapped-reads-by-the-mapping-quality-filtering-optinal"><a href="#4-1-Filtering-mapped-reads-by-the-mapping-quality-filtering-optinal" class="headerlink" title="4.1 Filtering mapped reads by the mapping quality filtering [optinal]"></a>4.1 Filtering mapped reads by the mapping quality filtering [<font color="red">optinal</font>]</h2><p>Some project may require more stringent filtering on the alignment quality score. This <a target="_blank" rel="noopener" href="http://biofinysics.blogspot.com/2014/05/how-does-bowtie2-assign-mapq-scores.html">blog</a> detailedly discussed how does bowtie assign quality score with examples.</p>
<p>MAPQ(x) = -10 * log10(P(x is mapped wrongly)) = -10 * log10(p)</p>
<p>which ranges from 0 to 37, 40 or 42.</p>
<p><code>samtools view -q minQualityScore</code> will <strong>eliminate</strong> all the alignment results that are below the minQualityScore defined by user.</p>
<figure class="highlight sh"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">##== linux command ==##</span></span><br><span class="line">minQualityScore=2<span class="comment">#20?30?</span></span><br><span class="line">samtools view -q <span class="variable">$minQualityScore</span> <span class="variable">$&#123;projPath&#125;</span>/alignment/sam/<span class="variable">$&#123;histName&#125;</span>_bowtie2.sam &gt;<span class="variable">$&#123;projPath&#125;</span>/alignment/sam/<span class="variable">$&#123;histName&#125;</span>_bowtie2.qualityScore<span class="variable">$minQualityScore</span>.sam</span><br></pre></td></tr></table></figure>

<ul>
<li>If you do implement this filtering, please replace the <code>$&#123;histName&#125;_bowtie2.sam</code> in the following steps by this filtered sam file <code>$&#123;histName&#125;_bowtie2.qualityScore$minQualityScore.sam</code>. <font color="red">Use the MAPQ20/30 reads</font></li>
</ul>
<h2 id="4-2-File-format-conversion-required"><a href="#4-2-File-format-conversion-required" class="headerlink" title="4.2 File format conversion [required]"></a>4.2 File format conversion [required]</h2><p>This section is <strong>required</strong> in preparation for the <font color="red">peak calling</font> and <font color="red">visualization</font> where there are a few filtering and file format conversion that need to be done.</p>
<p>For example, in MACS2, it has bed(shift extent mode) and pe mode</p>
<figure class="highlight sh"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">##== linux command ==##</span></span><br><span class="line"><span class="comment">## Filter and keep the mapped read pairs</span></span><br><span class="line">samtools view -bS -F 0x4 <span class="variable">$projPath</span>/alignment/sam/<span class="variable">$&#123;histName&#125;</span>_bowtie2.sam &gt;<span class="variable">$projPath</span>/alignment/bam/<span class="variable">$&#123;histName&#125;</span>_bowtie2.mapped.bam</span><br><span class="line"></span><br><span class="line"><span class="comment">## Convert into bed file format</span></span><br><span class="line">bedtools bamtobed -i <span class="variable">$projPath</span>/alignment/bam/<span class="variable">$&#123;histName&#125;</span>_bowtie2.mapped.bam -bedpe &gt;<span class="variable">$projPath</span>/alignment/bed/<span class="variable">$&#123;histName&#125;</span>_bowtie2.bed</span><br><span class="line"></span><br><span class="line"><span class="comment">## Keep the read pairs that are on the same chromosome and fragment length less than 1000bp.</span></span><br><span class="line">awk <span class="string">&#x27;$1==$4 &amp;&amp; $6-$2 &lt; 1000 &#123;print $0&#125;&#x27;</span> <span class="variable">$projPath</span>/alignment/bed/<span class="variable">$&#123;histName&#125;</span>_bowtie2.bed &gt;<span class="variable">$projPath</span>/alignment/bed/<span class="variable">$&#123;histName&#125;</span>_bowtie2.clean.bed</span><br><span class="line"></span><br><span class="line"><span class="comment">## Only extract the fragment related columns</span></span><br><span class="line">cut -f 1,2,6 <span class="variable">$projPath</span>/alignment/bed/<span class="variable">$&#123;histName&#125;</span>_bowtie2.clean.bed | sort -k1,1 -k2,2n -k3,3n  &gt;<span class="variable">$projPath</span>/alignment/bed/<span class="variable">$&#123;histName&#125;</span>_bowtie2.fragments.bed</span><br></pre></td></tr></table></figure>

<h2 id="4-3-Assess-replicate-reproducibility-continue-section-3-5"><a href="#4-3-Assess-replicate-reproducibility-continue-section-3-5" class="headerlink" title="4.3 Assess replicate reproducibility (continue section 3.5)"></a>4.3 Assess replicate reproducibility (continue section 3.5)</h2><p>To study the reproducibility between replicates and across conditions, the <strong>genome is split into 500 bp bins</strong>, and a <strong>Pearson correlation of the log2-transformed values</strong> of <strong>read counts in each bin</strong> is calculated between replicate datasets. Multiple replicates and IgG control datasets are displayed in a hierarchically clustered correlation matrix.</p>
<figure class="highlight sh"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">##== linux command ==##</span></span><br><span class="line"><span class="comment">## We use the mid point of each fragment to infer which 500bp bins does this fragment belong to.</span></span><br><span class="line">binLen=500</span><br><span class="line">awk -v w=<span class="variable">$binLen</span> <span class="string">&#x27;&#123;print $1, int(($2 + $3)/(2*w))*w + w/2&#125;&#x27;</span> <span class="variable">$projPath</span>/alignment/bed/<span class="variable">$&#123;histName&#125;</span>_bowtie2.fragments.bed | sort -k1,1V -k2,2n | uniq -c | awk -v OFS=<span class="string">&quot;\t&quot;</span> <span class="string">&#x27;&#123;print $2, $3, $1&#125;&#x27;</span> |  sort -k1,1V -k2,2n  &gt;<span class="variable">$projPath</span>/alignment/bed/<span class="variable">$&#123;histName&#125;</span>_bowtie2.fragmentsCount.bin<span class="variable">$binLen</span>.bed</span><br></pre></td></tr></table></figure>



<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">##== R command ==##</span></span><br><span class="line">reprod = <span class="built_in">c</span>()</span><br><span class="line">fragCount = <span class="literal">NULL</span></span><br><span class="line"><span class="keyword">for</span>(hist <span class="keyword">in</span> sampleList)&#123;</span><br><span class="line">  </span><br><span class="line">  <span class="keyword">if</span>(<span class="built_in">is.null</span>(fragCount))&#123;</span><br><span class="line">    </span><br><span class="line">    fragCount = read.table(paste0(projPath, <span class="string">&quot;/alignment/bed/&quot;</span>, hist, <span class="string">&quot;_bowtie2.fragmentsCount.bin500.bed&quot;</span>), header = <span class="literal">FALSE</span>) </span><br><span class="line">    colnames(fragCount) = <span class="built_in">c</span>(<span class="string">&quot;chrom&quot;</span>, <span class="string">&quot;bin&quot;</span>, hist)</span><br><span class="line">  </span><br><span class="line">  &#125;<span class="keyword">else</span>&#123;</span><br><span class="line">    </span><br><span class="line">    fragCountTmp = read.table(paste0(projPath, <span class="string">&quot;/alignment/bed/&quot;</span>, hist, <span class="string">&quot;_bowtie2.fragmentsCount.bin500.bed&quot;</span>), header = <span class="literal">FALSE</span>)</span><br><span class="line">    colnames(fragCountTmp) = <span class="built_in">c</span>(<span class="string">&quot;chrom&quot;</span>, <span class="string">&quot;bin&quot;</span>, hist)</span><br><span class="line">    fragCount = full_join(fragCount, fragCountTmp, by = <span class="built_in">c</span>(<span class="string">&quot;chrom&quot;</span>, <span class="string">&quot;bin&quot;</span>))</span><br><span class="line">    </span><br><span class="line">  &#125;</span><br><span class="line">&#125;</span><br><span class="line"></span><br><span class="line">M = cor(fragCount %&gt;% select(-<span class="built_in">c</span>(<span class="string">&quot;chrom&quot;</span>, <span class="string">&quot;bin&quot;</span>)) %&gt;% log2(), use = <span class="string">&quot;complete.obs&quot;</span>) </span><br><span class="line"></span><br><span class="line">corrplot(M, method = <span class="string">&quot;color&quot;</span>, outline = <span class="built_in">T</span>, addgrid.col = <span class="string">&quot;darkgray&quot;</span>, order=<span class="string">&quot;hclust&quot;</span>, addrect = <span class="number">3</span>, rect.col = <span class="string">&quot;black&quot;</span>, rect.lwd = <span class="number">3</span>,cl.pos = <span class="string">&quot;b&quot;</span>, tl.col = <span class="string">&quot;indianred4&quot;</span>, tl.cex = <span class="number">1</span>, cl.cex = <span class="number">1</span>, addCoef.col = <span class="string">&quot;black&quot;</span>, number.digits = <span class="number">2</span>, number.cex = <span class="number">1</span>, col = colorRampPalette(<span class="built_in">c</span>(<span class="string">&quot;midnightblue&quot;</span>,<span class="string">&quot;white&quot;</span>,<span class="string">&quot;darkred&quot;</span>))(<span class="number">100</span>))</span><br></pre></td></tr></table></figure>

<p><img src="/blog/2022/03/21/2022-03-21-CUTTag/5.png"></p>
<hr>
<h1 id="V-Spike-in-calibration"><a href="#V-Spike-in-calibration" class="headerlink" title="V. Spike-in calibration"></a>V. Spike-in calibration</h1><p>This section is <strong>optional</strong> but <strong>recommended</strong> depending on your experimental protocol. We have shown the alignment to the spike-in genome in Section 3.1.2 and the spike-in alignment summary in Section 3.2.2.</p>
<p>The underlying assumption is that the ratio of fragments mapped to the primary genome to the E. coli genome is the same for a series of samples, each using the same number of cells. Because of this assumption, we do not normalize between experiments or between batches of purified pATn5, which can have very different amounts of carry-over E. coli DNA. Using a constant C to avoid small fractions in normalized data, we define a scaling factor S as</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">S = C / (fragments mapped to E. coli genome)</span><br></pre></td></tr></table></figure>

<p>Normalized coverage is then calculated as:</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">Normalized coverage = (primary_genome_coverage) * S</span><br></pre></td></tr></table></figure>

<p>The Constant is an arbitrary multiplier, typically 10,000. The resulting file will be comparatively small as a genomic coverage bedgraph file.</p>
<figure class="highlight sh"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">##== linux command ==##</span></span><br><span class="line"><span class="keyword">if</span> [[ <span class="string">&quot;<span class="variable">$seqDepth</span>&quot;</span> -gt <span class="string">&quot;1&quot;</span> ]]; <span class="keyword">then</span></span><br><span class="line">    </span><br><span class="line">    mkdir -p <span class="variable">$projPath</span>/alignment/bedgraph</span><br><span class="line"></span><br><span class="line">    scale_factor=`<span class="built_in">echo</span> <span class="string">&quot;10000 / <span class="variable">$seqDepth</span>&quot;</span> | bc -l`</span><br><span class="line">    <span class="built_in">echo</span> <span class="string">&quot;Scaling factor for <span class="variable">$histName</span> is: <span class="variable">$scale_factor</span>!&quot;</span></span><br><span class="line">    bedtools genomecov -<span class="built_in">bg</span> -scale <span class="variable">$scale_factor</span> -i <span class="variable">$projPath</span>/alignment/bed/<span class="variable">$&#123;histName&#125;</span>_bowtie2.fragments.bed -g <span class="variable">$chromSize</span> &gt; <span class="variable">$projPath</span>/alignment/bedgraph/<span class="variable">$&#123;histName&#125;</span>_bowtie2.fragments.normalized.bedgraph</span><br><span class="line">    </span><br><span class="line"><span class="keyword">fi</span></span><br></pre></td></tr></table></figure>

<h2 id="5-1-Scaling-factor"><a href="#5-1-Scaling-factor" class="headerlink" title="5.1 Scaling factor"></a>5.1 Scaling factor</h2><figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">##=== R command ===## </span></span><br><span class="line">scaleFactor = <span class="built_in">c</span>()</span><br><span class="line">multiplier = <span class="number">10000</span></span><br><span class="line"><span class="keyword">for</span>(hist <span class="keyword">in</span> sampleList)&#123;</span><br><span class="line">  spikeDepth = read.table(paste0(projPath, <span class="string">&quot;/alignment/sam/bowtie2_summary/&quot;</span>, hist, <span class="string">&quot;_bowtie2_spikeIn.seqDepth&quot;</span>), header = <span class="literal">FALSE</span>, fill = <span class="literal">TRUE</span>)$V1[<span class="number">1</span>]</span><br><span class="line">  </span><br><span class="line">  histInfo = strsplit(hist, <span class="string">&quot;_&quot;</span>)[[<span class="number">1</span>]]</span><br><span class="line">  scaleFactor = data.frame(scaleFactor = multiplier/spikeDepth, Histone = histInfo[<span class="number">1</span>], Replicate = histInfo[<span class="number">2</span>])  %&gt;% rbind(scaleFactor, .)</span><br><span class="line">&#125;</span><br><span class="line">scaleFactor$Histone = factor(scaleFactor$Histone, levels = histList)</span><br><span class="line">left_join(alignDupSummary, scaleFactor, by = <span class="built_in">c</span>(<span class="string">&quot;Histone&quot;</span>, <span class="string">&quot;Replicate&quot;</span>))</span><br></pre></td></tr></table></figure>

<p>Histone<br>Replicate<br>SequencingDepth<br>MappedFragNum_hg38<br>AlignmentRate_hg38<br>MappedFragNum_spikeIn<br>AlignmentRate_spikeIn<br><strong>DuplicationRate</strong><br><strong>EstimatedLibrarySize</strong><br><strong>UniqueFragNum</strong><br><strong>scaleFactor</strong></p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">##=== R command ===##</span></span><br><span class="line"><span class="comment">## Generate sequencing depth boxplot</span></span><br><span class="line">fig6A = scaleFactor %&gt;% ggplot(aes(x = Histone, y = scaleFactor, fill = Histone)) +</span><br><span class="line">    geom_boxplot() +</span><br><span class="line">    geom_jitter(aes(color = Replicate), position = position_jitter(<span class="number">0.15</span>)) +</span><br><span class="line">    scale_fill_viridis(discrete = <span class="literal">TRUE</span>, begin = <span class="number">0.1</span>, end = <span class="number">0.9</span>, option = <span class="string">&quot;magma&quot;</span>, alpha = <span class="number">0.8</span>) +</span><br><span class="line">    scale_color_viridis(discrete = <span class="literal">TRUE</span>, begin = <span class="number">0.1</span>, end = <span class="number">0.9</span>) +</span><br><span class="line">    theme_bw(base_size = <span class="number">20</span>) +</span><br><span class="line">    ylab(<span class="string">&quot;Spike-in Scalling Factor&quot;</span>) +</span><br><span class="line">    xlab(<span class="string">&quot;&quot;</span>)</span><br><span class="line"></span><br><span class="line">normDepth = inner_join(scaleFactor, alignResult, by = <span class="built_in">c</span>(<span class="string">&quot;Histone&quot;</span>, <span class="string">&quot;Replicate&quot;</span>)) %&gt;% mutate(normDepth = MappedFragNum_hg38 * scaleFactor)</span><br><span class="line"></span><br><span class="line">fig6B = normDepth %&gt;% ggplot(aes(x = Histone, y = normDepth, fill = Histone)) +</span><br><span class="line">    geom_boxplot() +</span><br><span class="line">    geom_jitter(aes(color = Replicate), position = position_jitter(<span class="number">0.15</span>)) +</span><br><span class="line">    scale_fill_viridis(discrete = <span class="literal">TRUE</span>, begin = <span class="number">0.1</span>, end = <span class="number">0.9</span>, option = <span class="string">&quot;magma&quot;</span>, alpha = <span class="number">0.8</span>) +</span><br><span class="line">    scale_color_viridis(discrete = <span class="literal">TRUE</span>, begin = <span class="number">0.1</span>, end = <span class="number">0.9</span>) +</span><br><span class="line">    theme_bw(base_size = <span class="number">20</span>) +</span><br><span class="line">    ylab(<span class="string">&quot;Normalization Fragment Count&quot;</span>) +</span><br><span class="line">    xlab(<span class="string">&quot;&quot;</span>) + </span><br><span class="line">    coord_cartesian(ylim = <span class="built_in">c</span>(<span class="number">1000000</span>, <span class="number">130000000</span>))</span><br><span class="line">ggarrange(fig6A, fig6B, ncol = <span class="number">2</span>, common.legend = <span class="literal">TRUE</span>, legend=<span class="string">&quot;bottom&quot;</span>)</span><br></pre></td></tr></table></figure>

<p><img src="/blog/2022/03/21/2022-03-21-CUTTag/6.png"></p>
<hr>
<h1 id="VI-Peak-calling"><a href="#VI-Peak-calling" class="headerlink" title="VI. Peak calling"></a>VI. Peak calling</h1><h2 id="6-1-SEACR"><a href="#6-1-SEACR" class="headerlink" title="6.1. SEACR"></a>6.1. SEACR</h2><p>The Sparse Enrichment Analysis for CUT&amp;RUN, <a target="_blank" rel="noopener" href="https://github.com/FredHutch/SEACR/">SEACR</a>, package is designed to call peaks and enriched regions from chromatin profiling data with very low backgrounds (i.e., regions with no read coverage) that are typical for CUT&amp;Tag chromatin profiling experiments. SEACR requires bedGraph files from paired-end sequencing as input and defines peaks as contiguous blocks of basepair coverage that do not overlap with blocks of background signal delineated in the IgG control dataset. SEACR is effective for calling both narrow peaks from factor binding sites and broad domains characteristic of some histone modifications. The description of the method is published at <a target="_blank" rel="noopener" href="https://epigeneticsandchromatin.biomedcentral.com/articles/10.1186/s13072-019-0287-4">Meers et al. 2019</a>, and the user’s manual is available on <a target="_blank" rel="noopener" href="https://github.com/FredHutch/SEACR/">github</a>. Since we have normalized fragment counts with the E. coli read count, we set the normalization option of SEACR to “non”. Otherwise, the “norm” is recommended.</p>
<figure class="highlight sh"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">##== linux command ==##</span></span><br><span class="line">seacr=<span class="string">&quot;/fh/fast/gottardo_r/yezheng_working/Software/SEACR/SEACR_1.3.sh&quot;</span></span><br><span class="line">histControl=<span class="variable">$2</span></span><br><span class="line">mkdir -p <span class="variable">$projPath</span>/peakCalling/SEACR</span><br><span class="line"></span><br><span class="line">bash <span class="variable">$seacr</span> <span class="variable">$projPath</span>/alignment/bedgraph/<span class="variable">$&#123;histName&#125;</span>_bowtie2.fragments.normalized.bedgraph \</span><br><span class="line">     <span class="variable">$projPath</span>/alignment/bedgraph/<span class="variable">$&#123;histControl&#125;</span>_bowtie2.fragments.normalized.bedgraph \</span><br><span class="line">     non stringent <span class="variable">$projPath</span>/peakCalling/SEACR/<span class="variable">$&#123;histName&#125;</span>_seacr_control.peaks</span><br><span class="line"></span><br><span class="line">bash <span class="variable">$seacr</span> <span class="variable">$projPath</span>/alignment/bedgraph/<span class="variable">$&#123;histName&#125;</span>_bowtie2.fragments.normalized.bedgraph 0.01 non stringent <span class="variable">$projPath</span>/peakCalling/SEACR/<span class="variable">$&#123;histName&#125;</span>_seacr_top0.01.peaks</span><br></pre></td></tr></table></figure>

<h3 id="6-1-1-Number-of-peaks-called"><a href="#6-1-1-Number-of-peaks-called" class="headerlink" title="6.1.1 Number of peaks called"></a>6.1.1 Number of peaks called</h3><figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">##=== R command ===## </span></span><br><span class="line">peakN = <span class="built_in">c</span>()</span><br><span class="line">peakWidth = <span class="built_in">c</span>()</span><br><span class="line">peakType = <span class="built_in">c</span>(<span class="string">&quot;control&quot;</span>, <span class="string">&quot;top0.01&quot;</span>)</span><br><span class="line"><span class="keyword">for</span>(hist <span class="keyword">in</span> sampleList)&#123;</span><br><span class="line">  histInfo = strsplit(hist, <span class="string">&quot;_&quot;</span>)[[<span class="number">1</span>]]</span><br><span class="line">  <span class="keyword">if</span>(histInfo[<span class="number">1</span>] != <span class="string">&quot;IgG&quot;</span>)&#123;</span><br><span class="line">    <span class="keyword">for</span>(type <span class="keyword">in</span> peakType)&#123;</span><br><span class="line">      peakInfo = read.table(paste0(projPath, <span class="string">&quot;/peakCalling/SEACR/&quot;</span>, hist, <span class="string">&quot;_seacr_&quot;</span>, type, <span class="string">&quot;.peaks.stringent.bed&quot;</span>), header = <span class="literal">FALSE</span>, fill = <span class="literal">TRUE</span>)  %&gt;% mutate(width = <span class="built_in">abs</span>(V3-V2))</span><br><span class="line">      peakN = data.frame(peakN = nrow(peakInfo), peakType = type, Histone = histInfo[<span class="number">1</span>], Replicate = histInfo[<span class="number">2</span>]) %&gt;% rbind(peakN, .)</span><br><span class="line">      peakWidth = data.frame(width = peakInfo$width, peakType = type, Histone = histInfo[<span class="number">1</span>], Replicate = histInfo[<span class="number">2</span>])  %&gt;% rbind(peakWidth, .)</span><br><span class="line">    &#125;</span><br><span class="line">  &#125;</span><br><span class="line">&#125;</span><br><span class="line">peakN %&gt;% select(Histone, Replicate, peakType, peakN)</span><br></pre></td></tr></table></figure>

<table>
<thead>
<tr>
<th align="left">Histone<chr></chr></th>
<th align="left">Replicate<chr></chr></th>
<th align="left">peakType<chr></chr></th>
<th align="right">peakN<int></int></th>
</tr>
</thead>
<tbody><tr>
<td align="left">K27me3</td>
<td align="left">rep1</td>
<td align="left">control</td>
<td align="right">144906</td>
</tr>
<tr>
<td align="left">K27me3</td>
<td align="left">rep1</td>
<td align="left">top0.01</td>
<td align="right">5829</td>
</tr>
<tr>
<td align="left">K27me3</td>
<td align="left">rep2</td>
<td align="left">control</td>
<td align="right">74444</td>
</tr>
<tr>
<td align="left">K27me3</td>
<td align="left">rep2</td>
<td align="left">top0.01</td>
<td align="right">9642</td>
</tr>
<tr>
<td align="left">K4me3</td>
<td align="left">rep1</td>
<td align="left">control</td>
<td align="right">8709</td>
</tr>
<tr>
<td align="left">K4me3</td>
<td align="left">rep1</td>
<td align="left">top0.01</td>
<td align="right">878</td>
</tr>
</tbody></table>
<h3 id="6-1-2-Reproducibility-of-the-peak-across-biological-replicates"><a href="#6-1-2-Reproducibility-of-the-peak-across-biological-replicates" class="headerlink" title="6.1.2 Reproducibility of the peak across biological replicates"></a>6.1.2 Reproducibility of the peak across biological replicates</h3><p>Peak calling on replicate datasets is compared to define reproducible peaks. The top 1% of peaks (ranked by total signal in each block) are selected as high-confidence sites.</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">##=== R command ===## </span></span><br><span class="line">histL = <span class="built_in">c</span>(<span class="string">&quot;K27me3&quot;</span>, <span class="string">&quot;K4me3&quot;</span>)</span><br><span class="line">repL = paste0(<span class="string">&quot;rep&quot;</span>, <span class="number">1</span>:<span class="number">2</span>)</span><br><span class="line">peakType = <span class="built_in">c</span>(<span class="string">&quot;control&quot;</span>, <span class="string">&quot;top0.01&quot;</span>)</span><br><span class="line">peakOverlap = <span class="built_in">c</span>()</span><br><span class="line"><span class="keyword">for</span>(type <span class="keyword">in</span> peakType)&#123;</span><br><span class="line">  <span class="keyword">for</span>(hist <span class="keyword">in</span> histL)&#123;</span><br><span class="line">    overlap.gr = GRanges()</span><br><span class="line">    <span class="keyword">for</span>(<span class="built_in">rep</span> <span class="keyword">in</span> repL)&#123;</span><br><span class="line">      peakInfo = read.table(paste0(projPath, <span class="string">&quot;/peakCalling/SEACR/&quot;</span>, hist, <span class="string">&quot;_&quot;</span>, <span class="built_in">rep</span>, <span class="string">&quot;_seacr_&quot;</span>, type, <span class="string">&quot;.peaks.stringent.bed&quot;</span>), header = <span class="literal">FALSE</span>, fill = <span class="literal">TRUE</span>)</span><br><span class="line">      peakInfo.gr = GRanges(peakInfo$V1, IRanges(start = peakInfo$V2, end = peakInfo$V3), strand = <span class="string">&quot;*&quot;</span>)</span><br><span class="line">      <span class="keyword">if</span>(<span class="built_in">length</span>(overlap.gr) &gt;<span class="number">0</span>)&#123;</span><br><span class="line">        overlap.gr = overlap.gr[findOverlaps(overlap.gr, peakInfo.gr)@from]</span><br><span class="line">      &#125;<span class="keyword">else</span>&#123;</span><br><span class="line">        overlap.gr = peakInfo.gr</span><br><span class="line">        </span><br><span class="line">      &#125;</span><br><span class="line">    &#125;</span><br><span class="line">    peakOverlap = data.frame(peakReprod = <span class="built_in">length</span>(overlap.gr), Histone = hist, peakType = type) %&gt;% rbind(peakOverlap, .)</span><br><span class="line">  &#125;</span><br><span class="line">&#125;</span><br><span class="line"></span><br><span class="line">peakReprod = left_join(peakN, peakOverlap, by = <span class="built_in">c</span>(<span class="string">&quot;Histone&quot;</span>, <span class="string">&quot;peakType&quot;</span>)) %&gt;% mutate(peakReprodRate = peakReprod/peakN * <span class="number">100</span>)</span><br><span class="line">peakReprod %&gt;% select(Histone, Replicate, peakType, peakN, peakReprodNum = peakReprod, peakReprodRate)</span><br></pre></td></tr></table></figure>

<table>
<thead>
<tr>
<th align="left">Histone<chr></chr></th>
<th align="left">Replicate<chr></chr></th>
<th align="left">peakType<chr></chr></th>
<th align="right">peakN<int></int></th>
<th align="right">peakReprodNum<int></int></th>
<th align="right">peakReprodRate<dbl></dbl></th>
</tr>
</thead>
<tbody><tr>
<td align="left">K27me3</td>
<td align="left">rep1</td>
<td align="left">control</td>
<td align="right">144906</td>
<td align="right">71692</td>
<td align="right">49.47483</td>
</tr>
<tr>
<td align="left">K27me3</td>
<td align="left">rep1</td>
<td align="left">top0.01</td>
<td align="right">5829</td>
<td align="right">5314</td>
<td align="right">91.16487</td>
</tr>
<tr>
<td align="left">K27me3</td>
<td align="left">rep2</td>
<td align="left">control</td>
<td align="right">74444</td>
<td align="right">71692</td>
<td align="right">96.30326</td>
</tr>
<tr>
<td align="left">K27me3</td>
<td align="left">rep2</td>
<td align="left">top0.01</td>
<td align="right">9642</td>
<td align="right">5314</td>
<td align="right">55.11305</td>
</tr>
</tbody></table>
<p>The reproducibility is calculated by</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">`# peaks overlapping rep1 and rep2/# peaks of rep1 or rep2 * 100</span><br></pre></td></tr></table></figure>

<p>Therefore, it is sensitive to the total number of peaks called in each replicate.</p>
<h3 id="6-1-3-FRagment-proportion-in-Peaks-regions-FRiPs"><a href="#6-1-3-FRagment-proportion-in-Peaks-regions-FRiPs" class="headerlink" title="6.1.3 FRagment proportion in Peaks regions (FRiPs)."></a>6.1.3 FRagment proportion in Peaks regions (FRiPs).</h3><p>We calculate the fraction of reads in peaks (FRiPs) as a measure of signal-to-noise, and contrast it to FRiPs in the IgG control dataset for illustration. Although sequencing depths for CUT&amp;Tag are typically only 1-5 million reads, the low background of the method results in high FRiP scores.</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br></pre></td><td class="code"><pre><span class="line">##=== R command ===## </span><br><span class="line">library(chromVAR)</span><br><span class="line"></span><br><span class="line">bamDir = paste0(projPath, &quot;/alignment/bam&quot;)</span><br><span class="line">inPeakData = c()</span><br><span class="line">## overlap with bam file to get count</span><br><span class="line">for(hist in histL)&#123;</span><br><span class="line">  for(rep in repL)&#123;</span><br><span class="line">    peakRes = read.table(paste0(projPath, &quot;/peakCalling/SEACR/&quot;, hist, &quot;_&quot;, rep, &quot;_seacr_control.peaks.stringent.bed&quot;), header = FALSE, fill = TRUE)</span><br><span class="line">    peak.gr = GRanges(seqnames = peakRes$V1, IRanges(start = peakRes$V2, end = peakRes$V3), strand = &quot;*&quot;)</span><br><span class="line">    bamFile = paste0(bamDir, &quot;/&quot;, hist, &quot;_&quot;, rep, &quot;_bowtie2.mapped.bam&quot;)</span><br><span class="line">    fragment_counts &lt;- getCounts(bamFile, peak.gr, paired = TRUE, by_rg = FALSE, format = &quot;bam&quot;)</span><br><span class="line">    inPeakN = counts(fragment_counts)[,1] %&gt;% sum</span><br><span class="line">    inPeakData = rbind(inPeakData, data.frame(inPeakN = inPeakN, Histone = hist, Replicate = rep))</span><br><span class="line">  &#125;</span><br><span class="line">&#125;</span><br><span class="line"></span><br><span class="line">frip = left_join(inPeakData, alignResult, by = c(&quot;Histone&quot;, &quot;Replicate&quot;)) %&gt;% mutate(frip = inPeakN/MappedFragNum_hg38 * 100)</span><br><span class="line">frip %&gt;% select(Histone, Replicate, SequencingDepth, MappedFragNum_hg38, AlignmentRate_hg38, FragInPeakNum = inPeakN, FRiPs = frip)</span><br></pre></td></tr></table></figure>

<table>
<thead>
<tr>
<th align="left">Histone<chr></chr></th>
<th align="left">Replicate<chr></chr></th>
<th align="right">SequencingDepth<dbl></dbl></th>
<th align="right">MappedFragNum_hg38<dbl></dbl></th>
<th align="right">AlignmentRate_hg38<dbl></dbl></th>
<th align="right">FragInPeakNum<dbl></dbl></th>
<th align="left"><strong>FRiPs</strong></th>
</tr>
</thead>
<tbody><tr>
<td align="left">K27me3</td>
<td align="left">rep1</td>
<td align="right">2984630</td>
<td align="right">2859520</td>
<td align="right">95.81</td>
<td align="right">2223813</td>
<td align="left">77.76875</td>
</tr>
<tr>
<td align="left">K27me3</td>
<td align="left">Rep2</td>
<td align="right"></td>
<td align="right"></td>
<td align="right"></td>
<td align="right"></td>
<td align="left">40.68507</td>
</tr>
<tr>
<td align="left">K4me3</td>
<td align="left">rep1</td>
<td align="right"></td>
<td align="right"></td>
<td align="right"></td>
<td align="right"></td>
<td align="left">91.60169</td>
</tr>
<tr>
<td align="left">K4me3</td>
<td align="left">Rep2</td>
<td align="right"></td>
<td align="right"></td>
<td align="right"></td>
<td align="right"></td>
<td align="left">71.46116</td>
</tr>
</tbody></table>
<h3 id="6-1-4-Visualization-of-peak-number-peak-width-peak-reproducibility-and-FRiPs"><a href="#6-1-4-Visualization-of-peak-number-peak-width-peak-reproducibility-and-FRiPs" class="headerlink" title="6.1.4 Visualization of peak number, peak width, peak reproducibility and FRiPs."></a>6.1.4 Visualization of peak number, peak width, peak reproducibility and FRiPs.</h3><figure class="highlight sh"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br></pre></td><td class="code"><pre><span class="line">fig7A = peakN %&gt;% ggplot(aes(x = Histone, y = peakN, fill = Histone)) +</span><br><span class="line">    geom_boxplot() +</span><br><span class="line">    geom_jitter(aes(color = Replicate), position = position_jitter(0.15)) +</span><br><span class="line">    facet_grid(~peakType) +</span><br><span class="line">    scale_fill_viridis(discrete = TRUE, begin = 0.1, end = 0.55, option = <span class="string">&quot;magma&quot;</span>, alpha = 0.8) +</span><br><span class="line">    scale_color_viridis(discrete = TRUE, begin = 0.1, end = 0.9) +</span><br><span class="line">    theme_bw(base_size = 18) +</span><br><span class="line">    ylab(<span class="string">&quot;Number of Peaks&quot;</span>) +</span><br><span class="line">    xlab(<span class="string">&quot;&quot;</span>)</span><br><span class="line"></span><br><span class="line">fig7B = peakWidth %&gt;% ggplot(aes(x = Histone, y = width, fill = Histone)) +</span><br><span class="line">    geom_violin() +</span><br><span class="line">    facet_grid(Replicate~peakType) +</span><br><span class="line">    scale_fill_viridis(discrete = TRUE, begin = 0.1, end = 0.55, option = <span class="string">&quot;magma&quot;</span>, alpha = 0.8) +</span><br><span class="line">    scale_color_viridis(discrete = TRUE, begin = 0.1, end = 0.9) +</span><br><span class="line">    scale_y_continuous(trans = <span class="string">&quot;log&quot;</span>, breaks = c(400, 3000, 22000)) +</span><br><span class="line">    theme_bw(base_size = 18) +</span><br><span class="line">    ylab(<span class="string">&quot;Width of Peaks&quot;</span>) +</span><br><span class="line">    xlab(<span class="string">&quot;&quot;</span>)</span><br><span class="line"></span><br><span class="line">fig7C = peakReprod %&gt;% ggplot(aes(x = Histone, y = peakReprodRate, fill = Histone, label = round(peakReprodRate, 2))) +</span><br><span class="line">    geom_bar(<span class="built_in">stat</span> = <span class="string">&quot;identity&quot;</span>) +</span><br><span class="line">    geom_text(vjust = 0.1) +</span><br><span class="line">    facet_grid(Replicate~peakType) +</span><br><span class="line">    scale_fill_viridis(discrete = TRUE, begin = 0.1, end = 0.55, option = <span class="string">&quot;magma&quot;</span>, alpha = 0.8) +</span><br><span class="line">    scale_color_viridis(discrete = TRUE, begin = 0.1, end = 0.9) +</span><br><span class="line">    theme_bw(base_size = 18) +</span><br><span class="line">    ylab(<span class="string">&quot;% of Peaks Reproduced&quot;</span>) +</span><br><span class="line">    xlab(<span class="string">&quot;&quot;</span>)</span><br><span class="line"></span><br><span class="line">fig7D = frip %&gt;% ggplot(aes(x = Histone, y = frip, fill = Histone, label = round(frip, 2))) +</span><br><span class="line">    geom_boxplot() +</span><br><span class="line">    geom_jitter(aes(color = Replicate), position = position_jitter(0.15)) +</span><br><span class="line">    scale_fill_viridis(discrete = TRUE, begin = 0.1, end = 0.55, option = <span class="string">&quot;magma&quot;</span>, alpha = 0.8) +</span><br><span class="line">    scale_color_viridis(discrete = TRUE, begin = 0.1, end = 0.9) +</span><br><span class="line">    theme_bw(base_size = 18) +</span><br><span class="line">    ylab(<span class="string">&quot;% of Fragments in Peaks&quot;</span>) +</span><br><span class="line">    xlab(<span class="string">&quot;&quot;</span>)</span><br><span class="line"></span><br><span class="line">ggarrange(fig7A, fig7B, fig7C, fig7D, ncol = 2, nrow=2, common.legend = TRUE, legend=<span class="string">&quot;bottom&quot;</span>)</span><br></pre></td></tr></table></figure>

<p><img src="/blog/2022/03/21/2022-03-21-CUTTag/7.png" alt="7"></p>
<hr>
<h1 id="VII-Visualization"><a href="#VII-Visualization" class="headerlink" title="VII. Visualization"></a>VII. Visualization</h1><p>Typically we are interested in visualizing a chromatin landscape in regions using a genome browser. The <a target="_blank" rel="noopener" href="http://software.broadinstitute.org/software/igv/home">Integrative Genomic Viewer</a> provides a web app version and a local desktop version that are easy to use. The <a target="_blank" rel="noopener" href="https://genome.ucsc.edu/">UCSC Genome Browser</a> provides the most comprehensive supplementary genome information.</p>
<h2 id="7-1-Browser-display-of-normalized-bedgraph-files"><a href="#7-1-Browser-display-of-normalized-bedgraph-files" class="headerlink" title="7.1. Browser display of normalized bedgraph files."></a>7.1. Browser display of normalized bedgraph files.</h2><p><img src="/blog/2022/03/21/2022-03-21-CUTTag/8.png" alt="7"></p>
<h2 id="7-2-Heatmap-visualization-on-specific-regions"><a href="#7-2-Heatmap-visualization-on-specific-regions" class="headerlink" title="7.2. Heatmap visualization on specific regions"></a>7.2. Heatmap visualization on specific regions</h2><p>We are also interested in looking at chromatin features at a list of annotated sites, for example histone modification signal at gene promoters.We will use the <code>computeMatrix</code> and <code>plotHeatmap</code> functions from <a target="_blank" rel="noopener" href="https://deeptools.readthedocs.io/en/develop/">deepTools</a> to generate the heatmap.</p>
<figure class="highlight sh"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">##== linux command ==##</span></span><br><span class="line">mkdir -p <span class="variable">$projPath</span>/alignment/bigwig                                                                                                                                        </span><br><span class="line">samtools sort -o <span class="variable">$projPath</span>/alignment/bam/<span class="variable">$&#123;histName&#125;</span>.sorted.bam <span class="variable">$projPath</span>/alignment/bam/<span class="variable">$&#123;histName&#125;</span>_bowtie2.mapped.bam                                                     </span><br><span class="line">samtools index <span class="variable">$projPath</span>/alignment/bam/<span class="variable">$&#123;histName&#125;</span>.sorted.bam                                                                                                              </span><br><span class="line">bamCoverage -b <span class="variable">$projPath</span>/alignment/bam/<span class="variable">$&#123;histName&#125;</span>.sorted.bam -o <span class="variable">$projPath</span>/alignment/bigwig/<span class="variable">$&#123;histName&#125;</span>_raw.bw         </span><br></pre></td></tr></table></figure>

<h3 id="7-2-1-Heatmap-over-transcription-units"><a href="#7-2-1-Heatmap-over-transcription-units" class="headerlink" title="7.2.1 Heatmap over transcription units"></a>7.2.1 Heatmap over transcription units</h3><figure class="highlight sh"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">##== linux command ==##</span></span><br><span class="line">cores=8</span><br><span class="line">computeMatrix scale-regions -S <span class="variable">$projPath</span>/alignment/bigwig/K27me3_rep1_raw.bw \</span><br><span class="line">                               <span class="variable">$projPath</span>/alignment/bigwig/K27me3_rep2_raw.bw \</span><br><span class="line">                               <span class="variable">$projPath</span>/alignment/bigwig/K4me3_rep1_raw.bw \</span><br><span class="line">                               <span class="variable">$projPath</span>/alignment/bigwig/K4me3_rep2_raw.bw \</span><br><span class="line">                              -R <span class="variable">$projPath</span>/data/hg38_gene/hg38_gene.tsv \</span><br><span class="line">                              --beforeRegionStartLength 3000 \</span><br><span class="line">                              --regionBodyLength 5000 \</span><br><span class="line">                              --afterRegionStartLength 3000 \</span><br><span class="line">                              --skipZeros -o <span class="variable">$projPath</span>/data/hg38_gene/matrix_gene.mat.gz -p <span class="variable">$cores</span></span><br><span class="line"></span><br><span class="line">plotHeatmap -m <span class="variable">$projPath</span>/data/hg38_gene/matrix_gene.mat.gz -out <span class="variable">$projPath</span>/data/hg38_gene/Histone_gene.png --sortUsing sum</span><br></pre></td></tr></table></figure>

<p><img src="/blog/2022/03/21/2022-03-21-CUTTag/9.png" alt="7"></p>
<h3 id="7-2-2-Heatmap-on-CUT-amp-Tag-peaks"><a href="#7-2-2-Heatmap-on-CUT-amp-Tag-peaks" class="headerlink" title="7.2.2. Heatmap on CUT&amp;Tag peaks"></a>7.2.2. Heatmap on CUT&amp;Tag peaks</h3><p>We use the midpoint of the signal block returned from SEACR to align signals in heatmaps. The sixth column of the SEACR output is an entry in the form chr:start-end that represents the first and ending bases of the region with the maximum signal of the region. We first generate a new bed file containing this midpoint information in column 6 and use deeptools for the heatmap visualization.</p>
<figure class="highlight sh"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#== linux command ==##</span></span><br><span class="line">awk <span class="string">&#x27;&#123;split($6, summit, &quot;:&quot;); split(summit[2], region, &quot;-&quot;); print summit[1]&quot;\t&quot;region[1]&quot;\t&quot;region[2]&#125;&#x27;</span> <span class="variable">$projPath</span>/peakCalling/SEACR/<span class="variable">$&#123;histName&#125;</span>_<span class="variable">$&#123;repName&#125;</span>_seacr_control.pe\</span><br><span class="line">aks.stringent.bed &gt;<span class="variable">$projPath</span>/peakCalling/SEACR/<span class="variable">$&#123;histName&#125;</span>_<span class="variable">$&#123;repName&#125;</span>_seacr_control.peaks.summitRegion.bed</span><br><span class="line"></span><br><span class="line">computeMatrix reference-point -S <span class="variable">$projPath</span>/alignment/bigwig/<span class="variable">$&#123;histName&#125;</span>_<span class="variable">$&#123;repName&#125;</span>_raw.bw \</span><br><span class="line">              -R <span class="variable">$projPath</span>/peakCalling/SEACR/<span class="variable">$&#123;histName&#125;</span>_<span class="variable">$&#123;repName&#125;</span>_seacr_control.peaks.summitRegion.bed \</span><br><span class="line">              --skipZeros -o <span class="variable">$projPath</span>/peakCalling/SEACR/<span class="variable">$&#123;histName&#125;</span>_<span class="variable">$&#123;repName&#125;</span>_SEACR.mat.gz -p <span class="variable">$cores</span> -a 3000 -b 3000 --referencePoint center</span><br><span class="line"></span><br><span class="line">plotHeatmap -m <span class="variable">$projPath</span>/peakCalling/SEACR/<span class="variable">$&#123;histName&#125;</span>_SEACR.mat.gz -out <span class="variable">$projPath</span>/peakCalling/SEACR/<span class="variable">$&#123;histName&#125;</span>_SEACR_heatmap.png --sortUsing sum --startLabel <span class="string">&quot;Peak Start&quot;</span> -\</span><br><span class="line">-endLabel <span class="string">&quot;Peak End&quot;</span> --xAxisLabel <span class="string">&quot;&quot;</span> --regionsLabel <span class="string">&quot;Peaks&quot;</span> --samplesLabel <span class="string">&quot;<span class="variable">$&#123;histName&#125;</span> <span class="variable">$&#123;repName&#125;</span>&quot;</span></span><br></pre></td></tr></table></figure>

<p><img src="/blog/2022/03/21/2022-03-21-CUTTag/10.png" alt="7"></p>
<hr>
<h1 id="VIII-Differention-analysis"><a href="#VIII-Differention-analysis" class="headerlink" title="VIII. Differention analysis"></a>VIII. Differention analysis</h1><ul>
<li>DESeq2: <a target="_blank" rel="noopener" href="https://genomebiology.biomedcentral.com/articles/10.1186/s13059-014-0550-8">Moderated estimation of fold change and dispersion for RNA-seq data with DESeq2</a></li>
</ul>
<p>Estimate variance-mean dependence in count data from high-throughput sequencing assays and test for differential expression based on a model using the negative binomial distribution.</p>
<h2 id="8-1-Create-the-peak-x-sample-matrix"><a href="#8-1-Create-the-peak-x-sample-matrix" class="headerlink" title="8.1. Create the peak x sample matrix."></a>8.1. Create the peak x sample matrix.</h2><p>Usually, the differential tests compare two or more conditions of the same histone modification. In this tutorial, limited by the demonstration data, we will illustrate the differential detection by comparing two replicates of H3K27me3 and two replicates of H3K4me3. We will use DESeq2 (<a target="_blank" rel="noopener" href="http://bioconductor.org/packages/release/bioc/vignettes/DESeq2/inst/doc/DESeq2.html#why-un-normalized-counts">complete tutorial</a>) as illustration.</p>
<h3 id="8-1-1-Create-a-master-peak-list-merging-all-the-peaks-called-for-each-sample"><a href="#8-1-1-Create-a-master-peak-list-merging-all-the-peaks-called-for-each-sample" class="headerlink" title="8.1.1 Create a master peak list merging all the peaks called for each sample."></a>8.1.1 Create a master peak list merging all the peaks called for each sample.</h3><figure class="highlight sh"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">##=== R command ===## </span></span><br><span class="line">mPeak = GRanges()</span><br><span class="line"><span class="comment">## overlap with bam file to get count</span></span><br><span class="line"><span class="keyword">for</span>(hist <span class="keyword">in</span> histL)&#123;</span><br><span class="line">  <span class="keyword">for</span>(rep <span class="keyword">in</span> repL)&#123;</span><br><span class="line">    peakRes = read.table(paste0(projPath, <span class="string">&quot;/peakCalling/SEACR/&quot;</span>, hist, <span class="string">&quot;_&quot;</span>, rep, <span class="string">&quot;_seacr_control.peaks.stringent.bed&quot;</span>), header = FALSE, fill = TRUE)</span><br><span class="line">    mPeak = GRanges(seqnames = peakRes<span class="variable">$V1</span>, IRanges(start = peakRes<span class="variable">$V2</span>, end = peakRes<span class="variable">$V3</span>), strand = <span class="string">&quot;*&quot;</span>) %&gt;% append(mPeak, .)</span><br><span class="line">  &#125;</span><br><span class="line">&#125;</span><br><span class="line">masterPeak = reduce(mPeak)</span><br></pre></td></tr></table></figure>

<h3 id="8-1-2-Get-the-fragment-counts-for-each-peak-in-the-master-peak-list"><a href="#8-1-2-Get-the-fragment-counts-for-each-peak-in-the-master-peak-list" class="headerlink" title="8.1.2 Get the fragment counts for each peak in the master peak list."></a>8.1.2 Get the fragment counts for each peak in the master peak list.</h3><figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">##=== R command ===## </span></span><br><span class="line">library(DESeq2)</span><br><span class="line">bamDir = paste0(projPath, <span class="string">&quot;/alignment/bam&quot;</span>)</span><br><span class="line">countMat = matrix(<span class="literal">NA</span>, <span class="built_in">length</span>(masterPeak), <span class="built_in">length</span>(histL)*<span class="built_in">length</span>(repL))</span><br><span class="line"><span class="comment">## overlap with bam file to get count</span></span><br><span class="line">i = <span class="number">1</span></span><br><span class="line"><span class="keyword">for</span>(hist <span class="keyword">in</span> histL)&#123;</span><br><span class="line">  <span class="keyword">for</span>(<span class="built_in">rep</span> <span class="keyword">in</span> repL)&#123;</span><br><span class="line">    </span><br><span class="line">    bamFile = paste0(bamDir, <span class="string">&quot;/&quot;</span>, hist, <span class="string">&quot;_&quot;</span>, <span class="built_in">rep</span>, <span class="string">&quot;_bowtie2.mapped.bam&quot;</span>)</span><br><span class="line">    fragment_counts &lt;- getCounts(bamFile, masterPeak, paired = <span class="literal">TRUE</span>, by_rg = <span class="literal">FALSE</span>, format = <span class="string">&quot;bam&quot;</span>)</span><br><span class="line">    countMat[, i] = counts(fragment_counts)[,<span class="number">1</span>]</span><br><span class="line">    i = i + <span class="number">1</span></span><br><span class="line">  &#125;</span><br><span class="line">&#125;</span><br><span class="line">colnames(countMat) = paste(<span class="built_in">rep</span>(histL, <span class="number">2</span>), <span class="built_in">rep</span>(repL, each = <span class="number">2</span>), sep = <span class="string">&quot;_&quot;</span>)</span><br></pre></td></tr></table></figure>



<h2 id="8-2-Sequencing-depth-normalization-and-differential-enriched-peaks-detection"><a href="#8-2-Sequencing-depth-normalization-and-differential-enriched-peaks-detection" class="headerlink" title="8.2. Sequencing depth normalization and differential enriched peaks detection"></a>8.2. Sequencing depth normalization and differential enriched peaks detection</h2><figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">##=== R command ===## </span></span><br><span class="line">selectR = which(rowSums(countMat) &gt; <span class="number">5</span>) <span class="comment">## remove low count genes</span></span><br><span class="line">dataS = countMat[selectR,]</span><br><span class="line">condition = factor(<span class="built_in">rep</span>(histL, each = <span class="built_in">length</span>(repL)))</span><br><span class="line">dds = DESeqDataSetFromMatrix(countData = dataS,</span><br><span class="line">                              colData = DataFrame(condition),</span><br><span class="line">                              design = ~ condition)</span><br><span class="line">DDS = DESeq(dds)</span><br><span class="line">normDDS = counts(DDS, normalized = <span class="literal">TRUE</span>) <span class="comment">## normalization with respect to the sequencing depth</span></span><br><span class="line">colnames(normDDS) = paste0(colnames(normDDS), <span class="string">&quot;_norm&quot;</span>)</span><br><span class="line">res = results(DDS, independentFiltering = <span class="literal">FALSE</span>, altHypothesis = <span class="string">&quot;greaterAbs&quot;</span>)</span><br><span class="line"></span><br><span class="line">countMatDiff = cbind(dataS, normDDS, res)</span><br><span class="line">head(countMatDiff)</span><br></pre></td></tr></table></figure>

<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br></pre></td><td class="code"><pre><span class="line">DataFrame with <span class="number">6</span> rows and <span class="number">14</span> columns</span><br><span class="line">  K27me3_rep1 K4me3_rep1 K27me3_rep2 K4me3_rep2 K27me3_rep1_norm</span><br><span class="line">    &lt;numeric&gt;  &lt;numeric&gt;   &lt;numeric&gt;  &lt;numeric&gt;        &lt;numeric&gt;</span><br><span class="line"><span class="number">1</span>           <span class="number">6</span>          <span class="number">2</span>           <span class="number">1</span>          <span class="number">6</span>         <span class="number">1.408657</span></span><br><span class="line"><span class="number">2</span>           <span class="number">1</span>          <span class="number">0</span>         <span class="number">242</span>        <span class="number">182</span>         <span class="number">0.234776</span></span><br><span class="line"><span class="number">3</span>           <span class="number">0</span>          <span class="number">0</span>         <span class="number">176</span>         <span class="number">88</span>         <span class="number">0.000000</span></span><br><span class="line"><span class="number">4</span>           <span class="number">0</span>          <span class="number">0</span>         <span class="number">274</span>        <span class="number">194</span>         <span class="number">0.000000</span></span><br><span class="line"><span class="number">5</span>           <span class="number">3</span>          <span class="number">4</span>           <span class="number">0</span>          <span class="number">1</span>         <span class="number">0.704328</span></span><br><span class="line"><span class="number">6</span>           <span class="number">0</span>          <span class="number">1</span>         <span class="number">109</span>         <span class="number">59</span>         <span class="number">0.000000</span></span><br><span class="line">  K4me3_rep1_norm K27me3_rep2_norm K4me3_rep2_norm  baseMean log2FoldChange</span><br><span class="line">        &lt;numeric&gt;        &lt;numeric&gt;       &lt;numeric&gt; &lt;numeric&gt;      &lt;numeric&gt;</span><br><span class="line"><span class="number">1</span>        <span class="number">0.620403</span>            <span class="number">4.170</span>         <span class="number">18.2724</span>   <span class="number">6.11787</span>       <span class="number">3.496854</span></span><br><span class="line"><span class="number">2</span>        <span class="number">0.000000</span>         <span class="number">1009.141</span>        <span class="number">554.2634</span> <span class="number">390.90978</span>      <span class="number">12.510325</span></span><br><span class="line"><span class="number">3</span>        <span class="number">0.000000</span>          <span class="number">733.921</span>        <span class="number">267.9955</span> <span class="number">250.47905</span>      <span class="number">13.297304</span></span><br><span class="line"><span class="number">4</span>        <span class="number">0.000000</span>         <span class="number">1142.581</span>        <span class="number">590.8082</span> <span class="number">433.34733</span>      <span class="number">14.089840</span></span><br><span class="line"><span class="number">5</span>        <span class="number">1.240806</span>            <span class="number">0.000</span>          <span class="number">3.0454</span>   <span class="number">1.24763</span>       <span class="number">0.846266</span></span><br><span class="line"><span class="number">6</span>        <span class="number">0.310202</span>          <span class="number">454.530</span>        <span class="number">179.6788</span> <span class="number">158.62986</span>      <span class="number">11.189689</span></span><br><span class="line">      lfcSE      stat      pvalue        padj</span><br><span class="line">  &lt;numeric&gt; &lt;numeric&gt;   &lt;numeric&gt;   &lt;numeric&gt;</span><br><span class="line"><span class="number">1</span>   <span class="number">1.19893</span>  <span class="number">2.916635</span> <span class="number">3.53829e-03</span> <span class="number">4.22134e-02</span></span><br><span class="line"><span class="number">2</span>   <span class="number">1.50039</span>  <span class="number">8.338074</span> <span class="number">7.55102e-17</span> <span class="number">2.18197e-15</span></span><br><span class="line"><span class="number">3</span>   <span class="number">1.58547</span>  <span class="number">8.386969</span> <span class="number">4.98837e-17</span> <span class="number">1.50548e-15</span></span><br><span class="line"><span class="number">4</span>   <span class="number">1.55196</span>  <span class="number">9.078730</span> <span class="number">1.09850e-19</span> <span class="number">6.73560e-18</span></span><br><span class="line"><span class="number">5</span>   <span class="number">2.18326</span>  <span class="number">0.387617</span> <span class="number">6.98300e-01</span> <span class="number">9.72755e-01</span></span><br><span class="line"><span class="number">6</span>   <span class="number">1.53046</span>  <span class="number">7.311313</span> <span class="number">2.64545e-13</span> <span class="number">4.33546e-12</span></span><br></pre></td></tr></table></figure>

<ul>
<li>DESeq2 requires the input matrix should be un-normalized counts or estimated counts of sequencing reads.</li>
<li>DESeq2 model internally corrects for library size.</li>
<li><strong>countMatDiff</strong> summarizes the differential analysis results:<ul>
<li>First 4 columns: raw reads counts after filtering the peak regions with low counts</li>
<li>Second 4 columns: normalized read counts eliminating library size difference.</li>
<li>Remaining columns: differential detection results.</li>
</ul>
</li>
</ul>
<h1 id="References"><a href="#References" class="headerlink" title="References"></a>References</h1><figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line">Kaya-Okur HS, Wu SJ, Codomo CA, Pledger ES, Bryson TD, Henikoff JG, Ahmad K, Henikoff S: CUT&amp;Tag for efficient epigenomic profiling of small samples and single cells. Nature Communications 2019 10:1930 (PMID:31036827).</span><br><span class="line"></span><br><span class="line">Meers, M.P., Tenenbaum, D. &amp; Henikoff, S. Peak calling by Sparse Enrichment Analysis for CUT&amp;RUN chromatin profiling. Epigenetics &amp; Chromatin 12, 42 (2019). https://doi.org/10.1186/s13072-019-0287-4</span><br><span class="line"></span><br></pre></td></tr></table></figure>

<h1 id="IX-Additional-Alternatives"><a href="#IX-Additional-Alternatives" class="headerlink" title="IX. Additional Alternatives"></a>IX. Additional Alternatives</h1><h2 id="9-1-ChIPseqSpikeInFree-for-normalizing-data-without-spike-in-DNA-Optional"><a href="#9-1-ChIPseqSpikeInFree-for-normalizing-data-without-spike-in-DNA-Optional" class="headerlink" title="9.1 ChIPseqSpikeInFree for normalizing data without spike-in DNA [Optional]"></a>9.1 ChIPseqSpikeInFree for normalizing data without spike-in DNA [Optional]</h2><p><a target="_blank" rel="noopener" href="https://academic.oup.com/bioinformatics/article/36/4/1270/5578481">ChIPseqSpikeInFree: a ChIP-seq normalization approach to reveal global changes in histone modifications without spike-in</a> is a novel ChIP-seq normalization method to effectively determine scaling factors for samples across various conditions and treatments, which does not rely on exogenous spike-in chromatin or peak detection to reveal global changes in histone modification occupancy. The installation details can be found on <a target="_blank" rel="noopener" href="https://github.com/stjude/ChIPseqSpikeInFree">github</a>.</p>
<ul>
<li><a target="_blank" rel="noopener" href="https://github.com/stjude/ChIPseqSpikeInFree#interpretation-of-scaling-factor-table">Interpretation of the ChIPseqSpikeInFree output.</a></li>
<li><a target="_blank" rel="noopener" href="https://github.com/stjude/ChIPseqSpikeInFree#how-to-use-chipseqspikein-scaling-factor">How to use ChIPseqSpikeInFree scaling factor.</a></li>
</ul>
<h2 id="9-2-Other-peak-calling-methods"><a href="#9-2-Other-peak-calling-methods" class="headerlink" title="9.2. Other peak calling methods."></a>9.2. Other peak calling methods.</h2><ul>
<li>MACS2: <a target="_blank" rel="noopener" href="https://genomebiology.biomedcentral.com/articles/10.1186/gb-2008-9-9-r137">Model-based Analysis of ChIP-Seq (MACS)</a>. Installation details can be found <a target="_blank" rel="noopener" href="https://github.com/taoliu/MACS/wiki">here</a>.</li>
</ul>
<figure class="highlight sh"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">##== linux command ==##</span></span><br><span class="line">histName=<span class="string">&quot;K27me3&quot;</span></span><br><span class="line">controlName=<span class="string">&quot;IgG&quot;</span></span><br><span class="line"></span><br><span class="line">mkdir -p <span class="variable">$projPath</span>/peakCalling</span><br><span class="line">macs2 callpeak -t <span class="variable">$&#123;projPath&#125;</span>/alignment/bam/<span class="variable">$&#123;histName&#125;</span>_rep1_bowtie2.mapped.bam \</span><br><span class="line">      -c <span class="variable">$&#123;projPath&#125;</span>/alignment/bam/<span class="variable">$&#123;controlName&#125;</span>_rep1_bowtie2.mapped.bam \</span><br><span class="line">      -g hs -f BAMPE -n macs2_peak_q0.1 --outdir <span class="variable">$projPath</span>/peakCalling/MACS2 -q 0.1 --keep-dup all 2&gt;<span class="variable">$&#123;projPath&#125;</span>/peakCalling/MACS2/macs2Peak_summary.txt</span><br></pre></td></tr></table></figure>

<ul>
<li>dPeak: <a target="_blank" rel="noopener" href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1003246">dPeak: High Resolution Identification of Transcription Factor Binding Sites from PET and SET ChIP-Seq Data</a></li>
<li>MOSAiCS: <a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4608541/">A Statistical Framework for the Analysis of ChIP-Seq Data</a></li>
</ul>
<h2 id="9-3-Other-packages-for-differential-analysis-of-binding-sites"><a href="#9-3-Other-packages-for-differential-analysis-of-binding-sites" class="headerlink" title="9.3 Other packages for differential analysis of binding sites"></a>9.3 Other packages for differential analysis of binding sites</h2><ul>
<li>Limma: <a target="_blank" rel="noopener" href="https://academic.oup.com/nar/article/43/7/e47/2414268">limma powers differential expression analyses for RNA-sequencing and microarray studies</a></li>
</ul>
<p>Limma is an R package for the analysis of gene expression microarray data, especially the use of linear models for analysing designed experiments and the assessment of differential expression. Limma provides the ability to analyse comparisons between many RNA targets simultaneously in arbitrary complicated designed experiments. Empirical Bayesian methods are used to provide stable results even when the number of arrays is small. Limma can be extended to study differential fragment enrichment analysis within peak regions. Notably, limma can deal with both fixed effect model and random effect model.</p>
<ul>
<li>edgeR: <a target="_blank" rel="noopener" href="https://academic.oup.com/nar/article/40/10/4288/2411520">Differential Expression Analysis of Multifactor RNA-Seq Experiments With Respect to Biological Variation</a></li>
</ul>
<p>Differential expression analysis of RNA-seq expression profiles with biological replication. Implements a range of statistical methodology based on the negative binomial distributions, including empirical Bayes estimation, exact tests, generalized linear models and quasi-likelihood tests. As well as RNA-seq, it be applied to differential signal analysis of other types of genomic data that produce read counts, including ChIP-seq, ATAC-seq, Bisulfite-seq, SAGE and CAGE. edgeR can deal with multifactor problem.</p>
<h1 id="X-Troubleshooting-Generating-your-data"><a href="#X-Troubleshooting-Generating-your-data" class="headerlink" title="X. Troubleshooting: Generating your data"></a>X. Troubleshooting: Generating your data</h1><p>This workflow can be followed with your own data and will generate a standardized set of quality-control reports. However, many sequencing facilities do not perform 25x25 PE sequencing, and alternate parameters for trimming and mapping are provided here. Control datasets for non-specific antibody (IgG) profiling or ATAC-seq profiling of your material can also be used for optional analysis detailed here.</p>
<p>Stringent washing with 300 mM NaCl is critical to limit the affinity of Tn5 for exposed DNA. We describe here the need for controlling background Tn5 affinities and describe how our CUT&amp;Tag protocol effectively suppresses this artifact for unambiguous mapping of chromatin epitopes. We present a protocol that can process either native or fixed nuclei and includes alternative methods for DNA isolation. To illustrate the method, we describe a typical experiment, including evaluation of the results using a new metric for peak-calling information. Further, we validate a single-tube format for CUT&amp;Tag that requires no DNA isolation but instead uses tagmented material directly for library amplification. We document critical steps for the CUT&amp;Tag protocol, informed by our experiences, helping users establish this method in their research.</p>
<hr>
<p>Cite:<a target="_blank" rel="noopener" href="https://yezhengstat.github.io/CUTTag_tutorial/#">https://yezhengstat.github.io/CUTTag_tutorial/#</a> Zheng Y et al (2020). Protocol.io</p>

    </div>

    
    
    
        

<div>
<ul class="post-copyright">
  <li class="post-copyright-author">
    <strong>Post author:  </strong>Wang C
  </li>
  <li class="post-copyright-link">
    <strong>Post link: </strong>
    <a href="https://wangc1106.github.io/blog/2022/03/21/2022-03-21-CUTTag/" title="【CUT&amp;Tag-pipeline】">https://wangc1106.github.io/blog/2022/03/21/2022-03-21-CUTTag/</a>
  </li>
  <li class="post-copyright-license">
    <strong>Copyright Notice:  </strong>All articles in this blog are licensed under <a href="https://creativecommons.org/licenses/by-nc-sa/4.0/" rel="noopener" target="_blank"><i class="fab fa-fw fa-creative-commons"></i>BY-NC-SA</a> unless stating additionally.
  </li>
</ul>
</div>


      <footer class="post-footer">
          <div class="post-tags">
              <a href="/blog/tags/pipeline/" rel="tag"># pipeline</a>
          </div>

        


        
    <div class="post-nav">
      <div class="post-nav-item">
    <a href="/blog/2022/03/20/2022-03-20-ATAC-seq-6/" rel="prev" title="【ATAC-seq-pipeline】">
      <i class="fa fa-chevron-left"></i> 【ATAC-seq-pipeline】
    </a></div>
      <div class="post-nav-item"></div>
    </div>
      </footer>
    
  </article>
  
  
  



          </div>
          

<script>
  window.addEventListener('tabs:register', () => {
    let { activeClass } = CONFIG.comments;
    if (CONFIG.comments.storage) {
      activeClass = localStorage.getItem('comments_active') || activeClass;
    }
    if (activeClass) {
      let activeTab = document.querySelector(`a[href="#comment-${activeClass}"]`);
      if (activeTab) {
        activeTab.click();
      }
    }
  });
  if (CONFIG.comments.storage) {
    window.addEventListener('tabs:click', event => {
      if (!event.target.matches('.tabs-comment .tab-content .tab-pane')) return;
      let commentClass = event.target.classList[1];
      localStorage.setItem('comments_active', commentClass);
    });
  }
</script>

        </div>
          
  
  <div class="toggle sidebar-toggle">
    <span class="toggle-line toggle-line-first"></span>
    <span class="toggle-line toggle-line-middle"></span>
    <span class="toggle-line toggle-line-last"></span>
  </div>

  <aside class="sidebar">
    <div class="sidebar-inner">

      <ul class="sidebar-nav motion-element">
        <li class="sidebar-nav-toc">
          Table of Contents
        </li>
        <li class="sidebar-nav-overview">
          Overview
        </li>
      </ul>

      <!--noindex-->
      <div class="post-toc-wrap sidebar-panel">
          <div class="post-toc motion-element"><ol class="nav"><li class="nav-item nav-level-1"><a class="nav-link" href="#I-Introduction"><span class="nav-number">1.</span> <span class="nav-text">I. Introduction</span></a><ol class="nav-child"><li class="nav-item nav-level-2"><a class="nav-link" href="#1-1-Overview-of-CUT-amp-Tag"><span class="nav-number">1.1.</span> <span class="nav-text">1.1. Overview of CUT&amp;Tag</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#1-2-Objectives"><span class="nav-number">1.2.</span> <span class="nav-text">1.2. Objectives</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#1-3-CUT-amp-Tag-data-processing-and-analysis-outline"><span class="nav-number">1.3.</span> <span class="nav-text">1.3 CUT&amp;Tag data processing and analysis outline.</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#1-4-Requirements"><span class="nav-number">1.4.</span> <span class="nav-text">1.4. Requirements</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#1-5-Data-Downloading"><span class="nav-number">1.5.</span> <span class="nav-text">1.5. Data Downloading</span></a></li></ol></li><li class="nav-item nav-level-1"><a class="nav-link" href="#II-Data-Pre-processing"><span class="nav-number">2.</span> <span class="nav-text">II. Data Pre-processing</span></a><ol class="nav-child"><li class="nav-item nav-level-2"><a class="nav-link" href="#2-2-Merge-technical-replicates-lanes-if-needed-Optional"><span class="nav-number">2.1.</span> <span class="nav-text">2.2. Merge technical replicates&#x2F;lanes if needed [Optional]</span></a></li></ol></li><li class="nav-item nav-level-1"><a class="nav-link" href="#III-Alignment"><span class="nav-number">3.</span> <span class="nav-text">III. Alignment</span></a><ol class="nav-child"><li class="nav-item nav-level-2"><a class="nav-link" href="#3-1-Bowtie2-alignment-required"><span class="nav-number">3.1.</span> <span class="nav-text">3.1. Bowtie2 alignment [required]</span></a><ol class="nav-child"><li class="nav-item nav-level-3"><a class="nav-link" href="#3-1-1-Alignment-to-ref"><span class="nav-number">3.1.1.</span> <span class="nav-text">3.1.1 Alignment to ref.</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#3-1-2-Alignment-to-spike-in-genome-for-spike-in-calibration-optional-recommended"><span class="nav-number">3.1.2.</span> <span class="nav-text">3.1.2 Alignment to spike-in genome for spike-in calibration [optional&#x2F;recommended]</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#3-1-3-Alignment-summary"><span class="nav-number">3.1.3.</span> <span class="nav-text">3.1.3 Alignment summary</span></a></li></ol></li><li class="nav-item nav-level-2"><a class="nav-link" href="#3-2-Report-sequencing-mapping-summary-required"><span class="nav-number">3.2.</span> <span class="nav-text">3.2 Report sequencing mapping summary [required]</span></a><ol class="nav-child"><li class="nav-item nav-level-3"><a class="nav-link" href="#3-2-1-Sequencing-depth"><span class="nav-number">3.2.1.</span> <span class="nav-text">3.2.1 Sequencing depth</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#3-2-2-Spike-in-alignment"><span class="nav-number">3.2.2.</span> <span class="nav-text">3.2.2 Spike-in alignment</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#3-2-3-Summarize-the-alignment-to-hg38-and-E-coli"><span class="nav-number">3.2.3.</span> <span class="nav-text">3.2.3 Summarize the alignment to hg38 and E.coli</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#3-2-4-Visualizing-the-sequencing-depth-and-alignment-results"><span class="nav-number">3.2.4.</span> <span class="nav-text">3.2.4 Visualizing the sequencing depth and alignment results</span></a></li></ol></li><li class="nav-item nav-level-2"><a class="nav-link" href="#3-3-Remove-duplicates-optional-required"><span class="nav-number">3.3.</span> <span class="nav-text">3.3. Remove duplicates [optional&#x2F;required]</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#3-4-Assess-mapped-fragment-size-distribution-Required"><span class="nav-number">3.4.</span> <span class="nav-text">3.4. Assess mapped fragment size distribution [Required]</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#3-5-Assess-replicate-reproducibility"><span class="nav-number">3.5.</span> <span class="nav-text">3.5 Assess replicate reproducibility</span></a></li></ol></li><li class="nav-item nav-level-1"><a class="nav-link" href="#IV-Alignments-results-filtering-and-file-format-conversion"><span class="nav-number">4.</span> <span class="nav-text">IV. Alignments results filtering and file format conversion</span></a><ol class="nav-child"><li class="nav-item nav-level-2"><a class="nav-link" href="#4-1-Filtering-mapped-reads-by-the-mapping-quality-filtering-optinal"><span class="nav-number">4.1.</span> <span class="nav-text">4.1 Filtering mapped reads by the mapping quality filtering [optinal]</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#4-2-File-format-conversion-required"><span class="nav-number">4.2.</span> <span class="nav-text">4.2 File format conversion [required]</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#4-3-Assess-replicate-reproducibility-continue-section-3-5"><span class="nav-number">4.3.</span> <span class="nav-text">4.3 Assess replicate reproducibility (continue section 3.5)</span></a></li></ol></li><li class="nav-item nav-level-1"><a class="nav-link" href="#V-Spike-in-calibration"><span class="nav-number">5.</span> <span class="nav-text">V. Spike-in calibration</span></a><ol class="nav-child"><li class="nav-item nav-level-2"><a class="nav-link" href="#5-1-Scaling-factor"><span class="nav-number">5.1.</span> <span class="nav-text">5.1 Scaling factor</span></a></li></ol></li><li class="nav-item nav-level-1"><a class="nav-link" href="#VI-Peak-calling"><span class="nav-number">6.</span> <span class="nav-text">VI. Peak calling</span></a><ol class="nav-child"><li class="nav-item nav-level-2"><a class="nav-link" href="#6-1-SEACR"><span class="nav-number">6.1.</span> <span class="nav-text">6.1. SEACR</span></a><ol class="nav-child"><li class="nav-item nav-level-3"><a class="nav-link" href="#6-1-1-Number-of-peaks-called"><span class="nav-number">6.1.1.</span> <span class="nav-text">6.1.1 Number of peaks called</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#6-1-2-Reproducibility-of-the-peak-across-biological-replicates"><span class="nav-number">6.1.2.</span> <span class="nav-text">6.1.2 Reproducibility of the peak across biological replicates</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#6-1-3-FRagment-proportion-in-Peaks-regions-FRiPs"><span class="nav-number">6.1.3.</span> <span class="nav-text">6.1.3 FRagment proportion in Peaks regions (FRiPs).</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#6-1-4-Visualization-of-peak-number-peak-width-peak-reproducibility-and-FRiPs"><span class="nav-number">6.1.4.</span> <span class="nav-text">6.1.4 Visualization of peak number, peak width, peak reproducibility and FRiPs.</span></a></li></ol></li></ol></li><li class="nav-item nav-level-1"><a class="nav-link" href="#VII-Visualization"><span class="nav-number">7.</span> <span class="nav-text">VII. Visualization</span></a><ol class="nav-child"><li class="nav-item nav-level-2"><a class="nav-link" href="#7-1-Browser-display-of-normalized-bedgraph-files"><span class="nav-number">7.1.</span> <span class="nav-text">7.1. Browser display of normalized bedgraph files.</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#7-2-Heatmap-visualization-on-specific-regions"><span class="nav-number">7.2.</span> <span class="nav-text">7.2. Heatmap visualization on specific regions</span></a><ol class="nav-child"><li class="nav-item nav-level-3"><a class="nav-link" href="#7-2-1-Heatmap-over-transcription-units"><span class="nav-number">7.2.1.</span> <span class="nav-text">7.2.1 Heatmap over transcription units</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#7-2-2-Heatmap-on-CUT-amp-Tag-peaks"><span class="nav-number">7.2.2.</span> <span class="nav-text">7.2.2. Heatmap on CUT&amp;Tag peaks</span></a></li></ol></li></ol></li><li class="nav-item nav-level-1"><a class="nav-link" href="#VIII-Differention-analysis"><span class="nav-number">8.</span> <span class="nav-text">VIII. Differention analysis</span></a><ol class="nav-child"><li class="nav-item nav-level-2"><a class="nav-link" href="#8-1-Create-the-peak-x-sample-matrix"><span class="nav-number">8.1.</span> <span class="nav-text">8.1. Create the peak x sample matrix.</span></a><ol class="nav-child"><li class="nav-item nav-level-3"><a class="nav-link" href="#8-1-1-Create-a-master-peak-list-merging-all-the-peaks-called-for-each-sample"><span class="nav-number">8.1.1.</span> <span class="nav-text">8.1.1 Create a master peak list merging all the peaks called for each sample.</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#8-1-2-Get-the-fragment-counts-for-each-peak-in-the-master-peak-list"><span class="nav-number">8.1.2.</span> <span class="nav-text">8.1.2 Get the fragment counts for each peak in the master peak list.</span></a></li></ol></li><li class="nav-item nav-level-2"><a class="nav-link" href="#8-2-Sequencing-depth-normalization-and-differential-enriched-peaks-detection"><span class="nav-number">8.2.</span> <span class="nav-text">8.2. Sequencing depth normalization and differential enriched peaks detection</span></a></li></ol></li><li class="nav-item nav-level-1"><a class="nav-link" href="#References"><span class="nav-number">9.</span> <span class="nav-text">References</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#IX-Additional-Alternatives"><span class="nav-number">10.</span> <span class="nav-text">IX. Additional Alternatives</span></a><ol class="nav-child"><li class="nav-item nav-level-2"><a class="nav-link" href="#9-1-ChIPseqSpikeInFree-for-normalizing-data-without-spike-in-DNA-Optional"><span class="nav-number">10.1.</span> <span class="nav-text">9.1 ChIPseqSpikeInFree for normalizing data without spike-in DNA [Optional]</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#9-2-Other-peak-calling-methods"><span class="nav-number">10.2.</span> <span class="nav-text">9.2. Other peak calling methods.</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#9-3-Other-packages-for-differential-analysis-of-binding-sites"><span class="nav-number">10.3.</span> <span class="nav-text">9.3 Other packages for differential analysis of binding sites</span></a></li></ol></li><li class="nav-item nav-level-1"><a class="nav-link" href="#X-Troubleshooting-Generating-your-data"><span class="nav-number">11.</span> <span class="nav-text">X. Troubleshooting: Generating your data</span></a></li></ol></div>
      </div>
      <!--/noindex-->

      <div class="site-overview-wrap sidebar-panel">
        <div class="site-author motion-element" itemprop="author" itemscope itemtype="http://schema.org/Person">
    <img class="site-author-image" itemprop="image" alt="Wang C"
      src="/blog/images/banner.png">
  <p class="site-author-name" itemprop="name">Wang C</p>
  <div class="site-description" itemprop="description">Make more time</div>
</div>
<div class="site-state-wrap motion-element">
  <nav class="site-state">
      <div class="site-state-item site-state-posts">
          <a href="/blog/archives">
          <span class="site-state-item-count">46</span>
          <span class="site-state-item-name">posts</span>
        </a>
      </div>
      <div class="site-state-item site-state-categories">
            <a href="/blog/categories/">
          
        <span class="site-state-item-count">13</span>
        <span class="site-state-item-name">categories</span></a>
      </div>
      <div class="site-state-item site-state-tags">
            <a href="/blog/tags/">
          
        <span class="site-state-item-count">19</span>
        <span class="site-state-item-name">tags</span></a>
      </div>
  </nav>
</div>
  <div class="links-of-author motion-element">
      <span class="links-of-author-item">
        <a href="https://github.com/wangc1106" title="GitHub → https:&#x2F;&#x2F;github.com&#x2F;wangc1106" rel="noopener" target="_blank"><i class="fab fa-github fa-fw"></i>GitHub</a>
      </span>
      <span class="links-of-author-item">
        <a href="mailto:wangchen116@126.com" title="E-Mail → mailto:wangchen116@126.com" rel="noopener" target="_blank"><i class="fa fa-envelope fa-fw"></i>E-Mail</a>
      </span>
  </div>
  <div class="cc-license motion-element" itemprop="license">
    <a href="https://creativecommons.org/licenses/by-nc-sa/4.0/" class="cc-opacity" rel="noopener" target="_blank"><img src="/blog/images/cc-by-nc-sa.svg" alt="Creative Commons"></a>
  </div>



      </div>
        <div class="back-to-top motion-element">
          <i class="fa fa-arrow-up"></i>
          <span>0%</span>
        </div>

    </div>
  </aside>
  <div id="sidebar-dimmer"></div>


      </div>
    </main>

    <footer class="footer">
      <div class="footer-inner">
        

        

<div class="copyright">
  
  &copy; 2021 – 
  <span itemprop="copyrightYear">2022</span>
  <span class="with-love">
    <i class="fas fa-dna"></i>
  </span>
  <span class="author" itemprop="copyrightHolder">Wang C</span>
</div>

        






<script>
  (function() {
    function leancloudSelector(url) {
      url = encodeURI(url);
      return document.getElementById(url).querySelector('.leancloud-visitors-count');
    }

    function addCount(Counter) {
      var visitors = document.querySelector('.leancloud_visitors');
      var url = decodeURI(visitors.id);
      var title = visitors.dataset.flagTitle;

      Counter('get', '/classes/Counter?where=' + encodeURIComponent(JSON.stringify({ url })))
        .then(response => response.json())
        .then(({ results }) => {
          if (results.length > 0) {
            var counter = results[0];
            leancloudSelector(url).innerText = counter.time + 1;
            Counter('put', '/classes/Counter/' + counter.objectId, { time: { '__op': 'Increment', 'amount': 1 } })
              .catch(error => {
                console.error('Failed to save visitor count', error);
              });
          } else {
              Counter('post', '/classes/Counter', { title, url, time: 1 })
                .then(response => response.json())
                .then(() => {
                  leancloudSelector(url).innerText = 1;
                })
                .catch(error => {
                  console.error('Failed to create', error);
                });
          }
        })
        .catch(error => {
          console.error('LeanCloud Counter Error', error);
        });
    }

    function showTime(Counter) {
      var visitors = document.querySelectorAll('.leancloud_visitors');
      var entries = [...visitors].map(element => {
        return decodeURI(element.id);
      });

      Counter('get', '/classes/Counter?where=' + encodeURIComponent(JSON.stringify({ url: { '$in': entries } })))
        .then(response => response.json())
        .then(({ results }) => {
          for (let url of entries) {
            let target = results.find(item => item.url === url);
            leancloudSelector(url).innerText = target ? target.time : 0;
          }
        })
        .catch(error => {
          console.error('LeanCloud Counter Error', error);
        });
    }

    let { app_id, app_key, server_url } = {"enable":true,"app_id":"UqIYufmNk2sE8MiBKKWc4Kja-gzGzoHsz","app_key":"iYpeWQcmNqpHwr2uVlnJYVji","server_url":null,"security":false};
    function fetchData(api_server) {
      var Counter = (method, url, data) => {
        return fetch(`${api_server}/1.1${url}`, {
          method,
          headers: {
            'X-LC-Id'     : app_id,
            'X-LC-Key'    : app_key,
            'Content-Type': 'application/json',
          },
          body: JSON.stringify(data)
        });
      };
      if (CONFIG.page.isPost) {
        if (CONFIG.hostname !== location.hostname) return;
        addCount(Counter);
      } else if (document.querySelectorAll('.post-title-link').length >= 1) {
        showTime(Counter);
      }
    }

    let api_server = app_id.slice(-9) !== '-MdYXbMMI' ? server_url : `https://${app_id.slice(0, 8).toLowerCase()}.api.lncldglobal.com`;

    if (api_server) {
      fetchData(api_server);
    } else {
      fetch('https://app-router.leancloud.cn/2/route?appId=' + app_id)
        .then(response => response.json())
        .then(({ api_server }) => {
          fetchData('https://' + api_server);
        });
    }
  })();
</script>


      </div>
      <div style="height: 200px">
      <script type="text/javascript" src="//rf.revolvermaps.com/0/0/8.js?i=5knf34gs07t&amp;m=0&amp;c=ff0000&amp;cr1=ffffff&amp;f=arial&amp;l=33" async="async"></script>
      </div>
    </footer>
  </div>

  
  <script src="/blog/lib/anime.min.js"></script>
  <script src="/blog/lib/velocity/velocity.min.js"></script>
  <script src="/blog/lib/velocity/velocity.ui.min.js"></script>

<script src="/blog/js/utils.js"></script>

<script src="/blog/js/motion.js"></script>


<script src="/blog/js/schemes/pisces.js"></script>


<script src="/blog/js/next-boot.js"></script>




  




  
<script src="/blog/js/local-search.js"></script>













  

  

</body>
</html>
