<!DOCTYPE html>
<html lang="zh-Hans">
<head>
  <meta charset="UTF-8">
<meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=2">
<meta name="theme-color" content="#222">
<meta name="generator" content="Hexo 5.4.0">
  <link rel="apple-touch-icon" sizes="180x180" href="/blog/images/apple-touch-icon-next.png">
  <link rel="icon" type="image/png" sizes="32x32" href="/blog/images/favicon-32x32-next.png">
  <link rel="icon" type="image/png" sizes="16x16" href="/blog/images/favicon-16x16-next.png">
  <link rel="mask-icon" href="/blog/images/logo.svg" color="#222">
  <meta name="google-site-verification" content="1osTo_gOe1SW9Hevqtz8vl8HfSgTO4DSMZ0DiMPMX6w">

<link rel="stylesheet" href="/blog/css/main.css">


<link rel="stylesheet" href="/blog/lib/font-awesome/css/all.min.css">

<script id="hexo-configurations">
    var NexT = window.NexT || {};
    var CONFIG = {"hostname":"wangc1106.github.io","root":"/blog/","scheme":"Gemini","version":"7.8.0","exturl":false,"sidebar":{"position":"left","display":"post","padding":18,"offset":12,"onmobile":false},"copycode":{"enable":false,"show_result":false,"style":null},"back2top":{"enable":true,"sidebar":true,"scrollpercent":"ture"},"bookmark":{"enable":false,"color":"#222","save":"auto"},"fancybox":false,"mediumzoom":false,"lazyload":false,"pangu":false,"comments":{"style":"tabs","active":null,"storage":true,"lazyload":false,"nav":null},"algolia":{"hits":{"per_page":10},"labels":{"input_placeholder":"Search for Posts","hits_empty":"We didn't find any results for the search: ${query}","hits_stats":"${hits} results found in ${time} ms"}},"localsearch":{"enable":true,"trigger":"auto","top_n_per_article":1,"unescape":false,"preload":false},"motion":{"enable":true,"async":false,"transition":{"post_block":"fadeIn","post_header":"slideDownIn","post_body":"slideDownIn","coll_header":"slideLeftIn","sidebar":"slideUpIn"}},"path":"search.xml"};
  </script>

  <meta name="description" content="purge haplotigs and overlaps in an assembly based on read depth">
<meta property="og:type" content="article">
<meta property="og:title" content="【hifi mapping】【purge_dups】">
<meta property="og:url" content="https://wangc1106.github.io/blog/2022/03/18/2022-03-18-purgedups/index.html">
<meta property="og:site_name" content="Banner&#39;s Blog">
<meta property="og:description" content="purge haplotigs and overlaps in an assembly based on read depth">
<meta property="og:locale">
<meta property="og:image" content="https://wangc1106.github.io/blog/2022/03/18/2022-03-18-purgedups/1.png">
<meta property="article:published_time" content="2022-03-18T02:19:22.778Z">
<meta property="article:modified_time" content="2022-03-19T14:15:23.482Z">
<meta property="article:author" content="Wang C">
<meta property="article:tag" content="denovo-III">
<meta name="twitter:card" content="summary">
<meta name="twitter:image" content="https://wangc1106.github.io/blog/2022/03/18/2022-03-18-purgedups/1.png">

<link rel="canonical" href="https://wangc1106.github.io/blog/2022/03/18/2022-03-18-purgedups/">


<script id="page-configurations">
  // https://hexo.io/docs/variables.html
  CONFIG.page = {
    sidebar: "",
    isHome : false,
    isPost : true,
    lang   : 'zh-Hans'
  };
</script>

  <title>【hifi mapping】【purge_dups】 | Banner's Blog</title>
  






  <noscript>
  <style>
  .use-motion .brand,
  .use-motion .menu-item,
  .sidebar-inner,
  .use-motion .post-block,
  .use-motion .pagination,
  .use-motion .comments,
  .use-motion .post-header,
  .use-motion .post-body,
  .use-motion .collection-header { opacity: initial; }

  .use-motion .site-title,
  .use-motion .site-subtitle {
    opacity: initial;
    top: initial;
  }

  .use-motion .logo-line-before i { left: initial; }
  .use-motion .logo-line-after i { right: initial; }
  </style>
</noscript>

</head>

<body itemscope itemtype="http://schema.org/WebPage">
  <div class="container use-motion">
    <div class="headband"></div>

    <header class="header" itemscope itemtype="http://schema.org/WPHeader">
      <div class="header-inner"><div class="site-brand-container">
  <div class="site-nav-toggle">
    <div class="toggle" aria-label="Toggle navigation bar">
      <span class="toggle-line toggle-line-first"></span>
      <span class="toggle-line toggle-line-middle"></span>
      <span class="toggle-line toggle-line-last"></span>
    </div>
  </div>

  <div class="site-meta">

    <a href="/blog/" class="brand" rel="start">
      <span class="logo-line-before"><i></i></span>
      <h1 class="site-title">Banner's Blog</h1>
      <span class="logo-line-after"><i></i></span>
    </a>
      <p class="site-subtitle" itemprop="description">Dream big work hard</p>
  </div>

  <div class="site-nav-right">
    <div class="toggle popup-trigger">
        <i class="fa fa-search fa-fw fa-lg"></i>
    </div>
  </div>
</div>




<nav class="site-nav">
  <ul id="menu" class="main-menu menu">
        <li class="menu-item menu-item-home">

    <a href="/blog/" rel="section"><i class="home fa-fw"></i>Home</a>

  </li>
        <li class="menu-item menu-item-categories">

    <a href="/blog/categories/" rel="section"><i class="th fa-fw"></i>Categories</a>

  </li>
        <li class="menu-item menu-item-tags">

    <a href="/blog/tags/" rel="section"><i class="tags fa-fw"></i>Tags</a>

  </li>
        <li class="menu-item menu-item-about">

    <a href="https://wangc1106.github.io/" rel="section"><i class="user fa-fw"></i>About</a>

  </li>
      <li class="menu-item menu-item-search">
        <a role="button" class="popup-trigger"><i class="fa fa-search fa-fw"></i>Search
        </a>
      </li>
  </ul>
</nav>



  <div class="search-pop-overlay">
    <div class="popup search-popup">
        <div class="search-header">
  <span class="search-icon">
    <i class="fa fa-search"></i>
  </span>
  <div class="search-input-container">
    <input autocomplete="off" autocapitalize="off"
           placeholder="Searching..." spellcheck="false"
           type="search" class="search-input">
  </div>
  <span class="popup-btn-close">
    <i class="fa fa-times-circle"></i>
  </span>
</div>
<div id="search-result">
  <div id="no-result">
    <i class="fa fa-spinner fa-pulse fa-5x fa-fw"></i>
  </div>
</div>

    </div>
  </div>

</div>
    </header>

    


    <main class="main">
      <div class="main-inner">
        <div class="content-wrap">
          

          <div class="content post posts-expand">
            

    
  
  
  <article itemscope itemtype="http://schema.org/Article" class="post-block" lang="zh-Hans">
    <link itemprop="mainEntityOfPage" href="https://wangc1106.github.io/blog/2022/03/18/2022-03-18-purgedups/">

    <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
      <meta itemprop="image" content="/blog/images/banner.png">
      <meta itemprop="name" content="Wang C">
      <meta itemprop="description" content="Make more time">
    </span>

    <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
      <meta itemprop="name" content="Banner's Blog">
    </span>
      <header class="post-header">
        <h1 class="post-title" itemprop="name headline">
          【hifi mapping】【purge_dups】
        </h1>

        <div class="post-meta">
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-calendar"></i>
              </span>
              <span class="post-meta-item-text">Posted on</span>

              <time title="Created: 2022-03-18 10:19:22" itemprop="dateCreated datePublished" datetime="2022-03-18T10:19:22+08:00">2022-03-18</time>
            </span>
              <span class="post-meta-item">
                <span class="post-meta-item-icon">
                  <i class="far fa-calendar-check"></i>
                </span>
                <span class="post-meta-item-text">Edited on</span>
                <time title="Modified: 2022-03-19 22:15:23" itemprop="dateModified" datetime="2022-03-19T22:15:23+08:00">2022-03-19</time>
              </span>
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-folder"></i>
              </span>
              <span class="post-meta-item-text">In</span>
                <span itemprop="about" itemscope itemtype="http://schema.org/Thing">
                  <a href="/blog/categories/Software/" itemprop="url" rel="index"><span itemprop="name">Software</span></a>
                </span>
            </span>

          
            <span id="/blog/2022/03/18/2022-03-18-purgedups/" class="post-meta-item leancloud_visitors" data-flag-title="【hifi mapping】【purge_dups】" title="Views">
              <span class="post-meta-item-icon">
                <i class="fa fa-eye"></i>
              </span>
              <span class="post-meta-item-text">Views: </span>
              <span class="leancloud-visitors-count"></span>
            </span>
            <span class="post-meta-item" title="Symbols count in article">
              <span class="post-meta-item-icon">
                <i class="far fa-file-word"></i>
              </span>
              <span>6.1k</span>
            </span>
            <span class="post-meta-item" title="Reading time">
              <span class="post-meta-item-icon">
                <i class="far fa-clock"></i>
              </span>
              <span>6 mins.</span>
            </span>

        </div>
      </header>

    
    
    
    <div class="post-body" itemprop="articleBody">

      
        <p>purge haplotigs and overlaps in an assembly based on read depth</p>
<span id="more"></span>

<p><a target="_blank" rel="noopener" href="https://github.com/dfguan/purge_dups">https://github.com/dfguan/purge_dups</a></p>
<h1 id="overview"><a href="#overview" class="headerlink" title="overview"></a>overview</h1><p>purge_dups is designed to remove haplotigs and contig overlaps in a <em>de novo</em> assembly based on read depth.</p>
<p><img src="/blog/2022/03/18/2022-03-18-purgedups/1.png" alt="img"></p>
<p><code>purge_dups</code>根据read深度分析组装中haplotigs和overlaps。相对于<code>purge_haplotigs</code>，运行速度更快，而且<strong>自动确定阈值</strong>。</p>
<p><code>purge_dups</code>分为三个部分，<strong>第一部分</strong>是将序列<strong>回贴</strong>到基因组并分析<strong>覆盖度确定阈值</strong>，<strong>第二部分</strong>是将<strong>组装自我比对</strong>，<strong>第三部分</strong>是利用前两部分得到的信息<strong>鉴定</strong>原来序列中的<strong>haplotigs和overlaps</strong>.</p>
<h1 id="Directory-Structure"><a href="#Directory-Structure" class="headerlink" title="Directory Structure"></a>Directory Structure</h1><ul>
<li>scripts/pd_config.py: script to generate a configuration file used by run_purge_dups.py.</li>
<li>scripts/run_purge_dups.py: script to run the <strong>purge_dups pipeline</strong>.</li>
<li>scripts/run_busco: script to run busco, dependency: busco.</li>
<li>scripts/run_kcm: script to make k-mer comparison plot.</li>
<li>scripts/sub.sh: shell script to submit a <strong>farm</strong> job.</li>
<li>src: purge_dups source files.</li>
<li><font color="blue"><strong>src/split_fa</strong></font>: split fasta file by ‘N’s.</li>
<li><font color="blue"><strong>src/pbcstat</strong></font>: create <strong>read depth histogram and base-level read depth</strong> for an assembly based on <strong>pacbio</strong> data.</li>
<li><font color="blue"><strong>src/ngstat</strong></font>: create read depth histogram and base-level read detph for an assembly based on <strong>illumina</strong> data.</li>
<li><font color="blue"><strong>src/calcuts</strong></font>: <strong>calculate coverage cutoffs</strong>.</li>
<li><font color="blue"><strong>src/purge_dups</strong></font>: <strong>purge haplotigs and overlaps</strong> for an <strong>assembly</strong>.</li>
<li><font color="blue"><strong>src/get_seqs</strong></font>: <strong>obtain seqeuences after purging</strong>.</li>
<li>bin/* : all purge_dups excutables.</li>
</ul>
<h1 id="installation"><a href="#installation" class="headerlink" title="installation"></a>installation</h1><figure class="highlight sh"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">git <span class="built_in">clone</span> https://github.com/dfguan/purge_dups.git</span><br><span class="line"><span class="built_in">cd</span> purge_dups/src &amp;&amp; make</span><br><span class="line"><span class="comment">#脚本在scripts目录，编译程序在bin目录</span></span><br></pre></td></tr></table></figure>

<h1 id="Pipeline-Guide"><a href="#Pipeline-Guide" class="headerlink" title="Pipeline Guide"></a>Pipeline Guide</h1><p>Given a primary assembly <em><strong>pri_asm</strong></em> and an alternative assembly <em><strong>hap_asm</strong></em> (<strong>optional</strong>, if you have one), follow the steps shown below to build your own purge_dups pipeline, steps with same number can be run simultaneously. Among all the steps, although <strong>step 5 is optional</strong>, we <strong>highly recommend</strong> our users to do so, because <strong>assemblers may produce overrepresented seqeuences</strong>. In such a case, The final <strong>step 5</strong> can be applied to <strong>remove those seqeuences</strong>.</p>
<h2 id="Step-1-Run-minimap2-to-align-pacbio-data-and-generate-paf-files-then-calculate-read-depth-histogram-and-base-level-read-depth-根据覆盖度计算分界点-cutoff"><a href="#Step-1-Run-minimap2-to-align-pacbio-data-and-generate-paf-files-then-calculate-read-depth-histogram-and-base-level-read-depth-根据覆盖度计算分界点-cutoff" class="headerlink" title="Step 1. Run minimap2 to align pacbio data and generate paf files, then calculate read depth histogram and base-level read depth. 根据覆盖度计算分界点(cutoff)"></a>Step 1. Run minimap2 to align pacbio data and generate paf files, then calculate read depth histogram and base-level read depth. 根据覆盖度计算分界点(cutoff)</h2><p>For PacBio CLR reads</p>
<figure class="highlight sh"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line"><span class="keyword">for</span> i <span class="keyword">in</span> <span class="variable">$pb_list</span></span><br><span class="line"><span class="keyword">do</span></span><br><span class="line">	minimap2 -xmap-pb <span class="variable">$pri_asm</span> <span class="variable">$i</span> | gzip -c - &gt; <span class="variable">$i</span>.paf.gz</span><br><span class="line"><span class="keyword">done</span></span><br><span class="line">bin/pbcstat *.paf.gz (produces PB.base.cov and PB.stat files)</span><br><span class="line">bin/calcuts PB.stat &gt; cutoffs 2&gt;calcults.log</span><br></pre></td></tr></table></figure>

<p><strong>For PacBio CCS reads</strong></p>
<figure class="highlight sh"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># gzip可以替换成pigz, 进行多线程压缩</span></span><br><span class="line"><span class="keyword">for</span> i <span class="keyword">in</span> <span class="variable">$pb_list</span></span><br><span class="line"><span class="keyword">do</span></span><br><span class="line">	minimap2 -xasm20 <span class="variable">$pri_asm</span> <span class="variable">$i</span> | gzip -c - &gt; <span class="variable">$i</span>.paf.gz</span><br><span class="line"><span class="keyword">done</span><span class="comment"># 统计paf, 输出PB.base.cov和PB.stat文件</span></span><br><span class="line">bin/pbcstat *.paf.gz (produces PB.base.cov and PB.stat files)</span><br><span class="line">bin/calcuts PB.stat &gt; cutoffs 2&gt;calcults.log</span><br></pre></td></tr></table></figure>

<p><strong>Notice</strong> If you have a large genome, please set minimap2 <code>-I</code> option to ensure the genome can be indexed once, otherwise read depth can be wrong.</p>
<p>二代测序，用<code>bwa mem</code>进行比对，然后用<code>bin/ngscstat</code>统计bam覆盖度，然后用<code>bin/calcuts</code>计算分界点。</p>
<h2 id="Step-2-Split-an-assembly-and-do-a-self-self-alignment-Commands-are-following"><a href="#Step-2-Split-an-assembly-and-do-a-self-self-alignment-Commands-are-following" class="headerlink" title="Step 2. Split an assembly and do a self-self alignment. Commands are following:"></a>Step 2. Split an assembly and do a self-self alignment. Commands are following:</h2><figure class="highlight sh"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#将assembly【从N处打断】，如果assembly中没有N就不打断，然后用minimap2进行contig自我比对。</span></span><br><span class="line"><span class="comment"># Split an assembly</span></span><br><span class="line">bin/split_fa <span class="variable">$pri_asm</span> &gt; <span class="variable">$pri_asm</span>.split</span><br><span class="line"><span class="comment"># do a self-self alignment</span></span><br><span class="line">minimap2 -xasm5 -DP <span class="variable">$pri_asm</span>.split <span class="variable">$pri_asm</span>.split | gzip -c - &gt; <span class="variable">$pri_asm</span>.split.self.paf.gz</span><br></pre></td></tr></table></figure>

<p>run step1&amp;step2 simultaneously</p>
<h2 id="Step-3-Purge-haplotigs-and-overlaps-with-the-following-command"><a href="#Step-3-Purge-haplotigs-and-overlaps-with-the-following-command" class="headerlink" title="Step 3. Purge haplotigs and overlaps with the following command."></a>Step 3. Purge haplotigs and overlaps with the following command.</h2><figure class="highlight sh"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">根据每个碱基覆盖度以及组装自我比对结果对contig分类</span><br><span class="line"><span class="comment"># purge haplotigs and overlap</span></span><br><span class="line">bin/purge_dups -2 -T cutoffs -c PB.base.cov <span class="variable">$pri_asm</span>.split.self.paf.gz &gt; dups.bed 2&gt; purge_dups.log</span><br></pre></td></tr></table></figure>

<p>dups.bed<strong>第四列</strong>是每个<strong>contig的分类信息</strong>，分为**”JUNK”, “HIGHCOV”, “HAPLOTIG”, “PRIMARY”, “REPEAT”, “OVLP”** 6类，<code>purge_dups</code>可以先以默认参数运行，如果结果不理想，可以调整如下参数</p>
<ul>
<li><code>-f</code>默认是.8, 根据80%区域的覆盖度来对contig进行分类。例如80%的区域都低于5x，将该序列定义为JUNK。对应源码中的<code>classify_seq</code>函数的<code>min_frac</code>参数</li>
<li><code>-a</code>和<code>-b</code>过滤alignment, 对于源码中的<code>flt_by_bm_mm</code>的<code>min_bmf</code>和<code>min_mmf</code>参数</li>
<li><code>-m</code>表示将两个联配衔接时，最低的匹配碱基数</li>
<li><code>-M</code>和<code>-G</code>:分别表示第一轮和第二轮将前后两个联配衔接时最大空缺大小</li>
<li><code>-E</code>表示 如果合并之后的alignment在contig末尾的前15k内，那么就把alignment延伸至contig末尾</li>
<li><code>-l</code>: 用于控制overlap的大小，该值越小，overlap越多</li>
</ul>
<h2 id="Step-4-Get-purged-primary-and-haplotig-sequences-from-draft-assembly"><a href="#Step-4-Get-purged-primary-and-haplotig-sequences-from-draft-assembly" class="headerlink" title="Step 4. Get purged primary and haplotig sequences from draft assembly."></a>Step 4. Get purged primary and haplotig sequences from draft assembly.</h2><figure class="highlight sh"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">bin/get_seqs -e dups.bed <span class="variable">$pri_asm</span> </span><br><span class="line"><span class="comment"># If you also want to remove the duplications in the middle, please remove 【`-e`】 option at your own risk, it may delete false positive duplications. For more options, please refer to `get_seqs -h`.</span></span><br></pre></td></tr></table></figure>

<p>这里的purged.fa就是最终结果，junk, haplotig和duplication都会在hap.fa中。</p>
<p>可选步骤: 将<strong>alternative assembly和输出hap.fa进行合并</strong>，然后运行上面四步，得到的<strong>purge.fa就是新的alternative assembly</strong>，而<strong>再次输出的hap.fa则是junk或overrepresented序列</strong>。</p>
<p>PS: 能不能用来过滤纯合基因组组装的垃圾序列呢？根据物种测试，过滤前后的BUSCO值，几乎没有变化，missing rate只提高了0.1%，</p>
<figure class="highlight sh"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># 运行前</span></span><br><span class="line">C:98.8%[S:96.3%,D:2.5%],F:0.5%,M:0.7%,n:1375</span><br><span class="line"><span class="comment"># 运行后</span></span><br><span class="line">C:98.7%[S:96.3%,D:2.4%],F:0.5%,M:0.8%,n:1375</span><br></pre></td></tr></table></figure>

<p>用法可行，且Canu的作者建议用<code>purge_dups</code>处理，参考<a target="_blank" rel="noopener" href="https://links.jianshu.com/go?to=https://github.com/marbl/canu/issues/1717%23issuecomment-629644894">canu-issues-1717</a>。</p>
<p>作者尚未在ONT和Illumina数据中测试软件，但是作者认为只需要修改<code>minimap2</code>的<code>-x map-pb</code>为<code>-x map-ont</code>就可以用在ONT数据上。</p>
<p><strong>Notice</strong> this command will only remove haplotypic duplications at the ends of the contigs. If you also want to remove the duplications in the middle, please remove <code>-e</code> option at your own risk, it may delete false positive duplications. For more options, please refer to <code>get_seqs -h</code>.</p>
<h2 id="Step-5-Merge-hap-fa-and-hap-asm-and-redo-the-above-steps-to-get-a-decent-haplotig-set"><a href="#Step-5-Merge-hap-fa-and-hap-asm-and-redo-the-above-steps-to-get-a-decent-haplotig-set" class="headerlink" title="Step 5. Merge hap.fa and $hap_asm and redo the above steps to get a decent haplotig set."></a>Step 5. Merge hap.fa and $hap_asm and redo the above steps to get a decent haplotig set.</h2><h1 id="Limitation"><a href="#Limitation" class="headerlink" title="Limitation"></a>Limitation</h1><ul>
<li>Read depth cutoffs calculation: the coverage cutoffs can be larger for a low heterozygosity species, which causes the purged assembly size smaller than expected. In such a case, please use script/hist_plot.py to make the histogram plot and set coverage cutoffs manually.</li>
<li>Repeats: purge_dups has a limited ability to process repeats.</li>
</ul>
<h1 id="FAQ"><a href="#FAQ" class="headerlink" title="FAQ"></a>FAQ</h1><p><strong>Q:</strong> Can I use purge_dups with <strong>short reads</strong>?</p>
<p><strong>A:</strong> Yes, purge_dups does have a program to process Illumina reads, it’s called <strong>ngscstat</strong> under the bin directory. But I have not got time to test it. If you want to play with it, please follow this workflow:</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">bwa mem $pri_asm $sr_1.fq $sr_2.fq | samtools view -b -o - &gt; $sr.bam </span><br><span class="line">ngscstat $sr.bam... # The program will generate two/three outputs, TX.stat and TX.base.cov which functions the same way as PB.stat and PB.base.cov respectively.  </span><br></pre></td></tr></table></figure>

<p>After you get the TX.stat and TX.base.cov file, you can following the normal purge_dups routine to clean your assembly.</p>
<p><strong>Q1:</strong> Can I validate the cutoffs used by purge_dups?</p>
<p><strong>A1:</strong> Yes, we also recommend this step. A script “hist_plot.py” under the scripts directory is available, you can also use it to manually select the cutoffs.</p>
<p><strong>Q2:</strong> How can I <strong>validate the purged assembly?</strong> Is it clean enough or overpurged?</p>
<p><strong>A2:</strong> There are many ways to validate the purged assembly. One way is to make a coverage plot for it which can also be <strong>hist_plot.py</strong>, the 2nd way is to run <strong>BUSCO</strong> and another way is to make a KAT plot with KAT (<a target="_blank" rel="noopener" href="https://github.com/TGAC/KAT">https://github.com/TGAC/KAT</a>) or KMC (<a target="_blank" rel="noopener" href="https://github.com/dfguan/KMC">https://github.com/dfguan/KMC</a>, use this if you only have a small memory machine) if short reads or some accurate reads are available.</p>
<p><strong>Q3:</strong> Why do I get <strong>much fewer haplotypic duplications</strong> than expected?</p>
<p><strong>A3:</strong> First check the original contig names, they should not contain any colons. Then check the cutoffs, if purge_dups automatically use <strong>a fairly low read depth for haplotypic duplications</strong>, it may <strong>remove nothing</strong>. In this case, you <strong>need to set the cutoffs manually.</strong></p>
<p><strong>Q4:</strong> why does purge_dups remove <strong>middle sequence in a contig?</strong></p>
<p><strong>A4:</strong> <strong>Some of them are real, while others may not</strong>. We are currently investigating them. Please use <code>-e</code> for <code>get_seqs</code> command if you <strong>only want to remove the duplications at the ends of the contigs.</strong></p>
<hr>
<p>Ref:<a target="_blank" rel="noopener" href="https://www.jianshu.com/p/e218a1192d12">https://www.jianshu.com/p/e218a1192d12</a></p>

    </div>

    
    
    
        

<div>
<ul class="post-copyright">
  <li class="post-copyright-author">
    <strong>Post author:  </strong>Wang C
  </li>
  <li class="post-copyright-link">
    <strong>Post link: </strong>
    <a href="https://wangc1106.github.io/blog/2022/03/18/2022-03-18-purgedups/" title="【hifi mapping】【purge_dups】">https://wangc1106.github.io/blog/2022/03/18/2022-03-18-purgedups/</a>
  </li>
  <li class="post-copyright-license">
    <strong>Copyright Notice:  </strong>All articles in this blog are licensed under <a href="https://creativecommons.org/licenses/by-nc-sa/4.0/" rel="noopener" target="_blank"><i class="fab fa-fw fa-creative-commons"></i>BY-NC-SA</a> unless stating additionally.
  </li>
</ul>
</div>


      <footer class="post-footer">
          <div class="post-tags">
              <a href="/blog/tags/denovo-III/" rel="tag"># denovo-III</a>
          </div>

        


        
    <div class="post-nav">
      <div class="post-nav-item">
    <a href="/blog/2022/03/17/2022-03-17-juicebox/" rel="prev" title="【Hi-C】【Juicebox】">
      <i class="fa fa-chevron-left"></i> 【Hi-C】【Juicebox】
    </a></div>
      <div class="post-nav-item">
    <a href="/blog/2022/03/18/2022-03-18-mito-chro/" rel="next" title="Organelle assembly">
      Organelle assembly <i class="fa fa-chevron-right"></i>
    </a></div>
    </div>
      </footer>
    
  </article>
  
  
  



          </div>
          

<script>
  window.addEventListener('tabs:register', () => {
    let { activeClass } = CONFIG.comments;
    if (CONFIG.comments.storage) {
      activeClass = localStorage.getItem('comments_active') || activeClass;
    }
    if (activeClass) {
      let activeTab = document.querySelector(`a[href="#comment-${activeClass}"]`);
      if (activeTab) {
        activeTab.click();
      }
    }
  });
  if (CONFIG.comments.storage) {
    window.addEventListener('tabs:click', event => {
      if (!event.target.matches('.tabs-comment .tab-content .tab-pane')) return;
      let commentClass = event.target.classList[1];
      localStorage.setItem('comments_active', commentClass);
    });
  }
</script>

        </div>
          
  
  <div class="toggle sidebar-toggle">
    <span class="toggle-line toggle-line-first"></span>
    <span class="toggle-line toggle-line-middle"></span>
    <span class="toggle-line toggle-line-last"></span>
  </div>

  <aside class="sidebar">
    <div class="sidebar-inner">

      <ul class="sidebar-nav motion-element">
        <li class="sidebar-nav-toc">
          Table of Contents
        </li>
        <li class="sidebar-nav-overview">
          Overview
        </li>
      </ul>

      <!--noindex-->
      <div class="post-toc-wrap sidebar-panel">
          <div class="post-toc motion-element"><ol class="nav"><li class="nav-item nav-level-1"><a class="nav-link" href="#overview"><span class="nav-number">1.</span> <span class="nav-text">overview</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#Directory-Structure"><span class="nav-number">2.</span> <span class="nav-text">Directory Structure</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#installation"><span class="nav-number">3.</span> <span class="nav-text">installation</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#Pipeline-Guide"><span class="nav-number">4.</span> <span class="nav-text">Pipeline Guide</span></a><ol class="nav-child"><li class="nav-item nav-level-2"><a class="nav-link" href="#Step-1-Run-minimap2-to-align-pacbio-data-and-generate-paf-files-then-calculate-read-depth-histogram-and-base-level-read-depth-%E6%A0%B9%E6%8D%AE%E8%A6%86%E7%9B%96%E5%BA%A6%E8%AE%A1%E7%AE%97%E5%88%86%E7%95%8C%E7%82%B9-cutoff"><span class="nav-number">4.1.</span> <span class="nav-text">Step 1. Run minimap2 to align pacbio data and generate paf files, then calculate read depth histogram and base-level read depth. 根据覆盖度计算分界点(cutoff)</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#Step-2-Split-an-assembly-and-do-a-self-self-alignment-Commands-are-following"><span class="nav-number">4.2.</span> <span class="nav-text">Step 2. Split an assembly and do a self-self alignment. Commands are following:</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#Step-3-Purge-haplotigs-and-overlaps-with-the-following-command"><span class="nav-number">4.3.</span> <span class="nav-text">Step 3. Purge haplotigs and overlaps with the following command.</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#Step-4-Get-purged-primary-and-haplotig-sequences-from-draft-assembly"><span class="nav-number">4.4.</span> <span class="nav-text">Step 4. Get purged primary and haplotig sequences from draft assembly.</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#Step-5-Merge-hap-fa-and-hap-asm-and-redo-the-above-steps-to-get-a-decent-haplotig-set"><span class="nav-number">4.5.</span> <span class="nav-text">Step 5. Merge hap.fa and $hap_asm and redo the above steps to get a decent haplotig set.</span></a></li></ol></li><li class="nav-item nav-level-1"><a class="nav-link" href="#Limitation"><span class="nav-number">5.</span> <span class="nav-text">Limitation</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#FAQ"><span class="nav-number">6.</span> <span class="nav-text">FAQ</span></a></li></ol></div>
      </div>
      <!--/noindex-->

      <div class="site-overview-wrap sidebar-panel">
        <div class="site-author motion-element" itemprop="author" itemscope itemtype="http://schema.org/Person">
    <img class="site-author-image" itemprop="image" alt="Wang C"
      src="/blog/images/banner.png">
  <p class="site-author-name" itemprop="name">Wang C</p>
  <div class="site-description" itemprop="description">Make more time</div>
</div>
<div class="site-state-wrap motion-element">
  <nav class="site-state">
      <div class="site-state-item site-state-posts">
          <a href="/blog/archives">
          <span class="site-state-item-count">42</span>
          <span class="site-state-item-name">posts</span>
        </a>
      </div>
      <div class="site-state-item site-state-categories">
            <a href="/blog/categories/">
          
        <span class="site-state-item-count">12</span>
        <span class="site-state-item-name">categories</span></a>
      </div>
      <div class="site-state-item site-state-tags">
            <a href="/blog/tags/">
          
        <span class="site-state-item-count">19</span>
        <span class="site-state-item-name">tags</span></a>
      </div>
  </nav>
</div>
  <div class="links-of-author motion-element">
      <span class="links-of-author-item">
        <a href="https://github.com/wangc1106" title="GitHub → https:&#x2F;&#x2F;github.com&#x2F;wangc1106" rel="noopener" target="_blank"><i class="fab fa-github fa-fw"></i>GitHub</a>
      </span>
      <span class="links-of-author-item">
        <a href="mailto:wangchen116@126.com" title="E-Mail → mailto:wangchen116@126.com" rel="noopener" target="_blank"><i class="fa fa-envelope fa-fw"></i>E-Mail</a>
      </span>
  </div>
  <div class="cc-license motion-element" itemprop="license">
    <a href="https://creativecommons.org/licenses/by-nc-sa/4.0/" class="cc-opacity" rel="noopener" target="_blank"><img src="/blog/images/cc-by-nc-sa.svg" alt="Creative Commons"></a>
  </div>



      </div>
        <div class="back-to-top motion-element">
          <i class="fa fa-arrow-up"></i>
          <span>0%</span>
        </div>

    </div>
  </aside>
  <div id="sidebar-dimmer"></div>


      </div>
    </main>

    <footer class="footer">
      <div class="footer-inner">
        

        

<div class="copyright">
  
  &copy; 2021 – 
  <span itemprop="copyrightYear">2022</span>
  <span class="with-love">
    <i class="fas fa-dna"></i>
  </span>
  <span class="author" itemprop="copyrightHolder">Wang C</span>
</div>

        






<script>
  (function() {
    function leancloudSelector(url) {
      url = encodeURI(url);
      return document.getElementById(url).querySelector('.leancloud-visitors-count');
    }

    function addCount(Counter) {
      var visitors = document.querySelector('.leancloud_visitors');
      var url = decodeURI(visitors.id);
      var title = visitors.dataset.flagTitle;

      Counter('get', '/classes/Counter?where=' + encodeURIComponent(JSON.stringify({ url })))
        .then(response => response.json())
        .then(({ results }) => {
          if (results.length > 0) {
            var counter = results[0];
            leancloudSelector(url).innerText = counter.time + 1;
            Counter('put', '/classes/Counter/' + counter.objectId, { time: { '__op': 'Increment', 'amount': 1 } })
              .catch(error => {
                console.error('Failed to save visitor count', error);
              });
          } else {
              Counter('post', '/classes/Counter', { title, url, time: 1 })
                .then(response => response.json())
                .then(() => {
                  leancloudSelector(url).innerText = 1;
                })
                .catch(error => {
                  console.error('Failed to create', error);
                });
          }
        })
        .catch(error => {
          console.error('LeanCloud Counter Error', error);
        });
    }

    function showTime(Counter) {
      var visitors = document.querySelectorAll('.leancloud_visitors');
      var entries = [...visitors].map(element => {
        return decodeURI(element.id);
      });

      Counter('get', '/classes/Counter?where=' + encodeURIComponent(JSON.stringify({ url: { '$in': entries } })))
        .then(response => response.json())
        .then(({ results }) => {
          for (let url of entries) {
            let target = results.find(item => item.url === url);
            leancloudSelector(url).innerText = target ? target.time : 0;
          }
        })
        .catch(error => {
          console.error('LeanCloud Counter Error', error);
        });
    }

    let { app_id, app_key, server_url } = {"enable":true,"app_id":"UqIYufmNk2sE8MiBKKWc4Kja-gzGzoHsz","app_key":"iYpeWQcmNqpHwr2uVlnJYVji","server_url":null,"security":false};
    function fetchData(api_server) {
      var Counter = (method, url, data) => {
        return fetch(`${api_server}/1.1${url}`, {
          method,
          headers: {
            'X-LC-Id'     : app_id,
            'X-LC-Key'    : app_key,
            'Content-Type': 'application/json',
          },
          body: JSON.stringify(data)
        });
      };
      if (CONFIG.page.isPost) {
        if (CONFIG.hostname !== location.hostname) return;
        addCount(Counter);
      } else if (document.querySelectorAll('.post-title-link').length >= 1) {
        showTime(Counter);
      }
    }

    let api_server = app_id.slice(-9) !== '-MdYXbMMI' ? server_url : `https://${app_id.slice(0, 8).toLowerCase()}.api.lncldglobal.com`;

    if (api_server) {
      fetchData(api_server);
    } else {
      fetch('https://app-router.leancloud.cn/2/route?appId=' + app_id)
        .then(response => response.json())
        .then(({ api_server }) => {
          fetchData('https://' + api_server);
        });
    }
  })();
</script>


      </div>
      <div style="height: 200px">
      <script type="text/javascript" src="//rf.revolvermaps.com/0/0/8.js?i=5knf34gs07t&amp;m=0&amp;c=ff0000&amp;cr1=ffffff&amp;f=arial&amp;l=33" async="async"></script>
      </div>
    </footer>
  </div>

  
  <script src="/blog/lib/anime.min.js"></script>
  <script src="/blog/lib/velocity/velocity.min.js"></script>
  <script src="/blog/lib/velocity/velocity.ui.min.js"></script>

<script src="/blog/js/utils.js"></script>

<script src="/blog/js/motion.js"></script>


<script src="/blog/js/schemes/pisces.js"></script>


<script src="/blog/js/next-boot.js"></script>




  




  
<script src="/blog/js/local-search.js"></script>













  

  

</body>
</html>
