<!DOCTYPE html>
<html lang="zh-Hans">
<head>
  <meta charset="UTF-8">
<meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=2">
<meta name="theme-color" content="#222">
<meta name="generator" content="Hexo 5.4.0">
  <link rel="apple-touch-icon" sizes="180x180" href="/blog/images/apple-touch-icon-next.png">
  <link rel="icon" type="image/png" sizes="32x32" href="/blog/images/favicon-32x32-next.png">
  <link rel="icon" type="image/png" sizes="16x16" href="/blog/images/favicon-16x16-next.png">
  <link rel="mask-icon" href="/blog/images/logo.svg" color="#222">
  <meta name="google-site-verification" content="1osTo_gOe1SW9Hevqtz8vl8HfSgTO4DSMZ0DiMPMX6w">

<link rel="stylesheet" href="/blog/css/main.css">


<link rel="stylesheet" href="/blog/lib/font-awesome/css/all.min.css">

<script id="hexo-configurations">
    var NexT = window.NexT || {};
    var CONFIG = {"hostname":"wangc1106.github.io","root":"/blog/","scheme":"Gemini","version":"7.8.0","exturl":false,"sidebar":{"position":"left","display":"post","padding":18,"offset":12,"onmobile":false},"copycode":{"enable":false,"show_result":false,"style":null},"back2top":{"enable":true,"sidebar":true,"scrollpercent":"ture"},"bookmark":{"enable":false,"color":"#222","save":"auto"},"fancybox":false,"mediumzoom":false,"lazyload":false,"pangu":false,"comments":{"style":"tabs","active":null,"storage":true,"lazyload":false,"nav":null},"algolia":{"hits":{"per_page":10},"labels":{"input_placeholder":"Search for Posts","hits_empty":"We didn't find any results for the search: ${query}","hits_stats":"${hits} results found in ${time} ms"}},"localsearch":{"enable":true,"trigger":"auto","top_n_per_article":1,"unescape":false,"preload":false},"motion":{"enable":true,"async":false,"transition":{"post_block":"fadeIn","post_header":"slideDownIn","post_body":"slideDownIn","coll_header":"slideLeftIn","sidebar":"slideUpIn"}},"path":"search.xml"};
  </script>

  <meta name="description" content="https:&#x2F;&#x2F;github.com&#x2F;hbctraining&#x2F;In-depth-NGS-Data-Analysis-Course&#x2F;tree&#x2F;master&#x2F;sessionV&#x2F;lessons">
<meta property="og:type" content="article">
<meta property="og:title" content="ATAC-seq-3【hbctraining_pipeline】">
<meta property="og:url" content="https://wangc1106.github.io/blog/2021/10/26/ATAC-seq-3-hbctraining/index.html">
<meta property="og:site_name" content="Banner&#39;s Blog">
<meta property="og:description" content="https:&#x2F;&#x2F;github.com&#x2F;hbctraining&#x2F;In-depth-NGS-Data-Analysis-Course&#x2F;tree&#x2F;master&#x2F;sessionV&#x2F;lessons">
<meta property="og:locale">
<meta property="og:image" content="https://img-blog.csdnimg.cn/f2741798d4f04f708412c97b2e6ef045.png?x-oss-process=image/watermark,type_ZHJvaWRzYW5zZmFsbGJhY2s,shadow_50,text_Q1NETiBAYmFubmVyRHI=,size_20,color_FFFFFF,t_70,g_se,x_16">
<meta property="og:image" content="https://img-blog.csdnimg.cn/a5f4429afb3348be91f68d90d4011065.png?x-oss-process=image/watermark,type_ZHJvaWRzYW5zZmFsbGJhY2s,shadow_50,text_Q1NETiBAYmFubmVyRHI=,size_20,color_FFFFFF,t_70,g_se,x_16">
<meta property="og:image" content="https://img-blog.csdnimg.cn/3929b0f60a01477ea09402760205f200.png?x-oss-process=image/watermark,type_ZHJvaWRzYW5zZmFsbGJhY2s,shadow_50,text_Q1NETiBAYmFubmVyRHI=,size_17,color_FFFFFF,t_70,g_se,x_16">
<meta property="og:image" content="https://img-blog.csdnimg.cn/eeefe600965542478444f761de31db17.png?x-oss-process=image/watermark,type_ZHJvaWRzYW5zZmFsbGJhY2s,shadow_50,text_Q1NETiBAYmFubmVyRHI=,size_10,color_FFFFFF,t_70,g_se,x_16">
<meta property="og:image" content="https://img-blog.csdnimg.cn/5571c4b228cb4706805ba0853c546c3c.png">
<meta property="og:image" content="https://img-blog.csdnimg.cn/eaac9c4b4c294be3b8b3b3caf7c35a21.png">
<meta property="og:image" content="https://img-blog.csdnimg.cn/3c7c645b299e45b2b7975d6327d421bc.png">
<meta property="og:image" content="https://img-blog.csdnimg.cn/7bd1ca4fcb3847daacc450a91989b5cd.png?x-oss-process=image/watermark,type_ZHJvaWRzYW5zZmFsbGJhY2s,shadow_50,text_Q1NETiBAYmFubmVyRHI=,size_20,color_FFFFFF,t_70,g_se,x_16">
<meta property="og:image" content="https://img-blog.csdnimg.cn/3d941b11b37c4b7687b11e6766a9daff.png?x-oss-process=image/watermark,type_ZHJvaWRzYW5zZmFsbGJhY2s,shadow_50,text_Q1NETiBAYmFubmVyRHI=,size_13,color_FFFFFF,t_70,g_se,x_16">
<meta property="article:published_time" content="2021-10-26T02:07:16.720Z">
<meta property="article:modified_time" content="2021-10-30T07:28:35.752Z">
<meta property="article:author" content="Wang C">
<meta property="article:tag" content="ATAC-seq">
<meta name="twitter:card" content="summary">
<meta name="twitter:image" content="https://img-blog.csdnimg.cn/f2741798d4f04f708412c97b2e6ef045.png?x-oss-process=image/watermark,type_ZHJvaWRzYW5zZmFsbGJhY2s,shadow_50,text_Q1NETiBAYmFubmVyRHI=,size_20,color_FFFFFF,t_70,g_se,x_16">

<link rel="canonical" href="https://wangc1106.github.io/blog/2021/10/26/ATAC-seq-3-hbctraining/">


<script id="page-configurations">
  // https://hexo.io/docs/variables.html
  CONFIG.page = {
    sidebar: "",
    isHome : false,
    isPost : true,
    lang   : 'zh-Hans'
  };
</script>

  <title>ATAC-seq-3【hbctraining_pipeline】 | Banner's Blog</title>
  






  <noscript>
  <style>
  .use-motion .brand,
  .use-motion .menu-item,
  .sidebar-inner,
  .use-motion .post-block,
  .use-motion .pagination,
  .use-motion .comments,
  .use-motion .post-header,
  .use-motion .post-body,
  .use-motion .collection-header { opacity: initial; }

  .use-motion .site-title,
  .use-motion .site-subtitle {
    opacity: initial;
    top: initial;
  }

  .use-motion .logo-line-before i { left: initial; }
  .use-motion .logo-line-after i { right: initial; }
  </style>
</noscript>

</head>

<body itemscope itemtype="http://schema.org/WebPage">
  <div class="container use-motion">
    <div class="headband"></div>

    <header class="header" itemscope itemtype="http://schema.org/WPHeader">
      <div class="header-inner"><div class="site-brand-container">
  <div class="site-nav-toggle">
    <div class="toggle" aria-label="Toggle navigation bar">
      <span class="toggle-line toggle-line-first"></span>
      <span class="toggle-line toggle-line-middle"></span>
      <span class="toggle-line toggle-line-last"></span>
    </div>
  </div>

  <div class="site-meta">

    <a href="/blog/" class="brand" rel="start">
      <span class="logo-line-before"><i></i></span>
      <h1 class="site-title">Banner's Blog</h1>
      <span class="logo-line-after"><i></i></span>
    </a>
      <p class="site-subtitle" itemprop="description">Dream big work hard</p>
  </div>

  <div class="site-nav-right">
    <div class="toggle popup-trigger">
        <i class="fa fa-search fa-fw fa-lg"></i>
    </div>
  </div>
</div>




<nav class="site-nav">
  <ul id="menu" class="main-menu menu">
        <li class="menu-item menu-item-home">

    <a href="/blog/" rel="section"><i class="home fa-fw"></i>Home</a>

  </li>
        <li class="menu-item menu-item-categories">

    <a href="/blog/categories/" rel="section"><i class="th fa-fw"></i>Categories</a>

  </li>
        <li class="menu-item menu-item-tags">

    <a href="/blog/tags/" rel="section"><i class="tags fa-fw"></i>Tags</a>

  </li>
        <li class="menu-item menu-item-about">

    <a href="https://wangc1106.github.io/" rel="section"><i class="user fa-fw"></i>About</a>

  </li>
      <li class="menu-item menu-item-search">
        <a role="button" class="popup-trigger"><i class="fa fa-search fa-fw"></i>Search
        </a>
      </li>
  </ul>
</nav>



  <div class="search-pop-overlay">
    <div class="popup search-popup">
        <div class="search-header">
  <span class="search-icon">
    <i class="fa fa-search"></i>
  </span>
  <div class="search-input-container">
    <input autocomplete="off" autocapitalize="off"
           placeholder="Searching..." spellcheck="false"
           type="search" class="search-input">
  </div>
  <span class="popup-btn-close">
    <i class="fa fa-times-circle"></i>
  </span>
</div>
<div id="search-result">
  <div id="no-result">
    <i class="fa fa-spinner fa-pulse fa-5x fa-fw"></i>
  </div>
</div>

    </div>
  </div>

</div>
    </header>

    


    <main class="main">
      <div class="main-inner">
        <div class="content-wrap">
          

          <div class="content post posts-expand">
            

    
  
  
  <article itemscope itemtype="http://schema.org/Article" class="post-block" lang="zh-Hans">
    <link itemprop="mainEntityOfPage" href="https://wangc1106.github.io/blog/2021/10/26/ATAC-seq-3-hbctraining/">

    <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
      <meta itemprop="image" content="/blog/images/banner.png">
      <meta itemprop="name" content="Wang C">
      <meta itemprop="description" content="Make more time">
    </span>

    <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
      <meta itemprop="name" content="Banner's Blog">
    </span>
      <header class="post-header">
        <h1 class="post-title" itemprop="name headline">
          ATAC-seq-3【hbctraining_pipeline】
        </h1>

        <div class="post-meta">
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-calendar"></i>
              </span>
              <span class="post-meta-item-text">Posted on</span>

              <time title="Created: 2021-10-26 10:07:16" itemprop="dateCreated datePublished" datetime="2021-10-26T10:07:16+08:00">2021-10-26</time>
            </span>
              <span class="post-meta-item">
                <span class="post-meta-item-icon">
                  <i class="far fa-calendar-check"></i>
                </span>
                <span class="post-meta-item-text">Edited on</span>
                <time title="Modified: 2021-10-30 15:28:35" itemprop="dateModified" datetime="2021-10-30T15:28:35+08:00">2021-10-30</time>
              </span>
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-folder"></i>
              </span>
              <span class="post-meta-item-text">In</span>
                <span itemprop="about" itemscope itemtype="http://schema.org/Thing">
                  <a href="/blog/categories/ATAC-seq/" itemprop="url" rel="index"><span itemprop="name">ATAC-seq</span></a>
                </span>
            </span>

          
            <span id="/blog/2021/10/26/ATAC-seq-3-hbctraining/" class="post-meta-item leancloud_visitors" data-flag-title="ATAC-seq-3【hbctraining_pipeline】" title="Views">
              <span class="post-meta-item-icon">
                <i class="fa fa-eye"></i>
              </span>
              <span class="post-meta-item-text">Views: </span>
              <span class="leancloud-visitors-count"></span>
            </span>
            <span class="post-meta-item" title="Symbols count in article">
              <span class="post-meta-item-icon">
                <i class="far fa-file-word"></i>
              </span>
              <span>11k</span>
            </span>
            <span class="post-meta-item" title="Reading time">
              <span class="post-meta-item-icon">
                <i class="far fa-clock"></i>
              </span>
              <span>10 mins.</span>
            </span>

        </div>
      </header>

    
    
    
    <div class="post-body" itemprop="articleBody">

      
        <h2 id="https-github-com-hbctraining-In-depth-NGS-Data-Analysis-Course-tree-master-sessionV-lessons"><a href="#https-github-com-hbctraining-In-depth-NGS-Data-Analysis-Course-tree-master-sessionV-lessons" class="headerlink" title="https://github.com/hbctraining/In-depth-NGS-Data-Analysis-Course/tree/master/sessionV/lessons"></a><a target="_blank" rel="noopener" href="https://github.com/hbctraining/In-depth-NGS-Data-Analysis-Course/tree/master/sessionV/lessons">https://github.com/hbctraining/In-depth-NGS-Data-Analysis-Course/tree/master/sessionV/lessons</a></h2><span id="more"></span>

<p>算法，参数，输出。</p>
<h2 id="Peak-calling"><a href="#Peak-calling" class="headerlink" title="Peak calling"></a>Peak calling</h2><p><img src="https://img-blog.csdnimg.cn/f2741798d4f04f708412c97b2e6ef045.png?x-oss-process=image/watermark,type_ZHJvaWRzYW5zZmFsbGJhY2s,shadow_50,text_Q1NETiBAYmFubmVyRHI=,size_20,color_FFFFFF,t_70,g_se,x_16" alt="在这里插入图片描述"><br>ChIP-seq实验，从比对文件中观察到<code>正/负链</code>上以<code>结合位点为中心</code>的非对称reads<br>密度。<br>For ChIP-seq experiments, what we observe from the alignment files is a strand asymmetry with read densities on the +/- strand, centered around the binding site. <code>The 5’ ends of the selected fragments</code> will form <code>groups</code> on the positive- and negative-strand. The <code>distributions</code> of these groups are then assessed using <code>statistical measures</code> and <code>compared against background</code> (input or mock IP samples) to determine if the site of enrichment is likely to be a real binding site.</p>
<p><img src="https://img-blog.csdnimg.cn/a5f4429afb3348be91f68d90d4011065.png?x-oss-process=image/watermark,type_ZHJvaWRzYW5zZmFsbGJhY2s,shadow_50,text_Q1NETiBAYmFubmVyRHI=,size_20,color_FFFFFF,t_70,g_se,x_16" alt="加粗样式"><br><code>ChIP-seq analysis algorithms</code> are specialized in identifying one of <code>two types of enrichment</code> (or have specific methods for each): <code>broad peaks or broad domains</code> (i.e. <code>histone modifications that cover entire gene bodies</code>) or <code>narrow peaks</code> ( <code>transcription factor binding</code>). <code>Narrow peaks are easier to detect</code> as we are looking for regions that have higher amplitude and are easier to distinguish from the background, compared to broad or dispersed marks. There are also <code>‘mixed’ binding profiles</code> which can be hard for algorithms to discern. An example of this is the binding properties of <code>PolII</code>, which binds at promotor and across the length of the gene resulting in mixed signals (<code>narrow and broad</code>).</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">NOTE：ChIP-seq的分析方法可以鉴定两种类型的富集模式：broad domains和narrow peaks。broad domains，如组蛋白修饰在整个基因body区域的分布；narrow peak，如转录因子的结合。narrow peak相对于broad 或者分散的marks更易被检测到。也有一些混合的结合图谱，如PolII包括narrow和broad信号。</span><br></pre></td></tr></table></figure>
<h2 id="MACS2"><a href="#MACS2" class="headerlink" title="MACS2"></a>MACS2</h2><p>MACS2是最常用的call peaks工具。 The MACS algorithm captures the influence of <code>genome complexity</code> to evaluate the significance of enriched ChIP regions。全称<code>Model-based Analysis of ChIP-Seq</code>，最初设计用来<code>鉴定转录因子结合位点</code>（also suited for <code>larger</code> regions），也可用于<code>其他类型</code>的富集方式测序。<br>MACS通过整合<code>序列标签位置信息</code>和<code>方向</code>信息提高<code>结合位点</code>的<code>空间分辨率</code>（ <code>improves</code> the <code>spatial resolution of binding sites</code> through <code>combining the information of both sequencing tag position and orientation</code>）。单独使用或加对照（ <code>increases specificity</code> of the peak calls）。<br><img src="https://img-blog.csdnimg.cn/3929b0f60a01477ea09402760205f200.png?x-oss-process=image/watermark,type_ZHJvaWRzYW5zZmFsbGJhY2s,shadow_50,text_Q1NETiBAYmFubmVyRHI=,size_17,color_FFFFFF,t_70,g_se,x_16" alt="在这里插入图片描述"></p>
<h3 id="1-Remove-redundancy"><a href="#1-Remove-redundancy" class="headerlink" title="1. Remove redundancy"></a>1. Remove redundancy</h3><p><code>Why worry about duplicates?</code> <code>Reads with the same start position</code> are <code>considered duplicates</code>. These duplicates can arise from experimental artifacts, but can also contribute to genuine ChIP-signal. (相同起点的reads被认为是duplicates，可能由于实验误差造成，<code>也可能是ChIP信号</code>)</p>
<ul>
<li><code>坏 duplicates</code>: If initial starting <code>material is low</code> this can lead to <code>overamplification</code> of this material before sequencing. Any <code>biases in PCR</code> will compound this problem and can lead to <code>artificially enriched regions</code>. Also <code>blacklisted (repeat) regions</code> with ultra high signal will also be high in duplicates. <code>Masking these regions</code> prior to analysis can help remove this problem。 实验材料量少，过度扩增，PCR偏差，人为富集区域。<code>blacklist（重复）区域？怎么获得</code>，屏蔽该区域。</li>
<li><code>好 duplicates</code>: You can expect some <code>biological duplicates with ChIP-seq</code> since you are only sequencing a small <code>part of the genome</code>. This number can increase if your depth of coverage is excessive or if your protein only binds to few sites. If there are a good proportion of biological dupicates, <code>removal</code> can lead to an <code>underestimation of the ChIP signal</code>. 有ChIP-seq意义的生物学意义duplicates ，测序基因组小部分。覆盖深度过多，蛋白质与少数位点结合，duplicates会增加。去除此类会导致对ChIP信号的低估。</li>
<li>Consider your <code>enrichment efficiency</code> and <code>sequencing depth</code>. Try to <code>discriminate</code> via genome browser of your non-deduplicated data. Bona fide peaks will have <code>multiple overlapping</code> reads with <code>offsets</code>, while samples with only PCR duplicates will stack up perfectly without offsets. A possible solution to distinguishing biological duplicate from PCR artifact would be to include UMIs into your experimental setup. 考虑<code>富集效率</code>和<code>测序深度</code>。通过<code>基因组浏览器区别</code>。<code>真正的峰</code>会有多个<code>重叠reads和偏移量</code>，<code>PCR重复没有偏移量</code>。如何区分：实验设计考虑<a target="_blank" rel="noopener" href="https://www.sohu.com/a/471483238_120055884">UMIs</a>（<code>UMI将被用于合并PCR复制物</code>）。</li>
<li>Retain duplicates for differential binding analysis. 保留duplicates 用于差异结合分析 <code>why</code>？retain和keep区别？</li>
<li>If you are expecting binding in repetitive regions, use paired-end sequencing and keep duplicates. 研究重复区域，使用pariedend测序并<code>保持重复？</code></li>
<li>call peak之前就remove dup<h3 id="2-0-Shift-size-d"><a href="#2-0-Shift-size-d" class="headerlink" title="2.0 Shift size d"></a>2.0 <code>Shift size d</code></h3></li>
<li>真实结合位点周围的tag density应显示<code>双峰富集</code>(<code>成对峰</code>)。MACS利用这种双峰模式 <code>empirically model the shifting size</code>，精确定位结合位点。</li>
<li>为了找到<code>成对峰</code>来<code>构建模型</code>，MACS首先<code>扫描整个数据集</code>，寻找<code>高度显著富集区域</code>。只利用ChIP样本，给定超声大小sonication size(<code>bandwidth</code>)和high-confidence fold-enrichment(<code>mfold</code>)， MACS <code>slides two bandwidth windows</code> across the genome to find <code>regions with tags more than mfold enriched</code> relative to a <code>random tag genome distribution</code>.</li>
<li>MACS<code>随机抽取1000个高质量峰</code>，<code>分离正链和负链标签</code>，aligns them by the midpoint between their centers。The <code>distance between the modes of the two peaks in the alignment is defined as ‘d’</code> and represents the estimated fragment length. MACS shifts all the tags by d/2 toward the 3’ ends to the most likely protein-DNA interaction sites.<br><img src="https://img-blog.csdnimg.cn/eeefe600965542478444f761de31db17.png?x-oss-process=image/watermark,type_ZHJvaWRzYW5zZmFsbGJhY2s,shadow_50,text_Q1NETiBAYmFubmVyRHI=,size_10,color_FFFFFF,t_70,g_se,x_16" alt="在这里插入图片描述"></li>
</ul>
<h3 id="2-Select-1000-regions-with-a-10-to-30-fold-enrichment-relative-to-the-genome-background"><a href="#2-Select-1000-regions-with-a-10-to-30-fold-enrichment-relative-to-the-genome-background" class="headerlink" title="2. Select 1000 regions with a 10- to 30-fold enrichment relative to the genome background"></a>2. Select 1000 regions with a 10- to 30-fold enrichment relative to the genome background</h3><h3 id="3-Build-model-and-estimate-DNA-fragment-size-d"><a href="#3-Build-model-and-estimate-DNA-fragment-size-d" class="headerlink" title="3. Build model and estimate DNA fragment size d"></a>3. Build model and estimate DNA fragment size d</h3><h3 id="4-Shift-reads-toward-3’end-by-d"><a href="#4-Shift-reads-toward-3’end-by-d" class="headerlink" title="4. Shift reads toward 3’end by d"></a>4. Shift reads toward 3’end by d</h3><h3 id="5-Scale-two-libraries-if-have-control"><a href="#5-Scale-two-libraries-if-have-control" class="headerlink" title="5. Scale two libraries(if have control)"></a>5. Scale two libraries(if have control)</h3><p>For experiments in which <code>sequence depth differs</code> between input and treatment samples, MACS linearly scales the total control tag count to be the same as the total ChIP tag count. The default behaviour is for the <code>larger sample to be scaled down</code>. </p>
<h3 id="6-0-Effective-genome-length"><a href="#6-0-Effective-genome-length" class="headerlink" title="6.0 Effective genome length"></a>6.0 Effective genome length</h3><p>To calculate λBG from tag count, MAC2 requires the <code>effective genome size or the size of the genome that is mappable</code>. Mappability is related to the <code>uniqueness of the k-mers</code> at a particular position the genome. <code>Low-complexity</code> and <code>repetitive regions</code> have <code>low uniqueness</code>, which means <code>low mappability</code>. Therefore we need to provide the effective genome length to correct for the loss of true signals in low-mappable regions. 提供<code>有效基因组长度</code>，以<code>纠正低定位区域真实信号丢失</code>。<br><img src="https://img-blog.csdnimg.cn/5571c4b228cb4706805ba0853c546c3c.png" alt="在这里插入图片描述"><br><code>如何获得？</code>The MACS2 software has some <code>pre-computed values</code> for <code>commonly used</code> organisms (human, mouse, worm and fly，<code>rice？</code>). more accurate values? The <code>deepTools</code> docs has additional pre-computed values for more recent builds but also has some good materials on <code>how to go about computing</code> it.</p>
<h3 id="6-Call-candidate-peaks-relative-to-genome-background"><a href="#6-Call-candidate-peaks-relative-to-genome-background" class="headerlink" title="6. Call candidate peaks relative to genome background"></a>6. Call candidate peaks relative to genome background</h3><p>After MACS <code>shifts every tag by d/2</code>, it then <code>slides across the genome</code> using a window size of <code>2d</code> to <code>find candidate peaks</code>.  The <code>tag distribution</code> along the genome can be modeled by a <code>Poisson distribution</code>.  The Poisson is a one parameter model, where the parameter <code>λ is the expected number of reads in that window</code>.</p>
<h3 id="7-Calculate-dynamic-λ-for-candidate-peaks"><a href="#7-Calculate-dynamic-λ-for-candidate-peaks" class="headerlink" title="7. Calculate dynamic λ for candidate peaks"></a>7. Calculate dynamic λ for candidate peaks</h3><p><img src="https://img-blog.csdnimg.cn/eaac9c4b4c294be3b8b3b3caf7c35a21.png" alt="在这里插入图片描述"><br>泊松分布的参数λ是单位时间(或单位面积)内随机事件的平均发生次数。 泊松分布的期望和方差均为λ.<br>Instead of using a uniform λ estimated from the whole genome, MACS uses a <code>dynamic parameter, λlocal</code>, defined for <code>each candidate peak</code>. The lambda parameter is estimated from the control sample and is deduced by taking the maximum value across <code>various window sizes</code>:</p>
<blockquote>
<p>λlocal = max(λBG, λ1k, λ5k, λ10k).</p>
</blockquote>
<p>In this way lambda <code>captures the influence of local biases</code>, and is robust against occasional low tag counts at small local regions. Possible sources for these biases include local chromatin structure, DNA amplification and sequencing bias, and genome copy number variation. 通过使用动态lambda捕获局部偏差的影响，并且对小局部区域中<code>偶尔出现的low tag counts</code>表现较好。<code>偏差可能来源</code>包括局部染色质结构、DNA扩增和测序偏差以及基因组拷贝数变化。<br><img src="https://img-blog.csdnimg.cn/3c7c645b299e45b2b7975d6327d421bc.png" alt="在这里插入图片描述"><br>A region is considered to have a <code>significant tag enrichment</code> if the <code>p-value &lt; 10e-5</code> (可设置). This is a Poisson distribution p-value based on λ.</p>
<p>Overlapping enriched peaks are merged, and each tag position is <code>extended ‘d’ bases？</code> from its center. The location in the <code>peak with the highest fragment pileup堆积</code>, hereafter referred to as the summit峰顶, is <code>predicted as the precise binding location</code>. The <code>ratio between the ChIP-seq tag count and λlocal？</code> is reported as the fold enrichment.</p>
<h3 id="8-Calculate-P-value-and-filter-candidate-peaks"><a href="#8-Calculate-P-value-and-filter-candidate-peaks" class="headerlink" title="8. Calculate P value and filter candidate peaks"></a>8. Calculate P value and filter candidate peaks</h3><h3 id="9-Calculate-FDR-by-exchanging-treatment-and-control"><a href="#9-Calculate-FDR-by-exchanging-treatment-and-control" class="headerlink" title="9. Calculate FDR by exchanging treatment and control"></a>9. Calculate FDR by exchanging treatment and control</h3><p>Each peak is considered an independent test and thus, when we encounter thousands of significant peaks detected in a sample we have a multiple testing problem. In <code>MACSv1.4</code>, the FDR was determined <code>empirically by exchanging the ChIP and control samples</code>. However, in <code>MACS2</code>, p-values are now corrected for multiple comparison using the <code>Benjamini-Hochberg correction</code>.</p>
<h2 id="MACS2-参数"><a href="#MACS2-参数" class="headerlink" title="MACS2 参数"></a>MACS2 参数</h2><h3 id="1-Input-file-options"><a href="#1-Input-file-options" class="headerlink" title="1. Input file options"></a>1. Input file options</h3><ul>
<li><code>-t</code> : The IP data file (this is the only REQUIRED parameter for MACS) 实验组</li>
<li><code>-c</code> : Control or mock data 对照</li>
<li><code>-f</code> : 输入文件格式，默认值“AUTO” ，bam sam bed</li>
<li><code>-g</code> :  <code>mappable genome size</code> which is defined as the genome size which can be sequenced; some precompiled values provided.</li>
<li>hs: 2.7e9人类基因组有效大小(UCSC human hg18 assembly)<h3 id="2-Output-arguments"><a href="#2-Output-arguments" class="headerlink" title="2. Output arguments"></a>2. Output arguments</h3></li>
<li><code>-outdir</code> : 输出文件夹</li>
<li><code>-n</code> : 文件前缀</li>
<li><code>-B/--bdg</code> : store the fragment pileup, control lambda, -log10pvalue and -log10qvalue scores in bedGraph files 输出bedgraph格式的文件<h3 id="3-Shifting-model-arguments"><a href="#3-Shifting-model-arguments" class="headerlink" title="3. Shifting model arguments"></a>3. Shifting model arguments</h3></li>
<li><code>-s</code> : <code>size of sequencing tags</code>. Default, MACS will use the <code>first 10 sequences</code> from your input treatment file to determine it</li>
<li><code>--bw</code> : The bandwidth which is used to <code>scan the genome</code> ONLY for model building. Can be set to the expected sonication fragment size.</li>
<li><code>--mfold</code> : <code>upper and lower limit</code> for model building</li>
<li><code>--nomodel</code>：<code>和extsize、shift是配套使用</code>，有这个参数才可设置extsize和shift。</li>
<li><code>--extsize</code>：当设置了nomodel时，MACS会用–extsize这个参数从<code>5&#39;-&gt;3&#39;方向扩展reads修复fragments</code>。如转录因子结合范围200bp，设置这个参数是200。</li>
<li><code>--shift</code>：当设置了–nomodel，MACS用这个参数从<code>5&#39; 端移动剪切</code>，然后用–extsize延伸，如果–shift是负值表示从3’端方向移动。建议ChIP-seq数据集这个值保持默认值为0？，对于检测富集剪切位点如DNAseq数据集设置为EXTSIZE的一半。</li>
</ul>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line">想找富集剪切位点，如DNAse-seq，所有5&#x27;端的序列reads应该从两个方向延伸，如果想设置移动的窗口是200bp，参数设置如下：</span><br><span class="line">--nomodel --shift -100 --extsize 200</span><br><span class="line">对nucleosome-seq数据，用核小体大小的一半进行extsize,所以参数设置如下：</span><br><span class="line">--nomodel --shift 37 --extsize 73</span><br></pre></td></tr></table></figure>

<p><code>ATAC-seq</code>关心的是在哪切断，断点才是peak的中心，所以使用shift模型，–shift -75或-100<br>对人细胞系ATAC-seq 数据call peak的参数设置：</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">macs2 callpeak -t H1hesc.final.bam -n sample --shift -100 --extsize 200 --nomodel -B --SPMR -g hs --outdir Macs2_out 2&gt; sample.macs2.log</span><br></pre></td></tr></table></figure>

<h3 id="4-Peak-calling-arguments"><a href="#4-Peak-calling-arguments" class="headerlink" title="4. Peak calling arguments"></a>4. Peak calling arguments</h3><ul>
<li><code>-q</code> : q-value (minimum FDR) cutoff <code>q value默认值是0.05，与pvalue不能同时使用。</code></li>
<li><code>-p</code> : p-value cutoff (instead of q-value cutoff)</li>
<li><code>--nolambda</code> : do not consider the local bias/lambda at peak candidate regions。不要考虑在峰值候选区域的局部偏差/λ</li>
<li><code>--broad</code> : broad peak calling，narrow peak和broad peak</li>
</ul>
<p> Relaxing the <code>q-value</code> does not behave as expected in this case since it is partially tied to peak widths. Ideally, if you <code>relaxed the thresholds</code>, you would simply get <code>more peaks</code> but with MACS2 relaxing thresholds also results in <code>wider peaks</code>. q值与峰宽有一定的联系。理想情况下，如果放宽阈值，您将简单地获得更多的峰值，但是使用MACS2放松阈值也会导致更宽的峰值。</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line">macs2 callpeak -t bowtie2/H1hesc_Nanog_Rep1_aln.bam \</span><br><span class="line">	-c bowtie2/H1hesc_Input_Rep1_aln.bam \</span><br><span class="line"> 	-f BAM -g 1.3e+8 \</span><br><span class="line">	-n Nanog-rep1 \</span><br><span class="line">	--outdir macs2 2&gt; macs2/Nanog-rep1-macs2.log</span><br></pre></td></tr></table></figure>


<h2 id="MACS2-Output-files"><a href="#MACS2-Output-files" class="headerlink" title="MACS2 Output files"></a>MACS2 Output files</h2><h3 id="narrowPeak"><a href="#narrowPeak" class="headerlink" title="narrowPeak"></a>narrowPeak</h3><p>A narrowPeak (<code>.narrowPeak</code>) file is used by the ENCODE project to provide <code>called peaks of signal enrichment</code> based on pooled, normalized (interpreted) data. It is a <code>BED 6+4 format</code>, which means the <code>first 6 columns of a standard BED file</code> with <code>4 additional</code> fields:<br><img src="https://img-blog.csdnimg.cn/7bd1ca4fcb3847daacc450a91989b5cd.png?x-oss-process=image/watermark,type_ZHJvaWRzYW5zZmFsbGJhY2s,shadow_50,text_Q1NETiBAYmFubmVyRHI=,size_20,color_FFFFFF,t_70,g_se,x_16" alt="在这里插入图片描述"></p>
<h3 id="WIG-format"><a href="#WIG-format" class="headerlink" title="WIG format"></a>WIG format</h3><p>Wiggle format (WIG) allows the <code>display</code> of continuous-valued data in a track format. Wiggle format is <code>line-oriented</code>. It is composed of declaration lines and data lines, and require a separate wiggle track definition line. There are two options for formatting wiggle data: variableStep and fixedStep. These formats were developed to allow the file to be written as compactly as possible.</p>
<h3 id="BedGraph-format"><a href="#BedGraph-format" class="headerlink" title="BedGraph format"></a>BedGraph format</h3><p>The BedGraph format also allows display of continuous-valued data in track format. This display type is useful for probability scores and transcriptome data. This track type is similar to the wiggle (WIG) format, but unlike the wiggle format, data exported in the bedGraph format are preserved in their original state. For the purposes of visualization, these can be interchangeable.</p>
<ul>
<li><p><code>_peaks.narrowPeak</code>: BED6+4 format file which contains the peak locations together with peak summit, pvalue and qvalue。<code>BED6+4格式，可以上传到UCSC浏览。</code></p>
<ul>
<li>1：染色体号</li>
<li>2：peak起始位点</li>
<li>3：结束位点</li>
<li>4：peak name</li>
<li>5：int(-10*log10qvalue)</li>
<li>6 ：正负链</li>
<li>7：fold change</li>
<li>8：-log10pvalue</li>
<li>9：-log10qvalue</li>
<li>10：relative summit position to peak start（？）</li>
</ul>
</li>
<li><p><code>_peaks.xls</code>: a tabular file which contains information about called peaks. Additional information includes pileup and fold enrichment。<code>含peak信息的tab分割的文件，前几行显示callpeak命令</code>。</p>
<ul>
<li>染色体号</li>
<li>peak起始位点</li>
<li>peak结束位点</li>
<li>peak区域长度</li>
<li>peak的峰值位点（summit position）</li>
<li>peak 峰值的高度（pileup height at peak summit, -log10(pvalue) for the peak summit）</li>
<li>peak的富集倍数（相对于random Poisson distribution with local lambda）</li>
<li>XLS里的坐标和bed格式的坐标还不一样，起始坐标需要减1才与narrowPeak的起始坐标一样。</li>
</ul>
</li>
<li><p><code>_summits.bed</code>: peak summits locations for every peak. To find the motifs at the binding sites, this file is recommended。BED格式的文件，包含peak的summits位置，第5列是-log10pvalue。如果想找motif，推荐使用此文件。</p>
</li>
<li><p><code>_model.R</code>: an R script which you can use to produce a PDF image about the model based on your data and cross-correlation plot</p>
</li>
<li><p><code>_control_lambda.bdg</code>: bedGraph format for input sample</p>
</li>
<li><p><code>_treat_pileup.bdg</code>: bedGraph format for treatment sample。bedGraph格式，可以导入UCSC或者转换为bigwig格式。两种bfg文件：treat_pileup, and control_lambda.</p>
</li>
<li><p><code>NAME_peaks.broadPeak</code>： BED6+3格式与narrowPeak类似，只是没有第10列。</p>
</li>
</ul>
<p>R作图the first plot illustrates the distance between the modes from which the shift size was determined？<br><img src="https://img-blog.csdnimg.cn/3d941b11b37c4b7687b11e6766a9daff.png?x-oss-process=image/watermark,type_ZHJvaWRzYW5zZmFsbGJhY2s,shadow_50,text_Q1NETiBAYmFubmVyRHI=,size_13,color_FFFFFF,t_70,g_se,x_16" alt="在这里插入图片描述"><br>The second plot is the cross-correlation plot. This is a graphical representation of the Pearson correlation of positive- and negative- strand tag densities, shifting the strands relative to each other by increasing distance？</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br></pre></td><td class="code"><pre><span class="line">xls文件</span><br><span class="line">文件包含信息还是比较多的，和narrowPeak唯一不同的是peak的起始位置需要减1才是bed格式的文件，另外还包含fold_enrichment 和narrowPeak的fold change 对应，-log10pvalue,-log10qvalue,peak长度，peak 峰值位置等。</span><br><span class="line">narrowPeak文件</span><br><span class="line">和xls文件信息类似</span><br><span class="line">summits.bed文件</span><br><span class="line">包含峰的位置信息和-log10pvalue</span><br><span class="line">bdg文件</span><br><span class="line">bdg文件适合导入UCSC或IGV进行谱图可视化，或者转换为bigwig格式再进行可视化。</span><br></pre></td></tr></table></figure>

    </div>

    
    
    
        

<div>
<ul class="post-copyright">
  <li class="post-copyright-author">
    <strong>Post author:  </strong>Wang C
  </li>
  <li class="post-copyright-link">
    <strong>Post link: </strong>
    <a href="https://wangc1106.github.io/blog/2021/10/26/ATAC-seq-3-hbctraining/" title="ATAC-seq-3【hbctraining_pipeline】">https://wangc1106.github.io/blog/2021/10/26/ATAC-seq-3-hbctraining/</a>
  </li>
  <li class="post-copyright-license">
    <strong>Copyright Notice:  </strong>All articles in this blog are licensed under <a href="https://creativecommons.org/licenses/by-nc-sa/4.0/" rel="noopener" target="_blank"><i class="fab fa-fw fa-creative-commons"></i>BY-NC-SA</a> unless stating additionally.
  </li>
</ul>
</div>


      <footer class="post-footer">
          <div class="post-tags">
              <a href="/blog/tags/ATAC-seq/" rel="tag"># ATAC-seq</a>
          </div>

        


        
    <div class="post-nav">
      <div class="post-nav-item">
    <a href="/blog/2021/10/20/hifiasm/" rel="prev" title="单倍型基因组组装算法hifiasm">
      <i class="fa fa-chevron-left"></i> 单倍型基因组组装算法hifiasm
    </a></div>
      <div class="post-nav-item">
    <a href="/blog/2021/10/29/ATAC-seq-5-macs2-info/" rel="next" title="ATAC-seq-5【PeakCalling-MACS2】">
      ATAC-seq-5【PeakCalling-MACS2】 <i class="fa fa-chevron-right"></i>
    </a></div>
    </div>
      </footer>
    
  </article>
  
  
  



          </div>
          

<script>
  window.addEventListener('tabs:register', () => {
    let { activeClass } = CONFIG.comments;
    if (CONFIG.comments.storage) {
      activeClass = localStorage.getItem('comments_active') || activeClass;
    }
    if (activeClass) {
      let activeTab = document.querySelector(`a[href="#comment-${activeClass}"]`);
      if (activeTab) {
        activeTab.click();
      }
    }
  });
  if (CONFIG.comments.storage) {
    window.addEventListener('tabs:click', event => {
      if (!event.target.matches('.tabs-comment .tab-content .tab-pane')) return;
      let commentClass = event.target.classList[1];
      localStorage.setItem('comments_active', commentClass);
    });
  }
</script>

        </div>
          
  
  <div class="toggle sidebar-toggle">
    <span class="toggle-line toggle-line-first"></span>
    <span class="toggle-line toggle-line-middle"></span>
    <span class="toggle-line toggle-line-last"></span>
  </div>

  <aside class="sidebar">
    <div class="sidebar-inner">

      <ul class="sidebar-nav motion-element">
        <li class="sidebar-nav-toc">
          Table of Contents
        </li>
        <li class="sidebar-nav-overview">
          Overview
        </li>
      </ul>

      <!--noindex-->
      <div class="post-toc-wrap sidebar-panel">
          <div class="post-toc motion-element"><ol class="nav"><li class="nav-item nav-level-2"><a class="nav-link" href="#https-github-com-hbctraining-In-depth-NGS-Data-Analysis-Course-tree-master-sessionV-lessons"><span class="nav-number">1.</span> <span class="nav-text">https:&#x2F;&#x2F;github.com&#x2F;hbctraining&#x2F;In-depth-NGS-Data-Analysis-Course&#x2F;tree&#x2F;master&#x2F;sessionV&#x2F;lessons</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#Peak-calling"><span class="nav-number">2.</span> <span class="nav-text">Peak calling</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#MACS2"><span class="nav-number">3.</span> <span class="nav-text">MACS2</span></a><ol class="nav-child"><li class="nav-item nav-level-3"><a class="nav-link" href="#1-Remove-redundancy"><span class="nav-number">3.1.</span> <span class="nav-text">1. Remove redundancy</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#2-0-Shift-size-d"><span class="nav-number">3.2.</span> <span class="nav-text">2.0 Shift size d</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#2-Select-1000-regions-with-a-10-to-30-fold-enrichment-relative-to-the-genome-background"><span class="nav-number">3.3.</span> <span class="nav-text">2. Select 1000 regions with a 10- to 30-fold enrichment relative to the genome background</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#3-Build-model-and-estimate-DNA-fragment-size-d"><span class="nav-number">3.4.</span> <span class="nav-text">3. Build model and estimate DNA fragment size d</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#4-Shift-reads-toward-3%E2%80%99end-by-d"><span class="nav-number">3.5.</span> <span class="nav-text">4. Shift reads toward 3’end by d</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#5-Scale-two-libraries-if-have-control"><span class="nav-number">3.6.</span> <span class="nav-text">5. Scale two libraries(if have control)</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#6-0-Effective-genome-length"><span class="nav-number">3.7.</span> <span class="nav-text">6.0 Effective genome length</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#6-Call-candidate-peaks-relative-to-genome-background"><span class="nav-number">3.8.</span> <span class="nav-text">6. Call candidate peaks relative to genome background</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#7-Calculate-dynamic-%CE%BB-for-candidate-peaks"><span class="nav-number">3.9.</span> <span class="nav-text">7. Calculate dynamic λ for candidate peaks</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#8-Calculate-P-value-and-filter-candidate-peaks"><span class="nav-number">3.10.</span> <span class="nav-text">8. Calculate P value and filter candidate peaks</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#9-Calculate-FDR-by-exchanging-treatment-and-control"><span class="nav-number">3.11.</span> <span class="nav-text">9. Calculate FDR by exchanging treatment and control</span></a></li></ol></li><li class="nav-item nav-level-2"><a class="nav-link" href="#MACS2-%E5%8F%82%E6%95%B0"><span class="nav-number">4.</span> <span class="nav-text">MACS2 参数</span></a><ol class="nav-child"><li class="nav-item nav-level-3"><a class="nav-link" href="#1-Input-file-options"><span class="nav-number">4.1.</span> <span class="nav-text">1. Input file options</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#2-Output-arguments"><span class="nav-number">4.2.</span> <span class="nav-text">2. Output arguments</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#3-Shifting-model-arguments"><span class="nav-number">4.3.</span> <span class="nav-text">3. Shifting model arguments</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#4-Peak-calling-arguments"><span class="nav-number">4.4.</span> <span class="nav-text">4. Peak calling arguments</span></a></li></ol></li><li class="nav-item nav-level-2"><a class="nav-link" href="#MACS2-Output-files"><span class="nav-number">5.</span> <span class="nav-text">MACS2 Output files</span></a><ol class="nav-child"><li class="nav-item nav-level-3"><a class="nav-link" href="#narrowPeak"><span class="nav-number">5.1.</span> <span class="nav-text">narrowPeak</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#WIG-format"><span class="nav-number">5.2.</span> <span class="nav-text">WIG format</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#BedGraph-format"><span class="nav-number">5.3.</span> <span class="nav-text">BedGraph format</span></a></li></ol></li></ol></div>
      </div>
      <!--/noindex-->

      <div class="site-overview-wrap sidebar-panel">
        <div class="site-author motion-element" itemprop="author" itemscope itemtype="http://schema.org/Person">
    <img class="site-author-image" itemprop="image" alt="Wang C"
      src="/blog/images/banner.png">
  <p class="site-author-name" itemprop="name">Wang C</p>
  <div class="site-description" itemprop="description">Make more time</div>
</div>
<div class="site-state-wrap motion-element">
  <nav class="site-state">
      <div class="site-state-item site-state-posts">
          <a href="/blog/archives">
          <span class="site-state-item-count">32</span>
          <span class="site-state-item-name">posts</span>
        </a>
      </div>
      <div class="site-state-item site-state-categories">
            <a href="/blog/categories/">
          
        <span class="site-state-item-count">11</span>
        <span class="site-state-item-name">categories</span></a>
      </div>
      <div class="site-state-item site-state-tags">
            <a href="/blog/tags/">
          
        <span class="site-state-item-count">18</span>
        <span class="site-state-item-name">tags</span></a>
      </div>
  </nav>
</div>
  <div class="links-of-author motion-element">
      <span class="links-of-author-item">
        <a href="https://github.com/wangc1106" title="GitHub → https:&#x2F;&#x2F;github.com&#x2F;wangc1106" rel="noopener" target="_blank"><i class="fab fa-github fa-fw"></i>GitHub</a>
      </span>
      <span class="links-of-author-item">
        <a href="mailto:wangchen116@126.com" title="E-Mail → mailto:wangchen116@126.com" rel="noopener" target="_blank"><i class="fa fa-envelope fa-fw"></i>E-Mail</a>
      </span>
  </div>
  <div class="cc-license motion-element" itemprop="license">
    <a href="https://creativecommons.org/licenses/by-nc-sa/4.0/" class="cc-opacity" rel="noopener" target="_blank"><img src="/blog/images/cc-by-nc-sa.svg" alt="Creative Commons"></a>
  </div>



      </div>
        <div class="back-to-top motion-element">
          <i class="fa fa-arrow-up"></i>
          <span>0%</span>
        </div>

    </div>
  </aside>
  <div id="sidebar-dimmer"></div>


      </div>
    </main>

    <footer class="footer">
      <div class="footer-inner">
        

        

<div class="copyright">
  
  &copy; 2021 – 
  <span itemprop="copyrightYear">2022</span>
  <span class="with-love">
    <i class="fas fa-dna"></i>
  </span>
  <span class="author" itemprop="copyrightHolder">Wang C</span>
</div>

        






<script>
  (function() {
    function leancloudSelector(url) {
      url = encodeURI(url);
      return document.getElementById(url).querySelector('.leancloud-visitors-count');
    }

    function addCount(Counter) {
      var visitors = document.querySelector('.leancloud_visitors');
      var url = decodeURI(visitors.id);
      var title = visitors.dataset.flagTitle;

      Counter('get', '/classes/Counter?where=' + encodeURIComponent(JSON.stringify({ url })))
        .then(response => response.json())
        .then(({ results }) => {
          if (results.length > 0) {
            var counter = results[0];
            leancloudSelector(url).innerText = counter.time + 1;
            Counter('put', '/classes/Counter/' + counter.objectId, { time: { '__op': 'Increment', 'amount': 1 } })
              .catch(error => {
                console.error('Failed to save visitor count', error);
              });
          } else {
              Counter('post', '/classes/Counter', { title, url, time: 1 })
                .then(response => response.json())
                .then(() => {
                  leancloudSelector(url).innerText = 1;
                })
                .catch(error => {
                  console.error('Failed to create', error);
                });
          }
        })
        .catch(error => {
          console.error('LeanCloud Counter Error', error);
        });
    }

    function showTime(Counter) {
      var visitors = document.querySelectorAll('.leancloud_visitors');
      var entries = [...visitors].map(element => {
        return decodeURI(element.id);
      });

      Counter('get', '/classes/Counter?where=' + encodeURIComponent(JSON.stringify({ url: { '$in': entries } })))
        .then(response => response.json())
        .then(({ results }) => {
          for (let url of entries) {
            let target = results.find(item => item.url === url);
            leancloudSelector(url).innerText = target ? target.time : 0;
          }
        })
        .catch(error => {
          console.error('LeanCloud Counter Error', error);
        });
    }

    let { app_id, app_key, server_url } = {"enable":true,"app_id":"UqIYufmNk2sE8MiBKKWc4Kja-gzGzoHsz","app_key":"iYpeWQcmNqpHwr2uVlnJYVji","server_url":null,"security":false};
    function fetchData(api_server) {
      var Counter = (method, url, data) => {
        return fetch(`${api_server}/1.1${url}`, {
          method,
          headers: {
            'X-LC-Id'     : app_id,
            'X-LC-Key'    : app_key,
            'Content-Type': 'application/json',
          },
          body: JSON.stringify(data)
        });
      };
      if (CONFIG.page.isPost) {
        if (CONFIG.hostname !== location.hostname) return;
        addCount(Counter);
      } else if (document.querySelectorAll('.post-title-link').length >= 1) {
        showTime(Counter);
      }
    }

    let api_server = app_id.slice(-9) !== '-MdYXbMMI' ? server_url : `https://${app_id.slice(0, 8).toLowerCase()}.api.lncldglobal.com`;

    if (api_server) {
      fetchData(api_server);
    } else {
      fetch('https://app-router.leancloud.cn/2/route?appId=' + app_id)
        .then(response => response.json())
        .then(({ api_server }) => {
          fetchData('https://' + api_server);
        });
    }
  })();
</script>


      </div>
      <div style="height: 200px">
      <script type="text/javascript" src="//rf.revolvermaps.com/0/0/8.js?i=5knf34gs07t&amp;m=0&amp;c=ff0000&amp;cr1=ffffff&amp;f=arial&amp;l=33" async="async"></script>
      </div>
    </footer>
  </div>

  
  <script src="/blog/lib/anime.min.js"></script>
  <script src="/blog/lib/velocity/velocity.min.js"></script>
  <script src="/blog/lib/velocity/velocity.ui.min.js"></script>

<script src="/blog/js/utils.js"></script>

<script src="/blog/js/motion.js"></script>


<script src="/blog/js/schemes/pisces.js"></script>


<script src="/blog/js/next-boot.js"></script>




  




  
<script src="/blog/js/local-search.js"></script>













  

  

</body>
</html>
