<!DOCTYPE html>
<html lang="zh-Hans">
<head>
  <meta charset="UTF-8">
<meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=2">
<meta name="theme-color" content="#222">
<meta name="generator" content="Hexo 5.4.0">
  <link rel="apple-touch-icon" sizes="180x180" href="/blog/images/apple-touch-icon-next.png">
  <link rel="icon" type="image/png" sizes="32x32" href="/blog/images/favicon-32x32-next.png">
  <link rel="icon" type="image/png" sizes="16x16" href="/blog/images/favicon-16x16-next.png">
  <link rel="mask-icon" href="/blog/images/logo.svg" color="#222">
  <meta name="google-site-verification" content="1osTo_gOe1SW9Hevqtz8vl8HfSgTO4DSMZ0DiMPMX6w">

<link rel="stylesheet" href="/blog/css/main.css">


<link rel="stylesheet" href="/blog/lib/font-awesome/css/all.min.css">

<script id="hexo-configurations">
    var NexT = window.NexT || {};
    var CONFIG = {"hostname":"wangc1106.github.io","root":"/blog/","scheme":"Gemini","version":"7.8.0","exturl":false,"sidebar":{"position":"left","display":"post","padding":18,"offset":12,"onmobile":false},"copycode":{"enable":false,"show_result":false,"style":null},"back2top":{"enable":true,"sidebar":true,"scrollpercent":"ture"},"bookmark":{"enable":false,"color":"#222","save":"auto"},"fancybox":false,"mediumzoom":false,"lazyload":false,"pangu":false,"comments":{"style":"tabs","active":null,"storage":true,"lazyload":false,"nav":null},"algolia":{"hits":{"per_page":10},"labels":{"input_placeholder":"Search for Posts","hits_empty":"We didn't find any results for the search: ${query}","hits_stats":"${hits} results found in ${time} ms"}},"localsearch":{"enable":true,"trigger":"auto","top_n_per_article":1,"unescape":false,"preload":false},"motion":{"enable":true,"async":false,"transition":{"post_block":"fadeIn","post_header":"slideDownIn","post_body":"slideDownIn","coll_header":"slideLeftIn","sidebar":"slideUpIn"}},"path":"search.xml"};
  </script>

  <meta name="description" content="soapdenovo2:">
<meta property="og:type" content="article">
<meta property="og:title" content="Soapdenovo2">
<meta property="og:url" content="https://wangc1106.github.io/blog/2022/02/24/2022-02-25-soapdenovo2/index.html">
<meta property="og:site_name" content="Banner&#39;s Blog">
<meta property="og:description" content="soapdenovo2:">
<meta property="og:locale">
<meta property="og:image" content="https://wangc1106.github.io/blog/blog/2022/02/24/2022-02-25-soapdenovo2/1.png">
<meta property="og:image" content="https://wangc1106.github.io/blog/blog/2022/02/24/2022-02-25-soapdenovo2/2.png">
<meta property="og:image" content="https://wangc1106.github.io/blog/blog/2022/02/24/2022-02-25-soapdenovo2/3.png">
<meta property="og:image" content="https://wangc1106.github.io/blog/blog/2022/02/24/2022-02-25-soapdenovo2/4.png">
<meta property="og:image" content="https://wangc1106.github.io/blog/blog/2022/02/24/2022-02-25-soapdenovo2/5.png">
<meta property="og:image" content="https://wangc1106.github.io/blog/blog/2022/02/24/2022-02-25-soapdenovo2/6.png">
<meta property="og:image" content="https://wangc1106.github.io/blog/blog/2022/02/24/2022-02-25-soapdenovo2/7.png">
<meta property="og:image" content="https://wangc1106.github.io/blog/blog/2022/02/24/2022-02-25-soapdenovo2/8.png">
<meta property="og:image" content="https://wangc1106.github.io/blog/blog/2022/02/24/2022-02-25-soapdenovo2/9.png">
<meta property="og:image" content="https://wangc1106.github.io/blog/blog/2022/02/24/2022-02-25-soapdenovo2/10.png">
<meta property="og:image" content="https://wangc1106.github.io/blog/blog/2022/02/24/2022-02-25-soapdenovo2/11.png">
<meta property="og:image" content="https://wangc1106.github.io/blog/blog/2022/02/24/2022-02-25-soapdenovo2/12.png">
<meta property="og:image" content="https://wangc1106.github.io/blog/blog/2022/02/24/2022-02-25-soapdenovo2/13.png">
<meta property="og:image" content="https://wangc1106.github.io/blog/blog/2022/02/24/2022-02-25-soapdenovo2/14.png">
<meta property="og:image" content="https://wangc1106.github.io/blog/blog/2022/02/24/2022-02-25-soapdenovo2/15.png">
<meta property="og:image" content="https://wangc1106.github.io/blog/blog/2022/02/24/2022-02-25-soapdenovo2/16.png">
<meta property="og:image" content="https://wangc1106.github.io/blog/blog/2022/02/24/2022-02-25-soapdenovo2/17.png">
<meta property="og:image" content="https://wangc1106.github.io/blog/blog/2022/02/24/2022-02-25-soapdenovo2/18.png">
<meta property="og:image" content="https://wangc1106.github.io/blog/blog/2022/02/24/2022-02-25-soapdenovo2/19.png">
<meta property="og:image" content="https://wangc1106.github.io/blog/blog/2022/02/24/2022-02-25-soapdenovo2/20.png">
<meta property="og:image" content="https://wangc1106.github.io/blog/blog/2022/02/24/2022-02-25-soapdenovo2/21.png">
<meta property="og:image" content="https://wangc1106.github.io/blog/blog/2022/02/24/2022-02-25-soapdenovo2/22.png">
<meta property="og:image" content="https://wangc1106.github.io/blog/blog/2022/02/24/2022-02-25-soapdenovo2/23.png">
<meta property="og:image" content="https://wangc1106.github.io/blog/blog/2022/02/24/2022-02-25-soapdenovo2/24.png">
<meta property="article:published_time" content="2022-02-24T13:58:19.224Z">
<meta property="article:modified_time" content="2022-03-04T02:01:28.273Z">
<meta property="article:author" content="Wang C">
<meta property="article:tag" content="SOAPdenovo2">
<meta name="twitter:card" content="summary">
<meta name="twitter:image" content="https://wangc1106.github.io/blog/blog/2022/02/24/2022-02-25-soapdenovo2/1.png">

<link rel="canonical" href="https://wangc1106.github.io/blog/2022/02/24/2022-02-25-soapdenovo2/">


<script id="page-configurations">
  // https://hexo.io/docs/variables.html
  CONFIG.page = {
    sidebar: "",
    isHome : false,
    isPost : true,
    lang   : 'zh-Hans'
  };
</script>

  <title>Soapdenovo2 | Banner's Blog</title>
  






  <noscript>
  <style>
  .use-motion .brand,
  .use-motion .menu-item,
  .sidebar-inner,
  .use-motion .post-block,
  .use-motion .pagination,
  .use-motion .comments,
  .use-motion .post-header,
  .use-motion .post-body,
  .use-motion .collection-header { opacity: initial; }

  .use-motion .site-title,
  .use-motion .site-subtitle {
    opacity: initial;
    top: initial;
  }

  .use-motion .logo-line-before i { left: initial; }
  .use-motion .logo-line-after i { right: initial; }
  </style>
</noscript>

</head>

<body itemscope itemtype="http://schema.org/WebPage">
  <div class="container use-motion">
    <div class="headband"></div>

    <header class="header" itemscope itemtype="http://schema.org/WPHeader">
      <div class="header-inner"><div class="site-brand-container">
  <div class="site-nav-toggle">
    <div class="toggle" aria-label="Toggle navigation bar">
      <span class="toggle-line toggle-line-first"></span>
      <span class="toggle-line toggle-line-middle"></span>
      <span class="toggle-line toggle-line-last"></span>
    </div>
  </div>

  <div class="site-meta">

    <a href="/blog/" class="brand" rel="start">
      <span class="logo-line-before"><i></i></span>
      <h1 class="site-title">Banner's Blog</h1>
      <span class="logo-line-after"><i></i></span>
    </a>
      <p class="site-subtitle" itemprop="description">Dream big work hard</p>
  </div>

  <div class="site-nav-right">
    <div class="toggle popup-trigger">
        <i class="fa fa-search fa-fw fa-lg"></i>
    </div>
  </div>
</div>




<nav class="site-nav">
  <ul id="menu" class="main-menu menu">
        <li class="menu-item menu-item-home">

    <a href="/blog/" rel="section"><i class="home fa-fw"></i>Home</a>

  </li>
        <li class="menu-item menu-item-categories">

    <a href="/blog/categories/" rel="section"><i class="th fa-fw"></i>Categories</a>

  </li>
        <li class="menu-item menu-item-tags">

    <a href="/blog/tags/" rel="section"><i class="tags fa-fw"></i>Tags</a>

  </li>
        <li class="menu-item menu-item-about">

    <a href="https://wangc1106.github.io/" rel="section"><i class="user fa-fw"></i>About</a>

  </li>
      <li class="menu-item menu-item-search">
        <a role="button" class="popup-trigger"><i class="fa fa-search fa-fw"></i>Search
        </a>
      </li>
  </ul>
</nav>



  <div class="search-pop-overlay">
    <div class="popup search-popup">
        <div class="search-header">
  <span class="search-icon">
    <i class="fa fa-search"></i>
  </span>
  <div class="search-input-container">
    <input autocomplete="off" autocapitalize="off"
           placeholder="Searching..." spellcheck="false"
           type="search" class="search-input">
  </div>
  <span class="popup-btn-close">
    <i class="fa fa-times-circle"></i>
  </span>
</div>
<div id="search-result">
  <div id="no-result">
    <i class="fa fa-spinner fa-pulse fa-5x fa-fw"></i>
  </div>
</div>

    </div>
  </div>

</div>
    </header>

    


    <main class="main">
      <div class="main-inner">
        <div class="content-wrap">
          

          <div class="content post posts-expand">
            

    
  
  
  <article itemscope itemtype="http://schema.org/Article" class="post-block" lang="zh-Hans">
    <link itemprop="mainEntityOfPage" href="https://wangc1106.github.io/blog/2022/02/24/2022-02-25-soapdenovo2/">

    <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
      <meta itemprop="image" content="/blog/images/banner.png">
      <meta itemprop="name" content="Wang C">
      <meta itemprop="description" content="Make more time">
    </span>

    <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
      <meta itemprop="name" content="Banner's Blog">
    </span>
      <header class="post-header">
        <h1 class="post-title" itemprop="name headline">
          Soapdenovo2
        </h1>

        <div class="post-meta">
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-calendar"></i>
              </span>
              <span class="post-meta-item-text">Posted on</span>

              <time title="Created: 2022-02-24 21:58:19" itemprop="dateCreated datePublished" datetime="2022-02-24T21:58:19+08:00">2022-02-24</time>
            </span>
              <span class="post-meta-item">
                <span class="post-meta-item-icon">
                  <i class="far fa-calendar-check"></i>
                </span>
                <span class="post-meta-item-text">Edited on</span>
                <time title="Modified: 2022-03-04 10:01:28" itemprop="dateModified" datetime="2022-03-04T10:01:28+08:00">2022-03-04</time>
              </span>
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-folder"></i>
              </span>
              <span class="post-meta-item-text">In</span>
                <span itemprop="about" itemscope itemtype="http://schema.org/Thing">
                  <a href="/blog/categories/assemble/" itemprop="url" rel="index"><span itemprop="name">assemble</span></a>
                </span>
            </span>

          
            <span id="/blog/2022/02/24/2022-02-25-soapdenovo2/" class="post-meta-item leancloud_visitors" data-flag-title="Soapdenovo2" title="Views">
              <span class="post-meta-item-icon">
                <i class="fa fa-eye"></i>
              </span>
              <span class="post-meta-item-text">Views: </span>
              <span class="leancloud-visitors-count"></span>
            </span>
            <span class="post-meta-item" title="Symbols count in article">
              <span class="post-meta-item-icon">
                <i class="far fa-file-word"></i>
              </span>
              <span>33k</span>
            </span>
            <span class="post-meta-item" title="Reading time">
              <span class="post-meta-item-icon">
                <i class="far fa-clock"></i>
              </span>
              <span>30 mins.</span>
            </span>

        </div>
      </header>

    
    
    
    <div class="post-body" itemprop="articleBody">

      
        <h1 id="soapdenovo2"><a class="markdownIt-Anchor" href="#soapdenovo2"></a> soapdenovo2:</h1>
<span id="more"></span>
<p>SOAPdenovo使用说明，内容包括程序<font color="green"><strong>使用</strong></font>、<font color="green"><strong>参数说明</strong></font>、<font color="green"><strong>参数调整</strong></font>、主要<font color="green"><strong>输出文件</strong></font>的格式说明等。</p>
<p><img src="/blog/blog/2022/02/24/2022-02-25-soapdenovo2/1.png" alt="img"></p>
<h1 id="contiging"><a class="markdownIt-Anchor" href="#contiging"></a> contiging</h1>
<p><em><strong>a.</strong></em>  基因组DNA随机打断，使用paired-end测序。长度在150-500bp的short clones扩增直接测序。在2-10kb的长paired-end libraries 通过DNA环化、fragmentation破碎，然后净化400-600bp的碎片为了cluster 结构。</p>
<p><em><strong>b.</strong></em>  raw reads 或预修正reads被存储，de Bruijn graph data structure 被用于表示reads间overlap。</p>
<p>Kmer-graph构建：所有reads被切割成某一固定Kmer长度的序列（21bp=&lt;kmer&lt;=127bp），形成等长的Kmers。将Kmers连成图。<font color="green"><strong>相邻的kmers是通过K-1 overlaping连接</strong></font>，所以它不需要成对序列比对（The neighboring kmers are K-1 overlaping which generated from read sequences, so it doesn’t need pair-wise reads alignment.）。重复序列在图中被压缩。</p>
<p><img src="/blog/blog/2022/02/24/2022-02-25-soapdenovo2/2.png" alt="img"></p>
<p><em><strong>c.</strong></em>  产生<font color="red"><strong>tips翼尖、bubbles气泡、low coverage links低覆盖率链接、tiny repeat微小重复</strong></font>等问题。</p>
<p><font color="blue"><strong>tips翼尖（a）</strong></font>和<font color="blue"><strong>bubbles气泡（c）</strong></font>：由于<strong>测序错误</strong>或<strong>杂合</strong>或<strong>高重复序列</strong>相似性，将会导致翼尖和气泡出现。</p>
<p><font color="blue"><strong>low coverage links低覆盖率链接：（b）（d）</strong></font>。</p>
<p><font color="blue"><strong>tiny repeat微小重复(e)</strong></font>：重复在graph中被压缩，并作为不同路径的共享边缘，但是能够通过reads 穿过来解决。</p>
<p><img src="/blog/blog/2022/02/24/2022-02-25-soapdenovo2/3.png" alt="img"></p>
<p><img src="/blog/blog/2022/02/24/2022-02-25-soapdenovo2/4.png" alt="img"></p>
<p><img src="/blog/blog/2022/02/24/2022-02-25-soapdenovo2/5.png" alt="img"></p>
<p>移除错误链接和graph simplification图形简化，得到contigs or contig  graphs：tips翼尖移除；删除低覆盖链接；bubbles合并气泡；解决微小重复；</p>
<p><img src="/blog/blog/2022/02/24/2022-02-25-soapdenovo2/6.png" alt="img"></p>
<p><img src="/blog/blog/2022/02/24/2022-02-25-soapdenovo2/7.png" alt="img"></p>
<p><img src="/blog/blog/2022/02/24/2022-02-25-soapdenovo2/8.png" alt="img"></p>
<p><img src="/blog/blog/2022/02/24/2022-02-25-soapdenovo2/9.png" alt="img"></p>
<p><em><strong>d.</strong></em>  contig  graphs,在重复的节点处剪断?，输出contigs</p>
<h1 id="scaffolding"><a class="markdownIt-Anchor" href="#scaffolding"></a> scaffolding</h1>
<p><em><strong>e.</strong></em>  重新用reads和contigs进行比对，使用<font color="red"><strong>paired-end信息</strong></font>把单一的contigs连接成scaffolds。reads 比对到contigs上，临近的contig建立连接；repeat将会引来冲突矛盾信息；<font color="red"><strong>在组装成scaffold时，repeat contigs将会被屏蔽</strong></font>；paired-end信息的不同插入片段被用来一步步从短到长的建立scaffold graph(Scaffolding iteratively from short to long insert PEs./Various insert size of paired-end information is used to build contig graph step by step from short to long)。</p>
<p><img src="/blog/blog/2022/02/24/2022-02-25-soapdenovo2/10.png" alt="img"></p>
<p><img src="/blog/blog/2022/02/24/2022-02-25-soapdenovo2/11.png" alt="img"></p>
<p><img src="/blog/blog/2022/02/24/2022-02-25-soapdenovo2/12.png" alt="img"></p>
<h1 id="gap-filling"><a class="markdownIt-Anchor" href="#gap-filling"></a> Gap Filling:</h1>
<p><em><strong>f.</strong></em>  使用paired-end reads来填补scaffolds内部可能是由重复序列所造成的Gap。contig N50 通常比较小（&lt;3KB）,但是，gap filling之后能够显著提高N50值（i.e.,&gt;20KB）;Most of the gaps are repeat relative sequences.;Reads locate at gaps can collected by their paired-end which uniquely map to the contig.</p>
<p><img src="/blog/blog/2022/02/24/2022-02-25-soapdenovo2/13.png" alt="img"></p>
<h1 id="简介"><a class="markdownIt-Anchor" href="#简介"></a> 简介</h1>
<p>SOAPdenovo（目前最新版是SOAPdenovo2）是利用一种新的组装短read的方法，它以kerm为节点单位，利用<font color="green"><strong>de Bruijn图</strong></font>的方法实现全基因组的组装，和其他短序列组装软件相比，它可以进行大型基因组比如人类基因组的组装，组装结果更加准确可靠，可以通过组装的结果非常准确地鉴别出基因组上的序列结构性变异，为构建全基因组参考序列和以低测序成本对未知基因组实施精确分析创造了可能。</p>
<p>SOAPdenovo aims for large plant and animal genomes, although it also works well on bacteria and fungi genomes. It runs on 64-bit Linux system with a minimum of 5G physical memory. For <strong>big genomes</strong> like human, about <strong>150 GB memory</strong> would be required.</p>
<p>程序的下载及安装：<br>
下载地址：<a target="_blank" rel="noopener" href="http://soap.genomics.org.cn/soapdenovo.html">http://soap.genomics.org.cn/soapdenovo.html</a><br>
安装：<br>
(a) 下载SOAPdenovo的压缩包</p>
<figure class="highlight sh"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">git <span class="built_in">clone</span> https://github.com/aquaskyline/SOAPdenovo2.git</span><br><span class="line"><span class="built_in">cd</span> SOAPdenovo2</span><br><span class="line">make</span><br></pre></td></tr></table></figure>
<p>(b) 解压缩<br>
©将得到可执行文件SOAPdenovo和一个配置文件的模板example.contig</p>
<p><strong>3个可执行文件</strong></p>
<ol>
<li><font color="green"><strong>SOAPdenovo-63mer</strong></font></li>
<li><font color="green"><strong>SOAPdenovo-127mer</strong></font></li>
<li><font color="green"><strong>SOAPdenovo-fusion</strong></font></li>
</ol>
<p>第一个支持最大k-mer为63，第二个最大kmer为127，第三个文件用于单细胞测序和宏基因组测序数据。</p>
<h1 id="使用程序及参数"><a class="markdownIt-Anchor" href="#使用程序及参数"></a> 使用程序及参数</h1>
<p>SOAPdenovo可以一步跑完，也可以分成四步单独跑<br>
<font color="green"><strong>一步跑完脚本:</strong></font><br>
./ SOAPdenovo <font color="red"><strong>all</strong></font> <strong>-s</strong> lib.cfg <strong>-K</strong> 29 <strong>-D</strong> 1 <strong>-o</strong> ant &gt;&gt;ass.log<br>
<font color="green"><strong>四步单独跑的脚本</strong></font><br>
./ SOAPdenovo <font color="red"><strong>pregraph</strong></font> <strong>-s</strong> lib.cfg <strong>-d</strong> 1  <strong>-K</strong> 29 <strong>-o</strong> ant &gt;pregraph.log<br>
./ SOAPdenovo <font color="red"><strong>contig</strong></font> <strong>-g</strong> ant <strong>-D</strong> 1 <strong>-M</strong> 3 &gt;contig.log<br>
./ SOAPdenovo <font color="red"><strong>map</strong></font> <strong>-s</strong> lib23.cfg <strong>-g</strong> ant &gt;map.log<br>
./ SOAPdenovo <font color="red"><strong>scaff</strong></font> <strong>-g</strong> ant <strong>-F</strong> &gt;scaff.log</p>
<h1 id="参数说明"><a class="markdownIt-Anchor" href="#参数说明"></a> 参数说明</h1>
<p>用法：/PathToProgram/SOAPdenovo all -s configFile [-K kmer -d KmerFreqCutOff -D EdgeCovCutoff -M mergeLevel -R -u -G gapLenDiff -L minContigLen -p n_cpu] -o Output</p>
<figure class="highlight sh"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br></pre></td><td class="code"><pre><span class="line">-s    STR     配置文件√</span><br><span class="line">-o    STR     图形输出的文件名前缀 √</span><br><span class="line">-g    STR     输入文件的文件名前缀</span><br><span class="line">-K    INT     输入的K-mer值大小，默认值23，取值范围 13-63 √</span><br><span class="line">-p    INT     程序运行时设定的cpu线程数，默认值8</span><br><span class="line">-R            <span class="string">&quot;利用read鉴别短的重复序列，默认值不进行此操作&quot;</span> √</span><br><span class="line">-d    INT     去除频数不大于该值的k-mer，默认值为0;<span class="string">&quot;最小化错误测序带来的影响&quot;</span> √</span><br><span class="line">-D    INT     去除频数不大于该值的由k-mer连接的边，默认值为1，即该边上每个点的频数都小于等于1时才去除,最小化错误测序带来的影响 ?</span><br><span class="line">-M    INT     连接contig时合并相似序列的等级(强度)，默认值为1，最大值3。</span><br><span class="line">-F            利用<span class="built_in">read</span>对scaffold中的gap进行填补，默认不执行</span><br><span class="line">-u            构建scaffold前不屏蔽高覆盖度的contig，这里高频率覆盖度指平均contig覆盖深度的2倍。默认屏蔽</span><br><span class="line">-G    INT     估计gap的大小和实际补gap的大小的差异，默认值为50bp。</span><br><span class="line">-L            用于构建scaffold的contig的最短长度，默认为：Kmer参数值 ×2</span><br></pre></td></tr></table></figure>
<p><font color="blue"><strong>全局配置</strong></font></p>
<p><code>max_rd_len</code></p>
<p><font color="blue"><strong>文库配置</strong></font></p>
<p>每个文库的配置以<code>[LIB]</code>开头，主要指定输入文件的路径，支持多种格式的输入文件，用不同的前缀表示， <code>q</code>代表输入序列为fastq格式；<code>f</code>代笔输入序列为fasta格式，<code>b</code>代表输入文件为bam格式，对于双端数据，分别用后缀<code>1</code>和<code>2</code>表示R1端和R2端的reads。</p>
<p><font color="blue"><strong>主要参数</strong></font></p>
<figure class="highlight sh"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br></pre></td><td class="code"><pre><span class="line">-avg_ins    This value indicates the average insert size of this library or the peak value position <span class="keyword">in</span> the <span class="string">&quot;insert size distribution figure?&quot;</span>. <span class="comment">#文库平均插入长度(或取插入大小分布图中的峰值位置)，根据建库情况设置。 </span></span><br><span class="line"></span><br><span class="line">-reverse_seq    This option takes value 0 or 1. It tells the assembler <span class="keyword">if</span> the <span class="built_in">read</span> sequences need to be complementarily reversed. Illumima GA produces two types of paired-end libraries: a) <span class="string">&quot;forward-reverse&quot;</span>, generated from fragmented DNA ends with typical <span class="string">&quot;insert size less than 500 bp&quot;</span>; b) <span class="string">&quot;reverse-forward&quot;</span>, generated from <span class="string">&quot;circularizing libraries&quot;</span> with typical <span class="string">&quot;insert size greater than 2 Kb&quot;</span>. The parameter <span class="string">&quot;reverse_seq&quot;</span> should be <span class="built_in">set</span> to indicate this: 0, forward-reverse; 1, reverse-forward. </span><br><span class="line"></span><br><span class="line">-asm_flags    This indicator decides <span class="keyword">in</span> <span class="built_in">which</span> part(s) the reads are used. It takes value 1(only contig assembly), 2 (only scaffold assembly), <span class="string">&quot;3(both contig and scaffold assembly)&quot;</span><span class="string">&quot;, or 4 (only gap closure). </span></span><br><span class="line"><span class="string"></span></span><br><span class="line"><span class="string">-rd_len_cutof    The assembler will &quot;</span>cut the reads from the current library to this length<span class="string">&quot;.</span></span><br><span class="line"><span class="string"></span></span><br><span class="line"><span class="string">-rank    It takes integer values and decides in which &quot;</span>order the reads are used <span class="keyword">for</span> scaffold assembly<span class="string">&quot;. Libraries with the same &quot;</span>rank<span class="string">&quot; are used at the same time during scaffold assembly.</span></span><br><span class="line"><span class="string"></span></span><br><span class="line"><span class="string">-pair_num_cutoff    This parameter is the cutoff value of pair number for a &quot;</span>reliable connection<span class="string">&quot; between two contigs or pre-scaffolds. The minimum number for &quot;</span>paired-end reads and mate-pair reads is 3 and 5 respectively<span class="string">&quot;. </span></span><br><span class="line"><span class="string"></span></span><br><span class="line"><span class="string">-map_len    This takes effect in the &quot;</span>map<span class="string">&quot; step and is the minimun alignment length between a read and a contig required for a reliable read location. The minimum length for &quot;</span>paired-end reads and mate-pair reads is 32 and 35 respectively<span class="string">&quot;.</span></span><br></pre></td></tr></table></figure>
<p><font color="blue"><strong>实例</strong></font></p>
<figure class="highlight sh"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br></pre></td><td class="code"><pre><span class="line">max_rd_len=100</span><br><span class="line">[LIB]</span><br><span class="line"><span class="comment">#avg_ins 文库插入片段的平均长度，在实际设置时，可以参考文库size分布图，取峰值即可???</span></span><br><span class="line">avg_ins=200</span><br><span class="line"><span class="comment">#reverse_seq 是否需要将序列反向互补，pair-end数据，不需反向互补，设置为0；mate-pair需要设置为1</span></span><br><span class="line">reverse_seq=0</span><br><span class="line"><span class="comment">#asm_flags 1只组装contig. 2只组装scaffold,3同时组装contig和scaffold,4只补gap</span></span><br><span class="line">asm_flags=3</span><br><span class="line"><span class="comment">#序列长度阈值，作用和max_rd_len相同，大于该长度的序列会被切除到该长度</span></span><br><span class="line">rd_len_cutoff=100</span><br><span class="line"><span class="comment">#设置不同文库数据的优先级顺序，取值范围为整数，rank值相同的多个文库，在组装scaffold时，会同时使用。</span></span><br><span class="line">rank=1</span><br><span class="line">q1=fastq1_read_1.fq</span><br><span class="line">q2=fastq1_read_2.fq</span><br><span class="line">-pair_num_cutoff</span><br><span class="line"><span class="comment">#contig或者scaffold之前的最小overlap个数，pair-end数据默认为3；mate-paird数据默认为5</span></span><br><span class="line">-map_len</span><br><span class="line"><span class="comment">#比对长度的最小阈值，pair-end数据默认为32；mate-pair数据默认为35</span></span><br></pre></td></tr></table></figure>
<figure class="highlight sh"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">SOAPdenovo-63mer all -s config_file -K 63 -R -o graph_prefix</span><br></pre></td></tr></table></figure>
<h1 id="使用方法及示例"><a class="markdownIt-Anchor" href="#使用方法及示例"></a> 使用方法及示例</h1>
<figure class="highlight sh"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#（1）示例</span></span><br><span class="line">SOAPdenovo all -s HCB.lib -K 25 -d -o <span class="built_in">test</span></span><br><span class="line"><span class="comment">#（2）配置文件 configFile</span></span><br><span class="line"><span class="comment">#maximal read length （read的最大长度），该值【一般设置的比实际read读长稍微短一些？】，截去测序最后的部分，具体长度看测序质量</span></span><br><span class="line">max_rd_len=50    </span><br><span class="line"></span><br><span class="line"><span class="comment">#[LIB]文库信息以此开头</span></span><br><span class="line"></span><br><span class="line">avg_ins=200<span class="comment">#?</span></span><br><span class="line"></span><br><span class="line">reverse_seq=0</span><br><span class="line"><span class="comment">#序列是否需反转，目前测序技术，插入片段大于等于2k的采用了环化，对于插入长度大于等于2k文库，序列需要反转，reverse_seq＝1，小片段设为0</span></span><br><span class="line"></span><br><span class="line">asm_flags=3</span><br><span class="line"><span class="comment">#该文库中的read序列在组装的哪些过程（contig/scaff/fill）中用到</span></span><br><span class="line"><span class="comment">#短插入片段(&lt;2K)的设为3，同时构建contig和scaffold，长插入片段(&gt;=2k)设为2，不构建contig，只构建scaffold，454single 长reads只用于补洞。</span></span><br><span class="line"></span><br><span class="line">rank=1</span><br><span class="line"><span class="comment">#该值取整数，决定了reads用于构建scaffold的次序，值越低数据越优先用于构建scaffold。【相同rank的文库数据会同时用于组装scaffold】。一般将短插入片段设为1；2k设为2；5k设为3；10k设为4；当【某个档的数据量较大】时可以将其分为多个档，同样当【某档数据量不足够】时，可以将多个档的数据合在一起构建scaffold?。这里说的数据量够与不够是从该档的测序覆盖度和物理覆盖度两个方面考虑。</span></span><br><span class="line"></span><br><span class="line">pair_num_cutoff=3</span><br><span class="line"><span class="comment">#【可选参数】，该参数规定连接两个contig 或者是pre-scaffold 的可信连接的阈值，当连接数大于该值，连接才算有效。短插入片段(&lt;2k)默认值为3，长插入长度序列默认值为5</span></span><br><span class="line"></span><br><span class="line">map_len=32</span><br><span class="line"><span class="comment">#map_len该参数规定在map过程中 reads和contig的比对长度必须达到该值（比对不容mismacth和gap），该比对才能作为一个可信的比对。可选参数，短插入片段(&lt;2k)一般设置为32，长插入片段设置为35，默认值是K＋2。</span></span><br><span class="line"></span><br><span class="line">q1=/path/LIBNAMEA/fastq_read_1.fq<span class="comment">#read 1的fastq</span></span><br><span class="line">q2=/path/LIBNAMEA/fastq_read_2.fq<span class="comment">#read 2的fastq【注意顺序】</span></span><br><span class="line">f1=/path/LIBNAMEA/fasta_read_1.fa<span class="comment">#read 1的fa</span></span><br><span class="line">f2=/path/LIBNAMEA/fasta_read_2.fa</span><br><span class="line">q=/path/LIBNAMEA/fastq_read_single.fq<span class="comment">#单向测序fastq</span></span><br><span class="line">f=/path/LIBNAMEA/fasta_read_single.fa<span class="comment">#单向测序fa</span></span><br><span class="line">p=/path/LIBNAMEA/pairs_in_one_file.fa<span class="comment">#双向测序一个fasta</span></span><br></pre></td></tr></table></figure>
<h1 id="输出文件及说明"><a class="markdownIt-Anchor" href="#输出文件及说明"></a> 输出文件及说明</h1>
<p>SOAPdenovo 分四部分别对应的输出文件：<br>
● pregraph  生成7个文件 *.kmerFreq  *.edge  *.preArc  *.markOnEdge  *.path  *.vertex  *.preGraphBasic<br>
● contig       生成4个文件 <font color="red"> *<strong>.contig</strong></font>  *.ContigIndex  *.updated.edge  *.Arc<br>
● map          生成3个文件 *.readOnContig  *.peGrads  *.readInGap<br>
● scaff          生成6个文件 *.newContigIndex  *.links  *.scaf  *.scaf_gap  <font color="red"> *<strong>.scafSeq</strong></font>  *.gapSeq</p>
<p>*.kmerFreq             #每行显示一个数，这个数是kmer值出现的频率等于行号的kmer个数。</p>
<p><font color="red">*<strong>.contig</strong></font>：没有使用mate pair 信息的contig sequences<br>
<font color="red">*<strong>.scafSeq</strong></font>：fasta格式scaffold序列文件，contig之间的gap用N填充；<br>
<strong>对于得到的.scafSeq文件还需要用<font color="red">GapCloser</font>去合并其中的gap，最后的contig文件则是对补洞之后的scaffold文件通过打断N区的方法得到?</strong><br>
以上两个文件是组装结果中最主要的输出。</p>
<p><font color="red"><em><strong>.scaf</strong></em></font>：包括scaffold中contig的详细信息；<br>
在<em>scaffold行</em><em>中包括scaffold名字、contig长度和该scaffold长度。<br>
在</em><em>contig行</em>*包括contig名字、contig在scaffold上的起始位置、正反链、长度和contig间的链接信息<br>
<font color="red"> *.links</font>：contig间的pair-end连接信息<br>
<font color="red"> *.readOnContig</font>：reads在contig上的位置<br>
<font color="red"> *.peGrads</font>： 通过调整本文件中参数来显示构建scaffold所用到的插入片段库的个数，总共用到的read数，最长read，每个库对应的哪些reads，rank设置，pair_num_cutoff设置。例如：</p>
<blockquote>
<p>grads&amp;num: 10   522083934       70<br>
323     104577616       1       3<br>
334     180770522       1       3<br>
345     226070520       1       3<br>
486     361955834       2       3<br>
2200    392088076       3       5<br>
2290    422272580       3       5<br>
2400    445522690       3       5<br>
4870    475666064       4       5<br>
9000    511030930       5       8<br>
9110    522083934       5       5</p>
<p>该文件中共分成4列。<strong>组装的配置文件中有n个文库，该文件则有n+1行，且按照文库大小顺序排列。</strong><br>
<strong>第1行</strong>，二三四列分别是所用文库数目，reads总数和<font color="red">组装中用到的最长的reads长度？</font>。<br>
<strong>第2行</strong>，四列分别是文库大小，文库reads数目，文库reads的rank等级和pair_num_cutoff。<br>
<strong>第3～n+1行</strong>，四列分别是文库大小，文库reads数目加上前面文库中的reads总数，文库reads的rank等级pair_num_cutoff。#contig或者scaffold之前的最小overlap个数，pair-end数据默认为3；mate-paird数据默认为5,若配置文件没设置pair_num_cutoff，即使用默认参数0。</p>
</blockquote>
<p><strong>对于SOAPdenovo的每个步骤都有日志文件输出，要保存好日志文件，日志文件中包含有很多有用的信息。</strong></p>
<p>SOAPdenovo日志输出说明：</p>
<p>1）<font color="red">pregraph.log</font>:  其中有很多的统计信息，包括构建debruijn-graph所用reads数目，构图产生多少uniq的kmer以及设置-d参数后去除了多少kmer。</p>
<blockquote>
<p>-R            利用read鉴别短的重复序列，默认值不进行此操作</p>
<p>-d    INT     去除频数不大于该值的k-mer，默认值为0</p>
</blockquote>
<p>在pregraph中，可选参数有 –R –K –d  结果如：<br>
5467781332 <strong>nodes</strong> allocated, 70662750348 <strong>kmer</strong> <strong>in reads</strong>, 70662750348 <strong>kmer processed</strong><br>
3283081670 <strong>kmer removed</strong><br>
其中Kmer 数取决于k值和数据量，nodes数即特异性kmer数，当nodes数目过高**（一般和基因组大小差不多大小），可能是数据的错误率比较高，也可能是存在杂合**。若nodes数目偏小，且kmer数目多，则基因组重复度较高。</p>
<p>k值的选取，当数据量充足（&gt;=40X），<strong>植物基因组一般采用大kmer效果较好</strong>，而<strong>动物基因组k值一般取27和29足够</strong>。kmer removed表示的 <strong>–d 参数所去除的低频的kmer</strong>。</p>
<p>2）<font color="red">contig.log</font>: contig 中，可选参数 –R –D –M，注，-R 参数的选定，必须pregraph和contig中同时选择才有效。</p>
<blockquote>
<p>-R            利用read鉴别短的重复序列，默认值不进行此操作</p>
</blockquote>
<p>结果例子：<br>
16430183 pairs found, 2334584 pairs of paths compared, 1674493 pairs merged<br>
<strong>merged的数量可作为估计杂合以及测序错误的程度</strong>。<br>
sum up 1932549703bp, with average length 1170<br>
the longest is 36165bp, contig N50 is 2871 bp,contig N90 is 553 bp<br>
相关统计信息，通常<strong>植物基因组的contig N90在200bp左右</strong>，若<strong>N90高于200bp则该基因组scaffold构建不会有太大问题</strong>。动物基因组scaffold构建很少有出问题的。</p>
<p>3）<font color="red">map.log</font>:<br>
Output 415219610 out of 1956217742 (21.2)% reads in gaps<br>
<strong>1661094582 out of 1956217742 (84.9)% reads mapped to contigs</strong><br>
一般情况下，reads in gap比例和map to contig 比例<strong>总和大于1（21.2%+84.9%）</strong>。因为reads map到多个地方都被统计。当map to contig的比例很高（80%左右时），但组装效果不好可能是重复序列较多。reads in gap比例较高（大于40%）是因为基因组组装的较碎gap区域较多。<br>
map_len 默认值=K+5，短插入片段(&lt;2k)一般设置为32，长插入片段为35。取值为max{K+5,map_len}</p>
<p>4）<font color="red">scaff.log</font>:<br>
average contig coverage is 23, 5832270 contig masked<br>
构建scaffold是对高频覆盖的contig进行屏蔽（即<strong>频率高于average contig coverage的两倍的contig不用于构建scaffold(average-dep 2倍以上repeat sequences)?</strong>），从这里可以看出组装的基因组一定的重复情况。<br>
<strong>estimated PE size 162</strong>, by 40034765 pairs<br>
on contigs longer than 173, 38257479 pairs found,SD=8, insert_size estimated: 163<br>
<font color="green"><strong>173 是配置文件文库insertsize</strong></font>，<font color="green"><strong>163 是reads  map到contig上的距离的估计值</strong></font>，8是分布标准偏差。一般考虑比对上去的pair数目和SD值。若pair对数很多且SD值很小？（小片段文库数据不超过三位数，大片段文库数据不超过500），<font color="green">一般可将配置文件中<strong>文库插入片段值修改</strong>为对短插入片段文库（&lt;1k）的大小<strong>估计值</strong>，下次<strong>组装</strong>以及<strong>补洞</strong>时应根据估计值对原配置文件中的<strong>insertsize修正</strong></font>。对于大片段文库（&gt;=2K），因为是把reads map到contig上，若最长contig较短，可能找不到成pair比对上去的reads，无法估计文库大小，需要自己将大片段一级一级的map到前一级的组装结果上，然后再分析大片段文库的插入片段大小。注，<em><em>需要调整insertsize时，只需修改</em> .peGrads文件中第一列，后删除</em>.links文件，重跑scaff即可**。即构建scaffold时，主要是根据*.links文件的信息进行连接。<br>
Cutoff for number of pairs to make a reliable connection: 3<br>
1124104 weak connects removed (there were 4773564 active cnnects))<br>
Cutoff for number是在配置文件中设的pair_num_cutoff值，weak connects是低于这个值被认定为无效的连接数，active connects是满足cutoff的连接数，根据这个数值可对pair_num_cutoff做调整<br>
Picked  25241 subgraphs,4 have conflicting connections<br>
conflicting connections 是表示构建scaffold时的矛盾数，矛盾数比较高（&gt;100）时，可根据前面的有效连接数，适当提高pair_num_cutoff值，即提高scaffold连接要求的最少关系数<br>
182483 scaffolds&amp;singleton sum up 1990259817bp, with average length 10906<br>
the longest is 6561520bp,scaffold N50 is 836795 bp, scaffold N90 is 157667 bp<br>
scaffold 统计信息，将是根据rank分梯度的统计<br>
Done with 13301 scaffolds, 2161915 gaps finished, 2527441 gaps overall<br>
<strong>-F 参数补洞的统计信息。</strong></p>
<blockquote>
<p>发现scafstatistics 文件中的sca N50是整个scafSeq文件中的N50（包括不存在于scaf内的contig）。log文件中的sca N50只包括scaf 中的N50.</p>
<p>sacf.seq中可能有N，也可能无N。大部分有</p>
<p>gap finish？</p>
</blockquote>
<h1 id="参数调整"><a class="markdownIt-Anchor" href="#参数调整"></a> 参数调整</h1>
<p>组装时需调整的参数，主要分两种：</p>
<p>一种是脚本中参数改动：如调整  -K  -R  -d  -D  -M<br>
-K 值一般与基因组的特性和数据量相关，目前用到的SOAPdenovo软件主要有两个版本，SOAPdenovo-63mer是可使用大kmer组装，K值范围13-63。<br>
【<strong>经验</strong>】：植物基因组组装采用大kmer效果会比较好（要求短片段reads长度75bp），动物基因组很少有用到大kmer后有明显改进效果的，且<strong>动物基因组组装K值一般设置为27和29较多</strong>。</p>
<p><font color="red"><strong>组装杂合子基因组的K-mer值应该小一点；组装含有高repeats基因组且要求其有高的测序深度和长的reads,的K-mer应该大一点。</strong></font></p>
<p>-R参数，对于<font color="green">动物基因组R参数一般不设置</font>，<font color="red"><strong>植物基因组</strong>由于<strong>较多的repeat区</strong>则<strong>设置R参数</strong>后效果更好</font>。注意<strong>设置-R时，一般使用-M 的默认值</strong>。（熊猫基因组组装时得出的结论）</p>
<p>-M 参数，0-3,<strong>默认值1</strong>。一般杂合率为千分之几就设为几。熊猫基因组组装时-M 2 。</p>
<p>-d 参数，对于没有纠错，没有处理的质量又较差的原始数据，kmer频数为1数据较多的组装，<font color="red"><strong>设置为-d 1 足够</strong></font>。对于处理过或是测序质量较好数据可不设置。数据量较多时，也可设置-d 去除部分质量稍差数据。</p>
<p>-D 参数，默认为1，一般不用另行设置。</p>
<p>第二种，从<strong>map</strong>过程去调节参数。可以<font color="red"> 调整配置文件map_len的值和调整文件 * .peGrads  </font>。<br>
当<strong>文库插入片段分布图</strong>中文库大小与<strong>实验给出的文库大小</strong><font color="red">差异很大</font>，<font color="red"><em><em>调整</em>.peGrad</em>*s</font>文件中的插入片段大小。</p>
<p>根据每一档数据的数据量去调整文库的<strong>rank等级</strong>。</p>
<p>当该文库的数据量很多或者是在构建scaffold的过程中的冲突数很多时，可是适当的调大第四列的<strong>pair_num_cutoff</strong>，把条件设置的更严。</p>
<h1 id="内存估计"><a class="markdownIt-Anchor" href="#内存估计"></a> 内存估计</h1>
<p>SOAPdenovo四个步骤消耗的内存不一样</p>
<p>第一步消耗内存最多，使用没有纠错的的reads，(<font color="blue"> <strong>K&lt;=31</strong></font>)第一步消耗的内存在<font color="blue"><strong>基因组大小80－100倍</strong>左右</font>，<font color="blue">纠过错在<strong>40－50倍</strong>左右</font></p>
<p>第二步相对消耗内存会少很多</p>
<p>第三步消耗内存仅次于第一步，在第一步的一半左右</p>
<p>第四步消耗的内存也会比较少</p>
<p>对于CPU的使用，默认是8个，如果申请内存时申请一个计算节点的所有内存, CPU就设置为该计算节点的CPU个数充分利用计算资源，如果仅申请一个节点的部分内存则根据实际情况考虑。</p>
<p>对于<font color="blue"><strong>大kemr(K&gt;31)</strong></font>其内存使用是<font color="blue"><strong>(k&lt;=31)</strong></font>的<font color="blue"><strong>1.5倍</strong></font>左右，有时甚至更多，要充分估计内存使用，在第一次运行时考虑不能太保守。</p>
<h1 id="常见错误"><a class="markdownIt-Anchor" href="#常见错误"></a> 常见错误</h1>
<p>1）配置文件reads存储路径错误<br>
只输出日志文件。<br>
pregraph.log中的错误信息：“Cannot open /path/LIBNAMEA/fastq_read_1.fq. Now exit to system…”<br>
2）-g 后所跟参数与pregraph（第一步） -o  后所跟参数名不一致【<strong>-o对应后面的-g</strong>】<br>
contig  map  scaff 这三个步骤都只是输出日志文件。<br>
<font color="green">contig.log</font>错误信息：“Cannot open *.preGraphBasic. Now exit to system…”<br>
<font color="green">map.log</font>错误信息：“Cannot open *.contig. Now exit to system…”<br>
<font color="green">scaff.log</font>错误信息：“Cannot open *.preGraphBasic. Now exit to system…”<br>
3）<font color="blue">从map开始重新跑时，需要删除 *.links文件</font>，否则会生成core文件，程序退出。</p>
<hr>
<h1 id="github-readmemd"><a class="markdownIt-Anchor" href="#github-readmemd"></a> <a target="_blank" rel="noopener" href="http://Github-Readme.md">Github-Readme.md</a></h1>
<h2 id="configuration-file"><a class="markdownIt-Anchor" href="#configuration-file"></a> Configuration file</h2>
<p>For big genome projects with deep sequencing, <font color="red">the data is usually organized as <strong>multiple read sequence files</strong> generated from <strong>multiple libraries</strong></font>. The configuration file: where? information? example.config</p>
<p>The configuration file has a section for <font color="red"><strong>global information</strong></font>, and then <font color="red"><strong>multiple library sections</strong></font>. Right now only “max_rd_len” is included in the global information section. Any read longer than max_rd_len will be cut to this length.</p>
<p>The library information and the information of sequencing data generated from the library should be organized in the corresponding library section. <font color="red"><strong>Each library section starts with tag [LIB]</strong></font> and includes the following items:</p>
<figure class="highlight sh"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line">-avg_ins</span><br><span class="line">-reverse_seq</span><br><span class="line">-asm_flags</span><br><span class="line">-rd_len_cutof</span><br><span class="line">-rank</span><br><span class="line">-pair_num_cutoff</span><br><span class="line">-map_len</span><br></pre></td></tr></table></figure>
<p>The assembler accepts read file in three kinds of formats: <font color="green">FASTA, FASTQ and BAM</font>. Mate-pair relationship could be indicated in two ways: two sequence files with reads in the same order belonging to a pair, or two adjacent reads in a single file (FASTA only) belonging to a pair. If a read in <strong>bam</strong> file <strong>fails</strong> platform/vendor <strong>quality checks</strong>(the flag field 0x0200 is set), itself and it’s <strong>paired read</strong> would be <strong>ignored</strong>.</p>
<p>All the above items in each library section are optional. The assembler assigns default values for most of them. <strong>If you are not sure how to set a parameter, you can remove it from your configuration file</strong>.</p>
<h2 id="get-started"><a class="markdownIt-Anchor" href="#get-started"></a> Get started</h2>
<figure class="highlight sh"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br></pre></td><td class="code"><pre><span class="line"><span class="string">&quot;one step&quot;</span></span><br><span class="line">SOAPdenvo all -s config_file -K 63 -R -o graph_prefix 1&gt;ass.log 2&gt;ass.err</span><br><span class="line"><span class="comment">#-R            利用read鉴别短的重复序列，默认值不进行此操作</span></span><br><span class="line"></span><br><span class="line"><span class="comment">#step by step </span></span><br><span class="line"><span class="string">&quot;step1:&quot;</span></span><br><span class="line">SOAPdenvo pregraph -s config_file -K 63 -R -o graph_prefix 1&gt;pregraph.log 2&gt;pregraph.err</span><br><span class="line"><span class="string">&quot;OR&quot;</span></span><br><span class="line">SOAPdenvo sparse_pregraph -s config_file -K 63 -z 5000000000 -R -o graph_prefix 1&gt;pregraph.log 2&gt;pregraph.err</span><br><span class="line"><span class="comment">#-z genomize(required):estimated genome size</span></span><br><span class="line"></span><br><span class="line"><span class="string">&quot;step2:&quot;</span></span><br><span class="line">SOAPdenvo contig -g graph_prefix -R 1&gt;contig.log 2&gt;contig.err</span><br><span class="line"><span class="comment">#-g    STR     输入文件的文件名前缀</span></span><br><span class="line"></span><br><span class="line"><span class="string">&quot;step3:&quot;</span></span><br><span class="line">SOAPdenvo map -s config_file -g graph_prefix 1&gt;map.log 2&gt;map.err</span><br><span class="line"></span><br><span class="line"><span class="string">&quot;step4:&quot;</span></span><br><span class="line">SOAPdenvo scaff -g graph_prefix -F 1&gt;scaff.log 2&gt;scaff.err</span><br><span class="line"><span class="comment">#-F            利用read对scaffold中的gap进行填补，默认不执行       </span></span><br></pre></td></tr></table></figure>
<h2 id="options"><a class="markdownIt-Anchor" href="#options"></a> Options</h2>
<h3 id="options-for-all-font-colorredpregraph-contig-map-scafffont"><a class="markdownIt-Anchor" href="#options-for-all-font-colorredpregraph-contig-map-scafffont"></a> Options for all (<font color="red">pregraph-contig-map-scaff</font>)</h3>
<figure class="highlight sh"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br></pre></td><td class="code"><pre><span class="line"><span class="string">&quot;-s&quot;</span>    configFile: the config file of solexa reads</span><br><span class="line"><span class="string">&quot;-o&quot;</span>     outputGraph: prefix of <span class="string">&quot;output graph file name&quot;</span></span><br><span class="line"><span class="string">&quot;-K&quot;</span>        kmer(min 13, max 63/127): kmer size, [23]</span><br><span class="line"><span class="string">&quot;-p&quot;</span>        n_cpu: number of cpu <span class="keyword">for</span> use, [8]</span><br><span class="line"><span class="string">&quot;-a?&quot;</span>        initMemoryAssumption: memory assumption initialized to avoid further reallocation, unit G, [0]</span><br><span class="line"><span class="string">&quot;-d&quot;</span>        KmerFreqCutoff: kmers with frequency no larger than KmerFreqCutoff will be deleted, [0]	INT     去除频数不大于该值的k-mer，默认值为0</span><br><span class="line"><span class="string">&quot;-R&quot;</span> (optional)  resolve repeats by reads, [NO]</span><br><span class="line"><span class="string">&quot;-D&quot;</span>        EdgeCovCutoff: edges with coverage no larger than EdgeCovCutoff will be deleted, [1]	INT     去除频数不大于该值的由k-mer连接的边，默认值为1，即该边上每个点的频数都小于等于1时才去除</span><br><span class="line"><span class="string">&quot;-M&quot;</span>        mergeLevel(min 0, max 3): the strength of merging similar sequences during contiging, [1]	INT     连接contig时合并相似序列的等级，默认值为1，最大值3。</span><br><span class="line">-m        max k when using multi kmer</span><br><span class="line">-e        weight to filter arc when linearize two edges(default 0)</span><br><span class="line">-r (optional)  keep available <span class="built_in">read</span>(*.<span class="built_in">read</span>)</span><br><span class="line">-E (optional)  merge clean bubble before iterate</span><br><span class="line">-f (optional)  output gap related reads <span class="keyword">in</span> map step <span class="keyword">for</span> using SRkgf to fill gap, [NO]</span><br><span class="line">-k        kmer_R2C(min 13, max 63): kmer size used <span class="keyword">for</span> mapping <span class="built_in">read</span> to contig, [K]</span><br><span class="line">-F (optional)  fill gaps <span class="keyword">in</span> scaffold, [NO]利用<span class="built_in">read</span>对scaffold中的gap进行填补，默认不执行</span><br><span class="line">-u (optional)  un-mask contigs with high/low coverage before scaffolding, [mask]构建scaffold前不屏蔽高覆盖度的contig，这里高频率覆盖度指平均contig覆盖深度的2倍。默认屏蔽</span><br><span class="line">-w (optional)  keep contigs weakly connected to other contigs <span class="keyword">in</span> scaffold, [NO]</span><br><span class="line">-G        gapLenDiff: allowed length difference between estimated and filled gap, [50]INT     估计gap的大小和实际补gap的大小的差异，默认值为50bp。</span><br><span class="line">-L        minContigLen: shortest contig <span class="keyword">for</span> scaffolding, [K+2]用于构建scaffold的contig的最短长度，默认为：Kmer参数值 ×2</span><br><span class="line">-c      minContigCvg: <span class="string">&quot;minimum contig coverage&quot;</span> (c*avgCvg), contigs shorter than 100bp with coverage smaller than c*avgCvg will be masked before scaffolding unless -u is <span class="built_in">set</span>, [0.1]</span><br><span class="line">-C      maxContigCvg: <span class="string">&quot;maximum contig coverage&quot;</span> (C*avgCvg), contigs with coverage larger than C*avgCvg or contigs shorter than 100bp with coverage larger than 0.8*C*avgCvg will be masked before scaffolding unless -u is <span class="built_in">set</span>, [2]</span><br><span class="line">-b      insertSizeUpperBound: (b*avg_ins) will be used as upper bound of insert size <span class="keyword">for</span> large insert size ( &gt; 1000) when handling pair-end connections between contigs <span class="keyword">if</span> b is <span class="built_in">set</span> to larger than 1, [1.5]</span><br><span class="line">-B      bubbleCoverage: remove contig with lower cvoerage <span class="keyword">in</span> bubble structure <span class="keyword">if</span> both contigs<span class="string">&#x27; coverage are smaller than bubbleCoverage*avgCvg, [0.6]</span></span><br><span class="line"><span class="string">-N        genomeSize: genome size for statistics, [0]</span></span><br><span class="line"><span class="string">-V (optional)  output visualization information of assembly, [NO]</span></span><br></pre></td></tr></table></figure>
<h3 id="options-for-sparse_pregraph"><a class="markdownIt-Anchor" href="#options-for-sparse_pregraph"></a> Options for sparse_pregraph</h3>
<figure class="highlight sh"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br></pre></td><td class="code"><pre><span class="line">Usage: ./SOAPdenovo2 sparse_pregraph -s configFile -K kmer -z genomeSize -o outputGraph [-g maxKmerEdgeLength -d kmerFreqCutoff -e kmerEdgeFreqCutoff -R -r runMode -p n_cpu]</span><br><span class="line">  -s      configFile: the config file of solexa reads</span><br><span class="line">  -K         kmer(min 13, max 63/127): kmer size, [23]</span><br><span class="line">  -g         maxKmerEdgeLength(min 1, max 25): number of skipped intermediate kmers, [15]</span><br><span class="line">  <span class="string">&quot;-z&quot;</span>         genomeSize(mandatory): estimated genome size</span><br><span class="line">  <span class="string">&quot;-d&quot;</span>         kmerFreqCutoff: delete kmers with frequency no larger than,[1]</span><br><span class="line">  -e         kmerEdgeFreqCutoff: delete kmers<span class="string">&#x27; related edge with frequency no larger than [1]</span></span><br><span class="line"><span class="string">  -R (optional)   output extra information for resolving repeats in contig step, [NO]</span></span><br><span class="line"><span class="string">  -r         runMode: 0 build graph &amp; build edge and preArc, 1 load graph by prefix &amp; build edge and preArc, 2 build graph only, 3 build edges only, 4 build preArcs only [0]</span></span><br><span class="line"><span class="string">  &quot;-p&quot;         n_cpu: number of cpu for use,[8]</span></span><br><span class="line"><span class="string">  -o         outputGraph: prefix of output graph file name</span></span><br></pre></td></tr></table></figure>
<figure class="highlight sh"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br></pre></td><td class="code"><pre><span class="line">contigs：</span><br><span class="line">      -g  &lt;string&gt;      输入graph file文件名前缀 </span><br><span class="line">    -R  (optional)    移除repeats，使用pregraph步骤中产生的结果，如果参数-R在pregraph步骤中被设置的话，默认[NO]</span><br><span class="line">    -D  &lt;int&gt;         去除频数不大于该值（edgeCovCutoff）的由k-mer连接的边，默认值[1]，即该边上每个点的频数都小于等于1时才去除。edges with coverage no larger than EdgeCovCutoff will be deleted, [1] <span class="comment">#最小化错误测序带来的影响</span></span><br><span class="line">    -M  &lt;int&gt;         在contiging操作时，合并相似序列的强度，默认值为[1]，最小值0，最大值3。<span class="comment">#deal with heterozygosis</span></span><br><span class="line">    -e  &lt;int&gt;         两边缘之间的弧的权重大于该值（arcWeight），将被线性化，默认值[0]</span><br><span class="line">    -m  &lt;int&gt;         最大的kmer size（max 127）用于multi-kmer，默认[NO]</span><br><span class="line">    -E  (optional)    在iterate迭代之前，合并clean bubble功能，仅在当使用multi-kmer时且设置-M参数，默认[NO]</span><br><span class="line"> </span><br><span class="line">map：</span><br><span class="line">    -g  &lt;string&gt;      输入graph file文件名前缀 </span><br><span class="line">    -k  &lt;int&gt;         该值（kmer_R2C）是kmer size用于mapping reads到contigs上时的值，默认值[K]</span><br><span class="line">    -f  (optional)    在map那一步中，对于使用SRkgf去填充gap，输出与gap相关的reads,默认[NO]</span><br><span class="line"> </span><br><span class="line">scaffold：</span><br><span class="line">      -F  (optional)    利用<span class="built_in">read</span>对scaffold中的gap进行填补，默认[NO]</span><br><span class="line">    -u  (optional)    构建scaffolding前不屏蔽高/低覆盖度的contigs，这里高频率覆盖度指平均contig覆盖深度的2倍。默认[mask]屏蔽</span><br><span class="line">    -w  (optional)    在scaffold中，保持contigs弱连接于其他contigs，默认[NO]</span><br><span class="line">    -G  &lt;int&gt;         估计gap的大小和实际补gap的大小的差异值，默认值[50]bp。</span><br><span class="line">    -L  &lt;int&gt;         用于构建scaffold的contig的最短长度(minContigLen)，默认为：[Kmer参数值+2]</span><br><span class="line">    -c  &lt;<span class="built_in">float</span>&gt;       在scaffolding之前，contigs小于100bp，且覆盖率小于该最小contig覆盖率（c*avgCvg），将被屏蔽，除非参数-u已设定，默认值[0.1]</span><br><span class="line">    -C  &lt;<span class="built_in">float</span>&gt;       在scaffolding之前，contigs覆盖率大于该最大contigs覆盖率（C*avgCvg），或者contigs小于100bp且覆盖率大于0.8*（C*avgCvg），将被屏蔽，除非参数-u已设定，默认值[2]</span><br><span class="line">    -b  &lt;<span class="built_in">float</span>&gt;       当处理contigs间的pair-end连接时，如果参数-b&gt;1，该插入片段的上限值（b*avg_ins）将被用来作为大的插入片段（&gt;1000）的上限，默认值[1.5]</span><br><span class="line">    -B  &lt;<span class="built_in">float</span>&gt;       如果两个contigs的覆盖率都小于bubble覆盖率（bubbleCoverage）乘以contigs平均覆盖率（bubbleCoverage*avgCvg），则去除在bubble结构中的低覆盖率contig，默认值[0.6]</span><br><span class="line">    -N  &lt;int&gt;         统计基因组大小，默认值[0]</span><br><span class="line">    -V  (optional)    输出相关信息为了可视化组装，默认[NO]</span><br></pre></td></tr></table></figure>
<h2 id="output-files"><a class="markdownIt-Anchor" href="#output-files"></a> Output files</h2>
<figure class="highlight sh"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line">a. <span class="string">&quot;*.contig&quot;</span></span><br><span class="line">  contig sequences without using mate pair information.</span><br><span class="line">b. <span class="string">&quot;*.scafSeq&quot;</span></span><br><span class="line">  scaffold sequences (final contig sequences can be extracted by breaking down scaffold sequences at gap regions).</span><br></pre></td></tr></table></figure>
<h2 id="faq"><a class="markdownIt-Anchor" href="#faq"></a> FAQ</h2>
<h3 id="how-to-set-k-mer-size"><a class="markdownIt-Anchor" href="#how-to-set-k-mer-size"></a> How to set K-mer size?</h3>
<p>The program accepts odd numbers between 13 and 31? <strong>Larger K-mers would have higher rate of uniqueness in the genome and would make the graph simpler, but it requires deep sequencing depth and longer read length to guarantee the overlap at any genomic location.</strong> <font color="red"><strong>Balance?</strong></font></p>
<p>The <strong>sparse_pregraph</strong> module usually needs <strong>2-10bp smaller kmer</strong> length to achieve the same performance as the original <strong>pregraph</strong> module. <font color="red"><strong>Difference?</strong></font></p>
<h3 id="how-to-set-genome-size-z-for-sparse_pregraph-module"><a class="markdownIt-Anchor" href="#how-to-set-genome-size-z-for-sparse_pregraph-module"></a> How to set genome size(-z) for sparse_pregraph module?</h3>
<p>The -z parameter for sparse pregraph should be set <font color="red"><strong>a litter larger than the real genome size</strong></font>, it is used to allocate memory.</p>
<h3 id="how-to-set-library-rank"><a class="markdownIt-Anchor" href="#how-to-set-library-rank"></a> How to set library rank?</h3>
<p>SOAPdenovo will <font color="red">use the <strong>pair-end libraries</strong> with <strong>insert size</strong> from <strong>smaller to larger</strong> to <strong>construct</strong> scaffolds</font>. Libraries with the same rank would be used at the same time. For example, in a dataset of a human genome, we set <strong><font color="red">five ranks</font> for five libraries with insert size 200-bp, 500-bp, 2-Kb, 5-Kb and 10-Kb</strong>, separately. It is desired that the pairs in each rank provide adequate physical coverage of the genome.</p>
<h2 id="appendix-a-an-exampleconfig"><a class="markdownIt-Anchor" href="#appendix-a-an-exampleconfig"></a> APPENDIX A: an example.config</h2>
<figure class="highlight sh"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br><span class="line">47</span><br><span class="line">48</span><br><span class="line">49</span><br><span class="line">50</span><br><span class="line">51</span><br><span class="line">52</span><br><span class="line">53</span><br><span class="line">54</span><br><span class="line">55</span><br><span class="line">56</span><br><span class="line">57</span><br><span class="line">58</span><br><span class="line">59</span><br><span class="line">60</span><br><span class="line">61</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#maximal read length</span></span><br><span class="line">max_rd_len=100</span><br><span class="line"><span class="string">&quot;[LIB]&quot;</span></span><br><span class="line"><span class="comment">#average insert size</span></span><br><span class="line">avg_ins=200</span><br><span class="line"><span class="comment">#if sequence needs to be reversed</span></span><br><span class="line">reverse_seq=0</span><br><span class="line"><span class="comment">#in which part(s) the reads are used</span></span><br><span class="line">asm_flags=3</span><br><span class="line"><span class="comment">#use only first 100 bps of each read</span></span><br><span class="line">rd_len_cutoff=100</span><br><span class="line"><span class="comment">#in which order the reads are used while scaffolding</span></span><br><span class="line">rank=1</span><br><span class="line"><span class="comment"># cutoff of pair number for a reliable connection (at least 3 for short insert size)</span></span><br><span class="line">pair_num_cutoff=3</span><br><span class="line"><span class="comment">#minimum aligned length to contigs for a reliable read location (at least 32 for short insert size)</span></span><br><span class="line">map_len=32</span><br><span class="line"><span class="comment">#a pair of fastq file, read 1 file should always be followed by read 2 file</span></span><br><span class="line">q1=/path/**LIBNAMEA**/fastq1_read_1.fq</span><br><span class="line">q2=/path/**LIBNAMEA**/fastq1_read_2.fq</span><br><span class="line"><span class="comment">#another pair of fastq file, read 1 file should always be followed by read 2 file</span></span><br><span class="line">q1=/path/**LIBNAMEA**/fastq2_read_1.fq</span><br><span class="line">q2=/path/**LIBNAMEA**/fastq2_read_2.fq</span><br><span class="line"><span class="comment">#a pair of fasta file, read 1 file should always be followed by read 2 file</span></span><br><span class="line">f1=/path/**LIBNAMEA**/fasta1_read_1.fa</span><br><span class="line">f2=/path/**LIBNAMEA**/fasta1_read_2.fa</span><br><span class="line"><span class="comment">#another pair of fasta file, read 1 file should always be followed by read 2 file</span></span><br><span class="line">f1=/path/**LIBNAMEA**/fasta2_read_1.fa</span><br><span class="line">f2=/path/**LIBNAMEA**/fasta2_read_2.fa</span><br><span class="line"><span class="comment">#fastq file for single reads</span></span><br><span class="line">q=/path/**LIBNAMEA**/fastq1_read_single.fq</span><br><span class="line"><span class="comment">#another fastq file for single reads</span></span><br><span class="line">q=/path/**LIBNAMEA**/fastq2_read_single.fq</span><br><span class="line"><span class="comment">#fasta file for single reads</span></span><br><span class="line">f=/path/**LIBNAMEA**/fasta1_read_single.fa</span><br><span class="line"><span class="comment">#another fasta file for single reads</span></span><br><span class="line">f=/path/**LIBNAMEA**/fasta2_read_single.fa</span><br><span class="line"><span class="comment">#a single fasta file for paired reads</span></span><br><span class="line">p=/path/**LIBNAMEA**/pairs1_in_one_file.fa</span><br><span class="line"><span class="comment">#another single fasta file for paired reads</span></span><br><span class="line">p=/path/**LIBNAMEA**/pairs2_in_one_file.fa</span><br><span class="line"><span class="comment">#bam file for single or paired reads, reads 1 in paired reads file should always be followed by reads 2</span></span><br><span class="line"><span class="comment">#	<span class="doctag">NOTE:</span> If a read in bam file fails platform/vendor quality checks(the flag field 0x0200 is set), itself and it&#x27;s paired read would be ignored.</span></span><br><span class="line">b=/path/**LIBNAMEA**/reads1_in_file.bam</span><br><span class="line"><span class="comment">#another bam file for single or paired reads</span></span><br><span class="line">b=/path/**LIBNAMEA**/reads2_in_file.bam</span><br><span class="line"><span class="string">&quot;[LIB]&quot;</span></span><br><span class="line">avg_ins=2000</span><br><span class="line">reverse_seq=1</span><br><span class="line">asm_flags=2</span><br><span class="line">rank=2</span><br><span class="line"><span class="comment"># cutoff of pair number for a reliable connection (at least 5 for large insert size)</span></span><br><span class="line">pair_num_cutoff=5</span><br><span class="line"><span class="comment">#minimum aligned length to contigs for a reliable read location (at least 35 for large insert size)</span></span><br><span class="line">map_len=35</span><br><span class="line">q1=/path/**LIBNAMEB**/fastq_read_1.fq</span><br><span class="line">q2=/path/**LIBNAMEB**/fastq_read_2.fq</span><br><span class="line">f1=/path/**LIBNAMEA**/fasta_read_1.fa</span><br><span class="line">f2=/path/**LIBNAMEA**/fasta_read_2.fa</span><br><span class="line">p=/path/**LIBNAMEA**/pairs_in_one_file.fa</span><br><span class="line">b=/path/**LIBNAMEA**/reads_in_file.bam</span><br></pre></td></tr></table></figure>
<h2 id="appendix-b-output-files"><a class="markdownIt-Anchor" href="#appendix-b-output-files"></a> Appendix B: output files</h2>
<h3 id="1-output-files-from-the-command-pregraph"><a class="markdownIt-Anchor" href="#1-output-files-from-the-command-pregraph"></a> 1. Output files from the command “pregraph”</h3>
<figure class="highlight sh"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br></pre></td><td class="code"><pre><span class="line">a. *.kmerFreq</span><br><span class="line">   Each row shows the number of Kmers with a frequency equals the row number. Note that those peaks of frequencies <span class="built_in">which</span> are the integral multiple of 63 are due to the data structure.</span><br><span class="line">b. *.edge</span><br><span class="line">   Each record gives the information of an edge <span class="keyword">in</span> the pre-graph: length, Kmers on both ends, average kmer coverage, whether it<span class="string">&#x27;s reverse-complementarily identical and the sequence.</span></span><br><span class="line"><span class="string">c. *.markOnEdge &amp; *.path</span></span><br><span class="line"><span class="string">   These two files are for using reads to solve small repeats.</span></span><br><span class="line"><span class="string">e. *.preArc</span></span><br><span class="line"><span class="string">   Connections between edges which are established by the read paths.</span></span><br><span class="line"><span class="string">f. *.vertex</span></span><br><span class="line"><span class="string">   Kmers at the ends of edges.</span></span><br><span class="line"><span class="string">g. *.preGraphBasic</span></span><br><span class="line"><span class="string">   Some basic information about the pre-graph: number of vertex, K value, number of edges, maximum read length etc.</span></span><br></pre></td></tr></table></figure>
<h3 id="2-output-files-from-the-command-contig"><a class="markdownIt-Anchor" href="#2-output-files-from-the-command-contig"></a> 2. Output files from the command “contig”</h3>
<figure class="highlight sh"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br></pre></td><td class="code"><pre><span class="line">a. *.contig</span><br><span class="line">      Contig information: corresponding edge index, length, kmer coverage, whether it<span class="string">&#x27;s tip and the sequence. Either a contig or its reverse complementry counterpart is included. Each reverse complementary contig index is indicated in the *.ContigIndex file.</span></span><br><span class="line"><span class="string">   b. *.Arc</span></span><br><span class="line"><span class="string">      Arcs coming out of each edge and their corresponding coverage by reads</span></span><br><span class="line"><span class="string">   c. *.updated.edge</span></span><br><span class="line"><span class="string">      Some information for each edge in graph: length, Kmers at both ends, index difference between the reverse-complementary edge and this one.</span></span><br><span class="line"><span class="string">   d. *.ContigIndex</span></span><br><span class="line"><span class="string">      Each record gives information about each contig in the *.contig: it&#x27;</span>s edge index, length, the index difference between its reverse-complementary counterpart and itself.</span><br></pre></td></tr></table></figure>
<h3 id="3-output-files-from-the-command-map"><a class="markdownIt-Anchor" href="#3-output-files-from-the-command-map"></a> 3. Output files from the command “map”</h3>
<figure class="highlight sh"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line">a. *.peGrads</span><br><span class="line">      Information <span class="keyword">for</span> each <span class="built_in">clone</span> library: insert-size, <span class="built_in">read</span> index upper bound, rank and pair number cutoff <span class="keyword">for</span> a reliable link. This file can be revised manually <span class="keyword">for</span> scaffolding tuning.</span><br><span class="line">   b. *.readOnContig</span><br><span class="line">      Reads<span class="string">&#x27; locations on contigs. Here contigs are referred by their edge index. Howerver about half of them are not listed in the *.contig file for their reverse-complementary counterparts are included already.</span></span><br><span class="line"><span class="string">   c. *.readInGap</span></span><br><span class="line"><span class="string">      This file includes reads that could be located in gaps between contigs. This information will be used to close gaps in scaffolds if &quot;-F&quot; is set.</span></span><br></pre></td></tr></table></figure>
<h3 id="4-output-files-from-the-command-scaff"><a class="markdownIt-Anchor" href="#4-output-files-from-the-command-scaff"></a> 4. Output files from the command “scaff”</h3>
<figure class="highlight sh"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br></pre></td><td class="code"><pre><span class="line">a. *.newContigIndex</span><br><span class="line">      Contigs are sorted according their length before scaffolding. Their new index are listed <span class="keyword">in</span> this file.  This is useful <span class="keyword">if</span> one wants to corresponds contigs <span class="keyword">in</span> *.contig with those <span class="keyword">in</span> *.links.</span><br><span class="line">   b. *.links</span><br><span class="line">      Links between contigs <span class="built_in">which</span> are established by <span class="built_in">read</span> pairs. New index are used.</span><br><span class="line">   c. *.scaf_gap</span><br><span class="line">      Contigs <span class="keyword">in</span> gaps found by contig graph outputted by the contiging procedure. Here new index are used.</span><br><span class="line">   d. *.scaf</span><br><span class="line">      Contigs <span class="keyword">for</span> each scaffold: contig index (concordant to index <span class="keyword">in</span> *.contig),  approximate start position on scaffold, orientation, contig length, and its links to others contigs.</span><br><span class="line">   e. *.gapSeq</span><br><span class="line">      Gap sequences between contigs.</span><br><span class="line">   f. *.scafSeq</span><br><span class="line">      Sequences of each scaffolds.</span><br><span class="line">   g. *.contigPosInscaff</span><br><span class="line">      Contigs<span class="string">&#x27; positions in each scaffold.</span></span><br><span class="line"><span class="string">   h. *.bubbleInScaff</span></span><br><span class="line"><span class="string">      Contigs that form bubble structures in scaffolds. Every two contigs form a bubble and the contig with higher coverage will be kept in scaffold.</span></span><br><span class="line"><span class="string">   i. *.scafStatistics</span></span><br><span class="line"><span class="string">      Statistic information of final scaffold and contig.</span></span><br></pre></td></tr></table></figure>
<hr>
<h1 id="小记1"><a class="markdownIt-Anchor" href="#小记1"></a> 小记1</h1>
<p>ref：<a target="_blank" rel="noopener" href="https://www.cnblogs.com/Formulate0303/p/6879841.html">https://www.cnblogs.com/Formulate0303/p/6879841.html</a></p>
<h2 id="背景"><a class="markdownIt-Anchor" href="#背景"></a> 背景</h2>
<p>确定基因组缺失什么；确定难以生化研究的基因和pathways；研究感兴趣pathway通路中每一个基因；研究基因组非编码区（introns、promoters、telomeres端粒等）的调控机理和结构特征；基因组提供可进行各种统计的大型数据库（provide large databases that are amenable to statistical methods）;识别不同可能有细微表型的序列；研究物种和基因组的进化过程。一般都是用不同梯度的插入片段来测序，小片段（200,500,800）和大片段（1k, 2kb 5kb 10kb 20kb 40kb）。</p>
<h2 id="测序前准备"><a class="markdownIt-Anchor" href="#测序前准备"></a> 测序前准备</h2>
<p>搜集物种相关信息（染色体的倍型、基因组大小、杂合度、重复序列比例、是否有可用的遗传图谱、GC含量 和 GC分布）。提供已经发表的近源物种。根据近源物种分析以上信息，尤其是GC含量以及对应的GC分布，重复程度。</p>
<h3 id="1-获取基因组大小"><a class="markdownIt-Anchor" href="#1-获取基因组大小"></a> 1 获取基因组大小</h3>
<ul>
<li>
<p>基因组太大（&gt;10Gb）超出了目前denovo组装基因组软件对机器内存的要求，则无法实现组装；对组装结果的大小的进行正确性与否判断。</p>
</li>
<li>
<p>动物基因组大小数据库（ANIMAL GENOME SIZE DATABASE）: <a target="_blank" rel="noopener" href="http://www.genomesize.com/%EF%BC%9B%E5%AF%B9%E4%BA%8E%E6%9F%A5%E4%B8%8D%E5%88%B0%E7%9A%84%E7%89%A9%E7%A7%8D%E5%9F%BA%E5%9B%A0%E7%BB%84%E5%A4%A7%E5%B0%8F%E7%9A%84%EF%BC%8C%E5%8F%AF%E4%BB%A5%E9%80%9A%E8%BF%87%E4%B8%80%E4%BA%9B%E6%96%B9%E6%B3%95%E5%8E%BB%E4%BC%B0%E8%AE%A1%E3%80%82">http://www.genomesize.com/；对于查不到的物种基因组大小的，可以通过一些方法去估计。</a></p>
</li>
<li>
<p>实验（流式细胞仪）估计基因组大小的例子： “A full-length enriched cDNA library and expressed sequence tag analysis of the parasitic weed, Striga hermonthica.” BMC Plant Biol (2010)</p>
</li>
<li>
<p>基于福尔摩根染色估计基因组大小：书The evolution of the genome《基因组进化》, Gregory, T. (2005).</p>
</li>
<li>
<p>定量PCR估计基因组大小: “Real-time PCR-based method for the estimation of genome sizes.” Nucleic Acids Res (2003); “The nuclear genome of the phytoseiid Metaseiulus occidentalis (Acari: Phytoseiidae) is among the smallest known in arthropods.” Exp Appl Acarol (2009)</p>
</li>
<li>
<p>通过Kmer估计基因组大小：Kim, E. B., X. Fang, et al. (2011). “<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3319411/"><strong>Genome sequencing reveals insights into physiology and longevity of the naked mole rat</strong></a>.” Nature 479(7372): 223-227. [Obtained 2.5 G contig sequences with N50 19.3 kbp and N90 4.7 kbp, and 2.7 G scaffold sequences with N50 1.6 Mbp and N90 0.3 Mbp.]</p>
</li>
</ul>
<h3 id="2-杂合度评估"><a class="markdownIt-Anchor" href="#2-杂合度评估"></a> 2 杂合度评估</h3>
<ul>
<li>
<p>主要体现在<font color="green">不能合并姊妹染色体</font>，杂合度高的区域，会把两条姊妹染色单体都组装出来，造成组装的基因组<font color="green">偏大</font>。</p>
</li>
<li>
<p><font color="red"><strong>杂合度&gt;0.5%则组装有一定难度</strong></font>。杂合度&gt;1%则很难组装出来。</p>
</li>
<li>
<p>杂合度高则组装的序列不适合用于后续生物学分析（eg:拷贝数、基因完整结构等）。</p>
</li>
</ul>
<h3 id="3-是否有遗传图谱可用"><a class="markdownIt-Anchor" href="#3-是否有遗传图谱可用"></a> 3 是否有遗传图谱可用</h3>
<p>随着测序对质量提高和相关技术成熟，<font color="red"><strong>遗传图谱？</strong></font>也快成了denovo基因组的必须组成。</p>
<p><a target="_blank" rel="noopener" href="https://cdmd.cnki.com.cn/Article/CDMD-10504-1019065822.htm">矮牵牛遗传图谱构建及重瓣分子标记筛选</a></p>
<p>遗传图谱是某一物种的染色体图，显示所知的基因和/或遗传标记的相对位置，而不是在每条染色体上特殊的物理位置。由<a target="_blank" rel="noopener" href="https://baike.baidu.com/item/%E9%81%97%E4%BC%A0%E9%87%8D%E7%BB%84/6391485">遗传重组</a>测验结果推算出来的、在一条染色体上可以发生的突变座位的直线排列（<a target="_blank" rel="noopener" href="https://baike.baidu.com/item/%E5%9F%BA%E5%9B%A0%E4%BD%8D%E7%82%B9/3613206">基因位点</a>的排列）图。</p>
<p>Genetic map，<a target="_blank" rel="noopener" href="https://baike.baidu.com/item/%E9%81%97%E4%BC%A0%E5%9B%BE/2951032">遗传图</a>或遗传<a target="_blank" rel="noopener" href="https://baike.baidu.com/item/%E8%BF%9E%E9%94%81%E5%9B%BE/1586423">连锁图</a>是以<a target="_blank" rel="noopener" href="https://baike.baidu.com/item/%E5%9F%BA%E5%9B%A0%E8%BF%9E%E9%94%81/7740242">基因连锁</a>、重组<a target="_blank" rel="noopener" href="https://baike.baidu.com/item/%E4%BA%A4%E6%8D%A2%E5%80%BC/5106108">交换值</a>构建的图谱，图距为cM（<a target="_blank" rel="noopener" href="https://baike.baidu.com/item/%E5%8E%98%E6%91%A9/1586351">厘摩</a>），1%交换值为1cM，约相当于1000kb。人基因组全长约3300cM，如两个标记之间相距1cM，则需3300个标记，如相距2～5cM，则需660～l650个标记。标记可以是体质性状，也可以是可检出的DNA序列，例如基因、<a target="_blank" rel="noopener" href="https://baike.baidu.com/item/%E9%99%90%E5%88%B6%E7%89%87%E6%AE%B5%E9%95%BF%E5%BA%A6%E5%A4%9A%E6%80%81%E6%80%A7/3841692">限制片段长度多态性</a>（<a target="_blank" rel="noopener" href="https://baike.baidu.com/item/RFLP/5060944">RFLP</a>）和特定的<a target="_blank" rel="noopener" href="https://baike.baidu.com/item/%E5%8D%95%E4%B8%80%E5%BA%8F%E5%88%97/1851974">单一序列</a>等。每一个标记都要用专一序列位点（STS）作鉴定。STS是指其位置及核苷酸序列均已知的、人基因组中只有一份拷贝的DNA短片段（一般长200～500碱基对），它很容易用多聚酶链反应（PCR）加以验证。当各个实验室报道定位和测序的数据时，可用STS来确定这些DNA片段的定位与取向。人类基因组计划的研究目标是，完成一个完全连接的人类遗传图，标记之间平均相距2～5cM。</p>
<p>遗传图谱构建相关概念推荐参考书：<strong>The handbook of plant genome mapping: genetic and physical mapping</strong></p>
<h3 id="4-生物学问题的调研"><a class="markdownIt-Anchor" href="#4-生物学问题的调研"></a> 4 生物学问题的调研</h3>
<h2 id="组装"><a class="markdownIt-Anchor" href="#组装"></a> 组装</h2>
<ul>
<li>
<p>BAC-by-BAC:测序和组装每一个BAC, 合并BAC和移除BAC冗余部分，获得参考基因组序列。</p>
</li>
<li>
<p>whole genome shotgun：全基因组鸟枪法，染色体DNA被随机打断成片段，依次测序和组装。</p>
</li>
</ul>
<h3 id="算法"><a class="markdownIt-Anchor" href="#算法"></a> 算法</h3>
<p>二代测序数据从头组装的解决overlap的三种算法</p>
<ul>
<li>
<p>overlap-layout-consensus：重叠布局一致OLC法，【软件：PHRAP.NEWBLER.CABOG.CELERA.SHORTY.EDENA,popular for long reads】，1. Overlap discovery involves all-against-all, pair-wise read comparison. 2. Construction an approximate read layout according to the pair-wise alignment 3. Multiple sequence alignment determines the precise layout and the consensus.</p>
</li>
<li>
<p>De bruijn graph:DBG法，【软件：SOAPdenovo2.Velvet.EULER,popular for illumina ，for short reads】,1.所有的测序reads都被切割成某一固定Kmer长度的序列（21bp=&lt;kmer&lt;=127bp）.2.相邻kmers链接是来自read序列，所以它不需要成对序列比对(The links between neighboring Kmers are derived from read sequences,so it doesn’t need pair-wise reads alignment.）3.冗余的数据自动被压缩。</p>
</li>
<li>
<p>greedy method：贪婪法(use OLC or DBG)，【软件：SSAKE.SHARCGS.VCAKE】，从给定的reads和contigs开始，使用下一个得分最高的overlap去做下一个连接，这样一直做下去，直到不能进行下去为止。</p>
</li>
</ul>
<h3 id="评估"><a class="markdownIt-Anchor" href="#评估"></a> 评估</h3>
<p>组装short reads的挑战</p>
<p>1.基因组的复杂性，<strong>重复序列</strong>、<strong>杂合</strong>的<strong>二倍体</strong>基因组heterozygous diploid genome、<strong>多倍性</strong>polyploidy.</p>
<p>2.illumina reads 的数据特征，<strong>测序错误</strong>率~1%、short read 长度<sub>100bp、</sub>100X的高测序深度、不同级别的文库插入片段（200bp~40Kbp）。</p>
<p>3.Complexity of computation</p>
<h2 id><a class="markdownIt-Anchor" href="#"></a> </h2>
<hr>
<h1 id="小记2"><a class="markdownIt-Anchor" href="#小记2"></a> 小记2</h1>
<p>ref <a target="_blank" rel="noopener" href="https://www.jianshu.com/p/e02ecd537c83">https://www.jianshu.com/p/e02ecd537c83</a></p>
<h2 id="reads质量控制"><a class="markdownIt-Anchor" href="#reads质量控制"></a> reads质量控制</h2>
<ul>
<li>认识数据。(read类型，read数量，其GC含量，可能的污染和其他问题)</li>
<li>数据清理。(组装前清理原始数据可以使组装结果更好,因为移除了低质量和污染的reads)</li>
<li>为组装软件提供一些组装的参数</li>
</ul>
<h2 id="检查原始read数据的质量"><a class="markdownIt-Anchor" href="#检查原始read数据的质量"></a> 检查原始read数据的质量</h2>
<p>(1)read长度:在设置组装的最大k-mer值时将很重要。</p>
<p>(2)质量编码类型:对于quality trimming软件很重要。(Phred33、64)</p>
<p>(3)GC含量:<strong>高GC的生物体往往组装得不好，并且read覆盖率分布可能不均匀</strong></p>
<p>(4)总的reads数:<strong>了解coverage范围</strong></p>
<p>(4)在read的开始、中间或结尾附近质量下降:确定可能的修整/清除方法和参数。</p>
<p>(5)出现<strong>高度重复的k-mers</strong>:说明序列可能存在污染。</p>
<p>(6)reads中存在<strong>大量N</strong>:可能表明测序质量较差。你需要修剪这些read，以删除N.</p>
<h2 id="genome-survey"><a class="markdownIt-Anchor" href="#genome-survey"></a> genome survey</h2>
<p>正式组装前进行kmer分析，以了解以下信息。</p>
<p>(1)基因组杂合度。</p>
<p>(2)重复序列比例。</p>
<p>(3)预估基因组大小。</p>
<p>(4)测序深度。</p>
<p>所用软件为kmergenie，<a target="_blank" rel="noopener" href="https://links.jianshu.com/go?to=http%3A%2F%2Fkmergenie.bx.psu.edu%2F">http://kmergenie.bx.psu.edu/</a></p>
<figure class="highlight sh"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#安装(使用时，确保服务器装有Pthon与R)</span></span><br><span class="line">tar -xzvf kmergenie-1.7051.tar.gz &amp;&amp; <span class="built_in">cd</span> kmergenie-1.7051</span><br><span class="line">python setup.py install</span><br><span class="line"> </span><br><span class="line"><span class="comment">#添加可执行权限</span></span><br><span class="line">chmod -R 755 *</span><br></pre></td></tr></table></figure>
<p>接下来介绍怎么用，参考：<a target="_blank" rel="noopener" href="https://links.jianshu.com/go?to=http%3A%2F%2Fwap.sciencenet.cn%2Fblog-3406804-1159967.html">http://wap.sciencenet.cn/blog-3406804-1159967.html</a></p>
<figure class="highlight sh"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br></pre></td><td class="code"><pre><span class="line">touch fq_501.txt                                 <span class="comment">#存储fastq文件的路径。</span></span><br><span class="line">vim fq_501.txt                                   <span class="comment">#编辑fq.txt，输入fastq文件路径。</span></span><br><span class="line">mkdir kmergenie_result     </span><br><span class="line">./tools/kmergenie-1.7051/kmergenie fq_501.list -o ./kmergenie_result/PB_501 -k 105 -</span><br><span class="line">l 15 -s 10 -t 12                               </span><br><span class="line"><span class="comment">#-o:输出文件存放的地址。</span></span><br><span class="line"><span class="comment">#-k:最大的k值</span></span><br><span class="line"><span class="comment">#-l:最小的k值</span></span><br><span class="line"><span class="comment">#-s:从最小的k----最大的k，每次增加的k。</span></span><br><span class="line"><span class="comment">#-t:线程数。</span></span><br><span class="line"><span class="comment">#注:刚开始s可以设大点，可以根据判断出的最佳k值，缩小s再进行判断((第一次:I:15,k:105,s:10 第二次I:85,k:140,s:6)。</span></span><br></pre></td></tr></table></figure>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br><span class="line">47</span><br></pre></td><td class="code"><pre><span class="line">usage:</span><br><span class="line"></span><br><span class="line">    ./kmergenie reads_file</span><br><span class="line"></span><br><span class="line">reads_file is either a single FASTA, FASTQ, FASTA.gz, FASTQ.gz file or a list of file names, one per line. For example:</span><br><span class="line">    </span><br><span class="line">    ls -1 *.fastq.gz &gt; list_files</span><br><span class="line">    ./kmergenie list_files</span><br><span class="line"></span><br><span class="line">    type ./kmergenie to see extra options</span><br><span class="line"></span><br><span class="line">input:</span><br><span class="line"></span><br><span class="line">    Input reads should be exactly those the de novo assembler will use to create contigs, i.e. the list of all single and paired-end reads.</span><br><span class="line"></span><br><span class="line">    The order does not matter, KmerGenie treats the reads as an unordered set of k-mers. Orientation of the reads also does not matter.</span><br><span class="line">    With Velvet, if you have mate-pairs, Velvet uses them to create contigs, so do include them in KmerGenie.</span><br><span class="line">    Otherwise, if the mate-pairs are used only for scaffolding (i.e. asm_flag=2 in SOAPdenovo), do not include them.</span><br><span class="line"></span><br><span class="line">tips:</span><br><span class="line"></span><br><span class="line">    * Take a look at the generated HTML report. It provides a summary of the results, and contains an Advanced Help section to help analysis.</span><br><span class="line"></span><br><span class="line">    * There is no need to use Kmergenie for a multi-k assembler, like SPAdes. Default parameters of multi-k assemblers are generally better than a single best k.</span><br><span class="line"></span><br><span class="line">    * By default, KmerGenie will perform another pass to estimate k more precisely. You may skip it by using the &quot;--one-pass&quot; option (roughly 2x faster).</span><br><span class="line"></span><br><span class="line">    * To run multiple instances of KmerGenie on the same folder, specify the &quot;-o&quot; and &quot;-t&quot; parameters (output prefix, number of threads per instance).</span><br><span class="line"></span><br><span class="line">    * The sampling value e (&quot;-e&quot; parameter) makes sure that roughly n/e k-mers are sampled out of the n distinct kmers in the dataset. The datasets are fully analyzed, though. Kmergenie samples from the set of distinct kmers, not from the set of reads.</span><br><span class="line"></span><br><span class="line">    * The diploid model should only be used for moderate-high heterozygosity rates. The haploid model works better when only one peak is visible in the histograms (indicating low heterozygosity).</span><br><span class="line"></span><br><span class="line">outputs:</span><br><span class="line"></span><br><span class="line">    histograms_report.html</span><br><span class="line">        html report of all the results</span><br><span class="line">    histograms.dat</span><br><span class="line">        file containing the raw data for the number of genomic kmers for each k value</span><br><span class="line">    histograms-k*.histo</span><br><span class="line">        files containing raw sampled histograms for each k value</span><br><span class="line">    histograms-k*.histo.pdf</span><br><span class="line">        plots of the histograms and the fits. Colors: red is the fit of the complete statistical model of the histogram (erroneous k-mers + genomic k-mers). With the diploid model, green are only the heterozygous k-mers, blue are only the homozygous k-mers.</span><br><span class="line">    &lt;stdout&gt;</span><br><span class="line">        last line is the best k value</span><br><span class="line">    &lt;return code / error code&gt;</span><br><span class="line">        0 if and only if a best k was found (note: no longer returns the best k as error code)</span><br></pre></td></tr></table></figure>
<p>结果文件生成report。报告以折线图形式给出每种长度的kmer下，估计的基因组大小，同时给出了最佳的kmer(评估基因组总大小最高)</p>
<p><img src="/blog/blog/2022/02/24/2022-02-25-soapdenovo2/14.png" alt="img"></p>
<p>理想条件该图应该为一条平滑的曲线，有明确最大值。然而在一些情况下有多个局部最大值。</p>
<p>这些情况表明，对于某些k值，Kmergenie中的统计模型并不总是正确地拟合输入数据。因此由Kmergenie预测的最佳k值可能不是最佳的。在随后的分析中，还尝试使用比Kmergenie预测的k大的k(其他峰值)</p>
<p>若基因组k-mers的数量不会随着k值的升高而下降，则是一个高覆盖率(high coverage)数据集的迹象？<br>
什么是覆盖率呢？<br>
覆盖率(coverage)：指的是测序过程中读取单个碱基的平均次数。如果覆盖度为100×，说明平均每个碱基测序了100次。对碱基测序的频率越高，数据的质量越高。</p>
<p><img src="/blog/blog/2022/02/24/2022-02-25-soapdenovo2/15.png" alt="img"></p>
<h2 id="soapdevovo2"><a class="markdownIt-Anchor" href="#soapdevovo2"></a> soapdevovo2</h2>
<figure class="highlight sh"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#正式组装(kmer选103,113,123,127分别跑)</span></span><br><span class="line"><span class="comment">#SOAPdenovo2既可以一步组装也可以分四步，如果基因组大且复杂建议分四步。</span></span><br><span class="line"> nohup ./tools/SOAPdenovo2-master/SOAPdenovo-127mer all -s config_file -d 1 -R -F -K 113 -p 60 -o PB_501_113 &gt; PB_501_113.log &amp;  <span class="comment">#注:加nohup和&amp;是为了让它在后台运行(防止远程连接断开程序停止)</span></span><br></pre></td></tr></table></figure>
<figure class="highlight sh"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#分四步(在k=103、113、123、及127(支持的最大kmer)跑了后发现127在同样参数下contig N50最大)，下面是为了优化结果跑的。</span></span><br><span class="line"><span class="comment">#pregraph(d=1,2,3,4)</span></span><br><span class="line">nohup ./tools/SOAPdenovo2-master/SOAPdenovo-127mer pregraph -s config_file -o PB_501_127_d_3 -R -K 127 -p 60 -d 2 &gt;PB_501_127_d_3_pre.log &amp;</span><br><span class="line">nohup ./tools/SOAPdenovo2-master/SOAPdenovo-127mer contig -g PB_501_127_d_3 -R -p 60 &gt;PB_501_127_d_3_con.log &amp;</span><br><span class="line">nohup ./tools/SOAPdenovo2-master/SOAPdenovo-127mer map -s config_file -g PB_501_127_d_3 -p 60 -k 127 &gt;PB_501_127_d_3_map.log &amp;  </span><br><span class="line">nohup ./tools/SOAPdenovo2-master/SOAPdenovo-127mer scaff -g PB_501_127_d_3 -F -p 60 &gt;PB_501_127_d_3_scaf.log &amp;</span><br></pre></td></tr></table></figure>
<h2 id="统计"><a class="markdownIt-Anchor" href="#统计"></a> 统计</h2>
<p><img src="/blog/blog/2022/02/24/2022-02-25-soapdenovo2/16.png" alt="img"></p>
<h2 id="组装评估"><a class="markdownIt-Anchor" href="#组装评估"></a> 组装评估</h2>
<p>QUAST软件下载:<a target="_blank" rel="noopener" href="https://links.jianshu.com/go?to=https%3A%2F%2Fsourceforge.net%2Fprojects%2Fquast%2Ffiles%2Fquast-5.0.2.tar.gz%2Fdownload">https://sourceforge.net/projects/quast/files/quast-5.0.2.tar.gz/download</a></p>
<figure class="highlight sh"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#解压安装</span></span><br><span class="line">tar -zxvf quast-5.0.2.tar.gz &amp;&amp; <span class="built_in">cd</span> quast-5.0.2</span><br><span class="line">python  setup.py install_full</span><br></pre></td></tr></table></figure>
<h3 id="参考基因组下载"><a class="markdownIt-Anchor" href="#参考基因组下载"></a> 参考基因组下载</h3>
<h3 id="quast使用"><a class="markdownIt-Anchor" href="#quast使用"></a> QUAST使用</h3>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line">./tools/quast-5.0.2/quast.py -o ./quast_results -R ./reference/IRGSP-1.0_genome.fasta.gz -t 70 ./PB_501_results/PB_501_127_d_1/PB_501_127_d_1.scafSeq ./PB_501_results/PB_501_127_d_2/PB_501_127_d_2.scafSeq ./PB_501_results/PB_501_127_d_3/PB_501_127_d_3.scafSeq ./PB_501_results/PB_501_127_d_6/PB_501_127_d_6.scafSeq</span><br><span class="line"><span class="comment">#-o输出目录</span></span><br><span class="line"><span class="comment">#-R参考基因组序列</span></span><br><span class="line"><span class="comment">#-t线程数</span></span><br><span class="line"><span class="comment">#后面为要比较的contig/scaffold所在目录。</span></span><br></pre></td></tr></table></figure>
<h2 id="结果"><a class="markdownIt-Anchor" href="#结果"></a> 结果</h2>
<h3 id="统计结果"><a class="markdownIt-Anchor" href="#统计结果"></a> 统计结果</h3>
<p>将contig/scaffold比对到参考基因组上，下面是比对得到的结果。</p>
<p><img src="/blog/blog/2022/02/24/2022-02-25-soapdenovo2/17.png" alt="img"></p>
<p><img src="/blog/blog/2022/02/24/2022-02-25-soapdenovo2/18.png" alt="img"></p>
<p>累计长度:若与灰色线重合越好，组装结果越好。</p>
<p><img src="/blog/blog/2022/02/24/2022-02-25-soapdenovo2/19.png" alt="img"></p>
<p>错误组装情况：</p>
<p><img src="/blog/blog/2022/02/24/2022-02-25-soapdenovo2/20.png" alt="img"></p>
<p>GC含量:</p>
<p><img src="/blog/blog/2022/02/24/2022-02-25-soapdenovo2/21.png" alt="img"></p>
<h3 id="contigscaffold大小可视化"><a class="markdownIt-Anchor" href="#contigscaffold大小可视化"></a> contig/scaffold大小可视化</h3>
<p><img src="/blog/blog/2022/02/24/2022-02-25-soapdenovo2/22.png" alt="img"></p>
<h3 id="比对到每条染色体情况"><a class="markdownIt-Anchor" href="#比对到每条染色体情况"></a> 比对到每条染色体情况</h3>
<p><img src="/blog/blog/2022/02/24/2022-02-25-soapdenovo2/23.png" alt="img"></p>
<p>比对到每条染色体可视化情况:</p>
<p><img src="/blog/blog/2022/02/24/2022-02-25-soapdenovo2/24.png" alt="img"></p>

    </div>

    
    
    
        

<div>
<ul class="post-copyright">
  <li class="post-copyright-author">
    <strong>Post author:  </strong>Wang C
  </li>
  <li class="post-copyright-link">
    <strong>Post link: </strong>
    <a href="https://wangc1106.github.io/blog/2022/02/24/2022-02-25-soapdenovo2/" title="Soapdenovo2">https://wangc1106.github.io/blog/2022/02/24/2022-02-25-soapdenovo2/</a>
  </li>
  <li class="post-copyright-license">
    <strong>Copyright Notice:  </strong>All articles in this blog are licensed under <a href="https://creativecommons.org/licenses/by-nc-sa/4.0/" rel="noopener" target="_blank"><i class="fab fa-fw fa-creative-commons"></i>BY-NC-SA</a> unless stating additionally.
  </li>
</ul>
</div>


      <footer class="post-footer">
          <div class="post-tags">
              <a href="/blog/tags/SOAPdenovo2/" rel="tag"># SOAPdenovo2</a>
          </div>

        


        
    <div class="post-nav">
      <div class="post-nav-item">
    <a href="/blog/2022/02/24/2022-02-24-Seqkit/" rel="prev" title="seqkit">
      <i class="fa fa-chevron-left"></i> seqkit
    </a></div>
      <div class="post-nav-item">
    <a href="/blog/2022/02/28/2022-02-28-genome-assemble/" rel="next" title="Genome assemble pipeline">
      Genome assemble pipeline <i class="fa fa-chevron-right"></i>
    </a></div>
    </div>
      </footer>
    
  </article>
  
  
  



          </div>
          

<script>
  window.addEventListener('tabs:register', () => {
    let { activeClass } = CONFIG.comments;
    if (CONFIG.comments.storage) {
      activeClass = localStorage.getItem('comments_active') || activeClass;
    }
    if (activeClass) {
      let activeTab = document.querySelector(`a[href="#comment-${activeClass}"]`);
      if (activeTab) {
        activeTab.click();
      }
    }
  });
  if (CONFIG.comments.storage) {
    window.addEventListener('tabs:click', event => {
      if (!event.target.matches('.tabs-comment .tab-content .tab-pane')) return;
      let commentClass = event.target.classList[1];
      localStorage.setItem('comments_active', commentClass);
    });
  }
</script>

        </div>
          
  
  <div class="toggle sidebar-toggle">
    <span class="toggle-line toggle-line-first"></span>
    <span class="toggle-line toggle-line-middle"></span>
    <span class="toggle-line toggle-line-last"></span>
  </div>

  <aside class="sidebar">
    <div class="sidebar-inner">

      <ul class="sidebar-nav motion-element">
        <li class="sidebar-nav-toc">
          Table of Contents
        </li>
        <li class="sidebar-nav-overview">
          Overview
        </li>
      </ul>

      <!--noindex-->
      <div class="post-toc-wrap sidebar-panel">
          <div class="post-toc motion-element"><ol class="nav"><li class="nav-item nav-level-1"><a class="nav-link" href="#soapdenovo2"><span class="nav-number">1.</span> <span class="nav-text"> soapdenovo2:</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#contiging"><span class="nav-number">2.</span> <span class="nav-text"> contiging</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#scaffolding"><span class="nav-number">3.</span> <span class="nav-text"> scaffolding</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#gap-filling"><span class="nav-number">4.</span> <span class="nav-text"> Gap Filling:</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#%E7%AE%80%E4%BB%8B"><span class="nav-number">5.</span> <span class="nav-text"> 简介</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#%E4%BD%BF%E7%94%A8%E7%A8%8B%E5%BA%8F%E5%8F%8A%E5%8F%82%E6%95%B0"><span class="nav-number">6.</span> <span class="nav-text"> 使用程序及参数</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#%E5%8F%82%E6%95%B0%E8%AF%B4%E6%98%8E"><span class="nav-number">7.</span> <span class="nav-text"> 参数说明</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#%E4%BD%BF%E7%94%A8%E6%96%B9%E6%B3%95%E5%8F%8A%E7%A4%BA%E4%BE%8B"><span class="nav-number">8.</span> <span class="nav-text"> 使用方法及示例</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#%E8%BE%93%E5%87%BA%E6%96%87%E4%BB%B6%E5%8F%8A%E8%AF%B4%E6%98%8E"><span class="nav-number">9.</span> <span class="nav-text"> 输出文件及说明</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#%E5%8F%82%E6%95%B0%E8%B0%83%E6%95%B4"><span class="nav-number">10.</span> <span class="nav-text"> 参数调整</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#%E5%86%85%E5%AD%98%E4%BC%B0%E8%AE%A1"><span class="nav-number">11.</span> <span class="nav-text"> 内存估计</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#%E5%B8%B8%E8%A7%81%E9%94%99%E8%AF%AF"><span class="nav-number">12.</span> <span class="nav-text"> 常见错误</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#github-readmemd"><span class="nav-number">13.</span> <span class="nav-text"> Github-Readme.md</span></a><ol class="nav-child"><li class="nav-item nav-level-2"><a class="nav-link" href="#configuration-file"><span class="nav-number">13.1.</span> <span class="nav-text"> Configuration file</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#get-started"><span class="nav-number">13.2.</span> <span class="nav-text"> Get started</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#options"><span class="nav-number">13.3.</span> <span class="nav-text"> Options</span></a><ol class="nav-child"><li class="nav-item nav-level-3"><a class="nav-link" href="#options-for-all-font-colorredpregraph-contig-map-scafffont"><span class="nav-number">13.3.1.</span> <span class="nav-text"> Options for all (pregraph-contig-map-scaff)</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#options-for-sparse_pregraph"><span class="nav-number">13.3.2.</span> <span class="nav-text"> Options for sparse_pregraph</span></a></li></ol></li><li class="nav-item nav-level-2"><a class="nav-link" href="#output-files"><span class="nav-number">13.4.</span> <span class="nav-text"> Output files</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#faq"><span class="nav-number">13.5.</span> <span class="nav-text"> FAQ</span></a><ol class="nav-child"><li class="nav-item nav-level-3"><a class="nav-link" href="#how-to-set-k-mer-size"><span class="nav-number">13.5.1.</span> <span class="nav-text"> How to set K-mer size?</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#how-to-set-genome-size-z-for-sparse_pregraph-module"><span class="nav-number">13.5.2.</span> <span class="nav-text"> How to set genome size(-z) for sparse_pregraph module?</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#how-to-set-library-rank"><span class="nav-number">13.5.3.</span> <span class="nav-text"> How to set library rank?</span></a></li></ol></li><li class="nav-item nav-level-2"><a class="nav-link" href="#appendix-a-an-exampleconfig"><span class="nav-number">13.6.</span> <span class="nav-text"> APPENDIX A: an example.config</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#appendix-b-output-files"><span class="nav-number">13.7.</span> <span class="nav-text"> Appendix B: output files</span></a><ol class="nav-child"><li class="nav-item nav-level-3"><a class="nav-link" href="#1-output-files-from-the-command-pregraph"><span class="nav-number">13.7.1.</span> <span class="nav-text"> 1. Output files from the command “pregraph”</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#2-output-files-from-the-command-contig"><span class="nav-number">13.7.2.</span> <span class="nav-text"> 2. Output files from the command “contig”</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#3-output-files-from-the-command-map"><span class="nav-number">13.7.3.</span> <span class="nav-text"> 3. Output files from the command “map”</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#4-output-files-from-the-command-scaff"><span class="nav-number">13.7.4.</span> <span class="nav-text"> 4. Output files from the command “scaff”</span></a></li></ol></li></ol></li><li class="nav-item nav-level-1"><a class="nav-link" href="#%E5%B0%8F%E8%AE%B01"><span class="nav-number">14.</span> <span class="nav-text"> 小记1</span></a><ol class="nav-child"><li class="nav-item nav-level-2"><a class="nav-link" href="#%E8%83%8C%E6%99%AF"><span class="nav-number">14.1.</span> <span class="nav-text"> 背景</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#%E6%B5%8B%E5%BA%8F%E5%89%8D%E5%87%86%E5%A4%87"><span class="nav-number">14.2.</span> <span class="nav-text"> 测序前准备</span></a><ol class="nav-child"><li class="nav-item nav-level-3"><a class="nav-link" href="#1-%E8%8E%B7%E5%8F%96%E5%9F%BA%E5%9B%A0%E7%BB%84%E5%A4%A7%E5%B0%8F"><span class="nav-number">14.2.1.</span> <span class="nav-text"> 1 获取基因组大小</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#2-%E6%9D%82%E5%90%88%E5%BA%A6%E8%AF%84%E4%BC%B0"><span class="nav-number">14.2.2.</span> <span class="nav-text"> 2 杂合度评估</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#3-%E6%98%AF%E5%90%A6%E6%9C%89%E9%81%97%E4%BC%A0%E5%9B%BE%E8%B0%B1%E5%8F%AF%E7%94%A8"><span class="nav-number">14.2.3.</span> <span class="nav-text"> 3 是否有遗传图谱可用</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#4-%E7%94%9F%E7%89%A9%E5%AD%A6%E9%97%AE%E9%A2%98%E7%9A%84%E8%B0%83%E7%A0%94"><span class="nav-number">14.2.4.</span> <span class="nav-text"> 4 生物学问题的调研</span></a></li></ol></li><li class="nav-item nav-level-2"><a class="nav-link" href="#%E7%BB%84%E8%A3%85"><span class="nav-number">14.3.</span> <span class="nav-text"> 组装</span></a><ol class="nav-child"><li class="nav-item nav-level-3"><a class="nav-link" href="#%E7%AE%97%E6%B3%95"><span class="nav-number">14.3.1.</span> <span class="nav-text"> 算法</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#%E8%AF%84%E4%BC%B0"><span class="nav-number">14.3.2.</span> <span class="nav-text"> 评估</span></a></li></ol></li><li class="nav-item nav-level-2"><a class="nav-link"><span class="nav-number">14.4.</span> <span class="nav-text"> </span></a></li></ol></li><li class="nav-item nav-level-1"><a class="nav-link" href="#%E5%B0%8F%E8%AE%B02"><span class="nav-number">15.</span> <span class="nav-text"> 小记2</span></a><ol class="nav-child"><li class="nav-item nav-level-2"><a class="nav-link" href="#reads%E8%B4%A8%E9%87%8F%E6%8E%A7%E5%88%B6"><span class="nav-number">15.1.</span> <span class="nav-text"> reads质量控制</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#%E6%A3%80%E6%9F%A5%E5%8E%9F%E5%A7%8Bread%E6%95%B0%E6%8D%AE%E7%9A%84%E8%B4%A8%E9%87%8F"><span class="nav-number">15.2.</span> <span class="nav-text"> 检查原始read数据的质量</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#genome-survey"><span class="nav-number">15.3.</span> <span class="nav-text"> genome survey</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#soapdevovo2"><span class="nav-number">15.4.</span> <span class="nav-text"> soapdevovo2</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#%E7%BB%9F%E8%AE%A1"><span class="nav-number">15.5.</span> <span class="nav-text"> 统计</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#%E7%BB%84%E8%A3%85%E8%AF%84%E4%BC%B0"><span class="nav-number">15.6.</span> <span class="nav-text"> 组装评估</span></a><ol class="nav-child"><li class="nav-item nav-level-3"><a class="nav-link" href="#%E5%8F%82%E8%80%83%E5%9F%BA%E5%9B%A0%E7%BB%84%E4%B8%8B%E8%BD%BD"><span class="nav-number">15.6.1.</span> <span class="nav-text"> 参考基因组下载</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#quast%E4%BD%BF%E7%94%A8"><span class="nav-number">15.6.2.</span> <span class="nav-text"> QUAST使用</span></a></li></ol></li><li class="nav-item nav-level-2"><a class="nav-link" href="#%E7%BB%93%E6%9E%9C"><span class="nav-number">15.7.</span> <span class="nav-text"> 结果</span></a><ol class="nav-child"><li class="nav-item nav-level-3"><a class="nav-link" href="#%E7%BB%9F%E8%AE%A1%E7%BB%93%E6%9E%9C"><span class="nav-number">15.7.1.</span> <span class="nav-text"> 统计结果</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#contigscaffold%E5%A4%A7%E5%B0%8F%E5%8F%AF%E8%A7%86%E5%8C%96"><span class="nav-number">15.7.2.</span> <span class="nav-text"> contig&#x2F;scaffold大小可视化</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#%E6%AF%94%E5%AF%B9%E5%88%B0%E6%AF%8F%E6%9D%A1%E6%9F%93%E8%89%B2%E4%BD%93%E6%83%85%E5%86%B5"><span class="nav-number">15.7.3.</span> <span class="nav-text"> 比对到每条染色体情况</span></a></li></ol></li></ol></li></ol></div>
      </div>
      <!--/noindex-->

      <div class="site-overview-wrap sidebar-panel">
        <div class="site-author motion-element" itemprop="author" itemscope itemtype="http://schema.org/Person">
    <img class="site-author-image" itemprop="image" alt="Wang C"
      src="/blog/images/banner.png">
  <p class="site-author-name" itemprop="name">Wang C</p>
  <div class="site-description" itemprop="description">Make more time</div>
</div>
<div class="site-state-wrap motion-element">
  <nav class="site-state">
      <div class="site-state-item site-state-posts">
          <a href="/blog/archives">
          <span class="site-state-item-count">36</span>
          <span class="site-state-item-name">posts</span>
        </a>
      </div>
      <div class="site-state-item site-state-categories">
            <a href="/blog/categories/">
          
        <span class="site-state-item-count">12</span>
        <span class="site-state-item-name">categories</span></a>
      </div>
      <div class="site-state-item site-state-tags">
            <a href="/blog/tags/">
          
        <span class="site-state-item-count">19</span>
        <span class="site-state-item-name">tags</span></a>
      </div>
  </nav>
</div>
  <div class="links-of-author motion-element">
      <span class="links-of-author-item">
        <a href="https://github.com/wangc1106" title="GitHub → https:&#x2F;&#x2F;github.com&#x2F;wangc1106" rel="noopener" target="_blank"><i class="fab fa-github fa-fw"></i>GitHub</a>
      </span>
      <span class="links-of-author-item">
        <a href="mailto:wangchen116@126.com" title="E-Mail → mailto:wangchen116@126.com" rel="noopener" target="_blank"><i class="fa fa-envelope fa-fw"></i>E-Mail</a>
      </span>
  </div>
  <div class="cc-license motion-element" itemprop="license">
    <a href="https://creativecommons.org/licenses/by-nc-sa/4.0/" class="cc-opacity" rel="noopener" target="_blank"><img src="/blog/images/cc-by-nc-sa.svg" alt="Creative Commons"></a>
  </div>



      </div>
        <div class="back-to-top motion-element">
          <i class="fa fa-arrow-up"></i>
          <span>0%</span>
        </div>

    </div>
  </aside>
  <div id="sidebar-dimmer"></div>


      </div>
    </main>

    <footer class="footer">
      <div class="footer-inner">
        

        

<div class="copyright">
  
  &copy; 2021 – 
  <span itemprop="copyrightYear">2022</span>
  <span class="with-love">
    <i class="fas fa-dna"></i>
  </span>
  <span class="author" itemprop="copyrightHolder">Wang C</span>
</div>

        






<script>
  (function() {
    function leancloudSelector(url) {
      url = encodeURI(url);
      return document.getElementById(url).querySelector('.leancloud-visitors-count');
    }

    function addCount(Counter) {
      var visitors = document.querySelector('.leancloud_visitors');
      var url = decodeURI(visitors.id);
      var title = visitors.dataset.flagTitle;

      Counter('get', '/classes/Counter?where=' + encodeURIComponent(JSON.stringify({ url })))
        .then(response => response.json())
        .then(({ results }) => {
          if (results.length > 0) {
            var counter = results[0];
            leancloudSelector(url).innerText = counter.time + 1;
            Counter('put', '/classes/Counter/' + counter.objectId, { time: { '__op': 'Increment', 'amount': 1 } })
              .catch(error => {
                console.error('Failed to save visitor count', error);
              });
          } else {
              Counter('post', '/classes/Counter', { title, url, time: 1 })
                .then(response => response.json())
                .then(() => {
                  leancloudSelector(url).innerText = 1;
                })
                .catch(error => {
                  console.error('Failed to create', error);
                });
          }
        })
        .catch(error => {
          console.error('LeanCloud Counter Error', error);
        });
    }

    function showTime(Counter) {
      var visitors = document.querySelectorAll('.leancloud_visitors');
      var entries = [...visitors].map(element => {
        return decodeURI(element.id);
      });

      Counter('get', '/classes/Counter?where=' + encodeURIComponent(JSON.stringify({ url: { '$in': entries } })))
        .then(response => response.json())
        .then(({ results }) => {
          for (let url of entries) {
            let target = results.find(item => item.url === url);
            leancloudSelector(url).innerText = target ? target.time : 0;
          }
        })
        .catch(error => {
          console.error('LeanCloud Counter Error', error);
        });
    }

    let { app_id, app_key, server_url } = {"enable":true,"app_id":"UqIYufmNk2sE8MiBKKWc4Kja-gzGzoHsz","app_key":"iYpeWQcmNqpHwr2uVlnJYVji","server_url":null,"security":false};
    function fetchData(api_server) {
      var Counter = (method, url, data) => {
        return fetch(`${api_server}/1.1${url}`, {
          method,
          headers: {
            'X-LC-Id'     : app_id,
            'X-LC-Key'    : app_key,
            'Content-Type': 'application/json',
          },
          body: JSON.stringify(data)
        });
      };
      if (CONFIG.page.isPost) {
        if (CONFIG.hostname !== location.hostname) return;
        addCount(Counter);
      } else if (document.querySelectorAll('.post-title-link').length >= 1) {
        showTime(Counter);
      }
    }

    let api_server = app_id.slice(-9) !== '-MdYXbMMI' ? server_url : `https://${app_id.slice(0, 8).toLowerCase()}.api.lncldglobal.com`;

    if (api_server) {
      fetchData(api_server);
    } else {
      fetch('https://app-router.leancloud.cn/2/route?appId=' + app_id)
        .then(response => response.json())
        .then(({ api_server }) => {
          fetchData('https://' + api_server);
        });
    }
  })();
</script>


      </div>
      <div style="height: 200px">
      <script type="text/javascript" src="//rf.revolvermaps.com/0/0/8.js?i=5knf34gs07t&amp;m=0&amp;c=ff0000&amp;cr1=ffffff&amp;f=arial&amp;l=33" async="async"></script>
      </div>
    </footer>
  </div>

  
  <script src="/blog/lib/anime.min.js"></script>
  <script src="/blog/lib/velocity/velocity.min.js"></script>
  <script src="/blog/lib/velocity/velocity.ui.min.js"></script>

<script src="/blog/js/utils.js"></script>

<script src="/blog/js/motion.js"></script>


<script src="/blog/js/schemes/pisces.js"></script>


<script src="/blog/js/next-boot.js"></script>




  




  
<script src="/blog/js/local-search.js"></script>













  

  

</body>
</html>
