<!DOCTYPE html>
<html lang="zh-Hans">
<head>
  <meta charset="UTF-8">
<meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=2">
<meta name="theme-color" content="#222">
<meta name="generator" content="Hexo 5.4.0">
  <link rel="apple-touch-icon" sizes="180x180" href="/blog/images/apple-touch-icon-next.png">
  <link rel="icon" type="image/png" sizes="32x32" href="/blog/images/favicon-32x32-next.png">
  <link rel="icon" type="image/png" sizes="16x16" href="/blog/images/favicon-16x16-next.png">
  <link rel="mask-icon" href="/blog/images/logo.svg" color="#222">
  <meta name="google-site-verification" content="1osTo_gOe1SW9Hevqtz8vl8HfSgTO4DSMZ0DiMPMX6w">

<link rel="stylesheet" href="/blog/css/main.css">


<link rel="stylesheet" href="/blog/lib/font-awesome/css/all.min.css">

<script id="hexo-configurations">
    var NexT = window.NexT || {};
    var CONFIG = {"hostname":"wangc1106.github.io","root":"/blog/","scheme":"Gemini","version":"7.8.0","exturl":false,"sidebar":{"position":"left","display":"post","padding":18,"offset":12,"onmobile":false},"copycode":{"enable":false,"show_result":false,"style":null},"back2top":{"enable":true,"sidebar":true,"scrollpercent":"ture"},"bookmark":{"enable":false,"color":"#222","save":"auto"},"fancybox":false,"mediumzoom":false,"lazyload":false,"pangu":false,"comments":{"style":"tabs","active":null,"storage":true,"lazyload":false,"nav":null},"algolia":{"hits":{"per_page":10},"labels":{"input_placeholder":"Search for Posts","hits_empty":"We didn't find any results for the search: ${query}","hits_stats":"${hits} results found in ${time} ms"}},"localsearch":{"enable":true,"trigger":"auto","top_n_per_article":1,"unescape":false,"preload":false},"motion":{"enable":true,"async":false,"transition":{"post_block":"fadeIn","post_header":"slideDownIn","post_body":"slideDownIn","coll_header":"slideLeftIn","sidebar":"slideUpIn"}},"path":"search.xml"};
  </script>

  <meta name="description" content="SeqKit - a cross-platform and ultrafast toolkit for FASTA&#x2F;Q file manipulation">
<meta property="og:type" content="article">
<meta property="og:title" content="seqkit">
<meta property="og:url" content="https://wangc1106.github.io/blog/2022/02/24/2022-02-24-Seqkit/index.html">
<meta property="og:site_name" content="Banner&#39;s Blog">
<meta property="og:description" content="SeqKit - a cross-platform and ultrafast toolkit for FASTA&#x2F;Q file manipulation">
<meta property="og:locale">
<meta property="article:published_time" content="2022-02-24T08:21:15.010Z">
<meta property="article:modified_time" content="2022-02-24T13:38:53.525Z">
<meta property="article:author" content="Wang C">
<meta property="article:tag" content="sequence-tool">
<meta name="twitter:card" content="summary">

<link rel="canonical" href="https://wangc1106.github.io/blog/2022/02/24/2022-02-24-Seqkit/">


<script id="page-configurations">
  // https://hexo.io/docs/variables.html
  CONFIG.page = {
    sidebar: "",
    isHome : false,
    isPost : true,
    lang   : 'zh-Hans'
  };
</script>

  <title>seqkit | Banner's Blog</title>
  






  <noscript>
  <style>
  .use-motion .brand,
  .use-motion .menu-item,
  .sidebar-inner,
  .use-motion .post-block,
  .use-motion .pagination,
  .use-motion .comments,
  .use-motion .post-header,
  .use-motion .post-body,
  .use-motion .collection-header { opacity: initial; }

  .use-motion .site-title,
  .use-motion .site-subtitle {
    opacity: initial;
    top: initial;
  }

  .use-motion .logo-line-before i { left: initial; }
  .use-motion .logo-line-after i { right: initial; }
  </style>
</noscript>

</head>

<body itemscope itemtype="http://schema.org/WebPage">
  <div class="container use-motion">
    <div class="headband"></div>

    <header class="header" itemscope itemtype="http://schema.org/WPHeader">
      <div class="header-inner"><div class="site-brand-container">
  <div class="site-nav-toggle">
    <div class="toggle" aria-label="Toggle navigation bar">
      <span class="toggle-line toggle-line-first"></span>
      <span class="toggle-line toggle-line-middle"></span>
      <span class="toggle-line toggle-line-last"></span>
    </div>
  </div>

  <div class="site-meta">

    <a href="/blog/" class="brand" rel="start">
      <span class="logo-line-before"><i></i></span>
      <h1 class="site-title">Banner's Blog</h1>
      <span class="logo-line-after"><i></i></span>
    </a>
      <p class="site-subtitle" itemprop="description">Dream big work hard</p>
  </div>

  <div class="site-nav-right">
    <div class="toggle popup-trigger">
        <i class="fa fa-search fa-fw fa-lg"></i>
    </div>
  </div>
</div>




<nav class="site-nav">
  <ul id="menu" class="main-menu menu">
        <li class="menu-item menu-item-home">

    <a href="/blog/" rel="section"><i class="home fa-fw"></i>Home</a>

  </li>
        <li class="menu-item menu-item-categories">

    <a href="/blog/categories/" rel="section"><i class="th fa-fw"></i>Categories</a>

  </li>
        <li class="menu-item menu-item-tags">

    <a href="/blog/tags/" rel="section"><i class="tags fa-fw"></i>Tags</a>

  </li>
        <li class="menu-item menu-item-about">

    <a href="https://wangc1106.github.io/" rel="section"><i class="user fa-fw"></i>About</a>

  </li>
      <li class="menu-item menu-item-search">
        <a role="button" class="popup-trigger"><i class="fa fa-search fa-fw"></i>Search
        </a>
      </li>
  </ul>
</nav>



  <div class="search-pop-overlay">
    <div class="popup search-popup">
        <div class="search-header">
  <span class="search-icon">
    <i class="fa fa-search"></i>
  </span>
  <div class="search-input-container">
    <input autocomplete="off" autocapitalize="off"
           placeholder="Searching..." spellcheck="false"
           type="search" class="search-input">
  </div>
  <span class="popup-btn-close">
    <i class="fa fa-times-circle"></i>
  </span>
</div>
<div id="search-result">
  <div id="no-result">
    <i class="fa fa-spinner fa-pulse fa-5x fa-fw"></i>
  </div>
</div>

    </div>
  </div>

</div>
    </header>

    


    <main class="main">
      <div class="main-inner">
        <div class="content-wrap">
          

          <div class="content post posts-expand">
            

    
  
  
  <article itemscope itemtype="http://schema.org/Article" class="post-block" lang="zh-Hans">
    <link itemprop="mainEntityOfPage" href="https://wangc1106.github.io/blog/2022/02/24/2022-02-24-Seqkit/">

    <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
      <meta itemprop="image" content="/blog/images/banner.png">
      <meta itemprop="name" content="Wang C">
      <meta itemprop="description" content="Make more time">
    </span>

    <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
      <meta itemprop="name" content="Banner's Blog">
    </span>
      <header class="post-header">
        <h1 class="post-title" itemprop="name headline">
          seqkit
        </h1>

        <div class="post-meta">
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-calendar"></i>
              </span>
              <span class="post-meta-item-text">Posted on</span>
              

              <time title="Created: 2022-02-24 16:21:15 / Modified: 21:38:53" itemprop="dateCreated datePublished" datetime="2022-02-24T16:21:15+08:00">2022-02-24</time>
            </span>
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-folder"></i>
              </span>
              <span class="post-meta-item-text">In</span>
                <span itemprop="about" itemscope itemtype="http://schema.org/Thing">
                  <a href="/blog/categories/software/" itemprop="url" rel="index"><span itemprop="name">software</span></a>
                </span>
            </span>

          
            <span id="/blog/2022/02/24/2022-02-24-Seqkit/" class="post-meta-item leancloud_visitors" data-flag-title="seqkit" title="Views">
              <span class="post-meta-item-icon">
                <i class="fa fa-eye"></i>
              </span>
              <span class="post-meta-item-text">Views: </span>
              <span class="leancloud-visitors-count"></span>
            </span>
            <span class="post-meta-item" title="Symbols count in article">
              <span class="post-meta-item-icon">
                <i class="far fa-file-word"></i>
              </span>
              <span>16k</span>
            </span>
            <span class="post-meta-item" title="Reading time">
              <span class="post-meta-item-icon">
                <i class="far fa-clock"></i>
              </span>
              <span>14 mins.</span>
            </span>

        </div>
      </header>

    
    
    
    <div class="post-body" itemprop="articleBody">

      
        <h1 id="seqkit-a-cross-platform-and-ultrafast-toolkit-for-fastaq-file-manipulation"><a class="markdownIt-Anchor" href="#seqkit-a-cross-platform-and-ultrafast-toolkit-for-fastaq-file-manipulation"></a> SeqKit - a cross-platform and ultrafast toolkit for FASTA/Q file manipulation</h1>
<span id="more"></span>
<p><a target="_blank" rel="noopener" href="https://bioinf.shenwei.me/seqkit/usage/#fx2tab-tab2fx">Official doc</a></p>
<h1 id="quick-guide"><a class="markdownIt-Anchor" href="#quick-guide"></a> Quick Guide</h1>
<ul>
<li>Basic: <a target="_blank" rel="noopener" href="https://bioinf.shenwei.me/seqkit/usage/#seq">seq</a>, <a target="_blank" rel="noopener" href="https://bioinf.shenwei.me/seqkit/usage/#stats">stats</a>, <a target="_blank" rel="noopener" href="https://bioinf.shenwei.me/seqkit/usage/#subseq">subseq</a>, <a target="_blank" rel="noopener" href="https://bioinf.shenwei.me/seqkit/usage/#sliding">sliding</a>, <a target="_blank" rel="noopener" href="https://bioinf.shenwei.me/seqkit/usage/#faidx">faidx</a>, <a target="_blank" rel="noopener" href="https://bioinf.shenwei.me/seqkit/usage/#watch">watch</a>, <a target="_blank" rel="noopener" href="https://bioinf.shenwei.me/seqkit/usage/#sana">sana</a>, <a target="_blank" rel="noopener" href="https://bioinf.shenwei.me/seqkit/usage/#scat">scat</a></li>
<li>Format conversion: <a target="_blank" rel="noopener" href="https://bioinf.shenwei.me/seqkit/usage/#fq2fa">fq2fa</a>, <a target="_blank" rel="noopener" href="https://bioinf.shenwei.me/seqkit/usage/#fx2tab-tab2fx">fx2tab</a>, <a target="_blank" rel="noopener" href="https://bioinf.shenwei.me/seqkit/usage/#fx2tab-tab2fx">tab2fx</a>, <a target="_blank" rel="noopener" href="https://bioinf.shenwei.me/seqkit/usage/#convert">convert</a>, <a target="_blank" rel="noopener" href="https://bioinf.shenwei.me/seqkit/usage/#translate">translate</a></li>
<li>Searching: <a target="_blank" rel="noopener" href="https://bioinf.shenwei.me/seqkit/usage/#grep">grep</a>, <a target="_blank" rel="noopener" href="https://bioinf.shenwei.me/seqkit/usage/#locate">locate</a>, <a target="_blank" rel="noopener" href="https://bioinf.shenwei.me/seqkit/usage/#amplicon">amplicon</a>, <a target="_blank" rel="noopener" href="https://bioinf.shenwei.me/seqkit/usage/#fish">fish</a></li>
<li>Set operation: <a target="_blank" rel="noopener" href="https://bioinf.shenwei.me/seqkit/usage/#sample">sample</a>, <a target="_blank" rel="noopener" href="https://bioinf.shenwei.me/seqkit/usage/#rmdup">rmdup</a>, <a target="_blank" rel="noopener" href="https://bioinf.shenwei.me/seqkit/usage/#common">common</a>, <a target="_blank" rel="noopener" href="https://bioinf.shenwei.me/seqkit/usage/#duplicate">duplicate</a>, <a target="_blank" rel="noopener" href="https://bioinf.shenwei.me/seqkit/usage/#split">split</a>, <a target="_blank" rel="noopener" href="https://bioinf.shenwei.me/seqkit/usage/#split2">split2</a>, <a target="_blank" rel="noopener" href="https://bioinf.shenwei.me/seqkit/usage/#head">head</a>, <a target="_blank" rel="noopener" href="https://bioinf.shenwei.me/seqkit/usage/#head-genome">head-genome</a>, <a target="_blank" rel="noopener" href="https://bioinf.shenwei.me/seqkit/usage/#range">range</a>, <a target="_blank" rel="noopener" href="https://bioinf.shenwei.me/seqkit/usage/#pair">pair</a></li>
<li>Edit: <a target="_blank" rel="noopener" href="https://bioinf.shenwei.me/seqkit/usage/#concat">concat</a>, <a target="_blank" rel="noopener" href="https://bioinf.shenwei.me/seqkit/usage/#replace">replace</a>, <a target="_blank" rel="noopener" href="https://bioinf.shenwei.me/seqkit/usage/#restart">restart</a>, <a target="_blank" rel="noopener" href="https://bioinf.shenwei.me/seqkit/usage/#mutate">mutate</a>, <a target="_blank" rel="noopener" href="https://bioinf.shenwei.me/seqkit/usage/#rename">rename</a></li>
<li>Ordering: <a target="_blank" rel="noopener" href="https://bioinf.shenwei.me/seqkit/usage/#sort">sort</a>, <a target="_blank" rel="noopener" href="https://bioinf.shenwei.me/seqkit/usage/#shuffle">shuffle</a></li>
<li>BAM processing: <a target="_blank" rel="noopener" href="https://bioinf.shenwei.me/seqkit/usage/#bam">bam</a></li>
</ul>
<figure class="highlight sh"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">conda install -c bioconda seqkit</span><br></pre></td></tr></table></figure>
<h1 id="command"><a class="markdownIt-Anchor" href="#command"></a> command</h1>
<figure class="highlight sh"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br></pre></td><td class="code"><pre><span class="line">amplicon        通过引物检索扩增子(或其周围的特定区域)</span><br><span class="line">bam             检查和在线绘制BAM记录文件的直方图</span><br><span class="line">common          通过id/名称/序列查找多个文件的公共序列</span><br><span class="line">concat          连接多个文件中具有相同ID的序列</span><br><span class="line">convert         转换FASTQ质量编码格式：支持格式包括：桑格，Solexa和Illumina</span><br><span class="line">duplicate       重复序列N次</span><br><span class="line">faidx           创建FASTA索引文件并提取子序列</span><br><span class="line">fish            使用局部比对在较大的序列中寻找短序列</span><br><span class="line">fq2fa           转换FASTQ到FASTA</span><br><span class="line">fx2tab          将FASTA/Q转换为表格格式(包含长度/GC含量/GC偏好)</span><br><span class="line">genautocomplete 生成shell自动完成脚本</span><br><span class="line">grep            通过ID/name/sequence/sequence motif搜索序列，允许错配</span><br><span class="line">head            打印第一条序列</span><br><span class="line"><span class="built_in">help</span>            打印帮助信息</span><br><span class="line">locate          定位序列，或者motifs，允许错配</span><br><span class="line">mutate          编辑序列(点突变、插入、删除)</span><br><span class="line">pair            匹配双端序列文件</span><br><span class="line">range           打印一个范围内的序列</span><br><span class="line">rename          重命名重复序列ID</span><br><span class="line">replace         使用正则表达式修改名称或者序列</span><br><span class="line">restart         重置环状基因组的起始位置</span><br><span class="line">rmdup           通过id/名称/序列删除重复的序列</span><br><span class="line">sample          按数量或比例对序列进行抽样</span><br><span class="line">sana            清理损坏的单行fastq文件</span><br><span class="line">scat            real time recursive concatenation and streaming of fastx files</span><br><span class="line">seq             转换序列(反向，补充，提取ID…)</span><br><span class="line">shuffle         随机序列</span><br><span class="line">sliding         序列滑窗提取，支持环形基因组</span><br><span class="line">sort            按id/名称/序列/长度排序序列</span><br><span class="line">split           按id/seq区域/大小/部件将序列拆分为文件(主要用于FASTA)</span><br><span class="line">split2          按序列数量/文件数将序列拆分为多个文件(FASTA, PE/SE FASTQ)</span><br><span class="line">stats           FASTA/Q文件的简单统计</span><br><span class="line">subseq          通过region/gtf/bed得到子序列，包括侧翼序列</span><br><span class="line">tab2fx          转换表格格式为FASTA/Q格式</span><br><span class="line">translate       翻译DNA/RNA到蛋白质序列(支持歧义碱基)</span><br><span class="line">version         打印版本信息并检查是否更新</span><br><span class="line">watch           序列特征的监测和在线直方图</span><br><span class="line"></span><br><span class="line"></span><br></pre></td></tr></table></figure>
<h1 id="parameter"><a class="markdownIt-Anchor" href="#parameter"></a> Parameter:</h1>
<figure class="highlight sh"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br></pre></td><td class="code"><pre><span class="line">Flags:</span><br><span class="line">      --alphabet-guess-seq-length int   seqkit根据第一个FASTA记录猜测序列类型的序列前缀的长度(0表示整个序列)(默认10000)</span><br><span class="line">  -h, --<span class="built_in">help</span>                            显示帮助</span><br><span class="line">      --id-ncbi                         FASTA头是ncbi风格的，例如&gt;gi|110645304|ref|NC_002516.2 </span><br><span class="line">      --id-regexp string                用于解析ID的正则表达式(default <span class="string">&quot;^(\\S+)\\s?&quot;</span>)，匹配空格前的部分为序列名</span><br><span class="line">      --infile-list string              输入文件列表中的文件 (one file per line), <span class="keyword">if</span> given, they are appended to files from cli arguments</span><br><span class="line">  -w, --line-width int                  输出FASTA格式时的行宽 (0 <span class="keyword">for</span> no wrap) (default 60)</span><br><span class="line">  -o, --out-file string                 输出 (<span class="string">&quot;-&quot;</span> <span class="keyword">for</span> stdout, suffix .gz <span class="keyword">for</span> gzipped out) (default <span class="string">&quot;-&quot;</span>) -代表标准输出，加.gz可输出压缩文件</span><br><span class="line">      --quiet                           保持安静，不要显示额外的信息</span><br><span class="line">  -t, --seq-type string                 序列类型 (dna|rna|protein|<span class="built_in">unlimit</span>|auto) (auto, 按第一个序列自动检测) (default <span class="string">&quot;auto&quot;</span>)</span><br><span class="line">  -j, --threads int                     CPU数量 (默认单核为1，多核为2) (default 2)</span><br><span class="line"></span><br><span class="line"></span><br></pre></td></tr></table></figure>
<h1 id="stats-fastaq文件的简单统计"><a class="markdownIt-Anchor" href="#stats-fastaq文件的简单统计"></a> stats FASTA/Q文件的简单统计</h1>
<p>统计序列格式fa/fq、内容类型DNA/RNA/Protein，<font color="red">序列数量、总长度，最小、平均和最大长度</font></p>
<figure class="highlight sh"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># FASTA DNA</span></span><br><span class="line"><span class="built_in">echo</span> -e <span class="string">&quot;&gt;seq\nacgtryswkmbdhvACGTRYSWKMBDHV&quot;</span> | seqkit stats</span><br><span class="line">file  format  <span class="built_in">type</span>  num_seqs  sum_len  min_len  avg_len  max_len</span><br><span class="line">-     FASTA   DNA          1       28       28       28       28</span><br><span class="line"></span><br><span class="line"><span class="comment"># RNA </span></span><br><span class="line"><span class="built_in">echo</span> -e <span class="string">&quot;&gt;seq\nACGUN\nACGUN&quot;</span> | seqkit stats</span><br><span class="line">file  format  <span class="built_in">type</span>  num_seqs  sum_len  min_len  avg_len  max_len</span><br><span class="line">-     FASTA   RNA          1       10       10       10       10</span><br><span class="line"></span><br><span class="line"><span class="comment"># Protein</span></span><br><span class="line"><span class="built_in">echo</span> -e <span class="string">&quot;&gt;seq\nabcdefghijklmnpqrstvwyz&quot;</span> | seqkit stats</span><br><span class="line">file  format  <span class="built_in">type</span>     num_seqs  sum_len  min_len  avg_len  max_len</span><br><span class="line">-     FASTA   Protein         1       23       23       23       23</span><br><span class="line"></span><br><span class="line"><span class="comment"># FASTQ DNA</span></span><br><span class="line"><span class="built_in">echo</span> -e <span class="string">&quot;@read\nACTGCN\n+\n@IICCG&quot;</span> | seqkit stats</span><br><span class="line">file  format  <span class="built_in">type</span>  num_seqs  sum_len  min_len  avg_len  max_len</span><br><span class="line">-     FASTQ   DNA          1        6        6        6        6</span><br><span class="line"></span><br><span class="line">file                             format  <span class="built_in">type</span>     num_seqs         sum_len  min_len  avg_len  max_len</span><br><span class="line">R1.fastq.gz  FASTQ   DNA   254,677,261  34,439,769,981       50    135.2      138</span><br><span class="line"></span><br></pre></td></tr></table></figure>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br></pre></td><td class="code"><pre><span class="line"># 一般模式</span><br><span class="line">seqkit stats C1_1.fq.gz</span><br><span class="line">#--输出结果tab分隔</span><br><span class="line">seqkit stats C1_1.fq.gz -T</span><br><span class="line">#--输出文件转化其他格式</span><br><span class="line">seqkit stats C1_1.fq.gz -T| csvtk pretty -t</span><br><span class="line">seqkit stats C1_1.fq.gz -T| csvtk csv2md -t</span><br><span class="line"># 统计更多信息 -a</span><br><span class="line">seqkit stats C1_1.fq.gz -a</span><br><span class="line"># j多线程加速，尤其是对于具有多个序列文件会加速</span><br><span class="line"># seqkit stats -j 2 *.fq.gz</span><br><span class="line"></span><br><span class="line"></span><br></pre></td></tr></table></figure>
<h1 id="seq-转换序列-反向-互补提取id"><a class="markdownIt-Anchor" href="#seq-转换序列-反向-互补提取id"></a> seq 转换序列 （反向、互补/提取ID）</h1>
<p>“-n”: 提取序列ID，包括“&gt;”后面的全部内容</p>
<p>“-n -i”: 仅提取第一个空格前的ID</p>
<h1 id="按长度过滤常用"><a class="markdownIt-Anchor" href="#按长度过滤常用"></a> 按长度过滤(常用)</h1>
<p>扩增子分析时要筛选扩增长度相近的片段，过长或过短一般都要删除。宏基因组中比如组装的结果，经常要过滤&lt;200/300bp的短片段，分箱时要筛选&gt;1000/2000的长片段使用。本条命令非常多的应用场景。筛选后结果可用 &gt; 写入文件</p>
<p>-m 按照序列长度过滤，表示保留的最小长度，-M 此为保留的最大长度</p>
<figure class="highlight sh"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#--提取序列长度大于60的并统计长度信息</span></span><br><span class="line">zcat hairpin.fa.gz | seqkit seq -m 60 | seqkit stats</span><br><span class="line"><span class="comment"># 设置最小序列长度和最大序列长度，用于过滤序列，并统计</span></span><br><span class="line">zcat hairpin.fa.gz | seqkit seq -m 100 -M 1000 | seqkit stats</span><br><span class="line"><span class="comment"># 保存&gt;100且&lt;1000长度的序列</span></span><br><span class="line">seqkit seq -m 100 -M 1000 hairpin.fa.gz &gt; hairpin100-1000.fa</span><br><span class="line">seqkit <span class="built_in">stat</span> hairpin100-1000.fa</span><br><span class="line"></span><br></pre></td></tr></table></figure>
<h1 id="提取id"><a class="markdownIt-Anchor" href="#提取id"></a> 提取ID</h1>
<figure class="highlight sh"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br></pre></td><td class="code"><pre><span class="line">head gene.fa</span><br><span class="line"><span class="comment">## 名称全行</span></span><br><span class="line">seqkit seq gene.fa -n | head</span><br><span class="line"><span class="comment"># 仅仅打印ID</span></span><br><span class="line">seqkit seq gene.fa -n -i | head</span><br><span class="line"> </span><br><span class="line"><span class="comment"># 使用正则表达式提取名字中的信息</span></span><br><span class="line">zcat hairpin.fa.gz | head</span><br><span class="line"><span class="comment"># 提取ID中第二个字段作为ID</span></span><br><span class="line">seqkit seq hairpin.fa.gz -n -i --id-regexp <span class="string">&quot;^[^\s]+\s([^\s]+)\s&quot;</span> | head</span><br><span class="line"></span><br><span class="line"></span><br></pre></td></tr></table></figure>
<h1 id="单行多行转换"><a class="markdownIt-Anchor" href="#单行多行转换"></a> 单行/多行转换</h1>
<ul>
<li>-s提取并展示序列</li>
<li>-w 代表每行的碱基数量，0代表不换行</li>
</ul>
<figure class="highlight sh"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#仅仅提取序列 -s</span></span><br><span class="line">seqkit seq gene.fa -s -w 0|head</span><br><span class="line"><span class="comment">#--将多行序列转化为标准4行FASTQ</span></span><br><span class="line">seqkit seq C1_1.fq.gz -w 0|head</span><br></pre></td></tr></table></figure>
<h1 id="反向互补"><a class="markdownIt-Anchor" href="#反向互补"></a> 反向/互补</h1>
<ul>
<li>-r 序列反向</li>
<li>-p序列互补</li>
</ul>
<figure class="highlight sh"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># 序列反向互补,-r反向，-p互补</span></span><br><span class="line">seqkit seq hairpin.fa.gz -r -p|head</span><br></pre></td></tr></table></figure>
<h1 id="删除gap大小写转换"><a class="markdownIt-Anchor" href="#删除gap大小写转换"></a> 删除gap/大小写转换</h1>
<ul>
<li>-g 去除序列中的间隔,将中间的横杠去掉</li>
<li>-u转化序列为大写字母展示</li>
</ul>
<figure class="highlight sh"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line"><span class="built_in">echo</span> -e <span class="string">&quot;&gt;seq\nACGT-ACTGC-acc&quot;</span> | seqkit seq -g -u</span><br></pre></td></tr></table></figure>
<h1 id="rna转为dna"><a class="markdownIt-Anchor" href="#rna转为dna"></a> RNA转为DNA</h1>
<ul>
<li>—rna2dna 将RNA序列转化为DNA序列</li>
</ul>
<figure class="highlight sh"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line"><span class="built_in">echo</span> -e <span class="string">&quot;&gt;seq\nUCAUAUGCUUGUCUCAAAGAUUA&quot;</span> | seqkit seq --rna2dna</span><br></pre></td></tr></table></figure>
<h1 id="subseq通过指定区域"><a class="markdownIt-Anchor" href="#subseq通过指定区域"></a> subseq通过指定区域</h1>
<ul>
<li>-r 通过区域来截取序列</li>
</ul>
<p>如1:12提取前12个碱基，-12:-1提取序列结尾12个碱基；<br>
for last 12 bases, 13:-1 for cutting first 12 bases. type “seqkit subseq -h” for more examples</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line">#-提取序列前1：12个碱基</span><br><span class="line">zcat C1_1.fq.gz | seqkit subseq -r 1:12 |head</span><br><span class="line">#-提取序列最后1：12个碱基</span><br><span class="line">zcat C1_1.fq.gz | seqkit subseq -r -12:-1 |head</span><br><span class="line"> </span><br><span class="line">#取第12至倒数第12个碱基，即前11和后11个碱基去掉</span><br><span class="line">zcat C1_1.fq.gz | seqkit subseq -r 12:-12| head</span><br></pre></td></tr></table></figure>
<p>基于gtf/bed信息挑选子序列。</p>
<p>—gtf  根据gtf文件挑选基因，这部分功能用于根据基因注释快速提取基因序列，在宏基因组、转录组、重测序中常用。—chr 选择染色体，—feature cds选择序列类型</p>
<p>以拟南芥基因组的序列和注释数据演示：提取第一条染色体上的CDS基因信息，并统计基本信息</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line">seqkit subseq --gtf Arabidopsis_thaliana.TAIR10.49.gtf.gz --chr 1 --feature cds  Arabidopsis_thaliana.TAIR10.dna.toplevel.fa.gz  &gt; chr1.gtf.cds.fa</span><br><span class="line">seqkit stats chr1.gtf.cds.fa</span><br><span class="line"></span><br><span class="line"></span><br></pre></td></tr></table></figure>
<ul>
<li>-u 可以提取目标基因上游的序列</li>
<li>-f 目标区域不展示</li>
</ul>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line">#--挑选序列并多加上上游的3个碱基</span><br><span class="line">seqkit subseq --gtf Arabidopsis_thaliana.TAIR10.49.gtf.gz Arabidopsis_thaliana.TAIR10.dna.toplevel.fa.gz -u 3 |head</span><br><span class="line"> </span><br><span class="line"># 仅提取上游序列，如提取启动子区2k：-f仅定位不输出位置序列，-u输出上游序列，此处示例3bp</span><br><span class="line">seqkit subseq --gtf Arabidopsis_thaliana.TAIR10.49.gtf.gz Arabidopsis_thaliana.TAIR10.dna.toplevel.fa.gz -u 3 -f |head</span><br><span class="line"></span><br><span class="line"></span><br></pre></td></tr></table></figure>
<h1 id="sliding-滑窗提取序列支持环状基因组"><a class="markdownIt-Anchor" href="#sliding-滑窗提取序列支持环状基因组"></a> sliding 滑窗提取序列，支持环状基因组</h1>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br></pre></td><td class="code"><pre><span class="line">#-s 步长为3，-W 序列长度为6个碱基</span><br><span class="line">echo -e &quot;&gt;seq\nACGTacgtNN&quot; | seqkit sliding -s 3 -W 6</span><br><span class="line"># -g 贪婪模式，后面不足6个那也取</span><br><span class="line">echo -e &quot;&gt;seq\nACGTacgtNN&quot; | seqkit sliding -s 3 -W 6 -g</span><br><span class="line"># 环状DNA模式-C，首尾不算中断，环状</span><br><span class="line">echo -e &quot;&gt;seq\nACGTacgtNN&quot; | seqkit sliding -s 3 -W 6 -C</span><br><span class="line"></span><br><span class="line"></span><br></pre></td></tr></table></figure>
<ul>
<li>步长为5，取30个碱基序列，然后统计GC含量</li>
</ul>
<ul>
<li>fx2tab：统计fasta/fastq序列的信息为表格</li>
<li>-n仅输出ID，不输出序列</li>
<li>-g为GC含量</li>
</ul>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">zcat hairpin.fa.gz | seqkit sliding -s 5 -W 30 | seqkit fx2tab -n -g |head</span><br></pre></td></tr></table></figure>
<h1 id="faidx-创建fasta索引文件并提取子序列"><a class="markdownIt-Anchor" href="#faidx-创建fasta索引文件并提取子序列"></a> faidx 创建FASTA索引文件并提取子序列</h1>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br></pre></td><td class="code"><pre><span class="line"># 解压</span><br><span class="line">zcat hairpin.fa.gz &gt; hairpin.fa</span><br><span class="line"> </span><br><span class="line"># 建索引*.fai文件</span><br><span class="line">seqkit faidx hairpin.fa</span><br><span class="line"> </span><br><span class="line"># ID信息：hsa-let-7a-1 hsa-let-7a-2</span><br><span class="line">seqkit faidx hairpin.fa hsa-let-7a-1 hsa-let-7a-2</span><br><span class="line"># -f 标题行全部显示</span><br><span class="line">seqkit faidx hairpin.fa hsa-let-7a-1 hsa-let-7a-2 -f</span><br><span class="line"># 提取序列并选择区域显示</span><br><span class="line">seqkit faidx hairpin.fa hsa-let-7a-1:1-10</span><br><span class="line">seqkit faidx hairpin.fa hsa-let-7a-1:-10--1</span><br><span class="line">seqkit faidx hairpin.fa hsa-let-7a-1:1</span><br><span class="line"># 检查hsa开头的序列并统计</span><br><span class="line">seqkit faidx hairpin.fa hsa -r | seqkit stats</span><br><span class="line"></span><br><span class="line"></span><br></pre></td></tr></table></figure>
<h1 id="watch-序列质量的监测和在线直方图"><a class="markdownIt-Anchor" href="#watch-序列质量的监测和在线直方图"></a> watch 序列质量的监测和在线直方图</h1>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line">#-取对数展示直方图</span><br><span class="line">seqkit watch -L -f ReadLen hairpin.fa</span><br><span class="line"> </span><br><span class="line"># 每五千个做一个图保存在pdf文件中</span><br><span class="line">seqkit watch -p 500000 -O qhist.pdf -f MeanQual C1_1.fq.gz</span><br></pre></td></tr></table></figure>
<p>从有错误记录的fastq文件中挽救可用的读取</p>
<h1 id="sana清理损坏fastq文件"><a class="markdownIt-Anchor" href="#sana清理损坏fastq文件"></a> sana：清理损坏fastq文件</h1>
<p>这里我专门将C1_1.fq的第一个序列进行了错位，进行测试。这个操作往往在进行数据整合的时候可以有很大作用。</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">zcat C1_1.fq.gz|sed &#x27;2 s/^/A/&#x27; &gt; C1_1_bad.fq</span><br><span class="line">seqkit sana C1_1_bad.fq -o rescued.fq.gz</span><br></pre></td></tr></table></figure>
<h1 id="fq2fa-将fq转为fa格式"><a class="markdownIt-Anchor" href="#fq2fa-将fq转为fa格式"></a> fq2fa 将fq转为fa格式</h1>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">seqkit fq2fa C1_1.fq.gz -o C1_1.fa</span><br></pre></td></tr></table></figure>
<h1 id="fx2tab-tab2fx-序列转化表格格式"><a class="markdownIt-Anchor" href="#fx2tab-tab2fx-序列转化表格格式"></a> fx2tab &amp; tab2fx 序列转化表格格式</h1>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br></pre></td><td class="code"><pre><span class="line">这一转化很有用，往往用于表格/矩阵处理的时候。</span><br><span class="line">seqkit fx2tab hairpin.fa.gz | head -n 2</span><br><span class="line">通过矩阵格式的序列文件统计序列长度和质量值</span><br><span class="line"></span><br><span class="line">-l 统计序列长度</span><br><span class="line"></span><br><span class="line">-g 统计平均GC含量</span><br><span class="line"></span><br><span class="line">-i 只打印名称(不打印序列)</span><br><span class="line"></span><br><span class="line">-H 打印标题行</span><br><span class="line"># 打印序列长度、GC含量</span><br><span class="line">seqkit fx2tab hairpin.fa.gz -l -g -n -i -H | head</span><br><span class="line"></span><br><span class="line"></span><br></pre></td></tr></table></figure>
<p>tab2fx 和表格格式转化为序列格式</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br></pre></td><td class="code"><pre><span class="line"># seqkit tab2fx 表格形式转化为序列形式</span><br><span class="line">zcat hairpin.fa.gz | seqkit fx2tab | seqkit tab2fx | head</span><br><span class="line"> </span><br><span class="line"># 转化为表格，然后排序，然后转化回去</span><br><span class="line">zcat hairpin.fa.gz \</span><br><span class="line">    | seqkit fx2tab -l \</span><br><span class="line">    | sort -t &quot;`echo -e &#x27;\t&#x27;`&quot; -n -k4,4 \</span><br><span class="line">    | seqkit tab2fx | head</span><br><span class="line"># 等同于下面的命令</span><br><span class="line">seqkit sort -l hairpin.fa.gz | head</span><br><span class="line"> </span><br><span class="line"># 通过这个转化可以将很多在表格中实现的数据处理方法用于序列</span><br><span class="line">例如下面的提取1000个序列：seqkit fx2tab hairpin.fa.gz | head -n 1000 | seqkit tab2fx | head</span><br></pre></td></tr></table></figure>
<h1 id="translate-翻译dnarna为蛋白质序列"><a class="markdownIt-Anchor" href="#translate-翻译dnarna为蛋白质序列"></a> translate 翻译DNA/RNA为蛋白质序列</h1>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line">#--转化为蛋白序列</span><br><span class="line">seqkit translate gene.fa|head</span><br><span class="line"> </span><br><span class="line"># 去除&#x27;X&#x27; 和 &#x27;*&#x27;</span><br><span class="line">seqkit translate hairpin.fa </span><br><span class="line">seqkit translate hairpin.fa --trim | head</span><br></pre></td></tr></table></figure>
<h1 id="grep-序列匹配"><a class="markdownIt-Anchor" href="#grep-序列匹配"></a> grep  序列匹配</h1>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br></pre></td><td class="code"><pre><span class="line"># 生成一个ID列表</span><br><span class="line">grep &#x27;&gt;&#x27; C1_1.fa|cut -c2-|head -n 10 &gt; id.txt</span><br><span class="line"> </span><br><span class="line"># 使用序列id列表进行搜索(不包含空格)</span><br><span class="line">seqkit grep -f id.txt C1_1.fq.gz -o result.fq.gz</span><br><span class="line"># 使用序列名称列表进行搜索(它们可能包含空格)</span><br><span class="line">seqkit grep -n -f id.txt C1_1.fq.gz -o result.fa.gz</span><br><span class="line">zcat  result.fa.gz</span><br><span class="line"># 提取hsa开头的序列</span><br><span class="line">zcat hairpin.fa.gz | seqkit grep -r -p ^hsa |head</span><br><span class="line"> </span><br><span class="line"># -v参数v用于移除序列</span><br><span class="line">zcat hairpin.fa.gz | seqkit grep -r -p ^hsa -p ^mmu -v | head</span><br><span class="line"> </span><br><span class="line"># 提取ID</span><br><span class="line">zcat miRNA.diff.gz | grep ^# -v | grep NEW | cut -f 2 &gt; list</span><br><span class="line">head list</span><br><span class="line"># 根据ID提取文件</span><br><span class="line">zcat hairpin.fa.gz | seqkit grep -f list &gt; new.fa</span><br><span class="line">head new.fa</span><br><span class="line"> </span><br><span class="line"># 提取包含特定碱基组合的序列</span><br><span class="line">cat hairpin.fa.gz | seqkit grep -s -i -p aggcg |head</span><br><span class="line"># 统计</span><br><span class="line">cat hairpin.fa.gz | seqkit grep -s -i -p aggcg | seqkit stats</span><br><span class="line"> </span><br><span class="line">#  去除 包含特定组合碱基的序列</span><br><span class="line">zcat hairpin.fa.gz | seqkit grep -s -r -i -p ^aggcg |head</span><br><span class="line"></span><br><span class="line"></span><br></pre></td></tr></table></figure>
<h1 id="locate-输出匹配位置"><a class="markdownIt-Anchor" href="#locate-输出匹配位置"></a> locate 输出匹配位置</h1>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line">＃　其他两种输出格式</span><br><span class="line">zcat hairpin.fa.gz | seqkit locate -i -d -p AUGGACUN --bed</span><br><span class="line">zcat hairpin.fa.gz | seqkit locate -i -d -p AUGGACUN --gtf</span><br><span class="line"></span><br><span class="line"></span><br></pre></td></tr></table></figure>
<h1 id="fish-使用局部比对在较大的序列中寻找短序列"><a class="markdownIt-Anchor" href="#fish-使用局部比对在较大的序列中寻找短序列"></a> fish 使用局部比对在较大的序列中寻找短序列</h1>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line">echo -e &#x27;&gt;seq\nACGACGACGA&#x27; \</span><br><span class="line">    | seqkit locate -p ACGA -G | csvtk -t pretty</span><br><span class="line"> </span><br><span class="line">echo -e &#x27;&gt;seq\nACGACGACGA&#x27; \</span><br><span class="line">    | seqkit fish -F ACGA -a 2&gt;&amp;1 | csvtk -t pretty</span><br><span class="line"></span><br><span class="line"></span><br></pre></td></tr></table></figure>
<h1 id="amplicon-通过引物检索扩增子或其周围的特定区域"><a class="markdownIt-Anchor" href="#amplicon-通过引物检索扩增子或其周围的特定区域"></a> amplicon 通过引物检索扩增子(或其周围的特定区域)</h1>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br></pre></td><td class="code"><pre><span class="line">echo -ne &quot;&gt;seq\nacgcccactgaaatga\n&quot; \</span><br><span class="line">    | seqkit amplicon -F ccc -R ttt</span><br><span class="line"> </span><br><span class="line"># 设置输出格式为bed，匹配的位点等信息</span><br><span class="line">echo -ne &quot;&gt;seq\nacgcccactgaaatga\n&quot; \</span><br><span class="line">    | seqkit amplicon -F ccc -R ttt --bed</span><br><span class="line"> </span><br><span class="line">#- 使用引物文件，这用于刘老师组的高通量分菌的序列拆分速度应该很客观</span><br><span class="line"># cat seqs4amplicon.fa | seqkit amplicon -p primers.tsv --bed</span><br><span class="line"></span><br><span class="line"></span><br></pre></td></tr></table></figure>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br></pre></td><td class="code"><pre><span class="line"># 输出除去引物之外的部分，-r输出几个碱基</span><br><span class="line">echo -ne &quot;&gt;seq\nacgcccactgaaatga\n&quot; \</span><br><span class="line">    | seqkit amplicon -F ccc -R ttt -r 4:7</span><br><span class="line"># 输出格式为bed</span><br><span class="line">echo -ne &quot;&gt;seq\nacgcccactgaaatga\n&quot; \</span><br><span class="line">    | seqkit amplicon -F ccc -R ttt -r 4:7 --bed</span><br><span class="line"></span><br><span class="line"></span><br></pre></td></tr></table></figure>
<h1 id="duplicate-对序列重复n次"><a class="markdownIt-Anchor" href="#duplicate-对序列重复n次"></a> duplicate 对序列重复N次</h1>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br></pre></td><td class="code"><pre><span class="line"># 重复序列1次，但是名字没有修改</span><br><span class="line">cat hairpin.fa | seqkit head -n 1 \</span><br><span class="line">    | seqkit duplicate -n 2</span><br><span class="line"># 对重复序列改名，使其独一无二</span><br><span class="line">cat hairpin.fa | seqkit head -n 1 \</span><br><span class="line">    | seqkit duplicate -n 2 | seqkit rename</span><br><span class="line"></span><br><span class="line"></span><br></pre></td></tr></table></figure>
<h1 id="rmdup-通过id名称序列删除重复的序列"><a class="markdownIt-Anchor" href="#rmdup-通过id名称序列删除重复的序列"></a> rmdup 通过id/名称/序列删除重复的序列</h1>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br></pre></td><td class="code"><pre><span class="line"># 去除重复的序列</span><br><span class="line">zcat hairpin.fa.gz | seqkit rmdup -s -o clean.fa.gz</span><br><span class="line"> </span><br><span class="line"># 保存重复序列得到文件 -D duplicated.detail.txt</span><br><span class="line">zcat hairpin.fa.gz \</span><br><span class="line">    | seqkit rmdup -s -i -o clean.fa.gz -d duplicated.fa.gz -D duplicated.detail.txt</span><br><span class="line"></span><br><span class="line"></span><br></pre></td></tr></table></figure>
<h1 id="common-通过id名称序列查找多个文件的公共序列"><a class="markdownIt-Anchor" href="#common-通过id名称序列查找多个文件的公共序列"></a> common ：通过id/名称/序列查找多个文件的公共序列</h1>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br></pre></td><td class="code"><pre><span class="line"># 通过ID匹配，文件夹下全部的fa序列公共部分输出来</span><br><span class="line">seqkit common *.fq.gz</span><br><span class="line"> </span><br><span class="line"># 通过-n实现全部名字匹配，-o输出结果</span><br><span class="line">seqkit common *.fq.gz -n -o common.fq </span><br><span class="line"># 通过-s序列匹配</span><br><span class="line">seqkit common *.fq.gz -s -i -o common.fq</span><br><span class="line"></span><br><span class="line"></span><br></pre></td></tr></table></figure>
<h1 id="split-拆分序列为子文件"><a class="markdownIt-Anchor" href="#split-拆分序列为子文件"></a> split 拆分序列为子文件</h1>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br></pre></td><td class="code"><pre><span class="line">按名称ID、给定区域的子序列、文件大小或序列数量将序列拆分为文件</span><br><span class="line"></span><br><span class="line">可用于将大文件拆分后，并行处理，加速分析。如从contig中预测基因。</span><br><span class="line">#按照10000个序列为一个文件拆分，结果为hairpin.fa.gz.split/目录 ，文件名为hairpin.part_00x.fasta，-s</span><br><span class="line">seqkit split hairpin.fa.gz -s 10000 -2</span><br><span class="line"> </span><br><span class="line"># 将序列拆分为四个部分(常用，等分然后并行)</span><br><span class="line">seqkit split hairpin.fa.gz -p 5 -2</span><br><span class="line"> </span><br><span class="line"># 复杂一点的就是按照ID区分</span><br><span class="line">seqkit split hairpin.fa.gz -i --id-regexp &quot;^([\w]+)\-&quot; -2</span><br><span class="line"> </span><br><span class="line"># 按照前三个序列碱基来区分</span><br><span class="line">seqkit split hairpin.fa.gz -r 1:3 -2</span><br></pre></td></tr></table></figure>
<h1 id="split2-拆分文件-升级版本"><a class="markdownIt-Anchor" href="#split2-拆分文件-升级版本"></a> split2 拆分文件 升级版本</h1>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br></pre></td><td class="code"><pre><span class="line">Flags:</span><br><span class="line">  -l, --by-length string   split sequences into chunks of &gt;=N bases, supports K/M/G suffix</span><br><span class="line">  -p, --by-part int        按照拆分出来的数量，比如：拆分成两个子文件2。-s, --by-size int        按照序列数量拆分</span><br><span class="line">  -f, --force              强制覆盖文件</span><br><span class="line">  -h, --help               查看帮助文件</span><br><span class="line">  -O, --out-dir string     输出文件夹 (default value is $infile.split)</span><br><span class="line">  -1, --read1 string       (gzipped) 双端序列第一个</span><br><span class="line">  -2, --read2 string       (gzipped) 双端序列第二个</span><br><span class="line"></span><br><span class="line"></span><br></pre></td></tr></table></figure>
<p>同时支持fa和fq文件。单端和双端序列拆分实例</p>
<p>-f强制覆盖结果，适合重复计算时使用</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line">seqkit split2 hairpin.fa.gz -s 10000 -f</span><br><span class="line"> </span><br><span class="line"># 双端序列拆分(重点)，p指定拆分数量，-O指定输出目录，-f覆盖结果，默认为压缩</span><br><span class="line">seqkit split2 -1 C1_1.fq.gz -2 C1_2.fq.gz -p 2 -O out -f</span><br><span class="line"></span><br><span class="line"></span><br></pre></td></tr></table></figure>
<h1 id="pair-拼接两个fastq文件"><a class="markdownIt-Anchor" href="#pair-拼接两个fastq文件"></a> pair 拼接两个fastq文件</h1>
<p>留下匹配的，去除不匹配的，这里我们使用扩增子的双端序列做一个演示：</p>
<p>注意：双端序列在两个文件中的顺序最好是一样的，否则会消耗大量内存去匹配。</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line">seqkit pair -1 C1_1.fq.gz -2 C1_2.fq.gz -O result</span><br><span class="line"># -u 输出未匹配上的文件</span><br><span class="line">seqkit pair -1 C1_1.fq.gz -2 C1_2.fq.gz -O result -u -f</span><br><span class="line"></span><br></pre></td></tr></table></figure>
<h1 id="sample-按数量或比例对序列进行抽样"><a class="markdownIt-Anchor" href="#sample-按数量或比例对序列进行抽样"></a> sample 按数量或比例对序列进行抽样。</h1>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line">按照百分比例和序列数量进行抽样</span><br><span class="line"># 抽样百分之十</span><br><span class="line">zcat C1_1.fq.gz | seqkit sample -p 0.1 -o sample.fq.gz</span><br><span class="line"># 抽样1000条</span><br><span class="line">zcat C1_1.fq.gz | seqkit sample -n 1000 -o sample.fq.gz</span><br><span class="line"></span><br><span class="line">注意：1000条并不是很准确，可能是900多条，为什么呢？看这里了解问题。https://bioinf.shenwei.me/seqkit/note/#effect-of-random-seed-on-results-of-seqkit-sample</span><br></pre></td></tr></table></figure>
<p>这里为大家展示一下减少内存的序列抽样方法</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br></pre></td><td class="code"><pre><span class="line"># 抽样 seqkit sample </span><br><span class="line">zcat hairpin.fa.gz \</span><br><span class="line">    | seqkit sample -p 0.1 \</span><br><span class="line">    | seqkit head -n 1000 -o sample.fa.gz</span><br><span class="line"> </span><br><span class="line"># 设置随机种子，方便重复结果: -s 11</span><br><span class="line">zcat hairpin.fa.gz \</span><br><span class="line">    | seqkit sample -p 0.1 -s 11 |head</span><br><span class="line"> </span><br><span class="line"># 抽样后打乱序列 :seqkit shuffle</span><br><span class="line">zcat hairpin.fa.gz \</span><br><span class="line">    | seqkit sample -p 0.1 \</span><br><span class="line">    | seqkit shuffle -o sample.fa.gz</span><br><span class="line"></span><br><span class="line"></span><br></pre></td></tr></table></figure>
<h1 id="range-打印序列-按照一个范围"><a class="markdownIt-Anchor" href="#range-打印序列-按照一个范围"></a> range 打印序列 按照一个范围</h1>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br></pre></td><td class="code"><pre><span class="line"># 打印一个范围内的序列</span><br><span class="line">cat hairpin.fa | seqkit range -r 1:10</span><br><span class="line"># 打印最后几行的序列</span><br><span class="line">cat hairpin.fa | seqkit range -r -100:-1 | seqkit stats</span><br><span class="line"># 打印中间范围的序列</span><br><span class="line">cat hairpin.fa | seqkit range -r 101:150 | seqkit stats</span><br><span class="line"></span><br><span class="line"></span><br></pre></td></tr></table></figure>
<h1 id="repeat-使用正则表达式替换名称序列"><a class="markdownIt-Anchor" href="#repeat-使用正则表达式替换名称序列"></a> repeat 使用正则表达式替换名称/序列。</h1>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br></pre></td><td class="code"><pre><span class="line"># 修改序列名称：删除空格后内存</span><br><span class="line">echo -e &quot;&gt;seq1 abc-123\nACGT-ACGT&quot; \</span><br><span class="line">    | seqkit replace -p &quot;\s.+&quot;</span><br><span class="line"> </span><br><span class="line"># 修改序列名：替换</span><br><span class="line">echo -e &quot;&gt;seq1 abc-123\nACGT-ACGT&quot; \</span><br><span class="line">    | seqkit replace -p &quot;\-&quot; -r &#x27;=&#x27;</span><br><span class="line"> </span><br><span class="line"># 修改序列：去除序列间隔</span><br><span class="line">echo -e &quot;&gt;seq1 abc-123\nACGT-ACGT&quot; \</span><br><span class="line">    | seqkit replace -p &quot; |-&quot; -s</span><br><span class="line"> </span><br><span class="line"># 修改序列：给每一个碱基加上空格</span><br><span class="line">echo -e &quot;&gt;seq1 abc-123\nACGT-ACGT&quot; \</span><br><span class="line">    | seqkit replace -p &quot;(.)&quot; -r &#x27;$1 &#x27; -s</span><br><span class="line"> </span><br><span class="line"># 使用字符加数据重命名序列-用于扩增子代表序列改名非常优秀</span><br><span class="line">echo -e &quot;&gt;abc\nACTG\n&gt;123\nATTT&quot; \</span><br><span class="line">    |  seqkit replace -p .+ -r &quot;ASV_&#123;nr&#125;&quot;</span><br><span class="line"> </span><br><span class="line">echo -e &quot;&gt;abc\nACTG\n&gt;123\nATTT&quot; \</span><br><span class="line">    |  seqkit replace -p .+ -r &quot;SAV_&#123;nr&#125;&quot; --nr-width 5</span><br><span class="line"></span><br><span class="line"></span><br></pre></td></tr></table></figure>
<h1 id="rename-重命名重复的id"><a class="markdownIt-Anchor" href="#rename-重命名重复的id"></a> rename 重命名重复的ID</h1>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line"># 重命名：相同序列会在后面加上_2 来处理</span><br><span class="line">echo -e &quot;&gt;a comment\nacgt\n&gt;b comment of b\nACTG\n&gt;a comment\naaaa&quot; \</span><br><span class="line">    | seqkit rename</span><br><span class="line"></span><br><span class="line"></span><br></pre></td></tr></table></figure>
<h1 id="concat-连接多个文件中具有相同id的序列"><a class="markdownIt-Anchor" href="#concat-连接多个文件中具有相同id的序列"></a> concat 连接多个文件中具有相同ID的序列</h1>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">#这里演示组合前面两个碱基和最后两个碱基的用法</span><br><span class="line">seqkit concat &lt;(seqkit subseq -r 1:2 C1_1.fq.gz) &lt;(seqkit subseq -r -2:-1 C1_2.fq.gz)|head</span><br><span class="line"></span><br></pre></td></tr></table></figure>
<h1 id="shuffle-随机打乱序列-默认全部读入内存"><a class="markdownIt-Anchor" href="#shuffle-随机打乱序列-默认全部读入内存"></a> shuffle 随机打乱序列 默认全部读入内存</h1>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">seqkit shuffle hairpin.fa.gz -2  &gt; shuffled.fa</span><br></pre></td></tr></table></figure>
<h1 id="sort-按id名称序列长度排序序列"><a class="markdownIt-Anchor" href="#sort-按id名称序列长度排序序列"></a> sort 按id/名称/序列/长度排序序列</h1>
<p>—quiet 屏幕不输出过程</p>
<p>-i 排序忽略大小写</p>
<p>-l 按照序列长度排序</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br></pre></td><td class="code"><pre><span class="line"># ID排序</span><br><span class="line">echo -e &quot;&gt;seq1\nACGTNcccc\n&gt;SEQ2\nacgtnAAAA&quot; \</span><br><span class="line">    | seqkit sort --quiet</span><br><span class="line"> </span><br><span class="line"># 按照ID排序，忽略大小写</span><br><span class="line">echo -e &quot;&gt;seq1\nACGTNcccc\n&gt;SEQ2\nacgtnAAAA&quot; \</span><br><span class="line">    | seqkit sort --quiet -i</span><br><span class="line"> </span><br><span class="line"># 按照序列长度排序，由小到大</span><br><span class="line">echo -e &quot;&gt;seq1\nACGTNcccc\n&gt;SEQ2\nacgtnAAAAnnn\n&gt;seq3\nacgt&quot; \</span><br><span class="line">    | seqkit sort --quiet -l</span><br><span class="line"></span><br><span class="line"></span><br></pre></td></tr></table></figure>
<h1 id="mutate-编辑序列点突变-插入-删除"><a class="markdownIt-Anchor" href="#mutate-编辑序列点突变-插入-删除"></a> mutate 编辑序列(点突变、插入、删除)</h1>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br></pre></td><td class="code"><pre><span class="line">echo -ne &quot;&gt;1\nACTGNactgn\n&gt;2\nactgnACTGN\n&quot;</span><br><span class="line"> </span><br><span class="line"># 修改第一个碱基</span><br><span class="line">echo -ne &quot;&gt;1\nACTGNactgn\n&gt;2\nactgnACTGN\n&quot; \</span><br><span class="line">    | seqkit mutate -p 1:x</span><br><span class="line"> </span><br><span class="line"># 修改第五个位置的碱基，输出信息隐藏    </span><br><span class="line">echo -ne &quot;&gt;1\nACTGNactgn\n&gt;2\nactgnACTGN\n&quot; \</span><br><span class="line">    | seqkit mutate -p 5:x --quiet</span><br><span class="line"> </span><br><span class="line"># 可以同时修改多个碱基</span><br><span class="line">echo -ne &quot;&gt;1\nACTGNactgn\n&gt;2\nactgnACTGN\n&quot; \</span><br><span class="line">    | seqkit mutate -p 1:x -p -1:x --quiet</span><br><span class="line"> </span><br><span class="line"># 删除碱基</span><br><span class="line">echo -ne &quot;&gt;1\nACTGNactgn\n&gt;2\nactgnACTGN\n&quot; \</span><br><span class="line">    | seqkit mutate -d 1:1 --quiet</span><br><span class="line"> </span><br><span class="line"># 删除倒数三个碱基</span><br><span class="line">echo -ne &quot;&gt;1\nACTGNactgn\n&gt;2\nactgnACTGN\n&quot; \</span><br><span class="line">    | seqkit mutate -d -3:-1 --quiet</span><br><span class="line"> </span><br><span class="line"># 插入碱基</span><br><span class="line">echo -ne &quot;&gt;1\nACTGNactgn\n&gt;2\nactgnACTGN\n&quot; \</span><br><span class="line">    | seqkit mutate -i 0:xx --quiet</span><br><span class="line"></span><br><span class="line"></span><br></pre></td></tr></table></figure>
<h1 id="head-展示开头n行的序列"><a class="markdownIt-Anchor" href="#head-展示开头n行的序列"></a> head 展示开头N行的序列</h1>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">seqkit head -n 1 hairpin.fa.gz</span><br><span class="line"></span><br><span class="line"></span><br></pre></td></tr></table></figure>
<h1 id="locate-定位子序列或者保守序列位置"><a class="markdownIt-Anchor" href="#locate-定位子序列或者保守序列位置"></a> locate 定位子序列或者保守序列位置</h1>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br></pre></td><td class="code"><pre><span class="line">cat gene.fa | seqkit locate -p ACT | csvtk pretty -t</span><br><span class="line"># 调整错配 最大错配为1</span><br><span class="line">cat gene.fa \</span><br><span class="line">  | seqkit locate -p ACTG -m 1 \</span><br><span class="line">  | csvtk pretty -t</span><br><span class="line"> </span><br><span class="line"># 简并碱基</span><br><span class="line">zcat hairpin.fa.gz \</span><br><span class="line">    | seqkit locate -i -d -p AUGGACUN \</span><br><span class="line">    | head -n 4</span><br><span class="line"></span><br><span class="line"></span><br></pre></td></tr></table></figure>
<h1 id="restart-重置循环基因组的起始位置"><a class="markdownIt-Anchor" href="#restart-重置循环基因组的起始位置"></a> restart 重置循环基因组的起始位置</h1>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line">echo -e &quot;&gt;seq\nacgtnACGTN&quot;</span><br><span class="line"> </span><br><span class="line">echo -e &quot;&gt;seq\nacgtnACGTN&quot; | seqkit restart -i 2</span><br><span class="line">echo -e &quot;&gt;seq\nacgtnACGTN&quot; | seqkit restart -i -2</span><br><span class="line"></span><br><span class="line"></span><br></pre></td></tr></table></figure>
<h1 id="convet-二代测序质量值的转化为-sanger"><a class="markdownIt-Anchor" href="#convet-二代测序质量值的转化为-sanger"></a> convet 二代测序质量值的转化为 Sanger</h1>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br></pre></td><td class="code"><pre><span class="line"># Illumina-1.8+ -&gt; Sanger</span><br><span class="line"># seqkit convert tests/Illimina1.8.fq.gz  | seqkit head -n 1</span><br><span class="line"># 40分以上的都认为40 Illumina-1.8+ -&gt; Sanger</span><br><span class="line"># seqkit convert tests/Illimina1.8.fq.gz -f | seqkit head -n 1</span><br><span class="line"> </span><br><span class="line">#Illumina-1.8+ -&gt; Illumina-1.5+</span><br><span class="line"># seqkit convert tests/Illimina1.8.fq.gz --to Illumina-1.5+ | seqkit head -n 1</span><br><span class="line"> </span><br><span class="line"># Illumina-1.8+ -&gt; Illumina-1.5+ -&gt;  Sanger.</span><br><span class="line"># seqkit convert tests/Illimina1.8.fq.gz --to Illumina-1.5+ | seqkit convert | seqkit head -n 1</span><br><span class="line"> </span><br><span class="line"># Solexa -&gt; Sanger</span><br><span class="line"># seqkit convert tests/Illimina1.8.fq.gz --from Solexa</span><br><span class="line"> </span><br><span class="line"># Illumina-1.5+ -&gt; Sanger</span><br><span class="line"># seqkit convert tests/Illimina1.5.fq | seqkit head -n 1</span><br><span class="line"></span><br><span class="line"></span><br></pre></td></tr></table></figure>
<h1 id="技术细节和使用"><a class="markdownIt-Anchor" href="#技术细节和使用"></a> 技术细节和使用</h1>
<p>seqkit处理压缩文件</p>
<p>pigz 或者 gzip 在部分操作中不能加速，所以在v.0.8.1版本以便关注了，然而还是可以使用命令：</p>
<p>pigz -d -c seqs.fq.gz | seqkit xxx<br>
因为seqkit使用了pgzip去写gzip。这比gzip和pigz更快。（10X of gzip, 4X of pigz），而且gzip压缩文件比较大。</p>
<p>从数据处理格式来讲</p>
<p>seqkit无缝支持fa和fq格式数据，并且可以自动识别。除了 faidx之外，全部命令都可以处理这两种格式的数据。</p>
<p>只有fa格式支持命令(subseq, split, sort和shuffle)利用FASTA索引(by flag -two-pass)下提高大文件的性能。</p>
<p>序列类型的检测DNA/RNA/Protein，会使用子序列进行，默认检测第一条子序列，通过—alphabet-guess-seq-length参数默认为10000，如果长度小于10000，则检查整条序列。</p>
<p>序列名字</p>
<p>所有的软件，包括seqkit，使用第一个空格之前的字符作为序列的名字：<br>
需要注意的NCBI等一些序列的格式并不是如此，例如：</p>
<blockquote>
<p>gi|110645304|ref|NC_002516.2| Pseudomona<br>
想要在seqkit中识别出来的序列ID为：NC_002516.2。</p>
</blockquote>
<p>此时使用参数–id-regexp &quot;|([^|]+)| &quot;，或者添加参数–id-ncbi，但如果是只要前面的gi数字作为ID的话，添加参数：–id-regexp “<sup>gi|([</sup>|]+)|”。</p>
<p>注意：.seqkit.fai不同于samtools产生的.fai格式文件，seqkit使用整个序列开头而不是ID作为索引。</p>
<p>并行运算</p>
<p>单核CPU默认线程：—threads 1，多个CPU，线程默认2.</p>
<p>内存占用</p>
<p>seqkit许多的命令都不需要将整个序列读入到内存中。包括：stat, fq2fa, fx2tab, tab2fx, grep, locate, replace, seq, sliding, subseq。</p>
<p>注意：如果使用subseq —gtf | —bed时，如果GTF或者BED文件太大，内存使用量会暴增，可以通过指定染色体：—chr，或者—feature去限制特征。</p>
<p>有一些命令需要将文件读入内存，但是可以用过rmdup 和 common减少内存使用。</p>
<p>随机—抽样</p>
<p>抽样命令sample和shuffle使用了随机功能，为了保证重现性，可以使用-s设置随机种子。这可以保证在不同的环境中可以有相同的抽样结果。</p>
<h1 id="序列长度分布"><a class="markdownIt-Anchor" href="#序列长度分布"></a> 序列长度分布</h1>
<figure class="highlight sh"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># -j 是线程数</span></span><br><span class="line">seqkit fx2tab -j 30 -l  -n -i -H file.fastq.gz | cut -f 4 &gt; Length.txt</span><br><span class="line"><span class="comment"># 查看Length.txt</span></span><br></pre></td></tr></table></figure>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br></pre></td><td class="code"><pre><span class="line">library(tidyverse)</span><br><span class="line"></span><br><span class="line">length &lt;- read_tsv(<span class="string">&quot;Length.txt&quot;</span>) %&gt;% group_by(<span class="built_in">length</span>) %&gt;%</span><br><span class="line">  summarise(Count = n())</span><br><span class="line"><span class="built_in">length</span>$length &lt;- <span class="built_in">as.character</span>(<span class="built_in">length</span>$<span class="built_in">length</span>)</span><br><span class="line">sum &lt;- <span class="built_in">sum</span>(<span class="built_in">length</span>$Count)</span><br><span class="line">ggplot(<span class="built_in">length</span>) + geom_col(aes(<span class="built_in">length</span>, Count), width = <span class="number">0.8</span>) + </span><br><span class="line">  geom_line(aes(<span class="built_in">length</span>, Count), group = <span class="number">1</span>) + geom_point(aes(<span class="built_in">length</span>, Count)) + </span><br><span class="line">  scale_y_continuous(sec.axis = sec_axis(~.*<span class="number">100</span>/<span class="built_in">sum</span>, name = <span class="string">&quot;% Relative Abundance&quot;</span>)) + xlab(<span class="string">&quot;Length&quot;</span>) +</span><br><span class="line">  theme_bw() + theme(panel.grid = element_blank(), </span><br><span class="line">                     axis.title = element_text(size = <span class="number">15</span>))</span><br><span class="line"></span><br><span class="line">ggsave(<span class="string">&quot;Length.png&quot;</span>, height = <span class="number">5</span>, width = <span class="number">8</span>)</span><br><span class="line">ggsave(<span class="string">&quot;Length.pdf&quot;</span>, height = <span class="number">5</span>, width = <span class="number">8</span>)</span><br></pre></td></tr></table></figure>
    </div>

    
    
    
        

<div>
<ul class="post-copyright">
  <li class="post-copyright-author">
    <strong>Post author:  </strong>Wang C
  </li>
  <li class="post-copyright-link">
    <strong>Post link: </strong>
    <a href="https://wangc1106.github.io/blog/2022/02/24/2022-02-24-Seqkit/" title="seqkit">https://wangc1106.github.io/blog/2022/02/24/2022-02-24-Seqkit/</a>
  </li>
  <li class="post-copyright-license">
    <strong>Copyright Notice:  </strong>All articles in this blog are licensed under <a href="https://creativecommons.org/licenses/by-nc-sa/4.0/" rel="noopener" target="_blank"><i class="fab fa-fw fa-creative-commons"></i>BY-NC-SA</a> unless stating additionally.
  </li>
</ul>
</div>


      <footer class="post-footer">
          <div class="post-tags">
              <a href="/blog/tags/sequence-tool/" rel="tag"># sequence-tool</a>
          </div>

        


        
    <div class="post-nav">
      <div class="post-nav-item">
    <a href="/blog/2022/02/24/2022-02-24-memory/" rel="prev" title="Linux 内存 存储">
      <i class="fa fa-chevron-left"></i> Linux 内存 存储
    </a></div>
      <div class="post-nav-item">
    <a href="/blog/2022/02/24/2022-02-25-soapdenovo2/" rel="next" title="Soapdenovo2">
      Soapdenovo2 <i class="fa fa-chevron-right"></i>
    </a></div>
    </div>
      </footer>
    
  </article>
  
  
  



          </div>
          

<script>
  window.addEventListener('tabs:register', () => {
    let { activeClass } = CONFIG.comments;
    if (CONFIG.comments.storage) {
      activeClass = localStorage.getItem('comments_active') || activeClass;
    }
    if (activeClass) {
      let activeTab = document.querySelector(`a[href="#comment-${activeClass}"]`);
      if (activeTab) {
        activeTab.click();
      }
    }
  });
  if (CONFIG.comments.storage) {
    window.addEventListener('tabs:click', event => {
      if (!event.target.matches('.tabs-comment .tab-content .tab-pane')) return;
      let commentClass = event.target.classList[1];
      localStorage.setItem('comments_active', commentClass);
    });
  }
</script>

        </div>
          
  
  <div class="toggle sidebar-toggle">
    <span class="toggle-line toggle-line-first"></span>
    <span class="toggle-line toggle-line-middle"></span>
    <span class="toggle-line toggle-line-last"></span>
  </div>

  <aside class="sidebar">
    <div class="sidebar-inner">

      <ul class="sidebar-nav motion-element">
        <li class="sidebar-nav-toc">
          Table of Contents
        </li>
        <li class="sidebar-nav-overview">
          Overview
        </li>
      </ul>

      <!--noindex-->
      <div class="post-toc-wrap sidebar-panel">
          <div class="post-toc motion-element"><ol class="nav"><li class="nav-item nav-level-1"><a class="nav-link" href="#seqkit-a-cross-platform-and-ultrafast-toolkit-for-fastaq-file-manipulation"><span class="nav-number">1.</span> <span class="nav-text"> SeqKit - a cross-platform and ultrafast toolkit for FASTA&#x2F;Q file manipulation</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#quick-guide"><span class="nav-number">2.</span> <span class="nav-text"> Quick Guide</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#command"><span class="nav-number">3.</span> <span class="nav-text"> command</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#parameter"><span class="nav-number">4.</span> <span class="nav-text"> Parameter:</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#stats-fastaq%E6%96%87%E4%BB%B6%E7%9A%84%E7%AE%80%E5%8D%95%E7%BB%9F%E8%AE%A1"><span class="nav-number">5.</span> <span class="nav-text"> stats FASTA&#x2F;Q文件的简单统计</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#seq-%E8%BD%AC%E6%8D%A2%E5%BA%8F%E5%88%97-%E5%8F%8D%E5%90%91-%E4%BA%92%E8%A1%A5%E6%8F%90%E5%8F%96id"><span class="nav-number">6.</span> <span class="nav-text"> seq 转换序列 （反向、互补&#x2F;提取ID）</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#%E6%8C%89%E9%95%BF%E5%BA%A6%E8%BF%87%E6%BB%A4%E5%B8%B8%E7%94%A8"><span class="nav-number">7.</span> <span class="nav-text"> 按长度过滤(常用)</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#%E6%8F%90%E5%8F%96id"><span class="nav-number">8.</span> <span class="nav-text"> 提取ID</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#%E5%8D%95%E8%A1%8C%E5%A4%9A%E8%A1%8C%E8%BD%AC%E6%8D%A2"><span class="nav-number">9.</span> <span class="nav-text"> 单行&#x2F;多行转换</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#%E5%8F%8D%E5%90%91%E4%BA%92%E8%A1%A5"><span class="nav-number">10.</span> <span class="nav-text"> 反向&#x2F;互补</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#%E5%88%A0%E9%99%A4gap%E5%A4%A7%E5%B0%8F%E5%86%99%E8%BD%AC%E6%8D%A2"><span class="nav-number">11.</span> <span class="nav-text"> 删除gap&#x2F;大小写转换</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#rna%E8%BD%AC%E4%B8%BAdna"><span class="nav-number">12.</span> <span class="nav-text"> RNA转为DNA</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#subseq%E9%80%9A%E8%BF%87%E6%8C%87%E5%AE%9A%E5%8C%BA%E5%9F%9F"><span class="nav-number">13.</span> <span class="nav-text"> subseq通过指定区域</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#sliding-%E6%BB%91%E7%AA%97%E6%8F%90%E5%8F%96%E5%BA%8F%E5%88%97%E6%94%AF%E6%8C%81%E7%8E%AF%E7%8A%B6%E5%9F%BA%E5%9B%A0%E7%BB%84"><span class="nav-number">14.</span> <span class="nav-text"> sliding 滑窗提取序列，支持环状基因组</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#faidx-%E5%88%9B%E5%BB%BAfasta%E7%B4%A2%E5%BC%95%E6%96%87%E4%BB%B6%E5%B9%B6%E6%8F%90%E5%8F%96%E5%AD%90%E5%BA%8F%E5%88%97"><span class="nav-number">15.</span> <span class="nav-text"> faidx 创建FASTA索引文件并提取子序列</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#watch-%E5%BA%8F%E5%88%97%E8%B4%A8%E9%87%8F%E7%9A%84%E7%9B%91%E6%B5%8B%E5%92%8C%E5%9C%A8%E7%BA%BF%E7%9B%B4%E6%96%B9%E5%9B%BE"><span class="nav-number">16.</span> <span class="nav-text"> watch 序列质量的监测和在线直方图</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#sana%E6%B8%85%E7%90%86%E6%8D%9F%E5%9D%8Ffastq%E6%96%87%E4%BB%B6"><span class="nav-number">17.</span> <span class="nav-text"> sana：清理损坏fastq文件</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#fq2fa-%E5%B0%86fq%E8%BD%AC%E4%B8%BAfa%E6%A0%BC%E5%BC%8F"><span class="nav-number">18.</span> <span class="nav-text"> fq2fa 将fq转为fa格式</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#fx2tab-tab2fx-%E5%BA%8F%E5%88%97%E8%BD%AC%E5%8C%96%E8%A1%A8%E6%A0%BC%E6%A0%BC%E5%BC%8F"><span class="nav-number">19.</span> <span class="nav-text"> fx2tab &amp; tab2fx 序列转化表格格式</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#translate-%E7%BF%BB%E8%AF%91dnarna%E4%B8%BA%E8%9B%8B%E7%99%BD%E8%B4%A8%E5%BA%8F%E5%88%97"><span class="nav-number">20.</span> <span class="nav-text"> translate 翻译DNA&#x2F;RNA为蛋白质序列</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#grep-%E5%BA%8F%E5%88%97%E5%8C%B9%E9%85%8D"><span class="nav-number">21.</span> <span class="nav-text"> grep  序列匹配</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#locate-%E8%BE%93%E5%87%BA%E5%8C%B9%E9%85%8D%E4%BD%8D%E7%BD%AE"><span class="nav-number">22.</span> <span class="nav-text"> locate 输出匹配位置</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#fish-%E4%BD%BF%E7%94%A8%E5%B1%80%E9%83%A8%E6%AF%94%E5%AF%B9%E5%9C%A8%E8%BE%83%E5%A4%A7%E7%9A%84%E5%BA%8F%E5%88%97%E4%B8%AD%E5%AF%BB%E6%89%BE%E7%9F%AD%E5%BA%8F%E5%88%97"><span class="nav-number">23.</span> <span class="nav-text"> fish 使用局部比对在较大的序列中寻找短序列</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#amplicon-%E9%80%9A%E8%BF%87%E5%BC%95%E7%89%A9%E6%A3%80%E7%B4%A2%E6%89%A9%E5%A2%9E%E5%AD%90%E6%88%96%E5%85%B6%E5%91%A8%E5%9B%B4%E7%9A%84%E7%89%B9%E5%AE%9A%E5%8C%BA%E5%9F%9F"><span class="nav-number">24.</span> <span class="nav-text"> amplicon 通过引物检索扩增子(或其周围的特定区域)</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#duplicate-%E5%AF%B9%E5%BA%8F%E5%88%97%E9%87%8D%E5%A4%8Dn%E6%AC%A1"><span class="nav-number">25.</span> <span class="nav-text"> duplicate 对序列重复N次</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#rmdup-%E9%80%9A%E8%BF%87id%E5%90%8D%E7%A7%B0%E5%BA%8F%E5%88%97%E5%88%A0%E9%99%A4%E9%87%8D%E5%A4%8D%E7%9A%84%E5%BA%8F%E5%88%97"><span class="nav-number">26.</span> <span class="nav-text"> rmdup 通过id&#x2F;名称&#x2F;序列删除重复的序列</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#common-%E9%80%9A%E8%BF%87id%E5%90%8D%E7%A7%B0%E5%BA%8F%E5%88%97%E6%9F%A5%E6%89%BE%E5%A4%9A%E4%B8%AA%E6%96%87%E4%BB%B6%E7%9A%84%E5%85%AC%E5%85%B1%E5%BA%8F%E5%88%97"><span class="nav-number">27.</span> <span class="nav-text"> common ：通过id&#x2F;名称&#x2F;序列查找多个文件的公共序列</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#split-%E6%8B%86%E5%88%86%E5%BA%8F%E5%88%97%E4%B8%BA%E5%AD%90%E6%96%87%E4%BB%B6"><span class="nav-number">28.</span> <span class="nav-text"> split 拆分序列为子文件</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#split2-%E6%8B%86%E5%88%86%E6%96%87%E4%BB%B6-%E5%8D%87%E7%BA%A7%E7%89%88%E6%9C%AC"><span class="nav-number">29.</span> <span class="nav-text"> split2 拆分文件 升级版本</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#pair-%E6%8B%BC%E6%8E%A5%E4%B8%A4%E4%B8%AAfastq%E6%96%87%E4%BB%B6"><span class="nav-number">30.</span> <span class="nav-text"> pair 拼接两个fastq文件</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#sample-%E6%8C%89%E6%95%B0%E9%87%8F%E6%88%96%E6%AF%94%E4%BE%8B%E5%AF%B9%E5%BA%8F%E5%88%97%E8%BF%9B%E8%A1%8C%E6%8A%BD%E6%A0%B7"><span class="nav-number">31.</span> <span class="nav-text"> sample 按数量或比例对序列进行抽样。</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#range-%E6%89%93%E5%8D%B0%E5%BA%8F%E5%88%97-%E6%8C%89%E7%85%A7%E4%B8%80%E4%B8%AA%E8%8C%83%E5%9B%B4"><span class="nav-number">32.</span> <span class="nav-text"> range 打印序列 按照一个范围</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#repeat-%E4%BD%BF%E7%94%A8%E6%AD%A3%E5%88%99%E8%A1%A8%E8%BE%BE%E5%BC%8F%E6%9B%BF%E6%8D%A2%E5%90%8D%E7%A7%B0%E5%BA%8F%E5%88%97"><span class="nav-number">33.</span> <span class="nav-text"> repeat 使用正则表达式替换名称&#x2F;序列。</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#rename-%E9%87%8D%E5%91%BD%E5%90%8D%E9%87%8D%E5%A4%8D%E7%9A%84id"><span class="nav-number">34.</span> <span class="nav-text"> rename 重命名重复的ID</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#concat-%E8%BF%9E%E6%8E%A5%E5%A4%9A%E4%B8%AA%E6%96%87%E4%BB%B6%E4%B8%AD%E5%85%B7%E6%9C%89%E7%9B%B8%E5%90%8Cid%E7%9A%84%E5%BA%8F%E5%88%97"><span class="nav-number">35.</span> <span class="nav-text"> concat 连接多个文件中具有相同ID的序列</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#shuffle-%E9%9A%8F%E6%9C%BA%E6%89%93%E4%B9%B1%E5%BA%8F%E5%88%97-%E9%BB%98%E8%AE%A4%E5%85%A8%E9%83%A8%E8%AF%BB%E5%85%A5%E5%86%85%E5%AD%98"><span class="nav-number">36.</span> <span class="nav-text"> shuffle 随机打乱序列 默认全部读入内存</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#sort-%E6%8C%89id%E5%90%8D%E7%A7%B0%E5%BA%8F%E5%88%97%E9%95%BF%E5%BA%A6%E6%8E%92%E5%BA%8F%E5%BA%8F%E5%88%97"><span class="nav-number">37.</span> <span class="nav-text"> sort 按id&#x2F;名称&#x2F;序列&#x2F;长度排序序列</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#mutate-%E7%BC%96%E8%BE%91%E5%BA%8F%E5%88%97%E7%82%B9%E7%AA%81%E5%8F%98-%E6%8F%92%E5%85%A5-%E5%88%A0%E9%99%A4"><span class="nav-number">38.</span> <span class="nav-text"> mutate 编辑序列(点突变、插入、删除)</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#head-%E5%B1%95%E7%A4%BA%E5%BC%80%E5%A4%B4n%E8%A1%8C%E7%9A%84%E5%BA%8F%E5%88%97"><span class="nav-number">39.</span> <span class="nav-text"> head 展示开头N行的序列</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#locate-%E5%AE%9A%E4%BD%8D%E5%AD%90%E5%BA%8F%E5%88%97%E6%88%96%E8%80%85%E4%BF%9D%E5%AE%88%E5%BA%8F%E5%88%97%E4%BD%8D%E7%BD%AE"><span class="nav-number">40.</span> <span class="nav-text"> locate 定位子序列或者保守序列位置</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#restart-%E9%87%8D%E7%BD%AE%E5%BE%AA%E7%8E%AF%E5%9F%BA%E5%9B%A0%E7%BB%84%E7%9A%84%E8%B5%B7%E5%A7%8B%E4%BD%8D%E7%BD%AE"><span class="nav-number">41.</span> <span class="nav-text"> restart 重置循环基因组的起始位置</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#convet-%E4%BA%8C%E4%BB%A3%E6%B5%8B%E5%BA%8F%E8%B4%A8%E9%87%8F%E5%80%BC%E7%9A%84%E8%BD%AC%E5%8C%96%E4%B8%BA-sanger"><span class="nav-number">42.</span> <span class="nav-text"> convet 二代测序质量值的转化为 Sanger</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#%E6%8A%80%E6%9C%AF%E7%BB%86%E8%8A%82%E5%92%8C%E4%BD%BF%E7%94%A8"><span class="nav-number">43.</span> <span class="nav-text"> 技术细节和使用</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#%E5%BA%8F%E5%88%97%E9%95%BF%E5%BA%A6%E5%88%86%E5%B8%83"><span class="nav-number">44.</span> <span class="nav-text"> 序列长度分布</span></a></li></ol></div>
      </div>
      <!--/noindex-->

      <div class="site-overview-wrap sidebar-panel">
        <div class="site-author motion-element" itemprop="author" itemscope itemtype="http://schema.org/Person">
    <img class="site-author-image" itemprop="image" alt="Wang C"
      src="/blog/images/banner.png">
  <p class="site-author-name" itemprop="name">Wang C</p>
  <div class="site-description" itemprop="description">Make more time</div>
</div>
<div class="site-state-wrap motion-element">
  <nav class="site-state">
      <div class="site-state-item site-state-posts">
          <a href="/blog/archives">
          <span class="site-state-item-count">30</span>
          <span class="site-state-item-name">posts</span>
        </a>
      </div>
      <div class="site-state-item site-state-categories">
            <a href="/blog/categories/">
          
        <span class="site-state-item-count">11</span>
        <span class="site-state-item-name">categories</span></a>
      </div>
      <div class="site-state-item site-state-tags">
            <a href="/blog/tags/">
          
        <span class="site-state-item-count">18</span>
        <span class="site-state-item-name">tags</span></a>
      </div>
  </nav>
</div>
  <div class="links-of-author motion-element">
      <span class="links-of-author-item">
        <a href="https://github.com/wangc1106" title="GitHub → https:&#x2F;&#x2F;github.com&#x2F;wangc1106" rel="noopener" target="_blank"><i class="fab fa-github fa-fw"></i>GitHub</a>
      </span>
      <span class="links-of-author-item">
        <a href="mailto:wangchen116@126.com" title="E-Mail → mailto:wangchen116@126.com" rel="noopener" target="_blank"><i class="fa fa-envelope fa-fw"></i>E-Mail</a>
      </span>
  </div>
  <div class="cc-license motion-element" itemprop="license">
    <a href="https://creativecommons.org/licenses/by-nc-sa/4.0/" class="cc-opacity" rel="noopener" target="_blank"><img src="/blog/images/cc-by-nc-sa.svg" alt="Creative Commons"></a>
  </div>



      </div>
        <div class="back-to-top motion-element">
          <i class="fa fa-arrow-up"></i>
          <span>0%</span>
        </div>

    </div>
  </aside>
  <div id="sidebar-dimmer"></div>


      </div>
    </main>

    <footer class="footer">
      <div class="footer-inner">
        

        

<div class="copyright">
  
  &copy; 2021 – 
  <span itemprop="copyrightYear">2022</span>
  <span class="with-love">
    <i class="fas fa-dna"></i>
  </span>
  <span class="author" itemprop="copyrightHolder">Wang C</span>
</div>

        






<script>
  (function() {
    function leancloudSelector(url) {
      url = encodeURI(url);
      return document.getElementById(url).querySelector('.leancloud-visitors-count');
    }

    function addCount(Counter) {
      var visitors = document.querySelector('.leancloud_visitors');
      var url = decodeURI(visitors.id);
      var title = visitors.dataset.flagTitle;

      Counter('get', '/classes/Counter?where=' + encodeURIComponent(JSON.stringify({ url })))
        .then(response => response.json())
        .then(({ results }) => {
          if (results.length > 0) {
            var counter = results[0];
            leancloudSelector(url).innerText = counter.time + 1;
            Counter('put', '/classes/Counter/' + counter.objectId, { time: { '__op': 'Increment', 'amount': 1 } })
              .catch(error => {
                console.error('Failed to save visitor count', error);
              });
          } else {
              Counter('post', '/classes/Counter', { title, url, time: 1 })
                .then(response => response.json())
                .then(() => {
                  leancloudSelector(url).innerText = 1;
                })
                .catch(error => {
                  console.error('Failed to create', error);
                });
          }
        })
        .catch(error => {
          console.error('LeanCloud Counter Error', error);
        });
    }

    function showTime(Counter) {
      var visitors = document.querySelectorAll('.leancloud_visitors');
      var entries = [...visitors].map(element => {
        return decodeURI(element.id);
      });

      Counter('get', '/classes/Counter?where=' + encodeURIComponent(JSON.stringify({ url: { '$in': entries } })))
        .then(response => response.json())
        .then(({ results }) => {
          for (let url of entries) {
            let target = results.find(item => item.url === url);
            leancloudSelector(url).innerText = target ? target.time : 0;
          }
        })
        .catch(error => {
          console.error('LeanCloud Counter Error', error);
        });
    }

    let { app_id, app_key, server_url } = {"enable":true,"app_id":"UqIYufmNk2sE8MiBKKWc4Kja-gzGzoHsz","app_key":"iYpeWQcmNqpHwr2uVlnJYVji","server_url":null,"security":false};
    function fetchData(api_server) {
      var Counter = (method, url, data) => {
        return fetch(`${api_server}/1.1${url}`, {
          method,
          headers: {
            'X-LC-Id'     : app_id,
            'X-LC-Key'    : app_key,
            'Content-Type': 'application/json',
          },
          body: JSON.stringify(data)
        });
      };
      if (CONFIG.page.isPost) {
        if (CONFIG.hostname !== location.hostname) return;
        addCount(Counter);
      } else if (document.querySelectorAll('.post-title-link').length >= 1) {
        showTime(Counter);
      }
    }

    let api_server = app_id.slice(-9) !== '-MdYXbMMI' ? server_url : `https://${app_id.slice(0, 8).toLowerCase()}.api.lncldglobal.com`;

    if (api_server) {
      fetchData(api_server);
    } else {
      fetch('https://app-router.leancloud.cn/2/route?appId=' + app_id)
        .then(response => response.json())
        .then(({ api_server }) => {
          fetchData('https://' + api_server);
        });
    }
  })();
</script>


      </div>
      <div style="height: 200px">
      <script type="text/javascript" src="//rf.revolvermaps.com/0/0/8.js?i=5knf34gs07t&amp;m=0&amp;c=ff0000&amp;cr1=ffffff&amp;f=arial&amp;l=33" async="async"></script>
      </div>
    </footer>
  </div>

  
  <script src="/blog/lib/anime.min.js"></script>
  <script src="/blog/lib/velocity/velocity.min.js"></script>
  <script src="/blog/lib/velocity/velocity.ui.min.js"></script>

<script src="/blog/js/utils.js"></script>

<script src="/blog/js/motion.js"></script>


<script src="/blog/js/schemes/pisces.js"></script>


<script src="/blog/js/next-boot.js"></script>




  




  
<script src="/blog/js/local-search.js"></script>













  

  

</body>
</html>
