<!DOCTYPE html>
<html lang="zh-Hans">
<head>
  <meta charset="UTF-8">
<meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=2">
<meta name="theme-color" content="#222">
<meta name="generator" content="Hexo 5.4.0">
  <link rel="apple-touch-icon" sizes="180x180" href="/blog/images/apple-touch-icon-next.png">
  <link rel="icon" type="image/png" sizes="32x32" href="/blog/images/favicon-32x32-next.png">
  <link rel="icon" type="image/png" sizes="16x16" href="/blog/images/favicon-16x16-next.png">
  <link rel="mask-icon" href="/blog/images/logo.svg" color="#222">
  <meta name="google-site-verification" content="1osTo_gOe1SW9Hevqtz8vl8HfSgTO4DSMZ0DiMPMX6w">

<link rel="stylesheet" href="/blog/css/main.css">


<link rel="stylesheet" href="/blog/lib/font-awesome/css/all.min.css">

<script id="hexo-configurations">
    var NexT = window.NexT || {};
    var CONFIG = {"hostname":"wangc1106.github.io","root":"/blog/","scheme":"Gemini","version":"7.8.0","exturl":false,"sidebar":{"position":"left","display":"post","padding":18,"offset":12,"onmobile":false},"copycode":{"enable":false,"show_result":false,"style":null},"back2top":{"enable":true,"sidebar":true,"scrollpercent":"ture"},"bookmark":{"enable":false,"color":"#222","save":"auto"},"fancybox":false,"mediumzoom":false,"lazyload":false,"pangu":false,"comments":{"style":"tabs","active":null,"storage":true,"lazyload":false,"nav":null},"algolia":{"hits":{"per_page":10},"labels":{"input_placeholder":"Search for Posts","hits_empty":"We didn't find any results for the search: ${query}","hits_stats":"${hits} results found in ${time} ms"}},"localsearch":{"enable":true,"trigger":"auto","top_n_per_article":1,"unescape":false,"preload":false},"motion":{"enable":true,"async":false,"transition":{"post_block":"fadeIn","post_header":"slideDownIn","post_body":"slideDownIn","coll_header":"slideLeftIn","sidebar":"slideUpIn"}},"path":"search.xml"};
  </script>

  <meta name="description" content="Bowtie 2 is an ultrafast and memory-efficient tool for aligning sequencing reads to long reference sequences. It is particularly good at aligning reads of about 50 up to 100s or 1,000s of characters,">
<meta property="og:type" content="article">
<meta property="og:title" content="【bowtie2】">
<meta property="og:url" content="https://wangc1106.github.io/blog/2022/03/20/2022-03-20-bowtie2/index.html">
<meta property="og:site_name" content="Banner&#39;s Blog">
<meta property="og:description" content="Bowtie 2 is an ultrafast and memory-efficient tool for aligning sequencing reads to long reference sequences. It is particularly good at aligning reads of about 50 up to 100s or 1,000s of characters,">
<meta property="og:locale">
<meta property="og:image" content="https://wangc1106.github.io/blog/2022/03/20/2022-03-20-bowtie2/1.png">
<meta property="article:published_time" content="2022-03-20T02:15:39.370Z">
<meta property="article:modified_time" content="2022-03-20T12:38:03.299Z">
<meta property="article:author" content="Wang C">
<meta property="article:tag" content="alignment">
<meta name="twitter:card" content="summary">
<meta name="twitter:image" content="https://wangc1106.github.io/blog/2022/03/20/2022-03-20-bowtie2/1.png">

<link rel="canonical" href="https://wangc1106.github.io/blog/2022/03/20/2022-03-20-bowtie2/">


<script id="page-configurations">
  // https://hexo.io/docs/variables.html
  CONFIG.page = {
    sidebar: "",
    isHome : false,
    isPost : true,
    lang   : 'zh-Hans'
  };
</script>

  <title>【bowtie2】 | Banner's Blog</title>
  






  <noscript>
  <style>
  .use-motion .brand,
  .use-motion .menu-item,
  .sidebar-inner,
  .use-motion .post-block,
  .use-motion .pagination,
  .use-motion .comments,
  .use-motion .post-header,
  .use-motion .post-body,
  .use-motion .collection-header { opacity: initial; }

  .use-motion .site-title,
  .use-motion .site-subtitle {
    opacity: initial;
    top: initial;
  }

  .use-motion .logo-line-before i { left: initial; }
  .use-motion .logo-line-after i { right: initial; }
  </style>
</noscript>

</head>

<body itemscope itemtype="http://schema.org/WebPage">
  <div class="container use-motion">
    <div class="headband"></div>

    <header class="header" itemscope itemtype="http://schema.org/WPHeader">
      <div class="header-inner"><div class="site-brand-container">
  <div class="site-nav-toggle">
    <div class="toggle" aria-label="Toggle navigation bar">
      <span class="toggle-line toggle-line-first"></span>
      <span class="toggle-line toggle-line-middle"></span>
      <span class="toggle-line toggle-line-last"></span>
    </div>
  </div>

  <div class="site-meta">

    <a href="/blog/" class="brand" rel="start">
      <span class="logo-line-before"><i></i></span>
      <h1 class="site-title">Banner's Blog</h1>
      <span class="logo-line-after"><i></i></span>
    </a>
      <p class="site-subtitle" itemprop="description">Dream big work hard</p>
  </div>

  <div class="site-nav-right">
    <div class="toggle popup-trigger">
        <i class="fa fa-search fa-fw fa-lg"></i>
    </div>
  </div>
</div>




<nav class="site-nav">
  <ul id="menu" class="main-menu menu">
        <li class="menu-item menu-item-home">

    <a href="/blog/" rel="section"><i class="home fa-fw"></i>Home</a>

  </li>
        <li class="menu-item menu-item-categories">

    <a href="/blog/categories/" rel="section"><i class="th fa-fw"></i>Categories</a>

  </li>
        <li class="menu-item menu-item-tags">

    <a href="/blog/tags/" rel="section"><i class="tags fa-fw"></i>Tags</a>

  </li>
        <li class="menu-item menu-item-about">

    <a href="https://wangc1106.github.io/" rel="section"><i class="user fa-fw"></i>About</a>

  </li>
      <li class="menu-item menu-item-search">
        <a role="button" class="popup-trigger"><i class="fa fa-search fa-fw"></i>Search
        </a>
      </li>
  </ul>
</nav>



  <div class="search-pop-overlay">
    <div class="popup search-popup">
        <div class="search-header">
  <span class="search-icon">
    <i class="fa fa-search"></i>
  </span>
  <div class="search-input-container">
    <input autocomplete="off" autocapitalize="off"
           placeholder="Searching..." spellcheck="false"
           type="search" class="search-input">
  </div>
  <span class="popup-btn-close">
    <i class="fa fa-times-circle"></i>
  </span>
</div>
<div id="search-result">
  <div id="no-result">
    <i class="fa fa-spinner fa-pulse fa-5x fa-fw"></i>
  </div>
</div>

    </div>
  </div>

</div>
    </header>

    


    <main class="main">
      <div class="main-inner">
        <div class="content-wrap">
          

          <div class="content post posts-expand">
            

    
  
  
  <article itemscope itemtype="http://schema.org/Article" class="post-block" lang="zh-Hans">
    <link itemprop="mainEntityOfPage" href="https://wangc1106.github.io/blog/2022/03/20/2022-03-20-bowtie2/">

    <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
      <meta itemprop="image" content="/blog/images/banner.png">
      <meta itemprop="name" content="Wang C">
      <meta itemprop="description" content="Make more time">
    </span>

    <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
      <meta itemprop="name" content="Banner's Blog">
    </span>
      <header class="post-header">
        <h1 class="post-title" itemprop="name headline">
          【bowtie2】
        </h1>

        <div class="post-meta">
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-calendar"></i>
              </span>
              <span class="post-meta-item-text">Posted on</span>
              

              <time title="Created: 2022-03-20 10:15:39 / Modified: 20:38:03" itemprop="dateCreated datePublished" datetime="2022-03-20T10:15:39+08:00">2022-03-20</time>
            </span>
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-folder"></i>
              </span>
              <span class="post-meta-item-text">In</span>
                <span itemprop="about" itemscope itemtype="http://schema.org/Thing">
                  <a href="/blog/categories/Software/" itemprop="url" rel="index"><span itemprop="name">Software</span></a>
                </span>
            </span>

          
            <span id="/blog/2022/03/20/2022-03-20-bowtie2/" class="post-meta-item leancloud_visitors" data-flag-title="【bowtie2】" title="Views">
              <span class="post-meta-item-icon">
                <i class="fa fa-eye"></i>
              </span>
              <span class="post-meta-item-text">Views: </span>
              <span class="leancloud-visitors-count"></span>
            </span>
            <span class="post-meta-item" title="Symbols count in article">
              <span class="post-meta-item-icon">
                <i class="far fa-file-word"></i>
              </span>
              <span>73k</span>
            </span>
            <span class="post-meta-item" title="Reading time">
              <span class="post-meta-item-icon">
                <i class="far fa-clock"></i>
              </span>
              <span>1:06</span>
            </span>

        </div>
      </header>

    
    
    
    <div class="post-body" itemprop="articleBody">

      
        <p><strong>Bowtie 2</strong> is an ultrafast and memory-efficient tool for aligning sequencing reads to long reference sequences. It is particularly good at aligning reads of about <strong>50 up to 100s or 1,000s</strong> of characters, and particularly <strong>good at aligning to relatively long</strong> (e.g. mammalian) genomes. Bowtie 2 indexes the genome with an <strong>FM Index</strong> to keep its <strong>memory footprint small</strong>: for the human genome, its memory footprint is typically around 3.2 GB. Bowtie 2 supports <strong>gapped</strong>, <strong>local</strong>, and <strong>paired-end</strong> alignment modes.</p>
<span id="more"></span>

<p><a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml">http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml</a></p>
<h1 id="Introduction"><a href="#Introduction" class="headerlink" title="Introduction"></a>Introduction</h1><p><a target="_blank" rel="noopener" href="http://bowtie-bio.sf.net/bowtie2">Bowtie 2</a> is an ultrafast and memory-efficient tool for aligning sequencing reads to long reference sequences. It is particularly good at aligning reads of about 50 up to 100s of characters to relatively long (e.g. mammalian) genomes. Bowtie 2 indexes the genome with an <a target="_blank" rel="noopener" href="http://en.wikipedia.org/wiki/FM-index">FM Index</a> (based on the <a target="_blank" rel="noopener" href="http://en.wikipedia.org/wiki/Burrows-Wheeler_transform">Burrows-Wheeler Transform</a> or <a target="_blank" rel="noopener" href="http://en.wikipedia.org/wiki/Burrows-Wheeler_transform">BWT</a>) to keep its memory footprint small: for the human genome, its memory footprint is typically around 3.2 gigabytes of RAM. Bowtie 2 supports gapped, local, and paired-end alignment modes. Multiple processors can be used simultaneously to achieve greater alignment speed.</p>
<p>Bowtie 2 outputs alignments in <a target="_blank" rel="noopener" href="http://samtools.sourceforge.net/SAM1.pdf">SAM</a> format, enabling interoperation with a large number of other tools (e.g. <a target="_blank" rel="noopener" href="http://samtools.sourceforge.net/">SAMtools</a>, <a target="_blank" rel="noopener" href="http://www.broadinstitute.org/gsa/wiki/index.php/The_Genome_Analysis_Toolkit">GATK</a>) that use SAM. Bowtie 2 is distributed under the <a target="_blank" rel="noopener" href="http://www.gnu.org/licenses/gpl-3.0.html">GPLv3 license</a>, and it runs on the command line under Windows, Mac OS X and Linux and BSD.</p>
<p><a target="_blank" rel="noopener" href="http://bowtie-bio.sf.net/bowtie2">Bowtie 2</a> is often the first step in pipelines for <font color="blue"><strong>comparative genomics</strong></font>, including for <font color="blue"><strong>variation calling, ChIP-seq, RNA-seq, BS-seq</strong></font>. <a target="_blank" rel="noopener" href="http://bowtie-bio.sf.net/bowtie2">Bowtie 2</a> and <a target="_blank" rel="noopener" href="http://bowtie-bio.sf.net/">Bowtie</a>(also called “<a target="_blank" rel="noopener" href="http://bowtie-bio.sf.net/">Bowtie 1</a>“ here) are also tightly integrated into many other tools, some of which <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/other_tools.shtml">are listed here</a>.</p>
<p>If you use <a target="_blank" rel="noopener" href="http://bowtie-bio.sf.net/bowtie2">Bowtie 2</a> for your published research, please cite our work. Papers describing Bowtie 2 are:</p>
<ul>
<li>Langmead B, Wilks C, Antonescu V, Charles R. <a target="_blank" rel="noopener" href="https://doi.org/10.1093/bioinformatics/bty648">Scaling read aligners to hundreds of threads on general-purpose processors</a>. <em>Bioinformatics</em>. 2018 Jul 18. doi: 10.1093/bioinformatics/bty648.</li>
<li>Langmead B, Salzberg SL. <a target="_blank" rel="noopener" href="https://www.nature.com/articles/nmeth.1923">Fast gapped-read alignment with Bowtie 2</a>. <em>Nature Methods</em>. 2012 Mar 4;9(4):357-9. doi: 10.1038/nmeth.1923.</li>
</ul>
<h2 id="How-is-Bowtie-2-different-from-Bowtie-1"><a href="#How-is-Bowtie-2-different-from-Bowtie-1" class="headerlink" title="How is Bowtie 2 different from Bowtie 1?"></a>How is Bowtie 2 different from Bowtie 1?</h2><p><font color="blue"><strong>Bowtie 1</strong></font> was released in 2009 and was geared toward <strong>aligning the relatively short sequencing reads</strong> (up to <strong>25-50 nucl</strong>eotides) prevalent at the time. Since then, technology has improved both sequencing throughput (more nucleotides produced per sequencer per day) and read length (more nucleotides per read).</p>
<p>The chief differences between Bowtie 1 and Bowtie 2 are:</p>
<ol>
<li>For reads <strong>longer than about 50 bp</strong> Bowtie 2 is generally <strong>faster</strong>, more <strong>sensitive</strong>, and uses <strong>less memory</strong> than Bowtie 1. For relatively short reads (e.g. <strong>less than 50 bp) Bowtie 1</strong> is sometimes <strong>faster</strong> and/or more <strong>sensitive</strong>.</li>
<li>Bowtie 2 <strong>supports gapped alignment</strong> with affine gap penalties. Number of gaps and gap lengths are not restricted, except by way of the configurable scoring scheme. <strong>Bowtie 1</strong> finds just <strong>ungapped alignments</strong>.</li>
<li>Bowtie 2 <strong>supports</strong> <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#end-to-end-alignment-versus-local-alignment"><strong>local alignment</strong></a>, which doesn’t require reads to align end-to-end. Local alignments might be “trimmed” (“soft clipped”) at one or both extremes in a way that optimizes alignment score. Bowtie 2 <strong>also supports</strong> <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#end-to-end-alignment-versus-local-alignment">end-to-end alignment</a> which, like <strong>Bowtie 1, requires that the read align entirely.</strong></li>
<li>There is <strong>no upper limit on read length</strong> in <strong>Bowtie 2</strong>. Bowtie 1 had an upper limit of around 1000 bp.</li>
<li>Bowtie 2 <strong>allows</strong> alignments to <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#ambiguous-characters">overlap ambiguous characters</a> (e.g. <code>N</code>s) in the reference. Bowtie 1 does not.</li>
<li>Bowtie 2 does away with Bowtie 1’s notion of alignment “stratum”, and its distinction between “Maq-like” and “end-to-end” modes. In Bowtie 2 all alignments <strong>lie along a continuous spectrum of alignment scores</strong> where the <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#scores-higher-more-similar">scoring scheme</a>, <strong>similar to</strong> <a target="_blank" rel="noopener" href="http://en.wikipedia.org/wiki/Needleman-Wunsch_algorithm">Needleman-Wunsch</a> and <a target="_blank" rel="noopener" href="http://en.wikipedia.org/wiki/Smith_waterman">Smith-Waterman</a>.</li>
<li>Bowtie 2’s <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#aligning-pairs">paired-end alignment</a> is <strong>more flexible</strong>. E.g. for pairs that do not align in a paired fashion, Bowtie 2 attempts to <strong>find unpaired alignments for each mate</strong>.</li>
<li>Bowtie 2 <strong>reports</strong> a spectrum of <strong>mapping qualities</strong>, in contrast for Bowtie 1 which reports either 0 or high.</li>
<li>Bowtie 2 does not align <strong>colorspace reads?</strong>.</li>
</ol>
<p>Bowtie 2 is not a “drop-in” replacement for Bowtie 1. Bowtie 2’s command-line <strong>arguments and genome index format are both different</strong> from Bowtie 1’s.</p>
<h2 id="What-isn’t-Bowtie-2"><a href="#What-isn’t-Bowtie-2" class="headerlink" title="What isn’t Bowtie 2?"></a>What isn’t Bowtie 2?</h2><p>Bowtie 2 is geared toward aligning relatively short sequencing reads to long genomes. That said, it handles arbitrarily small reference sequences (e.g. amplicons) and very long reads (i.e. upwards of 10s or 100s of kilobases), though it is slower in those settings. It is optimized for the read lengths and error modes yielded by typical Illumina sequencers.</p>
<p>Bowtie 2 <strong>does not support</strong> alignment of colorspace reads. (Bowtie 1 does.)</p>
<h1 id="Obtaining-Bowtie-2"><a href="#Obtaining-Bowtie-2" class="headerlink" title="Obtaining Bowtie 2"></a>Obtaining Bowtie 2</h1><p>Bowtie 2 is available from various package managers, notably <a target="_blank" rel="noopener" href="https://anaconda.org/bioconda/bowtie2">Bioconda</a>. With Bioconda installed, you should be able to install Bowtie 2 with <code>conda install bowtie2</code>.</p>
<p>Containerized versions of Bowtie 2 are also available via the <a target="_blank" rel="noopener" href="https://biocontainers.pro/">Biocontainers</a> project (e.g. <a target="_blank" rel="noopener" href="https://hub.docker.com/r/biocontainers/bowtie2/">via Docker Hub</a>).</p>
<p>You can also download Bowtie 2 sources and binaries from the <a target="_blank" rel="noopener" href="https://sourceforge.net/projects/bowtie-bio/files/bowtie2/">Download</a> section of the Sourceforge site. <strong>Binaries</strong> are available for the <code>x86_64</code>architecture running Linux, Mac OS X, and Windows.  If you plan to <strong>compile Bowtie 2 yourself</strong>, make sure to get the <strong>source package</strong>, i.e., the filename that ends in “-source.zip”.</p>
<h2 id="Building-from-source"><a href="#Building-from-source" class="headerlink" title="Building from source"></a>Building from source</h2><p>Building from source</p>
<p>Unzip the <strong>-source.zip</strong>, change to the unzipped directory, and build the Bowtie 2 tools by running GNU <code>make</code> (usually with the command <code>make</code>, but sometimes with <code>gmake</code>) with no arguments. </p>
<p>The Bowtie 2 Makefile also includes recipes for basic automatic dependency management. Running <code>make static-libs &amp;&amp; make STATIC_BUILD=1</code>will issue a series of commands that will: 1. download <strong>zstd</strong> and <strong>zlib 2</strong>. compile them as static libraries 3. <strong>link the resulting libraries to the compiled Bowtie 2 binaries</strong></p>
<p>As of version 2.3.5 bowtie2 now <strong>supports aligning SRA reads</strong>. Prepackaged builds will include a package that supports SRA. If you’re building bowtie2 from source please make sure that the Java runtime is available on your system. You can then proceed with the build by running <code>make sra-deps &amp;&amp; make USE_SRA=1</code>.</p>
<h1 id="The-bowtie2-aligner"><a href="#The-bowtie2-aligner" class="headerlink" title="The bowtie2 aligner"></a>The <code>bowtie2</code> aligner</h1><p><code>bowtie2</code> takes a Bowtie 2 <strong>index</strong> and a set of sequencing <strong>read</strong> files and outputs a set of alignments in <strong>SAM</strong> format.</p>
<p>“Alignment” is the process by which we discover <strong>how and where the read sequences are similar to the reference</strong> sequence. An “alignment” is a result from this process, specifically: an alignment is a way of “lining up” some or all of the characters in the read with some characters from the reference in a way that reveals how they’re similar. For example:</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">Read:      GACTGGGCGATCTCGACTTCG</span><br><span class="line">           |||||  |||||||||| |||</span><br><span class="line">Reference: GACTG--CGATCTCGACATCG</span><br></pre></td></tr></table></figure>

<p>Where dash symbols represent <strong>gaps</strong> and <strong>vertical bars</strong> show where aligned characters <strong>match</strong>.</p>
<p>We use alignment to <strong>make an educated guess</strong> as to where a read originated with respect to the reference genome. <strong>It’s not always possible to determine this with certainty</strong>. For instance, if the reference genome contains several long stretches of As (<code>AAAAAAAAA</code> etc.) and the read sequence is a short stretch of As (<code>AAAAAAA</code>), we cannot know for certain exactly where in the sea of <code>A</code>s the read originated.</p>
<h2 id="End-to-end-alignment-versus-local-alignment"><a href="#End-to-end-alignment-versus-local-alignment" class="headerlink" title="End-to-end alignment versus local alignment"></a>End-to-end alignment versus local alignment</h2><p><font color="blue">By default, Bowtie 2 performs <strong>end-to-end</strong> read alignment</font>. That is, it searches for alignments involving all of the read characters. This is also called an “untrimmed” or “unclipped” alignment.</p>
<p>When the <font color="blue"><strong>–local option</strong></font> is specified, Bowtie 2 performs local read alignment. In this mode, Bowtie 2 <font color="blue">might “trim” or “clip” some read characters from one or both ends of the alignment</font> if doing so maximizes the alignment score.</p>
<h3 id="End-to-end-alignment-example"><a href="#End-to-end-alignment-example" class="headerlink" title="End-to-end alignment example"></a>End-to-end alignment example</h3><p>The following is an “end-to-end” alignment because it involves all the characters in the read. Such an alignment can be produced by Bowtie 2 in either end-to-end mode or in local mode.</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line">Read:      GACTGGGCGATCTCGACTTCG</span><br><span class="line">Reference: GACTGCGATCTCGACATCG</span><br><span class="line"></span><br><span class="line">Alignment:</span><br><span class="line">  Read:      GACTGGGCGATCTCGACTTCG</span><br><span class="line">             |||||  |||||||||| |||</span><br><span class="line">  Reference: GACTG--CGATCTCGACATCG</span><br></pre></td></tr></table></figure>

<h3 id="Local-alignment-example"><a href="#Local-alignment-example" class="headerlink" title="Local alignment example"></a>Local alignment example</h3><p>The following is a “local” alignment because <font color="blue">some of the characters <strong>at the ends of the read do not participate</strong></font>. In this case, 4 characters are <font color="red"><strong>omitted</strong> (or “<strong>soft trimmed</strong>“ or “<strong>soft clipped</strong>“)</font> from the beginning and 3 characters are omitted from the end. This sort of alignment can be produced by Bowtie 2 <strong>only in local mode</strong>.</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line">Read:      ACGGTTGCGTTAATCCGCCACG</span><br><span class="line">Reference: TAACTTGCGTTAAATCCGCCTGG</span><br><span class="line"></span><br><span class="line">Alignment:</span><br><span class="line">  Read:      ACGGTTGCGTTAA-TCCGCCACG</span><br><span class="line">                 ||||||||| ||||||</span><br><span class="line">  Reference: TAACTTGCGTTAAATCCGCCTGG</span><br></pre></td></tr></table></figure>

<h2 id="Scores-higher-more-similar"><a href="#Scores-higher-more-similar" class="headerlink" title="Scores: higher = more similar"></a>Scores: higher = more similar</h2><p>An alignment score quantifies how similar the read sequence is to the reference sequence aligned to. The higher the score, the more similar they are. A score is <font color="blue"><strong>calculated</strong> by <strong>subtracting penalties</strong> for each difference (<strong>mismatch</strong>, <strong>gap</strong>, etc.)</font> and, in local alignment mode, adding bonuses for each match.</p>
<p>The scores can be configured with the <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-options-ma"><code>--ma</code></a> (match bonus), <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-options-mp"><code>--mp</code></a> (mismatch penalty), <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-options-np"><code>--np</code></a> (penalty for having an N in either the read or the reference), <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-options-rdg"><code>--rdg</code></a> (affine read gap penalty) and <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-options-rfg"><code>--rfg</code></a> (affine reference gap penalty) options.</p>
<h3 id="End-to-end-alignment-score-example-Default"><a href="#End-to-end-alignment-score-example-Default" class="headerlink" title="End-to-end alignment score example[Default]"></a>End-to-end alignment score example[Default]</h3><p>A <strong>mismatched</strong> base at a high-quality position in the read receives a penalty of <strong>-6</strong> by default. A length-2 read <strong>gap</strong> receives a penalty of -11 by default (<strong>-5 for the gap open</strong>, <strong>-3 for the first extension</strong>, -3 for the second extension). Thus, if 50 bp, one mismatch, one length-2 read gap, the overall score is -(6 + 11) = -<strong>17</strong>.</p>
<p>The <strong>best</strong> possible alignment <strong>score</strong> in end-to-end mode is <strong>0</strong>, which happens when there are <strong>no differences</strong> between the read and the reference. <font color="red"><strong>No match bonus</strong>. </font> </p>
<h3 id="Local-alignment-score-example"><a href="#Local-alignment-score-example" class="headerlink" title="Local alignment score example"></a>Local alignment score example</h3><p>A mismatched base at a high-quality position in the read receives a penalty of -6 by default. A length-2 read gap receives a penalty of -11 by default (-5 for the gap open, -3 for the first extension, -3 for the second extension). A base that <font color="red"><strong>matches receives a bonus of +2 be default</strong></font>. Thus, if 50 bp, one mismatch, one length-2 read gap, <strong>bonus, 2 * 49</strong>, minus the total penalty, 6 + 11, = <strong>81</strong>.</p>
<p>The best possible score in local mode equals the match bonus times the length of the read. This happens when there are no differences between the read and the reference.</p>
<h3 id="Valid-alignments-meet-or-exceed-the-minimum-score-threshold"><a href="#Valid-alignments-meet-or-exceed-the-minimum-score-threshold" class="headerlink" title="Valid alignments meet or exceed the minimum score threshold"></a><font color="red">Valid alignments meet or exceed the minimum score threshold</font></h3><p>For an alignment to be considered “valid” (i.e. “good enough”) by Bowtie 2, it must have an alignment score no less than the minimum score threshold. The threshold is configurable and is expressed as a function of the read length. In <font color="red"><strong>end-to-end</strong> alignment mode, the <strong>default minimum score threshold</strong></font> is <code>-0.6 + -0.6 * L</code>, where <code>L</code> is the read length. In <font color="red"><strong>local</strong> alignment mode, the <strong>default minimum score threshold</strong></font> is <code>20 + 8.0 * ln(L)</code>, where L is the read length. This can be configured with the <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-options-score-min"><code>--score-min</code></a> option. For details on how to set options like <code>--score-min</code> that correspond to functions, see the section on <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#setting-function-options">setting function options</a>.</p>
<h2 id="MAPQ-Mapping-quality-higher-more-unique"><a href="#MAPQ-Mapping-quality-higher-more-unique" class="headerlink" title="MAPQ,Mapping quality: higher = more unique"></a><font color="red">MAPQ</font>,Mapping quality: higher = more unique</h2><p>The aligner cannot always assign a read to its point of origin with high confidence. For instance, a read that originated inside a repeat element might align equally well to many occurrences of the element throughout the genome, leaving the aligner with no basis for preferring one over the others.</p>
<p>Aligners characterize their degree of confidence in the point of origin by reporting a mapping quality: a non-negative integer <font color="red"><strong>Q = -10 log10 p</strong></font>, where <font color="blue">p is an estimate of the probability that the alignment does not correspond to the read’s true point of origin</font>. Mapping quality is sometimes abbreviated <font color="red"><strong>MAPQ</strong></font>, and is recorded in the <a target="_blank" rel="noopener" href="http://samtools.sourceforge.net/SAM1.pdf">SAM</a> <code>MAPQ</code> field.</p>
<p>Mapping quality is related to “uniqueness.” We say an alignment is unique if it has a much higher alignment score than all the other possible alignments. The bigger the gap between the best alignment’s score and the second-best alignment’s score, the more unique the best alignment, and the higher its mapping quality should be.</p>
<p><strong>Accurate mapping qualities</strong> are <strong>useful</strong> for <strong>downstream tools</strong> like <strong>variant callers</strong>. For instance, a variant caller might choose to ignore evidence from alignments with mapping quality less than, say, 10. A mapping quality of 10 or less indicates that there is at least a 1 in 10 chance that the read truly originated elsewhere.</p>
<h2 id="Aligning-pairs"><a href="#Aligning-pairs" class="headerlink" title="Aligning pairs"></a>Aligning pairs</h2><p>A “paired-end” or “mate-pair” read consists of pair of mates, called mate 1 and mate 2. Pairs come with a prior expectation about (a) the relative orientation of the mates, and (b) the distance separating them on the original DNA molecule. Exactly what expectations hold for a given dataset depends on the lab procedures used to generate the data. For example, a common lab procedure for producing pairs is Illumina’s Paired-end Sequencing Assay, which yields pairs with a <font color="blue"><strong>relative orientation of FR (“forward, reverse”)</strong></font> meaning that if <font color="blue">mate 1 came from the <strong>Watson strand</strong></font>, <font color="blue">mate 2 very likely came from the <strong>Crick strand</strong></font> and vice versa. Also, this protocol yields pairs where the expected genomic distance from end to end is about <strong>200-500 base pairs</strong>.</p>
<p>For simplicity, this manual uses the term “paired-end” to refer to any pair of reads with some expected relative orientation and distance. Depending on the protocol, these might actually be referred to as “paired-end” or “mate-paired.” Also, we always refer to the individual sequences making up the pair as “mates.”</p>
<h3 id="Paired-inputs"><a href="#Paired-inputs" class="headerlink" title="Paired inputs"></a>Paired inputs</h3><p>Bowtie 2, mate 1s mates using the <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-options-1"><code>-1</code></a> ,  mate 2s using the <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-options-2"><code>-2</code></a> argument. </p>
<h3 id="Paired-SAM-output"><a href="#Paired-SAM-output" class="headerlink" title="Paired SAM output"></a>Paired SAM output</h3><p>When Bowtie 2 prints a SAM alignment for a pair, it <font color="blue"><strong>prints two records</strong> (i.e. two lines of output), one for each mate(1&amp;2)</font>. In both records, some of the fields of the SAM record describe various properties of the alignment; for instance, the <font color="blue">7th and 8th</font> fields (<code>RNEXT</code> and <code>PNEXT</code> respectively) indicate the <font color="blue">reference <strong>name</strong> and <strong>position</strong> where the other mate aligned</font>, and the <font color="blue"><strong>9th</strong> field indicates the inferred length of the <strong>DNA fragment</strong></font> from which the two mates were sequenced. See the <a target="_blank" rel="noopener" href="http://samtools.sourceforge.net/SAM1.pdf">SAM specification</a> for more details regarding these fields.</p>
<h3 id="Concordant-pairs-match-pair-expectations-discordant-pairs-don’t"><a href="#Concordant-pairs-match-pair-expectations-discordant-pairs-don’t" class="headerlink" title="Concordant pairs match pair expectations, discordant pairs don’t"></a><font color="red">Concordant pairs</font> match pair expectations, discordant pairs don’t</h3><p>A pair that aligns <font color="blue">with the <strong>expected relative mate orientation</strong> and with the <strong>expected range</strong> of distances between mates</font> is said to align “<font color="red"><strong>concordantly</strong></font>“. If both mates have unique alignments, but the alignments do not match paired-end expectations (i.e. the mates aren’t in the expected relative orientation, or aren’t within the expected distance range, or both), the pair is said to align “<font color="red"><strong>discordantly</strong></font>“. Discordant alignments may be of particular interest, for instance, when seeking <a target="_blank" rel="noopener" href="http://www.ncbi.nlm.nih.gov/dbvar/content/overview/"><font color="red"><strong>structural variants</strong></font></a>.</p>
<p>The expected relative orientation of the mates is set using the <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-options-fr"><code>--ff</code></a>, <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-options-fr"><code>--fr</code></a>, or <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-options-fr"><code>--rf</code></a> options. The <strong>expected range of inter-mates distances</strong> (as measured from the furthest extremes of the mates; also called “outer distance”) is set with the <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-options-I"><code>-I</code></a> and <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-options-X"><code>-X</code></a> options. Note that setting <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-options-I"><code>-I</code></a> and <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-options-X"><code>-X</code></a> far apart <font color="blue">makes Bowtie 2 slower</font>. See documentation for <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-options-I"><code>-I</code></a> and <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-options-X"><code>-X</code></a>.</p>
<p>To declare that a pair aligns discordantly, Bowtie 2 requires that both mates align uniquely. This is a conservative threshold, but this is often desirable when seeking structural variants.</p>
<p><font color="red">By default, Bowtie 2 searches for both concordant and discordant alignments</font>, though searching for discordant alignments can be <strong>disabled</strong> with the <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-options-no-discordant"><code>--no-discordant</code></a> option.</p>
<h3 id="Mixed-mode-paired-where-possible-unpaired-otherwise"><a href="#Mixed-mode-paired-where-possible-unpaired-otherwise" class="headerlink" title="Mixed mode: paired where possible, unpaired otherwise"></a>Mixed mode: paired where possible, unpaired otherwise</h3><p>If Bowtie 2 <font color="blue"><strong>cannot find</strong> a <strong>paired-end</strong> alignment for a <strong>pair</strong></font>, <font color="blue">by default it will go on to <strong>look for unpaired alignments</strong> for the constituent mates</font>. This is called “mixed mode.” To disable mixed mode, set the <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-options-no-mixed"><code>--no-mixed</code></a> option.</p>
<p>Bowtie 2 <strong>runs a little faster</strong> in <code>--no-mixed</code> mode, but will only consider alignment status of pairs per se, not individual mates.</p>
<h3 id="Some-SAM-FLAGS-describe-paired-end-properties"><a href="#Some-SAM-FLAGS-describe-paired-end-properties" class="headerlink" title="Some SAM FLAGS describe paired-end properties"></a>Some <font color="red">SAM FLAGS</font> describe <font color="red">paired-end properties</font></h3><p>The SAM <code>FLAGS</code> field, the <font color="red"><strong>second field</strong></font> in a SAM record, has multiple bits that describe the paired-end nature of the read and alignment. </p>
<p>The first (least significant) bit (1 in decimal, 0x1 in hexadecimal) is set if the read is part of a pair. </p>
<p>The second bit (2 in decimal, 0x2 in hexadecimal) is set if <font color="blue">the read is part of a pair that aligned in a paired-end fashion</font>. </p>
<p>The fourth bit (8 in decimal, 0x8 in hexadecimal) is set if the read is part of a pair and the other mate in the pair had at least one valid alignment. </p>
<p>The sixth bit (32 in decimal, 0x20 in hexadecimal) is set if the read is part of a pair and the other mate in the pair aligned to the Crick strand (or, equivalently, if the reverse complement of the other mate aligned to the Watson strand). </p>
<p>The seventh bit (64 in decimal, 0x40 in hexadecimal) is set if the read is mate 1 in a pair. </p>
<p>The eighth bit (128 in decimal, 0x80 in hexadecimal) is set if the read is mate 2 in a pair. </p>
<p>See the <a target="_blank" rel="noopener" href="http://samtools.sourceforge.net/SAM1.pdf">SAM specification</a> for a more detailed description of the <code>FLAGS</code> field.</p>
<h3 id="Some-SAM-optional-fields-describe-more-paired-end-properties"><a href="#Some-SAM-optional-fields-describe-more-paired-end-properties" class="headerlink" title="Some SAM optional fields describe more paired-end properties"></a>Some SAM optional fields describe more paired-end properties</h3><p>The last several fields of each SAM record usually contain SAM optional fields, which are simply tab-separated strings conveying additional information about the reads and alignments. A SAM optional field is formatted like this: “XP:i:1” where “XP” is the <code>TAG</code>, “i” is the <code>TYPE</code> (“integer” in this case), and “1” is the <code>VALUE</code>. See the <a target="_blank" rel="noopener" href="http://samtools.sourceforge.net/SAM1.pdf">SAM specification</a> for details regarding SAM optional fields.</p>
<h3 id="Mates-can-overlap-contain-or-dovetail-each-other"><a href="#Mates-can-overlap-contain-or-dovetail-each-other" class="headerlink" title="Mates can overlap, contain, or dovetail each other"></a>Mates can <font color="red">overlap, contain, or dovetail</font> each other</h3><p>The fragment and read lengths might be such that alignments for the two mates from a pair overlap each other. Consider this example:</p>
<p>(For these examples, assume we expect mate 1 to align to the left of mate 2.)</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">Mate 1:    GCAGATTATATGAGTCAGCTACGATATTGTT</span><br><span class="line">Mate 2:                               TGTTTGGGGTGACACATTACGCGTCTTTGAC</span><br><span class="line">Reference: GCAGATTATATGAGTCAGCTACGATATTGTTTGGGGTGACACATTACGCGTCTTTGAC</span><br></pre></td></tr></table></figure>

<p>It’s also possible, though unusual, for one mate alignment to contain the other, as in these examples:</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line">Mate 1:    GCAGATTATATGAGTCAGCTACGATATTGTTTGGGGTGACACATTACGC</span><br><span class="line">Mate 2:                               TGTTTGGGGTGACACATTACGC</span><br><span class="line">Reference: GCAGATTATATGAGTCAGCTACGATATTGTTTGGGGTGACACATTACGCGTCTTTGAC</span><br><span class="line"></span><br><span class="line">Mate 1:                   CAGCTACGATATTGTTTGGGGTGACACATTACGC</span><br><span class="line">Mate 2:                      CTACGATATTGTTTGGGGTGAC</span><br><span class="line">Reference: GCAGATTATATGAGTCAGCTACGATATTGTTTGGGGTGACACATTACGCGTCTTTGAC</span><br></pre></td></tr></table></figure>

<p>And it’s also possible, though unusual, for the mates to “dovetail”, with the mates seemingly extending “past” each other as in this example:</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">Mate 1:                 GTCAGCTACGATATTGTTTGGGGTGACACATTACGC</span><br><span class="line">Mate 2:            TATGAGTCAGCTACGATATTGTTTGGGGTGACACAT                   </span><br><span class="line">Reference: GCAGATTATATGAGTCAGCTACGATATTGTTTGGGGTGACACATTACGCGTCTTTGAC</span><br></pre></td></tr></table></figure>

<p>In some situations, it’s desirable for the aligner to consider all these cases as “concordant” as long as other paired-end constraints are not violated. <font color="red"><strong>Bowtie 2’s default</strong></font> behavior is to consider <font color="blue">overlapping and containing</font> as being consistent with <font color="blue">concordant</font> alignment. By default, <font color="blue">dovetailing is considered inconsistent with concordant alignment</font>.</p>
<p>These <font color="red"><strong>defaults can be overridden</strong></font>. Setting <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-options-no-overlap"><code>--no-overlap</code></a> causes Bowtie 2 to consider overlapping mates as non-concordant. Setting <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-options-no-contain"><code>--no-contain</code></a>causes Bowtie 2 to consider cases where one mate alignment contains the other as non-concordant. Setting <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-options-dovetail"><code>--dovetail</code></a> causes Bowtie 2 to consider cases where the mate alignments dovetail as concordant.</p>
<h2 id="Reporting"><a href="#Reporting" class="headerlink" title="Reporting"></a>Reporting</h2><p>The reporting mode governs how many alignments Bowtie 2 looks for, and how to report them. Bowtie 2 has three distinct reporting modes. The default reporting mode is similar to the default reporting mode of many other read alignment tools, including <a target="_blank" rel="noopener" href="http://bio-bwa.sourceforge.net/">BWA</a>. It is also similar to Bowtie 1’s <code>-M</code>alignment mode.</p>
<p>In general, when we say that a read has an alignment, we mean that it has a <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#valid-alignments-meet-or-exceed-the-minimum-score-threshold">valid alignment</a>. When we say that a read has multiple alignments, we mean that it has multiple alignments that are valid and distinct from one another.</p>
<h3 id="Distinct-alignments-map-a-read-to-different-places"><a href="#Distinct-alignments-map-a-read-to-different-places" class="headerlink" title="Distinct alignments map a read to different places"></a>Distinct alignments map a read to different places</h3><p>Two alignments for the same individual read are “distinct” if they map the same read to different places. Specifically, we say that two alignments are distinct if there are no alignment positions where a particular read offset is aligned opposite a particular reference offset in both alignments with the same orientation. E.g. if the first alignment is in the forward orientation and aligns the read character at read offset 10 to the reference character at chromosome 3, offset 3,445,245, and the second alignment is also in the forward orientation and also aligns the read character at read offset 10 to the reference character at chromosome 3, offset 3,445,245, they are not distinct alignments.</p>
<p>Two alignments for the same pair are distinct if either the mate 1s in the two paired-end alignments are distinct or the mate 2s in the two alignments are distinct or both.</p>
<h3 id="Default-mode-search-for-multiple-alignments-report-the-best-one"><a href="#Default-mode-search-for-multiple-alignments-report-the-best-one" class="headerlink" title="Default mode: search for multiple alignments, report the best one"></a><font color="red">Default mode</font>: search for multiple alignments, <font color="red">report the best one</font></h3><p>By default, Bowtie 2 searches for distinct, valid alignments for each read. When it finds a valid alignment, it generally will continue to look for alignments that are nearly as good or better. It will eventually stop looking, either because it <font color="red">exceeded a limit placed on search effort</font> (see <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-options-D"><code>-D</code></a> and <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-options-R"><code>-R</code></a>) or because it already knows all it needs to know to report an alignment. <font color="blue">Information from the best alignments are used to estimate mapping quality</font> (the <code>MAPQ</code> <a target="_blank" rel="noopener" href="http://samtools.sourceforge.net/SAM1.pdf">SAM</a> field) and to set SAM optional fields, such as <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-build-opt-fields-as"><code>AS:i</code></a> and <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-build-opt-fields-xs"><code>XS:i</code></a>. Bowtie 2 does not guarantee that the alignment reported is the best possible in terms of alignment score.</p>
<p>See also: <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-options-D"><code>-D</code></a>, which puts an <strong>upper limit</strong> on the number of dynamic programming problems (i.e. seed extensions) that can “fail” in a row before Bowtie 2 stops searching. Increasing <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-options-D"><code>-D</code></a> makes Bowtie 2 <strong>slower</strong>, but increases the likelihood that it will report the correct alignment for a read that aligns many places.</p>
<p>See also: <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-options-R"><code>-R</code></a>, which sets the maximum number of times Bowtie 2 will “re-seed” when attempting to align a read with repetitive seeds. Increasing <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-options-R"><code>-R</code></a>makes Bowtie 2 <strong>slower</strong>, but increases the likelihood that it will report the correct alignment for a read that aligns many places.</p>
<h3 id="k-mode-search-for-one-or-more-alignments-report-each"><a href="#k-mode-search-for-one-or-more-alignments-report-each" class="headerlink" title="-k mode: search for one or more alignments, report each"></a><font color="red">-k mode</font>: search for one or more alignments, report each</h3><p>In <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-options-k"><code>-k</code></a> mode, Bowtie 2 <font color="red">searches for up to N distinct</font>, valid alignments for each read, where N equals the integer specified with the <code>-k</code> parameter. That is, if <code>-k 2</code> is specified, Bowtie 2 will search for at most 2 distinct alignments. It reports all alignments found, in descending order by alignment score. The alignment score for a paired-end alignment equals the sum of the alignment scores of the individual mates. Each reported read or pair alignment beyond the first has the SAM ‘secondary’ bit (which equals 256) set in its FLAGS field. Supplementary alignments will also be assigned a MAPQ of 255. See the <a target="_blank" rel="noopener" href="http://samtools.sourceforge.net/SAM1.pdf">SAM specification</a> for details.</p>
<p>Bowtie 2 does not “find” alignments in any specific order, so for reads that have more than N distinct, valid alignments, Bowtie 2 does not guarantee that the N alignments reported are the best possible in terms of alignment score. Still, this mode can be effective and fast in situations where the user <font color="red">cares more about whether a read aligns (or aligns a certain number of times) than where exactly it originated</font>.</p>
<h3 id="a-mode-search-for-and-report-all-alignments"><a href="#a-mode-search-for-and-report-all-alignments" class="headerlink" title="-a mode: search for and report all alignments"></a>-a mode: search for and report all alignments</h3><p><a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-options-a"><code>-a</code></a> mode is similar to <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-options-k"><code>-k</code></a> mode except that there is <font color="red">no upper limit on the number of alignments Bowtie 2 should report</font>. Alignments are reported in descending order by alignment score. The alignment score for a paired-end alignment equals the sum of the alignment scores of the individual mates. Each reported read or pair alignment beyond the first has the SAM ‘secondary’ bit (which equals 256) set in its FLAGS field. Supplementary alignments will be assigned a MAPQ of 255. See the <a target="_blank" rel="noopener" href="http://samtools.sourceforge.net/SAM1.pdf">SAM specification</a> for details.</p>
<p>Some tools are designed with this reporting mode in mind. Bowtie 2 is not! For very large genomes, this mode is <strong>very slow</strong>.</p>
<h3 id="Randomness-in-Bowtie-2"><a href="#Randomness-in-Bowtie-2" class="headerlink" title="Randomness in Bowtie 2"></a>Randomness in Bowtie 2</h3><p>Bowtie 2’s search for alignments for a given read is “randomized.” That is, when Bowtie 2 encounters <font color="blue">a set of equally-good choices, it uses a <strong>pseudo-random number</strong> to choose?</font>. For example, if Bowtie 2 discovers a set of 3 equally-good alignments and wants to decide which to report, it picks a pseudo-random integer 0, 1 or 2 and reports the corresponding alignment. Arbitrary choices can crop up at various points during alignment.</p>
<p>The pseudo-random number generator is re-initialized for every read, and the seed used to initialize it is a function of the read name, nucleotide string, quality string, and the value specified with <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-options-seed"><code>--seed</code></a>. If you run the same version of Bowtie 2 on two reads with identical names, nucleotide strings, and quality strings, and if <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-options-seed"><code>--seed</code></a> is <font color="blue">set the same for both runs, Bowtie 2 will produce the same output</font>; i.e., it will align the read to the same place, even if there are multiple equally good alignments. This is intuitive and desirable in most cases. <font color="red">Most users expect Bowtie to produce the <strong>same output</strong> when run twice on the <strong>same input</strong></font>.</p>
<p>However, when the user specifies the <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-options-non-deterministic"><code>--non-deterministic</code></a> option, Bowtie 2 will use the current time to re-initialize the pseudo-random number generator. When this is specified, Bowtie 2 might report different alignments for identical reads. This is counter-intuitive for some users, but might be more appropriate in situations where the input consists of many identical reads.</p>
<h2 id="Multiseed-heuristic"><a href="#Multiseed-heuristic" class="headerlink" title="Multiseed heuristic"></a>Multiseed heuristic</h2><p>To rapidly narrow the number of possible alignments that must be considered, Bowtie 2 begins by extracting substrings (“seeds”) from the read and its reverse complement and aligning them in an ungapped fashion with the help of the <a target="_blank" rel="noopener" href="http://portal.acm.org/citation.cfm?id=796543">FM Index</a>. This is “multiseed alignment” and it is similar to what <a target="_blank" rel="noopener" href="http://genomebiology.com/2009/10/3/R25">Bowtie 1 does</a>, except Bowtie 1 attempts to align the entire read this way.</p>
<p>This initial step makes Bowtie 2 much faster than it would be without such a filter, but at the expense of missing some valid alignments. For instance, it is possible for a read to have a valid overall alignment but to have no valid seed alignments because each potential seed alignment is interrupted by too many mismatches or gaps.</p>
<p>The <font color="red">trade-off between <strong>speed</strong> and <strong>sensitivity/accuracy</strong> can be adjusted by setting the seed length</font> (<a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-options-L"><code>-L</code></a>), the interval between extracted seeds (<a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-options-i"><code>-i</code></a>), and the number of mismatches permitted per seed (<a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-options-N"><code>-N</code></a>). For more sensitive alignment, set these parameters to (a) make the seeds closer together, (b) make the seeds shorter, and/or (c) allow more mismatches. You can adjust these options one-by-one, though Bowtie 2 comes with some useful combinations of options prepackaged as “<a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#presets-setting-many-settings-at-once">preset options</a>.”</p>
<p><a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-options-D"><code>-D</code></a> and <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-options-R"><code>-R</code></a> are also options that adjust the trade-off between speed and sensitivity/accuracy.</p>
<h3 id="FM-Index-memory-footprint"><a href="#FM-Index-memory-footprint" class="headerlink" title="FM Index memory footprint"></a>FM Index memory footprint</h3><p>Bowtie 2 uses the <a target="_blank" rel="noopener" href="http://portal.acm.org/citation.cfm?id=796543">FM Index</a> to find ungapped alignments for seeds. This step accounts for the bulk of Bowtie 2’s memory footprint, as the <a target="_blank" rel="noopener" href="http://portal.acm.org/citation.cfm?id=796543">FM Index</a>itself is typically the largest data structure used. For instance, the memory footprint of the <a target="_blank" rel="noopener" href="http://portal.acm.org/citation.cfm?id=796543">FM Index</a> for the human genome is about 3.2 gigabytes of RAM.</p>
<h2 id="Ambiguous-characters"><a href="#Ambiguous-characters" class="headerlink" title="Ambiguous characters"></a>Ambiguous characters</h2><p>Non-whitespace characters besides A, C, G or T are considered “ambiguous.” N is a common ambiguous character that appears in reference sequences. Bowtie 2 considers all ambiguous characters in the reference (including <a target="_blank" rel="noopener" href="http://www.bioinformatics.org/sms/iupac.html">IUPAC nucleotide codes</a>) to be Ns.</p>
<p>Bowtie 2 <font color="red"><strong>allows</strong> alignments to overlap ambiguous characters in the reference</font>. An alignment position that contains an ambiguous character in the read, reference, or both, is penalized according to <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-options-np"><code>--np</code></a>. <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-options-n-ceil"><code>--n-ceil</code></a> sets an upper limit on the number of positions that may contain ambiguous reference characters in a valid alignment. The optional field <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-build-opt-fields-xn"><code>XN:i</code></a> reports the number of ambiguous reference characters overlapped by an alignment.</p>
<p>Note that the <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#multiseed-heuristic">multiseed heuristic</a> cannot find <em>seed</em> alignments that overlap ambiguous reference characters. For an alignment overlapping an ambiguous reference character to be found, it must have one or more seed alignments that do not overlap ambiguous reference characters.</p>
<h2 id="Presets-setting-many-settings-at-once"><a href="#Presets-setting-many-settings-at-once" class="headerlink" title="Presets: setting many settings at once"></a>Presets: setting many settings at once</h2><p>Bowtie 2 comes with some useful combinations of parameters packaged into shorter “preset” parameters. For example, running Bowtie 2 with the <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-options-very-sensitive"><code>--very-sensitive</code></a> option is the same as running with options: <code>-D 20 -R 3 -N 0 -L 20 -i S,1,0.50</code>. The preset options that come with Bowtie 2 are <font color="red">designed to cover a wide area of the speed/sensitivity/accuracy trade-off space</font>, with the presets ending in <code>fast</code> generally being faster but less sensitive and less accurate, and the presets ending in <code>sensitive</code> generally being slower but <strong>more sensitive and more accurate</strong>. See the <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#preset-options-in---end-to-end-mode">documentation for the preset options</a> for details.</p>
<p>As of Bowtie2 v2.4.0, individual preset values can be overridden by providing the specific options e.g. the configured seed length of 20 in the [<code>--very-senitive</code>] preset above can be changed to 25 by <strong>also specifying</strong> the <code>-L 25</code> parameter anywhere on the command line[masked -L 20].</p>
<h2 id="Filtering"><a href="#Filtering" class="headerlink" title="Filtering"></a>Filtering</h2><p>Some <font color="red"><strong>reads</strong></font> are skipped or “<font color="red"><strong>filtered out</strong></font>“ by Bowtie 2. For example, reads may be filtered out because they are extremely short or have a high proportion of ambiguous nucleotides. Bowtie 2 will still print a SAM record for such a read, but no alignment will be reported and the <code>YF:i</code> SAM optional field will be set to indicate the reason the read was filtered.</p>
<ul>
<li><code>YF:Z:LN</code>: the read was filtered because it had <font color="blue">length less than or equal</font> to the number of seed mismatches set with the <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-options-N"><code>-N</code></a> option.</li>
<li><code>YF:Z:NS</code>: the read was filtered because it <font color="blue">contains a number of ambiguous characters</font> (usually <code>N</code> or <code>.</code>) greater than the ceiling specified with <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-options-n-ceil"><code>--n-ceil</code></a>.</li>
<li><code>YF:Z:SC</code>: the read was filtered because the read length and the match bonus (set with <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-options-ma"><code>--ma</code></a>) are such that the read <font color="blue">can’t possibly earn an alignment score</font> greater than or equal to the threshold set with <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-options-score-min"><code>--score-min</code></a></li>
<li><code>YF:Z:QC</code>: the read was filtered because it was <font color="blue">marked as failing quality control</font> and the user specified the <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-options-qc-filter"><code>--qc-filter</code></a> option. This only happens when the input is in Illumina’s QSEQ format (i.e. when <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-options-qseq"><code>--qseq</code></a> is specified) and the last (11th) field of the read’s QSEQ record contains <code>1</code>.</li>
</ul>
<p>If a read could be filtered for more than one reason, the value <code>YF:Z</code> flag will reflect only one of those reasons.</p>
<h2 id="Alignment-summary"><a href="#Alignment-summary" class="headerlink" title="Alignment summary"></a>Alignment summary</h2><p>When Bowtie 2 finishes running, it prints messages summarizing what happened. These messages are printed to the “standard error” (“stderr”) filehandle. For datasets <font color="blue">consisting of unpaired reads</font>, the summary might look like this:</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line">20000 reads; of these:</span><br><span class="line">  20000 (100.00%) were unpaired; of these:</span><br><span class="line">    1247 (6.24%) aligned 0 times</span><br><span class="line">    18739 (93.69%) aligned exactly 1 time</span><br><span class="line">    14 (0.07%) aligned &gt;1 times</span><br><span class="line">93.77% overall alignment rate</span><br></pre></td></tr></table></figure>

<p>For datasets <font color="blue">consisting of pairs</font>, the summary might look like this:</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br></pre></td><td class="code"><pre><span class="line">10000 reads; of these:</span><br><span class="line">  10000 (100.00%) were paired; of these:</span><br><span class="line">    650 (6.50%) aligned concordantly 0 times</span><br><span class="line">    8823 (88.23%) aligned concordantly exactly 1 time</span><br><span class="line">    527 (5.27%) aligned concordantly &gt;1 times</span><br><span class="line">    ----</span><br><span class="line">    650 pairs aligned concordantly 0 times; of these:</span><br><span class="line">      34 (5.23%) aligned discordantly 1 time</span><br><span class="line">    ----</span><br><span class="line">    616 pairs aligned 0 times concordantly or discordantly; of these:</span><br><span class="line">      1232 mates make up the pairs; of these:</span><br><span class="line">        660 (53.57%) aligned 0 times</span><br><span class="line">        571 (46.35%) aligned exactly 1 time</span><br><span class="line">        1 (0.08%) aligned &gt;1 times</span><br><span class="line">96.70% overall alignment rate</span><br></pre></td></tr></table></figure>

<p>The indentation indicates how subtotals relate to totals.</p>
<h2 id="Wrapper-scripts"><a href="#Wrapper-scripts" class="headerlink" title="Wrapper scripts"></a>Wrapper scripts</h2><p>The <code>bowtie2</code>, <code>bowtie2-build</code> and <code>bowtie2-inspect</code> executables are actually wrapper scripts that call binary programs as appropriate. The wrappers shield users from having to distinguish between “small” and “large” index formats, discussed briefly in the following section. Also, the <code>bowtie2</code>wrapper provides some key functionality, like the ability to handle compressed inputs, and the functionality for <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-options-un"><code>--un</code></a>, <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-options-al"><code>--al</code></a> and related options.</p>
<p>It is recommended that you always run the bowtie2 wrappers and <strong>not run the binaries directly.</strong></p>
<h2 id="Small-and-large-indexes"><a href="#Small-and-large-indexes" class="headerlink" title="Small and large indexes"></a>Small and large indexes</h2><p><code>bowtie2-build</code> can index reference genomes of any size. For genomes less than about 4 billion nucleotides in length, <code>bowtie2-build</code> builds a “small” index using 32-bit numbers in various parts of the index. When the genome is longer, <code>bowtie2-build</code> builds a “large” index using 64-bit numbers. <font color="blue">Small indexes are stored in files with the <code>.bt2</code> extension, and large indexes are stored in files with the <code>.bt2l</code> extension. </font>The user need not worry about whether a particular index is small or large; the wrapper scripts will automatically build and use the appropriate index.</p>
<h2 id="Performance-tuning"><a href="#Performance-tuning" class="headerlink" title="Performance tuning"></a>Performance tuning</h2><ol>
<li><p>If your computer has multiple processors/cores, use <code>-p</code></p>
<p>The <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-options-p"><code>-p</code></a> option causes Bowtie 2 to launch a specified number of parallel search threads. Each thread runs on a different processor/core and all threads find alignments in parallel, increasing alignment throughput by approximately a multiple of the number of threads (though in practice, speedup is somewhat worse than linear).</p>
</li>
<li><p>If reporting many alignments per read, try reducing <code>bowtie2-build --offrate</code></p>
<p><font color="blue">If you are using <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-options-k"><code>-k</code></a> or <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-options-a"><code>-a</code></a> options and Bowtie 2 is reporting many alignments per read, using an index with a denser SA sample can speed things up considerably</font>. To do this, specify a smaller-than-default <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-options-o"><code>-o</code>/<code>--offrate</code></a> value when running <code>bowtie2-build</code>. A denser SA sample yields a larger index, but is also particularly effective at speeding up alignment when many alignments are reported per read.</p>
</li>
<li><p>If <code>bowtie2</code> “thrashes”, try increasing <code>bowtie2-build --offrate</code></p>
<p>If <code>bowtie2</code> runs very slowly on a relatively low-memory computer, try setting <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-options-o"><code>-o</code>/<code>--offrate</code></a> to a <em>larger</em> value when building the index. This decreases the memory footprint of the index.</p>
</li>
</ol>
<h2 id="Command-Line"><a href="#Command-Line" class="headerlink" title="Command Line"></a>Command Line</h2><h3 id="Alignment"><a href="#Alignment" class="headerlink" title="Alignment"></a><font color="red">Alignment</font></h3><p>Here are a few that may benefit the alignment of an <a target="_blank" rel="noopener" href="https://informatics.fas.harvard.edu/atac-seq-guidelines.html">ATAC-seq dataset on Cannon</a>:</p>
<table>
<thead>
<tr>
<th align="left">Argument</th>
<th align="left">Description</th>
</tr>
</thead>
<tbody><tr>
<td align="left"><code>-X &lt;int&gt;</code></td>
<td align="left"><strong>Maximum DNA fragment length</strong> (default 500bp). If you anticipate that you may have DNA fragments longer than the default value, you should increase this parameter accordingly; otherwise, alignments from such fragments are considered not properly paired (see Fig. 3B below).</td>
</tr>
<tr>
<td align="left"><code>--very-sensitive</code></td>
<td align="left">Bowtie2 has a number of alignment and effort parameters that interact in complex (and sometimes unexpected) ways. Preset collections of these parameters are provided for convenience; the default is <code>--sensitive</code>, but <strong>better alignment results</strong> are frequently achieved with <code>--very-sensitive</code>.</td>
</tr>
<tr>
<td align="left"><code>-k &lt;int&gt;</code></td>
<td align="left">Maximum number of <strong>alignments to report per read</strong>. By default, Bowtie2 reports at most one alignment per read, and if multiple equivalent alignments exist, it chooses one randomly. Generate huge output</td>
</tr>
<tr>
<td align="left"><code>-p &lt;int&gt;</code></td>
<td align="left">Number of cores on which to run</td>
</tr>
</tbody></table>
<p>The output is a <a target="_blank" rel="noopener" href="https://samtools.github.io/hts-specs/SAMv1.pdf">SAM file</a>, which contains alignment information for each input read. The SAM <strong>should be compressed to a binary format (BAM)</strong> and sorted by queryname with <a target="_blank" rel="noopener" href="http://www.htslib.org/doc/samtools.html">SAMtools</a>. This is best accomplished by piping the output from Bowtie2 directly to <code>samtools view</code> and <code>samtools sort</code>, </p>
<figure class="highlight sh"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line">module load bowtie2   <span class="comment"># if not already loaded</span></span><br><span class="line">module load samtools</span><br><span class="line">bowtie2  --very-sensitive  -k 10  -x &lt;genomeIndexName&gt;  -1 &lt;name&gt;_1.fastq.gz  -2 &lt;name&gt;_2.fastq.gz  \</span><br><span class="line">  |  samtools view  -u  -  \</span><br><span class="line">  |  samtools sort  -n  -o &lt;BAM&gt;  -</span><br></pre></td></tr></table></figure>

<p>For input files of <strong>20 million paired reads</strong>, this command takes around <strong>five hours</strong> on Cannon. This can be decreased by increasing the number of cores in the Bowtie2 command. For example, one could specify eight cores for Bowtie2 with <code>-p 8</code> and adjust the request in the SLURM script to <code>#SBATCH -n 10</code>(that is, <strong>eight cores for Bowtie2 and one each for SAMtools view and sort, simultaneously processing</strong>). The memory usage of Bowtie2 depends primarily on the genome length; enough must be requested to load the genome indexes.</p>
<p>Bowtie2 also provides (via <code>stderr</code>) a summary of the mapping results, separated according to uniqueness and alignment type (concordant, discordant, and non-concordant/non-discordant). In terms of alignment interpretation, it is conceptually easier to divide alignments into two <strong>basic categories: properly paired and unpaired</strong> (Fig. 3).</p>
<p><img src="/blog/2022/03/20/2022-03-20-bowtie2/1.png" alt="Alignment types"></p>
<p><strong>Figure 3. Alignment types for paired-end reads. A:</strong> Properly paired alignments (“concordant”) have the reads aligned in opposite orientations on the same reference sequence (chromosome). The reads may overlap to some extent (bottom).  <strong>B:</strong> A read alignment (for R1) can be unpaired for several reasons: if the read’s mate (R2) is unaligned (upper left), aligns to a different chromosome (upper right), aligns in the incorrect orientation (middle cases), or aligns in the correct orientation but at an invalid distance (bottom). In all cases except the upper left, the R2 read alignment is also unpaired, and the read pair align discordantly (though Bowtie2 also requires uniqueness for such alignments to be counted as discordant).</p>
<h3 id="Usage"><a href="#Usage" class="headerlink" title="Usage"></a>Usage</h3><figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">bowtie2 [options]* -x &lt;bt2-idx&gt; &#123;-1 &lt;m1&gt; -2 &lt;m2&gt; | -U &lt;r&gt; | --interleaved &lt;i&gt; | --sra-acc &lt;acc&gt; | b &lt;bam&gt;&#125; -S [&lt;sam&gt;]</span><br></pre></td></tr></table></figure>

<h3 id="Main-arguments"><a href="#Main-arguments" class="headerlink" title="Main arguments"></a>Main arguments</h3><table>
<thead>
<tr>
<th><code>-x &lt;bt2-idx&gt;</code></th>
<th>The basename of the index for the reference genome.</th>
</tr>
</thead>
<tbody><tr>
<td><code>-1 &lt;m1&gt;</code></td>
<td>Comma-separated list of files containing mate 1s (filename usually includes <code>_1</code>), e.g. <code>-1 flyA_1.fq,flyB_1.fq</code>.</td>
</tr>
<tr>
<td><code>-2 &lt;m2&gt;</code></td>
<td>Comma-separated list of files containing mate 2s (filename usually includes <code>_2</code>), e.g. <code>-2 flyA_2.fq,flyB_2.fq</code>.</td>
</tr>
<tr>
<td><code>-U &lt;r&gt;</code></td>
<td>Comma-separated list of files containing unpaired reads to be aligned, e.g. <code>lane1.fq,lane2.fq,lane3.fq,lane4.fq</code>. Reads may be a mix of different lengths. If <code>-</code> is specified, <code>bowtie2</code> gets the reads from the “standard in” or “stdin” filehandle.</td>
</tr>
<tr>
<td><code>--interleaved</code></td>
<td>Reads interleaved FASTQ files where the first two records (8 lines) represent a mate pair.</td>
</tr>
<tr>
<td><code>--sra-acc</code></td>
<td>Reads are SRA accessions.</td>
</tr>
<tr>
<td><code>-b &lt;bam&gt;</code></td>
<td>Reads are unaligned BAM records sorted by read name. The <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-options-align-paired-reads"><code>--align-paired-reads</code></a> and <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-options-preserve-tags"><code>--preserve-tags</code></a> options affect the way Bowtie 2 processes records.</td>
</tr>
<tr>
<td><code>-S &lt;sam&gt;</code></td>
<td>File to write SAM alignments to. By default, alignments are written to the “standard out” or “stdout” filehandle (i.e. the console).</td>
</tr>
</tbody></table>
<h3 id="Options"><a href="#Options" class="headerlink" title="Options"></a>Options</h3><h4 id="Input-options"><a href="#Input-options" class="headerlink" title="Input options"></a>Input options</h4><table>
<thead>
<tr>
<th><code>-q</code></th>
<th>Reads (specified with <code>&lt;m1&gt;</code>, <code>&lt;m2&gt;</code>, <code>&lt;s&gt;</code>) are FASTQ files. FASTQ files usually have extension <code>.fq</code> or <code>.fastq</code>. FASTQ is the default format. See also: <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-options-solexa-quals"><code>--solexa-quals</code></a> and <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-options-int-quals"><code>--int-quals</code></a>.</th>
</tr>
</thead>
<tbody><tr>
<td><code>--tab5</code></td>
<td>Each read or pair is on a single line. An unpaired read line is <code>[name]\t[seq]\t[qual]\n</code>. A paired-end read line is <code>[name]\t[seq1]\t[qual1]\t[seq2]\t[qual2]\n</code>. An input file can be a mix of unpaired and paired-end reads and Bowtie 2 recognizes each according to the number of fields, handling each as it should.</td>
</tr>
<tr>
<td><code>--tab6</code></td>
<td>Similar to <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-options-tab5"><code>--tab5</code></a> except, for paired-end reads, the second end can have a different name from the first: <code>[name1]\t[seq1]\t[qual1]\t[name2]\t[seq2]\t[qual2]\n</code></td>
</tr>
<tr>
<td><code>--qseq</code></td>
<td>Reads (specified with <code>&lt;m1&gt;</code>, <code>&lt;m2&gt;</code>, <code>&lt;s&gt;</code>) are QSEQ files. QSEQ files usually end in <code>_qseq.txt</code>. See also: <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-options-solexa-quals"><code>--solexa-quals</code></a> and <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-options-int-quals"><code>--int-quals</code></a>.</td>
</tr>
<tr>
<td><code>-f</code></td>
<td>Reads (specified with <code>&lt;m1&gt;</code>, <code>&lt;m2&gt;</code>, <code>&lt;s&gt;</code>) are <a target="_blank" rel="noopener" href="https://en.wikipedia.org/wiki/FASTA"><code>FASTA</code></a> files. <a target="_blank" rel="noopener" href="https://en.wikipedia.org/wiki/FASTA"><code>FASTA</code></a> files usually have extension <code>.fa</code>, <code>.fasta</code>, <code>.mfa</code>, <code>.fna</code>or similar. <a target="_blank" rel="noopener" href="https://en.wikipedia.org/wiki/FASTA"><code>FASTA</code></a> files do not have a way of specifying quality values, so when <code>-f</code> is set, the result is as if <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-options-ignore-quals"><code>--ignore-quals</code></a> is also set.</td>
</tr>
<tr>
<td><code>-r</code></td>
<td>Reads (specified with <code>&lt;m1&gt;</code>, <code>&lt;m2&gt;</code>, <code>&lt;s&gt;</code>) are files with one input sequence per line, without any other information (no read names, no qualities). When <code>-r</code> is set, the result is as if <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-options-ignore-quals"><code>--ignore-quals</code></a> is also set.</td>
</tr>
<tr>
<td><code>-F k:&lt;int&gt;,i:&lt;int&gt;</code></td>
<td>Reads are substrings (k-mers) extracted from a FASTA file <code>&lt;s&gt;</code>. Specifically, for every reference sequence in FASTA file <code>&lt;s&gt;</code>, Bowtie 2 aligns the k-mers at offsets 1, 1+i, 1+2i, … until reaching the end of the reference. Each k-mer is aligned as a separate read. Quality values are set to all Is (40 on Phred scale). Each k-mer (read) is given a name like <code>&lt;sequence&gt;_&lt;offset&gt;</code>, where <code>&lt;sequence&gt;</code> is the name of the FASTA sequence it was drawn from and <code>&lt;offset&gt;</code> is its 0-based offset of origin with respect to the sequence. Only single k-mers, i.e. unpaired reads, can be aligned in this way.</td>
</tr>
<tr>
<td><code>-c</code></td>
<td>The read sequences are given on command line. I.e. <code>&lt;m1&gt;</code>, <code>&lt;m2&gt;</code> and <code>&lt;singles&gt;</code> are comma-separated lists of reads rather than lists of read files. There is no way to specify read names or qualities, so <code>-c</code> also implies <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-options-ignore-quals"><code>--ignore-quals</code></a>.</td>
</tr>
<tr>
<td><code>-s/--skip &lt;int&gt;</code></td>
<td><strong>Skip (i.e. do not align) the first <code>&lt;int&gt;</code> reads or pairs in the input.</strong></td>
</tr>
<tr>
<td><code>-u/--qupto &lt;int&gt;</code></td>
<td><strong>Align the first <code>&lt;int&gt;</code> reads or read pairs from the input</strong> (after the <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-options-s"><code>-s</code>/<code>--skip</code></a> reads or pairs have been skipped), then stop. Default: no limit.</td>
</tr>
<tr>
<td><code>-5/--trim5 &lt;int&gt;</code></td>
<td><strong>Trim <code>&lt;int&gt;</code> bases from 5’ (left) end</strong> of each read before alignment (default: <strong>0</strong>).</td>
</tr>
<tr>
<td><code>-3/--trim3 &lt;int&gt;</code></td>
<td><strong>Trim <code>&lt;int&gt;</code> bases from 3’ (right) end</strong> of each read before alignment (default: <strong>0</strong>).</td>
</tr>
<tr>
<td>`–trim-to [3:</td>
<td>5:]<int>`</int></td>
</tr>
<tr>
<td><code>--phred33</code></td>
<td>Input qualities are ASCII chars equal to the <a target="_blank" rel="noopener" href="http://en.wikipedia.org/wiki/Phred_quality_score">Phred quality</a> plus 33. This is also called the “Phred+33” encoding, which is <strong>used by the very latest Illumina pipelines.</strong></td>
</tr>
<tr>
<td><code>--phred64</code></td>
<td>Input qualities are ASCII chars equal to the <a target="_blank" rel="noopener" href="http://en.wikipedia.org/wiki/Phred_quality_score">Phred quality</a> plus 64. This is also called the “Phred+64” encoding.</td>
</tr>
<tr>
<td><code>--solexa-quals</code></td>
<td>Convert input qualities from <a target="_blank" rel="noopener" href="http://en.wikipedia.org/wiki/Phred_quality_score">Solexa</a> (which can be negative) to <a target="_blank" rel="noopener" href="http://en.wikipedia.org/wiki/Phred_quality_score">Phred</a> (which can’t). This scheme was used in older Illumina GA Pipeline versions (prior to 1.3). <strong>Default: off.</strong></td>
</tr>
<tr>
<td><code>--int-quals</code></td>
<td>Quality values are represented in the read input file as space-separated ASCII integers, e.g., <code>40 40 30 40</code>…, rather than ASCII characters, e.g., <code>II?I</code>…. Integers are treated as being on the <a target="_blank" rel="noopener" href="http://en.wikipedia.org/wiki/Phred_quality_score">Phred quality</a> scale unless <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-options-solexa-quals"><code>--solexa-quals</code></a> is also specified. <strong>Default: off.</strong></td>
</tr>
</tbody></table>
<h4 id="Preset-options-in-end-to-end-mode"><a href="#Preset-options-in-end-to-end-mode" class="headerlink" title="Preset options in --end-to-end mode"></a>Preset options in <code>--end-to-end</code> mode</h4><table>
<thead>
<tr>
<th><code>--very-fast</code></th>
<th>Same as: <code>-D 5 -R 1 -N 0 -L 22 -i S,0,2.50</code></th>
</tr>
</thead>
<tbody><tr>
<td><code>--fast</code></td>
<td>Same as: <code>-D 10 -R 2 -N 0 -L 22 -i S,0,2.50</code></td>
</tr>
<tr>
<td><code>--sensitive</code></td>
<td>Same as: <code>-D 15 -R 2 -N 0 -L 22 -i S,1,1.15</code> (default in <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-options-end-to-end"><code>--end-to-end</code></a> mode)</td>
</tr>
<tr>
<td><code>--very-sensitive</code></td>
<td>Same as: <code>-D 20 -R 3 -N 0 -L 20 -i S,1,0.50</code></td>
</tr>
</tbody></table>
<h4 id="Preset-options-in-local-mode"><a href="#Preset-options-in-local-mode" class="headerlink" title="Preset options in --local mode"></a>Preset options in <code>--local</code> mode</h4><table>
<thead>
<tr>
<th><code>--very-fast-local</code></th>
<th>Same as: <code>-D 5 -R 1 -N 0 -L 25 -i S,1,2.00</code></th>
</tr>
</thead>
<tbody><tr>
<td><code>--fast-local</code></td>
<td>Same as: <code>-D 10 -R 2 -N 0 -L 22 -i S,1,1.75</code></td>
</tr>
<tr>
<td><code>--sensitive-local</code></td>
<td>Same as: <code>-D 15 -R 2 -N 0 -L 20 -i S,1,0.75</code> (default in <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-options-local"><code>--local</code></a> mode)</td>
</tr>
<tr>
<td><code>--very-sensitive-local</code></td>
<td>Same as: <code>-D 20 -R 3 -N 0 -L 20 -i S,1,0.50</code></td>
</tr>
</tbody></table>
<h4 id="Alignment-options"><a href="#Alignment-options" class="headerlink" title="Alignment options"></a>Alignment options</h4><table>
<thead>
<tr>
<th><code>-N &lt;int&gt;</code></th>
<th>Sets the number of mismatches to allowed in a seed alignment during <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#multiseed-heuristic">multiseed alignment</a>. Can be set to 0 or 1. Setting this higher makes alignment slower (often much slower) but increases sensitivity. Default: 0.</th>
</tr>
</thead>
<tbody><tr>
<td><code>-L &lt;int&gt;</code></td>
<td><strong>Sets the length of the seed substrings to align</strong> during <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#multiseed-heuristic">multiseed alignment</a>. <font color="red"><strong>Smaller values make alignment slower but more sensitive</strong></font>. Default: the <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-options-sensitive"><code>--sensitive</code></a> preset is used by default, which sets <code>-L</code> to 22 and 20 in <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-options-end-to-end"><code>--end-to-end</code></a> mode and in <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-options-local"><code>--local</code></a> mode.</td>
</tr>
<tr>
<td><code>-i &lt;func&gt;</code></td>
<td>Sets a function governing the interval between seed substrings to use during <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#multiseed-heuristic">multiseed alignment</a>. For instance, if the read has 30 characters, and seed length is 10, and the seed interval is 6, the seeds extracted will be:<code>Read:      TAGCTACGCTCTACGCTATCATGCATAAAC Seed 1 fw: TAGCTACGCT Seed 1 rc: AGCGTAGCTA Seed 2 fw:       CGCTCTACGC Seed 2 rc:       GCGTAGAGCG Seed 3 fw:             ACGCTATCAT Seed 3 rc:             ATGATAGCGT Seed 4 fw:                   TCATGCATAA Seed 4 rc:                   TTATGCATGA</code>Since it’s best to use longer intervals for longer reads, this parameter sets the interval as a function of the read length, rather than a single one-size-fits-all number. For instance, specifying <code>-i S,1,2.5</code> sets the interval function <code>f</code> to <code>f(x) = 1 + 2.5 * sqrt(x)</code>, where x is the read length. See also: <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#setting-function-options">setting function options</a>. If the function returns a result less than 1, it is rounded up to 1. Default: the <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-options-sensitive"><code>--sensitive</code></a> preset is used by default, which sets <code>-i</code> to <code>S,1,1.15</code> in <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-options-end-to-end"><code>--end-to-end</code></a>mode to <code>-i S,1,0.75</code> in <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-options-local"><code>--local</code></a> mode.</td>
</tr>
<tr>
<td><code>--n-ceil &lt;func&gt;</code></td>
<td>Sets a function governing the maximum number of ambiguous characters (usually <code>N</code>s and/or <code>.</code>s) allowed in a read as a function of read length. For instance, specifying <code>-L,0,0.15</code> sets the N-ceiling function <code>f</code> to <code>f(x) = 0 + 0.15 * x</code>, where x is the read length. See also: <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#setting-function-options">setting function options</a>. Reads exceeding this ceiling are <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#filtering">filtered out</a>. Default: <code>L,0,0.15</code>.</td>
</tr>
<tr>
<td><code>--dpad &lt;int&gt;</code></td>
<td>“Pads” dynamic programming problems by <code>&lt;int&gt;</code> columns on either side to allow gaps. Default: 15.</td>
</tr>
<tr>
<td><code>--gbar &lt;int&gt;</code></td>
<td>Disallow gaps within <code>&lt;int&gt;</code> positions of the beginning or end of the read. Default: 4.</td>
</tr>
<tr>
<td><code>--ignore-quals</code></td>
<td>When calculating a mismatch penalty, always consider the quality value at the mismatched position to be the highest possible, regardless of the actual value. I.e. input is treated as though all quality values are high. This is also the default behavior when the input doesn’t specify quality values (e.g. in <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-options-f"><code>-f</code></a>, <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-options-r"><code>-r</code></a>, or <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-options-c"><code>-c</code></a> modes).</td>
</tr>
<tr>
<td><code>--nofw/--norc</code></td>
<td>If <code>--nofw</code> is specified, <code>bowtie2</code> will not attempt to align unpaired reads to the forward (Watson) reference strand. If <code>--norc</code> is specified, <code>bowtie2</code> will not attempt to align unpaired reads against the reverse-complement (Crick) reference strand. In paired-end mode, <code>--nofw</code> and <code>--norc</code> pertain to the fragments; i.e. specifying <code>--nofw</code> causes <code>bowtie2</code> to explore only those paired-end configurations corresponding to fragments from the reverse-complement (Crick) strand. Default: both strands enabled.</td>
</tr>
<tr>
<td><code>--no-1mm-upfront</code></td>
<td>By default, Bowtie 2 will attempt to find either an exact or a 1-mismatch end-to-end alignment for the read <em>before</em> trying the <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#multiseed-heuristic">multiseed heuristic</a>. Such alignments can be found very quickly, and many short read alignments have exact or near-exact end-to-end alignments. However, this can lead to unexpected alignments when the user also sets options governing the <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#multiseed-heuristic">multiseed heuristic</a>, like <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-options-L"><code>-L</code></a> and <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-options-N"><code>-N</code></a>. For instance, if the user specifies <code>-N 0</code> and <code>-L</code> equal to the length of the read, the user will be surprised to find 1-mismatch alignments reported. This option prevents Bowtie 2 from searching for 1-mismatch end-to-end alignments before using the <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#multiseed-heuristic">multiseed heuristic</a>, which leads to the expected behavior when combined with options such as <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-options-L"><code>-L</code></a> and <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-options-N"><code>-N</code></a>. This comes at the expense of speed.</td>
</tr>
<tr>
<td><code>--end-to-end</code></td>
<td>In this mode, Bowtie 2 requires that the entire read align from one end to the other, without any trimming (or “soft clipping”) of characters from either end. The match bonus <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-options-ma"><code>--ma</code></a> always equals 0 in this mode, so all alignment scores are less than or equal to 0, and the greatest possible alignment score is 0. This is mutually exclusive with <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-options-local"><code>--local</code></a>. <code>--end-to-end</code> is the default mode.</td>
</tr>
<tr>
<td><code>--local</code></td>
<td>In this mode, Bowtie 2 does not require that the entire read align from one end to the other. Rather, some characters may be omitted (“soft clipped”) from the ends in order to achieve the greatest possible alignment score. The match bonus <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-options-ma"><code>--ma</code></a>is used in this mode, and the best possible alignment score is equal to the match bonus (<a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-options-ma"><code>--ma</code></a>) times the length of the read. Specifying <code>--local</code> and one of the presets (e.g. <code>--local --very-fast</code>) is equivalent to specifying the local version of the preset (<code>--very-fast-local</code>). This is mutually exclusive with <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-options-end-to-end"><code>--end-to-end</code></a>. <code>--end-to-end</code> is the default mode.</td>
</tr>
</tbody></table>
<h4 id="Scoring-options"><a href="#Scoring-options" class="headerlink" title="Scoring options"></a>Scoring options</h4><table>
<thead>
<tr>
<th><code>--ma &lt;int&gt;</code></th>
<th>Sets the match bonus. In <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-options-local"><code>--local</code></a> mode <code>&lt;int&gt;</code> is added to the alignment score for each position where a read character aligns to a reference character and the characters match. Not used in <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-options-end-to-end"><code>--end-to-end</code></a> mode. Default: 2.</th>
</tr>
</thead>
<tbody><tr>
<td><code>--mp MX,MN</code></td>
<td>Sets the maximum (<code>MX</code>) and minimum (<code>MN</code>) mismatch penalties, both integers. A number less than or equal to <code>MX</code> and greater than or equal to <code>MN</code> is subtracted from the alignment score for each position where a read character aligns to a reference character, the characters do not match, and neither is an <code>N</code>. If <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-options-ignore-quals"><code>--ignore-quals</code></a> is specified, the number subtracted quals <code>MX</code>. Otherwise, the number subtracted is <code>MN + floor( (MX-MN)(MIN(Q, 40.0)/40.0) )</code> where Q is the Phred quality value. Default: <code>MX</code> = 6, <code>MN</code> = 2.</td>
</tr>
<tr>
<td><code>--np &lt;int&gt;</code></td>
<td>Sets penalty for positions where the read, reference, or both, contain an ambiguous character such as <code>N</code>. Default: 1.</td>
</tr>
<tr>
<td><code>--rdg &lt;int1&gt;,&lt;int2&gt;</code></td>
<td>Sets the read gap open (<code>&lt;int1&gt;</code>) and extend (<code>&lt;int2&gt;</code>) penalties. A read gap of length N gets a penalty of <code>&lt;int1&gt;</code> + N * <code>&lt;int2&gt;</code>. Default: 5, 3.</td>
</tr>
<tr>
<td><code>--rfg &lt;int1&gt;,&lt;int2&gt;</code></td>
<td>Sets the reference gap open (<code>&lt;int1&gt;</code>) and extend (<code>&lt;int2&gt;</code>) penalties. A reference gap of length N gets a penalty of <code>&lt;int1&gt;</code> + N * <code>&lt;int2&gt;</code>. Default: 5, 3.</td>
</tr>
<tr>
<td><code>--score-min &lt;func&gt;</code></td>
<td>Sets a function governing the minimum alignment score needed for an alignment to be considered “valid” (i.e. good enough to report). This is a function of read length. For instance, specifying <code>L,0,-0.6</code> sets the minimum-score function <code>f</code> to <code>f(x) = 0 + -0.6 * x</code>, where <code>x</code> is the read length. See also: <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#setting-function-options">setting function options</a>. The default in <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-options-end-to-end"><code>--end-to-end</code></a>mode is <code>L,-0.6,-0.6</code> and the default in <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-options-local"><code>--local</code></a> mode is <code>G,20,8</code>.</td>
</tr>
</tbody></table>
<h4 id="Reporting-options"><a href="#Reporting-options" class="headerlink" title="Reporting options"></a>Reporting options</h4><table>
<thead>
<tr>
<th><code>-k &lt;int&gt;</code></th>
<th>By default, <code>bowtie2</code> searches for distinct, valid alignments for each read. When it finds a valid alignment, it continues looking for alignments that are nearly as good or better. The best alignment found is reported (randomly selected from among best if tied). Information about the best alignments is used to estimate mapping quality and to set SAM optional fields, such as <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-build-opt-fields-as"><code>AS:i</code></a> and <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-build-opt-fields-xs"><code>XS:i</code></a>.When <code>-k</code> is specified, however, <code>bowtie2</code> behaves differently. Instead, it searches for at most <code>&lt;int&gt;</code> distinct, valid alignments for each read. The search terminates when it can’t find more distinct valid alignments, or when it finds <code>&lt;int&gt;</code>, whichever happens first. All alignments found are reported in descending order by alignment score. The alignment score for a paired-end alignment equals the sum of the alignment scores of the individual mates. Each reported read or pair alignment beyond the first has the SAM ‘secondary’ bit (which equals 256) set in its FLAGS field. For reads that have more than <code>&lt;int&gt;</code> distinct, valid alignments, <code>bowtie2</code>does not guarantee that the <code>&lt;int&gt;</code> alignments reported are the best possible in terms of alignment score. <code>-k</code> is mutually exclusive with <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-options-a"><code>-a</code></a>.Note: Bowtie 2 is not designed with large values for <code>-k</code> in mind, and when aligning reads to long, repetitive genomes large <code>-k</code> can be very, very slow.</th>
</tr>
</thead>
<tbody><tr>
<td><code>-a</code></td>
<td>Like <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-options-k"><code>-k</code></a> but with no upper limit on number of alignments to search for. <code>-a</code> is mutually exclusive with <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-options-k"><code>-k</code></a>.Note: Bowtie 2 is not designed with <code>-a</code> mode in mind, and when aligning reads to long, repetitive genomes this mode can be very, very slow.</td>
</tr>
</tbody></table>
<h4 id="Effort-options"><a href="#Effort-options" class="headerlink" title="Effort options"></a>Effort options</h4><table>
<thead>
<tr>
<th><code>-D &lt;int&gt;</code></th>
<th>Up to <code>&lt;int&gt;</code> consecutive seed extension attempts can “fail” before Bowtie 2 moves on, using the alignments found so far. A seed extension “fails” if it does not yield a new best or a new second-best alignment. This limit is automatically adjusted up when -k or -a are specified. Default: 15.</th>
</tr>
</thead>
<tbody><tr>
<td><code>-R &lt;int&gt;</code></td>
<td><code>&lt;int&gt;</code> is the maximum number of times Bowtie 2 will “re-seed” reads with repetitive seeds. When “re-seeding,” Bowtie 2 simply chooses a new set of reads (same length, same number of mismatches allowed) at different offsets and searches for more alignments. A read is considered to have repetitive seeds if the total number of seed hits divided by the number of seeds that aligned at least once is greater than 300. Default: 2.</td>
</tr>
</tbody></table>
<h4 id="Paired-end-options"><a href="#Paired-end-options" class="headerlink" title="Paired-end options"></a>Paired-end options</h4><table>
<thead>
<tr>
<th><code>-I/--minins &lt;int&gt;</code></th>
<th>The minimum fragment length for valid paired-end alignments. E.g. if <code>-I 60</code> is specified and a paired-end alignment consists of two 20-bp alignments in the appropriate orientation with a 20-bp gap between them, that alignment is considered valid (as long as <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-options-X"><code>-X</code></a> is also satisfied). A 19-bp gap would not be valid in that case. If trimming options <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-options-3"><code>-3</code></a> or <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-options-5"><code>-5</code></a>are also used, the <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-options-I"><code>-I</code></a> constraint is applied with respect to the untrimmed mates.The larger the difference between <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-options-I"><code>-I</code></a> and <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-options-X"><code>-X</code></a>, the slower Bowtie 2 will run. This is because larger differences between <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-options-I"><code>-I</code></a>and <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-options-X"><code>-X</code></a> require that Bowtie 2 scan a larger window to determine if a concordant alignment exists. For typical fragment length ranges (200 to 400 nucleotides), Bowtie 2 is very efficient.Default: 0 (essentially imposing no minimum)</th>
</tr>
</thead>
<tbody><tr>
<td><code>-X/--maxins &lt;int&gt;</code></td>
<td>The maximum fragment length for valid paired-end alignments. E.g. if <code>-X 100</code> is specified and a paired-end alignment consists of two 20-bp alignments in the proper orientation with a 60-bp gap between them, that alignment is considered valid (as long as <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-options-I"><code>-I</code></a> is also satisfied). A 61-bp gap would not be valid in that case. If trimming options <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-options-3"><code>-3</code></a> or <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-options-5"><code>-5</code></a> are also used, the <code>-X</code> constraint is applied with respect to the untrimmed mates, not the trimmed mates.The larger the difference between <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-options-I"><code>-I</code></a> and <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-options-X"><code>-X</code></a>, the slower Bowtie 2 will run. This is because larger differences between <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-options-I"><code>-I</code></a>and <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-options-X"><code>-X</code></a> require that Bowtie 2 scan a larger window to determine if a concordant alignment exists. For typical fragment length ranges (200 to 400 nucleotides), Bowtie 2 is very efficient.Default: 500.</td>
</tr>
<tr>
<td><code>--fr/--rf/--ff</code></td>
<td>The upstream/downstream mate orientations for a valid paired-end alignment against the forward reference strand. E.g., if <code>--fr</code> is specified and there is a candidate paired-end alignment where mate 1 appears upstream of the reverse complement of mate 2 and the fragment length constraints (<a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-options-I"><code>-I</code></a> and <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-options-X"><code>-X</code></a>) are met, that alignment is valid. Also, if mate 2 appears upstream of the reverse complement of mate 1 and all other constraints are met, that too is valid. <code>--rf</code> likewise requires that an upstream mate1 be reverse-complemented and a downstream mate2 be forward-oriented. <code>--ff</code> requires both an upstream mate 1 and a downstream mate 2 to be forward-oriented. Default: <code>--fr</code> (appropriate for Illumina’s Paired-end Sequencing Assay).</td>
</tr>
<tr>
<td><code>--no-mixed</code></td>
<td>By default, when <code>bowtie2</code> cannot find a concordant or discordant alignment for a pair, it then tries to find alignments for the individual mates. This option disables that behavior.</td>
</tr>
<tr>
<td><code>--no-discordant</code></td>
<td>By default, <code>bowtie2</code> looks for discordant alignments if it cannot find any concordant alignments. A discordant alignment is an alignment where both mates align uniquely, but that does not satisfy the paired-end constraints (<a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-options-fr"><code>--fr</code>/<code>--rf</code>/<code>--ff</code></a>, <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-options-I"><code>-I</code></a>, <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-options-X"><code>-X</code></a>). This option disables that behavior.</td>
</tr>
<tr>
<td><code>--dovetail</code></td>
<td>If the mates “dovetail”, that is if one mate alignment extends past the beginning of the other such that the wrong mate begins upstream, consider that to be concordant. See also: <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#mates-can-overlap-contain-or-dovetail-each-other">Mates can overlap, contain or dovetail each other</a>. Default: mates cannot dovetail in a concordant alignment.</td>
</tr>
<tr>
<td><code>--no-contain</code></td>
<td>If one mate alignment contains the other, consider that to be non-concordant. See also: <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#mates-can-overlap-contain-or-dovetail-each-other">Mates can overlap, contain or dovetail each other</a>. Default: a mate can contain the other in a concordant alignment.</td>
</tr>
<tr>
<td><code>--no-overlap</code></td>
<td>If one mate alignment overlaps the other at all, consider that to be non-concordant. See also: <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#mates-can-overlap-contain-or-dovetail-each-other">Mates can overlap, contain or dovetail each other</a>. Default: mates can overlap in a concordant alignment.</td>
</tr>
</tbody></table>
<h4 id="BAM-options"><a href="#BAM-options" class="headerlink" title="BAM options"></a>BAM options</h4><table>
<thead>
<tr>
<th><code>--align-paired-reads</code></th>
<th>Bowtie 2 will, by default, attempt to align unpaired BAM reads. Use this option to align paired-end reads instead.</th>
</tr>
</thead>
<tbody><tr>
<td><code>--preserve-tags</code></td>
<td>Preserve tags from the original BAM record by appending them to the end of the corresponding Bowtie 2 SAM output.</td>
</tr>
</tbody></table>
<h4 id="Output-options"><a href="#Output-options" class="headerlink" title="Output options"></a>Output options</h4><table>
<thead>
<tr>
<th><code>-t/--time</code></th>
<th>Print the wall-clock time required to load the index files and align the reads. This is printed to the “standard error” (“stderr”) filehandle. Default: off.</th>
</tr>
</thead>
<tbody><tr>
<td><code>--un &lt;path&gt; --un-gz &lt;path&gt; --un-bz2 &lt;path&gt; --un-lz4 &lt;path&gt;</code></td>
<td>Write unpaired reads that fail to align to file at <code>&lt;path&gt;</code>. These reads correspond to the SAM records with the FLAGS <code>0x4</code> bit set and neither the <code>0x40</code> nor <code>0x80</code> bits set. If <code>--un-gz</code> is specified, output will be gzip compressed. If <code>--un-bz2</code>or <code>--un-lz4</code> is specified, output will be bzip2 or lz4 compressed. Reads written in this way will appear exactly as they did in the input file, without any modification (same sequence, same name, same quality string, same quality encoding). Reads will not necessarily appear in the same order as they did in the input.</td>
</tr>
<tr>
<td><code>--al &lt;path&gt; --al-gz &lt;path&gt; --al-bz2 &lt;path&gt; --al-lz4 &lt;path&gt;</code></td>
<td>Write unpaired reads that align at least once to file at <code>&lt;path&gt;</code>. These reads correspond to the SAM records with the FLAGS <code>0x4</code>, <code>0x40</code>, and <code>0x80</code> bits unset. If <code>--al-gz</code> is specified, output will be gzip compressed. If <code>--al-bz2</code> is specified, output will be bzip2 compressed. Similarly if <code>--al-lz4</code> is specified, output will be lz4 compressed. Reads written in this way will appear exactly as they did in the input file, without any modification (same sequence, same name, same quality string, same quality encoding). Reads will not necessarily appear in the same order as they did in the input.</td>
</tr>
<tr>
<td><code>--un-conc &lt;path&gt; --un-conc-gz &lt;path&gt; --un-conc-bz2 &lt;path&gt; --un-conc-lz4 &lt;path&gt;</code></td>
<td>Write paired-end reads that fail to align concordantly to file(s) at <code>&lt;path&gt;</code>. These reads correspond to the SAM records with the FLAGS <code>0x4</code> bit set and either the <code>0x40</code> or <code>0x80</code> bit set (depending on whether it’s mate #1 or #2). <code>.1</code> and <code>.2</code>strings are added to the filename to distinguish which file contains mate #1 and mate #2. If a percent symbol, <code>%</code>, is used in <code>&lt;path&gt;</code>, the percent symbol is replaced with <code>1</code> or <code>2</code> to make the per-mate filenames. Otherwise, <code>.1</code> or <code>.2</code> are added before the final dot in <code>&lt;path&gt;</code> to make the per-mate filenames. Reads written in this way will appear exactly as they did in the input files, without any modification (same sequence, same name, same quality string, same quality encoding). Reads will not necessarily appear in the same order as they did in the inputs.</td>
</tr>
<tr>
<td><code>--al-conc &lt;path&gt; --al-conc-gz &lt;path&gt; --al-conc-bz2 &lt;path&gt; --al-conc-lz4 &lt;path&gt;</code></td>
<td>Write paired-end reads that align concordantly at least once to file(s) at <code>&lt;path&gt;</code>. These reads correspond to the SAM records with the FLAGS <code>0x4</code> bit unset and either the <code>0x40</code> or <code>0x80</code> bit set (depending on whether it’s mate #1 or #2). <code>.1</code> and <code>.2</code> strings are added to the filename to distinguish which file contains mate #1 and mate #2. If a percent symbol, <code>%</code>, is used in <code>&lt;path&gt;</code>, the percent symbol is replaced with <code>1</code> or <code>2</code> to make the per-mate filenames. Otherwise, <code>.1</code> or <code>.2</code> are added before the final dot in <code>&lt;path&gt;</code> to make the per-mate filenames. Reads written in this way will appear exactly as they did in the input files, without any modification (same sequence, same name, same quality string, same quality encoding). Reads will not necessarily appear in the same order as they did in the inputs.</td>
</tr>
<tr>
<td><code>--quiet</code></td>
<td>Print nothing besides alignments and serious errors.</td>
</tr>
<tr>
<td><code>--met-file &lt;path&gt;</code></td>
<td>Write <code>bowtie2</code> metrics to file <code>&lt;path&gt;</code>. Having alignment metric can be useful for debugging certain problems, especially performance issues. See also: <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-options-met"><code>--met</code></a>. Default: metrics disabled.</td>
</tr>
<tr>
<td><code>--met-stderr &lt;path&gt;</code></td>
<td>Write <code>bowtie2</code> metrics to the “standard error” (“stderr”) filehandle. This is not mutually exclusive with <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-options-met-file"><code>--met-file</code></a>. Having alignment metric can be useful for debugging certain problems, especially performance issues. See also: <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-options-met"><code>--met</code></a>. Default: metrics disabled.</td>
</tr>
<tr>
<td><code>--met &lt;int&gt;</code></td>
<td>Write a new <code>bowtie2</code> metrics record every <code>&lt;int&gt;</code> seconds. Only matters if either <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-options-met-stderr"><code>--met-stderr</code></a> or <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-options-met-file"><code>--met-file</code></a> are specified. Default: 1.</td>
</tr>
</tbody></table>
<h4 id="SAM-options"><a href="#SAM-options" class="headerlink" title="SAM options"></a>SAM options</h4><table>
<thead>
<tr>
<th><code>--no-unal</code></th>
<th>Suppress SAM records for reads that failed to align.</th>
</tr>
</thead>
<tbody><tr>
<td><code>--no-hd</code></td>
<td>Suppress SAM header lines (starting with <code>@</code>).</td>
</tr>
<tr>
<td><code>--no-sq</code></td>
<td>Suppress <code>@SQ</code> SAM header lines.</td>
</tr>
<tr>
<td><code>--rg-id &lt;text&gt;</code></td>
<td>Set the read group ID to <code>&lt;text&gt;</code>. This causes the SAM <code>@RG</code> header line to be printed, with <code>&lt;text&gt;</code> as the value associated with the <code>ID:</code> tag. It also causes the <code>RG:Z:</code> extra field to be attached to each SAM output record, with value set to <code>&lt;text&gt;</code>.</td>
</tr>
<tr>
<td><code>--rg &lt;text&gt;</code></td>
<td>Add <code>&lt;text&gt;</code> (usually of the form <code>TAG:VAL</code>, e.g. <code>SM:Pool1</code>) as a field on the <code>@RG</code> header line. Note: in order for the <code>@RG</code> line to appear, <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-options-rg-id"><code>--rg-id</code></a> must also be specified. This is because the <code>ID</code> tag is required by the <a target="_blank" rel="noopener" href="http://samtools.sourceforge.net/SAM1.pdf">SAM Spec</a>. Specify <code>--rg</code> multiple times to set multiple fields. See the <a target="_blank" rel="noopener" href="http://samtools.sourceforge.net/SAM1.pdf">SAM Spec</a> for details about what fields are legal.</td>
</tr>
<tr>
<td><code>--omit-sec-seq</code></td>
<td>When printing secondary alignments, Bowtie 2 by default will write out the <code>SEQ</code> and <code>QUAL</code> strings. Specifying this option causes Bowtie 2 to print an asterisk in those fields instead.</td>
</tr>
<tr>
<td><code>--soft-clipped-unmapped-tlen</code></td>
<td>Consider soft-clipped bases unmapped when calculating <code>TLEN</code>. Only available in <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-options-local"><code>--local</code></a> mode.</td>
</tr>
<tr>
<td><code>--sam-no-qname-trunc</code></td>
<td>Suppress standard behavior of truncating readname at first whitespace at the expense of generating non-standard SAM</td>
</tr>
<tr>
<td><code>--xeq</code></td>
<td>Use <code>&#39;=&#39;/&#39;X&#39;</code>, instead of <code>&#39;M&#39;</code>, to specify matches/mismatches in SAM record</td>
</tr>
<tr>
<td><code>--sam-append-comment</code></td>
<td>Append FASTA/FASTQ comment to SAM record, where a comment is everything after the first space in the read name.</td>
</tr>
</tbody></table>
<h4 id="Performance-options"><a href="#Performance-options" class="headerlink" title="Performance options"></a>Performance options</h4><table>
<thead>
<tr>
<th><code>-o/--offrate &lt;int&gt;</code></th>
<th>Override the offrate of the index with <code>&lt;int&gt;</code>. If <code>&lt;int&gt;</code> is greater than the offrate used to build the index, then some row markings are discarded when the index is read into memory. This reduces the memory footprint of the aligner but requires more time to calculate text offsets. <code>&lt;int&gt;</code> must be greater than the value used to build the index.</th>
</tr>
</thead>
<tbody><tr>
<td><code>-p/--threads NTHREADS</code></td>
<td>Launch <code>NTHREADS</code> parallel search threads (default: 1). Threads will run on separate processors/cores and synchronize when parsing reads and outputting alignments. Searching for alignments is highly parallel, and speedup is close to linear. Increasing <code>-p</code> increases Bowtie 2’s memory footprint. E.g. when aligning to a human genome index, increasing <code>-p</code> from 1 to 8 increases the memory footprint by a few hundred megabytes. This option is only available if <code>bowtie</code> is linked with the <code>pthreads</code> library (i.e. if <code>BOWTIE_PTHREADS=0</code> is not specified at build time).</td>
</tr>
<tr>
<td><code>--reorder</code></td>
<td>Guarantees that output SAM records are printed in an order corresponding to the order of the reads in the original input file, even when <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-options-p"><code>-p</code></a> is set greater than 1. Specifying <code>--reorder</code> and setting <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-options-p"><code>-p</code></a> greater than 1 causes Bowtie 2 to run somewhat slower and use somewhat more memory than if <code>--reorder</code> were not specified. Has no effect if <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-options-p"><code>-p</code></a> is set to 1, since output order will naturally correspond to input order in that case.</td>
</tr>
<tr>
<td><code>--mm</code></td>
<td>Use memory-mapped I/O to load the index, rather than typical file I/O. Memory-mapping allows many concurrent <code>bowtie</code> processes on the same computer to share the same memory image of the index (i.e. you pay the memory overhead just once). This facilitates memory-efficient parallelization of <code>bowtie</code> in situations where using <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-options-p"><code>-p</code></a> is not possible or not preferable.</td>
</tr>
</tbody></table>
<h4 id="Other-options"><a href="#Other-options" class="headerlink" title="Other options"></a>Other options</h4><table>
<thead>
<tr>
<th><code>--qc-filter</code></th>
<th>Filter out reads for which the QSEQ filter field is non-zero. Only has an effect when read format is <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-options-qseq"><code>--qseq</code></a>. Default: off.</th>
</tr>
</thead>
<tbody><tr>
<td><code>--seed &lt;int&gt;</code></td>
<td>Use <code>&lt;int&gt;</code> as the seed for pseudo-random number generator. Default: 0.</td>
</tr>
<tr>
<td><code>--non-deterministic</code></td>
<td>Normally, Bowtie 2 re-initializes its pseudo-random generator for each read. It seeds the generator with a number derived from (a) the read name, (b) the nucleotide sequence, (c) the quality sequence, (d) the value of the <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-options-seed"><code>--seed</code></a>option. This means that if two reads are identical (same name, same nucleotides, same qualities) Bowtie 2 will find and report the same alignment(s) for both, even if there was ambiguity. When <code>--non-deterministic</code> is specified, Bowtie 2 re-initializes its pseudo-random generator for each read using the current time. This means that Bowtie 2 will not necessarily report the same alignment for two identical reads. This is counter-intuitive for some users, but might be more appropriate in situations where the input consists of many identical reads.</td>
</tr>
<tr>
<td><code>--version</code></td>
<td>Print version information and quit.</td>
</tr>
<tr>
<td><code>-h/--help</code></td>
<td>Print usage information and quit.</td>
</tr>
</tbody></table>
<h2 id="SAM-output"><a href="#SAM-output" class="headerlink" title="SAM output"></a>SAM output</h2><p>Following is a brief description of the <a target="_blank" rel="noopener" href="http://samtools.sourceforge.net/SAM1.pdf">SAM</a> format as output by <code>bowtie2</code>. For more details, see the <a target="_blank" rel="noopener" href="http://samtools.sourceforge.net/SAM1.pdf">SAM format specification</a>.</p>
<p>By default, <code>bowtie2</code> prints a SAM header with <code>@HD</code>, <code>@SQ</code> and <code>@PG</code> lines. When one or more <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-options-rg"><code>--rg</code></a> arguments are specified, <code>bowtie2</code> will also print an <code>@RG</code> line that includes all user-specified <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-options-rg"><code>--rg</code></a> tokens separated by tabs.</p>
<p>Each subsequent line describes an alignment or, if the read failed to align, a read. Each line is a collection of at least 12 fields separated by tabs; from left to right, the fields are:</p>
<ol>
<li><p>Name of read that aligned.</p>
<p>Note that the <a target="_blank" rel="noopener" href="http://samtools.sourceforge.net/SAM1.pdf">SAM specification</a> disallows whitespace in the read name. If the read name contains any whitespace characters, Bowtie 2 will truncate the name at the first whitespace character. This is similar to the behavior of other tools. The standard behavior of truncating at the first whitespace can be suppressed with <code>--sam-no-qname-trunc</code> at the expense of generating non-standard SAM.</p>
</li>
<li><p><font color="red">Sum of all applicable flags. Flags relevant to Bowtie are:</font></p>
<table>
<thead>
<tr>
<th><code>1</code></th>
<th>The read is one of a pair</th>
</tr>
</thead>
<tbody><tr>
<td><code>2</code></td>
<td>The alignment is one end of a proper paired-end alignment</td>
</tr>
<tr>
<td><code>4</code></td>
<td>The read has no reported alignments</td>
</tr>
<tr>
<td><code>8</code></td>
<td>The read is one of a pair and has no reported alignments</td>
</tr>
<tr>
<td><code>16</code></td>
<td>The alignment is to the reverse reference strand</td>
</tr>
<tr>
<td><code>32</code></td>
<td>The other mate in the paired-end alignment is aligned to the reverse reference strand</td>
</tr>
<tr>
<td><code>64</code></td>
<td>The read is mate 1 in a pair</td>
</tr>
<tr>
<td><code>128</code></td>
<td>The read is mate 2 in a pair</td>
</tr>
</tbody></table>
<p>Thus, an unpaired read that aligns to the reverse reference strand will have flag 16. A paired-end read that aligns and is the first mate in the pair will have flag 83 (= 64 + 16 + 2 + 1).</p>
</li>
<li><p>Name of reference sequence where alignment occurs</p>
</li>
<li><p>1-based offset into the forward reference strand where leftmost character of the alignment occurs</p>
</li>
<li><p>Mapping quality</p>
</li>
<li><p>CIGAR string representation of alignment</p>
</li>
<li><p>Name of reference sequence where mate’s alignment occurs. Set to <code>=</code> if the mate’s reference sequence is the same as this alignment’s, or <code>*</code>if there is no mate.</p>
</li>
<li><p>1-based offset into the forward reference strand where leftmost character of the mate’s alignment occurs. Offset is 0 if there is no mate.</p>
</li>
<li><p>Inferred fragment length. Size is negative if the mate’s alignment occurs upstream of this alignment. Size is 0 if the mates did not align concordantly. However, size is non-0 if the mates aligned discordantly to the same chromosome.</p>
</li>
<li><p>Read sequence (reverse-complemented if aligned to the reverse strand)</p>
</li>
<li><p>ASCII-encoded read qualities (reverse-complemented if the read aligned to the reverse strand). The encoded quality values are on the <a target="_blank" rel="noopener" href="http://en.wikipedia.org/wiki/Phred_quality_score">Phred quality</a> scale and the encoding is ASCII-offset by 33 (ASCII char <code>!</code>), similarly to a <a target="_blank" rel="noopener" href="http://en.wikipedia.org/wiki/FASTQ_format">FASTQ</a> file.</p>
</li>
<li><p>Optional fields. Fields are tab-separated. <code>bowtie2</code> outputs zero or more of these optional fields for each alignment, depending on the type of the alignment:</p>
</li>
</ol>
<table>
<thead>
<tr>
<th><code>AS:i:&lt;N&gt;</code></th>
<th>Alignment score. Can be negative. Can be greater than 0 in <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-options-local"><code>--local</code></a> mode (but not in <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-options-end-to-end"><code>--end-to-end</code></a> mode). Only present if SAM record is for an aligned read.</th>
</tr>
</thead>
<tbody><tr>
<td><code>XS:i:&lt;N&gt;</code></td>
<td>Alignment score for the best-scoring alignment found other than the alignment reported. Can be negative. Can be greater than 0 in <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-options-local"><code>--local</code></a> mode (but not in <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-options-end-to-end"><code>--end-to-end</code></a> mode). Only present if the SAM record is for an aligned read and more than one alignment was found for the read. Note that, when the read is part of a concordantly-aligned pair, this score could be greater than <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-build-opt-fields-as"><code>AS:i</code></a>.</td>
</tr>
<tr>
<td><code>YS:i:&lt;N&gt;</code></td>
<td>Alignment score for opposite mate in the paired-end alignment. Only present if the SAM record is for a read that aligned as part of a paired-end alignment.</td>
</tr>
<tr>
<td><code>XN:i:&lt;N&gt;</code></td>
<td>The number of ambiguous bases in the reference covering this alignment. Only present if SAM record is for an aligned read.</td>
</tr>
<tr>
<td><code>XM:i:&lt;N&gt;</code></td>
<td>The number of mismatches in the alignment. Only present if SAM record is for an aligned read.</td>
</tr>
<tr>
<td><code>XO:i:&lt;N&gt;</code></td>
<td>The number of gap opens, for both read and reference gaps, in the alignment. Only present if SAM record is for an aligned read.</td>
</tr>
<tr>
<td><code>XG:i:&lt;N&gt;</code></td>
<td>The number of gap extensions, for both read and reference gaps, in the alignment. Only present if SAM record is for an aligned read.</td>
</tr>
<tr>
<td><code>NM:i:&lt;N&gt;</code></td>
<td>The edit distance; that is, the minimal number of one-nucleotide edits (substitutions, insertions and deletions) needed to transform the read string into the reference string. Only present if SAM record is for an aligned read.</td>
</tr>
<tr>
<td><code>YF:Z:&lt;S&gt;</code></td>
<td>String indicating reason why the read was filtered out. See also: <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#filtering">Filtering</a>. Only appears for reads that were filtered out.</td>
</tr>
<tr>
<td><code>YT:Z:&lt;S&gt;</code></td>
<td>Value of <code>UU</code> indicates the read was not part of a pair. Value of <code>CP</code> indicates the read was part of a pair and the pair aligned concordantly. Value of <code>DP</code> indicates the read was part of a pair and the pair aligned discordantly. Value of <code>UP</code> indicates the read was part of a pair but the pair failed to aligned either concordantly or discordantly.</td>
</tr>
<tr>
<td><code>MD:Z:&lt;S&gt;</code></td>
<td>A string representation of the mismatched reference bases in the alignment. See <a target="_blank" rel="noopener" href="https://samtools.github.io/hts-specs/SAMtags.pdf">SAM Tags format specification</a> for details. Only present if SAM record is for an aligned read.</td>
</tr>
</tbody></table>
<h1 id="The-bowtie2-build-indexer"><a href="#The-bowtie2-build-indexer" class="headerlink" title="The bowtie2-build indexer"></a>The <code>bowtie2-build</code> indexer</h1><p><code>bowtie2-build</code> builds a Bowtie index from a set of DNA sequences. <code>bowtie2-build</code> outputs a set of 6 files with suffixes <code>.1.bt2</code>, <code>.2.bt2</code>, <code>.3.bt2</code>, <code>.4.bt2</code>, <code>.rev.1.bt2</code>, and <code>.rev.2.bt2</code>. In the case of a large index these suffixes will have a <code>bt2l</code> termination. These files together constitute the index: they are all that is needed to align reads to that reference. The original sequence <a target="_blank" rel="noopener" href="https://en.wikipedia.org/wiki/FASTA"><code>FASTA</code></a> files are no longer used by Bowtie 2 once the index is built.</p>
<p>Bowtie 2’s <code>.bt2</code> index format is different from Bowtie 1’s <code>.ebwt</code> format, and they are not compatible with each other.</p>
<p>Use of Karkkainen’s <a target="_blank" rel="noopener" href="http://portal.acm.org/citation.cfm?id=1314852">blockwise algorithm</a> allows <code>bowtie2-build</code> to trade off between running time and memory usage. <code>bowtie2-build</code> has three options governing how it makes this trade: <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-build-options-p"><code>-p</code>/<code>--packed</code></a>, <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-build-options-bmax"><code>--bmax</code></a>/<a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-build-options-bmaxdivn"><code>--bmaxdivn</code></a>, and <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-build-options-dcv"><code>--dcv</code></a>. By default, <code>bowtie2-build</code> will automatically search for the settings that yield the best running time without exhausting memory. This behavior can be disabled using the <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-build-options-a"><code>-a</code>/<code>--noauto</code></a> option.</p>
<p>The indexer provides options pertaining to the “shape” of the index, e.g. <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-build-options-o"><code>--offrate</code></a> governs the fraction of <a target="_blank" rel="noopener" href="http://en.wikipedia.org/wiki/Burrows-Wheeler_transform">Burrows-Wheeler</a> rows that are “marked” (i.e., the density of the suffix-array sample; see the original <a target="_blank" rel="noopener" href="http://portal.acm.org/citation.cfm?id=796543">FM Index</a> paper for details). All of these options are potentially profitable trade-offs depending on the application. They have been set to defaults that are reasonable for most cases according to our experiments. See <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#performance-tuning">Performance tuning</a> for details.</p>
<p><code>bowtie2-build</code> can generate either <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#small-and-large-indexes">small or large indexes</a>. The wrapper will decide which based on the length of the input genome. If the reference does not exceed 4 billion characters but a large index is preferred, the user can specify <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-build-options-large-index"><code>--large-index</code></a> to force <code>bowtie2-build</code> to build a large index instead.</p>
<p>The Bowtie 2 index is based on the <a target="_blank" rel="noopener" href="http://portal.acm.org/citation.cfm?id=796543">FM Index</a> of Ferragina and Manzini, which in turn is based on the <a target="_blank" rel="noopener" href="http://en.wikipedia.org/wiki/Burrows-Wheeler_transform">Burrows-Wheeler</a> transform. The algorithm used to build the index is based on the <a target="_blank" rel="noopener" href="http://portal.acm.org/citation.cfm?id=1314852">blockwise algorithm</a> of Karkkainen.</p>
<h2 id="Command-Line-1"><a href="#Command-Line-1" class="headerlink" title="Command Line"></a>Command Line</h2><p>Usage:</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">bowtie2-build [options]* &lt;reference_in&gt; &lt;bt2_base&gt;</span><br></pre></td></tr></table></figure>

<h3 id="Main-arguments-1"><a href="#Main-arguments-1" class="headerlink" title="Main arguments"></a>Main arguments</h3><table>
<thead>
<tr>
<th><code>&lt;reference_in&gt;</code></th>
<th>A comma-separated list of <a target="_blank" rel="noopener" href="https://en.wikipedia.org/wiki/FASTA"><code>FASTA</code></a> files containing the reference sequences to be aligned to, or, if <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-build-options-c"><code>-c</code></a> is specified, the sequences themselves. E.g., <code>&lt;reference_in&gt;</code> might be <code>chr1.fa,chr2.fa,chrX.fa,chrY.fa</code>, or, if <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-build-options-c"><code>-c</code></a> is specified, this might be <code>GGTCATCCT,ACGGGTCGT,CCGTTCTATGCGGCTTA</code>.</th>
</tr>
</thead>
<tbody><tr>
<td><code>&lt;bt2_base&gt;</code></td>
<td>The basename of the index files to write. By default, <code>bowtie2-build</code> writes files named <code>NAME.1.bt2</code>, <code>NAME.2.bt2</code>, <code>NAME.3.bt2</code>, <code>NAME.4.bt2</code>, <code>NAME.rev.1.bt2</code>, and <code>NAME.rev.2.bt2</code>, where <code>NAME</code> is <code>&lt;bt2_base&gt;</code>.</td>
</tr>
</tbody></table>
<h3 id="Options-1"><a href="#Options-1" class="headerlink" title="Options"></a>Options</h3><table>
<thead>
<tr>
<th><code>-f</code></th>
<th>The reference input files (specified as <code>&lt;reference_in&gt;</code>) are <a target="_blank" rel="noopener" href="https://en.wikipedia.org/wiki/FASTA"><code>FASTA</code></a> files (usually having extension <code>.fa</code>, <code>.mfa</code>, <code>.fna</code> or similar).</th>
</tr>
</thead>
<tbody><tr>
<td><code>-c</code></td>
<td>The reference sequences are given on the command line. I.e. <code>&lt;reference_in&gt;</code> is a comma-separated list of sequences rather than a list of <a target="_blank" rel="noopener" href="https://en.wikipedia.org/wiki/FASTA"><code>FASTA</code></a> files.</td>
</tr>
<tr>
<td><code>--large-index</code></td>
<td>Force <code>bowtie2-build</code> to build a <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#small-and-large-indexes">large index</a>, even if the reference is less than ~ 4 billion nucleotides inlong.</td>
</tr>
<tr>
<td><code>-a/--noauto</code></td>
<td>Disable the default behavior whereby <code>bowtie2-build</code> automatically selects values for the <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-build-options-bmax"><code>--bmax</code></a>, <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-build-options-dcv"><code>--dcv</code></a> and <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-build-options-p"><code>--packed</code></a> parameters according to available memory. Instead, user may specify values for those parameters. If memory is exhausted during indexing, an error message will be printed; it is up to the user to try new parameters.</td>
</tr>
<tr>
<td><code>-p/--packed</code></td>
<td>Use a packed (2-bits-per-nucleotide) representation for DNA strings. This saves memory but makes indexing 2-3 times slower. Default: off. This is configured automatically by default; use <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-build-options-a"><code>-a</code>/<code>--noauto</code></a> to configure manually.</td>
</tr>
<tr>
<td><code>--bmax &lt;int&gt;</code></td>
<td>The maximum number of suffixes allowed in a block. Allowing more suffixes per block makes indexing faster, but increases peak memory usage. Setting this option overrides any previous setting for <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-build-options-bmax"><code>--bmax</code></a>, or <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-build-options-bmaxdivn"><code>--bmaxdivn</code></a>. Default (in terms of the <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-build-options-bmaxdivn"><code>--bmaxdivn</code></a> parameter) is <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-build-options-bmaxdivn"><code>--bmaxdivn</code></a> 4 * number of threads. This is configured automatically by default; use <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-build-options-a"><code>-a</code>/<code>--noauto</code></a> to configure manually.</td>
</tr>
<tr>
<td><code>--bmaxdivn &lt;int&gt;</code></td>
<td>The maximum number of suffixes allowed in a block, expressed as a fraction of the length of the reference. Setting this option overrides any previous setting for <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-build-options-bmax"><code>--bmax</code></a>, or <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-build-options-bmaxdivn"><code>--bmaxdivn</code></a>. Default: <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-build-options-bmaxdivn"><code>--bmaxdivn</code></a> 4 * number of threads. This is configured automatically by default; use <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-build-options-a"><code>-a</code>/<code>--noauto</code></a> to configure manually.</td>
</tr>
<tr>
<td><code>--dcv &lt;int&gt;</code></td>
<td>Use <code>&lt;int&gt;</code> as the period for the difference-cover sample. A larger period yields less memory overhead, but may make suffix sorting slower, especially if repeats are present. Must be a power of 2 no greater than 4096. Default: 1024. This is configured automatically by default; use <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-build-options-a"><code>-a</code>/<code>--noauto</code></a> to configure manually.</td>
</tr>
<tr>
<td><code>--nodc</code></td>
<td>Disable use of the difference-cover sample. Suffix sorting becomes quadratic-time in the worst case (where the worst case is an extremely repetitive reference). Default: off.</td>
</tr>
<tr>
<td><code>-r/--noref</code></td>
<td>Do not build the <code>NAME.3.bt2</code> and <code>NAME.4.bt2</code> portions of the index, which contain a bitpacked version of the reference sequences and are used for paired-end alignment.</td>
</tr>
<tr>
<td><code>-3/--justref</code></td>
<td>Build only the <code>NAME.3.bt2</code> and <code>NAME.4.bt2</code> portions of the index, which contain a bitpacked version of the reference sequences and are used for paired-end alignment.</td>
</tr>
<tr>
<td><code>-o/--offrate &lt;int&gt;</code></td>
<td>To map alignments back to positions on the reference sequences, it’s necessary to annotate (“mark”) some or all of the <a target="_blank" rel="noopener" href="http://en.wikipedia.org/wiki/Burrows-Wheeler_transform">Burrows-Wheeler</a> rows with their corresponding location on the genome. <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-build-options-o"><code>-o</code>/<code>--offrate</code></a> governs how many rows get marked: the indexer will mark every 2^<code>&lt;int&gt;</code> rows. Marking more rows makes reference-position lookups faster, but requires more memory to hold the annotations at runtime. The default is 5 (every 32nd row is marked; for human genome, annotations occupy about 340 megabytes).</td>
</tr>
<tr>
<td><code>-t/--ftabchars &lt;int&gt;</code></td>
<td>The ftab is the lookup table used to calculate an initial <a target="_blank" rel="noopener" href="http://en.wikipedia.org/wiki/Burrows-Wheeler_transform">Burrows-Wheeler</a> range with respect to the first <code>&lt;int&gt;</code>characters of the query. A larger <code>&lt;int&gt;</code> yields a larger lookup table but faster query times. The ftab has size 4^(<code>&lt;int&gt;</code>+1) bytes. The default setting is 10 (ftab is 4MB).</td>
</tr>
<tr>
<td><code>--seed &lt;int&gt;</code></td>
<td>Use <code>&lt;int&gt;</code> as the seed for pseudo-random number generator.</td>
</tr>
<tr>
<td><code>--cutoff &lt;int&gt;</code></td>
<td>Index only the first <code>&lt;int&gt;</code> bases of the reference sequences (cumulative across sequences) and ignore the rest.</td>
</tr>
<tr>
<td><code>-q/--quiet</code></td>
<td><code>bowtie2-build</code> is verbose by default. With this option <code>bowtie2-build</code> will print only error messages.</td>
</tr>
<tr>
<td><code>--threads &lt;int&gt;</code></td>
<td>By default <code>bowtie2-build</code> is using only one thread. Increasing the number of threads will speed up the index building considerably in most cases.</td>
</tr>
<tr>
<td><code>-h/--help</code></td>
<td>Print usage information and quit.</td>
</tr>
<tr>
<td><code>--version</code></td>
<td>Print version information and quit.</td>
</tr>
</tbody></table>
<h1 id="The-bowtie2-inspect-index-inspector"><a href="#The-bowtie2-inspect-index-inspector" class="headerlink" title="The bowtie2-inspect index inspector"></a>The <code>bowtie2-inspect</code> index inspector</h1><p><code>bowtie2-inspect</code> extracts information from a Bowtie index about what kind of index it is and what reference sequences were used to build it. When run without any options, the tool will output a <a target="_blank" rel="noopener" href="https://en.wikipedia.org/wiki/FASTA"><code>FASTA</code></a> file containing the sequences of the original references (with all non-<code>A</code>/<code>C</code>/<code>G</code>/<code>T</code> characters converted to <code>N</code>s). It can also be used to extract just the reference sequence names using the <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-inspect-options-n"><code>-n</code>/<code>--names</code></a> option or a more verbose summary using the <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#bowtie2-inspect-options-s"><code>-s</code>/<code>--summary</code></a> option.</p>
<h2 id="Command-Line-2"><a href="#Command-Line-2" class="headerlink" title="Command Line"></a>Command Line</h2><p>Usage:</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">bowtie2-inspect [options]* &lt;bt2_base&gt;</span><br></pre></td></tr></table></figure>

<h3 id="Main-arguments-2"><a href="#Main-arguments-2" class="headerlink" title="Main arguments"></a>Main arguments</h3><table>
<thead>
<tr>
<th><code>&lt;bt2_base&gt;</code></th>
<th>The basename of the index to be inspected. The basename is name of any of the index files but with the <code>.X.bt2</code> or <code>.rev.X.bt2</code>suffix omitted. <code>bowtie2-inspect</code> first looks in the current directory for the index files, then in the directory specified in the <code>BOWTIE2_INDEXES</code> environment variable.</th>
</tr>
</thead>
<tbody><tr>
<td></td>
<td></td>
</tr>
</tbody></table>
<h3 id="Options-2"><a href="#Options-2" class="headerlink" title="Options"></a>Options</h3><table>
<thead>
<tr>
<th><code>-a/--across &lt;int&gt;</code></th>
<th>When printing <a target="_blank" rel="noopener" href="https://en.wikipedia.org/wiki/FASTA"><code>FASTA</code></a> output, output a newline character every <code>&lt;int&gt;</code> bases (default: 60).</th>
</tr>
</thead>
<tbody><tr>
<td><code>-n/--names</code></td>
<td>Print reference sequence names, one per line, and quit.</td>
</tr>
<tr>
<td><code>-s/--summary</code></td>
<td>Print a summary that includes information about index settings, as well as the names and lengths of the input sequences. The summary has this format:<code>Colorspace  &lt;0 or 1&gt; SA-Sample   1 in &lt;sample&gt; FTab-Chars  &lt;chars&gt; Sequence-1  &lt;name&gt;  &lt;len&gt; Sequence-2  &lt;name&gt;  &lt;len&gt; ... Sequence-N  &lt;name&gt;  &lt;len&gt;</code>Fields are separated by tabs. Colorspace is always set to 0 for Bowtie 2.</td>
</tr>
<tr>
<td><code>-o/--output &lt;filename&gt;</code></td>
<td>Save output to user-specified filename (default: stdout)</td>
</tr>
<tr>
<td><code>-v/--verbose</code></td>
<td>Print verbose output (for debugging).</td>
</tr>
<tr>
<td><code>--version</code></td>
<td>Print version information and quit.</td>
</tr>
<tr>
<td><code>-h/--help</code></td>
<td>Print usage information and quit.</td>
</tr>
</tbody></table>
<h1 id="Getting-started-with-Bowtie-2-Lambda-phage-example"><a href="#Getting-started-with-Bowtie-2-Lambda-phage-example" class="headerlink" title="Getting started with Bowtie 2: Lambda phage example"></a>Getting started with Bowtie 2: Lambda phage example</h1><p>Bowtie 2 comes with some example files to get you started. The example files are not scientifically significant; we use the <a target="_blank" rel="noopener" href="http://en.wikipedia.org/wiki/Lambda_phage">Lambda phage</a> reference genome simply because it’s short, and the reads were generated by a computer program, not a sequencer. However, these files will let you start running Bowtie 2 and downstream tools right away.</p>
<p>First follow the manual instructions to <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#obtaining-bowtie-2">obtain Bowtie 2</a>. Set the <code>BT2_HOME</code> environment variable to point to the new Bowtie 2 directory containing the <code>bowtie2</code>, <code>bowtie2-build</code> and <code>bowtie2-inspect</code> binaries. This is important, as the <code>BT2_HOME</code> variable is used in the commands below to refer to that directory.</p>
<h2 id="Indexing-a-reference-genome"><a href="#Indexing-a-reference-genome" class="headerlink" title="Indexing a reference genome"></a>Indexing a reference genome</h2><p>To create an index for the <a target="_blank" rel="noopener" href="http://en.wikipedia.org/wiki/Lambda_phage">Lambda phage</a> reference genome included with Bowtie 2, create a new temporary directory (it doesn’t matter where), change into that directory, and run:</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">$BT2_HOME/bowtie2-build $BT2_HOME/example/reference/lambda_virus.fa lambda_virus</span><br></pre></td></tr></table></figure>

<p>The command should print many lines of output then quit. When the command completes, the current directory will contain four new files that all start with <code>lambda_virus</code> and end with <code>.1.bt2</code>, <code>.2.bt2</code>, <code>.3.bt2</code>, <code>.4.bt2</code>, <code>.rev.1.bt2</code>, and <code>.rev.2.bt2</code>. These files constitute the index - you’re done!</p>
<p>You can use <code>bowtie2-build</code> to create an index for a set of <a target="_blank" rel="noopener" href="https://en.wikipedia.org/wiki/FASTA"><code>FASTA</code></a> files obtained from any source, including sites such as <a target="_blank" rel="noopener" href="http://genome.ucsc.edu/cgi-bin/hgGateway">UCSC</a>, <a target="_blank" rel="noopener" href="http://www.ncbi.nlm.nih.gov/sites/genome">NCBI</a>, and <a target="_blank" rel="noopener" href="http://www.ensembl.org/">Ensembl</a>. When indexing multiple <a target="_blank" rel="noopener" href="https://en.wikipedia.org/wiki/FASTA"><code>FASTA</code></a> files, specify all the files using commas to separate file names. For more details on how to create an index with <code>bowtie2-build</code>, see the <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#the-bowtie2-build-indexer">manual section on index building</a>. You may also want to bypass this process by obtaining a pre-built index. See <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#using-a-pre-built-index">using a pre-built index</a> below for an example.</p>
<h2 id="Aligning-example-reads"><a href="#Aligning-example-reads" class="headerlink" title="Aligning example reads"></a>Aligning example reads</h2><p>Stay in the directory created in the previous step, which now contains the <code>lambda_virus</code> index files. Next, run:</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">$BT2_HOME/bowtie2 -x lambda_virus -U $BT2_HOME/example/reads/reads_1.fq -S eg1.sam</span><br></pre></td></tr></table></figure>

<p>This runs the Bowtie 2 aligner, which aligns a set of unpaired reads to the <a target="_blank" rel="noopener" href="http://en.wikipedia.org/wiki/Lambda_phage">Lambda phage</a> reference genome using the index generated in the previous step. The alignment results in SAM format are written to the file <code>eg1.sam</code>, and a short alignment summary is written to the console. (Actually, the summary is written to the “standard error” or “stderr” filehandle, which is typically printed to the console.)</p>
<p>To see the first few lines of the SAM output, run:</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">head eg1.sam</span><br></pre></td></tr></table></figure>

<p>You will see something like this:</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br></pre></td><td class="code"><pre><span class="line">@HD VN:1.0  SO:unsorted</span><br><span class="line">@SQ SN:gi|9626243|ref|NC_001416.1|  LN:48502</span><br><span class="line">@PG ID:bowtie2  PN:bowtie2  VN:2.0.1</span><br><span class="line">r1  0   gi|9626243|ref|NC_001416.1| 18401   42  122M    *   0   0   TGAATGCGAACTCCGGGACGCTCAGTAATGTGACGATAGCTGAAAACTGTACGATAAACNGTACGCTGAGGGCAGAAAAAATCGTCGGGGACATTNTAAAGGCGGCGAGCGCGGCTTTTCCG  +&quot;@6&lt;:27(F&amp;5)9&quot;B):%B+A-%5A?2$HCB0B+0=D&lt;7E/&lt;.03#!.F77@6B==?C&quot;7&gt;;))%;,3-$.A06+&lt;-1/@@?,26&quot;&gt;=?*@&#x27;0;$:;??G+:#+(A?9+10!8!?()?7C&gt;  AS:i:-5 XN:i:0  XM:i:3  XO:i:0  XG:i:0  NM:i:3  MD:Z:59G13G21G26    YT:Z:UU</span><br><span class="line">r2  0   gi|9626243|ref|NC_001416.1| 8886    42  275M    *   0   0   NTTNTGATGCGGGCTTGTGGAGTTCAGCCGATCTGACTTATGTCATTACCTATGAAATGTGAGGACGCTATGCCTGTACCAAATCCTACAATGCCGGTGAAAGGTGCCGGGATCACCCTGTGGGTTTATAAGGGGATCGGTGACCCCTACGCGAATCCGCTTTCAGACGTTGACTGGTCGCGTCTGGCAAAAGTTAAAGACCTGACGCCCGGCGAACTGACCGCTGAGNCCTATGACGACAGCTATCTCGATGATGAAGATGCAGACTGGACTGC (#!!&#x27;+!$&quot;&quot;%+(+)&#x27;%)%!+!(&amp;++)&#x27;&#x27;&quot;#&quot;#&amp;#&quot;!&#x27;!(&quot;%&#x27;&quot;&quot;(&quot;+&amp;%$%*%%#$%#%#!)*&#x27;(#&quot;)(($&amp;$&#x27;&amp;%+&amp;#%*)*#*%*&#x27;)(%+!%%*&quot;$%&quot;#+)$&amp;&amp;+)&amp;)*+!&quot;*)!*!(&quot;&amp;&amp;&quot;*#+&quot;&amp;&quot;&#x27;(%)*(&quot;&#x27;!$*!!%$&amp;&amp;&amp;$!!&amp;&amp;&quot;(*&quot;$&amp;&quot;#&amp;!$%&#x27;%&quot;#)$#+%*+)!&amp;*)+(&quot;&quot;#!)!%*#&quot;*)*&#x27;)&amp;&quot;)($+*%%)!*)!(&#x27;(%&quot;&quot;+%&quot;$##&quot;#+((&#x27;!*(($*&#x27;!&quot;*(&#x27;&quot;+)&amp;%#&amp;$+(&#x27;**$$&amp;+*&amp;!#%)&#x27;)&#x27;(+(!%+ AS:i:-14    XN:i:0  XM:i:8  XO:i:0  XG:i:0  NM:i:8  MD:Z:0A0C0G0A108C23G9T81T46 YT:Z:UU</span><br><span class="line">r3  16  gi|9626243|ref|NC_001416.1| 11599   42  338M    *   0   0   GGGCGCGTTACTGGGATGATCGTGAAAAGGCCCGTCTTGCGCTTGAAGCCGCCCGAAAGAAGGCTGAGCAGCAGACTCAAGAGGAGAAAAATGCGCAGCAGCGGAGCGATACCGAAGCGTCACGGCTGAAATATACCGAAGAGGCGCAGAAGGCTNACGAACGGCTGCAGACGCCGCTGCAGAAATATACCGCCCGTCAGGAAGAACTGANCAAGGCACNGAAAGACGGGAAAATCCTGCAGGCGGATTACAACACGCTGATGGCGGCGGCGAAAAAGGATTATGAAGCGACGCTGTAAAAGCCGAAACAGTCCAGCGTGAAGGTGTCTGCGGGCGAT  7F$%6=$:9B@/F&#x27;&gt;=?!D?@0(:A*)7/&gt;9C&gt;6#1&lt;6:C(.CC;#.;&gt;;2&#x27;$4D:?&amp;B!&gt;689?(0(G7+0=@37F)GG=&gt;?958.D2E04C&lt;E,*AD%G0.%$+A:&#x27;H;?8&lt;72:88?E6((CF)6DF#.)=&gt;B&gt;D-=&quot;C&#x27;B080E&#x27;5BH&quot;77&#x27;:&quot;@70#4%A5=6.2/1&gt;;9&quot;&amp;-H6)=$/0;5E:&lt;8G!@::1?2DC7C*;@*#.1C0.D&gt;H/20,!&quot;C-#,6@%&lt;+&lt;D(AG-).?&amp;#0.00&#x27;@)/F8?B!&amp;&quot;170,)&gt;:?&lt;A7#1(A@0E#&amp;A.*DC.E&quot;)AH&quot;+.,5,2&gt;5&quot;2?:G,F&quot;D0B8D-6$65D&lt;D!A/38860.*4;4B&lt;*31?6  AS:i:-22    XN:i:0  XM:i:8  XO:i:0  XG:i:0  NM:i:8  MD:Z:80C4C16A52T23G30A8T76A41   YT:Z:UU</span><br><span class="line">r4  0   gi|9626243|ref|NC_001416.1| 40075   42  184M    *   0   0   GGGCCAATGCGCTTACTGATGCGGAATTACGCCGTAAGGCCGCAGATGAGCTTGTCCATATGACTGCGAGAATTAACNGTGGTGAGGCGATCCCTGAACCAGTAAAACAACTTCCTGTCATGGGCGGTAGACCTCTAAATCGTGCACAGGCTCTGGCGAAGATCGCAGAAATCAAAGCTAAGT(=8B)GD04*G%&amp;4F,1&#x27;A&gt;.C&amp;7=F$,+#6!))43C,5/5+)?-/0&gt;/D3=-,2/+.1?@-&gt;;)00!&#x27;3!7BH$G)HG+ADC&#x27;#-9F)7&lt;7&quot;$?&amp;.&gt;0)@5;4,!0-#C!15CF8&amp;HB+B==H&gt;7,/)C5)5*+(F5A%D,EA&lt;(&gt;G9E0&gt;7&amp;/E?4%;#&#x27;92)&lt;5+@7:A.(BG@BG86@.G AS:i:-1 XN:i:0  XM:i:1  XO:i:0  XG:i:0  NM:i:1  MD:Z:77C106 YT:Z:UU</span><br><span class="line">r5  0   gi|9626243|ref|NC_001416.1| 48010   42  138M    *   0   0   GTCAGGAAAGTGGTAAAACTGCAACTCAATTACTGCAATGCCCTCGTAATTAAGTGAATTTACAATATCGTCCTGTTCGGAGGGAAGAACGCGGGATGTTCATTCTTCATCACTTTTAATTGATGTATATGCTCTCTT  9&#x27;&#x27;%&lt;D)A03E1-*7=),:F/0!6,D9:H,&lt;9D%:0B(%&#x27;E,(8EFG$E89B$27G8F*2+4,-!,0D5()&amp;=(FGG:5;3*@/.0F-G#5#3-&gt;(&#x27;FDFEG?)5.!)&quot;AGADB3?6(@H(:B&lt;&gt;6!&gt;;&gt;6&gt;G,.&quot;?%  AS:i:0  XN:i:0  XM:i:0  XO:i:0  XG:i:0  NM:i:0  MD:Z:138    YT:Z:UU</span><br><span class="line">r6  16  gi|9626243|ref|NC_001416.1| 41607   42  72M2D119M   *   0   0   TCGATTTGCAAATACCGGAACATCTCGGTAACTGCATATTCTGCATTAAAAAATCAACGCAAAAAATCGGACGCCTGCAAAGATGAGGAGGGATTGCAGCGTGTTTTTAATGAGGTCATCACGGGATNCCATGTGCGTGACGGNCATCGGGAAACGCCAAAGGAGATTATGTACCGAGGAAGAATGTCGCT 1H#G;H&quot;$E*E#&amp;&quot;*)2%66?=9/9&#x27;=;4)4/&gt;@%+5#@#$4A*!&lt;D==&quot;8#1*A9BA=:(1+#C&amp;.#(3#H=9E)AC*5,AC#E&#x27;536*2?)H14?&gt;9&#x27;B=7(3H/B:+A:8%1-+#(E%&amp;$$&amp;14&quot;76D?&gt;7(&amp;20H5%*&amp;CF8!G5B+A4F$7(:&quot;&#x27;?0$?G+$)B-?2&lt;0&lt;F=D!38BH,%=8&amp;5@+ AS:i:-13    XN:i:0  XM:i:2  XO:i:1  XG:i:2  NM:i:4  MD:Z:72^TT55C15A47  YT:Z:UU</span><br><span class="line">r7  16  gi|9626243|ref|NC_001416.1| 4692    42  143M    *   0   0   TCAGCCGGACGCGGGCGCTGCAGCCGTACTCGGGGATGACCGGTTACAACGGCATTATCGCCCGTCTGCAACAGGCTGCCAGCGATCCGATGGTGGACAGCATTCTGCTCGATATGGACANGCCCGGCGGGATGGTGGCGGGG -&quot;/@*7A0)&gt;2,AAH@&amp;&quot;%B)*5*23B/,)90.B@%=FE,E063C9?,:26$-0:,.,1849&#x27;4.;F&gt;FA;76+5&amp;$&lt;C&quot;:$!A*,&lt;B,&lt;)@&lt;&#x27;85D%C*:)30@85;?.B$05=@95DCDH&lt;53!8G:F:B7/A.E&#x27;:434&gt; AS:i:-6 XN:i:0  XM:i:2  XO:i:0  XG:i:0  NM:i:2  MD:Z:98G21C22   YT:Z:UU</span><br></pre></td></tr></table></figure>

<p>The first few lines (beginning with <code>@</code>) are SAM header lines, and the rest of the lines are SAM alignments, one line per read or mate. See the <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#sam-output">Bowtie 2 manual section on SAM output</a> and the <a target="_blank" rel="noopener" href="http://samtools.sourceforge.net/SAM1.pdf">SAM specification</a> for details about how to interpret the SAM file format.</p>
<h2 id="Paired-end-example"><a href="#Paired-end-example" class="headerlink" title="Paired-end example"></a>Paired-end example</h2><p>To align paired-end reads included with Bowtie 2, stay in the same directory and run:</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">$BT2_HOME/bowtie2 -x lambda_virus -1 $BT2_HOME/example/reads/reads_1.fq -2 $BT2_HOME/example/reads/reads_2.fq -S eg2.sam</span><br></pre></td></tr></table></figure>

<p>This aligns a set of paired-end reads to the reference genome, with results written to the file <code>eg2.sam</code>.</p>
<h2 id="Local-alignment-example-1"><a href="#Local-alignment-example-1" class="headerlink" title="Local alignment example"></a>Local alignment example</h2><p>To use <a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#end-to-end-alignment-versus-local-alignment">local alignment</a> to align some longer reads included with Bowtie 2, stay in the same directory and run:</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">$BT2_HOME/bowtie2 --local -x lambda_virus -U $BT2_HOME/example/reads/longreads.fq -S eg3.sam</span><br></pre></td></tr></table></figure>

<p>This aligns the long reads to the reference genome using local alignment, with results written to the file <code>eg3.sam</code>.</p>
<h2 id="Using-SAMtools-BCFtools-downstream"><a href="#Using-SAMtools-BCFtools-downstream" class="headerlink" title="Using SAMtools/BCFtools downstream"></a>Using SAMtools/BCFtools downstream</h2><p><a target="_blank" rel="noopener" href="http://samtools.sourceforge.net/">SAMtools</a> is a collection of tools for manipulating and analyzing SAM and BAM alignment files. <a target="_blank" rel="noopener" href="http://samtools.sourceforge.net/mpileup.shtml">BCFtools</a> is a collection of tools for calling variants and manipulating VCF and BCF files, and it is typically distributed with <a target="_blank" rel="noopener" href="http://samtools.sourceforge.net/">SAMtools</a>. Using these tools together allows you to get from alignments in SAM format to variant calls in VCF format. This example assumes that <code>samtools</code> and <code>bcftools</code> are installed and that the directories containing these binaries are in your <a target="_blank" rel="noopener" href="http://en.wikipedia.org/wiki/PATH_(variable)">PATH environment variable</a>.</p>
<p>Run the paired-end example:</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">$BT2_HOME/bowtie2 -x $BT2_HOME/example/index/lambda_virus -1 $BT2_HOME/example/reads/reads_1.fq -2 $BT2_HOME/example/reads/reads_2.fq -S eg2.sam</span><br></pre></td></tr></table></figure>

<p>Use <code>samtools view</code> to convert the SAM file into a BAM file. BAM is the binary format corresponding to the SAM text format. Run:</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">samtools view -bS eg2.sam &gt; eg2.bam</span><br></pre></td></tr></table></figure>

<p>Use <code>samtools sort</code> to convert the BAM file to a sorted BAM file.</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">samtools sort eg2.bam -o eg2.sorted.bam</span><br></pre></td></tr></table></figure>

<p>We now have a sorted BAM file called <code>eg2.sorted.bam</code>. Sorted BAM is a useful format because the alignments are (a) compressed, which is convenient for long-term storage, and (b) sorted, which is conveneint for variant discovery. To generate variant calls in VCF format, run:</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">bcftools mpileup -f $BT2_HOME/example/reference/lambda_virus.fa eg2.sorted.bam | bcftools view -Ov - &gt; eg2.raw.bcf</span><br></pre></td></tr></table></figure>

<p>Then to view the variants, run:</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">bcftools view eg2.raw.bcf</span><br></pre></td></tr></table></figure>

<p>See the official SAMtools guide to <a target="_blank" rel="noopener" href="http://samtools.sourceforge.net/mpileup.shtml">Calling SNPs/INDELs with SAMtools/BCFtools</a> for more details and variations on this process.</p>

    </div>

    
    
    
        

<div>
<ul class="post-copyright">
  <li class="post-copyright-author">
    <strong>Post author:  </strong>Wang C
  </li>
  <li class="post-copyright-link">
    <strong>Post link: </strong>
    <a href="https://wangc1106.github.io/blog/2022/03/20/2022-03-20-bowtie2/" title="【bowtie2】">https://wangc1106.github.io/blog/2022/03/20/2022-03-20-bowtie2/</a>
  </li>
  <li class="post-copyright-license">
    <strong>Copyright Notice:  </strong>All articles in this blog are licensed under <a href="https://creativecommons.org/licenses/by-nc-sa/4.0/" rel="noopener" target="_blank"><i class="fab fa-fw fa-creative-commons"></i>BY-NC-SA</a> unless stating additionally.
  </li>
</ul>
</div>


      <footer class="post-footer">
          <div class="post-tags">
              <a href="/blog/tags/alignment/" rel="tag"># alignment</a>
          </div>

        


        
    <div class="post-nav">
      <div class="post-nav-item">
    <a href="/blog/2022/03/20/2022-03-18-blast/" rel="prev" title="【blast+】">
      <i class="fa fa-chevron-left"></i> 【blast+】
    </a></div>
      <div class="post-nav-item">
    <a href="/blog/2022/03/20/2022-03-20-ATAC-seq-6/" rel="next" title="【ATAC-seq-pipeline】">
      【ATAC-seq-pipeline】 <i class="fa fa-chevron-right"></i>
    </a></div>
    </div>
      </footer>
    
  </article>
  
  
  



          </div>
          

<script>
  window.addEventListener('tabs:register', () => {
    let { activeClass } = CONFIG.comments;
    if (CONFIG.comments.storage) {
      activeClass = localStorage.getItem('comments_active') || activeClass;
    }
    if (activeClass) {
      let activeTab = document.querySelector(`a[href="#comment-${activeClass}"]`);
      if (activeTab) {
        activeTab.click();
      }
    }
  });
  if (CONFIG.comments.storage) {
    window.addEventListener('tabs:click', event => {
      if (!event.target.matches('.tabs-comment .tab-content .tab-pane')) return;
      let commentClass = event.target.classList[1];
      localStorage.setItem('comments_active', commentClass);
    });
  }
</script>

        </div>
          
  
  <div class="toggle sidebar-toggle">
    <span class="toggle-line toggle-line-first"></span>
    <span class="toggle-line toggle-line-middle"></span>
    <span class="toggle-line toggle-line-last"></span>
  </div>

  <aside class="sidebar">
    <div class="sidebar-inner">

      <ul class="sidebar-nav motion-element">
        <li class="sidebar-nav-toc">
          Table of Contents
        </li>
        <li class="sidebar-nav-overview">
          Overview
        </li>
      </ul>

      <!--noindex-->
      <div class="post-toc-wrap sidebar-panel">
          <div class="post-toc motion-element"><ol class="nav"><li class="nav-item nav-level-1"><a class="nav-link" href="#Introduction"><span class="nav-number">1.</span> <span class="nav-text">Introduction</span></a><ol class="nav-child"><li class="nav-item nav-level-2"><a class="nav-link" href="#How-is-Bowtie-2-different-from-Bowtie-1"><span class="nav-number">1.1.</span> <span class="nav-text">How is Bowtie 2 different from Bowtie 1?</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#What-isn%E2%80%99t-Bowtie-2"><span class="nav-number">1.2.</span> <span class="nav-text">What isn’t Bowtie 2?</span></a></li></ol></li><li class="nav-item nav-level-1"><a class="nav-link" href="#Obtaining-Bowtie-2"><span class="nav-number">2.</span> <span class="nav-text">Obtaining Bowtie 2</span></a><ol class="nav-child"><li class="nav-item nav-level-2"><a class="nav-link" href="#Building-from-source"><span class="nav-number">2.1.</span> <span class="nav-text">Building from source</span></a></li></ol></li><li class="nav-item nav-level-1"><a class="nav-link" href="#The-bowtie2-aligner"><span class="nav-number">3.</span> <span class="nav-text">The bowtie2 aligner</span></a><ol class="nav-child"><li class="nav-item nav-level-2"><a class="nav-link" href="#End-to-end-alignment-versus-local-alignment"><span class="nav-number">3.1.</span> <span class="nav-text">End-to-end alignment versus local alignment</span></a><ol class="nav-child"><li class="nav-item nav-level-3"><a class="nav-link" href="#End-to-end-alignment-example"><span class="nav-number">3.1.1.</span> <span class="nav-text">End-to-end alignment example</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#Local-alignment-example"><span class="nav-number">3.1.2.</span> <span class="nav-text">Local alignment example</span></a></li></ol></li><li class="nav-item nav-level-2"><a class="nav-link" href="#Scores-higher-more-similar"><span class="nav-number">3.2.</span> <span class="nav-text">Scores: higher &#x3D; more similar</span></a><ol class="nav-child"><li class="nav-item nav-level-3"><a class="nav-link" href="#End-to-end-alignment-score-example-Default"><span class="nav-number">3.2.1.</span> <span class="nav-text">End-to-end alignment score example[Default]</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#Local-alignment-score-example"><span class="nav-number">3.2.2.</span> <span class="nav-text">Local alignment score example</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#Valid-alignments-meet-or-exceed-the-minimum-score-threshold"><span class="nav-number">3.2.3.</span> <span class="nav-text">Valid alignments meet or exceed the minimum score threshold</span></a></li></ol></li><li class="nav-item nav-level-2"><a class="nav-link" href="#MAPQ-Mapping-quality-higher-more-unique"><span class="nav-number">3.3.</span> <span class="nav-text">MAPQ,Mapping quality: higher &#x3D; more unique</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#Aligning-pairs"><span class="nav-number">3.4.</span> <span class="nav-text">Aligning pairs</span></a><ol class="nav-child"><li class="nav-item nav-level-3"><a class="nav-link" href="#Paired-inputs"><span class="nav-number">3.4.1.</span> <span class="nav-text">Paired inputs</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#Paired-SAM-output"><span class="nav-number">3.4.2.</span> <span class="nav-text">Paired SAM output</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#Concordant-pairs-match-pair-expectations-discordant-pairs-don%E2%80%99t"><span class="nav-number">3.4.3.</span> <span class="nav-text">Concordant pairs match pair expectations, discordant pairs don’t</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#Mixed-mode-paired-where-possible-unpaired-otherwise"><span class="nav-number">3.4.4.</span> <span class="nav-text">Mixed mode: paired where possible, unpaired otherwise</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#Some-SAM-FLAGS-describe-paired-end-properties"><span class="nav-number">3.4.5.</span> <span class="nav-text">Some SAM FLAGS describe paired-end properties</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#Some-SAM-optional-fields-describe-more-paired-end-properties"><span class="nav-number">3.4.6.</span> <span class="nav-text">Some SAM optional fields describe more paired-end properties</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#Mates-can-overlap-contain-or-dovetail-each-other"><span class="nav-number">3.4.7.</span> <span class="nav-text">Mates can overlap, contain, or dovetail each other</span></a></li></ol></li><li class="nav-item nav-level-2"><a class="nav-link" href="#Reporting"><span class="nav-number">3.5.</span> <span class="nav-text">Reporting</span></a><ol class="nav-child"><li class="nav-item nav-level-3"><a class="nav-link" href="#Distinct-alignments-map-a-read-to-different-places"><span class="nav-number">3.5.1.</span> <span class="nav-text">Distinct alignments map a read to different places</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#Default-mode-search-for-multiple-alignments-report-the-best-one"><span class="nav-number">3.5.2.</span> <span class="nav-text">Default mode: search for multiple alignments, report the best one</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#k-mode-search-for-one-or-more-alignments-report-each"><span class="nav-number">3.5.3.</span> <span class="nav-text">-k mode: search for one or more alignments, report each</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#a-mode-search-for-and-report-all-alignments"><span class="nav-number">3.5.4.</span> <span class="nav-text">-a mode: search for and report all alignments</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#Randomness-in-Bowtie-2"><span class="nav-number">3.5.5.</span> <span class="nav-text">Randomness in Bowtie 2</span></a></li></ol></li><li class="nav-item nav-level-2"><a class="nav-link" href="#Multiseed-heuristic"><span class="nav-number">3.6.</span> <span class="nav-text">Multiseed heuristic</span></a><ol class="nav-child"><li class="nav-item nav-level-3"><a class="nav-link" href="#FM-Index-memory-footprint"><span class="nav-number">3.6.1.</span> <span class="nav-text">FM Index memory footprint</span></a></li></ol></li><li class="nav-item nav-level-2"><a class="nav-link" href="#Ambiguous-characters"><span class="nav-number">3.7.</span> <span class="nav-text">Ambiguous characters</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#Presets-setting-many-settings-at-once"><span class="nav-number">3.8.</span> <span class="nav-text">Presets: setting many settings at once</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#Filtering"><span class="nav-number">3.9.</span> <span class="nav-text">Filtering</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#Alignment-summary"><span class="nav-number">3.10.</span> <span class="nav-text">Alignment summary</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#Wrapper-scripts"><span class="nav-number">3.11.</span> <span class="nav-text">Wrapper scripts</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#Small-and-large-indexes"><span class="nav-number">3.12.</span> <span class="nav-text">Small and large indexes</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#Performance-tuning"><span class="nav-number">3.13.</span> <span class="nav-text">Performance tuning</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#Command-Line"><span class="nav-number">3.14.</span> <span class="nav-text">Command Line</span></a><ol class="nav-child"><li class="nav-item nav-level-3"><a class="nav-link" href="#Alignment"><span class="nav-number">3.14.1.</span> <span class="nav-text">Alignment</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#Usage"><span class="nav-number">3.14.2.</span> <span class="nav-text">Usage</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#Main-arguments"><span class="nav-number">3.14.3.</span> <span class="nav-text">Main arguments</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#Options"><span class="nav-number">3.14.4.</span> <span class="nav-text">Options</span></a><ol class="nav-child"><li class="nav-item nav-level-4"><a class="nav-link" href="#Input-options"><span class="nav-number">3.14.4.1.</span> <span class="nav-text">Input options</span></a></li><li class="nav-item nav-level-4"><a class="nav-link" href="#Preset-options-in-end-to-end-mode"><span class="nav-number">3.14.4.2.</span> <span class="nav-text">Preset options in --end-to-end mode</span></a></li><li class="nav-item nav-level-4"><a class="nav-link" href="#Preset-options-in-local-mode"><span class="nav-number">3.14.4.3.</span> <span class="nav-text">Preset options in --local mode</span></a></li><li class="nav-item nav-level-4"><a class="nav-link" href="#Alignment-options"><span class="nav-number">3.14.4.4.</span> <span class="nav-text">Alignment options</span></a></li><li class="nav-item nav-level-4"><a class="nav-link" href="#Scoring-options"><span class="nav-number">3.14.4.5.</span> <span class="nav-text">Scoring options</span></a></li><li class="nav-item nav-level-4"><a class="nav-link" href="#Reporting-options"><span class="nav-number">3.14.4.6.</span> <span class="nav-text">Reporting options</span></a></li><li class="nav-item nav-level-4"><a class="nav-link" href="#Effort-options"><span class="nav-number">3.14.4.7.</span> <span class="nav-text">Effort options</span></a></li><li class="nav-item nav-level-4"><a class="nav-link" href="#Paired-end-options"><span class="nav-number">3.14.4.8.</span> <span class="nav-text">Paired-end options</span></a></li><li class="nav-item nav-level-4"><a class="nav-link" href="#BAM-options"><span class="nav-number">3.14.4.9.</span> <span class="nav-text">BAM options</span></a></li><li class="nav-item nav-level-4"><a class="nav-link" href="#Output-options"><span class="nav-number">3.14.4.10.</span> <span class="nav-text">Output options</span></a></li><li class="nav-item nav-level-4"><a class="nav-link" href="#SAM-options"><span class="nav-number">3.14.4.11.</span> <span class="nav-text">SAM options</span></a></li><li class="nav-item nav-level-4"><a class="nav-link" href="#Performance-options"><span class="nav-number">3.14.4.12.</span> <span class="nav-text">Performance options</span></a></li><li class="nav-item nav-level-4"><a class="nav-link" href="#Other-options"><span class="nav-number">3.14.4.13.</span> <span class="nav-text">Other options</span></a></li></ol></li></ol></li><li class="nav-item nav-level-2"><a class="nav-link" href="#SAM-output"><span class="nav-number">3.15.</span> <span class="nav-text">SAM output</span></a></li></ol></li><li class="nav-item nav-level-1"><a class="nav-link" href="#The-bowtie2-build-indexer"><span class="nav-number">4.</span> <span class="nav-text">The bowtie2-build indexer</span></a><ol class="nav-child"><li class="nav-item nav-level-2"><a class="nav-link" href="#Command-Line-1"><span class="nav-number">4.1.</span> <span class="nav-text">Command Line</span></a><ol class="nav-child"><li class="nav-item nav-level-3"><a class="nav-link" href="#Main-arguments-1"><span class="nav-number">4.1.1.</span> <span class="nav-text">Main arguments</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#Options-1"><span class="nav-number">4.1.2.</span> <span class="nav-text">Options</span></a></li></ol></li></ol></li><li class="nav-item nav-level-1"><a class="nav-link" href="#The-bowtie2-inspect-index-inspector"><span class="nav-number">5.</span> <span class="nav-text">The bowtie2-inspect index inspector</span></a><ol class="nav-child"><li class="nav-item nav-level-2"><a class="nav-link" href="#Command-Line-2"><span class="nav-number">5.1.</span> <span class="nav-text">Command Line</span></a><ol class="nav-child"><li class="nav-item nav-level-3"><a class="nav-link" href="#Main-arguments-2"><span class="nav-number">5.1.1.</span> <span class="nav-text">Main arguments</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#Options-2"><span class="nav-number">5.1.2.</span> <span class="nav-text">Options</span></a></li></ol></li></ol></li><li class="nav-item nav-level-1"><a class="nav-link" href="#Getting-started-with-Bowtie-2-Lambda-phage-example"><span class="nav-number">6.</span> <span class="nav-text">Getting started with Bowtie 2: Lambda phage example</span></a><ol class="nav-child"><li class="nav-item nav-level-2"><a class="nav-link" href="#Indexing-a-reference-genome"><span class="nav-number">6.1.</span> <span class="nav-text">Indexing a reference genome</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#Aligning-example-reads"><span class="nav-number">6.2.</span> <span class="nav-text">Aligning example reads</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#Paired-end-example"><span class="nav-number">6.3.</span> <span class="nav-text">Paired-end example</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#Local-alignment-example-1"><span class="nav-number">6.4.</span> <span class="nav-text">Local alignment example</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#Using-SAMtools-BCFtools-downstream"><span class="nav-number">6.5.</span> <span class="nav-text">Using SAMtools&#x2F;BCFtools downstream</span></a></li></ol></li></ol></div>
      </div>
      <!--/noindex-->

      <div class="site-overview-wrap sidebar-panel">
        <div class="site-author motion-element" itemprop="author" itemscope itemtype="http://schema.org/Person">
    <img class="site-author-image" itemprop="image" alt="Wang C"
      src="/blog/images/banner.png">
  <p class="site-author-name" itemprop="name">Wang C</p>
  <div class="site-description" itemprop="description">Make more time</div>
</div>
<div class="site-state-wrap motion-element">
  <nav class="site-state">
      <div class="site-state-item site-state-posts">
          <a href="/blog/archives">
          <span class="site-state-item-count">46</span>
          <span class="site-state-item-name">posts</span>
        </a>
      </div>
      <div class="site-state-item site-state-categories">
            <a href="/blog/categories/">
          
        <span class="site-state-item-count">13</span>
        <span class="site-state-item-name">categories</span></a>
      </div>
      <div class="site-state-item site-state-tags">
            <a href="/blog/tags/">
          
        <span class="site-state-item-count">19</span>
        <span class="site-state-item-name">tags</span></a>
      </div>
  </nav>
</div>
  <div class="links-of-author motion-element">
      <span class="links-of-author-item">
        <a href="https://github.com/wangc1106" title="GitHub → https:&#x2F;&#x2F;github.com&#x2F;wangc1106" rel="noopener" target="_blank"><i class="fab fa-github fa-fw"></i>GitHub</a>
      </span>
      <span class="links-of-author-item">
        <a href="mailto:wangchen116@126.com" title="E-Mail → mailto:wangchen116@126.com" rel="noopener" target="_blank"><i class="fa fa-envelope fa-fw"></i>E-Mail</a>
      </span>
  </div>
  <div class="cc-license motion-element" itemprop="license">
    <a href="https://creativecommons.org/licenses/by-nc-sa/4.0/" class="cc-opacity" rel="noopener" target="_blank"><img src="/blog/images/cc-by-nc-sa.svg" alt="Creative Commons"></a>
  </div>



      </div>
        <div class="back-to-top motion-element">
          <i class="fa fa-arrow-up"></i>
          <span>0%</span>
        </div>

    </div>
  </aside>
  <div id="sidebar-dimmer"></div>


      </div>
    </main>

    <footer class="footer">
      <div class="footer-inner">
        

        

<div class="copyright">
  
  &copy; 2021 – 
  <span itemprop="copyrightYear">2022</span>
  <span class="with-love">
    <i class="fas fa-dna"></i>
  </span>
  <span class="author" itemprop="copyrightHolder">Wang C</span>
</div>

        






<script>
  (function() {
    function leancloudSelector(url) {
      url = encodeURI(url);
      return document.getElementById(url).querySelector('.leancloud-visitors-count');
    }

    function addCount(Counter) {
      var visitors = document.querySelector('.leancloud_visitors');
      var url = decodeURI(visitors.id);
      var title = visitors.dataset.flagTitle;

      Counter('get', '/classes/Counter?where=' + encodeURIComponent(JSON.stringify({ url })))
        .then(response => response.json())
        .then(({ results }) => {
          if (results.length > 0) {
            var counter = results[0];
            leancloudSelector(url).innerText = counter.time + 1;
            Counter('put', '/classes/Counter/' + counter.objectId, { time: { '__op': 'Increment', 'amount': 1 } })
              .catch(error => {
                console.error('Failed to save visitor count', error);
              });
          } else {
              Counter('post', '/classes/Counter', { title, url, time: 1 })
                .then(response => response.json())
                .then(() => {
                  leancloudSelector(url).innerText = 1;
                })
                .catch(error => {
                  console.error('Failed to create', error);
                });
          }
        })
        .catch(error => {
          console.error('LeanCloud Counter Error', error);
        });
    }

    function showTime(Counter) {
      var visitors = document.querySelectorAll('.leancloud_visitors');
      var entries = [...visitors].map(element => {
        return decodeURI(element.id);
      });

      Counter('get', '/classes/Counter?where=' + encodeURIComponent(JSON.stringify({ url: { '$in': entries } })))
        .then(response => response.json())
        .then(({ results }) => {
          for (let url of entries) {
            let target = results.find(item => item.url === url);
            leancloudSelector(url).innerText = target ? target.time : 0;
          }
        })
        .catch(error => {
          console.error('LeanCloud Counter Error', error);
        });
    }

    let { app_id, app_key, server_url } = {"enable":true,"app_id":"UqIYufmNk2sE8MiBKKWc4Kja-gzGzoHsz","app_key":"iYpeWQcmNqpHwr2uVlnJYVji","server_url":null,"security":false};
    function fetchData(api_server) {
      var Counter = (method, url, data) => {
        return fetch(`${api_server}/1.1${url}`, {
          method,
          headers: {
            'X-LC-Id'     : app_id,
            'X-LC-Key'    : app_key,
            'Content-Type': 'application/json',
          },
          body: JSON.stringify(data)
        });
      };
      if (CONFIG.page.isPost) {
        if (CONFIG.hostname !== location.hostname) return;
        addCount(Counter);
      } else if (document.querySelectorAll('.post-title-link').length >= 1) {
        showTime(Counter);
      }
    }

    let api_server = app_id.slice(-9) !== '-MdYXbMMI' ? server_url : `https://${app_id.slice(0, 8).toLowerCase()}.api.lncldglobal.com`;

    if (api_server) {
      fetchData(api_server);
    } else {
      fetch('https://app-router.leancloud.cn/2/route?appId=' + app_id)
        .then(response => response.json())
        .then(({ api_server }) => {
          fetchData('https://' + api_server);
        });
    }
  })();
</script>


      </div>
      <div style="height: 200px">
      <script type="text/javascript" src="//rf.revolvermaps.com/0/0/8.js?i=5knf34gs07t&amp;m=0&amp;c=ff0000&amp;cr1=ffffff&amp;f=arial&amp;l=33" async="async"></script>
      </div>
    </footer>
  </div>

  
  <script src="/blog/lib/anime.min.js"></script>
  <script src="/blog/lib/velocity/velocity.min.js"></script>
  <script src="/blog/lib/velocity/velocity.ui.min.js"></script>

<script src="/blog/js/utils.js"></script>

<script src="/blog/js/motion.js"></script>


<script src="/blog/js/schemes/pisces.js"></script>


<script src="/blog/js/next-boot.js"></script>




  




  
<script src="/blog/js/local-search.js"></script>













  

  

</body>
</html>
