<!DOCTYPE html>
<html lang="zh-Hans">
<head>
  <meta charset="UTF-8">
<meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=2">
<meta name="theme-color" content="#222">
<meta name="generator" content="Hexo 5.4.0">
  <link rel="apple-touch-icon" sizes="180x180" href="/blog/images/apple-touch-icon-next.png">
  <link rel="icon" type="image/png" sizes="32x32" href="/blog/images/favicon-32x32-next.png">
  <link rel="icon" type="image/png" sizes="16x16" href="/blog/images/favicon-16x16-next.png">
  <link rel="mask-icon" href="/blog/images/logo.svg" color="#222">
  <meta name="google-site-verification" content="1osTo_gOe1SW9Hevqtz8vl8HfSgTO4DSMZ0DiMPMX6w">

<link rel="stylesheet" href="/blog/css/main.css">


<link rel="stylesheet" href="/blog/lib/font-awesome/css/all.min.css">

<script id="hexo-configurations">
    var NexT = window.NexT || {};
    var CONFIG = {"hostname":"wangc1106.github.io","root":"/blog/","scheme":"Gemini","version":"7.8.0","exturl":false,"sidebar":{"position":"left","display":"post","padding":18,"offset":12,"onmobile":false},"copycode":{"enable":false,"show_result":false,"style":null},"back2top":{"enable":true,"sidebar":true,"scrollpercent":"ture"},"bookmark":{"enable":false,"color":"#222","save":"auto"},"fancybox":false,"mediumzoom":false,"lazyload":false,"pangu":false,"comments":{"style":"tabs","active":null,"storage":true,"lazyload":false,"nav":null},"algolia":{"hits":{"per_page":10},"labels":{"input_placeholder":"Search for Posts","hits_empty":"We didn't find any results for the search: ${query}","hits_stats":"${hits} results found in ${time} ms"}},"localsearch":{"enable":true,"trigger":"auto","top_n_per_article":1,"unescape":false,"preload":false},"motion":{"enable":true,"async":false,"transition":{"post_block":"fadeIn","post_header":"slideDownIn","post_body":"slideDownIn","coll_header":"slideLeftIn","sidebar":"slideUpIn"}},"path":"search.xml"};
  </script>

  <meta name="description" content="Hifiasm is a fast haplotype-resolved de novo assembler for PacBio HiFi reads(High Fidelity,high accuracy consensus reads). HiCanu, Hifiasm, BUSCO, Peregrine, minimap2">
<meta property="og:type" content="article">
<meta property="og:title" content="HiFi-3-assemble-software">
<meta property="og:url" content="https://wangc1106.github.io/blog/2022/03/02/2022-03-02-HiFi-3-software/index.html">
<meta property="og:site_name" content="Banner&#39;s Blog">
<meta property="og:description" content="Hifiasm is a fast haplotype-resolved de novo assembler for PacBio HiFi reads(High Fidelity,high accuracy consensus reads). HiCanu, Hifiasm, BUSCO, Peregrine, minimap2">
<meta property="og:locale">
<meta property="og:image" content="https://wangc1106.github.io/blog/2022/03/02/2022-03-02-HiFi-3-software/9.png">
<meta property="og:image" content="https://wangc1106.github.io/blog/2022/03/02/2022-03-02-HiFi-3-software/1.png">
<meta property="og:image" content="https://wangc1106.github.io/blog/640.jpeg">
<meta property="og:image" content="https://wangc1106.github.io/blog/2022/03/02/2022-03-02-HiFi-3-software/10.png">
<meta property="og:image" content="https://wangc1106.github.io/blog/2022/03/02/2022-03-02-HiFi-3-software/11.png">
<meta property="og:image" content="https://wangc1106.github.io/blog/2022/03/02/2022-03-02-HiFi-3-software/12.png">
<meta property="og:image" content="https://wangc1106.github.io/blog/2022/03/02/2022-03-02-HiFi-3-software/14.png">
<meta property="og:image" content="https://wangc1106.github.io/blog/2022/03/02/2022-03-02-HiFi-3-software/13.png">
<meta property="og:image" content="https://wangc1106.github.io/blog/2022/03/02/2022-03-02-HiFi-3-software/2.png">
<meta property="og:image" content="https://wangc1106.github.io/blog/2022/03/02/2022-03-02-HiFi-3-software/3.png">
<meta property="og:image" content="https://wangc1106.github.io/blog/2022/03/02/2022-03-02-HiFi-3-software/4.png">
<meta property="og:image" content="https://wangc1106.github.io/blog/2022/03/02/2022-03-02-HiFi-3-software/5.png">
<meta property="og:image" content="https://wangc1106.github.io/blog/2022/03/02/2022-03-02-HiFi-3-software/6.png">
<meta property="og:image" content="https://wangc1106.github.io/blog/2022/03/02/2022-03-02-HiFi-3-software/7.png">
<meta property="article:published_time" content="2022-03-02T01:08:25.090Z">
<meta property="article:modified_time" content="2022-03-06T15:12:39.754Z">
<meta property="article:author" content="Wang C">
<meta property="article:tag" content="denovo-III">
<meta name="twitter:card" content="summary">
<meta name="twitter:image" content="https://wangc1106.github.io/blog/2022/03/02/2022-03-02-HiFi-3-software/9.png">

<link rel="canonical" href="https://wangc1106.github.io/blog/2022/03/02/2022-03-02-HiFi-3-software/">


<script id="page-configurations">
  // https://hexo.io/docs/variables.html
  CONFIG.page = {
    sidebar: "",
    isHome : false,
    isPost : true,
    lang   : 'zh-Hans'
  };
</script>

  <title>HiFi-3-assemble-software | Banner's Blog</title>
  






  <noscript>
  <style>
  .use-motion .brand,
  .use-motion .menu-item,
  .sidebar-inner,
  .use-motion .post-block,
  .use-motion .pagination,
  .use-motion .comments,
  .use-motion .post-header,
  .use-motion .post-body,
  .use-motion .collection-header { opacity: initial; }

  .use-motion .site-title,
  .use-motion .site-subtitle {
    opacity: initial;
    top: initial;
  }

  .use-motion .logo-line-before i { left: initial; }
  .use-motion .logo-line-after i { right: initial; }
  </style>
</noscript>

</head>

<body itemscope itemtype="http://schema.org/WebPage">
  <div class="container use-motion">
    <div class="headband"></div>

    <header class="header" itemscope itemtype="http://schema.org/WPHeader">
      <div class="header-inner"><div class="site-brand-container">
  <div class="site-nav-toggle">
    <div class="toggle" aria-label="Toggle navigation bar">
      <span class="toggle-line toggle-line-first"></span>
      <span class="toggle-line toggle-line-middle"></span>
      <span class="toggle-line toggle-line-last"></span>
    </div>
  </div>

  <div class="site-meta">

    <a href="/blog/" class="brand" rel="start">
      <span class="logo-line-before"><i></i></span>
      <h1 class="site-title">Banner's Blog</h1>
      <span class="logo-line-after"><i></i></span>
    </a>
      <p class="site-subtitle" itemprop="description">Dream big work hard</p>
  </div>

  <div class="site-nav-right">
    <div class="toggle popup-trigger">
        <i class="fa fa-search fa-fw fa-lg"></i>
    </div>
  </div>
</div>




<nav class="site-nav">
  <ul id="menu" class="main-menu menu">
        <li class="menu-item menu-item-home">

    <a href="/blog/" rel="section"><i class="home fa-fw"></i>Home</a>

  </li>
        <li class="menu-item menu-item-categories">

    <a href="/blog/categories/" rel="section"><i class="th fa-fw"></i>Categories</a>

  </li>
        <li class="menu-item menu-item-tags">

    <a href="/blog/tags/" rel="section"><i class="tags fa-fw"></i>Tags</a>

  </li>
        <li class="menu-item menu-item-about">

    <a href="https://wangc1106.github.io/" rel="section"><i class="user fa-fw"></i>About</a>

  </li>
      <li class="menu-item menu-item-search">
        <a role="button" class="popup-trigger"><i class="fa fa-search fa-fw"></i>Search
        </a>
      </li>
  </ul>
</nav>



  <div class="search-pop-overlay">
    <div class="popup search-popup">
        <div class="search-header">
  <span class="search-icon">
    <i class="fa fa-search"></i>
  </span>
  <div class="search-input-container">
    <input autocomplete="off" autocapitalize="off"
           placeholder="Searching..." spellcheck="false"
           type="search" class="search-input">
  </div>
  <span class="popup-btn-close">
    <i class="fa fa-times-circle"></i>
  </span>
</div>
<div id="search-result">
  <div id="no-result">
    <i class="fa fa-spinner fa-pulse fa-5x fa-fw"></i>
  </div>
</div>

    </div>
  </div>

</div>
    </header>

    


    <main class="main">
      <div class="main-inner">
        <div class="content-wrap">
          

          <div class="content post posts-expand">
            

    
  
  
  <article itemscope itemtype="http://schema.org/Article" class="post-block" lang="zh-Hans">
    <link itemprop="mainEntityOfPage" href="https://wangc1106.github.io/blog/2022/03/02/2022-03-02-HiFi-3-software/">

    <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
      <meta itemprop="image" content="/blog/images/banner.png">
      <meta itemprop="name" content="Wang C">
      <meta itemprop="description" content="Make more time">
    </span>

    <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
      <meta itemprop="name" content="Banner's Blog">
    </span>
      <header class="post-header">
        <h1 class="post-title" itemprop="name headline">
          HiFi-3-assemble-software
        </h1>

        <div class="post-meta">
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-calendar"></i>
              </span>
              <span class="post-meta-item-text">Posted on</span>

              <time title="Created: 2022-03-02 09:08:25" itemprop="dateCreated datePublished" datetime="2022-03-02T09:08:25+08:00">2022-03-02</time>
            </span>
              <span class="post-meta-item">
                <span class="post-meta-item-icon">
                  <i class="far fa-calendar-check"></i>
                </span>
                <span class="post-meta-item-text">Edited on</span>
                <time title="Modified: 2022-03-06 23:12:39" itemprop="dateModified" datetime="2022-03-06T23:12:39+08:00">2022-03-06</time>
              </span>
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-folder"></i>
              </span>
              <span class="post-meta-item-text">In</span>
                <span itemprop="about" itemscope itemtype="http://schema.org/Thing">
                  <a href="/blog/categories/Software/" itemprop="url" rel="index"><span itemprop="name">Software</span></a>
                </span>
            </span>

          
            <span id="/blog/2022/03/02/2022-03-02-HiFi-3-software/" class="post-meta-item leancloud_visitors" data-flag-title="HiFi-3-assemble-software" title="Views">
              <span class="post-meta-item-icon">
                <i class="fa fa-eye"></i>
              </span>
              <span class="post-meta-item-text">Views: </span>
              <span class="leancloud-visitors-count"></span>
            </span>
            <span class="post-meta-item" title="Symbols count in article">
              <span class="post-meta-item-icon">
                <i class="far fa-file-word"></i>
              </span>
              <span>22k</span>
            </span>
            <span class="post-meta-item" title="Reading time">
              <span class="post-meta-item-icon">
                <i class="far fa-clock"></i>
              </span>
              <span>20 mins.</span>
            </span>

        </div>
      </header>

    
    
    
    <div class="post-body" itemprop="articleBody">

      
        <p>Hifiasm is a <font color="blue"><strong>fast haplotype-resolved de novo assembler</strong></font> for <font color="blue"><strong>PacBio HiFi reads(High Fidelity,high accuracy consensus reads)</strong></font>. HiCanu, Hifiasm, BUSCO, Peregrine, minimap2</p>
<span id="more"></span>



<h1 id="HiFi-快-准-狠"><a href="#HiFi-快-准-狠" class="headerlink" title="HiFi 快 准 狠"></a><a target="_blank" rel="noopener" href="https://zhuanlan.zhihu.com/p/138395531">HiFi 快 准 狠</a></h1><h2 id="组装快"><a href="#组装快" class="headerlink" title="组装快"></a>组装快</h2><p>CLR或Nanopore长Reads测序准确性较低，进行组装时需要利用高准确性Illumina短Reads纠错。而HiFi Reads本身即高准确性长Reads，可直接拼接，因此大大节省运算资源和时间。第一篇利用HiFi进行人基因组组装的报道表明，<font color="red"><strong>HiFi Reads组装速度比CLR Reads快10–100倍</strong></font> <a target="_blank" rel="noopener" href="https://pubmed.ncbi.nlm.nih.gov/31406327/">[1]</a>。另一项同时利用HiFi 和CLR 对人基因组组装的比较中，HiFi 组装仅花费约2,800个CPU时，而CLR 组装花费超过50,000个CPU时，利用Nanopore Reads进行组装更是花费约151,000个CPU时 <a target="_blank" rel="noopener" href="https://pubmed.ncbi.nlm.nih.gov/31711268/">[2]</a>。27 Gb六倍体加州红杉超大基因组，利用HiFi Reads两周就完成组装。</p>
<h2 id="组装准"><a href="#组装准" class="headerlink" title="组装准"></a>组装准</h2><h3 id="1-连续性与CLR和Nanopore相当"><a href="#1-连续性与CLR和Nanopore相当" class="headerlink" title="1 连续性与CLR和Nanopore相当"></a>1 连续性与CLR和Nanopore相当</h3><p>虽HiFi Reads在<font color="red"><strong>读长打折扣</strong></font>，但在<font color="red"><strong>连续性方面毫不逊色</strong></font>于PacBio CLR和Nanopore Reads。同时利用HiFi 和CLR 对人基因组组装的比较中，<font color="red"><strong>HiFi 和CLR 组装的Contigs N50相当</strong></font>（25.52 Mb和29.26 Mb）。一般认为<font color="red">Nanopore ultra-long</font>在基因组组装<font color="red"><strong>连续性更具优势</strong></font>，但同时利用HiFi 和Nanopore 进行水稻组装的报道中，三个组装软件中两个的组装结果都显示<font color="red"><strong>HiFi 组装Contigs N50高于Nanopore Reads</strong></font> <a target="_blank" rel="noopener" href="https://pubmed.ncbi.nlm.nih.gov/33319909/">[3]</a>。</p>
<h3 id="2-组装质量显著提升"><a href="#2-组装质量显著提升" class="headerlink" title="2 组装质量显著提升"></a>2 组装质量显著提升</h3><p>由于HiFi 与Illunima短Reads具有相当的碱基准确性，因此在组装上具有CLR和Nanopore无法比拟的优势。利用HiFi 对人基因组组装结果表明，<font color="red"><strong>HiFi 组装质量比CLR高6倍，比经Illumina校正的Nanopore高77倍</strong></font> <a target="_blank" rel="noopener" href="https://pubmed.ncbi.nlm.nih.gov/31406327/">[1]</a>。</p>
<p>值得一提，利用HiFi 和Nanopore 进行水稻组装文章表明，<font color="red"><strong>对于高重复区域利用高质量Illumina对Nanopore长Reads纠错效果一般</strong></font>。原因在于Illumina<font color="blue">短Reads在重复区域很难准确比对</font>，导致<strong>覆盖深度有限</strong>，因而<strong>无法起到准确纠错的作用</strong>。而HiFi 兼具长读长和高准确性双重优势，对<font color="red"><strong>重复区域</strong>仍然能够获得<strong>较高</strong>的组装<strong>质量</strong></font>。</p>
<h3 id="3-对重复区域和着丝粒区域具有较高的分辨率"><a href="#3-对重复区域和着丝粒区域具有较高的分辨率" class="headerlink" title="3 对重复区域和着丝粒区域具有较高的分辨率"></a>3 对重复区域和着丝粒区域具有较高的分辨率</h3><p>除了整个基因组的连续性和准确性，HiFi 在<font color="red"><strong>重复区域</strong>和<strong>着丝粒区域</strong></font>的组装也表现不俗。对人基因组的组装表现上，HiFi 组装对<font color="red"><strong>散在重复</strong>的<strong>分辨率为43%</strong></font>，<font color="red"><strong>高于</strong> CLR和已有的Nanopore版本</font>。对于<font color="red"><strong>数百拷贝的串联重复</strong></font>，HiFi 组装<font color="red">比CLR具有<strong>更高</strong>的<strong>分辨率</strong>和<strong>准确性</strong></font>。而专门针对HiFi Reads组装优化的<font color="blue"><strong>HiCanu</strong></font>在<font color="blue"><strong>重复序列和着丝粒组装</strong></font>方面更有良好表现。</p>
<p><img src="/blog/2022/03/02/2022-03-02-HiFi-3-software/9.png" alt="img"></p>
<p>HiFi和CLR对<strong>串联重复区域组装</strong>结果的比较 <a target="_blank" rel="noopener" href="https://pubmed.ncbi.nlm.nih.gov/31711268/">[2]</a></p>
<p><strong>variable number of tandem repeats (VNTRs)</strong></p>
<blockquote>
<p>b) Dot plot of a <strong>6.7 kbp VNTR</strong> in the <strong>intron</strong> of <em>RTEL1</em> (chr20:63693361–63693833) (top panel), <strong>which was resolved in the HiFi assembly only</strong>. The <strong>CLR assembly</strong> contained a <strong>gap</strong> over this region. The overall structure and length of this VNTR was <strong>supported by the ONT reads</strong> mapping to this location, which averaged 5,956 +/− 1799 bp (n = 5 ultra-long ONT reads), placing the HiFi sequence length at &lt;1 standard deviation away from the average ONT read. The motif homology plot (bottom panel) indicates that the content of the <em>RTEL1</em> VNTR is relatively pure, with an average sequence identity to the 15-mer repeat unit of 94.49% across the 439 copies.</p>
</blockquote>
<blockquote>
<p>c)Dot plot of the <strong>zinc finger protein gene <em>ZNF717</em></strong> (GRCh38 coordinates: chr3:75777127–75780446) (top panel), which was <strong>collapsed in the CLR</strong> assembly but <strong>fully represented in the HiFi assembly</strong>. The number of copies of this <strong>35 bp repeat unit</strong> increased from <strong>15 in the CLR</strong> assembly to <strong>89 in the HiFi</strong> assembly. The large amount of variation between individual copies of this VNTR is shown in the region between the <strong>red lines</strong> in the motif homology plots (bottom panels). The <strong>level of purity within the VNTR increased</strong> from 80.38% sequence identity in the CLR assembly to 90.75% sequence identity in the HiFi assembly. The red vertical lines indicate the start and end position of the VNTR.</p>
</blockquote>
<h2 id="狠"><a href="#狠" class="headerlink" title="狠"></a>狠</h2><h3 id="1-简单复杂基因组通吃"><a href="#1-简单复杂基因组通吃" class="headerlink" title="1 简单复杂基因组通吃"></a><strong>1 简单复杂基因组通吃</strong></h3><p>PacBio官方，除了已报道的<strong>人、水稻、果蝇和跗库蚊</strong>，目前HiFi 组装已经在<strong>人、大麻、燕麦、四倍体月季和六倍体加州红杉</strong>等物种中应用。即便是<strong>大麻、燕麦、月季、加州红杉</strong>这些<strong>复杂</strong>基因组，HiFi 组装不俗。11 Gb燕麦的Contig N50高达<strong>20 Mb</strong>！高重复高杂合的多倍体裸子植物的加州红杉超大基因组，HiFi 的Contig N50也达到1.92 Mb，远高于已报道其他裸子植物基因组。</p>
<h3 id="2-高质量单倍型基因组组装"><a href="#2-高质量单倍型基因组组装" class="headerlink" title="2 高质量单倍型基因组组装"></a><strong>2 高质量单倍型基因组组装</strong></h3><p>组装<strong>多倍体</strong>时，组装软件必须能<strong>区分不同的等位基因</strong>，并<strong>保存为不同的序列</strong>。用HiCanu对人基因组的HiFi Reads组装结果包含<strong>超过2 Gb的替代Contig</strong>，而<strong>其它软件</strong>只能产生<strong>不到400 Mb</strong>的<strong>替代Contig</strong>。说明HiCanu结合HiFi Reads具有更强的<strong>区分单倍型的能力</strong>。并且，HiCanu组装的<font color="blue">主要Contig</font>和<font color="blue">替代Contig</font>都具有较高的<font color="red">BUSCO完整性</font>（&gt; <strong>94％</strong>和&gt; 75％），远<strong>高</strong>于<strong>Nanopore ultra-long Reads</strong>组装的BUSCO完整性（分别只有**63%<strong>和</strong>0.3%**）。</p>
<h1 id="hifiasm-Official-doc"><a href="#hifiasm-Official-doc" class="headerlink" title="hifiasm Official doc"></a><a target="_blank" rel="noopener" href="https://github.com/chhylp123/hifiasm">hifiasm Official doc</a></h1><p><strong>install</strong></p>
<figure class="highlight sh"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># Install hifiasm (requiring g++ and zlib)</span></span><br><span class="line">git <span class="built_in">clone</span> https://github.com/chhylp123/hifiasm</span><br><span class="line"><span class="comment">#download zipfile unzip</span></span><br><span class="line"><span class="built_in">cd</span> hifiasm &amp;&amp; make</span><br></pre></td></tr></table></figure>

<p><strong>simple use</strong></p>
<figure class="highlight sh"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># Run on test data (use -f0 for small datasets)?</span></span><br><span class="line">wget https://github.com/chhylp123/hifiasm/releases/download/v0.7/chr11-2M.fa.gz</span><br><span class="line">./hifiasm -o <span class="built_in">test</span> -t4 -f0 chr11-2M.fa.gz 2&gt; test.log</span><br><span class="line">【awk <span class="string">&#x27;/^S/&#123;print &quot;&gt;&quot;$2;print $3&#125;&#x27;</span> test.bp.p_ctg.gfa &gt; test.p_ctg.fa】  <span class="comment"># get primary contigs in FASTA</span></span><br><span class="line"></span><br><span class="line"><span class="comment"># Assemble inbred/homozygous genomes (-l0 disables duplication purging)</span></span><br><span class="line"><span class="comment">#?</span></span><br><span class="line">hifiasm -o CHM13.asm -t32 -l0 CHM13-HiFi.fa.gz 2&gt; CHM13.asm.log</span><br><span class="line"><span class="comment"># Assemble heterozygous genomes with built-in duplication purging</span></span><br><span class="line"><span class="comment">#?</span></span><br><span class="line">hifiasm -o HG002.asm -t32 HG002-file1.fq.gz HG002-file2.fq.gz</span><br><span class="line"></span><br><span class="line"><span class="comment"># Hi-C phasing with paired-end short reads in two FASTQ files</span></span><br><span class="line"><span class="comment">#add Hi-C data</span></span><br><span class="line">hifiasm -o HG002.asm --h1 read1.fq.gz --h2 read2.fq.gz HG002-HiFi.fq.gz</span><br><span class="line"></span><br><span class="line"><span class="comment"># Trio binning assembly (requiring https://github.com/lh3/yak)</span></span><br><span class="line"><span class="comment">#combine parent sequencing data</span></span><br><span class="line">yak count -b37 -t16 -o pat.yak &lt;(cat pat_1.fq.gz pat_2.fq.gz) &lt;(cat pat_1.fq.gz pat_2.fq.gz)</span><br><span class="line">yak count -b37 -t16 -o mat.yak &lt;(cat mat_1.fq.gz mat_2.fq.gz) &lt;(cat mat_1.fq.gz mat_2.fq.gz)</span><br><span class="line">hifiasm -o HG002.asm -t32 -1 pat.yak -2 mat.yak HG002-HiFi.fa.gz</span><br></pre></td></tr></table></figure>

<h2 id="introduction"><a href="#introduction" class="headerlink" title="introduction"></a>introduction</h2><p>Hifiasm is a <font color="blue"><strong>fast haplotype-resolved de novo assembler</strong></font> for <font color="blue"><strong>PacBio HiFi reads</strong></font>. It can <strong>assemble a human genome in several hours</strong> and assemble a <strong>~30Gb</strong> California redwood genome in <strong>a few days</strong>. Hifiasm emits <font color="red"><strong>partially phased assemblies</strong></font> of quality competitive with the best assemblers. Given <font color="green"><strong>parental short reads</strong></font> or <font color="green"><strong>Hi-C data</strong></font>, it produces arguably <font color="red"><strong>the best haplotype-resolved assemblies</strong></font> so far.</p>
<h2 id="Why-Hifiasm"><a href="#Why-Hifiasm" class="headerlink" title="Why Hifiasm?"></a>Why Hifiasm?</h2><ul>
<li>Hifiasm delivers high-quality assemblies. It tends to generate <font color="blue">longer contigs</font> and resolve <font color="blue">more segmental duplications</font> than other assemblers.</li>
<li>Given <font color="blue">Hi-C reads</font> or <font color="blue">short reads from the parents</font>, hifiasm can produce overall <font color="blue"><strong>the best haplotype-resolved assembly</strong></font> so far. It is the assembler of choice by the <a target="_blank" rel="noopener" href="https://humanpangenome.org/">Human Pangenome Project</a> for the first batch of samples.</li>
<li>Hifiasm can <font color="blue">purge duplications?</font> between haplotigs without relying on third-party tools such as purge_dups. Hifiasm does <font color="blue">not need polishing tools?</font> like pilon or racon, either. This simplifies the assembly pipeline and saves running time.</li>
<li>Hifiasm is <font color="blue">fast</font>. It can assemble a human genome in half a day and assemble a ~30Gb redwood genome in three days. <font color="blue">No genome is too large for hifiasm</font>.</li>
<li>Hifiasm is trivial to install and easy to use. It does not required Python, R or C++11 compilers, and can be compiled into a single executable. The default setting works well with a variety of genomes.</li>
</ul>
<h1 id="hifiasm-paper"><a href="#hifiasm-paper" class="headerlink" title="hifiasm paper"></a>hifiasm paper</h1><p>ref：<a target="_blank" rel="noopener" href="https://www.jianshu.com/p/6d79690dce5d">https://www.jianshu.com/p/6d79690dce5d</a></p>
<p><a target="_blank" rel="noopener" href="https://hifiasm.readthedocs.io/en/latest/index.html">https://hifiasm.readthedocs.io/en/latest/index.html</a></p>
<p><a target="_blank" rel="noopener" href="https://links.jianshu.com/go?to=https://github.com/chhylp123/hifiasm">https://github.com/chhylp123/hifiasm</a></p>
<p><a target="_blank" rel="noopener" href="https://links.jianshu.com/go?to=https://www.nature.com/articles/s41592-020-01056-5">https://www.nature.com/articles/s41592-020-01056-5</a></p>
<h2 id="原理介绍1"><a href="#原理介绍1" class="headerlink" title="原理介绍1"></a>原理介绍1</h2><p>hifiasm有效利用PacBio HiFi测序技术，在<font color="red"><strong>分型组装图(pahsed assembly gprah)</strong></font>中可靠的表示单倍体信息。</p>
<p><strong>第一阶段</strong>：通过所有序列相互比对，对潜在测序错误纠正。如果<font color="red">一个位置只存在两种碱基类型</font>，且每个碱基类型<font color="red">至少有3条read支持</font>，那么这个位置会被当作<font color="red">杂合位点</font>，否则视作测序错误纠正。</p>
<p><strong>第二阶段</strong>：根据<font color="red">序列之间重叠关系</font>，构建<font color="red">分型的字符串图(phased string graph)</font>。其中调整朝向的序列作为<font color="red">顶点(vertex)</font>，一致重叠作为<font color="red">边(edge)</font>。字符串图中的<font color="red">气泡(bubble)</font>则是<font color="red">杂合位点</font>。</p>
<p><strong>第三阶段</strong>：如果没有额外的信息，hifiasm会<font color="red">随机选择气泡一边</font>构建primary assembly，另一边则是alternate assembly. 该策略和HiCanu，Falcon-Unzip一样。对于杂合基因组而言，由于存在一个以上的纯合haplotype，因此<font color="red">primary assembly可能还会包含haplotigs</font>。HiCanu依赖第三方的purge_dups, 而hifiasm内部实现purge_dups算法的变种，简化了流程。如果有<font color="red">额外信息</font>，hifiasm可正确的对<font color="red">haplotype进行分型</font>。</p>
<p><img src="/blog/2022/03/02/2022-03-02-HiFi-3-software/1.png" alt="img"></p>
<h2 id="hifiasm-原理介绍2"><a href="#hifiasm-原理介绍2" class="headerlink" title="hifiasm 原理介绍2"></a>hifiasm 原理介绍2</h2><p><strong>可识别单倍型的纠错</strong></p>
<p>Hifiasm将所有hifi reads读取到内存中进行<font color="red"><strong>all-vs-all比对并纠错</strong></font>。基于reads间overlap信息，如果read上有一个碱基与其他碱基不同，并有<font color="red"><strong>至少3条reads支持</strong></font>，则认为它是<font color="red"><strong>SNP</strong></font>并保留，<strong>否则</strong>认为是错误并<strong>纠正</strong>。</p>
<p><strong>组装图的构建</strong></p>
<p>在校正之后，大多数错误被去除，同时<font color="red"><strong>杂合变异信息被保留</strong></font>。基于这些信息，Hifiasm构建了<font color="red"><strong>以reads为顶点、重叠区为边的定相string-graph</strong></font>。</p>
<p><strong>组装序列的生成</strong></p>
<ul>
<li><p>如果<font color="red">没有其他数据</font>，Hifiasm在输出序列时会<font color="red"><strong>任意选择</strong>每个气泡<strong>bubble</strong>的<strong>一侧输出</strong>类似<strong>Falcon unzip</strong>和<strong>HiCanu</strong>的主要组装结果</font>（primary contigs）。</p>
</li>
<li><p>如果同时有<font color="red">父母本的测序数据</font>，Hifiasm可以通过<font color="red">亲本特有的kmer</font>在图上<font color="red">识别来自父母本的序列</font>，得到<font color="red"><strong>两套单倍体</strong>基因组</font>。</p>
</li>
<li><p>当然HiFiasm文章中也提到：</p>
<p>与其他基于图形的汇编程序不同，HiFiasm致力于<strong>保持所有单倍型的连续性。</strong><br><font color="red"><strong>HiCanu</strong>只试图<strong>保持一个亲本单倍型</strong>的<strong>连续性</strong></font>，并且经常<font color="red">破坏另一个单倍型的连续性</font>，当分离亲本单倍型时，这些突变点将导致单倍型分解的碎片—HiCanu<font color="red">没有充分利用HiFi Reads</font><br>Hifiasm针对HiFi特点开发，在hifi数据的组装表现上较同类软件更为突出，在多个基因组上表现出了更高的<strong>准确性</strong>和组装的<strong>连续性</strong>。</p>
</li>
</ul>
<h2 id="仅HiFi数据模式"><a href="#仅HiFi数据模式" class="headerlink" title="仅HiFi数据模式"></a>仅HiFi数据模式</h2><p>最基本的用法，会得到两个部分分型的组装</p>
<figure class="highlight sh"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">wget https://github.com/chhylp123/hifiasm/releases/download/v0.7/chr11-2M.fa.gz</span><br><span class="line">hifiasm -o <span class="built_in">test</span> -t 32 chr11-2M.fa.gz 2&gt; test.log</span><br><span class="line"><span class="comment">#-o输出文件前缀 -t线程</span></span><br></pre></td></tr></table></figure>

<p>结果文件关注如下几项 (prefix为前缀）</p>
<ul>
<li><code>prefix.bp.r_utg.gfa</code>: haplotype-resolved raw unitig graph，记录所有的单倍型信息</li>
<li><code>prefix.bp.p_utg.gfa</code>: 在raw unitig graph基础上过滤小的bubble，</li>
<li><code>prefix.bp.p_ctg.gfa</code>: 主要contig的assembly graph</li>
<li><code>prefix.bp.hap1.p_ctg.gfa</code>: haplotype1的部分分型的contig graph</li>
<li><code>prefix.bp.hap2.p_ctg.gfa</code>: haplotype2的部分分型的contig graph</li>
</ul>
<p>如果并不需要部分分型的组装，只要primary和alternate的组装结果，可以在之前命令的基础上，加上 <code>--primary</code>参数。</p>
<figure class="highlight sh"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">hifiasm --primary -o <span class="built_in">test</span> -t 32 chr11-2M.fa.gz 2&gt; test.log2</span><br></pre></td></tr></table></figure>

<p>由于hifiasm运行时会将步骤中纠错和相互比对的结果保存成 bin 文件，因此rerun速度很快(前缀名差异)。</p>
<p>primay模式下输出的文件和之前的类似，唯一的不同在于没有 <strong>bp</strong></p>
<p>prefix<code>.r_utg.gfa</code>: haplotype-resolved raw unitig graph: 保留了组装生成的所有单体型信息，包括体细胞突变和重复的测序错误。</p>
<p>prefix<code>.p_utg.gfa</code>: haplotype-resolved processed unitig graph without small bubbles：无小气泡的单倍型解析；去掉由于体细胞突变和数据背景噪音引起的small bubbles（这个并不是真正的单体型信息），对于高度杂合基因组物种优先选择这个结果。</p>
<p>prefix<code>.p_ctg.gfa</code>: assembly graph of primary contigs：对于低杂合度物种来说，优先选择该文件；对于高杂合度物种，该结果代表其中一个单倍型。</p>
<p>prefix<code>.a_ctg.gfa</code>: assembly graph of alternate contigs：组装出来的另一套单体型基因组结果。</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#contig，通过awk提取</span></span><br><span class="line">awk <span class="string">&#x27;/^S/&#123;print &quot;&gt;&quot;$2;print $3&#125;&#x27;</span> test.p_ctg.gfa &gt; test.p_ctg.fa</span><br></pre></td></tr></table></figure>

<h2 id="双亲二代测序Trio-binning模式"><a href="#双亲二代测序Trio-binning模式" class="headerlink" title="双亲二代测序Trio-binning模式"></a>双亲二代测序Trio-binning模式</h2><p>如果有双亲数据，则可使用trio-binning方法进行更可靠的分型。分为两步，<font color="red"><strong>先用yak统计k-mers， 然后用hifiasm进行组装</strong></font></p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">yak count -k31 -b37 -t16 -o pat.yak paternal.fq.gz</span><br><span class="line">yak count -k31 -b37 -t16 -o mat.yak maternal.fq.gz</span><br><span class="line">hifiasm -o NA12878.asm -t 32 -1 pat.yak -2 mat.yak NA12878.fq.gz</span><br></pre></td></tr></table></figure>

<p>输出和之前类似，主要关注文件名<font color="red">带 dip 的输出gfa文件</font></p>
<ul>
<li><strong>prefix.r_utg.gfa</strong>(Haplotype-resolved raw unitig graph in GFA format):保存了所有的单倍型信息。</li>
</ul>
<ul>
<li><code>prefix.dip.hap1.p_ctg.gfa</code>: 完成分型的<font color="red"><strong>父源单倍体 contig图</strong></font>。(Phased paternal/haplotype1 contig graph):保留了阶段性父系/单倍型1组装。</li>
<li><code>prefix.dip.hap2.p_ctg.gfa</code>: 完全分型的<font color="red"><strong>母源单倍体contig图</strong></font>。(Phased maternal/haplotype2 contig graph):保留了阶段性母系/单倍型2组装。</li>
</ul>
<h2 id="附加Hi-C模式"><a href="#附加Hi-C模式" class="headerlink" title="附加Hi-C模式"></a>附加Hi-C模式</h2><p>由于Hi-C数据能够提供<strong>远距信息</strong>，因此也能<strong>用于单倍体分型</strong>。需要加两个参数， h1为Hi-C的read1, h2 为Hi-C的read2</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">hifiasm -o NA12878.asm -t 32 --h1 read1.fq.gz --h2 read2.fq.gz HiFi-reads.fq.gz</span><br></pre></td></tr></table></figure>

<p>在该模式下，每个contig要么是来自于父亲，要么是来自于母亲。hifiasm将<font color="red">同一来源的contig放在同一个组装中</font>。需要注意的是，hifiasm未必能够处理好<font color="red">着丝粒附近区域</font>，另外<font color="red">hifiasm中Hi-C也不会用于进行scaffold</font>。</p>
<p>输出结果重点关注带hic的文件</p>
<ul>
<li><code>prefix.hic.p_ctg.gfa</code>: 主要contig的组装图</li>
<li><code>prefix.hic.hap1.p_ctg.gfa</code>: 完全分型的haplotype1的contig图</li>
<li><code>prefix.hic.hap2.p_ctg.gfa</code>: 完全分型的haplotype2的contig图</li>
<li><code>prefix.hic.a_ctg.gfa</code> : 如果设置了 <code>--primary</code>参数，还会输出该次要contig的组装图</li>
</ul>
<h2 id="日志和参数调整"><a href="#日志和参数调整" class="headerlink" title="日志和参数调整"></a>日志和参数调整</h2><p>绝大部分的时候，只需要使用默认参数即可得到相对比较好的结果。但是当默认参数无法达到目的，就需要检查日志信息，阅读相关参数从而优化结果。</p>
<p>日志信息主要分三项</p>
<ul>
<li><font color="red">k-mer图</font>: 纯合样本一个peak，杂合样本2个peak。</li>
<li><font color="red">纯合峰的覆盖度</font>: <code>[M::purge_dups] homozygous read coverage threshold: X</code> , 一般会由hifiasm自动推断。</li>
<li><font color="red">杂合/纯合碱基数目（Hi-C模式）</font>: Hi-C模式下，如果纯合的碱基数超过杂合碱基数，那么hifiasm就不容易找对纯合read的所在峰。</li>
</ul>
<p>对于日志信息，最主要关注<font color="red">k-mer图</font>，从而判断hifiasm是否能够正确的找到纯合峰，杂合峰所在位置。如果hifiasm没有找对纯合峰所在的位置，就需要<font color="red">根据k-mer图手动指定?</font> <code>--hom-cov</code>。</p>
<p>对于一个组装结果，最直接的<strong>评估标准</strong>就是<font color="red">基因组大小是否符合预期</font>，分型的<font color="red">两套基因组是否相差不大</font>，序列<strong>是否足够长</strong>，<strong>是否存在错误组装</strong>的情况。</p>
<p>如果基因组<font color="red"><strong>大小不符合预期</strong></font>，一般都是hifiasm<strong>找错纯合峰的位置</strong>，需要手动指定 <code>--hom-cov</code>；</p>
<p>如果分型的两套基因组<font color="red"><strong>差别过大</strong></font>，则通过降低 <code>-s</code> 调整。</p>
<p>如果<font color="red"><strong>序列不够长，片段化明显</strong></font>，则可尝试<font color="red">增加</font>  <code>-D</code> 和  <code>-N</code>, 虽增加运行时间，但会<font color="red">提高重复区域的分辨率</font>。</p>
<p>如果后续的Hi-C，或者BioNano发现hifiasm组装结果<font color="red"><strong>有比较多错误组装</strong></font>，则可以适当降低 <code>--purge-max</code>, <code>-s</code>和 <code>-O</code>。或者设置 <code>-u</code> 关闭post-join 步骤，hifiasm通过该步骤提高组装的连续性。</p>
<h2 id="输入格式-fq还是fa"><a href="#输入格式-fq还是fa" class="headerlink" title="输入格式 fq还是fa"></a>输入格式 fq还是fa</h2><figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line"># bam --&gt; fasta</span><br><span class="line">samtools view *.bam | awk &#x27;&#123;print &quot;&gt;&quot;$1&quot;\n&quot;$10&#125;&#x27; &gt; fasta</span><br><span class="line"># bam2fastq</span><br></pre></td></tr></table></figure>

<hr>
<h1 id="HiCanu-https-canu-readthedocs-io-en-latest-index-html"><a href="#HiCanu-https-canu-readthedocs-io-en-latest-index-html" class="headerlink" title="HiCanu[https://canu.readthedocs.io/en/latest/index.html]"></a>HiCanu[<a target="_blank" rel="noopener" href="https://canu.readthedocs.io/en/latest/index.html]">https://canu.readthedocs.io/en/latest/index.html]</a></h1><hr>
<p>ref <a target="_blank" rel="noopener" href="https://www.jianshu.com/p/c1aeeae77cb5">https://www.jianshu.com/p/c1aeeae77cb5</a></p>
<p>Canu分三个步骤，<font color="red"><strong>纠错</strong></font>，<font color="red"><strong>修整</strong></font>和<font color="red"><strong>组装</strong></font>，每一步<font color="red">步骤<strong>相似</strong></font>：</p>
<ul>
<li>加载read到read数据库，gkpStore</li>
<li>对k-mer进行记数，用于计算序列间的overlap</li>
<li>计算overlap</li>
<li>加载overlap到overlap数据库，OvlStore</li>
<li>根据read和overlap完成特定分析目标<ul>
<li>read<font color="blue"><strong>纠错</strong></font>时会从overlap中挑选一致性序列替换原始的噪声read</li>
<li>read<font color="blue"><strong>修整</strong></font>时会使用overlap确定read哪些区域是高质量区域，哪些区域质量较低需要修整。最后保留单个最高质量的序列块</li>
<li>序列<font color="blue"><strong>组装</strong></font>时根据一致的overlap对序列进行编排(layout), 最后得到contig</li>
</ul>
</li>
</ul>
<p>三步<font color="blue">可分开</font>运行，既可用<font color="blue">Canu纠错后结果</font>作为<font color="blue">其他组装软件输入</font>，也可将其他软件的纠错结果作为Canu的输入。</p>
<p><font color="red">几个<strong>全局参数</strong></font>：</p>
<ul>
<li><font color="red"><strong>genomeSize</strong></font>设置预估基因组大小，用于Canu<font color="blue">估计测序<strong>深度</strong></font>, The genome size should be your best guess of the haploid genome size of what is being assembled, It is used primarily to estimate coverage in reads, NOT as the desired assembly size.</li>
<li><font color="red">maxThreads</font>设置最大<font color="blue">线程</font></li>
<li><font color="red">rawErrorRate</font>设置<font color="blue">两个<strong>未纠错read之间</strong>最大<strong>期望差异碱基数</strong></font></li>
<li><font color="red">correctedErrorRate</font>设置<font color="blue"><strong>纠错后</strong>read之间最大<strong>期望差异碱基数</strong>，<strong>组装</strong>步骤<strong>多次调整</strong></font></li>
<li><font color="red">minReadLength</font>表示<font color="blue">只使用大于阈值的序列</font></li>
<li><font color="red">minOverlapLength</font>表示<font color="blue">Overlap最小长度</font></li>
<li>提高minReadLength可<strong>提</strong>高运行<strong>速</strong>度，增加minOverlapLength可<strong>降</strong>低<strong>假阳性overlap</strong></li>
</ul>
<p><font color="blue">测试数据</font> “<a target="_blank" rel="noopener" href="https://links.jianshu.com/go?to=https://www.nature.com/articles/s41467-018-03016-2">High contiguity Arabidopsis thaliana genome assembly with a single nanopore flow cell</a>“。 提供 <em>Arabidopsis thaliana</em> KBS-Mac-74 的30X短片段文库二代测序、PacBio和Nanopore的三代测序以及Bionano测序数据, 拟南芥基因组被认为是植物金标准，数据适合练习。项目编号”PRJEB21270”, 在European Nucleotide Archive上找到下载地址。</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line">## PacBio Sequal</span><br><span class="line">wget -c -q ftp://ftp.sra.ebi.ac.uk/vol1/ERA111/ERA1116568/bam/pb.bam</span><br><span class="line">## MinION</span><br><span class="line">wget -c -q ftp://ftp.sra.ebi.ac.uk/vol1/ERA111/ERA1116595/fastq/ont.fq.gz</span><br><span class="line"># Illuminia MiSeq</span><br><span class="line">wget -c -q ftp://ftp.sra.ebi.ac.uk/vol1/ERA111/ERA1116569/fastq/il_1.fq.gz</span><br><span class="line">wget -c -q ftp://ftp.sra.ebi.ac.uk/vol1/ERA111/ERA1116569/fastq/il_2.fq.gz</span><br></pre></td></tr></table></figure>

<p><font color="blue">PacBio数据以BAM格式存储</font>，通过安装PacBio的smrtlink工具套装，使用其中<font color="blue">bam2fasta</font>转换</p>
<figure class="highlight sh"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line"><span class="string">&#x27;build index for convert&#x27;</span></span><br><span class="line">~/opt/biosoft/smrtlink/smrtcmds/bin/pbindex pb.bam &amp;</span><br><span class="line"><span class="comment"># convert bam to fasta</span></span><br><span class="line">~/opt/biosoft/smrtlink/smrtcmds/bin/bam2fasta -o pb pb.bam &amp;</span><br></pre></td></tr></table></figure>

<p>PacBio的smrtlink工具套装大小1.4G，下载速度慢，安装手动确认各种选项, 好处是工具全。</p>
<h2 id="Install"><a href="#Install" class="headerlink" title="Install"></a>Install</h2><p>latest: <a target="_blank" rel="noopener" href="https://github.com/marbl/canu/releases/tag/v2.2">https://github.com/marbl/canu/releases/tag/v2.2</a></p>
<h2 id="Run-Canu"><a href="#Run-Canu" class="headerlink" title="Run Canu"></a>Run Canu</h2><p><strong>第一步</strong>：<font color="red"><strong>纠错</strong></font>。三代测序错误率高，原始数据充满噪音。通过<strong>序列</strong>间<strong>相互比较纠错得到高可信度的碱基</strong>。<font color="blue">主要<strong>调整两个参数</strong></font></p>
<ul>
<li><font color="blue">corOutCoverage</font>: 控制<strong>多少数据</strong>用于纠错。如拟南芥120M基因组，100X测序12G数据，如果只用最长6G数据纠错，参数设置为50(120m x 50)。<strong>设置大于测序深度的值</strong>如120表示<strong>使用所有数据</strong>。<font color="green"><strong>-pacbio-raw</strong></font></li>
</ul>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line">canu -correct \</span><br><span class="line">    -p ath -d pb_ath \</span><br><span class="line">    Threads=10 gnuplotTested=<span class="literal">true</span>\</span><br><span class="line">    genomeSize=120m minReadLength=2000 minOverlapLength=500\</span><br><span class="line">    corOutCoverage=120 corMinCoverage=2 \</span><br><span class="line">    -pacbio-raw pb.fasta.gz</span><br></pre></td></tr></table></figure>

<p>注: 有些服务器<strong>没有安装gnuplot</strong>, <strong>gnuplotTested=true</strong> 跳过检查。</p>
<p><strong>第二步</strong>：<font color="red"><strong>修整</strong></font>。为获取更高质量的序列，<font color="blue">移除可疑区域</font>（如残留的SMRTbell接头)。<font color="green"><strong>-pacbio-corrected</strong></font></p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line">canu -trim \</span><br><span class="line">        -p ath -d pb_ath</span><br><span class="line">        maxThreads=20 gnuplotTested=<span class="literal">true</span>\</span><br><span class="line">        genomeSize=120m minReadLength=2000 minOverlapLength=500\</span><br><span class="line">        -pacbio-corrected ath/pb_ath.correctedReads.fasta.gz</span><br></pre></td></tr></table></figure>

<p><strong>第三步</strong>: <font color="red"><strong>组装</strong></font>。前两步获得高质量序列后正式组装。 主要调整参数是<font color="red">纠错后的<strong>序列错误率 correctedErrorRate</strong></font>，它会影响utgOvlErrorRate。可尝试多个参数，因为速度较块。</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># error rate 0.035</span></span><br><span class="line">canu -assemble \</span><br><span class="line">    -p ath -d ath-erate-0.035 \</span><br><span class="line">    maxThreads=20 gnuplotTested=<span class="literal">true</span> \</span><br><span class="line">    genomeSize=120m\</span><br><span class="line">    correctedErrorRate=0.035 \</span><br><span class="line">    -pacbio-corrected atg/pb_ath.trimmedReads.fasta.gz</span><br><span class="line"><span class="comment"># error rate 0.050</span></span><br><span class="line">canu -assemble \</span><br><span class="line">    -p ath -d ath-erate-0.050 \</span><br><span class="line">    maxThreads=20 gnuplotTested=<span class="literal">true</span> \</span><br><span class="line">    genomeSize=120m\</span><br><span class="line">    correctedErrorRate=0.050 \</span><br><span class="line">    -pacbio-corrected atg/pb_ath.trimmedReads.fasta.gz</span><br></pre></td></tr></table></figure>

<p>输出文件<font color="red"><strong>ath.contigs.fasta</strong></font>结果文件。</p>
<h2 id="Debug"><a href="#Debug" class="headerlink" title="Debug"></a>Debug</h2><figure class="highlight sh"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#qsub: submit error (No default queue specified MSG=requested queue not found)</span></span><br><span class="line"><span class="comment">#invalid useGrid (1) and gridEngine (PBS); found no execution hosts - is grid available from this host?.</span></span><br></pre></td></tr></table></figure>

<p>issue: <a target="_blank" rel="noopener" href="https://github.com/marbl/canu/issues/1040">https://github.com/marbl/canu/issues/1040</a></p>
<figure class="highlight sh"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">advice : Canu uses the <span class="string">&quot;PBS command pbsnodes&quot;</span> to poll your available resources. On your compute nodes this is not working. See the FAQ: http://canu.readthedocs.io/en/latest/faq.html<span class="comment">#my-run-stopped-with-the-error-failed-to-submit-batch-jobs</span></span><br><span class="line">If your nodes <span class="string">&#x27;cannot access/submit&#x27;</span> to the <span class="string">&#x27;grid&#x27;</span> <span class="keyword">then</span> Canu <span class="string">&#x27;wont work with your grid system&#x27;</span>. These are grid configuration issues so youll have to check with your sysadmin <span class="keyword">if</span> it can be changed. If not, youll have to run Canu on a single node or use useGrid=remote from the head node <span class="built_in">which</span> will <span class="built_in">print</span> the commands <span class="keyword">for</span> you to manually run (see issue <span class="comment">#822).</span></span><br></pre></td></tr></table></figure>

<p><a target="_blank" rel="noopener" href="https://canu.readthedocs.io/en/latest/faq.html#my-run-stopped-with-the-error-failed-to-submit-batch-jobs">canu FAQ</a></p>
<p><strong>How do I run Canu on my SLURM / SGE / PBS / LSF / Torque system?</strong></p>
<figure class="highlight sh"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line">Canu will <span class="string">&#x27;detect and configure&#x27;</span> itself to use on most <span class="string">&#x27;grids&#x27;</span>. You <span class="keyword">do</span> not need to submit Canu to the grid <span class="keyword">in</span> this <span class="keyword">case</span>. Canu will query the system <span class="keyword">for</span> <span class="string">&#x27;grid support&#x27;</span>, configure itself <span class="keyword">for</span> the machines <span class="string">&#x27;available in the grid&#x27;</span>, <span class="keyword">then</span> submit itself to the grid <span class="keyword">for</span> execution. You can <span class="keyword">then</span> monitor the run <span class="string">&#x27;using your grid&#x27;</span>.</span><br><span class="line"></span><br><span class="line">Canu will <span class="string">&#x27;NOT request explicit time limits or queues/partitions&#x27;</span>. You can supply your own grid options, such as a partition on SLURM, an account code on SGE, and/or time limits with gridOptions=<span class="string">&quot;&lt;your options list&gt;&quot;</span> <span class="built_in">which</span> will passed to every job submitted by Canu. Similar options exist <span class="keyword">for</span> every stage of Canu, <span class="built_in">which</span> could be used to, <span class="keyword">for</span> example, restrict overlapping to a specific partition or queue.</span><br><span class="line"></span><br><span class="line">【【【<span class="string">&#x27;To disable grid support and run only on the local machine, specify useGrid=false&#x27;</span>】】】</span><br><span class="line"></span><br><span class="line">It is possible to <span class="built_in">limit</span> the number of grid <span class="built_in">jobs</span> running at the same time, but this isn’t directly supported by Canu. The various gridOptions parameters can pass grid-specific parameters to the submit commands used; see Issue <span class="comment">#756 for Slurm and SGE examples.</span></span><br></pre></td></tr></table></figure>

<figure class="highlight sh"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#common options:</span></span><br><span class="line">useGrid=string</span><br><span class="line">      - Run under grid control (<span class="literal">true</span>), <span class="string">&#x27;locally (false)&#x27;</span>, or <span class="built_in">set</span> up <span class="keyword">for</span> grid control</span><br><span class="line">        but dont submit any <span class="built_in">jobs</span> (remote)</span><br></pre></td></tr></table></figure>



<h2 id="Advice"><a href="#Advice" class="headerlink" title="Advice"></a>Advice</h2><h3 id="后续分析要去冗余"><a href="#后续分析要去冗余" class="headerlink" title="后续分析要去冗余"></a>后续分析要去冗余</h3><p>Canu2.0之前的contig尽管运行日志说没有bubble，其实是没有检测到。Canu2.0才加上该信息。但作者强烈推荐先用purge_dups去冗余，避免软件难以检测到的冗余序列存在影响后续分析。</p>
<p>作者：旧版Canu没有真正检测到bubble，因此contig包含着buble，建议后续分析用purge_dups 处理这些冗余序列</p>
<p>Canu改动基本上是增加新功能，提高组装准确性，提高组装速度，常规bug修复。</p>
<p>1.8版本更改了Nanopore的错误率，提高10倍左右组装速度，同时完整支持了trio-binning</p>
<p>1.9版开始增加PacBio的HiFi数据，极大提高相互比对速度</p>
<p>这些修改，可以直接将组装得到的contig用于后续分析</p>
<p>而2.0做了许多优化用来保证contig组装更长更准确，其中一个是在contig构建期间会检测bubbles, 防止他们打断杂合基因组</p>
<p>例如组装发现染色体明显偏大，用新Canu组装该物种时候发现，前后两次组装结果在bubbles一栏有明显差别。1.8版本是0，2.0版本是136，约6M基因组大小。</p>
<p>查看之前所有组装物种的日志，发现bubble一栏都是0（即便是杂合基因组）。</p>
<p>意味着，2.0版本装的基因组不能直接用于下游的HiC，而需要先把bubble这部分序列给过滤掉，不然基因组偏离实际值。</p>
<h3 id="高杂合物种组装"><a href="#高杂合物种组装" class="headerlink" title="高杂合物种组装"></a>高杂合物种组装</h3><p>高杂合物种组装，Canu建议是用 <code>batOptions=-dg 3 -db 3 -dr 1 -ca 500 -cp 50</code>参数尽量分出两套单倍型，然后对基因组去冗余。</p>
<p><code>batOptions</code>表示传递后续的参数给组装软件<code>bogart</code>, <code>-dg 3 -db3</code>降低自动确定阈值时的错误率离差(deviation)，从而更好分开单倍型。<code>-dr 1 -ca 500 -cp 50</code>会影响错误组装的拆分，对于一个模棱两可的contig，如果另一条可选路径的overlap长度至少是500bp，或另一条可选路径时在长度上和当前最佳路径存在50%的差异，那么将contig拆分。</p>
<p>关于杂合物种组装的讨论，参考<a target="_blank" rel="noopener" href="https://links.jianshu.com/go?to=https://github.com/marbl/canu/issues/201%23issuecomment-233750764">https://github.com/marbl/canu/issues/201#issuecomment-233750764</a></p>
<h3 id="购买SSD避免服务器IO瓶颈"><a href="#购买SSD避免服务器IO瓶颈" class="headerlink" title="购买SSD避免服务器IO瓶颈"></a>购买SSD避免服务器IO瓶颈</h3><p>如果服务器线程数很多，在普通机械硬盘上组装，而且系统还是CentOS，那么需要调整一个参数，避免其中一步的IO严重影响服务器性能。</p>
<p>Canu通过两个策略进行并行，bucketizing (‘ovb’ 任务) 和 sorting (‘ovs’ 任务)。 bucketizing会从1-overlap读取输出的overlap，将他们<strong>复制</strong>一份作为中间文件。sorting一步将这些文件加载到内存中进行排序然后<strong>写出</strong>到硬盘上。 如果你的overlap输出特别多，那么该步骤将会瞬间挤爆的你的IO.</p>
<p>为了避免悲剧发生，请增加如下参数: <code>ovsMemory=16G ovbConcurrency=15 ovsConcurrency=15</code>， 也就是降低这两步同时投递的任务数，缓解IO压力。</p>
<hr>
<h1 id="HiFI组装实例"><a href="#HiFI组装实例" class="headerlink" title="HiFI组装实例"></a>HiFI组装实例</h1><p>Contiguity, Completeness, and Correctness</p>
<h2 id="2018-10"><a href="#2018-10" class="headerlink" title="2018.10"></a>2018.10</h2><p>PacBio公司 高保真 单分子 长读长 测序【HiFi】模式</p>
<h2 id="2019-01"><a href="#2019-01" class="headerlink" title="2019.01"></a>2019.01</h2><p>首篇利用HiFi数据进行人基因组 结构变异检测 和 基因组组装 文章发布</p>
<h2 id="2020-01"><a href="#2020-01" class="headerlink" title="2020.01"></a>2020.01</h2><p>HiFi在 四倍体玫瑰 燕麦 大麻 白蛉 等多个物种基因组组装</p>
<h2 id="2020-03-27Gb六倍体加州红杉"><a href="#2020-03-27Gb六倍体加州红杉" class="headerlink" title="2020.03 27Gb六倍体加州红杉"></a>2020.03 27Gb六倍体加州红杉</h2><p><a target="_blank" rel="noopener" href="https://mp.weixin.qq.com/s?__biz=MzkxMjIzNzY3Mw==&mid=2247489444&idx=2&sn=e6e9e4eebac756a0d518ee4a0fa09956&source=41#wechat_redirect">PacBio科学家利用HiFi测序两周内完成基因组高达27Gb的六倍体加州红杉基因组组装</a></p>
<h3 id="初衷"><a href="#初衷" class="headerlink" title="初衷"></a>初衷</h3><p><font color="blue">15Gb六倍体小麦</font>基因组，选择<font color="blue">HiFi还是CLR？</font><font color="blue">11Gb燕麦</font>基因组组装说明<font color="blue">HiFi组装<strong>超大基因组</strong>的能力</font>，<strong>Contig N50</strong>高达<strong>20Mb</strong>！HiFi Reads <font color="blue"><strong>读取长度</strong></font> 和 <font color="blue"><strong>准确性</strong></font> 之间的 <font color="blue"><strong>平衡</strong></font> 是否能克服最艰巨的基因组（大且复杂）挑战？</p>
<h3 id="测序过程"><a href="#测序过程" class="headerlink" title="测序过程"></a>测序过程</h3><p>斯坦福大学校园内采样，DNA提取文库构建。</p>
<p>使用<font color="blue">BluePippin[高分子量DNA核酸片段分离和回收]</font>将DNA片段选择大小设为<strong>15 kb</strong>，更长的插入片段允许HiFi读长达到<strong>50 kb</strong>。 HiFi reads 的准确率分布可以发现所有reads准确率<strong>都在Q20以上</strong>，HiFi reads<strong>最密集</strong>的插入片段在<strong>20k-30k区间</strong>，最长的插入片段达到<strong>50k精度都高于Q20</strong>(99％)，大约有<strong>超过一半的reads</strong>准确率接近或超过<strong>Q30</strong>(99.9%)。</p>
<p><strong>7天连续测序</strong>共获得<strong>606Gb</strong> HiFi数据，相当于<strong>22X</strong>。在之前很多使用HiFi进行组装的项目中发现，只需<strong>20倍</strong>左右的覆盖度就能够产生<strong>出色</strong>的组装结果，且<strong>不需要NGS纠错</strong>。</p>
<h3 id="组装结果"><a href="#组装结果" class="headerlink" title="组装结果"></a>组装结果</h3><p>在获得HiFi数据之后，PacBio负责<strong>超大基因组组装</strong>的资深生信科学家<strong>Greg Concepcion</strong>开始了加州红杉基因组的组装工作。Greg选择的工具是<font color="red">Hifiasm</font>，因为这是目前<font color="blue"><strong>最快</strong>的且<strong>专注于解析单倍型</strong>的组装软件</font>。</p>
<p>Hifiasm软件运行<font color="red">6天</font>后，组装工作顺利完成。具体消耗计算资源如下；</p>
<p>服务器配置：64核512Gb RAM</p>
<p>生成HiFi数据：46,000 CPU hours</p>
<p>基因组组装：7,200 CPU hours，总共分析时间6天</p>
<p>对比2019年Oxford Nanopore + NGS同样组装加州红杉，使用2Tb内存,最终花费5-6个月才完成组装。</p>
<p>Sequencing and assembling mega-genomes of mega-trees: the giant sequoia and coast redwood genomes，<strong>Nanopore Community Meeting 2019</strong>【最大内存使用：2T；error correction：330，000CPU hours；Assembly post-error-correction：700，000CPU hours；wall clock time：5-6month】</p>
<p>获得的结果惊人，结果几乎是<font color="blue">预期基因大小<strong>两倍</strong></font>，而<font color="blue">Contig N50</font>达到<font color="blue"><strong>1.92Mb</strong></font> ！相对2019年Oxford Nanopore + NGS组装<font color="blue"><strong>提升18倍</strong></font>。大于预期基因组大小的组装中，大部分可是<font color="blue"><strong>两个类似的单倍型</strong></font>，而<font color="red"><strong>并非完全不同的六倍体</strong></font>，这与最近Scott发表的红杉相关研究类似，印证了加州红杉的多倍体化是<font color="red"><strong>自多倍体事件</strong></font>。</p>
<p><img src="/blog/./640.jpeg" alt="图片"></p>
<p><font color="blue">加州红杉组装结果中<strong>BUSCO评分只有59%??</strong> </font>不是一般<strong>较好</strong>的组装结果BUSCO值都要高达<font color="blue"><strong>90%以上吗？</strong></font>因为<font color="blue">裸子植物（加州红杉属于裸子植物）的BUSCO基因集与被子植物的<strong>BUSCO基因集相差很</strong>多</font>。意味着针对加州红杉使用<font color="red"><strong>通用BUSCO基因集</strong></font>来<font color="red"><strong>评估</strong>基因组完整性<strong>不准确</strong></font>，只是<strong>目前没有</strong>其它<strong>更好</strong>评估方法。</p>
<h3 id="总结"><a href="#总结" class="headerlink" title="总结"></a>总结</h3><p>从加州红杉基因组研究中得到什么呢？</p>
<p>首先，对PacBio SMRT测序充满信心，建议使用HiFi数据从任何生物体生成高质量的基因组数据。 </p>
<p>其次，它将彻底改变对<strong>大型，复杂</strong>的基因组装配的认知。之前经验是<strong>超大型复杂基因组</strong>组装需要花费<strong>大量时间</strong>和生信<strong>计算资源</strong>，还不包括测序时间。而加州红杉庞大基因组在短短17天内就可完成（包括4天样品准备和建库，7天测序和6天组装）。 </p>
<p><strong>通过PacBio HiFi数据，可以对<font color="blue">任何生物</font>进行<font color="blue">高质量</font>的基因组测序。</strong></p>
<h2 id="2020-05"><a href="#2020-05" class="headerlink" title="2020.05"></a>2020.05</h2><p>HiFi助力同源四倍体紫花苜蓿基因组组装，NC发表</p>
<p>Allele-aware chromosome-level genome assembly and efficient transgene-free genome editing for the autotetraploid cultivated alfalfa</p>
<p>（1）使用Canu默认参数，利用CCS clean reads组装contigs。组装得到的Contig N50值为459kb，总长度为3.15GB。</p>
<p>（2）使用HiC-Pro将Hi-C reads与contigs 进行比对，产生比对BAM文件。</p>
<p>（3）使用注释的蒺藜苜蓿蛋白作为参考，完全基于同源的策略注释contigs。对138,729个同源基因进行了结构注释。用MCscan用于鉴定contigs和参考基因组之间的共线性。显示紫花苜蓿和蒺藜苜蓿之间的高共线性。【下载父母本petunia ref】</p>
<p>（4）使用内部脚本处理BAM文件，去除等位基因contigs之间的links。使用ALLHiC软件，提取、聚类和重排Contigs (Contigs syntenic与蒺藜苜蓿染色体一致)，得到原始的scaffolds。</p>
<p>（5）Juicebox用于以图形和交互方式微调组装的scaffolds。剪裁了40个总长度达1800Mb的scaffolds 。</p>
<p>（6） 基于组装的scaffold，通过Hi-C数据，每个unplaced contig被分配到互作最强的那些contig cluster里。</p>
<p>（7）使用ALLHiC对那些contig clusters再次进行重排和构建scaffold。</p>
<p>（8）使用Juicebox对scaffolds进行微调，并从scaffolds上去除不一致的contigs，产生最终的染色体基因组，其包含32条染色体(8个同源组，每个组中有4个等位基因染色体)，总长度为2738Mb，和419Mb未挂载到染色体水平的序列。</p>
<h2 id="2020-07"><a href="#2020-07" class="headerlink" title="2020.07"></a>2020.07</h2><p>加州大学HiFi组装人完整X染色体，Nature</p>
<p>Miga Karen H,Koren Sergey,Rhie Arang et al. Telomere-to-telomere assembly of a complete human X chromosome.[J] .Nature, 2020.</p>
<h1 id="HiFi与不同类型基因组组装"><a href="#HiFi与不同类型基因组组装" class="headerlink" title="HiFi与不同类型基因组组装"></a><a target="_blank" rel="noopener" href="https://www.jianshu.com/p/4d42719d8b01">HiFi与不同类型基因组组装</a></h1><h2 id="高重复基因组—玉米"><a href="#高重复基因组—玉米" class="headerlink" title="高重复基因组—玉米"></a><strong>高重复基因组—玉米</strong></h2><p>玉米（<em>Z.mays</em>）是世界上产量最高种植最广的作物，也是遗传学和基因组学的基础model。玉米基因组经历<font color="blue"><strong>全基因组复制事件</strong></font>和<font color="blue"><strong>长末端重复转座子扩增</strong></font>导致基因组<strong>急剧扩大</strong>（<font color="red">2.3Gb</font>）。最终导致玉米基因组中<font color="red"><strong>转座子占比高达85%</strong></font>。</p>
<p>利用<strong>HiFi</strong>组装的<font color="blue"><strong>玉米B73</strong></font>基因组大小为<font color="blue"><strong>2.164 Gb</strong></font>，<strong>略大于</strong>2017年发表的利用PacBio <font color="blue">CLR</font>的组装版本（<font color="blue">2.104 Gb</font>）。在连续性方面，HiFi组装<font color="blue"><strong>Contig N50则达到28.2 Mb</strong></font>，提升近<font color="red"><strong>23倍</strong></font>，甚至<strong>超过</strong>最近发表的<strong>NC358</strong>基因组。<font color="red"><strong>BUSCO评估表明</strong></font>，HiFi组装版本的组装质量也<font color="red"><strong>高于所有</strong></font>已发表的<strong>玉米基因组版本</strong>。在时间和资源消耗上，最近发表的<strong>75× CLR数据</strong>组装NC358基因组花费了<strong>11,520 CPU hours</strong>，而HiFi数据组装仅在6小时完成，仅花费360 CPU hours，<font color="red"><strong>资源消耗节省了31倍</strong></font>。体现了HiFi在<font color="blue">高重复基因组组装</font>的明显<font color="blue">优势</font>。</p>
<p><img src="/blog/2022/03/02/2022-03-02-HiFi-3-software/10.png" alt="img"></p>
<p>HiFi 组装 <strong>玉米B73</strong> 基因组结果</p>
<h2 id="大型复杂基因组—黄腿山蛙"><a href="#大型复杂基因组—黄腿山蛙" class="headerlink" title="大型复杂基因组—黄腿山蛙"></a><strong>大型复杂基因组—黄腿山蛙</strong></h2><p>黄腿山蛙（<em>R. muscosa</em>）是一种两栖动物。<font color="blue"><strong>两栖动物</strong>是<strong>脊椎动物</strong>中<strong>基因组大小变化最大</strong>的一类</font>，已完成基因组测序的<strong>美西钝口螈</strong>高达<strong>32 Gb</strong>。蛙科物种中，最大基因组达10 Gb以上，黄腿山蛙的基因组<strong>预计9 Gb</strong>左右。</p>
<p>利用HiFi数据组装的黄腿山蛙基因组获得<font color="blue">两套单倍型基因组</font>，单倍型基因组大小分别为<font color="blue">9.03G和5.23 Gb</font>，<font color="red"><strong>主要单倍型基因组</strong></font>大小<strong>接近预估</strong>基因组大小。<font color="red"><strong>主要单倍型基因组</strong>Contig <strong>N50达2.8 M</strong>，<strong>远高于</strong>已发表的蛙科物种基因组</font>。HiFi 组装的<font color="blue"><strong>黄腿山蛙</strong>基因组的<strong>BUSCO评估结果</strong>也<strong>明显高于</strong>另外两个物种的基因组</font>。此外，虽然黄腿山蛙的基因组高达9 Gb，利用HiFi 组装也仅花费<strong>12小时</strong>。上述表明，HiFi快速组装<font color="blue">高质量大型基因组</font>出色。</p>
<p><img src="/blog/2022/03/02/2022-03-02-HiFi-3-software/11.png" alt="img"></p>
<p>HiFi组装 <strong>黄腿山蛙</strong> 基因组结果</p>
<h2 id="异源多倍体基因组—草莓"><a href="#异源多倍体基因组—草莓" class="headerlink" title="异源多倍体基因组—草莓"></a><strong>异源多倍体基因组—草莓</strong></h2><p>栽培草莓（<em>F. x ananassa</em>）是一个<font color="blue"><strong>异源</strong>八倍体物种（2n=8x=56）</font>，由两种<font color="blue">野生八倍体物种天然杂交</font>产生，这<font color="blue"><strong>两种野生八倍体</strong>都是<strong>100多万年</strong>前<strong>四种二倍体祖先物种合并</strong>的产物</font>。</p>
<p>利用HiFi数据组装的八倍体草莓基因组获得了<font color="blue">两套单倍型基因组</font>，大小分别为<font color="blue">776M和304 M</font>，<font color="red"><strong>主要单倍型基因组</strong></font>的大小与<strong>2019年</strong>最新<strong>发表</strong>的<strong>栽培草莓基因组相当</strong>，但<font color="blue">Contig N50达<strong>15.5 Mb</strong>，提升<strong>193倍</strong></font>。此外，整个基因组组装仅花费<strong>3小时</strong>。上述结果体现HiFi测序在<font color="blue">多倍体基因组组装中的良好表现</font>。</p>
<p><img src="/blog/2022/03/02/2022-03-02-HiFi-3-software/12.png" alt="img"></p>
<p>HiFi 数据组装 <strong>八倍体草莓</strong> 基因组结果</p>
<h2 id="同源多倍体基因组—苜蓿-√"><a href="#同源多倍体基因组—苜蓿-√" class="headerlink" title="同源多倍体基因组—苜蓿 √"></a><strong>同源多倍体基因组—苜</strong>蓿 √</h2><p>栽培紫花苜蓿（<em>M. sativa</em>）是一种<font color="blue">自交不亲和的<strong>同源</strong>四倍体（2n=4×=32）</font>植物。5月昆明动物研究所等单位利用HiFi测序（约23×）实现栽培紫花苜蓿的基因组组装 <a target="_blank" rel="noopener" href="https://pubmed.ncbi.nlm.nih.gov/32427850/">paper</a>。</p>
<p>利用专门针对HiFi数据开发的组装软件<font color="red"><strong>HiCanu和Hifiasm</strong></font>重新对该基因组进行组装。<font color="blue">HiCanu组装版本</font>基因组大小为<font color="blue">2.66 G</font>，<strong>略低</strong>于<font color="blue">Canu版本</font>；<font color="blue"><strong>Hifiasm组装版本</strong></font>的基因组大小（<font color="blue">3.22 G</font>）则略高于Canu版本。<font color="red"><strong>HiCanu</strong>和<strong>Hifiasm</strong>的组装的<strong>Contig N50</strong>分别为<strong>1.67</strong>和<strong>1.73</strong> Mb，<strong>明显高于Canu</strong>版本</font>。测试结果及已发表的苜蓿基因组均表明，HiFi能够实现<font color="blue">同源多倍体基因组的高质量组装</font>。</p>
<p><img src="/blog/2022/03/02/2022-03-02-HiFi-3-software/14.png" alt="同源四倍体苜蓿基因组组装结果比较"></p>
<h2 id="单倍型基因组-人"><a href="#单倍型基因组-人" class="headerlink" title="单倍型基因组   人"></a><strong>单倍型基因组   人</strong></h2><p>目前已发表多个高质量人基因组序列，且已有多个利用HiFi组装人基因组的案例，利用HiFi组装出<strong>人两套单倍型基因组</strong>也已经被报道。</p>
<p>利用HiFi 对人基因组组装，获得两套单倍型基因组。<font color="blue"><strong>两套单倍型基因组</strong>大小分别为<strong>3.14G</strong>和<strong>2.50 G</strong>，<strong>Contig N50</strong>分别为<strong>42.2M</strong>和<strong>1.6 M</strong></font>。这一结果与<strong>已发表的利用HiFi数据组装的HG002和HG00733基因组相当。整个组装过程仅花费2小时。</strong>上述结果体现HiFi在<font color="blue">单倍型基因组组装中的优势</font>。</p>
<p><img src="/blog/2022/03/02/2022-03-02-HiFi-3-software/13.png" alt="img"></p>
<p>HiFi 组装 人 基因组结果</p>
<hr>
<h1 id="？Pacbio序列结构变异germline-SV-calling"><a href="#？Pacbio序列结构变异germline-SV-calling" class="headerlink" title="？Pacbio序列结构变异germline SV calling"></a>？Pacbio序列结构变异germline SV calling</h1><p><a target="_blank" rel="noopener" href="https://blog.csdn.net/weixin_36416921/article/details/112277318">https://blog.csdn.net/weixin_36416921/article/details/112277318</a> ？？</p>
<p><img src="/blog/2022/03/02/2022-03-02-HiFi-3-software/2.png" alt="img"></p>
<ul>
<li><p>获取bam文件（Pacbio下机数据）。</p>
</li>
<li><p>subreads.bam文件 到ccs.bam文件</p>
</li>
<li><p>bam到fastq文件</p>
</li>
<li><p>数据分析，变异检测</p>
<ul>
<li><strong>mapping &amp; calling 【</strong>nextSV2流程】</li>
<li>常用的长序列比对软件有<strong>ngmlr</strong> 和 <strong>minimap2</strong>，不同的mapping软件得到结果略有差别，minimap2比对得到的结果较多？</li>
<li><strong>sniffles calling</strong> </li>
</ul>
</li>
</ul>
<hr>
<p>一般认为只有CLR的subreads需要用NGS校正，ccs或hifi reads已经自身校正了，不用额外校正。当ccs的阈值设定为3时，得到的序列就是HIFI reads了。</p>
<h1 id="HiFi-HiCanu"><a href="#HiFi-HiCanu" class="headerlink" title="HiFi+HiCanu"></a>HiFi+HiCanu</h1><p>ref: <a target="_blank" rel="noopener" href="http://www.genome.cn/News/Industry/1099.html">http://www.genome.cn/News/Industry/1099.html</a></p>
<p><a target="_blank" rel="noopener" href="https://pubmed.ncbi.nlm.nih.gov/32801147/">HiCanu: accurate assembly of segmental duplications, satellites, and allelic variants from high-fidelity long reads</a> <strong>2020.09 Genom Res</strong></p>
<p><strong>PacBio HiFi+HiCanu</strong>完成准确度超99.999%，<strong>Contig N50</strong>达到<strong>77Mb</strong>的<strong>人类</strong>基因组组装结果！</p>
<p>NIH科学家使用PacBio HiFi Reads组装基因组文章 <strong>HiCanu</strong>。使用HiCanu组装工具（专门针对 HiFi Reads优化组装流程），对<font color="red"><strong>果蝇</strong></font>和<font color="red"><strong>人类</strong></font><strong>标准基因组</strong>组装，重点探索HiFi Reads对<font color="blue"><strong>基因组单倍型的区分</strong></font>，主要<font color="blue"><strong>组织相容性复合体（MHC）变异</strong></font>，染色体<font color="blue"><strong>卫星区域</strong></font>和<font color="blue"><strong>片段重复</strong></font>的检测能力。</p>
<h2 id="基因组连续性和准确性提升"><a href="#基因组连续性和准确性提升" class="headerlink" title="基因组连续性和准确性提升"></a><strong>基因组连续性和准确性提升</strong></h2><p>30X HiFi Reads通过HiCanu将人CHM13细胞系的基因组组装的<strong>Contig N50</strong> 提升至<strong>77M</strong>，单碱基准确性超过99.999%（Q50）！在组装的准确性连续性两个方面，HiFi Reads的组装结果都超过了最新 高覆盖度 超长 牛津 纳米孔测序 的结果。</p>
<p><img src="/blog/2022/03/02/2022-03-02-HiFi-3-software/3.png" alt="3.png"></p>
<p>使用PacBio HiFi和Ultra-Long (Oxford Nanoproe) 组装人CHM13基因组结果对比，都<strong>未经短读长测序数据polish</strong>。Peregrine, Canu和HiCanu是不同的组装工具。</p>
<h2 id="基因组单倍型检测的提升-做此图类似评估"><a href="#基因组单倍型检测的提升-做此图类似评估" class="headerlink" title="基因组单倍型检测的提升[做此图类似评估]"></a><strong>基因组单倍型检测的提升</strong>[做此图类似评估]</h2><p>组装多倍体时，组装软件要能区分<font color="blue"><strong>不同等位基因</strong></font>，并将它们<font color="blue"><strong>保存为不同的序列</strong></font>，以<font color="blue"><strong>主要Contig</strong>和<strong>替代Contig</strong>模式</font>表示。对于<font color="blue">双倍体人</font>基因组组装，HiCanu组装结果<font color="blue">包含<strong>超过2 Gbp</strong>的<strong>替代Contig</strong></font>，而<font color="blue">其它组装软件</font>只能产生<font color="blue">不到400 Mbp的替代Contig [hifiasm如何?]</font>。说明<font color="blue">HiCanu</font>结合HiFi Reads具有<font color="blue">更强的<strong>区分单倍型能力</strong></font>。如下图，HiCanu组装的<font color="red">主要Contig和替代Contig<strong>都</strong></font>具有<font color="red">较高的<strong>BUSCO完整性</strong></font>（分别**&gt; 94％<strong>和</strong>&gt; 75％<strong>），而采用 <strong>超长 牛津纳米孔测序</strong>的BUSCO完整性分别只有</strong>63%<strong>和</strong>0.3%**。并且HiCanu + HiFi Reads的<font color="red"><strong>Phase Block NG50?</strong></font>是高覆盖度超长牛津纳米孔测序的2.5倍！</p>
<p><img src="/blog/2022/03/02/2022-03-02-HiFi-3-software/4.png" alt="4"></p>
<p>HiCanu与其它组装软件 <strong>组装 双倍体基因组 结果对比</strong>，其中HG00733的Haplotype asm hap1(hap2)a和HG002的Haplotype asm hap1(hap2)b 的数据是<font color="red"><strong>HiFi结合Hi-C的组装结果</strong></font>。</p>
<p>为了<font color="red">评估</font>不同组装工具对<strong>难以检测</strong>的<font color="red"><strong>复杂临床相关等位基因分型结果</strong></font>，通过HiCanu+ HiFi 得到的六个经典的<font color="blue">人类白细胞抗原（HLA）基因</font>的<font color="blue">装配分型结果</font>与之前通过<strong>多种检测方法</strong>获得的<strong>已知HG002和HG00733等位基因</strong>进行比较。<font color="red">只有HiCanu和TrioCanu能够<strong>恢复</strong>具有<strong>100％序列同一性</strong>的<strong>所有等位基因</strong></font>。这也体现HiCanu+ HiFi对于潜在的复杂临床相关基因的精确检测能力。</p>
<h2 id="应对复杂片段重复序列"><a href="#应对复杂片段重复序列" class="headerlink" title="应对复杂片段重复序列"></a><strong>应对复杂片段重复序列</strong></h2><p>为了评估对<font color="red"><strong>片段重复（SD，segmental duplications）</strong>序列的组装能力</font>，使用了BAC文库挑战。被<strong>选择</strong>的<strong>BAC文库序列包含大量片段重复序列</strong>，而片段重复区域是基因组<font color="blue"><strong>最难组装区域之一</strong></font>。经过初步比对，HiCanu+ HiFi在所有组装方案中<font color="red">解析最多的BAC</font>，并且实现<font color="red">最高的BAC对比质量</font>。</p>
<p><img src="/blog/2022/03/02/2022-03-02-HiFi-3-software/5.png" alt="不同测序数据与组装工具在BAC文库挑战中的结果"></p>
<p>不同测序数据与组装工具在BAC文库挑战中的结果</p>
<p>针对<font color="blue">CHM13中<strong>未能</strong>成功解析的BAC深入研究</font>，发现其中【11个BAC本身可能包含组装错误或克隆伪像?】。手动检查11个BAC的HiFi组装结果，并没有发现明显组装错误（如下图）。这表明HiCanu+ HiFi 实际上成功解析CHM13 341个BAC中的<strong>337个</strong>。</p>
<p><img src="/blog/2022/03/02/2022-03-02-HiFi-3-software/6.png" alt="1585618676593848.png"></p>
<p>人工检查可能本身存在错误的BAC</p>
<h2 id="人类染色体着丝粒的组装-how"><a href="#人类染色体着丝粒的组装-how" class="headerlink" title="人类染色体着丝粒的组装 [how?]"></a>人类染色体着丝粒的组装 [how?]</h2><p>使用HiCanu+HiFi 成功生成包括CHM13 <font color="blue"><strong>19号染色体着丝粒在内</strong></font>的<font color="blue"><strong>共9个人类染色体着丝粒装配图</strong></font>。CHM13 <font color="red"><strong>19号染色体着丝粒区域被认为是最难组装的着丝粒之一</strong></font>，因为它<font color="red"><strong>由多个HOR（Higher Order Repeat）区域组成</strong></font>并<font color="red"><strong>与1号和5号染色体的着丝粒区域共享α卫星序列</strong></font>。</p>
<p>HiCanu+HiFi 不仅能<font color="red">组装<strong>覆盖整个着丝粒</strong>的contig<strong>群</strong></font>，而且可以<font color="red">准确<strong>区分</strong>三个<strong>不同的HOR区域???</strong></font>：<strong>D19Z1，D19Z2和D19Z3</strong>。在之前的<font color="red"><strong>X染色体T2T研究</strong>中</font>，为了达到如此完整的人类染色体<strong>着丝粒组装</strong>结果，需要<font color="blue">使用包括<strong>高深度超长纳米孔测序</strong>，<strong>60X PacBio CLR</strong>测序和<strong>illumina短读长</strong>测序的数据<strong>共同</strong>组装</font>才能达到这样的效果。而现在<font color="blue">仅使用<strong>一种</strong>测序技术</font>，即<strong>HiFi 结合HiCanu</strong>就能<strong>解析</strong>人类染色体<strong>着丝粒</strong>结构。</p>
<p><img src="/blog/2022/03/02/2022-03-02-HiFi-3-software/7.png" alt="1585618881806532.png"></p>
<p>CHM13 19号染色体<font color="blue"><strong>着丝粒的HiCanu装配</strong></font>。tig00006497的<font color="red"><strong>RepeatMasker</strong>揭示了位于<strong>19号着丝粒</strong>（D19Z1，D19Z2α和D19Z3；标有黑条）中的<strong>三个α卫星HOR区域</strong></font>。这些<strong>HOR序列长度</strong>分别为<strong>606 kbp，289 kbp和3.96 Mbp</strong>，分别由<strong>13-mer</strong>，复杂的<strong>高阶HOR</strong>和<strong>二聚体HOR单元</strong>组成?</p>
<h2 id="总结与展望"><a href="#总结与展望" class="headerlink" title="总结与展望"></a><strong>总结与展望</strong></h2><p>通过本研究，证明<font color="blue">HiFi 结合HiCanu</font>能够生成迄今为止<font color="blue">最准确，最完整</font>的人类基因组<font color="blue">装配体</font>。其他应用，例如同样需要高精度Reads的<strong>宏基因组组装</strong>。<font color="red">HiFi擅长解决<strong>大型高度相似</strong>（但<strong>不完全相同</strong>）的<strong>重复序列</strong></font>。HiCanu+HiFi 对人类<strong>1号，7号，9号和16号</strong>染色体的重建<strong>显著改善</strong>先前<strong>超长纳米孔测序的组装连续性</strong>。这些染色体包含<font color="blue">多个长度超过<strong>200 kbp的片段重复</strong></font>，需要 <font color="blue">HiFi 才能<strong>识别</strong>变异并<strong>分离</strong>出<strong>单个拷贝</strong></font>。使用 <strong>HiFi <font color="red">独立完成</font><strong>人类染色体</strong>九个着丝粒区域</strong>的装配草图，这也是T2T项目的最难挑战之一。</p>
<p><font color="blue">HiFi 的长度</font>对组装结果有很大影响，目前最新HiFi Reads已经达到25k。HiCanu在亚马逊<strong>云平台</strong>上组装一个<strong>人</strong>类基因组只需要<strong>22小时</strong>，这还<strong>不是目前最快</strong>的针对 HiFi数据的组装软件，最快的软件只需要<strong>2小时</strong>。相信未来随着HiFi Reads的长度不断提高，以及更多针对PacBio HiFi Reads优化的组装软件出现，基因组的从头组装会<strong>更快更准</strong>。</p>
<h1 id><a href="#" class="headerlink" title></a></h1>
    </div>

    
    
    
        

<div>
<ul class="post-copyright">
  <li class="post-copyright-author">
    <strong>Post author:  </strong>Wang C
  </li>
  <li class="post-copyright-link">
    <strong>Post link: </strong>
    <a href="https://wangc1106.github.io/blog/2022/03/02/2022-03-02-HiFi-3-software/" title="HiFi-3-assemble-software">https://wangc1106.github.io/blog/2022/03/02/2022-03-02-HiFi-3-software/</a>
  </li>
  <li class="post-copyright-license">
    <strong>Copyright Notice:  </strong>All articles in this blog are licensed under <a href="https://creativecommons.org/licenses/by-nc-sa/4.0/" rel="noopener" target="_blank"><i class="fab fa-fw fa-creative-commons"></i>BY-NC-SA</a> unless stating additionally.
  </li>
</ul>
</div>


      <footer class="post-footer">
          <div class="post-tags">
              <a href="/blog/tags/denovo-III/" rel="tag"># denovo-III</a>
          </div>

        


        
    <div class="post-nav">
      <div class="post-nav-item">
    <a href="/blog/2022/03/01/2022-03-01-sequencing-data/" rel="prev" title="Sequencing data and fastq.gz">
      <i class="fa fa-chevron-left"></i> Sequencing data and fastq.gz
    </a></div>
      <div class="post-nav-item">
    <a href="/blog/2022/03/02/2022-03-02-paper-Petunia/" rel="next" title="【paper】Insight into the evolution of the Solanaceae from the parental genomes of Petunia hybrida">
      【paper】Insight into the evolution of the Solanaceae from the parental genomes of Petunia hybrida <i class="fa fa-chevron-right"></i>
    </a></div>
    </div>
      </footer>
    
  </article>
  
  
  



          </div>
          

<script>
  window.addEventListener('tabs:register', () => {
    let { activeClass } = CONFIG.comments;
    if (CONFIG.comments.storage) {
      activeClass = localStorage.getItem('comments_active') || activeClass;
    }
    if (activeClass) {
      let activeTab = document.querySelector(`a[href="#comment-${activeClass}"]`);
      if (activeTab) {
        activeTab.click();
      }
    }
  });
  if (CONFIG.comments.storage) {
    window.addEventListener('tabs:click', event => {
      if (!event.target.matches('.tabs-comment .tab-content .tab-pane')) return;
      let commentClass = event.target.classList[1];
      localStorage.setItem('comments_active', commentClass);
    });
  }
</script>

        </div>
          
  
  <div class="toggle sidebar-toggle">
    <span class="toggle-line toggle-line-first"></span>
    <span class="toggle-line toggle-line-middle"></span>
    <span class="toggle-line toggle-line-last"></span>
  </div>

  <aside class="sidebar">
    <div class="sidebar-inner">

      <ul class="sidebar-nav motion-element">
        <li class="sidebar-nav-toc">
          Table of Contents
        </li>
        <li class="sidebar-nav-overview">
          Overview
        </li>
      </ul>

      <!--noindex-->
      <div class="post-toc-wrap sidebar-panel">
          <div class="post-toc motion-element"><ol class="nav"><li class="nav-item nav-level-1"><a class="nav-link" href="#HiFi-%E5%BF%AB-%E5%87%86-%E7%8B%A0"><span class="nav-number">1.</span> <span class="nav-text">HiFi 快 准 狠</span></a><ol class="nav-child"><li class="nav-item nav-level-2"><a class="nav-link" href="#%E7%BB%84%E8%A3%85%E5%BF%AB"><span class="nav-number">1.1.</span> <span class="nav-text">组装快</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#%E7%BB%84%E8%A3%85%E5%87%86"><span class="nav-number">1.2.</span> <span class="nav-text">组装准</span></a><ol class="nav-child"><li class="nav-item nav-level-3"><a class="nav-link" href="#1-%E8%BF%9E%E7%BB%AD%E6%80%A7%E4%B8%8ECLR%E5%92%8CNanopore%E7%9B%B8%E5%BD%93"><span class="nav-number">1.2.1.</span> <span class="nav-text">1 连续性与CLR和Nanopore相当</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#2-%E7%BB%84%E8%A3%85%E8%B4%A8%E9%87%8F%E6%98%BE%E8%91%97%E6%8F%90%E5%8D%87"><span class="nav-number">1.2.2.</span> <span class="nav-text">2 组装质量显著提升</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#3-%E5%AF%B9%E9%87%8D%E5%A4%8D%E5%8C%BA%E5%9F%9F%E5%92%8C%E7%9D%80%E4%B8%9D%E7%B2%92%E5%8C%BA%E5%9F%9F%E5%85%B7%E6%9C%89%E8%BE%83%E9%AB%98%E7%9A%84%E5%88%86%E8%BE%A8%E7%8E%87"><span class="nav-number">1.2.3.</span> <span class="nav-text">3 对重复区域和着丝粒区域具有较高的分辨率</span></a></li></ol></li><li class="nav-item nav-level-2"><a class="nav-link" href="#%E7%8B%A0"><span class="nav-number">1.3.</span> <span class="nav-text">狠</span></a><ol class="nav-child"><li class="nav-item nav-level-3"><a class="nav-link" href="#1-%E7%AE%80%E5%8D%95%E5%A4%8D%E6%9D%82%E5%9F%BA%E5%9B%A0%E7%BB%84%E9%80%9A%E5%90%83"><span class="nav-number">1.3.1.</span> <span class="nav-text">1 简单复杂基因组通吃</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#2-%E9%AB%98%E8%B4%A8%E9%87%8F%E5%8D%95%E5%80%8D%E5%9E%8B%E5%9F%BA%E5%9B%A0%E7%BB%84%E7%BB%84%E8%A3%85"><span class="nav-number">1.3.2.</span> <span class="nav-text">2 高质量单倍型基因组组装</span></a></li></ol></li></ol></li><li class="nav-item nav-level-1"><a class="nav-link" href="#hifiasm-Official-doc"><span class="nav-number">2.</span> <span class="nav-text">hifiasm Official doc</span></a><ol class="nav-child"><li class="nav-item nav-level-2"><a class="nav-link" href="#introduction"><span class="nav-number">2.1.</span> <span class="nav-text">introduction</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#Why-Hifiasm"><span class="nav-number">2.2.</span> <span class="nav-text">Why Hifiasm?</span></a></li></ol></li><li class="nav-item nav-level-1"><a class="nav-link" href="#hifiasm-paper"><span class="nav-number">3.</span> <span class="nav-text">hifiasm paper</span></a><ol class="nav-child"><li class="nav-item nav-level-2"><a class="nav-link" href="#%E5%8E%9F%E7%90%86%E4%BB%8B%E7%BB%8D1"><span class="nav-number">3.1.</span> <span class="nav-text">原理介绍1</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#hifiasm-%E5%8E%9F%E7%90%86%E4%BB%8B%E7%BB%8D2"><span class="nav-number">3.2.</span> <span class="nav-text">hifiasm 原理介绍2</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#%E4%BB%85HiFi%E6%95%B0%E6%8D%AE%E6%A8%A1%E5%BC%8F"><span class="nav-number">3.3.</span> <span class="nav-text">仅HiFi数据模式</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#%E5%8F%8C%E4%BA%B2%E4%BA%8C%E4%BB%A3%E6%B5%8B%E5%BA%8FTrio-binning%E6%A8%A1%E5%BC%8F"><span class="nav-number">3.4.</span> <span class="nav-text">双亲二代测序Trio-binning模式</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#%E9%99%84%E5%8A%A0Hi-C%E6%A8%A1%E5%BC%8F"><span class="nav-number">3.5.</span> <span class="nav-text">附加Hi-C模式</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#%E6%97%A5%E5%BF%97%E5%92%8C%E5%8F%82%E6%95%B0%E8%B0%83%E6%95%B4"><span class="nav-number">3.6.</span> <span class="nav-text">日志和参数调整</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#%E8%BE%93%E5%85%A5%E6%A0%BC%E5%BC%8F-fq%E8%BF%98%E6%98%AFfa"><span class="nav-number">3.7.</span> <span class="nav-text">输入格式 fq还是fa</span></a></li></ol></li><li class="nav-item nav-level-1"><a class="nav-link" href="#HiCanu-https-canu-readthedocs-io-en-latest-index-html"><span class="nav-number">4.</span> <span class="nav-text">HiCanu[https:&#x2F;&#x2F;canu.readthedocs.io&#x2F;en&#x2F;latest&#x2F;index.html]</span></a><ol class="nav-child"><li class="nav-item nav-level-2"><a class="nav-link" href="#Install"><span class="nav-number">4.1.</span> <span class="nav-text">Install</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#Run-Canu"><span class="nav-number">4.2.</span> <span class="nav-text">Run Canu</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#Debug"><span class="nav-number">4.3.</span> <span class="nav-text">Debug</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#Advice"><span class="nav-number">4.4.</span> <span class="nav-text">Advice</span></a><ol class="nav-child"><li class="nav-item nav-level-3"><a class="nav-link" href="#%E5%90%8E%E7%BB%AD%E5%88%86%E6%9E%90%E8%A6%81%E5%8E%BB%E5%86%97%E4%BD%99"><span class="nav-number">4.4.1.</span> <span class="nav-text">后续分析要去冗余</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#%E9%AB%98%E6%9D%82%E5%90%88%E7%89%A9%E7%A7%8D%E7%BB%84%E8%A3%85"><span class="nav-number">4.4.2.</span> <span class="nav-text">高杂合物种组装</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#%E8%B4%AD%E4%B9%B0SSD%E9%81%BF%E5%85%8D%E6%9C%8D%E5%8A%A1%E5%99%A8IO%E7%93%B6%E9%A2%88"><span class="nav-number">4.4.3.</span> <span class="nav-text">购买SSD避免服务器IO瓶颈</span></a></li></ol></li></ol></li><li class="nav-item nav-level-1"><a class="nav-link" href="#HiFI%E7%BB%84%E8%A3%85%E5%AE%9E%E4%BE%8B"><span class="nav-number">5.</span> <span class="nav-text">HiFI组装实例</span></a><ol class="nav-child"><li class="nav-item nav-level-2"><a class="nav-link" href="#2018-10"><span class="nav-number">5.1.</span> <span class="nav-text">2018.10</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#2019-01"><span class="nav-number">5.2.</span> <span class="nav-text">2019.01</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#2020-01"><span class="nav-number">5.3.</span> <span class="nav-text">2020.01</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#2020-03-27Gb%E5%85%AD%E5%80%8D%E4%BD%93%E5%8A%A0%E5%B7%9E%E7%BA%A2%E6%9D%89"><span class="nav-number">5.4.</span> <span class="nav-text">2020.03 27Gb六倍体加州红杉</span></a><ol class="nav-child"><li class="nav-item nav-level-3"><a class="nav-link" href="#%E5%88%9D%E8%A1%B7"><span class="nav-number">5.4.1.</span> <span class="nav-text">初衷</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#%E6%B5%8B%E5%BA%8F%E8%BF%87%E7%A8%8B"><span class="nav-number">5.4.2.</span> <span class="nav-text">测序过程</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#%E7%BB%84%E8%A3%85%E7%BB%93%E6%9E%9C"><span class="nav-number">5.4.3.</span> <span class="nav-text">组装结果</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#%E6%80%BB%E7%BB%93"><span class="nav-number">5.4.4.</span> <span class="nav-text">总结</span></a></li></ol></li><li class="nav-item nav-level-2"><a class="nav-link" href="#2020-05"><span class="nav-number">5.5.</span> <span class="nav-text">2020.05</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#2020-07"><span class="nav-number">5.6.</span> <span class="nav-text">2020.07</span></a></li></ol></li><li class="nav-item nav-level-1"><a class="nav-link" href="#HiFi%E4%B8%8E%E4%B8%8D%E5%90%8C%E7%B1%BB%E5%9E%8B%E5%9F%BA%E5%9B%A0%E7%BB%84%E7%BB%84%E8%A3%85"><span class="nav-number">6.</span> <span class="nav-text">HiFi与不同类型基因组组装</span></a><ol class="nav-child"><li class="nav-item nav-level-2"><a class="nav-link" href="#%E9%AB%98%E9%87%8D%E5%A4%8D%E5%9F%BA%E5%9B%A0%E7%BB%84%E2%80%94%E7%8E%89%E7%B1%B3"><span class="nav-number">6.1.</span> <span class="nav-text">高重复基因组—玉米</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#%E5%A4%A7%E5%9E%8B%E5%A4%8D%E6%9D%82%E5%9F%BA%E5%9B%A0%E7%BB%84%E2%80%94%E9%BB%84%E8%85%BF%E5%B1%B1%E8%9B%99"><span class="nav-number">6.2.</span> <span class="nav-text">大型复杂基因组—黄腿山蛙</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#%E5%BC%82%E6%BA%90%E5%A4%9A%E5%80%8D%E4%BD%93%E5%9F%BA%E5%9B%A0%E7%BB%84%E2%80%94%E8%8D%89%E8%8E%93"><span class="nav-number">6.3.</span> <span class="nav-text">异源多倍体基因组—草莓</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#%E5%90%8C%E6%BA%90%E5%A4%9A%E5%80%8D%E4%BD%93%E5%9F%BA%E5%9B%A0%E7%BB%84%E2%80%94%E8%8B%9C%E8%93%BF-%E2%88%9A"><span class="nav-number">6.4.</span> <span class="nav-text">同源多倍体基因组—苜蓿 √</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#%E5%8D%95%E5%80%8D%E5%9E%8B%E5%9F%BA%E5%9B%A0%E7%BB%84-%E4%BA%BA"><span class="nav-number">6.5.</span> <span class="nav-text">单倍型基因组   人</span></a></li></ol></li><li class="nav-item nav-level-1"><a class="nav-link" href="#%EF%BC%9FPacbio%E5%BA%8F%E5%88%97%E7%BB%93%E6%9E%84%E5%8F%98%E5%BC%82germline-SV-calling"><span class="nav-number">7.</span> <span class="nav-text">？Pacbio序列结构变异germline SV calling</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#HiFi-HiCanu"><span class="nav-number">8.</span> <span class="nav-text">HiFi+HiCanu</span></a><ol class="nav-child"><li class="nav-item nav-level-2"><a class="nav-link" href="#%E5%9F%BA%E5%9B%A0%E7%BB%84%E8%BF%9E%E7%BB%AD%E6%80%A7%E5%92%8C%E5%87%86%E7%A1%AE%E6%80%A7%E6%8F%90%E5%8D%87"><span class="nav-number">8.1.</span> <span class="nav-text">基因组连续性和准确性提升</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#%E5%9F%BA%E5%9B%A0%E7%BB%84%E5%8D%95%E5%80%8D%E5%9E%8B%E6%A3%80%E6%B5%8B%E7%9A%84%E6%8F%90%E5%8D%87-%E5%81%9A%E6%AD%A4%E5%9B%BE%E7%B1%BB%E4%BC%BC%E8%AF%84%E4%BC%B0"><span class="nav-number">8.2.</span> <span class="nav-text">基因组单倍型检测的提升[做此图类似评估]</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#%E5%BA%94%E5%AF%B9%E5%A4%8D%E6%9D%82%E7%89%87%E6%AE%B5%E9%87%8D%E5%A4%8D%E5%BA%8F%E5%88%97"><span class="nav-number">8.3.</span> <span class="nav-text">应对复杂片段重复序列</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#%E4%BA%BA%E7%B1%BB%E6%9F%93%E8%89%B2%E4%BD%93%E7%9D%80%E4%B8%9D%E7%B2%92%E7%9A%84%E7%BB%84%E8%A3%85-how"><span class="nav-number">8.4.</span> <span class="nav-text">人类染色体着丝粒的组装 [how?]</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#%E6%80%BB%E7%BB%93%E4%B8%8E%E5%B1%95%E6%9C%9B"><span class="nav-number">8.5.</span> <span class="nav-text">总结与展望</span></a></li></ol></li><li class="nav-item nav-level-1"><a class="nav-link"><span class="nav-number">9.</span> <span class="nav-text"></span></a></li></ol></div>
      </div>
      <!--/noindex-->

      <div class="site-overview-wrap sidebar-panel">
        <div class="site-author motion-element" itemprop="author" itemscope itemtype="http://schema.org/Person">
    <img class="site-author-image" itemprop="image" alt="Wang C"
      src="/blog/images/banner.png">
  <p class="site-author-name" itemprop="name">Wang C</p>
  <div class="site-description" itemprop="description">Make more time</div>
</div>
<div class="site-state-wrap motion-element">
  <nav class="site-state">
      <div class="site-state-item site-state-posts">
          <a href="/blog/archives">
          <span class="site-state-item-count">46</span>
          <span class="site-state-item-name">posts</span>
        </a>
      </div>
      <div class="site-state-item site-state-categories">
            <a href="/blog/categories/">
          
        <span class="site-state-item-count">13</span>
        <span class="site-state-item-name">categories</span></a>
      </div>
      <div class="site-state-item site-state-tags">
            <a href="/blog/tags/">
          
        <span class="site-state-item-count">19</span>
        <span class="site-state-item-name">tags</span></a>
      </div>
  </nav>
</div>
  <div class="links-of-author motion-element">
      <span class="links-of-author-item">
        <a href="https://github.com/wangc1106" title="GitHub → https:&#x2F;&#x2F;github.com&#x2F;wangc1106" rel="noopener" target="_blank"><i class="fab fa-github fa-fw"></i>GitHub</a>
      </span>
      <span class="links-of-author-item">
        <a href="mailto:wangchen116@126.com" title="E-Mail → mailto:wangchen116@126.com" rel="noopener" target="_blank"><i class="fa fa-envelope fa-fw"></i>E-Mail</a>
      </span>
  </div>
  <div class="cc-license motion-element" itemprop="license">
    <a href="https://creativecommons.org/licenses/by-nc-sa/4.0/" class="cc-opacity" rel="noopener" target="_blank"><img src="/blog/images/cc-by-nc-sa.svg" alt="Creative Commons"></a>
  </div>



      </div>
        <div class="back-to-top motion-element">
          <i class="fa fa-arrow-up"></i>
          <span>0%</span>
        </div>

    </div>
  </aside>
  <div id="sidebar-dimmer"></div>


      </div>
    </main>

    <footer class="footer">
      <div class="footer-inner">
        

        

<div class="copyright">
  
  &copy; 2021 – 
  <span itemprop="copyrightYear">2022</span>
  <span class="with-love">
    <i class="fas fa-dna"></i>
  </span>
  <span class="author" itemprop="copyrightHolder">Wang C</span>
</div>

        






<script>
  (function() {
    function leancloudSelector(url) {
      url = encodeURI(url);
      return document.getElementById(url).querySelector('.leancloud-visitors-count');
    }

    function addCount(Counter) {
      var visitors = document.querySelector('.leancloud_visitors');
      var url = decodeURI(visitors.id);
      var title = visitors.dataset.flagTitle;

      Counter('get', '/classes/Counter?where=' + encodeURIComponent(JSON.stringify({ url })))
        .then(response => response.json())
        .then(({ results }) => {
          if (results.length > 0) {
            var counter = results[0];
            leancloudSelector(url).innerText = counter.time + 1;
            Counter('put', '/classes/Counter/' + counter.objectId, { time: { '__op': 'Increment', 'amount': 1 } })
              .catch(error => {
                console.error('Failed to save visitor count', error);
              });
          } else {
              Counter('post', '/classes/Counter', { title, url, time: 1 })
                .then(response => response.json())
                .then(() => {
                  leancloudSelector(url).innerText = 1;
                })
                .catch(error => {
                  console.error('Failed to create', error);
                });
          }
        })
        .catch(error => {
          console.error('LeanCloud Counter Error', error);
        });
    }

    function showTime(Counter) {
      var visitors = document.querySelectorAll('.leancloud_visitors');
      var entries = [...visitors].map(element => {
        return decodeURI(element.id);
      });

      Counter('get', '/classes/Counter?where=' + encodeURIComponent(JSON.stringify({ url: { '$in': entries } })))
        .then(response => response.json())
        .then(({ results }) => {
          for (let url of entries) {
            let target = results.find(item => item.url === url);
            leancloudSelector(url).innerText = target ? target.time : 0;
          }
        })
        .catch(error => {
          console.error('LeanCloud Counter Error', error);
        });
    }

    let { app_id, app_key, server_url } = {"enable":true,"app_id":"UqIYufmNk2sE8MiBKKWc4Kja-gzGzoHsz","app_key":"iYpeWQcmNqpHwr2uVlnJYVji","server_url":null,"security":false};
    function fetchData(api_server) {
      var Counter = (method, url, data) => {
        return fetch(`${api_server}/1.1${url}`, {
          method,
          headers: {
            'X-LC-Id'     : app_id,
            'X-LC-Key'    : app_key,
            'Content-Type': 'application/json',
          },
          body: JSON.stringify(data)
        });
      };
      if (CONFIG.page.isPost) {
        if (CONFIG.hostname !== location.hostname) return;
        addCount(Counter);
      } else if (document.querySelectorAll('.post-title-link').length >= 1) {
        showTime(Counter);
      }
    }

    let api_server = app_id.slice(-9) !== '-MdYXbMMI' ? server_url : `https://${app_id.slice(0, 8).toLowerCase()}.api.lncldglobal.com`;

    if (api_server) {
      fetchData(api_server);
    } else {
      fetch('https://app-router.leancloud.cn/2/route?appId=' + app_id)
        .then(response => response.json())
        .then(({ api_server }) => {
          fetchData('https://' + api_server);
        });
    }
  })();
</script>


      </div>
      <div style="height: 200px">
      <script type="text/javascript" src="//rf.revolvermaps.com/0/0/8.js?i=5knf34gs07t&amp;m=0&amp;c=ff0000&amp;cr1=ffffff&amp;f=arial&amp;l=33" async="async"></script>
      </div>
    </footer>
  </div>

  
  <script src="/blog/lib/anime.min.js"></script>
  <script src="/blog/lib/velocity/velocity.min.js"></script>
  <script src="/blog/lib/velocity/velocity.ui.min.js"></script>

<script src="/blog/js/utils.js"></script>

<script src="/blog/js/motion.js"></script>


<script src="/blog/js/schemes/pisces.js"></script>


<script src="/blog/js/next-boot.js"></script>




  




  
<script src="/blog/js/local-search.js"></script>













  

  

</body>
</html>
