<!DOCTYPE html>
<html lang="zh-Hans">
<head>
  <meta charset="UTF-8">
<meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=2">
<meta name="theme-color" content="#222">
<meta name="generator" content="Hexo 5.4.0">
  <link rel="apple-touch-icon" sizes="180x180" href="/blog/images/apple-touch-icon-next.png">
  <link rel="icon" type="image/png" sizes="32x32" href="/blog/images/favicon-32x32-next.png">
  <link rel="icon" type="image/png" sizes="16x16" href="/blog/images/favicon-16x16-next.png">
  <link rel="mask-icon" href="/blog/images/logo.svg" color="#222">
  <meta name="google-site-verification" content="1osTo_gOe1SW9Hevqtz8vl8HfSgTO4DSMZ0DiMPMX6w">

<link rel="stylesheet" href="/blog/css/main.css">


<link rel="stylesheet" href="/blog/lib/font-awesome/css/all.min.css">

<script id="hexo-configurations">
    var NexT = window.NexT || {};
    var CONFIG = {"hostname":"wangc1106.github.io","root":"/blog/","scheme":"Gemini","version":"7.8.0","exturl":false,"sidebar":{"position":"left","display":"post","padding":18,"offset":12,"onmobile":false},"copycode":{"enable":false,"show_result":false,"style":null},"back2top":{"enable":true,"sidebar":true,"scrollpercent":"ture"},"bookmark":{"enable":false,"color":"#222","save":"auto"},"fancybox":false,"mediumzoom":false,"lazyload":false,"pangu":false,"comments":{"style":"tabs","active":null,"storage":true,"lazyload":false,"nav":null},"algolia":{"hits":{"per_page":10},"labels":{"input_placeholder":"Search for Posts","hits_empty":"We didn't find any results for the search: ${query}","hits_stats":"${hits} results found in ${time} ms"}},"localsearch":{"enable":true,"trigger":"auto","top_n_per_article":1,"unescape":false,"preload":false},"motion":{"enable":true,"async":false,"transition":{"post_block":"fadeIn","post_header":"slideDownIn","post_body":"slideDownIn","coll_header":"slideLeftIn","sidebar":"slideUpIn"}},"path":"search.xml"};
  </script>

  <meta name="description" content="Hi-C技术，以高通量测序为手段，以3C技术为基础的染色质构象捕捉技术。-三维基因组。">
<meta property="og:type" content="article">
<meta property="og:title" content="染色质三维结构图谱 Hi-C">
<meta property="og:url" content="https://wangc1106.github.io/blog/2022/02/16/2022-0216-chromosome-structure-Hi-C/index.html">
<meta property="og:site_name" content="Banner&#39;s Blog">
<meta property="og:description" content="Hi-C技术，以高通量测序为手段，以3C技术为基础的染色质构象捕捉技术。-三维基因组。">
<meta property="og:locale">
<meta property="og:image" content="https://wangc1106.github.io/blog/blog/2022/02/16/2022-0216-chromosome-structure-Hi-C/0.png">
<meta property="og:image" content="https://wangc1106.github.io/blog/blog/2022/02/16/2022-0216-chromosome-structure-Hi-C/1.png">
<meta property="og:image" content="https://wangc1106.github.io/blog/blog/2022/02/16/2022-0216-chromosome-structure-Hi-C/8.png">
<meta property="og:image" content="https://wangc1106.github.io/blog/blog/2022/02/16/2022-0216-chromosome-structure-Hi-C/2.png">
<meta property="og:image" content="https://wangc1106.github.io/blog/blog/2022/02/16/2022-0216-chromosome-structure-Hi-C/3.png">
<meta property="og:image" content="https://wangc1106.github.io/blog/blog/2022/02/16/2022-0216-chromosome-structure-Hi-C/4.png">
<meta property="og:image" content="https://wangc1106.github.io/blog/blog/2022/02/16/2022-0216-chromosome-structure-Hi-C/5.png">
<meta property="og:image" content="https://wangc1106.github.io/blog/blog/2022/02/16/2022-0216-chromosome-structure-Hi-C/6.png">
<meta property="og:image" content="https://wangc1106.github.io/blog/blog/2022/02/16/2022-0216-chromosome-structure-Hi-C/7.png">
<meta property="og:image" content="https://wangc1106.github.io/blog/blog/2022/02/16/2022-0216-chromosome-structure-Hi-C/9.png">
<meta property="og:image" content="https://wangc1106.github.io/blog/blog/2022/02/16/2022-0216-chromosome-structure-Hi-C/10.png">
<meta property="og:image" content="https://wangc1106.github.io/blog/blog/2022/02/16/2022-0216-chromosome-structure-Hi-C/11.png">
<meta property="og:image" content="https://wangc1106.github.io/blog/blog/2022/02/16/2022-0216-chromosome-structure-Hi-C/12.png">
<meta property="og:image" content="https://wangc1106.github.io/blog/blog/2022/02/16/2022-0216-chromosome-structure-Hi-C/13.png">
<meta property="og:image" content="https://wangc1106.github.io/blog/blog/2022/02/16/2022-0216-chromosome-structure-Hi-C/14.png">
<meta property="og:image" content="https://wangc1106.github.io/blog/blog/2022/02/16/2022-0216-chromosome-structure-Hi-C/15.png">
<meta property="og:image" content="https://wangc1106.github.io/blog/blog/2022/02/16/2022-0216-chromosome-structure-Hi-C/16.png">
<meta property="og:image" content="https://wangc1106.github.io/blog/blog/2022/02/16/2022-0216-chromosome-structure-Hi-C/17.png">
<meta property="og:image" content="https://wangc1106.github.io/blog/blog/2022/02/16/2022-0216-chromosome-structure-Hi-C/18.png">
<meta property="og:image" content="https://wangc1106.github.io/blog/blog/2022/02/16/2022-0216-chromosome-structure-Hi-C/19.png">
<meta property="og:image" content="https://wangc1106.github.io/blog/blog/2022/02/16/2022-0216-chromosome-structure-Hi-C/20.png">
<meta property="og:image" content="https://wangc1106.github.io/blog/blog/2022/02/16/2022-0216-chromosome-structure-Hi-C/21.png">
<meta property="og:image" content="https://wangc1106.github.io/blog/blog/2022/02/16/2022-0216-chromosome-structure-Hi-C/22.png">
<meta property="og:image" content="https://wangc1106.github.io/blog/blog/2022/02/16/2022-0216-chromosome-structure-Hi-C/24.png">
<meta property="og:image" content="https://wangc1106.github.io/blog/blog/2022/02/16/2022-0216-chromosome-structure-Hi-C/23.png">
<meta property="og:image" content="https://wangc1106.github.io/blog/blog/2022/02/16/2022-0216-chromosome-structure-Hi-C/25.png">
<meta property="og:image" content="https://wangc1106.github.io/blog/blog/2022/02/16/2022-0216-chromosome-structure-Hi-C/26.png">
<meta property="og:image" content="https://wangc1106.github.io/blog/blog/2022/02/16/2022-0216-chromosome-structure-Hi-C/27.png">
<meta property="og:image" content="https://wangc1106.github.io/blog/blog/2022/02/16/2022-0216-chromosome-structure-Hi-C/28.png">
<meta property="og:image" content="https://wangc1106.github.io/blog/blog/2022/02/16/2022-0216-chromosome-structure-Hi-C/29.png">
<meta property="og:image" content="https://wangc1106.github.io/blog/blog/2022/02/16/2022-0216-chromosome-structure-Hi-C/30.png">
<meta property="og:image" content="https://wangc1106.github.io/blog/blog/2022/02/16/2022-0216-chromosome-structure-Hi-C/31.png">
<meta property="og:image" content="https://wangc1106.github.io/blog/blog/2022/02/16/2022-0216-chromosome-structure-Hi-C/33.png">
<meta property="og:image" content="https://wangc1106.github.io/blog/blog/2022/02/16/2022-0216-chromosome-structure-Hi-C/34.png">
<meta property="og:image" content="https://wangc1106.github.io/blog/blog/2022/02/16/2022-0216-chromosome-structure-Hi-C/35.png">
<meta property="og:image" content="https://wangc1106.github.io/blog/blog/2022/02/16/2022-0216-chromosome-structure-Hi-C/36.png">
<meta property="og:image" content="https://wangc1106.github.io/blog/blog/2022/02/16/2022-0216-chromosome-structure-Hi-C/37.png">
<meta property="og:image" content="https://wangc1106.github.io/blog/blog/2022/02/16/2022-0216-chromosome-structure-Hi-C/32.png">
<meta property="article:published_time" content="2022-02-16T12:37:36.271Z">
<meta property="article:modified_time" content="2022-02-20T12:54:56.226Z">
<meta property="article:author" content="Wang C">
<meta property="article:tag" content="technique">
<meta name="twitter:card" content="summary">
<meta name="twitter:image" content="https://wangc1106.github.io/blog/blog/2022/02/16/2022-0216-chromosome-structure-Hi-C/0.png">

<link rel="canonical" href="https://wangc1106.github.io/blog/2022/02/16/2022-0216-chromosome-structure-Hi-C/">


<script id="page-configurations">
  // https://hexo.io/docs/variables.html
  CONFIG.page = {
    sidebar: "",
    isHome : false,
    isPost : true,
    lang   : 'zh-Hans'
  };
</script>

  <title>染色质三维结构图谱 Hi-C | Banner's Blog</title>
  






  <noscript>
  <style>
  .use-motion .brand,
  .use-motion .menu-item,
  .sidebar-inner,
  .use-motion .post-block,
  .use-motion .pagination,
  .use-motion .comments,
  .use-motion .post-header,
  .use-motion .post-body,
  .use-motion .collection-header { opacity: initial; }

  .use-motion .site-title,
  .use-motion .site-subtitle {
    opacity: initial;
    top: initial;
  }

  .use-motion .logo-line-before i { left: initial; }
  .use-motion .logo-line-after i { right: initial; }
  </style>
</noscript>

</head>

<body itemscope itemtype="http://schema.org/WebPage">
  <div class="container use-motion">
    <div class="headband"></div>

    <header class="header" itemscope itemtype="http://schema.org/WPHeader">
      <div class="header-inner"><div class="site-brand-container">
  <div class="site-nav-toggle">
    <div class="toggle" aria-label="Toggle navigation bar">
      <span class="toggle-line toggle-line-first"></span>
      <span class="toggle-line toggle-line-middle"></span>
      <span class="toggle-line toggle-line-last"></span>
    </div>
  </div>

  <div class="site-meta">

    <a href="/blog/" class="brand" rel="start">
      <span class="logo-line-before"><i></i></span>
      <h1 class="site-title">Banner's Blog</h1>
      <span class="logo-line-after"><i></i></span>
    </a>
      <p class="site-subtitle" itemprop="description">Dream big work hard</p>
  </div>

  <div class="site-nav-right">
    <div class="toggle popup-trigger">
        <i class="fa fa-search fa-fw fa-lg"></i>
    </div>
  </div>
</div>




<nav class="site-nav">
  <ul id="menu" class="main-menu menu">
        <li class="menu-item menu-item-home">

    <a href="/blog/" rel="section"><i class="home fa-fw"></i>Home</a>

  </li>
        <li class="menu-item menu-item-categories">

    <a href="/blog/categories/" rel="section"><i class="th fa-fw"></i>Categories</a>

  </li>
        <li class="menu-item menu-item-tags">

    <a href="/blog/tags/" rel="section"><i class="tags fa-fw"></i>Tags</a>

  </li>
        <li class="menu-item menu-item-about">

    <a href="https://wangc1106.github.io/" rel="section"><i class="user fa-fw"></i>About</a>

  </li>
      <li class="menu-item menu-item-search">
        <a role="button" class="popup-trigger"><i class="fa fa-search fa-fw"></i>Search
        </a>
      </li>
  </ul>
</nav>



  <div class="search-pop-overlay">
    <div class="popup search-popup">
        <div class="search-header">
  <span class="search-icon">
    <i class="fa fa-search"></i>
  </span>
  <div class="search-input-container">
    <input autocomplete="off" autocapitalize="off"
           placeholder="Searching..." spellcheck="false"
           type="search" class="search-input">
  </div>
  <span class="popup-btn-close">
    <i class="fa fa-times-circle"></i>
  </span>
</div>
<div id="search-result">
  <div id="no-result">
    <i class="fa fa-spinner fa-pulse fa-5x fa-fw"></i>
  </div>
</div>

    </div>
  </div>

</div>
    </header>

    


    <main class="main">
      <div class="main-inner">
        <div class="content-wrap">
          

          <div class="content post posts-expand">
            

    
  
  
  <article itemscope itemtype="http://schema.org/Article" class="post-block" lang="zh-Hans">
    <link itemprop="mainEntityOfPage" href="https://wangc1106.github.io/blog/2022/02/16/2022-0216-chromosome-structure-Hi-C/">

    <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
      <meta itemprop="image" content="/blog/images/banner.png">
      <meta itemprop="name" content="Wang C">
      <meta itemprop="description" content="Make more time">
    </span>

    <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
      <meta itemprop="name" content="Banner's Blog">
    </span>
      <header class="post-header">
        <h1 class="post-title" itemprop="name headline">
          染色质三维结构图谱 Hi-C
        </h1>

        <div class="post-meta">
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-calendar"></i>
              </span>
              <span class="post-meta-item-text">Posted on</span>

              <time title="Created: 2022-02-16 20:37:36" itemprop="dateCreated datePublished" datetime="2022-02-16T20:37:36+08:00">2022-02-16</time>
            </span>
              <span class="post-meta-item">
                <span class="post-meta-item-icon">
                  <i class="far fa-calendar-check"></i>
                </span>
                <span class="post-meta-item-text">Edited on</span>
                <time title="Modified: 2022-02-20 20:54:56" itemprop="dateModified" datetime="2022-02-20T20:54:56+08:00">2022-02-20</time>
              </span>
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-folder"></i>
              </span>
              <span class="post-meta-item-text">In</span>
                <span itemprop="about" itemscope itemtype="http://schema.org/Thing">
                  <a href="/blog/categories/biology/" itemprop="url" rel="index"><span itemprop="name">biology</span></a>
                </span>
            </span>

          
            <span id="/blog/2022/02/16/2022-0216-chromosome-structure-Hi-C/" class="post-meta-item leancloud_visitors" data-flag-title="染色质三维结构图谱 Hi-C" title="Views">
              <span class="post-meta-item-icon">
                <i class="fa fa-eye"></i>
              </span>
              <span class="post-meta-item-text">Views: </span>
              <span class="leancloud-visitors-count"></span>
            </span>
            <span class="post-meta-item" title="Symbols count in article">
              <span class="post-meta-item-icon">
                <i class="far fa-file-word"></i>
              </span>
              <span>9.3k</span>
            </span>
            <span class="post-meta-item" title="Reading time">
              <span class="post-meta-item-icon">
                <i class="far fa-clock"></i>
              </span>
              <span>8 mins.</span>
            </span>

        </div>
      </header>

    
    
    
    <div class="post-body" itemprop="articleBody">

      
        <h1 id="hi-c技术以高通量测序为手段以3c技术为基础的染色质构象捕捉技术-三维基因组"><a class="markdownIt-Anchor" href="#hi-c技术以高通量测序为手段以3c技术为基础的染色质构象捕捉技术-三维基因组"></a> Hi-C技术，以高通量测序为手段，以3C技术为基础的染色质构象捕捉技术。-三维基因组。</h1>
<span id="more"></span>
<p><img src="/blog/blog/2022/02/16/2022-0216-chromosome-structure-Hi-C/0.png" alt="img"></p>
<h1 id="1-c技术"><a class="markdownIt-Anchor" href="#1-c技术"></a> 1. C技术</h1>
<h2 id="11-3c一对一"><a class="markdownIt-Anchor" href="#11-3c一对一"></a> 1.1 3C（一对一）</h2>
<ul>
<li>
<p>基因组捕获技术（Chromosome conformation capture，3C）是最早研究三维基因组的技术，需要<font color="green">提前知道互作区域</font>，才能<font color="green">量化一对基因组基因座之间的互作</font>。</p>
</li>
<li>
<p>3C技术2002年Deker提出，目的捕获染色体的interaction，假设染色体中的interaction以蛋白为介导。通过限制酶酶切，补平，连接，打断，PCR后，若染色体A，B两点相互作用，根据这两点特异序列的引物能PCR出产物，验证interaction。每<font color="green">一对位点的验证</font>需要<font color="green">设计一对特异性引物</font>。</p>
</li>
</ul>
<h2 id="12-4c一对多"><a class="markdownIt-Anchor" href="#12-4c一对多"></a> 1.2 4C（一对多）</h2>
<ul>
<li>
<p>染色体构象捕获芯片（Chromosome conformation capture-on-chip，4C ），可以捕获<font color="green">一个基因区域</font>和其他区域间的互相作用。该技术不需要知道作用区域的先验知识就可以使用。</p>
</li>
<li>
<p>3C较麻烦，后设计双酶切位点，然后<font color="green">成环</font>，保证只需要设计一对引物就可以检测一个位点对多个位点的相互作用，4C有circle的意思。</p>
</li>
</ul>
<h2 id="13-5c多对多"><a class="markdownIt-Anchor" href="#13-5c多对多"></a> 1.3 5C（多对多）</h2>
<p>染色体构象捕获碳拷贝（Chromosome conformation capture carbon copy，5C） ，可以检测<font color="green">某段区域内所有的互作</font>，但是该<font color="red">区域一般&lt;1 Mb</font>。覆盖度问题造成该技术<font color="green">不适用于全基因组测序</font>。</p>
<h2 id="14-font-colorredhi-c全部互作font"><a class="markdownIt-Anchor" href="#14-font-colorredhi-c全部互作font"></a> 1.4 <font color="red">Hi-C（全部互作）</font></h2>
<ul>
<li>
<p>高通量基因组捕获技术（High-throughput/resolution chromosome conformation capture），基本解决了上述技术的缺点，可以实现<font color="green">全基因组覆盖</font>检测<font color="green">全部未知互作区域</font>。</p>
</li>
<li>
<p>Hi-C基本步骤，甲醛交联，限制酶切，末端补平加biotin，平末端连接，超声破碎，biotin富集，建库测序。整个过程没有特异引物存在。</p>
</li>
<li>
<p>将线性距离远，空间结构近的DNA片段进行交联，并将交联的DNA片段富集，高通量测序，对测序数据分析可以揭示染色质的远程互作，推导出基因组的三维空间结构的可能的记忆之间的调控关系。</p>
</li>
<li>
<p>\1. 通过Scaffold间的交互频率大小，可以对已组装的基因组序列进行纠错（基因组更准确）。</p>
<p>\2. 基因信息不再仅仅是contig片段，而是被划分至染色体上，成为染色体水平。</p>
<p>\3. 无需辛苦的构建群体，单一一个体就能实现染色体定位。</p>
<p>\4. 相比遗传图谱，标记密度更大，序列定位更完整（能把更多的contig挂至染色体上!信息更全面！）</p>
<p>\5. 染色体重排等结构变异研究可以开展啦~(研究可以更深入！）</p>
<p>\6. QTL、GWAS可以定位区间到某个染色体啦~（追踪变异！）</p>
<p>\7. 该物种的三维基因结构、染色体互作及动态变化可以解析啦~（从基因到表观！全方位解析）</p>
</li>
</ul>
<h2 id="15-chip-loop-基于免疫沉淀"><a class="markdownIt-Anchor" href="#15-chip-loop-基于免疫沉淀"></a> 1.5 ChIP-loop (基于免疫沉淀)</h2>
<p>该技术将 3C 与 ChIP-seq 结合，可以检测<font color="red">目的蛋白质介导</font>的<font color="red">两个目的基因区域互作</font>。</p>
<h2 id="16-chia-pet-基于免疫沉淀"><a class="markdownIt-Anchor" href="#16-chia-pet-基于免疫沉淀"></a> 1.6 ChIA-PET (基于免疫沉淀)</h2>
<p>该技术将 HiC 与 ChIP-seq 结合，可以检测<font color="red">目的蛋白质的所有互相作用</font>。 用特异性抗体富集interaction信息，比如CTCF，pol II抗体，类似步骤建库测序。<font color="red">ChIA-PET数据是Hi-C数据的子集</font>。</p>
<hr>
<table>
<thead>
<tr>
<th></th>
<th>互作</th>
<th>覆盖</th>
<th>作用</th>
<th>研究应用</th>
</tr>
</thead>
<tbody>
<tr>
<td>3C</td>
<td>单对单</td>
<td>&lt;1Mb</td>
<td>检测已知<font color="red">基因组基因区域之间</font>的互作</td>
<td>确定已知启动子和增强子之间的互作</td>
</tr>
<tr>
<td>4C</td>
<td>一对多</td>
<td>全基因组</td>
<td>确定<font color="red">基因某组区域</font>与<font color="red">其他区域的互作</font></td>
<td>与已知LCR(locus control regions)区互作的全部基因和元件</td>
</tr>
<tr>
<td>5C</td>
<td>多对多</td>
<td>&lt;1Mb</td>
<td>确定<font color="red">染色体特定区域</font>的全部高级结构</td>
<td>确定染色体特定区域的全部高级结构</td>
</tr>
<tr>
<td>Hi-C</td>
<td>全部互作</td>
<td>全基因组</td>
<td>检测<font color="red">全基因组</font>范围的<font color="red">全部高级结构</font></td>
<td>全基因组范围所有染色体互作</td>
</tr>
<tr>
<td>ChIP-loop</td>
<td>一对一</td>
<td>&lt;1Mb</td>
<td>检测目的蛋白介导的<font color="red">两个目的基因区域互作</font></td>
<td></td>
</tr>
<tr>
<td>ChIA-PET</td>
<td>全部互作</td>
<td>全基因组</td>
<td>检测<font color="red">目的蛋白质的所有互作</font></td>
<td>构建已知转录因子介导的染色质互作</td>
</tr>
</tbody>
</table>
<hr>
<h1 id="2-技术原理"><a class="markdownIt-Anchor" href="#2-技术原理"></a> 2. 技术原理</h1>
<p><img src="/blog/blog/2022/02/16/2022-0216-chromosome-structure-Hi-C/1.png" alt="img"></p>
<p><img src="/blog/blog/2022/02/16/2022-0216-chromosome-structure-Hi-C/8.png" alt="img"></p>
<h2 id="21-甲醛固定"><a class="markdownIt-Anchor" href="#21-甲醛固定"></a> 2.1 甲醛固定</h2>
<p>先加入甲醛将基因组中参与<font color="red">染色质互作的蛋白质凝固</font>。一般将活体样本在室温用 1-3%的甲醛处理 10-30min，但是此步骤会<font color="green">减少限制内切酶</font>对DNA序列的消化<font color="green">效率</font>，需要严格控制。</p>
<p><img src="/blog/blog/2022/02/16/2022-0216-chromosome-structure-Hi-C/2.png" alt="img"></p>
<h2 id="22-酶切序列"><a class="markdownIt-Anchor" href="#22-酶切序列"></a> 2.2 酶切序列</h2>
<p>用限制性内切酶切割基因组，打断后的片段大小会影响测序分辨率，一般有两种酶可供选择：6bp 的限制性内切酶，4bp 的限制性内切酶。后者具有更高的分辨率。EcoR1 或 HindIII 用于每4000bp切割一次基因组，在人类基因组中产生约100万个片段。<br>
<img src="/blog/blog/2022/02/16/2022-0216-chromosome-structure-Hi-C/3.png" alt="img"></p>
<h2 id="23-末端修复"><a class="markdownIt-Anchor" href="#23-末端修复"></a> 2.3 末端修复</h2>
<p>得到的片段具有平末端或粘性末端，然后将<font color="red">末端补平修复，加入生物素</font>。【生物素pull-down保证只有连接的部分用于分析。读长reads被定位到基因组上，当这一对读长发现在不同片段上的时候，对<font color="red">互作片段进行相应评分</font>。一个包含基因组所有片段的<font color="red">连接频率矩阵被构建</font>。】</p>
<p><img src="/blog/blog/2022/02/16/2022-0216-chromosome-structure-Hi-C/4.png" alt="img"></p>
<h2 id="24-连接及解交联"><a class="markdownIt-Anchor" href="#24-连接及解交联"></a> 2.4 连接及解交联</h2>
<p>使用 <font color="red">T4 DNA连接酶连接互作片段形成环状</font>。将连接DNA片段的<font color="red">蛋白质消化掉</font>，得到交联片段。</p>
<p><img src="/blog/blog/2022/02/16/2022-0216-chromosome-structure-Hi-C/5.png" alt="img"></p>
<h2 id="25-序列打断"><a class="markdownIt-Anchor" href="#25-序列打断"></a> 2.5 序列打断</h2>
<p>使用超声波或其他方式，再次打断片段。</p>
<p><img src="/blog/blog/2022/02/16/2022-0216-chromosome-structure-Hi-C/6.png" alt="img"></p>
<h2 id="26-上机测序"><a class="markdownIt-Anchor" href="#26-上机测序"></a> 2.6 上机测序</h2>
<p>用<font color="red">磁珠将带生物素的捕获</font>，制作文库，上机测序。</p>
<p><img src="/blog/blog/2022/02/16/2022-0216-chromosome-structure-Hi-C/7.png" alt="img"></p>
<h1 id="3-hi-c应用"><a class="markdownIt-Anchor" href="#3-hi-c应用"></a> 3. Hi-C应用</h1>
<ul>
<li><font color="red">量化</font>在三维空间中基因组的<font color="red">染色质间交联</font>（cross-linked chromatin ）</li>
<li>解析<font color="red">全基因组互作<strong>模式</strong></font>，如<font color="red">启动子和增强子互作</font></li>
<li>构建<font color="red">三维空间结构<strong>模型</strong></font>，如研究基因组三维结构特征：<strong>compartment，TAD，loop</strong>等</li>
<li>构建<font color="red">全基因组互作<strong>图谱</strong></font></li>
<li><font color="red">辅助提升<strong>基因组组装</strong></font></li>
<li>构建<font color="red">基因组<strong>单体型图谱</strong></font></li>
</ul>
<h1 id="4-分析流程"><a class="markdownIt-Anchor" href="#4-分析流程"></a> 4. 分析流程</h1>
<h2 id="41-基本分析"><a class="markdownIt-Anchor" href="#41-基本分析"></a> 4.1 基本分析</h2>
<p><img src="/blog/blog/2022/02/16/2022-0216-chromosome-structure-Hi-C/9.png" alt="img"></p>
<p>Ferhat Ay et al;2015</p>
<ul>
<li>(a)首先是<font color="green">质控</font>，过滤后高质量的FASTQ数据（PE，150bp），如果比对软件<font color="red">不支持split mapping</font>，一般选用<font color="red">迭代比对</font>，因为连接处由于是基因组外的碱基，可能比对不上。<strong>从序列左端25bp开始比对，如果有唯一比对，则停止，如果多个比对位置，则再继续延伸5bp，直到出现唯一比对</strong>。或选择<strong>支持split mapping的软件进行比对</strong>，可以通过<strong>分段比对处理</strong>。</li>
<li>(b)选择高质量的比对数据</li>
<li>©HiC特异的<font color="green">比对标准</font></li>
<li>(d)对Vaild pairs进行<font color="green">矫正</font>。矫正完可以得到<font color="green">互作矩阵</font>。</li>
</ul>
<p><img src="/blog/blog/2022/02/16/2022-0216-chromosome-structure-Hi-C/10.png" alt="img"></p>
<p>Bryan R. Lajoie et al;2014</p>
<h2 id="42-hi-c可视化"><a class="markdownIt-Anchor" href="#42-hi-c可视化"></a> 4.2 Hi-C可视化</h2>
<p><img src="/blog/blog/2022/02/16/2022-0216-chromosome-structure-Hi-C/11.png" alt="img"></p>
<h2 id="43-序列过滤"><a class="markdownIt-Anchor" href="#43-序列过滤"></a> 4.3 序列过滤</h2>
<p>​	<img src="/blog/blog/2022/02/16/2022-0216-chromosome-structure-Hi-C/12.png" alt="img"></p>
<h2 id="44-数据矫正"><a class="markdownIt-Anchor" href="#44-数据矫正"></a> 4.4 数据矫正</h2>
<p>​	<img src="/blog/blog/2022/02/16/2022-0216-chromosome-structure-Hi-C/13.png" alt="img"></p>
<p>Eitan Yaffe &amp; Amos Tanay；2011</p>
<p>​	<img src="/blog/blog/2022/02/16/2022-0216-chromosome-structure-Hi-C/14.png" alt="img"></p>
<h1 id="5-分析"><a class="markdownIt-Anchor" href="#5-分析"></a> 5. 分析</h1>
<h2 id="51-cistrans互作比例"><a class="markdownIt-Anchor" href="#51-cistrans互作比例"></a> 5.1 cis/trans互作比例</h2>
<p>​	<img src="/blog/blog/2022/02/16/2022-0216-chromosome-structure-Hi-C/15.png" alt="img"></p>
<h2 id="52-互作频率与距离有关"><a class="markdownIt-Anchor" href="#52-互作频率与距离有关"></a> 5.2 互作频率与距离有关</h2>
<p>​	<img src="/blog/blog/2022/02/16/2022-0216-chromosome-structure-Hi-C/16.png" alt="img"></p>
<h2 id="53-compartment分析"><a class="markdownIt-Anchor" href="#53-compartment分析"></a> 5.3 compartment分析</h2>
<p><img src="/blog/blog/2022/02/16/2022-0216-chromosome-structure-Hi-C/17.png" alt="img"></p>
<h2 id="54-tad分析"><a class="markdownIt-Anchor" href="#54-tad分析"></a> 5.4 TAD分析</h2>
<p><img src="/blog/blog/2022/02/16/2022-0216-chromosome-structure-Hi-C/18.png" alt="img"></p>
<p>Hi-C辅助TAD结构研究（<em>Wang et al</em>., Nature Commuications 2018）</p>
<h2 id="55-显著互作分析"><a class="markdownIt-Anchor" href="#55-显著互作分析"></a> 5.5 显著互作分析</h2>
<p><img src="/blog/blog/2022/02/16/2022-0216-chromosome-structure-Hi-C/19.png" alt="img"></p>
<h1 id="6-应用"><a class="markdownIt-Anchor" href="#6-应用"></a> 6. 应用</h1>
<h2 id="61-解析全基因组互作模式"><a class="markdownIt-Anchor" href="#61-解析全基因组互作模式"></a> 6.1 解析全基因组互作模式</h2>
<p><img src="/blog/blog/2022/02/16/2022-0216-chromosome-structure-Hi-C/20.png" alt="img"></p>
<p><img src="/blog/blog/2022/02/16/2022-0216-chromosome-structure-Hi-C/21.png" alt="img"></p>
<h2 id="62-辅助提升基因组组装"><a class="markdownIt-Anchor" href="#62-辅助提升基因组组装"></a> 6.2 辅助提升基因组组装</h2>
<p><img src="/blog/blog/2022/02/16/2022-0216-chromosome-structure-Hi-C/22.png" alt="img"></p>
<ul>
<li>
<p>Hi-C辅助陆地棉与海岛棉基因组染色体水平组装（<em>Wang et al</em>.,Nature Genetics 2018）</p>
</li>
<li>
<p>Hi-C无需群体，单一个体就能实现染色体定位。很多物种都无法构建遗传群体，包括大部分高等动物、野生动植物、林木、果树等等。Hi-C是通过染色体上空间距离、线性距离的不同而导致的交互频率的不同来完成染色体的定位，所以不需要构建群体。</p>
</li>
<li>
<p>标记密度更大，序列定位更完整。相比遗传图谱，染色质之间的交互频率具有更高的标记密度，如此高密度的图谱不仅可以挂载上长的Scaffold，较短的Scaffold也可以被定位，所以通过Hi-C技术，一般可以将90%以上基因组序列定位到染色体。</p>
</li>
<li>
<p>可以对已组装的基因组进行纠错。通过Scaffold间的交互频率大小，可以对已组装的基因组序列进行纠错。</p>
</li>
</ul>
<h3 id="621-数据统计和过滤"><a class="markdownIt-Anchor" href="#621-数据统计和过滤"></a> 6.2.1 数据统计和过滤；</h3>
<h3 id="622-hi-c文库评估"><a class="markdownIt-Anchor" href="#622-hi-c文库评估"></a> 6.2.2 Hi-C文库评估；</h3>
<h3 id="623hi-c染色体定位"><a class="markdownIt-Anchor" href="#623hi-c染色体定位"></a> 6.2.3Hi-C染色体定位；</h3>
<p>​    ✓ 组装序列染色体群组划分；</p>
<p>✓  组装序列各群组内排序；</p>
<p>✓  组装序列各群组内定向；</p>
<h3 id="624-hi-c定位后统计评估"><a class="markdownIt-Anchor" href="#624-hi-c定位后统计评估"></a> 6.2.4 Hi-C定位后统计评估；</h3>
<p>​    ✓  近缘物种参考基因组评估；</p>
<p>​    ✓  遗传标记评估；</p>
<p><img src="/blog/blog/2022/02/16/2022-0216-chromosome-structure-Hi-C/24.png" alt="img"></p>
<h2 id="63-构建基因组单体型图谱"><a class="markdownIt-Anchor" href="#63-构建基因组单体型图谱"></a> 6.3 构建基因组单体型图谱</h2>
<p><img src="/blog/blog/2022/02/16/2022-0216-chromosome-structure-Hi-C/23.png" alt="img"></p>
<h1 id="7-文献案例"><a class="markdownIt-Anchor" href="#7-文献案例"></a> 7. 文献案例</h1>
<h2 id="71"><a class="markdownIt-Anchor" href="#71"></a> 7.1</h2>
<p><strong>物种：棉花</strong></p>
<p>**题目：**Genome sequence of<em>Gossypium herbaceum</em>and genome updates of<em>Gossypium arboreum</em>and<em>Gossypium hirsutum</em>provide insights into cotton A-genome evolution</p>
<p>**期刊：**Nature genetics</p>
<p>**影响因子：**25.455</p>
<p>**研究单位：**武汉大学</p>
<p><strong>主要研究内容：</strong></p>
<p>2020年4月13日，武汉大学朱玉贤院士团队关于草棉、亚洲棉和陆地棉基因组研究成果发表在国际著名期刊Nature genetics。该研究利用了PacBio三代分子测序和北京百迈客生物技术有限公司的Hi-C技术绘制了三个高质量棉花基因图谱，发现了三种棉花中多个染色体易位与倒位事件，解决了围绕A基因组起源的现有争议性概念，确定了与棉花纤维长度这一重要农艺性状相关的丰富的候选基因，并为棉花遗传改良提供了宝贵的基因组资源。</p>
<p><img src="/blog/blog/2022/02/16/2022-0216-chromosome-structure-Hi-C/25.png" alt="草棉与亚洲棉基因组Hi-C热图"></p>
<h2 id="72"><a class="markdownIt-Anchor" href="#72"></a> 7.2</h2>
<p><strong>物种：珙桐</strong></p>
<p><strong>题目：</strong><a href="https://link.zhihu.com/?target=http%3A//tour.biocloud.net/article/v1/into/articleDetail/31970919">Genomic analyses of a “living fossil”: The endangered dove-tree</a></p>
<p>**期刊：**Molecular Ecology Resources</p>
<p>**影响因子：**7.049</p>
<p>**研究单位：**四川大学</p>
<p>**主要研究内容：**2020年1月四川大学刘建全团队破译了中国特有的濒危物种珙桐（<em>Davidia involucrata</em>）基因组，相关成果发表在著名期刊Molecular Ecology Resources上。四川大学刘建全教授和杨勇志博士为该论文的通讯作者，陈阳和马涛教授为并列一作。该研究利用单分子实时测序SMRT和北京百迈客生物技术有限公司的Hi-C技术组装了一个高质量、染色体体水平的珙桐基因组，研究发现苞片中光合作用相关基因几乎缺失或表达减少，而抗菌、抗冷、抗水等抗逆相关基因在苞片中高度表达，突出了苞片在保护花和吸引授粉者中的重要作用。有效群体大小等研究分析了珙桐的生存机制和濒危原因。在未来气候持续变暖的背景下，研究结果为保护这独特的濒危物种提供了依据。</p>
<p><img src="/blog/blog/2022/02/16/2022-0216-chromosome-structure-Hi-C/26.png" alt="珙桐基因组Hi-C热图"></p>
<h2 id="73"><a class="markdownIt-Anchor" href="#73"></a> 7.3</h2>
<p><strong>物种：二倍体草莓</strong></p>
<p><strong>题目：</strong><a href="https://link.zhihu.com/?target=http%3A//tour.biocloud.net/article/v1/into/articleDetail/32003918">The high-quality genome of diploid strawberry (Fragaria nilgerrensis) provides new insights into anthocyanin accumulation</a></p>
<p>**期刊：**Plant Biotechnology Journal</p>
<p>**影响因子：**6.84</p>
<p>**研究单位：**沈阳农业大学</p>
<p><strong>主要研究内容：</strong></p>
<p>2020年1月31日沈阳农业大学张志宏研究团队与北京百迈客生物科技有限公司合作，基于单分子实时测序PacBio和染色体构象捕获Hi-C技术，成功构建亚洲东部和东南部地区特有的野生二倍体草莓——黄毛草莓（<em>Fragaria nilgerrensis</em>）高质量基因图谱，相关研究成果发表在知名期刊Plant Biotechnology Journal上。文中通过比较基因组学等系列分析找寻到可能与黄毛草莓特异表型以及环境适应性相关的扩张基因，以及可能与其白色果实相关的变异。该研究为草莓属物种的生物学以及比较基因组学研究提供了宝贵的遗传资源。</p>
<p><img src="/blog/blog/2022/02/16/2022-0216-chromosome-structure-Hi-C/27.png" alt="黄毛草莓基因组Hi-C热图"></p>
<h2 id="74"><a class="markdownIt-Anchor" href="#74"></a> 7.4</h2>
<p><strong>物种：油桐</strong></p>
<p><strong>题目：</strong><a href="https://link.zhihu.com/?target=http%3A//tour.biocloud.net/article/v1/into/articleDetail/32224189">Tung Tree (Vernicia fordii) Genome Provides A Resource for Understanding Genome Evolution and Improved Oil Production</a></p>
<p>**期刊：**Genomics Proteomics &amp; Bioinformatics</p>
<p>**影响因子：**6.597</p>
<p>**研究单位：**中南林业科技大学</p>
<p><strong>主要研究内容：</strong></p>
<p>2020年4月7日，Genomics Proteomics &amp; Bioinformatics 在线发表了由中南林业科技大学谭晓风团队牵头完成的油桐基因组研究论文。该研究利用Illumina+PacBio组装了1.12Gb油桐基因组，并通过北京百迈客生物技术有限公司的Hi-C技术将95.15%的序列挂载到11条染色体上。基于比较基因组等系列相关研究解析了油桐基因组进化和桐油生物合成的分子机制。该研究为理解基因组演化、以及油脂产量的分子育种与遗传改良都提供了宝贵的遗传资源，为油桐种质资源保护、分子遗传学研究、良种选育及加工利用，奠定了重要的科学基础。</p>
<p><img src="/blog/blog/2022/02/16/2022-0216-chromosome-structure-Hi-C/28.png" alt="油桐基因组circos图"></p>
<h2 id="75"><a class="markdownIt-Anchor" href="#75"></a> 7.5</h2>
<p><strong>物种：亚麻</strong></p>
<p><strong>题目：</strong><a href="https://link.zhihu.com/?target=http%3A//tour.biocloud.net/article/v1/into/articleDetail/32240956">Genomic Comparison and Population Diversity Analysis Provide Insights into the Domestication and Improvement of Flax</a></p>
<p>**期刊：**iScience</p>
<p><strong>影响因子：</strong>——</p>
<p>**研究单位：**甘肃省农业科学院</p>
<p><strong>主要研究内容：</strong></p>
<p>2020年4月24日，由甘肃省农科院和北京百迈客等单位合作的亚麻基因组研究成果荣登iScience, 其中张建平研究员，党占海研究员，刘头明研究员，百迈客CEO郑洪坤为本文通讯作者。该研究使用Illumina技术对油用亚麻品种Longya-10、纤维亚麻品种Heiya-14和pale flax分别进行测序与组装，后期利用HiC技术与遗传图谱辅助Longya-10基因组组装，将434 scaffolds组装至染色体水平。通过比较基因组及群体进化等研究探索油用及纤维用亚麻在驯化中受到的选择作用，以及可能在重塑亚麻形态结构中起着至关重要作用的相关基因。该研究揭示了对亚麻驯化和改良有重要影响的基因，这将有助于今后的分子育种。</p>
<p><img src="/blog/blog/2022/02/16/2022-0216-chromosome-structure-Hi-C/29.png" alt="亚麻基因组Hi-C热图"></p>
<h2 id="76"><a class="markdownIt-Anchor" href="#76"></a> 7.6</h2>
<p><strong>物种：枇杷</strong></p>
<p><strong>题目：</strong><a href="https://link.zhihu.com/?target=http%3A//tour.biocloud.net/article/v1/into/articleDetail/32141509">Chromosome-level genome assembly and annotation of the loquat (Eriobotrya japonica) genome</a></p>
<p><strong>期刊</strong>：Gigascience</p>
<p>**影响因子：**4.688</p>
<p>**研究单位：**上海市农业科学院</p>
<p><strong>主要研究内容：</strong></p>
<p>2020年2月6日，上海市农业科学院张学英研究员团队于知名期刊Gigascience上首次发表枇杷基因组相关研究论文，该研究中基于北京百迈客生物科技有限公司的Nanopore测序和Hi-C染色体构象捕获技术，构建了高质量的枇杷基因组。文中通过与苹果、水蜜桃、梨、覆盆莓、月季和野草莓的蛋白质序列比较，探索了枇杷的全基因组复制与进化事件，并进行了染色体重排分析。该研究提供了宝贵的染色体水平基因组数据，为研究枇杷性状提供了重要的基因组数据。</p>
<p><img src="/blog/blog/2022/02/16/2022-0216-chromosome-structure-Hi-C/30.png" alt="枇杷基因组circos图"></p>
<h2 id="77"><a class="markdownIt-Anchor" href="#77"></a> 7.7</h2>
<p><strong>物种：板蓝根（菘蓝）</strong></p>
<p><strong>题目：</strong><a href="https://link.zhihu.com/?target=http%3A//tour.biocloud.net/article/v1/into/articleDetail/32025321">A chromosome-scale genome assembly of Isatis indigotica, an important medicinal plant used in traditional Chinese medicine</a></p>
<p><strong>期刊：</strong> Horticulture Research</p>
<p>**影响因子：**3.640</p>
<p>**研究单位：**四川大学</p>
<p><strong>主要研究内：</strong></p>
<p>2020年2 月1日，四川大学生命科学学院刘建全课题组与华中农业大学作物遗传改良国家重点实验室李再云课题组首次完成了菘蓝（<em>Isatis indigotica</em>，2n=14）染色体级别基因组图谱绘制。该研究利用Pacbio测序(140X)结合北京百迈客生物技术有限公司的Hi-C技术（284X）组装得到294Mb高质量基因组（contigN50=1.2Mb）。基于同源搜索和功能注释，确定了该物种主要化合物（如吲哚类、萜类、黄酮、木脂素等物质）的可能生物合成通路和相关的候选基因。该研究强调了基因组测序在鉴定药用植物代谢产物合成候选基因中的重要性，为今后十字花科植物比较基因组等研究提供了重要遗传信息。</p>
<p><img src="/blog/blog/2022/02/16/2022-0216-chromosome-structure-Hi-C/31.png" alt="菘蓝基因组Hi-C热图"></p>
<h2 id="78"><a class="markdownIt-Anchor" href="#78"></a> 7.8</h2>
<h6 id="以更新的亚洲棉a基因组为基础的243份二倍体棉的重要农艺性状的研究"><a class="markdownIt-Anchor" href="#以更新的亚洲棉a基因组为基础的243份二倍体棉的重要农艺性状的研究"></a> 以更新的亚洲棉A基因组为基础的243份二倍体棉的重要农艺性状的研究</h6>
<h6 id="resequencing-of-243-diploid-cotton-accessions-based-on-an-updated-a-genome-identifies-the-genetic-basis-of-key-agronomic-traits"><a class="markdownIt-Anchor" href="#resequencing-of-243-diploid-cotton-accessions-based-on-an-updated-a-genome-identifies-the-genetic-basis-of-key-agronomic-traits"></a> RESEQUENCING OF 243 DIPLOID COTTON ACCESSIONS BASED ON AN UPDATED A GENOME IDENTIFIES THE GENETIC BASIS OF KEY AGRONOMIC TRAITS</h6>
<p>期刊：Nature Genetics</p>
<p>影响因子：27.125</p>
<p>发表单位：中国农业科学院棉花研究所、北京百迈客生物科技有限公司等</p>
<p>发表年份：2018年5月</p>
<p>研究背景：</p>
<p>棉花是研究植物多倍化的有价值的资源。亚洲棉(<em>Gossypium arboreum</em>)和草棉(<em>Gossypium herbaceum</em>)的祖先是现代栽培异源四倍体棉花A亚基因组的供体。 本研究中，利用了三代PacBio和Hi-C技术，重新组装了高质量的亚洲棉基因组，分析了243份二倍体棉花种质的群体结构和基因组分化趋势，同时确定了一些有助于棉花皮棉产量遗传改良的候选基因位点。</p>
<p>研究结果：</p>
<p>1、亚洲棉三代基因组组装：</p>
<p>利用三代测序和Hi-C相结合的方法进行亚洲棉基因组组装。共计获得了142.54 Gb ，组装1.71 Gb亚洲棉基因组，Contig N50=1.1 Mb，最长的Contig为12.37 Mb。利用Hi-C技术将组装的1573 Mb的数据定位到13条染色体上，与已经发表的基因组相比，当Hi-C数据比对到更新的基因组后，对角线外的不一致性明显减少（图1 a-b）</p>
<p><img src="/blog/blog/2022/02/16/2022-0216-chromosome-structure-Hi-C/33.png" alt="img"></p>
<p>图1 HI-C数据在两版亚洲棉基因组上的比对</p>
<p>2、二倍体棉花群体遗传进化分析：</p>
<p>对230份亚洲棉和13份草棉重测序，进行基因组比对、系统发育树、群体结构分析、PCA、LD和选择性清除分析得出亚洲棉和草棉（A）与雷蒙德氏棉同时进行了分化；亚洲棉起源于中国南部，随后被引入长江和黄河地区，大多数具有驯化相关特性的种质都经历了地理隔离（图2）。</p>
<p><img src="/blog/blog/2022/02/16/2022-0216-chromosome-structure-Hi-C/34.png" alt="img"></p>
<p>图2 二倍体棉群体进化和群体结构分析</p>
<p>3、亚洲棉的全基因组关联分析（GWAS）：</p>
<p>对来自不同环境下的11个重要性状进行全基因组关联分析，鉴定了亚洲棉11个重要农艺性状的98个显著关联位点，GaKASIII的非同义替换（半胱氨酸/精氨酸替换）使得棉籽中的脂肪酸组成（C16:0和C16:1）发生了变化；发现棉花枯萎病抗性与GaGSTF9基因的表达激活相关。选择了亚洲棉种质中的158份有绒毛和57份无绒毛材料进行GWAS关联分析，发现与毛状体和纤维发育有关信息（图3）。</p>
<p><img src="/blog/blog/2022/02/16/2022-0216-chromosome-structure-Hi-C/35.png" alt="img"></p>
<p>图3 二倍体棉群体进化和群体结构分析</p>
<p>研究结论：</p>
<p>利用三代测序+Hi-C技术完成了亚洲棉基因组的重新组装，将基因组组装指标从72 Kb提升到1.1 Mb，为亚洲棉后续的群体遗传学等相关研究奠定了基础；通过群体遗传进化等相关分析，发现亚洲棉和草棉(A型)与雷蒙德氏棉(D型)同时进行了分化，并证明了亚洲棉起源于中国南部，随后被引入长江和黄河地区；整合GWAS与QTL等分析方法，对亚洲棉脂肪酸含量，抗病性及棉绒生长发育相关基因进行定位，并进行相关功能验证，促进了亚洲棉复杂农艺性状的改良。</p>
<h2 id="79"><a class="markdownIt-Anchor" href="#79"></a> 7.9</h2>
<h6 id="二倍体-野生和栽培四倍体花生比较基因组分析揭示亚基因组不对称进化和改良"><a class="markdownIt-Anchor" href="#二倍体-野生和栽培四倍体花生比较基因组分析揭示亚基因组不对称进化和改良"></a> 二倍体、野生和栽培四倍体花生比较基因组分析揭示亚基因组不对称进化和改良</h6>
<h6 id="comparison-of-arachis-monticola-with-diploid-and-cultivated-tetraploid-genomes-reveals-asymmetric-subgenome-evolution-and-improvement-of-peanut"><a class="markdownIt-Anchor" href="#comparison-of-arachis-monticola-with-diploid-and-cultivated-tetraploid-genomes-reveals-asymmetric-subgenome-evolution-and-improvement-of-peanut"></a> COMPARISON OF <em>ARACHIS MONTICOLA</em> WITH DIPLOID AND CULTIVATED TETRAPLOID GENOMES REVEALS ASYMMETRIC SUBGENOME EVOLUTION AND IMPROVEMENT OF PEANUT</h6>
<p>期刊：Advanced Science</p>
<p>影响因子：15.804</p>
<p>发表单位：河南农业大学、北京百迈客生物科技有限公司等</p>
<p>发表年份：2019年11月</p>
<p>研究背景：</p>
<p>花生作为我国重要的经济作物，是提供重要的蛋白和油料的基础。花生属一共包括30个二倍体品种，1个异源四倍体野生花生(<em>A. monticola</em>)和1个栽培花生(<em>A. hypogaea</em>)。作为栽培花生农艺性状改良的重要野生资源供体，野生四倍体花生一直是国内外学者的研究热点。研究中对花生属唯一的野生异源四倍体花生<em>Arachis monticola</em>基因组进行了研究，同时对17个野生二倍体花生（AA;BB;EE;KK和CC）与30个野生和栽培四倍体花生进行了重测序分析。</p>
<p>研究结果：</p>
<p>1、野生四倍体花生基因组denovo及与栽培四倍体花生的比较分析：</p>
<p>基于 Illumina、PacBio 、Hi-C和光学图谱数据，组装<em>Arachis monticola</em>(2n = 4x = 40)基因组大小为2.62 Gb ，contigs N50=106.66 Kb，scaffolds N50=124.92 Mb；与栽培四倍体花生<em>A. hypogaea</em>基因组结构变异高度保守，且比野生祖先二倍体更加保守；</p>
<p>2、A、B亚基因组的单系起源和多样性：</p>
<p>对17个二倍体野生种（AA、BB、EE、KK和CC）和30个野生和栽培四倍体花生进行了进化树和PCA分析。结果表明，栽培四倍体花生与野生四倍体花生最接近， A和B亚基因组的单系起源（图1）；</p>
<p><img src="/blog/blog/2022/02/16/2022-0216-chromosome-structure-Hi-C/36.png" alt="img"></p>
<p>3、四倍体花生不对称亚基因组进化及表达差异 <strong>：</strong></p>
<p>栽培花生和野生花生的亚基因组间的同源序列交换率（HSE）分别为2.46%和2.54%。野生花生中A到B的HSE富集的基因为类黄酮生物合成和昼夜节律途径的基因，暗示不对称HSEs在生物学功能中的作用；</p>
<p>4、SV对荚发育和驯化相关基因表达的影响及抗病基因鉴定 <strong>：</strong></p>
<p>对野生四倍体花生和栽培四倍体花生不同发育阶段荚果的SV分析发现SV在荚果发育过程中基因表达的变化上可能起着重要作用；同时在栽培四倍体花生中鉴定到190个SV抗病基因（SV-RGAs），其中32个基因在接种后易感组或抗性组中表现出显著的表达变化（图2）。</p>
<p><img src="/blog/blog/2022/02/16/2022-0216-chromosome-structure-Hi-C/37.png" alt="img"></p>
<p>研究结论：</p>
<p>充分注释了高质量野生四倍体花生基因组，揭示了花生亚基因组单系起源和遗传进化模型，表明了野生和栽培四倍体花生亚基因组发生了不对称进化；此外，野生花生中存在的独特等位基因可以改善栽培花生的抗性和荚果大小等形状，为研究多倍体基因组进化、作物驯化和基因组辅助花生生产改良提供独特的价值。</p>
<hr>
<p><img src="/blog/blog/2022/02/16/2022-0216-chromosome-structure-Hi-C/32.png" alt="其他"></p>
<p>北京百迈客自启动Hi-C技术研究以来，自主开发Hi-C排图软件，具有实验+生物信息分析专利双保护，也是中国市场上能独立完成Hi-C实验——数据分析的测序企业之一！并完成上千个物种，近万个文库构建；文库含酶切位点有效数据比例最高达93%以上，针对物种推出定制化内切酶服务；实现多倍体物种的Hi-C辅助基因组组装，<strong>挂载效率最高达100%。</strong></p>
<p>另外为了让更多科研工作者体验Hi-C建库，百迈客在成功构建多种物种Hi-C建库测序的经验上，正式推出了Biomarker Hi-C Library Prep Kit for Illumina。</p>
<p>Biomarker Hi-C Library Prep Kit for Illumina涵盖了Hi-C实验全部环节，包括甲醛交联、Hi-C提取和Hi-C文库构建，适用于细胞系、动物和植物组织样本。1套Hi-C试剂盒包含5个反应，1个反应构建的文库可用于200Gb数据量的测序，试剂盒构建的文库适用于Illumina测序平台。</p>
<hr>
<p>Ref:<a target="_blank" rel="noopener" href="https://www.zhihu.com/question/48308074">https://www.zhihu.com/question/48308074</a></p>
<p><a target="_blank" rel="noopener" href="https://cloud.tencent.com/developer/article/1625366">https://cloud.tencent.com/developer/article/1625366</a></p>
<p><a target="_blank" rel="noopener" href="http://www.biomarker.com.cn/technology-services/hi-c">http://www.biomarker.com.cn/technology-services/hi-c</a></p>
<p><a target="_blank" rel="noopener" href="https://zhuanlan.zhihu.com/p/140795830">https://zhuanlan.zhihu.com/p/140795830</a></p>

    </div>

    
    
    
        

<div>
<ul class="post-copyright">
  <li class="post-copyright-author">
    <strong>Post author:  </strong>Wang C
  </li>
  <li class="post-copyright-link">
    <strong>Post link: </strong>
    <a href="https://wangc1106.github.io/blog/2022/02/16/2022-0216-chromosome-structure-Hi-C/" title="染色质三维结构图谱 Hi-C">https://wangc1106.github.io/blog/2022/02/16/2022-0216-chromosome-structure-Hi-C/</a>
  </li>
  <li class="post-copyright-license">
    <strong>Copyright Notice:  </strong>All articles in this blog are licensed under <a href="https://creativecommons.org/licenses/by-nc-sa/4.0/" rel="noopener" target="_blank"><i class="fab fa-fw fa-creative-commons"></i>BY-NC-SA</a> unless stating additionally.
  </li>
</ul>
</div>


      <footer class="post-footer">
          <div class="post-tags">
              <a href="/blog/tags/technique/" rel="tag"># technique</a>
          </div>

        


        
    <div class="post-nav">
      <div class="post-nav-item">
    <a href="/blog/2022/02/16/2022-0216-select-gene/" rel="prev" title="选择基因">
      <i class="fa fa-chevron-left"></i> 选择基因
    </a></div>
      <div class="post-nav-item">
    <a href="/blog/2022/02/18/20220218-ggplot2/" rel="next" title="">
       <i class="fa fa-chevron-right"></i>
    </a></div>
    </div>
      </footer>
    
  </article>
  
  
  



          </div>
          

<script>
  window.addEventListener('tabs:register', () => {
    let { activeClass } = CONFIG.comments;
    if (CONFIG.comments.storage) {
      activeClass = localStorage.getItem('comments_active') || activeClass;
    }
    if (activeClass) {
      let activeTab = document.querySelector(`a[href="#comment-${activeClass}"]`);
      if (activeTab) {
        activeTab.click();
      }
    }
  });
  if (CONFIG.comments.storage) {
    window.addEventListener('tabs:click', event => {
      if (!event.target.matches('.tabs-comment .tab-content .tab-pane')) return;
      let commentClass = event.target.classList[1];
      localStorage.setItem('comments_active', commentClass);
    });
  }
</script>

        </div>
          
  
  <div class="toggle sidebar-toggle">
    <span class="toggle-line toggle-line-first"></span>
    <span class="toggle-line toggle-line-middle"></span>
    <span class="toggle-line toggle-line-last"></span>
  </div>

  <aside class="sidebar">
    <div class="sidebar-inner">

      <ul class="sidebar-nav motion-element">
        <li class="sidebar-nav-toc">
          Table of Contents
        </li>
        <li class="sidebar-nav-overview">
          Overview
        </li>
      </ul>

      <!--noindex-->
      <div class="post-toc-wrap sidebar-panel">
          <div class="post-toc motion-element"><ol class="nav"><li class="nav-item nav-level-1"><a class="nav-link" href="#hi-c%E6%8A%80%E6%9C%AF%E4%BB%A5%E9%AB%98%E9%80%9A%E9%87%8F%E6%B5%8B%E5%BA%8F%E4%B8%BA%E6%89%8B%E6%AE%B5%E4%BB%A53c%E6%8A%80%E6%9C%AF%E4%B8%BA%E5%9F%BA%E7%A1%80%E7%9A%84%E6%9F%93%E8%89%B2%E8%B4%A8%E6%9E%84%E8%B1%A1%E6%8D%95%E6%8D%89%E6%8A%80%E6%9C%AF-%E4%B8%89%E7%BB%B4%E5%9F%BA%E5%9B%A0%E7%BB%84"><span class="nav-number">1.</span> <span class="nav-text"> Hi-C技术，以高通量测序为手段，以3C技术为基础的染色质构象捕捉技术。-三维基因组。</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#1-c%E6%8A%80%E6%9C%AF"><span class="nav-number">2.</span> <span class="nav-text"> 1. C技术</span></a><ol class="nav-child"><li class="nav-item nav-level-2"><a class="nav-link" href="#11-3c%E4%B8%80%E5%AF%B9%E4%B8%80"><span class="nav-number">2.1.</span> <span class="nav-text"> 1.1 3C（一对一）</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#12-4c%E4%B8%80%E5%AF%B9%E5%A4%9A"><span class="nav-number">2.2.</span> <span class="nav-text"> 1.2 4C（一对多）</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#13-5c%E5%A4%9A%E5%AF%B9%E5%A4%9A"><span class="nav-number">2.3.</span> <span class="nav-text"> 1.3 5C（多对多）</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#14-font-colorredhi-c%E5%85%A8%E9%83%A8%E4%BA%92%E4%BD%9Cfont"><span class="nav-number">2.4.</span> <span class="nav-text"> 1.4 Hi-C（全部互作）</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#15-chip-loop-%E5%9F%BA%E4%BA%8E%E5%85%8D%E7%96%AB%E6%B2%89%E6%B7%80"><span class="nav-number">2.5.</span> <span class="nav-text"> 1.5 ChIP-loop (基于免疫沉淀)</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#16-chia-pet-%E5%9F%BA%E4%BA%8E%E5%85%8D%E7%96%AB%E6%B2%89%E6%B7%80"><span class="nav-number">2.6.</span> <span class="nav-text"> 1.6 ChIA-PET (基于免疫沉淀)</span></a></li></ol></li><li class="nav-item nav-level-1"><a class="nav-link" href="#2-%E6%8A%80%E6%9C%AF%E5%8E%9F%E7%90%86"><span class="nav-number">3.</span> <span class="nav-text"> 2. 技术原理</span></a><ol class="nav-child"><li class="nav-item nav-level-2"><a class="nav-link" href="#21-%E7%94%B2%E9%86%9B%E5%9B%BA%E5%AE%9A"><span class="nav-number">3.1.</span> <span class="nav-text"> 2.1 甲醛固定</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#22-%E9%85%B6%E5%88%87%E5%BA%8F%E5%88%97"><span class="nav-number">3.2.</span> <span class="nav-text"> 2.2 酶切序列</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#23-%E6%9C%AB%E7%AB%AF%E4%BF%AE%E5%A4%8D"><span class="nav-number">3.3.</span> <span class="nav-text"> 2.3 末端修复</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#24-%E8%BF%9E%E6%8E%A5%E5%8F%8A%E8%A7%A3%E4%BA%A4%E8%81%94"><span class="nav-number">3.4.</span> <span class="nav-text"> 2.4 连接及解交联</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#25-%E5%BA%8F%E5%88%97%E6%89%93%E6%96%AD"><span class="nav-number">3.5.</span> <span class="nav-text"> 2.5 序列打断</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#26-%E4%B8%8A%E6%9C%BA%E6%B5%8B%E5%BA%8F"><span class="nav-number">3.6.</span> <span class="nav-text"> 2.6 上机测序</span></a></li></ol></li><li class="nav-item nav-level-1"><a class="nav-link" href="#3-hi-c%E5%BA%94%E7%94%A8"><span class="nav-number">4.</span> <span class="nav-text"> 3. Hi-C应用</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#4-%E5%88%86%E6%9E%90%E6%B5%81%E7%A8%8B"><span class="nav-number">5.</span> <span class="nav-text"> 4. 分析流程</span></a><ol class="nav-child"><li class="nav-item nav-level-2"><a class="nav-link" href="#41-%E5%9F%BA%E6%9C%AC%E5%88%86%E6%9E%90"><span class="nav-number">5.1.</span> <span class="nav-text"> 4.1 基本分析</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#42-hi-c%E5%8F%AF%E8%A7%86%E5%8C%96"><span class="nav-number">5.2.</span> <span class="nav-text"> 4.2 Hi-C可视化</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#43-%E5%BA%8F%E5%88%97%E8%BF%87%E6%BB%A4"><span class="nav-number">5.3.</span> <span class="nav-text"> 4.3 序列过滤</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#44-%E6%95%B0%E6%8D%AE%E7%9F%AB%E6%AD%A3"><span class="nav-number">5.4.</span> <span class="nav-text"> 4.4 数据矫正</span></a></li></ol></li><li class="nav-item nav-level-1"><a class="nav-link" href="#5-%E5%88%86%E6%9E%90"><span class="nav-number">6.</span> <span class="nav-text"> 5. 分析</span></a><ol class="nav-child"><li class="nav-item nav-level-2"><a class="nav-link" href="#51-cistrans%E4%BA%92%E4%BD%9C%E6%AF%94%E4%BE%8B"><span class="nav-number">6.1.</span> <span class="nav-text"> 5.1 cis&#x2F;trans互作比例</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#52-%E4%BA%92%E4%BD%9C%E9%A2%91%E7%8E%87%E4%B8%8E%E8%B7%9D%E7%A6%BB%E6%9C%89%E5%85%B3"><span class="nav-number">6.2.</span> <span class="nav-text"> 5.2 互作频率与距离有关</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#53-compartment%E5%88%86%E6%9E%90"><span class="nav-number">6.3.</span> <span class="nav-text"> 5.3 compartment分析</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#54-tad%E5%88%86%E6%9E%90"><span class="nav-number">6.4.</span> <span class="nav-text"> 5.4 TAD分析</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#55-%E6%98%BE%E8%91%97%E4%BA%92%E4%BD%9C%E5%88%86%E6%9E%90"><span class="nav-number">6.5.</span> <span class="nav-text"> 5.5 显著互作分析</span></a></li></ol></li><li class="nav-item nav-level-1"><a class="nav-link" href="#6-%E5%BA%94%E7%94%A8"><span class="nav-number">7.</span> <span class="nav-text"> 6. 应用</span></a><ol class="nav-child"><li class="nav-item nav-level-2"><a class="nav-link" href="#61-%E8%A7%A3%E6%9E%90%E5%85%A8%E5%9F%BA%E5%9B%A0%E7%BB%84%E4%BA%92%E4%BD%9C%E6%A8%A1%E5%BC%8F"><span class="nav-number">7.1.</span> <span class="nav-text"> 6.1 解析全基因组互作模式</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#62-%E8%BE%85%E5%8A%A9%E6%8F%90%E5%8D%87%E5%9F%BA%E5%9B%A0%E7%BB%84%E7%BB%84%E8%A3%85"><span class="nav-number">7.2.</span> <span class="nav-text"> 6.2 辅助提升基因组组装</span></a><ol class="nav-child"><li class="nav-item nav-level-3"><a class="nav-link" href="#621-%E6%95%B0%E6%8D%AE%E7%BB%9F%E8%AE%A1%E5%92%8C%E8%BF%87%E6%BB%A4"><span class="nav-number">7.2.1.</span> <span class="nav-text"> 6.2.1 数据统计和过滤；</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#622-hi-c%E6%96%87%E5%BA%93%E8%AF%84%E4%BC%B0"><span class="nav-number">7.2.2.</span> <span class="nav-text"> 6.2.2 Hi-C文库评估；</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#623hi-c%E6%9F%93%E8%89%B2%E4%BD%93%E5%AE%9A%E4%BD%8D"><span class="nav-number">7.2.3.</span> <span class="nav-text"> 6.2.3Hi-C染色体定位；</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#624-hi-c%E5%AE%9A%E4%BD%8D%E5%90%8E%E7%BB%9F%E8%AE%A1%E8%AF%84%E4%BC%B0"><span class="nav-number">7.2.4.</span> <span class="nav-text"> 6.2.4 Hi-C定位后统计评估；</span></a></li></ol></li><li class="nav-item nav-level-2"><a class="nav-link" href="#63-%E6%9E%84%E5%BB%BA%E5%9F%BA%E5%9B%A0%E7%BB%84%E5%8D%95%E4%BD%93%E5%9E%8B%E5%9B%BE%E8%B0%B1"><span class="nav-number">7.3.</span> <span class="nav-text"> 6.3 构建基因组单体型图谱</span></a></li></ol></li><li class="nav-item nav-level-1"><a class="nav-link" href="#7-%E6%96%87%E7%8C%AE%E6%A1%88%E4%BE%8B"><span class="nav-number">8.</span> <span class="nav-text"> 7. 文献案例</span></a><ol class="nav-child"><li class="nav-item nav-level-2"><a class="nav-link" href="#71"><span class="nav-number">8.1.</span> <span class="nav-text"> 7.1</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#72"><span class="nav-number">8.2.</span> <span class="nav-text"> 7.2</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#73"><span class="nav-number">8.3.</span> <span class="nav-text"> 7.3</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#74"><span class="nav-number">8.4.</span> <span class="nav-text"> 7.4</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#75"><span class="nav-number">8.5.</span> <span class="nav-text"> 7.5</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#76"><span class="nav-number">8.6.</span> <span class="nav-text"> 7.6</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#77"><span class="nav-number">8.7.</span> <span class="nav-text"> 7.7</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#78"><span class="nav-number">8.8.</span> <span class="nav-text"> 7.8</span></a><ol class="nav-child"><li class="nav-item nav-level-6"><a class="nav-link" href="#%E4%BB%A5%E6%9B%B4%E6%96%B0%E7%9A%84%E4%BA%9A%E6%B4%B2%E6%A3%89a%E5%9F%BA%E5%9B%A0%E7%BB%84%E4%B8%BA%E5%9F%BA%E7%A1%80%E7%9A%84243%E4%BB%BD%E4%BA%8C%E5%80%8D%E4%BD%93%E6%A3%89%E7%9A%84%E9%87%8D%E8%A6%81%E5%86%9C%E8%89%BA%E6%80%A7%E7%8A%B6%E7%9A%84%E7%A0%94%E7%A9%B6"><span class="nav-number">8.8.0.0.0.1.</span> <span class="nav-text"> 以更新的亚洲棉A基因组为基础的243份二倍体棉的重要农艺性状的研究</span></a></li><li class="nav-item nav-level-6"><a class="nav-link" href="#resequencing-of-243-diploid-cotton-accessions-based-on-an-updated-a-genome-identifies-the-genetic-basis-of-key-agronomic-traits"><span class="nav-number">8.8.0.0.0.2.</span> <span class="nav-text"> RESEQUENCING OF 243 DIPLOID COTTON ACCESSIONS BASED ON AN UPDATED A GENOME IDENTIFIES THE GENETIC BASIS OF KEY AGRONOMIC TRAITS</span></a></li></ol></li></ol></li></ol></li></ol></li><li class="nav-item nav-level-2"><a class="nav-link" href="#79"><span class="nav-number">8.9.</span> <span class="nav-text"> 7.9</span></a><ol class="nav-child"><li class="nav-item nav-level-6"><a class="nav-link" href="#%E4%BA%8C%E5%80%8D%E4%BD%93-%E9%87%8E%E7%94%9F%E5%92%8C%E6%A0%BD%E5%9F%B9%E5%9B%9B%E5%80%8D%E4%BD%93%E8%8A%B1%E7%94%9F%E6%AF%94%E8%BE%83%E5%9F%BA%E5%9B%A0%E7%BB%84%E5%88%86%E6%9E%90%E6%8F%AD%E7%A4%BA%E4%BA%9A%E5%9F%BA%E5%9B%A0%E7%BB%84%E4%B8%8D%E5%AF%B9%E7%A7%B0%E8%BF%9B%E5%8C%96%E5%92%8C%E6%94%B9%E8%89%AF"><span class="nav-number">8.9.0.0.0.1.</span> <span class="nav-text"> 二倍体、野生和栽培四倍体花生比较基因组分析揭示亚基因组不对称进化和改良</span></a></li><li class="nav-item nav-level-6"><a class="nav-link" href="#comparison-of-arachis-monticola-with-diploid-and-cultivated-tetraploid-genomes-reveals-asymmetric-subgenome-evolution-and-improvement-of-peanut"><span class="nav-number">8.9.0.0.0.2.</span> <span class="nav-text"> COMPARISON OF ARACHIS MONTICOLA WITH DIPLOID AND CULTIVATED TETRAPLOID GENOMES REVEALS ASYMMETRIC SUBGENOME EVOLUTION AND IMPROVEMENT OF PEANUT</span></a></li></ol></li></ol></li></ol></li></ol></li></ol></li></ol></div>
      </div>
      <!--/noindex-->

      <div class="site-overview-wrap sidebar-panel">
        <div class="site-author motion-element" itemprop="author" itemscope itemtype="http://schema.org/Person">
    <img class="site-author-image" itemprop="image" alt="Wang C"
      src="/blog/images/banner.png">
  <p class="site-author-name" itemprop="name">Wang C</p>
  <div class="site-description" itemprop="description">Make more time</div>
</div>
<div class="site-state-wrap motion-element">
  <nav class="site-state">
      <div class="site-state-item site-state-posts">
          <a href="/blog/archives">
          <span class="site-state-item-count">36</span>
          <span class="site-state-item-name">posts</span>
        </a>
      </div>
      <div class="site-state-item site-state-categories">
            <a href="/blog/categories/">
          
        <span class="site-state-item-count">12</span>
        <span class="site-state-item-name">categories</span></a>
      </div>
      <div class="site-state-item site-state-tags">
            <a href="/blog/tags/">
          
        <span class="site-state-item-count">19</span>
        <span class="site-state-item-name">tags</span></a>
      </div>
  </nav>
</div>
  <div class="links-of-author motion-element">
      <span class="links-of-author-item">
        <a href="https://github.com/wangc1106" title="GitHub → https:&#x2F;&#x2F;github.com&#x2F;wangc1106" rel="noopener" target="_blank"><i class="fab fa-github fa-fw"></i>GitHub</a>
      </span>
      <span class="links-of-author-item">
        <a href="mailto:wangchen116@126.com" title="E-Mail → mailto:wangchen116@126.com" rel="noopener" target="_blank"><i class="fa fa-envelope fa-fw"></i>E-Mail</a>
      </span>
  </div>
  <div class="cc-license motion-element" itemprop="license">
    <a href="https://creativecommons.org/licenses/by-nc-sa/4.0/" class="cc-opacity" rel="noopener" target="_blank"><img src="/blog/images/cc-by-nc-sa.svg" alt="Creative Commons"></a>
  </div>



      </div>
        <div class="back-to-top motion-element">
          <i class="fa fa-arrow-up"></i>
          <span>0%</span>
        </div>

    </div>
  </aside>
  <div id="sidebar-dimmer"></div>


      </div>
    </main>

    <footer class="footer">
      <div class="footer-inner">
        

        

<div class="copyright">
  
  &copy; 2021 – 
  <span itemprop="copyrightYear">2022</span>
  <span class="with-love">
    <i class="fas fa-dna"></i>
  </span>
  <span class="author" itemprop="copyrightHolder">Wang C</span>
</div>

        






<script>
  (function() {
    function leancloudSelector(url) {
      url = encodeURI(url);
      return document.getElementById(url).querySelector('.leancloud-visitors-count');
    }

    function addCount(Counter) {
      var visitors = document.querySelector('.leancloud_visitors');
      var url = decodeURI(visitors.id);
      var title = visitors.dataset.flagTitle;

      Counter('get', '/classes/Counter?where=' + encodeURIComponent(JSON.stringify({ url })))
        .then(response => response.json())
        .then(({ results }) => {
          if (results.length > 0) {
            var counter = results[0];
            leancloudSelector(url).innerText = counter.time + 1;
            Counter('put', '/classes/Counter/' + counter.objectId, { time: { '__op': 'Increment', 'amount': 1 } })
              .catch(error => {
                console.error('Failed to save visitor count', error);
              });
          } else {
              Counter('post', '/classes/Counter', { title, url, time: 1 })
                .then(response => response.json())
                .then(() => {
                  leancloudSelector(url).innerText = 1;
                })
                .catch(error => {
                  console.error('Failed to create', error);
                });
          }
        })
        .catch(error => {
          console.error('LeanCloud Counter Error', error);
        });
    }

    function showTime(Counter) {
      var visitors = document.querySelectorAll('.leancloud_visitors');
      var entries = [...visitors].map(element => {
        return decodeURI(element.id);
      });

      Counter('get', '/classes/Counter?where=' + encodeURIComponent(JSON.stringify({ url: { '$in': entries } })))
        .then(response => response.json())
        .then(({ results }) => {
          for (let url of entries) {
            let target = results.find(item => item.url === url);
            leancloudSelector(url).innerText = target ? target.time : 0;
          }
        })
        .catch(error => {
          console.error('LeanCloud Counter Error', error);
        });
    }

    let { app_id, app_key, server_url } = {"enable":true,"app_id":"UqIYufmNk2sE8MiBKKWc4Kja-gzGzoHsz","app_key":"iYpeWQcmNqpHwr2uVlnJYVji","server_url":null,"security":false};
    function fetchData(api_server) {
      var Counter = (method, url, data) => {
        return fetch(`${api_server}/1.1${url}`, {
          method,
          headers: {
            'X-LC-Id'     : app_id,
            'X-LC-Key'    : app_key,
            'Content-Type': 'application/json',
          },
          body: JSON.stringify(data)
        });
      };
      if (CONFIG.page.isPost) {
        if (CONFIG.hostname !== location.hostname) return;
        addCount(Counter);
      } else if (document.querySelectorAll('.post-title-link').length >= 1) {
        showTime(Counter);
      }
    }

    let api_server = app_id.slice(-9) !== '-MdYXbMMI' ? server_url : `https://${app_id.slice(0, 8).toLowerCase()}.api.lncldglobal.com`;

    if (api_server) {
      fetchData(api_server);
    } else {
      fetch('https://app-router.leancloud.cn/2/route?appId=' + app_id)
        .then(response => response.json())
        .then(({ api_server }) => {
          fetchData('https://' + api_server);
        });
    }
  })();
</script>


      </div>
      <div style="height: 200px">
      <script type="text/javascript" src="//rf.revolvermaps.com/0/0/8.js?i=5knf34gs07t&amp;m=0&amp;c=ff0000&amp;cr1=ffffff&amp;f=arial&amp;l=33" async="async"></script>
      </div>
    </footer>
  </div>

  
  <script src="/blog/lib/anime.min.js"></script>
  <script src="/blog/lib/velocity/velocity.min.js"></script>
  <script src="/blog/lib/velocity/velocity.ui.min.js"></script>

<script src="/blog/js/utils.js"></script>

<script src="/blog/js/motion.js"></script>


<script src="/blog/js/schemes/pisces.js"></script>


<script src="/blog/js/next-boot.js"></script>




  




  
<script src="/blog/js/local-search.js"></script>













  

  

</body>
</html>
